{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(\"plotgardener\")\n",
    "library(\"data.table\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"TxDb.Hsapiens.UCSC.hg38.knownGene\")\n",
    "library(png)\n",
    "options(bedtools.path = \"/rds/general/user/hrayjone/home/anaconda3/bin/\")\n",
    "library(bedtoolsr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "loops <- fread(\"~/HRJ_monocytes/CHiC/chicago/combinations_output/DpnII/combined_34reps_DpnII_binsize1500_maxL150K_mergeWeights/data/combined_34reps_DpnII_binsize1500_maxL150K_mergeWeights.bedpe\")\n",
    "# intersect loops with proxies so we can restrict loops to just incldue them\n",
    "\n",
    "proxies1 <- fread(\"~/HRJ_monocytes/findmotifs/find_new_dpnII/snps_went_into_loopingQTL_analysis.DpnII.marked.REFALTseqs.PURGED.DISTAL.txt\")\n",
    "proxies1[, chrom := paste0(\"chr\", Chr)]\n",
    "proxies1[, hg38Proxy_pos0 := hg38Proxy_pos-1]\n",
    "proxies_small <- proxies1[, .(chrom, hg38Proxy_pos0, hg38Proxy_pos, hg38Proxy_ID, hg38SNP_ID)]\n",
    "setkey(loops, V1, V2, V3)\n",
    "loops1 <- foverlaps(proxies_small, loops, by.x = c(\"chrom\", \"hg38Proxy_pos0\", \"hg38Proxy_pos\"), nomatch = NULL)\n",
    "setnames(loops1, c(\"chrom\", \"V2\", \"V3\", \"V4\", \"V5\", \"V6\"), c(\"chrom1\", \"start1\", \"end1\", \"chrom2\", \"start2\", \"end2\"))\n",
    "setkey(loops, V4, V5, V6)\n",
    "loops2 <- foverlaps(proxies_small, loops, by.x = c(\"chrom\", \"hg38Proxy_pos0\", \"hg38Proxy_pos\"), nomatch = NULL)\n",
    "setnames(loops2, c(\"chrom\", \"V1\", \"V2\", \"V3\", \"V5\", \"V6\"), c(\"chrom2\", \"chrom1\", \"start1\", \"end1\", \"start2\", \"end2\"))\n",
    "loops2_edit <- loops2[, .(chrom1, start1, end1, chrom2, start2, end2, hg38Proxy_pos0, hg38Proxy_pos, hg38Proxy_ID, hg38SNP_ID)]\n",
    "loops_proxies <- rbind(loops1, loops2_edit)\n",
    "\n",
    "## OR: we can instead look at eQTLs and only their interactions with eGenes\n",
    "eqtl_inters <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/consensus_DpnII_level_proxy_eGene_interactions_DpnIIFilt_withGeneTypes.txt\")\n",
    "sig_eqtl_inters <- eqtl_inters[score >= 5]\n",
    "sig_eqtl_inters <- sig_eqtl_inters[, c(\"Chr\", \"Proxy_dpnStart\", \"Proxy_dpnEnd\", \"DpnID\", \"Gene\", \"hg19SNP_ID\", \"hg19Proxy_ID\", \"hg38SNP_ID\")]\n",
    "# have to get other end coordinates\n",
    "dpnII <- fread(\"/rds/general/project/lms-spivakov-analysis/live/Design/Human_eQTL_CHiC_DpnII_hg38/hg38_dpnII.rmap\")\n",
    "setkey(dpnII, V4)\n",
    "sig_eqtl_inters_dpn <- sig_eqtl_inters[dpnII, on = c(DpnID = \"V4\"), nomatch = NULL]\n",
    "setnames(sig_eqtl_inters_dpn, c(\"V2\", \"V3\"), c(\"TSS_dpnStart\", \"TSS_dpnEnd\"))\n",
    "sig_eqtl_inters_dpn[, chrom1 := paste0(\"chr\", Chr)]\n",
    "sig_eqtl_inters_dpn[, chrom2 := paste0(\"chr\", V1)]\n",
    "eQTL_eGene_score5 <- sig_eqtl_inters_dpn[, .(chrom1, Proxy_dpnStart, Proxy_dpnEnd, chrom2, TSS_dpnStart, TSS_dpnEnd, Gene, hg19SNP_ID, hg19Proxy_ID, hg38SNP_ID)]\n",
    "\n",
    "\n",
    "# need to make proxy_sets only the ones that we keep!\n",
    "proxy_sets <- fread(\"~/HRJ_monocytes/Browser_tracks/proxy_setshg38.bed\")\n",
    "eqtls <- fread(\"~/HRJ_monocytes/Browser_tracks/eqtlshg38.bed\")\n",
    "#proxies <- fread(\"~/HRJ_monocytes/Browser_tracks/proxieshg38.bed\")\n",
    "proxies <- fread(\"~/HRJ_monocytes/findmotifs/find_new_dpnII/snps_went_into_loopingQTL_analysis.DpnII.marked.REFALTseqs.PURGED.DISTAL.txt\")\n",
    "proxies[, chrom := paste0(\"chr\", Chr)]\n",
    "proxies_plot <- proxies[, .(chrom, hg38Proxy_pos, hg38Proxy_end, hg38SNP_ID)]\n",
    "\n",
    "# Load in DpnII baits\n",
    "baits <- fread(\"/rds/general/project/lms-spivakov-analysis/live/Design/Human_eQTL_CHiC_DpnII_hg38/eCHiC_grch38.baitmap\")\n",
    "baits[, chrom := paste0(\"chr\", V1)]\n",
    "baits <- baits[, .(chrom, V2, V3, V4, V5)]\n",
    "names(baits) = c(\"chrom\", \"DpnStart\", \"DpnEnd\", \"DpnID\", \"DpnName\")\n",
    "\n",
    "# Bring in data for plotBaits: bg file. NO, too big!\n",
    "#int <- fread(\"~/HRJ_monocytes/CHiC/chicago/combinations_output/DpnII/combined_34reps_DpnII_binsize1500_maxL150K_mergeWeights/data/combined_34reps_DpnII_binsize1500_maxL150K_mergeWeights_interactions.txt\")\n",
    "\n",
    "# also make an overlap with genes\n",
    "proxies1[, TSS1 := TSS - 500]\n",
    "proxies1[, TSS2 := TSS + 500]\n",
    "genes_small <- proxies1[, .(chrom, TSS1, TSS2, rep_Gene, Gene, ENSG_ID)]\n",
    "setkey(loops, V1, V2, V3)\n",
    "loopsG1 <- foverlaps(genes_small, loops, by.x = c(\"chrom\", \"TSS1\", \"TSS2\"), nomatch = NULL)\n",
    "setnames(loopsG1, c(\"chrom\", \"V2\", \"V3\", \"V4\", \"V5\", \"V6\"), c(\"chrom1\", \"start1\", \"end1\", \"chrom2\", \"start2\", \"end2\"))\n",
    "setkey(loops, V4, V5, V6)\n",
    "loopsG2 <- foverlaps(genes_small, loops, by.x = c(\"chrom\", \"TSS1\", \"TSS2\"), nomatch = NULL)\n",
    "setnames(loopsG2, c(\"chrom\", \"V1\", \"V2\", \"V3\", \"V5\", \"V6\"), c(\"chrom2\", \"chrom1\", \"start1\", \"end1\", \"start2\", \"end2\"))\n",
    "loopsG2_edit <- loopsG2[, .(chrom1, start1, end1, chrom2, start2, end2, TSS1, TSS2, rep_Gene, Gene, ENSG_ID)]\n",
    "loops_genes <- rbind(loopsG1, loopsG2_edit)\n",
    "\n",
    "## bring in the regions that were tested in BaseQTL. for each proxy SNP, we can filter this file and merge the gene end to make one PE contact.\n",
    "regions <- fread(\"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/AS_CHiC/BaseQTL/scripts/bedpe_for_chic_egenes_with_Gene_number.txt\")\n",
    "regions[, chrom1 := paste0(\"chr\", V1)]\n",
    "regions[, chrom2 := paste0(\"chr\", V4)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Read in the BaseQTL results, in order to plot the contact QTLs\n",
    "\n",
    "BaseRes <- fread(\"~/HRJ_monocytes/BaseQTL/findings_round2_dpnIICorrection/BaseQTL_CHiC_Filt_on_Dpn_Distal_Rhat_AI_rsids.txt\")\n",
    "BaseBed <- BaseRes[Signif.99 == \"yes\", .(chrom, hg38Proxy_pos, hg38Proxy_end, Gene, hg19SNP_ID, BaseQTL_SNP_hg19, rsid, log_mean_aFC)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 30806 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr1</td><td> 270600</td><td> 271490</td></tr>\n",
       "\t<tr><td>chr1</td><td> 274244</td><td> 275580</td></tr>\n",
       "\t<tr><td>chr1</td><td> 844000</td><td> 844800</td></tr>\n",
       "\t<tr><td>chr1</td><td> 898600</td><td> 898800</td></tr>\n",
       "\t<tr><td>chr1</td><td>1103800</td><td>1105200</td></tr>\n",
       "\t<tr><td>chr1</td><td>1188800</td><td>1189800</td></tr>\n",
       "\t<tr><td>chr1</td><td>1405525</td><td>1405725</td></tr>\n",
       "\t<tr><td>chr1</td><td>1748598</td><td>1749567</td></tr>\n",
       "\t<tr><td>chr1</td><td>1806916</td><td>1807525</td></tr>\n",
       "\t<tr><td>chr1</td><td>1826525</td><td>1827827</td></tr>\n",
       "\t<tr><td>chr1</td><td>1847515</td><td>1847896</td></tr>\n",
       "\t<tr><td>chr1</td><td>1847515</td><td>1850200</td></tr>\n",
       "\t<tr><td>chr1</td><td>1867200</td><td>1873200</td></tr>\n",
       "\t<tr><td>chr1</td><td>1876509</td><td>1878600</td></tr>\n",
       "\t<tr><td>chr1</td><td>1995765</td><td>1997200</td></tr>\n",
       "\t<tr><td>chr1</td><td>2159000</td><td>2159800</td></tr>\n",
       "\t<tr><td>chr1</td><td>2162600</td><td>2163200</td></tr>\n",
       "\t<tr><td>chr1</td><td>2223800</td><td>2225200</td></tr>\n",
       "\t<tr><td>chr1</td><td>2242765</td><td>2243917</td></tr>\n",
       "\t<tr><td>chr1</td><td>2243068</td><td>2243965</td></tr>\n",
       "\t<tr><td>chr1</td><td>2265200</td><td>2267000</td></tr>\n",
       "\t<tr><td>chr1</td><td>2379600</td><td>2379800</td></tr>\n",
       "\t<tr><td>chr1</td><td>3168600</td><td>3169400</td></tr>\n",
       "\t<tr><td>chr1</td><td>3221600</td><td>3222800</td></tr>\n",
       "\t<tr><td>chr1</td><td>3592910</td><td>3594295</td></tr>\n",
       "\t<tr><td>chr1</td><td>3602600</td><td>3603200</td></tr>\n",
       "\t<tr><td>chr1</td><td>3716365</td><td>3718807</td></tr>\n",
       "\t<tr><td>chr1</td><td>3716938</td><td>3718807</td></tr>\n",
       "\t<tr><td>chr1</td><td>3719200</td><td>3720600</td></tr>\n",
       "\t<tr><td>chr1</td><td>3880600</td><td>3882000</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>chrX</td><td>153991800</td><td>153992800</td></tr>\n",
       "\t<tr><td>chrX</td><td>154007000</td><td>154011310</td></tr>\n",
       "\t<tr><td>chrX</td><td>154010282</td><td>154011310</td></tr>\n",
       "\t<tr><td>chrX</td><td>154012242</td><td>154012442</td></tr>\n",
       "\t<tr><td>chrX</td><td>154012541</td><td>154013400</td></tr>\n",
       "\t<tr><td>chrX</td><td>154056200</td><td>154062200</td></tr>\n",
       "\t<tr><td>chrX</td><td>154063400</td><td>154066200</td></tr>\n",
       "\t<tr><td>chrX</td><td>154066800</td><td>154069600</td></tr>\n",
       "\t<tr><td>chrX</td><td>154297800</td><td>154298800</td></tr>\n",
       "\t<tr><td>chrX</td><td>154300400</td><td>154301600</td></tr>\n",
       "\t<tr><td>chrX</td><td>154304200</td><td>154306600</td></tr>\n",
       "\t<tr><td>chrX</td><td>154310200</td><td>154311200</td></tr>\n",
       "\t<tr><td>chrX</td><td>154326200</td><td>154328042</td></tr>\n",
       "\t<tr><td>chrX</td><td>154333200</td><td>154335600</td></tr>\n",
       "\t<tr><td>chrX</td><td>154366242</td><td>154366642</td></tr>\n",
       "\t<tr><td>chrX</td><td>154423800</td><td>154425000</td></tr>\n",
       "\t<tr><td>chrX</td><td>154495445</td><td>154498200</td></tr>\n",
       "\t<tr><td>chrX</td><td>154511200</td><td>154512200</td></tr>\n",
       "\t<tr><td>chrX</td><td>154551800</td><td>154552400</td></tr>\n",
       "\t<tr><td>chrX</td><td>154661400</td><td>154662800</td></tr>\n",
       "\t<tr><td>chrX</td><td>154743200</td><td>154743400</td></tr>\n",
       "\t<tr><td>chrX</td><td>154746200</td><td>154746800</td></tr>\n",
       "\t<tr><td>chrX</td><td>154827000</td><td>154828600</td></tr>\n",
       "\t<tr><td>chrX</td><td>154829200</td><td>154829642</td></tr>\n",
       "\t<tr><td>chrX</td><td>155022400</td><td>155023000</td></tr>\n",
       "\t<tr><td>chrX</td><td>155188200</td><td>155188600</td></tr>\n",
       "\t<tr><td>chrX</td><td>155188800</td><td>155189800</td></tr>\n",
       "\t<tr><td>chrX</td><td>155241800</td><td>155243600</td></tr>\n",
       "\t<tr><td>chrX</td><td>155434765</td><td>155435425</td></tr>\n",
       "\t<tr><td>chrX</td><td>155890800</td><td>155891800</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 30806 × 3\n",
       "\\begin{tabular}{lll}\n",
       " V1 & V2 & V3\\\\\n",
       " <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr1 &  270600 &  271490\\\\\n",
       "\t chr1 &  274244 &  275580\\\\\n",
       "\t chr1 &  844000 &  844800\\\\\n",
       "\t chr1 &  898600 &  898800\\\\\n",
       "\t chr1 & 1103800 & 1105200\\\\\n",
       "\t chr1 & 1188800 & 1189800\\\\\n",
       "\t chr1 & 1405525 & 1405725\\\\\n",
       "\t chr1 & 1748598 & 1749567\\\\\n",
       "\t chr1 & 1806916 & 1807525\\\\\n",
       "\t chr1 & 1826525 & 1827827\\\\\n",
       "\t chr1 & 1847515 & 1847896\\\\\n",
       "\t chr1 & 1847515 & 1850200\\\\\n",
       "\t chr1 & 1867200 & 1873200\\\\\n",
       "\t chr1 & 1876509 & 1878600\\\\\n",
       "\t chr1 & 1995765 & 1997200\\\\\n",
       "\t chr1 & 2159000 & 2159800\\\\\n",
       "\t chr1 & 2162600 & 2163200\\\\\n",
       "\t chr1 & 2223800 & 2225200\\\\\n",
       "\t chr1 & 2242765 & 2243917\\\\\n",
       "\t chr1 & 2243068 & 2243965\\\\\n",
       "\t chr1 & 2265200 & 2267000\\\\\n",
       "\t chr1 & 2379600 & 2379800\\\\\n",
       "\t chr1 & 3168600 & 3169400\\\\\n",
       "\t chr1 & 3221600 & 3222800\\\\\n",
       "\t chr1 & 3592910 & 3594295\\\\\n",
       "\t chr1 & 3602600 & 3603200\\\\\n",
       "\t chr1 & 3716365 & 3718807\\\\\n",
       "\t chr1 & 3716938 & 3718807\\\\\n",
       "\t chr1 & 3719200 & 3720600\\\\\n",
       "\t chr1 & 3880600 & 3882000\\\\\n",
       "\t ⋮ & ⋮ & ⋮\\\\\n",
       "\t chrX & 153991800 & 153992800\\\\\n",
       "\t chrX & 154007000 & 154011310\\\\\n",
       "\t chrX & 154010282 & 154011310\\\\\n",
       "\t chrX & 154012242 & 154012442\\\\\n",
       "\t chrX & 154012541 & 154013400\\\\\n",
       "\t chrX & 154056200 & 154062200\\\\\n",
       "\t chrX & 154063400 & 154066200\\\\\n",
       "\t chrX & 154066800 & 154069600\\\\\n",
       "\t chrX & 154297800 & 154298800\\\\\n",
       "\t chrX & 154300400 & 154301600\\\\\n",
       "\t chrX & 154304200 & 154306600\\\\\n",
       "\t chrX & 154310200 & 154311200\\\\\n",
       "\t chrX & 154326200 & 154328042\\\\\n",
       "\t chrX & 154333200 & 154335600\\\\\n",
       "\t chrX & 154366242 & 154366642\\\\\n",
       "\t chrX & 154423800 & 154425000\\\\\n",
       "\t chrX & 154495445 & 154498200\\\\\n",
       "\t chrX & 154511200 & 154512200\\\\\n",
       "\t chrX & 154551800 & 154552400\\\\\n",
       "\t chrX & 154661400 & 154662800\\\\\n",
       "\t chrX & 154743200 & 154743400\\\\\n",
       "\t chrX & 154746200 & 154746800\\\\\n",
       "\t chrX & 154827000 & 154828600\\\\\n",
       "\t chrX & 154829200 & 154829642\\\\\n",
       "\t chrX & 155022400 & 155023000\\\\\n",
       "\t chrX & 155188200 & 155188600\\\\\n",
       "\t chrX & 155188800 & 155189800\\\\\n",
       "\t chrX & 155241800 & 155243600\\\\\n",
       "\t chrX & 155434765 & 155435425\\\\\n",
       "\t chrX & 155890800 & 155891800\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 30806 × 3\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| chr1 |  270600 |  271490 |\n",
       "| chr1 |  274244 |  275580 |\n",
       "| chr1 |  844000 |  844800 |\n",
       "| chr1 |  898600 |  898800 |\n",
       "| chr1 | 1103800 | 1105200 |\n",
       "| chr1 | 1188800 | 1189800 |\n",
       "| chr1 | 1405525 | 1405725 |\n",
       "| chr1 | 1748598 | 1749567 |\n",
       "| chr1 | 1806916 | 1807525 |\n",
       "| chr1 | 1826525 | 1827827 |\n",
       "| chr1 | 1847515 | 1847896 |\n",
       "| chr1 | 1847515 | 1850200 |\n",
       "| chr1 | 1867200 | 1873200 |\n",
       "| chr1 | 1876509 | 1878600 |\n",
       "| chr1 | 1995765 | 1997200 |\n",
       "| chr1 | 2159000 | 2159800 |\n",
       "| chr1 | 2162600 | 2163200 |\n",
       "| chr1 | 2223800 | 2225200 |\n",
       "| chr1 | 2242765 | 2243917 |\n",
       "| chr1 | 2243068 | 2243965 |\n",
       "| chr1 | 2265200 | 2267000 |\n",
       "| chr1 | 2379600 | 2379800 |\n",
       "| chr1 | 3168600 | 3169400 |\n",
       "| chr1 | 3221600 | 3222800 |\n",
       "| chr1 | 3592910 | 3594295 |\n",
       "| chr1 | 3602600 | 3603200 |\n",
       "| chr1 | 3716365 | 3718807 |\n",
       "| chr1 | 3716938 | 3718807 |\n",
       "| chr1 | 3719200 | 3720600 |\n",
       "| chr1 | 3880600 | 3882000 |\n",
       "| ⋮ | ⋮ | ⋮ |\n",
       "| chrX | 153991800 | 153992800 |\n",
       "| chrX | 154007000 | 154011310 |\n",
       "| chrX | 154010282 | 154011310 |\n",
       "| chrX | 154012242 | 154012442 |\n",
       "| chrX | 154012541 | 154013400 |\n",
       "| chrX | 154056200 | 154062200 |\n",
       "| chrX | 154063400 | 154066200 |\n",
       "| chrX | 154066800 | 154069600 |\n",
       "| chrX | 154297800 | 154298800 |\n",
       "| chrX | 154300400 | 154301600 |\n",
       "| chrX | 154304200 | 154306600 |\n",
       "| chrX | 154310200 | 154311200 |\n",
       "| chrX | 154326200 | 154328042 |\n",
       "| chrX | 154333200 | 154335600 |\n",
       "| chrX | 154366242 | 154366642 |\n",
       "| chrX | 154423800 | 154425000 |\n",
       "| chrX | 154495445 | 154498200 |\n",
       "| chrX | 154511200 | 154512200 |\n",
       "| chrX | 154551800 | 154552400 |\n",
       "| chrX | 154661400 | 154662800 |\n",
       "| chrX | 154743200 | 154743400 |\n",
       "| chrX | 154746200 | 154746800 |\n",
       "| chrX | 154827000 | 154828600 |\n",
       "| chrX | 154829200 | 154829642 |\n",
       "| chrX | 155022400 | 155023000 |\n",
       "| chrX | 155188200 | 155188600 |\n",
       "| chrX | 155188800 | 155189800 |\n",
       "| chrX | 155241800 | 155243600 |\n",
       "| chrX | 155434765 | 155435425 |\n",
       "| chrX | 155890800 | 155891800 |\n",
       "\n"
      ],
      "text/plain": [
       "      V1   V2        V3       \n",
       "1     chr1  270600    271490  \n",
       "2     chr1  274244    275580  \n",
       "3     chr1  844000    844800  \n",
       "4     chr1  898600    898800  \n",
       "5     chr1 1103800   1105200  \n",
       "6     chr1 1188800   1189800  \n",
       "7     chr1 1405525   1405725  \n",
       "8     chr1 1748598   1749567  \n",
       "9     chr1 1806916   1807525  \n",
       "10    chr1 1826525   1827827  \n",
       "11    chr1 1847515   1847896  \n",
       "12    chr1 1847515   1850200  \n",
       "13    chr1 1867200   1873200  \n",
       "14    chr1 1876509   1878600  \n",
       "15    chr1 1995765   1997200  \n",
       "16    chr1 2159000   2159800  \n",
       "17    chr1 2162600   2163200  \n",
       "18    chr1 2223800   2225200  \n",
       "19    chr1 2242765   2243917  \n",
       "20    chr1 2243068   2243965  \n",
       "21    chr1 2265200   2267000  \n",
       "22    chr1 2379600   2379800  \n",
       "23    chr1 3168600   3169400  \n",
       "24    chr1 3221600   3222800  \n",
       "25    chr1 3592910   3594295  \n",
       "26    chr1 3602600   3603200  \n",
       "27    chr1 3716365   3718807  \n",
       "28    chr1 3716938   3718807  \n",
       "29    chr1 3719200   3720600  \n",
       "30    chr1 3880600   3882000  \n",
       "⋮     ⋮    ⋮         ⋮        \n",
       "30777 chrX 153991800 153992800\n",
       "30778 chrX 154007000 154011310\n",
       "30779 chrX 154010282 154011310\n",
       "30780 chrX 154012242 154012442\n",
       "30781 chrX 154012541 154013400\n",
       "30782 chrX 154056200 154062200\n",
       "30783 chrX 154063400 154066200\n",
       "30784 chrX 154066800 154069600\n",
       "30785 chrX 154297800 154298800\n",
       "30786 chrX 154300400 154301600\n",
       "30787 chrX 154304200 154306600\n",
       "30788 chrX 154310200 154311200\n",
       "30789 chrX 154326200 154328042\n",
       "30790 chrX 154333200 154335600\n",
       "30791 chrX 154366242 154366642\n",
       "30792 chrX 154423800 154425000\n",
       "30793 chrX 154495445 154498200\n",
       "30794 chrX 154511200 154512200\n",
       "30795 chrX 154551800 154552400\n",
       "30796 chrX 154661400 154662800\n",
       "30797 chrX 154743200 154743400\n",
       "30798 chrX 154746200 154746800\n",
       "30799 chrX 154827000 154828600\n",
       "30800 chrX 154829200 154829642\n",
       "30801 chrX 155022400 155023000\n",
       "30802 chrX 155188200 155188600\n",
       "30803 chrX 155188800 155189800\n",
       "30804 chrX 155241800 155243600\n",
       "30805 chrX 155434765 155435425\n",
       "30806 chrX 155890800 155891800"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Bring in ENCODE ccREs\n",
    "encode <- fread(\"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/external_data/ENCODE/GRCh38-cCREs.bed\")\n",
    "#unique(encode$V6)\n",
    "ELS <- encode[V6 %like% \"ELS\"]\n",
    "mono_Enh <- fread(\"~/ext_datasets/blueprint_projected_segmentations/hg38/active_enhancers_mon.hg38.bed\")\n",
    "mono_Enh\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Prepare genotype-specific PCK2 data - NOT USING NOW, BUT DO GET LOOPS\n",
    "mydir <- \"~/HRJ_monocytes/CHiC/merge_genos/chicago/output/DpnII/data\"\n",
    "source(\"~/Rfunctions/helen_functions.R\")\n",
    "\n",
    "\n",
    "#ref_eQTL <- fread(paste(mydir, \"14:24527672:G:A_homozygous_REF_merged_DpnII_washU_NEW_eQTLinteractions.txt\", sep = \"/\"))\n",
    "#ref_eQTL_bpe <- washuNEW2bedpe(ref_eQTL)\n",
    "#ref_eQTL_bpe[, genotype_REF := \"TRUE\"]\n",
    "\n",
    "#alt_eQTL <- fread(paste(mydir, \"14:24527672:G:A_homozygous_ALT_merged_DpnII_washU_NEW_eQTLinteractions.txt\", sep = \"/\"))\n",
    "#alt_eQTL_bpe <- washuNEW2bedpe(alt_eQTL)\n",
    "#alt_eQTL_bpe[, genotype_ALT := \"TRUE\"]\n",
    "\n",
    "#both_eQTL_bpe <- merge.data.table(ref_eQTL_bpe, alt_eQTL_bpe, on = c(\"V1\", \"V2\", \"V3\", \"chr2\", \"start2\", \"end2\"), all = T)\n",
    "#both_eQTL_bpe[genotype_REF == T & genotype_ALT == T, allele := \"Either\"]\n",
    "#both_eQTL_bpe[genotype_REF == T & is.na(genotype_ALT), allele := \"REF\"]\n",
    "#both_eQTL_bpe[is.na(genotype_REF) & genotype_ALT == T, allele := \"ALT\"]\n",
    "#both_eQTL_bpe\n",
    "\n",
    "#ref_PCK2 <- fread(paste(mydir, \"14:24527672:G:A_homozygous_REF_merged_DpnII_washU_NEW_PCK2_allPromInteractions.txt\", sep = \"/\"))\n",
    "#ref_PCK2_bpe <- washuNEW2bedpe(ref_PCK2)\n",
    "#ref_PCK2_bpe[, genotype_REF := \"TRUE\"]\n",
    "\n",
    "#alt_PCK2 <- fread(paste(mydir, \"14:24527672:G:A_homozygous_ALT_merged_DpnII_washU_NEW_PCK2_allPromInteractions.txt\", sep = \"/\"))\n",
    "#alt_PCK2_bpe <- washuNEW2bedpe(alt_PCK2)\n",
    "#alt_PCK2_bpe[, genotype_ALT := \"TRUE\"]\n",
    "\n",
    "#both_PCK2_bpe <- merge.data.table(ref_PCK2_bpe, alt_PCK2_bpe, on = c(\"V1\", \"V2\", \"V3\", \"chr2\", \"start2\", \"end2\"), all = T)\n",
    "#both_PCK2_bpe[genotype_REF == T & genotype_ALT == T, allele := \"Either\"]\n",
    "#both_PCK2_bpe[genotype_REF == T & is.na(genotype_ALT), allele := \"REF\"]\n",
    "#both_PCK2_bpe[is.na(genotype_REF) & genotype_ALT == T, allele := \"ALT\"]\n",
    "#both_PCK2_bpe\n",
    "\n",
    "# Reading interactions from the chicdiff analysis\n",
    "mydir <- \"~/HRJ_monocytes/CHiC/merge_genos/chicdiff\"\n",
    "#list.files(mydir)\n",
    "PCK2counts <- readRDS(paste0(mydir, \"/PCK2_14:24527672:G:A_5K_expand3_countput.Rds\"))\n",
    "rmap5K <- fread(\"/rds/general/project/lms-spivakov-analysis/live/Design/Human_eQTL_CHiC_DpnII_hg38_5K_bins/human_DpnII_5K.rmap\")\n",
    "names(rmap5K) = c(\"chr\", \"start\", \"end\", \"ID\")\n",
    "rmap5K[, chrom := paste0(\"chr\", chr)]\n",
    "rmap5K[, chr := NULL]\n",
    "setkey(rmap5K, ID)\n",
    "\n",
    "sigLoops <- PCK2counts[score > 5]\n",
    "loops1 <- rmap5K[sigLoops, on = c(ID = \"baitID\")]\n",
    "setnames(loops1, c(\"chrom\", \"start\", \"end\", \"ID\"), c(\"baitChrom\", \"baitStart\", \"baitEnd\", \"baitID\"))\n",
    "loops2 <- rmap5K[loops1, on = c(ID = \"otherEndID\")]\n",
    "setnames(loops2, c(\"chrom\", \"start\", \"end\", \"ID\"), c(\"oeChrom\", \"oeStart\", \"oeEnd\", \"oeID\"))\n",
    "loops2[baitStart > oeStart, `:=` (leftChr = oeChrom, \n",
    "                                  leftStart = oeStart, \n",
    "                                  leftEnd = oeEnd, \n",
    "                                 rightChr = baitChrom, \n",
    "                                 rightStart = baitStart, \n",
    "                                 rightEnd = baitEnd)]\n",
    "loops2[baitStart < oeStart, `:=` (leftChr = baitChrom, \n",
    "                                  leftStart = baitStart, \n",
    "                                  leftEnd = baitEnd, \n",
    "                                 rightChr = oeChrom, \n",
    "                                 rightStart = oeStart, \n",
    "                                 rightEnd = oeEnd)]\n",
    "loops_plot <- loops2[, .(leftChr, leftStart, leftEnd, rightChr, rightStart, rightEnd, condition, baitID)]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Plot the PCK2 locus including loops for different genotypes. Will insert the plotbaits from chicdiff. Make sure to use 200kb to left and right of pck2."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "mid of the bait 135210 at PCK2 is 24095221, so start = 23895221 and end = 24295221"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>start</th><th scope=col>end</th><th scope=col>ID</th><th scope=col>chrom</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>24092611</td><td>24097831</td><td>135210</td><td>chr14</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 4\n",
       "\\begin{tabular}{llll}\n",
       " start & end & ID & chrom\\\\\n",
       " <int> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t 24092611 & 24097831 & 135210 & chr14\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 4\n",
       "\n",
       "| start &lt;int&gt; | end &lt;int&gt; | ID &lt;int&gt; | chrom &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| 24092611 | 24097831 | 135210 | chr14 |\n",
       "\n"
      ],
      "text/plain": [
       "  start    end      ID     chrom\n",
       "1 24092611 24097831 135210 chr14"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>start</th><th scope=col>end</th><th scope=col>ID</th><th scope=col>chrom</th><th scope=col>mid</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>24092611</td><td>24097831</td><td>135210</td><td>chr14</td><td>24095221</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " start & end & ID & chrom & mid\\\\\n",
       " <int> & <int> & <int> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 24092611 & 24097831 & 135210 & chr14 & 24095221\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 5\n",
       "\n",
       "| start &lt;int&gt; | end &lt;int&gt; | ID &lt;int&gt; | chrom &lt;chr&gt; | mid &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 24092611 | 24097831 | 135210 | chr14 | 24095221 |\n",
       "\n"
      ],
      "text/plain": [
       "  start    end      ID     chrom mid     \n",
       "1 24092611 24097831 135210 chr14 24095221"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "23895221"
      ],
      "text/latex": [
       "23895221"
      ],
      "text/markdown": [
       "23895221"
      ],
      "text/plain": [
       "[1] 23895221"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "24295221"
      ],
      "text/latex": [
       "24295221"
      ],
      "text/markdown": [
       "24295221"
      ],
      "text/plain": [
       "[1] 24295221"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "myBait <- 135210\n",
    "rmap5K[ID == myBait]\n",
    "rmap5K[, mid := (start+end)/2]\n",
    "rmap5K[ID == myBait]\n",
    "24095221 - 200000\n",
    "24095221 + 200000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "highlight[highlight1]\n",
      "\n",
      "highlight[highlight2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#### PCK2 locus: different genotypes\n",
    "#### check colorbind colors here check colours here https://colorbrewer2.org/#type=diverging&scheme=RdYlBu&n=7\n",
    "\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"14:24058463:G:A\"\n",
    "myGene = \"PCK2\"\n",
    "mychrom = \"chr14\"\n",
    "mystart = 23895221\n",
    "myend = 24295221\n",
    "###\n",
    "### restrict loops to PCK2\n",
    "REF_loops <- unique(loops_plot[condition == \"REF_5K\" & baitID == myBait])\n",
    "ALT_loops <- unique(loops_plot[condition == \"ALT_5K\" & baitID == myBait])\n",
    "\n",
    "###\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "# get CTCF predicted mono\n",
    "mybeds <- \"/rds/general/user/hrayjone/home/ext_datasets/blueprint_projected_segmentations/hg38\"\n",
    "CTCF <- fread(paste(mybeds, \"CTCF_monocyte.bed\", sep = \"/\"))\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/PCK2_refalt_plotgardener_genoSpecific_V02.pdf\")\n",
    "pageCreate(width = 15, height = 12, default.units = \"cm\", showGuides = F)\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.3, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 3.8, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(label = \"Genes\", fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_main,\n",
    "    y = 4.3, x = 0.4, height = 1,\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, \n",
    "                   scale = \"Mb\", fontsize = 7,\n",
    "                   y = 10.3)\n",
    "\n",
    "#### Add genotype specific 5Kb plotBaits NOT DOING NOW. PASTED BELOW.\n",
    "\n",
    "# Plot eQTLs\n",
    "#plotRanges( \n",
    "#    data = myeqtls, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "#    y = 1.2, height = 0.3\n",
    "#)\n",
    "# text for eQTLs\n",
    "#plotText(\n",
    "#    label = \"Lead eQTL\", \n",
    "#    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 1.2, height = 0.5,\n",
    "#)\n",
    "\n",
    "### Plot the contact QTLs\n",
    "contact <- BaseBed[Gene == myGene]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 2.8, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTL/ \\nlead eQTL\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 2.8, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 3.3, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD SNPs\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 3.5, height = 0.5\n",
    ")\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_main, height = 1, y = 0,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0,\n",
    "    params = params_text,\n",
    "    y = 0.2, height = 0.5,\n",
    ") \n",
    "\n",
    "# range for ATAC signal\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,13), label = T, # top ATAC -log10P\n",
    "    main = TRUE, axisLine = T, params = params_main, \n",
    "    y = 0, height = 1, fontsize = 7, fontcolor = \"seagreen\"\n",
    ")\n",
    "##### plot loops.\n",
    "plotPairsArches(\n",
    "    data = REF_loops, params = params_main, y = 6.5, height = 1,\n",
    "    fill = \"#0072B2\", linecolor = \"fill\", flip = TRUE, clip = T\n",
    ")\n",
    "# text for loops: REF\n",
    "plotText(\n",
    "    label = \"PCK2 significant\\ncontacts, REF\", fontsize = 7,\n",
    "    params = params_text,\n",
    "    y = 6.5, height = 0.5, fontcolor = \"#0072B2\"\n",
    ")\n",
    "plotPairsArches(\n",
    "    data = ALT_loops, params = params_main, y = 9, height = 1,\n",
    "    fill = \"#b2182b\", linecolor = \"fill\", flip = TRUE, clip = T\n",
    ")\n",
    "# text for loops: ALT\n",
    "plotText(\n",
    "    label = \"PCK2 significant\\ncontacts, ALT\", fontsize = 7,\n",
    "    params = params_text,\n",
    "    y = 9.1, height = 0.5, fontcolor = \"#b2182b\"\n",
    ")\n",
    "\n",
    "### put in text for plotbaits\n",
    "plotText(\n",
    "    label = \"PCK2 bait plot at\\n 5kb resolution, REF\", fontsize = 7,\n",
    "    params = params_text, x = 1, \n",
    "    y = 5.2, height = 0.5, fontcolor = \"gray40\"\n",
    ")\n",
    "plotText(\n",
    "    label = \"PCK2 bait plot at\\n 5kb resolution, ALT\", fontsize = 7,\n",
    "    params = params_text, x = 1,\n",
    "    y = 7.9, height = 0.5, fontcolor = \"gray40\"\n",
    ")\n",
    "\n",
    "### plot CTCF\n",
    "## Load ENCODE ChIP-seq: reps 1 and 2 from accession ENCFF167VRL http://dcc.blueprint-epigenome.eu/#/files\n",
    "CTCF_encode <- readBigwig(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/external_data/ENCODE/ENCFF167VRL_CTCF.bigWig\", \n",
    "                          chrom = mychrom,\n",
    "                          chromstart = mystart,\n",
    "                          chromend = myend)\n",
    "CTCFsignal <- plotSignal(\n",
    "    data = CTCF_encode,\n",
    "    params = params_main, height = 0.8, y = 1.2,\n",
    "    linecolor = \"maroon3\"\n",
    ")\n",
    "# text for ENCODE CTCF ChIP seq\n",
    "plotText(\n",
    "    label = \"ENCODE CTCF, \\nmono (-log10P)\", fontsize = 7, fontcolor = \"maroon3\",\n",
    "    rot = 0,\n",
    "    params = params_text,\n",
    "    y = 1.2, height = 0.5,\n",
    ") \n",
    "\n",
    "# range for CTCF\n",
    "annoYaxis(\n",
    "    CTCFsignal, at = c(0,478),\n",
    "    label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, \n",
    "    y = 1.2, height = 0.8, fontsize = 7, fontcolor = \"gray40\"\n",
    ")\n",
    "\n",
    "## plot enhancers\n",
    "encode <- plotRanges(\n",
    "    data= mono_Enh, \n",
    "    params = params_main, fill = \"burlywood\", # only one type of cCRE in this region - distal enhancer like signature\n",
    "    linecolor = \"fill\",\n",
    "    collapse = T, y = 2.2, height = 0.3\n",
    ")\n",
    "# text for enhancers\n",
    "plotText(\n",
    "    label = \"Blueprint projected\\nmonocyte enhancers\", fontsize = 7, fontcolor = \"burlywood\",\n",
    "    rot = 0,\n",
    "    params = params_text,\n",
    "    y = 2, height = 0.5,\n",
    ") \n",
    "\n",
    "## Going with ENCODE for CTCF, instead(above)\n",
    "## Add CTCF - (no other enhaner regions found in mono)\n",
    "#plotRanges( \n",
    "#    data = CTCF,\n",
    "#    params = params_main, linecolor=\"blue\", fill = \"blue\", \n",
    "#    collapse = T,\n",
    "#    y = 6, height = 0.5\n",
    "#)\n",
    "# text for CTCF\n",
    "#plotText(\n",
    "#    label = \"Blueprint \\nCTCF, mono\", fontsize = 7, fontcolor = \"blue\",\n",
    "#    rot = 0,\n",
    "#    params = params_text,\n",
    "#    y = 6.2, height = 0.5,\n",
    "#) \n",
    "\n",
    "# HIGHLIGHT OUR REGION: eQTL\n",
    "annoHighlight(\n",
    "    plot = CTCFsignal, chrom = mychrom, chromstart = 24054268, chromend = 24059365, # this is the 5Kb bait\n",
    "    y = 0, height = 10.3, just = c(\"left\", \"top\"), default.units = \"cm\"\n",
    ")\n",
    "\n",
    "# HIGHLIGHT OUR REGION: PCK2 gene\n",
    "# chr14:24092611-24097831\n",
    "annoHighlight(plot = CTCFsignal, chrom = \"chr14\", chromstart = 24092611, chromend = 24097831, # this is the 5kb bait\n",
    "    y = 0, height = 10.3, just = c(\"left\", \"top\"), default.units = \"cm\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 26 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>leftChr</th><th scope=col>leftStart</th><th scope=col>leftEnd</th><th scope=col>rightChr</th><th scope=col>rightStart</th><th scope=col>rightEnd</th><th scope=col>condition</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr14</td><td>23971624</td><td>23976928</td><td>chr14</td><td>24092611</td><td>24097831</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>23976929</td><td>23982303</td><td>chr14</td><td>24092611</td><td>24097831</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24046767</td><td>24054267</td><td>chr14</td><td>24092611</td><td>24097831</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24054268</td><td>24059365</td><td>chr14</td><td>24092611</td><td>24097831</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24059366</td><td>24066572</td><td>chr14</td><td>24092611</td><td>24097831</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24035498</td><td>24041675</td><td>chr14</td><td>24092611</td><td>24097831</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24133811</td><td>24139137</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24178165</td><td>24183603</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24193697</td><td>24198834</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24230327</td><td>24235501</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24235502</td><td>24241405</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24241406</td><td>24246687</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24291217</td><td>24296582</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24285940</td><td>24291216</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24296583</td><td>24301980</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24301981</td><td>24307429</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24307430</td><td>24312470</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24312471</td><td>24317623</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24322723</td><td>24327897</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24327898</td><td>24333283</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24333284</td><td>24338285</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24102963</td><td>24108266</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24155982</td><td>24161269</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24172310</td><td>24178164</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24317624</td><td>24322722</td><td>REF_5K</td></tr>\n",
       "\t<tr><td>chr14</td><td>24092611</td><td>24097831</td><td>chr14</td><td>24431974</td><td>24436979</td><td>REF_5K</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 26 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " leftChr & leftStart & leftEnd & rightChr & rightStart & rightEnd & condition\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr14 & 23971624 & 23976928 & chr14 & 24092611 & 24097831 & REF\\_5K\\\\\n",
       "\t chr14 & 23976929 & 23982303 & chr14 & 24092611 & 24097831 & REF\\_5K\\\\\n",
       "\t chr14 & 24046767 & 24054267 & chr14 & 24092611 & 24097831 & REF\\_5K\\\\\n",
       "\t chr14 & 24054268 & 24059365 & chr14 & 24092611 & 24097831 & REF\\_5K\\\\\n",
       "\t chr14 & 24059366 & 24066572 & chr14 & 24092611 & 24097831 & REF\\_5K\\\\\n",
       "\t chr14 & 24035498 & 24041675 & chr14 & 24092611 & 24097831 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24133811 & 24139137 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24178165 & 24183603 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24193697 & 24198834 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24230327 & 24235501 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24235502 & 24241405 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24241406 & 24246687 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24291217 & 24296582 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24285940 & 24291216 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24296583 & 24301980 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24301981 & 24307429 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24307430 & 24312470 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24312471 & 24317623 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24322723 & 24327897 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24327898 & 24333283 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24333284 & 24338285 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24102963 & 24108266 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24155982 & 24161269 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24172310 & 24178164 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24317624 & 24322722 & REF\\_5K\\\\\n",
       "\t chr14 & 24092611 & 24097831 & chr14 & 24431974 & 24436979 & REF\\_5K\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 26 × 7\n",
       "\n",
       "| leftChr &lt;chr&gt; | leftStart &lt;int&gt; | leftEnd &lt;int&gt; | rightChr &lt;chr&gt; | rightStart &lt;int&gt; | rightEnd &lt;int&gt; | condition &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| chr14 | 23971624 | 23976928 | chr14 | 24092611 | 24097831 | REF_5K |\n",
       "| chr14 | 23976929 | 23982303 | chr14 | 24092611 | 24097831 | REF_5K |\n",
       "| chr14 | 24046767 | 24054267 | chr14 | 24092611 | 24097831 | REF_5K |\n",
       "| chr14 | 24054268 | 24059365 | chr14 | 24092611 | 24097831 | REF_5K |\n",
       "| chr14 | 24059366 | 24066572 | chr14 | 24092611 | 24097831 | REF_5K |\n",
       "| chr14 | 24035498 | 24041675 | chr14 | 24092611 | 24097831 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24133811 | 24139137 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24178165 | 24183603 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24193697 | 24198834 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24230327 | 24235501 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24235502 | 24241405 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24241406 | 24246687 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24291217 | 24296582 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24285940 | 24291216 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24296583 | 24301980 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24301981 | 24307429 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24307430 | 24312470 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24312471 | 24317623 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24322723 | 24327897 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24327898 | 24333283 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24333284 | 24338285 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24102963 | 24108266 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24155982 | 24161269 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24172310 | 24178164 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24317624 | 24322722 | REF_5K |\n",
       "| chr14 | 24092611 | 24097831 | chr14 | 24431974 | 24436979 | REF_5K |\n",
       "\n"
      ],
      "text/plain": [
       "   leftChr leftStart leftEnd  rightChr rightStart rightEnd condition\n",
       "1  chr14   23971624  23976928 chr14    24092611   24097831 REF_5K   \n",
       "2  chr14   23976929  23982303 chr14    24092611   24097831 REF_5K   \n",
       "3  chr14   24046767  24054267 chr14    24092611   24097831 REF_5K   \n",
       "4  chr14   24054268  24059365 chr14    24092611   24097831 REF_5K   \n",
       "5  chr14   24059366  24066572 chr14    24092611   24097831 REF_5K   \n",
       "6  chr14   24035498  24041675 chr14    24092611   24097831 REF_5K   \n",
       "7  chr14   24092611  24097831 chr14    24133811   24139137 REF_5K   \n",
       "8  chr14   24092611  24097831 chr14    24178165   24183603 REF_5K   \n",
       "9  chr14   24092611  24097831 chr14    24193697   24198834 REF_5K   \n",
       "10 chr14   24092611  24097831 chr14    24230327   24235501 REF_5K   \n",
       "11 chr14   24092611  24097831 chr14    24235502   24241405 REF_5K   \n",
       "12 chr14   24092611  24097831 chr14    24241406   24246687 REF_5K   \n",
       "13 chr14   24092611  24097831 chr14    24291217   24296582 REF_5K   \n",
       "14 chr14   24092611  24097831 chr14    24285940   24291216 REF_5K   \n",
       "15 chr14   24092611  24097831 chr14    24296583   24301980 REF_5K   \n",
       "16 chr14   24092611  24097831 chr14    24301981   24307429 REF_5K   \n",
       "17 chr14   24092611  24097831 chr14    24307430   24312470 REF_5K   \n",
       "18 chr14   24092611  24097831 chr14    24312471   24317623 REF_5K   \n",
       "19 chr14   24092611  24097831 chr14    24322723   24327897 REF_5K   \n",
       "20 chr14   24092611  24097831 chr14    24327898   24333283 REF_5K   \n",
       "21 chr14   24092611  24097831 chr14    24333284   24338285 REF_5K   \n",
       "22 chr14   24092611  24097831 chr14    24102963   24108266 REF_5K   \n",
       "23 chr14   24092611  24097831 chr14    24155982   24161269 REF_5K   \n",
       "24 chr14   24092611  24097831 chr14    24172310   24178164 REF_5K   \n",
       "25 chr14   24092611  24097831 chr14    24317624   24322722 REF_5K   \n",
       "26 chr14   24092611  24097831 chr14    24431974   24436979 REF_5K   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "REF_loops"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr14</td><td>24058463</td><td>24058464</td><td>14:24058463:G:A</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 4\n",
       "\\begin{tabular}{llll}\n",
       " V1 & V2 & V3 & V4\\\\\n",
       " <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr14 & 24058463 & 24058464 & 14:24058463:G:A\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 4\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| chr14 | 24058463 | 24058464 | 14:24058463:G:A |\n",
       "\n"
      ],
      "text/plain": [
       "  V1    V2       V3       V4             \n",
       "1 chr14 24058463 24058464 14:24058463:G:A"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Make a plot of the region surrounding our SNP\n",
    "### See if can add the motif as a zoom in to the SNP\n",
    "\n",
    "\n",
    "mybeds <- \"/rds/general/user/hrayjone/home/ext_datasets/blueprint_projected_segmentations/hg38\"\n",
    "#h3K27ac <- fread(paste(mybeds, \"H3K27ac_monocyte.bed\", sep = \"/\"))\n",
    "#atacbed <- fread(paste(mybeds, \"consensus_Genrich.bed\", sep = \"/\"))\n",
    "# no other segmentations here\n",
    "CTCF <- fread(paste(mybeds, \"CTCF_monocyte.bed\", sep = \"/\"))\n",
    "### set parameters\n",
    "myeQTL <- \"14:24058463:G:A\"\n",
    "myGene = \"PCK2\"\n",
    "mychrom = \"chr14\"\n",
    "mystart = 24058000\n",
    "myend = 24059000\n",
    "###\n",
    "### the region is encode enhancer!! EH38E1703534 (enhancer-like signature, CTCF bound https://screen-v2.wenglab.org/search/?q=EH38E1703534&assembly=GRCh38)\n",
    "encode <- fread(\"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/external_data/ENCODE/GRCh38-cCREs.bed\")\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/PCK2_SNP_zoomIn_plotgardener.pdf\")\n",
    "pageCreate(width = 13, height = 7, default.units = \"cm\", showGuides = F)\n",
    "params_a <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2.5, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_a, stroke = 1, fontsize = 8, fontcolor = \"grey48\",\n",
    "                     fill = \"grey48\",\n",
    "                        y = 3.3, height = 1)\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", fontsize = 7, fontcolor = \"grey48\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 3.8, x = 0.1, height = 0.5,\n",
    ") \n",
    "\n",
    "# text for annotate genome\n",
    "plotText(\n",
    "    label = \"chr14\", fontsize = 7, fontcolor = \"black\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 4.2, x = 0.1, height = 0.5,\n",
    ")\n",
    "\n",
    "## Add eQTL\n",
    "plotRanges( \n",
    "    data = proxy_sets_thisone,\n",
    "    params = params_a, linecolor=\"tomato3\", fill = \"tomato3\", \n",
    "    collapse = T,\n",
    "    y = 0.5, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"eQTLs\", fontsize = 7, fontcolor = \"tomato3\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 0.6, x = 0.1, height = 0.5,\n",
    ")\n",
    "\n",
    "# text for rs7146599\n",
    "plotText(\n",
    "    label = \"rs7146599\", fontsize = 7, fontcolor = \"tomato3\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 0.5, x = 7.3, height = 0.5,\n",
    ")\n",
    "\n",
    "\n",
    "\n",
    "## Add CTCF - (no other enhaner regions found in mono)\n",
    "plotRanges( \n",
    "    data = CTCF,\n",
    "    params = params_a, linecolor=\"blue\", fill = \"blue\", \n",
    "    collapse = T,\n",
    "    y = 2.3, height = 0.2\n",
    ")\n",
    "# text for CTCF\n",
    "plotText(\n",
    "    label = \"Blueprint \\nCTCF, mono\", fontsize = 7, fontcolor = \"blue\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2, x = 0.1, height = 0.5,\n",
    ") \n",
    "\n",
    "\n",
    "## Add ENCODE ccREs\n",
    "plotRanges( \n",
    "    data = encode,\n",
    "    params = params_a,\n",
    "    fill = \"burlywood\", # only one type of cCRE in this region - distal enhancer like signature\n",
    "    linecolor = \"fill\",\n",
    "    collapse = T,\n",
    "    y = 2.8, height = 0.2\n",
    ")\n",
    "# text for enhancer\n",
    "plotText(\n",
    "    label = \"ENCODE cCRE, \\ndistal enhancer\", fontsize = 7, fontcolor = \"darkgoldenrod2\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.7, x = 0.1, height = 0.5,\n",
    ") \n",
    "\n",
    "## Load ENCODE ChIP-seq: reps 1 and 2 from accession ENCFF167VRL http://dcc.blueprint-epigenome.eu/#/files\n",
    "CTCF_encode <- readBigwig(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/external_data/ENCODE/ENCFF167VRL_CTCF.bigWig\", \n",
    "                          chrom = mychrom,\n",
    "                          chromstart = mystart,\n",
    "                          chromend = myend)\n",
    "CTCFsignal <- plotSignal(\n",
    "    data = CTCF_encode,\n",
    "    params = params_a, height = 1, y = 1,\n",
    "    linecolor = \"maroon3\"\n",
    ")\n",
    "# text for ENCODE CTCF ChIP seq\n",
    "plotText(\n",
    "    label = \"ENCODE CTCF, \\nmono (-log10P)\", fontsize = 7, fontcolor = \"maroon3\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 1.2, x = 0.1, height = 0.5,\n",
    ") \n",
    "\n",
    "# range for CTCF\n",
    "annoYaxis(\n",
    "    CTCFsignal,\n",
    "    at = c(0, 176),\n",
    "    label = c(0, 176), # top -log10P val\n",
    "    main = TRUE, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "## Annotate genome label\n",
    "annoXaxis(plot = genes_a, params = params_a, fontsize = 7,\n",
    "                   y = 4.2, at = c(24058000, 24058500, 24059000)\n",
    ")\n",
    "\n",
    "\n",
    "## going to add CTCF motif in inkscape\n",
    "#CTCF_motif <- readPNG(\"~/HRJ_monocytes/paper/CTCF_rs7146599.png\")\n",
    "\n",
    "#plotRaster(\n",
    "#    image = CTCF_motif,\n",
    "#    params = params_a,\n",
    "#    x = 3.2, y = 4, width = 5, height = 2, just = c(\"left\", \"top\")\n",
    "#)\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "### old code. including geno specific and maxatac ctcf.\n",
    "#################### ADD GENO SPECIFIC CHICAGO PLOTBAITS\n",
    "\n",
    "REF <- fread(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/14:24527672:G:A_homozygous_REF/14:24527672:G:A_homozygous_REFallReplicates_Chicago_135203.bg\")\n",
    "setnames(REF, \"V4\", \"score\")\n",
    "CHiC_signal_REF <- plotSignal(\n",
    "    data = REF, binSize = 50, binCap = F, range = c(0,52),\n",
    "    params = params_a, height = 0.8, y = 4.5,\n",
    "    linecolor = \"#0072B2\", fill = \"#56B4E9\"\n",
    ")\n",
    "annoYaxis(CHiC_signal_REF, at = NULL, label = T, main = TRUE, axisLine = T, params = params_a, y = 4.5, height = 0.5, fontsize = 7, fontcolor = \"deepskyblue2\")\n",
    "\n",
    "ALT <- fread(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/14:24527672:G:A_homozygous_ALT/14:24527672:G:A_homozygous_ALTallReplicates_Chicago_135203.bg\")\n",
    "setnames(ALT, \"V4\", \"score\")\n",
    "CHiC_signal_ALT <- plotSignal(\n",
    "    data = ALT, binSize = 50, binCap = F, range = c(0,52),\n",
    "    params = params_a, height = 0.8, y = 5.5,\n",
    "    linecolor = \"#b2182b\", fill = \"#D55E00\"\n",
    ")\n",
    "annoYaxis(CHiC_signal_ALT, at = NULL, label = T, main = TRUE, axisLine = T, params = params_a, y = 5.5, height = 0.5, fontsize = 7, fontcolor = \"firebrick2\")\n",
    "\n",
    "REFP <- fread(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/14:24527672:G:A_homozygous_REF/14:24527672:G:A_homozygous_REFallReplicates_Chicago_135211.bg\")\n",
    "setnames(REFP, \"V4\", \"score\")\n",
    "CHiC_signal_REFP <- plotSignal(\n",
    "    data = REFP, binSize = 50, binCap = F, range = c(0,52),\n",
    "    params = params_a, height = 0.8, y = 6.5,\n",
    "    linecolor = \"#0072B2\", fill = \"#56B4E9\"\n",
    ")\n",
    "annoYaxis(CHiC_signal_REFP, at = NULL, label = T, main = TRUE, axisLine = T, params = params_a, y = 6.5, height = 0.5, fontsize = 7, fontcolor = \"deepskyblue2\")\n",
    "\n",
    "## Bring in chic bw data from a bait for the SNP\n",
    "ALTP <- fread(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/14:24527672:G:A_homozygous_ALT/14:24527672:G:A_homozygous_ALTallReplicates_Chicago_135211.bg\")\n",
    "setnames(ALTP, \"V4\", \"score\")\n",
    "CHiC_signal_ALTP <- plotSignal(\n",
    "    data = ALTP, binSize = 50, binCap = F, range = c(0,52),\n",
    "    params = params_a, height = 0.8, y = 7.5,\n",
    "    linecolor = \"#b2182b\", fill = \"#D55E00\"\n",
    ")\n",
    "annoYaxis(CHiC_signal_ALTP, at = NULL, label = T, main = TRUE, axisLine = T, params = params_a, y = 7.5, height = 0.5, fontsize = 7, fontcolor = \"firebrick2\")\n",
    "\n",
    "# text for loops (5kb geno specific) \n",
    "plotText(\n",
    "    label = \"rs7146599 CHi-C\", fontsize = 7, fontcolor = \"black\",\n",
    "    rot = 90,\n",
    "    params = params_a,\n",
    "    y = 6.4, x = 0.3, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom REF \\nN reads\", fontsize = 7, fontcolor = \"#67a9cf\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 4.5, x = 0.6, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom ALT \\nN reads\", fontsize = 7, fontcolor = \"firebrick2\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 5.5, x = 0.6, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"PCK2 CHi-C\", fontsize = 7, fontcolor = \"black\",\n",
    "    rot = 90,\n",
    "    params = params_a,\n",
    "    y = 8.2, x = 0.3, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom REF \\nN reads\", fontsize = 7, fontcolor = \"#67a9cf\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 6.5, x = 0.6, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom ALT \\nN reads\", fontsize = 7, fontcolor = \"firebrick2\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 7.5, x = 0.6, height = 0.5,\n",
    ")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "##### Below: uncomment if you want to plot loops instead.\n",
    "## plot loops\n",
    "#plotPairsArches(\n",
    "#    data = both_eQTL_bpe,\n",
    "#    params = params_a,\n",
    "#    y = 4.5, height = 1.5,\n",
    "#    fill = colorby(\"allele\", palette = \n",
    "#                colorRampPalette(c(\"firebrick2\", \"deepskyblue2\"))), #darkorchid1\n",
    "#    linecolor = \"fill\", flip = TRUE\n",
    "#)\n",
    "# text for loops: eQTLs, all\n",
    "#plotText(\n",
    "#    label = \"eQTL loops\", fontsize = 7,\n",
    "#    rot = 90,\n",
    "#    params = params_a,\n",
    "#    y = 5.5, x = 0.4, height = 0.5,\n",
    "#)\n",
    "# text for loops: eQTLs, ref\n",
    "#plotText(\n",
    "#    label = \"Hom REF\", fontsize = 7, fontcolor = \"deepskyblue2\",\n",
    "#    rot = 0,\n",
    "#    params = params_a,\n",
    "#    y = 4.5, x = 1, height = 0.5,\n",
    "#)\n",
    "# text for loops: eQTLs, alt\n",
    "#plotText(\n",
    "#    label = \"Hom ALT\", fontsize = 7, fontcolor = \"firebrick2\",\n",
    "#    rot = 0,\n",
    "#    params = params_a,\n",
    "#    y = 5, x = 1, height = 0.5,\n",
    "#)\n",
    "\n",
    "## plot loops\n",
    "#plotPairsArches(\n",
    "#    data = both_PCK2_bpe,\n",
    "#    params = params_a,\n",
    "#    y = 6, height = 1.5,\n",
    "#    fill = colorby(\"allele\", palette = \n",
    "#                colorRampPalette(c(\"firebrick2\", \"darkblue\", \"deepskyblue2\"))), #darkorchid1\n",
    "#    linecolor = \"fill\", flip = TRUE\n",
    "#)\n",
    "# text for loops: PCK2 promoter, all\n",
    "#plotText(\n",
    "#    label = \"PCK2 loops\", fontsize = 7,\n",
    "#    rot =90,\n",
    "#    params = params_a,\n",
    "#    y = 7.3, x = 0.4, height = 0.5,\n",
    "#)\n",
    "# text for loops: PCK2 promoter, ref\n",
    "#plotText(\n",
    "#    label = \"Hom REF\", fontsize = 7, fontcolor = \"deepskyblue2\",\n",
    "#    rot = 0,\n",
    "#    params = params_a,\n",
    "#    y = 6, x = 1, height = 0.5,\n",
    "#)\n",
    "# text for loops: PCK2 promoter, alt\n",
    "#plotText(\n",
    "#    label = \"Hom ALT\", fontsize = 7, fontcolor = \"firebrick2\",\n",
    "#    rot = 0,\n",
    "#    params = params_a,\n",
    "#    y = 6.5, x = 1, height = 0.5,\n",
    "#)\n",
    "# text for loops: PCK2 promoter, both\n",
    "#plotText(\n",
    "#    label = \"REF/ALT\", fontsize = 7, fontcolor = \"darkblue\",\n",
    "#    rot = 0,\n",
    "#    params = params_a,\n",
    "#    y = 7, x = 1, height = 0.5,\n",
    "#)\n",
    "\n",
    "### plot CTCF from maxATAC\n",
    "atac_CTCF <- readBigwig(file = \"/rds/general/project/lms-spivakov-analysis/live/HRJ_monocytes/ATAC-seq/maxatac/output/CTCF/CTCF.averaged.chr14/CTCF.predict.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "plotCTCF <- plotSignal(\n",
    "    data = atac_CTCF, binCap = F, binSize = 1,\n",
    "    params = params_main, height = 0.5, y = 1.3,\n",
    "    linecolor = \"darkorange1\"\n",
    ")\n",
    "# text for CTCF (maxATAC)\n",
    "plotText(\n",
    "    label = \"CTCF \\nmaxATAC\", fontsize = 7, fontcolor = \"darkorange1\",\n",
    "    rot = 0,\n",
    "    params = params_text,\n",
    "    y = 1.3, x = 0.4, height = 0.5,\n",
    ") \n",
    "# range for CTCF (maxATAC)\n",
    "annoYaxis(\n",
    "    plotCTCF, at = c(0,1),\n",
    "    label = c(0, 1), # top maxATAC score\n",
    "    main = TRUE, axisLine = T, params = params_main, \n",
    "    y = 1.3, height = 0.5, fontsize = 7, fontcolor = \"darkorange1\"\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:plotgardener]",
   "language": "R",
   "name": "conda-env-plotgardener-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
