{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "58735"
      ],
      "text/latex": [
       "58735"
      ],
      "text/markdown": [
       "58735"
      ],
      "text/plain": [
       "[1] 58735"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "58749"
      ],
      "text/latex": [
       "58749"
      ],
      "text/markdown": [
       "58749"
      ],
      "text/plain": [
       "[1] 58749"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4742"
      ],
      "text/latex": [
       "4742"
      ],
      "text/markdown": [
       "4742"
      ],
      "text/plain": [
       "[1] 4742"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4762"
      ],
      "text/latex": [
       "4762"
      ],
      "text/markdown": [
       "4762"
      ],
      "text/plain": [
       "[1] 4762"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                   feature          hg38_location S025NM S025QG S025RE S025SC\n",
      "1:    ENSG00000213341_CHUK 10:100188298:100229619    802    551    579    506\n",
      "2: ENSG00000095485_CWF19L1 10:100232298:100267680    814    600    613    498\n",
      "3:  ENSG00000107593_PKD2L1 10:100288154:100330486      8      5      4      7\n",
      "4:    ENSG00000119906_SLF2 10:100912569:100965136    513    274    378    446\n",
      "5:  ENSG00000055950_MRPL43 10:100969458:100987515    233    183    240    297\n",
      "6:    ENSG00000107833_NPM3 10:101781325:101783413     46     35     55     41\n",
      "   S025TA S025U8 S025V6 S025W4 S025X2 S025Y0 S0260R S0262N S0263L S0264J S0265H\n",
      "1:    665    504    689    815    690    537    709    527    523    751    579\n",
      "2:    656    531    740   1049    542    716    616    531    720    758    576\n",
      "3:      5      9     18      6     13      1      6      7      9      8      9\n",
      "4:    574    495    647    612    497    469    485    444    358    323    240\n",
      "5:    302    244    367    394    392    367    372    199    327    193    211\n",
      "6:     41     55     24     64     49     49     58     30     47     19     44\n",
      "   S0267D S0268B S026A7 S026B5 S026C3 S026D1 S026FY S026GW S026HU S026JQ S026KO\n",
      "1:    787    750    780    921    507    469    553    907    845    654    736\n",
      "2:    847    708    899    912    544    455    687    601    765    773    755\n",
      "3:     14     11     11      3      2     22      9      2      6     17      0\n",
      "4:    403    337    519    387    399    353    392    493    542    564    284\n",
      "5:    261    216    257    225    213    209    271    362    306    298    235\n",
      "6:     39     35     54     32     40     33     38     54     31     55     36\n",
      "   S026MK S026NI S026S8 S026T6 S026XZ S026YX S0270N S0271L\n",
      "1:   1058    465    416   1265    754    579    859    545\n",
      "2:    939    578    508   1310    702    604    821    579\n",
      "3:      3      8     11     12      3     15      4      4\n",
      "4:    435    490    400    676    401    593    657    478\n",
      "5:    256    311    240    415    208    334    343    363\n",
      "6:     66     65     45     86     54     54     42     50\n",
      "                     feature avg_reads_noCorrection avg_reads_WASPCorrection\n",
      "   1:  ENSG00000088766_CRLS1              253.26471                161.20588\n",
      "   2: ENSG00000086475_SEPHS1              216.85294                140.32353\n",
      "   3: ENSG00000134571_MYBPC3              190.41176                120.35294\n",
      "   4:   ENSG00000198373_WWP2             1841.64706               1162.00000\n",
      "   5:   ENSG00000256043_CTSO              470.11765                302.05882\n",
      "  ---                                                                       \n",
      "4738: ENSG00000163374_YY1AP1             1615.91176               1058.32353\n",
      "4739: ENSG00000141542_RAB40B               24.76471                 15.97059\n",
      "4740:  ENSG00000182199_SHMT2              695.76471                426.50000\n",
      "4741:  ENSG00000276045_ORAI1              702.17647                416.85294\n",
      "4742:  ENSG00000110801_PSMD9              163.11765                103.67647\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.965232484818224"
      ],
      "text/latex": [
       "0.965232484818224"
      ],
      "text/markdown": [
       "0.965232484818224"
      ],
      "text/plain": [
       "[1] 0.9652325"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.998478943280205"
      ],
      "text/latex": [
       "0.998478943280205"
      ],
      "text/markdown": [
       "0.998478943280205"
      ],
      "text/plain": [
       "[1] 0.9984789"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3dB5wTZf7H8WeXZTsdpBdRD6Sp\n9I5g4SwUpSwqigh2mocFD1DKSfMQ1LPQVERFxfJXqigWlCYKIipNOrvARunssvX5T8tmkk2e\n/JLMLDOZ7+f1usxkMpMnwX1fkkkyYRwhFHHsYt8AhKIhQELIgAAJIQMCJIQMCJAQMiBAQsiA\nAAkhAwKksHqKvXaxbwKyVEZD2srUkq54cLd2/lntoi7sqDpzriy7Nrxr78TeVGduZGPUmQfY\nk8WucvvDjVNjUxs9/IfuBpWu1f+H8Mb07e3PpJNpdd8x5tpQlGQ8pNQ0qf7XVmRJXynnWfzv\n6kVFkOawy9jOsK59OuuvTLMS2VXqkrpsre9VLizFGt3xcFpNlrDcc4NurMNiXgrzPnlXbYgh\nV4OiK+MhXabOnH+I1cqXzzdj7QuVJUWQrk5dzkaFde2/swr58nQ5qxOjXNkOVjHf5ypPJZf6\nUJ4WjGfVsj03qGAOK32AMMQ5r0nx9jJAQsUzDRLPL8t+kc8/2p+9qixwQ1rPBhbUqJDttdmS\nrhVKV//nCnm2cF67MgkNnj4tzx9Iq5TUbP5R1tK9Xn2mPEEbxmaqT/JmsTt9r3INa6WuWzj0\niXT9DbrO87omd+ZVSVWuX6PMzm6ZmnDZMGlNPpZ9/r9qie6J1w0p2qCP/Dyxg/YaSbfts+yz\nTTeWT2j2XuT/hMiOmQeJX842yeeHHqtQ9oh83g3pbvYVf5It1G81l1V5cPyQijFvS/MDWZ3h\n/27Lmp7h/ERtdu1/Hqv6COvoXnEEG6tcdc1jLE2e+Sd7z/cq17F/FPi9QQ+xSdpcYQ/W4JG7\nUtlb0gPVTewfYybfxGoc5HwyezJlwH3uif6GeDZYdi9rO+tDFZJ+2ylsRKVnPpxRjn1t3L8l\nslHmQdodW+qUfP5eiUlveYEG6a/EeoV8J2un36op+1M6PVy2LecfsKtPSn+5w9gTnD/D+kmL\nj1ZlXdwrrmbNpdM97B7eqKLEJTup1Anfq8xtyHpt93eDrmULtLmFrHse5zuTk89Kt639BWnJ\nePnF1xRW7jteNNHfEN0GS5Sndgok/bZTWexP0vzLbJBR/5LIVpkFKfOTy9kjyvlBvLAz+5gX\nQZrBJkqn7dk23Va1Y47JkxwuPwGTdxHwk6WrcH4VWyfPP+uBlFNGXvNl9g4fyTZwvop1Ln6V\nOxoyVueuV371vkGFr7OU49q1dGWyFD5r9F7ega2UZ8/El86SMLSR57WJ/oboNtBB8t62pzz/\nM2sf6b8gsmVm7f6WGpTFVUh8Z0L1U25IhZfFSs+E+HyFmbsH2T/e0PZEpLITyvQatr8gnimv\n+Vd6IEmvUaTnY7fGHOcr5N3qj7EZfq4y/5O7akrj1/7PBV601+6meizuXfeVpDL3voTCRHZK\nmWnKNksYRsqz2kR3Q3Qb6CD5bPtveXY3uzqCfzxk30za/d2RXblfOz9IOp3IHnRDWslulJef\nTSkr/W3md5Fbw7MGl2Ks8ZPS87ssD8T1p1mich2bdZDelF4b5aRcw/n5BOlxoxH7w/cqtfa+\n3T+Ftc4pkl2qxh0/uy/LYgnu2TPu2W7So8tU9pw8q070N0S3gQ6Sz7bPy7N73LvlkcMy6ald\nVn31aY8GKadRzFoNUs+iv8+5nOcpM4ulxemv31aexS/h2SzmWa19p7S/1I06SJmxFQrWKO/G\nXhd74jCrz32v0tOxJvLzL93eD3dZLN49e8Y925WtcmNQJ/obotvAC1LxbQHJqZn1Gmklq3NG\nPT9InvwQ0/DCdTKkQ6XKD1Lqy1r4bJrzeqkKubwcc7+S4flx7Lw8/UQHibdhv/ybfcPlN2c/\nfYON4MWv8ox71VfZfX4h8TLaszbpqV2yNtuY/ewNSX9DdBvon9r52RaQnJppe+36sYfV84OU\nsw+yZ3rIkMayx7Q1G0qvLNwdyFAm10vP1G5kHynzf0v/u1Ld2TBKD+k/7KVOqfJeiV/YiEFs\nNS92lR1jVmhnJ8kvmvxB6srkj/nwKdetlx4ml8qzJ+KSLvhA0t8Q3Qa6nQ1+tgUkp2YapPSy\nMd9xD6RT1eOvliDlVmPaB4b4856PCGxh3WQaZ2uX+ot/wJqelebXxqVx/jjrK83uKq+H9Avr\nmajsIOPVml9WRtrM9yonsarKO62FK5Q3dfxBWsiaSw91+ysknuJvsTbyLomRbCj3gaS/IboN\nlir78hVIfrYFJKdm3vtIL7LLszyQpP8nZxKkDzzvrWbGp5xyz9/F6j/6zKN12Whp/g5W51/P\n9ilddiPnGVVYmyfuLTNMD4nXjmWvKDN3x7LbOS92lXl3MFbv5jv/WYfFTOL+IRXeyuo8dHcZ\neQdgYQ/WcNzELuwff/lC0t8Q3QZ/xpS+7yEVkp9tAcmpmQepoIXsogiSvEvgKL9W94mGfuxl\n92zBq+0via91w6fK/Lx2ZRIvvX+XPL+zV/nExnO2eUF6mCnv3nL+DmNvcO7nKr+464rk2DJN\nH97ifYN05f63WWJyJ+XpWt6LzZMTGj59kheDpL8hug2mVU5sqX1EqPi2gOTU7PB9pK1ekBCy\nYICEkAEBEkIGBEgIGRAgIWRAdoCEkOUDJIQMCJAQMiBAQsiAAAkhAwIkhAwIkBAyIEBCyIAA\nCSEDMhjSuVNBOp8XdJWIyz5r+hAX8k6bPkau6SOcyssxfYgzF0wf4mxeluljnD/vd3HRYQ2M\nhnTKFaRz/EywVSIuO+itiLgc/rfpYxSYPoKL55k+xIkc04c4xbNMH+PcOb+Liw7lAUjhBUjU\nAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjU\nAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjU\nAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjU\nAAmQBAESNUACJEGARA2QAEkQIFEDJEAS5BxI216euiKSIQAJkAQ5BtKsJMZY9/TwhwAkQBLk\nFEjfJDC5YeEPAUiAJMgpkB5RHLEK4Q8BSIAkyCmQ0lRIMUfDHgKQAEmQUyA9rUKqF/4QgARI\ngpwCaWdVBdJr4Q8BSIAkyCmQXN9cw1j5aREMAUiAJMgxkFyuPzaE/wLJBUiAJMxBkCIMkABJ\nECBRAyRAEgRI1AAJkAQBEjVAAiRBgEQNkABJECBRAyRAEgRI1AAprC4EK4/nBl0n0vLNH6KA\n55g+RqHpI1zg5o+RU2D6ELk83/Qx8vL8LzcL0pmTQcri54OtEnE5Z00fIpefMn2MAtNHOMnz\nTR/iTK7pQ5zl2aaPkZXld/FpsyDhqZ1x4akdsWh8agdIxgVIxADJpACJGiBRAyRzAiRqgARI\nggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARI\nggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARI\nggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARI\nggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARI\nggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqFoL0\nW0b4YwCSOQESNatAOj6+PCvdb0e4YwCSOQESNatAGsvk2h4LcwxAMidAomYRSAcTFUhsUZhj\nAJI5ARI1i0D6XnXExoU5BiCZEyBRswik7RqkmWGOAUjmBEjULALJ1VFxVHZ7mGMAkjkBEjWr\nQNp6meQodWG4YwCSOQESNatAcqXPG/3f38IeA5DMCZCoWQZSZAGSOQESNUACJEGARA2QAEkQ\nIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQ\nIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQ\nIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQ\nIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQ\nIFEDJEASBEjUAEnt0MQ70sbs4PzszAF9JxwPPAUkwwMkYnaAlHvPC0cyZqVl8ckj9xye/GhB\nwCkgGR4gEbMDpFOfZHF+pMc+V4+90qNP7y2BpoBkfIBEzA6Q5M68+nDu+j6F0tzwDwJNAcn4\nAImYPSAV3NZjzN981b3y/Li5gabSycG3pPadC1IOvxBslYjLyzZ9iHx+3vQxCk0f4RwvMH2I\nrHzTh8jmuaaPkZPjd/H5EB6RDv36n4fOrRosz46dG2gqnXzdQmpT0KtDKHoq2jlA2v2df+ey\njcpTuGFLAk2lk783SWWcClIWPx9slYjLOWf6EHn8tOljFJo+wimeb/oQZ/NMH+Icv2D6GNnZ\nfhefoUL65f5szgvvWnqi527pBVCv7YGm7vXxGsm48BqJmB1eI527Z9qho3P7pPPpw/ccemZ0\nYcApIBkeIBGzAyR+YOKdaU9u4/z87LR+U04EngKS4QESMVtACjFAMi5AIgZIJgVI1ACJGiCZ\nEyBRAyRAEgRI1AAJkAQBEjVAAiRBgEQNkABJECBRAyRAEgRI1AAJkAQBEjVAAiRBgEQNkABJ\nECBRAyRAEgRIXh1+5qZbp2X4vQiQAEkQIOk70IBJtfIrCZAASRAg6XuIKY33dxkgAZIgQNLX\nUIXU0d9lgARIggBJ32UqpDb+LgMkQBIESPoGqJBG+LsMkABJECDp+62y7KjePn+XARIgCQIk\nr7YNvLzh0F1+LwIkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIk\nQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIk\nQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEOQHSb8u2\nGjAEIAGSoOiH9Gdvxli37REPAUiAJCj6IfVUj0J8LNIhAAmQBEU9pM3qUYjZZ5EOAUiAJCjq\nIX2kQXop0iEACZAERT2k7zVIH0Y6BCABkqCoh+RqrzhqkB7pEIAESIKiH9K25rKj7yMeApAA\nSVD0Q3JlLn/5U/+/r6x15Kk68Vf+LzPIEIAESIIcAClofZUnf1OCrAVIgCQIkFwr1b0RiQfE\nqwESIAkCJNd0bb/eF+LVAAmQBAGS62UNUpD9EYAESIIAyfVrsuLo8uPi1QAJkAQBksv1v3jJ\nUfmvgqwFSIAkCJCk1o1KG7sz2EqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIka\nIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIka\nIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIka\nIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIka\nIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkhhdf5skC7w7GCrRFxululD5PFz\npo9RaPoIZ3mB6UOczzd9iCyeY/oYFy74XXzOLEhZwcrlOUHXibT8C+YPwbNNH6PQ9BGyeIHp\nQ1zIN38Inmf6GLm5/pebBQlP7YwLT+2IReNTO0AyLkAiBkgmBUjUAIkaIJkTIFEDJEASBEjU\nAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjU\nAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjU\nAAmQBAESNUdCOta/UgxTAiRxgETNkZD6x91w7xAlQBIHSNQcCany4nABAZLxARIx60FKPgBI\ntACJmiMhdV4KSLQAiZojIf3S+jtAIgVI1BwJqWNdllxXCZDEARI1R0Lq3O06LUASB0jUHAkp\n8gDJuACJmBUh/bV07vxVZwApWIBEzZGQCkaVlj/WkDIDkIIESNQcCel5dvv8FcvndGcLAUkc\nIFFzJKRGD6rTtOaAJA6QqDkSUsJqdfpZEiCJAyRqjoSU8rE6XZIKSOIAiZojIXXsnCNPsjp0\nBiRxgETNkZCWx9R5YNK4QZVKfQlI4gCJmiMh8U8byru/m60I1xEgGRggEbMgJM7Tf9x8LGxG\ngGRkgETMkpAiDJCMC5CIWQxSgym8QVGAJA6QqDkPUptZvE1RgCQOkKg5D5IhAZJxARIx60Fq\n8Yc6XYKPCAUJkKg5EhLbrEzyJiYDkjhAomZnSIc8s6FAYp7wiBQkQKJmW0iZ/63Nyg35UzsX\nCqRfXmS9lKNDDn3mMCCJAyRqtoU0RXlI6XpcPRfaU7vuu8MFBEjGB0jETIF0JEV9crZYPRvi\nXruMl6STzIkZgBQkQKJmV0jrtFc549SzoUHaWS1eOj3Aqu8BJHGARM2ukLZpkKarZ0OD1Pvy\nH+XJH5f3ASRxloD08YQXtkQ2BCAFrpXiKHmrei40SJXeUKdzqgCSOAtAOtxF+u+cODOiIQAp\ncJtrSv++Ca9p50KDlPiuOl2A95GCZAFI9yv/j5kw685u928OdwhAEnRw5v3jN7vPhAap3Y0F\n8uRcq/aAJM4CkMp63vZL+DzMIQCJWmiQVsQ2GDZ57KCKsSsBSdzFh5QRo3sDvfax8IYAJGoh\n7v5e3kz+z3LV8nAdAZKBBXlEqquDxNaGNwQgUQv509+ZW7a4wmYESEYWBNLrekhfhzcEIFEL\nFVLW9+9L/7qAFCwLQHLNrCg9eVAdlU8PbwhAohYipBnSK9gN/N+D8wFJnBUguTK3/Om6T4E0\nJ8whAIlaaJDmsZ6vS5AWxk0FJHGWgCR37L+tanb9ONwhAIlaaJCueohnS5D4040ASZxlIEUW\nIFEL8djfX6qQVuHY30ECJGqOhJT6uQppYVlAEgdI1BwJqVu3CzKkk01uBCRxgETNkZC+i6v/\nMBs8qFzpHwBJHCBRcyQkvqa5vDO19bfhOgIkAwMkYhaExLlry9YTPPwAybgAiZjlIOW0XBqB\nIUAyOEAiZjlIvMYsQKIFSNQcCWlp408i+JwdIBkbIBGzHqTOzVhc9bpygCTuIkA6uj3T8CEA\niVpokNp3uU4LkMSVOKQ/745nqY+H+SHvgAESNfwahTmVNKTMG5VPeT9i8BCARA177czJPEif\nPzFKO7inHtIy9XtHsX8YOxggUcNeO3MyDdI9spcbjsqzekgvaN+EDfcoJwECJGrYa2dOZkF6\nTfUyVp7XQ3ojsmMzBAqQqGGvnTmZBemfqpcr5Xk9pL2XKMubGrzjDpCoYa+dOZkFqb0KqZY8\n77XX7pMK8lG31hs8HCBRw147czIL0hAV0o3yvPf7SLtmPf7aYaOHAyRqoUL6a+nc+avOAFKw\nzIL0a0XZUeK38jw+2UDMepAKRpWW/zumzACkIJm21+67TqVLXbNUmQUkYtaD9Dy7ff6K5XO6\ns4WAJM7EN2TTj2gzgETMepAaPahO0/BjzEHCh1apORJSwmp1+hmOIhQkQKLmSEgpH6vTJamA\nJA6QqDkSUsfOOfIkq0NnQBIHSNQcCWl5TJ0HJo0bVKnUl4AkDpCoORIS/7ShvPu72YpwHQGS\ngQESMQtC4jz9x83H9Of//u/AtDG7OD87c0DfCccDTwHJ8ACJmOUgHc1UJhv0x+N6bMzejBkD\ns/nkkXsOT360IOAUkAwPkIhZDdLScurXkRpU3Vy07My0I5xn9tjl6rFXevTpvSXQFJCMD5CI\nWQzS7pQ6Xykz62pWz/R6urejp2t9n0JpZvgHgaayuT+kjp8MUhY/H2yViMs5a/oQufyU6WMU\nmD7CSZ5v+hCnc00f4izPNn2MrCy/i0/7gTQiaZc292vcBL2jM4/M46vulefGzQ00lU6+biG1\niSPknIpe0+ggNbizaLZPM926hx94tZCvGizPjp0baCqd7HpOand2kHJ5brBVIi4/x/QhCvgF\n08coNH2EbF5g+hAXzB8ih+eZPkZegCH8QEqZVjQ7uYzH0S93LpNONypP4YYtCTR1r43XSMaF\n10jELPYaKdnzw7HPli+a/f3On+XJiZ67JSW9tgeaApLxARIxi0Fq1LtotuvV7rmc+9+X18/m\n04fvOfTM6MKAU0AyPEAiZjFIj5dy7yn4nE0uemLXQ2kZPz87rd8UabNAU0AyPEAiZjFI6RUq\nvpsvTbNmJdZ08TADJOMCJGIWg8S/q8QqduvVKZXV2uyHCCDpAyRqDoTEjz55ZSxLumrSybAd\nAZKBARIxy0GSKjwfPiJAMjZAImY5SOc3/hDJ78cCkrEBEjGrQZqZwlipR3IAiZDRkHY91LzN\n4we8lwESMYtB+oBdOnpcJzYCkAgZDGlndfkLlY28j6QKSMQsBqn9peek0/tKnwKk4BkMaaB6\npOKnvBYCEjGLQUodL59uYj8AUvAMhlRfhdTJayEgEbMYJDZHPs1gqwApeAZDukyF1NlrISAR\nsxqkefLpUbYSkIJnMKRBut8XKwqQiAGSSdkP0u7aytGbvH+0HJCIWQ3S0xuklrNZ8gSQxBm9\n+3vPqHadx/r8/BEgEbMaJH2AJA5vyFJzHqRn9QGSOECi5jxIhgRIxgVIxCwHSTkkSsF37/wK\nSMECJGoOhLS4fhbn59pKr5BGAlKQAIma8yCtjK12mPMn2JC3b2GfApI4QKLmPEg31peW5lfq\nVMjzm9wKSOIAiZrzIFV+TjrZwOZLpxOqA5I4QKLmPEixi6WT6Wy/dDqvNCCJAyRqzoOU+KZ0\ncnMtefbFCoAkDpCoOQ/SlY9KDpIGybNDGwGSOECi5jxIo5MX7xnA5F+P3VU27P3fgGRcgETM\nYpAy6zLG7pBmPq6QuhOQxAESNedB4n/PfFL5VYlFV6wJ1xEgGRggEbMaJHcXCvk5QBIHSNQc\nDIlvur+Mn6WApAuQqDkW0t+zmzLWCZDEARI1Z0Iq/HJAAqsxZne4jgDJwACJmOUgHZ58KUu4\nVdkDDkjiAIma8yB9cnMp1mz2Xy5AIgRI1JwHiVUYLf9cLCBRAiRqzoOUwppNSwckWhaGdOyF\nnjePP0hbF5CohQDp9KvXsFI3f5QBSISsC+lYF/koUPX/JK0MSNRC22u3+YEyrDxbDEhBsy6k\naerx1O4lrQxI1EJ9H+nsvFaMtZt/FpDEWRdSdxVSbdLKgEQtjE82/PJIOZYCSOKsC6mbCqk6\naWVAohYCpKNFc+ffbAdI4oJAOjarT+9p6aI1CIUH6SkVUk/SyoBELQRIMc3H/pAfLiBA0ne0\nvXJM/MOCVQiFB+ngFfLY5beQVgYkaiFA6l2OsQppbx0DJEJiSBPUR4VRkY0R5u7v3Q80rJf2\nM21dQKIWymukvPUTO8axmObj1oX/wARIch1VSI0iGwNvyBKzGiS5s0tHNpYfmABJnBhSKxXS\n5ZGNAUjErAhJbv/IVPysS5DEkB5SIQ2IbAxAImZFSLlrx7cvzareAUjixJB215QdVdwe2RiA\nRMxykHa81KMMS71p5rbCcB0Bktr2u+rW7kfbcRY4QCJmMUiDa7H4DuO/yw0bESAZGyARsxgk\nxtq/kxmRIkAyNEAiZjFIy4b/g8U0G7Us7M/ZAZKxARIxi0GSOjivf2VWutPEdXmAJA6QqDkS\nklThTzN6VGFlAUkcIFFzKiRe+Osrd+B9pGABEjVnQjr69l1VGbvk7vcASRwgUXMepOzVjzdj\nLLbtxM3hv40ESAYGSMQsBimJsSoD3/s7fESAZGyARMxikNpO/NHroei1Ezz0AMm4AImYxSD5\nxrYDUsAAiRogAZIgQKIGSIAkCJCoARIgCQIkaoAESIIAiRogAZIgQKIGSIAkCJCoARIgCfID\nad/kgaO+NXIMQCIGSCZ1USBtrsoYi59q4BiARAyQTOqiQGqjHDgoYZ1xYwASMQtCOp/BefZb\nz++S51eG81VZx0LaqR6Biz1r3BiARMx6kHZcMpXntWMs8acwDDkb0hYN0hPGjQFIxKwH6fZm\n+/g7bO7edrcBkrhikI5WVCEtMm4MQCJmPUiXvMt5v2acv1sTkMQVf430quKo63HjxgAkYtaD\nFP81L6j4JOerEwBJnJ/d3wsal672yD4DxwAkYtaDVHs+/4p9w/mCGoAkzu8bspnGjgFIxKwH\naUi1p+rWy+fHm/UDJHH4ZAM1R0LKaMuq/MR5Wvlw3kICJIMDJGLWg8T5afnQkJuP8nADJOMC\nJGJWhBRpgGRcgETMYpBSdJUGJHGARM15kNLS0gY0KN2uT++rY1qNBCRxgETNeZCkljTJkCc7\n6nwOSOIAiZojITV5X53+7ypAEgdI1BwJKf5LdfohPtkQJECi5khINe5QjrVa0LM6IIkDJGqO\nhPQsazJ88uRHG7IxgCQOkKg5ElLh9OryR5grj88HJHGARM2RkCRKBzdt3FtwbjcgiQMkag6F\npPYVPv0dJECi5kxIy+7q1KFDh7ZlKgOSOECi5khIi1lcLVYjhXVdBkjiAImaIyG1uPkMT9iR\n99oN4RxASOn0X0E6x88GWyXiLgS9FREnQTJ9jALTR/iL55k+xMlc04c4zbNMH+PcOb+LT/qH\nVEZ6JErYwfnjj4QLKScvSNL/oMQAACAASURBVAU8P9gqEVdg/hCF3PQh8kpiiELTh8gvgSF4\ngeljFPgfItc/pHLLJUw/cr62WriQ8NTOuPDUjpj1ntq16ZvLG8zg/NMUQBIHSNQcCek91p2P\nSBgxsUZ7QBIHSNQcCYm/P4OfuIGx2psBSVxxSHsevLzWrQYe+RuQyFkQklL6jtziCwHJq2KQ\njlwpf7gq2UhJgETMipCyvn9f+tcN25EDIB1/Y8TYNcUhTVCPWNzVwJEAiZgFIc0oy9gG/u/B\n+NBqoA62lL08WQzSrSqkMgYOBUjErAdpHuv5ugRpYdxUQArQEBXMal9IfdTllQwcCpCIWQ/S\nVQ/xbAkSf7oRIAVI+9GJ+30hqcfQZwMMHAqQiFkPUsKXKqRVSYAUoHgVTH9fSJnKc7u6uwwc\nCpCIWQ9S6ucqpIVlASlATVVIk4rt/s58vf+t4w8aORQgEbMepG7dLsiQTja5EZACtERxVCcT\nb8gScySk7+LqP8wGDypX+gdACtTbDVh89834ZAM1R0Lia5rL/3/b+ttwHTkAkst1IAMfEaLn\nTEicu7ZsPcHDzwmQ5IpB2nXU8DEAiZjlIOW0XBqBIQdDypxamcX3/cPgMQCJmOUg8RqzAInU\njikPjNvqOTtV2QPRMsPYQQCJmPUgLW38SQSfs3MOpBXlJDdJ77rPppdV94nPM3YUQCJmPUid\nm7G46nXlAEnXvrdnr9Cfz6ituKm4Rzv/s+qIjTZ2WEAiZj1I7btcpwVInj6sIiHpss+zYKUG\n5y3t/O4Y9fxkQ4cFJGrWg+TVa+HsvYtCSL9VUJSkeZZ8pEF6xb3gOuVs8s9GDgtI5CwOiYXz\n0+ZRCEndk8Di9hct+aOUumite8H2K+TXTK8bOaoLkMgBkkkZC+lf2uOP7vFmtLLgbs+CjDmP\nTd1afNPIAiRigGRSxkJ6SXWUdMSz6NiUS2NqPp1u5Ch+AiRigGRSxkI6cJkC6XGvhTl5+IgQ\nLUACJK31rRmLH+b9ESB81o4aIAFSUVu/3K87t2tog/qDDgESLUACJP/tryc/06vyu5ljKAES\nMUAyKXMhjdK+bG7mGEqARAyQTMpcSB1USJeaOYYSIBEDJJMyF9K1KqQrzBxDCZCIWRzSynB+\nbcwBkCaqkB4ycwwlQCJmPUhxKVqpNW7+OgxGjoB0tK3sqNEhM8dQAiRi1oM0vDVrdNvtTVjH\nO68vG7sSkPx3dEaPm6ZlYfe3/759YugLug+BOBPS6rrr5cn6K37ip1p3BKTA4Q3ZAE1WjpO5\n3bPAkZCuWaROF3Th/L2wfrUPkIzLjpC+TVBeQN7gWeJISPGr1enKZM7/LxWQAgdI/ntK3RMT\n6znkrCMh1exfqEzvqcLzbmoJSIEDJP89qn3nZEfREkdCeoY1eWzG80+2ZsP5bWwxIAUOkPyn\n/ShHtcyiJY6EVDDpEvmfodxjOXz2e+E4AiQDsyOkjBYKpPmeJY6ElM8L92/auCfs3+tzGqQd\n86YvN28MO0Jy7RxYLrbBAt0CR0Kq/q+t4RtyHqQF8uHtupn2zqwtIUl5f3PYkZDaxrDG0w45\nG9KS29vdE/y3yWVIPyYrT2MGRzCYMLtC8s6RkPjB51uxmGsXnHYupHGyjfgPg60mQ3pafWGd\naPzh89UAiZgFIUntn96SJfZ3KqRN6k9bVgt2GG8Z0lBtV+/usEcTB0jErAmJ88JPLwv8mfAo\nh/SChuMb/cJ9Hy30PdaWDGmKumqlTBe1L3s3vfEN8tqARMySkPK/ebQGq3i/UyE9r0H6Srfs\nzYrSk737vbnIkP5UDwE+hXzli5T1R1FXByRi1oOUt/qBS1hy2me54TqyPaS1qqMKuj1P65KU\nRZO81lP22q1tyVjyWPIDUkZl9cq/J64PSMSsB6kii7tp0bmwFUUBJNcjvm8ouh5W//xre62m\nvSG7fW0IR4n8Vnu0m05cH5CIWQ9Sx1dcyvTcbsdCynyl02XdP9cv6a3++cd5rRbOJxu+0yDN\nIK4PSMSsB8ndVzUcC6l4w9Q//3peC8OBlFFFvabg71KpARIxC0JadlenDh06tC1TGZCK2pyq\n/Pk/77UwrM/avadcEfnnyACJmPUgLWZxtViNFNZ1GSB5+rAmY/Heh/4O80OrX/e7+qaF5LUB\niZj1ILW4+QxP2JH32g3hHEAoaiG50r/8ZJfPInz6m5ojIZWRHokSdnD++COAJA6QqDkSUrnl\nEqYfOV9bDZDEARI1R0Jq0zeXN5jB+adhHfgkCiEF/DwqIFFzJKT3WHc+ImHExBrtAUlqUeO4\nCgN3+r0IkKg5EhJ/fwY/cQNjtTcDkvujcdf4/ewCIFFzJiSl9B3O/aydvjrq+6ez/V0GSNQc\nDCmSogjSPu0TPUNdrkPrD/tcCEjUAMnpkNLjVEij9t4Zy0oN2u91ISBRAySnQ3LdokJafasy\nud3rMkCiBkiOh/R7PRnQWPe3H7w+awpI1ADJ8ZBcR2YOfuwr1xsaJK8PyQESNUACJLX/0yCt\n0C8EJGqABEhq6Zcrjhp6fcoBkKgBEiBpfSsf5KSu95EWAIkaIAGSu8NvTnzziPciQKIGSIAk\nCJCoAZKTIb1/5w2jfhetAEjUAMnBkEbIexfK/iBYA5CoAZLDIK24q9uQDersKnV/d3PB2oBE\nDZCcBWm6TEf7FQrtF4WZ74EadAESNUByFKRtiQqdKsqXjx7TIP0WeH1AogZIjoKk/YgwWyWf\nUY8/x+oIjuoNSNQAyVGQXtYgLVPO3aTMi35tDJCoAZKjIK1XHSUfUM6lj21a7boVovUBiRog\nOQqSdoTvWYFX2Lhote7TdoBEDZCcBen4S61rdF6sW3Bwzc+610j75W/5Xflt0XlAogZIzoLk\nU+bTiYw18fxuX5rygFW36PvmgEQNkBwNaaoCp6r7oHa7YtXXUK+5LwckaoDkaEhVVTjjtbPu\n3wgb574ckKgBkmMgHZz/7Bs+35I4pMEZqJ13PyLNca8ASNQAySmQ1tSU33z1/t5eZhnm/aNg\nA5Szlx5wnwckaoDkEEjpl/r5JrnrUfVtpY3u8wd6Smcbry26HJCoAZJDIH3s79gmrnT5YHbl\n39At+emdNcc85wCJGiA5BNJ8DdI7Psu/nv3m7sBbARI1QIpKSJ8/PmKR92dRv9EgbQywhf8A\niRogRSOkwTKZazO8lqlHJO4f2hDekBZfHV/5XsG3l8ILkIjZA9KR0b3kydmZA/pOOB54agtI\n89QHH+9fKN87sBQrfd+BAJsEyAvSYvULtRmBVw8rQCJmC0hr75mlQJo8cs/hyY8WBJzaApL6\n4MOu8Fl8ZIPfHxMT5QVJ3e/HXgz1SoIESMRsAWlN5gYZkqvHXunRp/eWQFN7QOqi/sVXvr9T\n77ciG0IPab/2KmtoZFdZLEAiZgtInCuQ1vcplE6HfxBoag9ID6l/8aXkkwd9Lzw+s2397suJ\nQ+ghuX9IaUQYN1UUIBGzE6RV98qz4+YGmsrr9ZTamh+kAl4QbJWIKww0xJFL5D947eM+3/tc\n+ICy9FPiEFx3pqd6hevCuKmiePBVIh6i0PQhCkpgiBK4G4X+h8gLGdJgeXbs3EBT6eT7rlI/\nFQaLB13DxH7rHl/qau2Z2LPeF21Ql1bNJV0R198N9cMRE4y7mdoYRl+hnyFKYowSGOKi/VHl\nhwppo/IUbtiSQFP3ytZ+aieVkf67Bulf3hdM0hav87+dT967vw9Pv3vEqtBvZ5Dw1I6YnZ7a\nnei5W1LSa3ugqW0gSdVTxfgc2+Q5DdIm0hB4Q5YaIKmdcK3uJf1p8unD9xx6ZnRhwKmNIH2q\ngOnts3St6qjucdIQgEQNkNSG9JD7jJ+fndZvirRZoKltIB3dsPrzW+q3mnLU94InZEcJS2lD\nABI1QAori0M6/MlTdRhLHuf3wvdv73Dfj8QhAIkaIEUhpKW1tBdC/450CECiBkjRB2lnFc0R\ni3nN/xrkAIkaIEUfpBmsqJRfIxsCkKgBUvRB+pcHEns5siEAiRogRR+kWTpI0yIbApCoAVL0\nQdpb2wMp0CHy960nfaUCkKgBUvRBcn3bxO2oj/8VfruVsfhhBEqARA2QohCS6/gP//f7Sw3i\naj9x2O/FR1sryh4OPgQgUQOkaIS0buITiwSfAXpf+75S8MMvABI1QIpCSBPiJSatAh+eYYr2\nxO+LoEMAEjVAij5IK1UmgwJu+LoGaUvQIQCJGiBFH6QHVSapATfco3yDlnUMPgQgUQOk6IM0\nQPt8UODjZv2f/CGiRtuCDwFI1AAp+iA9q0K6XLDp3nmT3jsmuNwdIFEDpOiDtK+uAultA4YA\nJGqAFH2QXD9eF8fqzAlwYUgBEjVAikJILle6QUfoBiRqgBSVkIwKkKgBUlRA2vnMwKd+MmEI\nQKIGSNEAaXV5xlj8XOOHACRqgBQFkI6pvxMR85LhQwASNUCyOaSML97e4P41vtJfGT0EIFED\nJHtD+raBBKid++tHtwff6PA25Scxv53xwgbCEIBEDZBsDWm/+t6r9sMTrHmwTX65KZaVn5SZ\nKf82Zvzo4EMAEjVAsjWkOdrH6jRINwfZIl396uxzM9XVFwYdApCoAZKtIT3DvPogyBavqKuV\nbaFOuwcdApCoAZKtIc3VM4qp+kCQDzSM0tbUjsTaMugQgEQNkGwN6WB974ekS/cLt5iorlVK\n+5HZAUGHACRqgGRrSBnjY70ljRJu8XOKstKtK5RJ8g9BhwAkaoBkZ0jrr2A+tRNvMk+W1HSX\n660ajF32cfAhAIkaINkY0l+NfR2xzkG22f7C02/L3+jL/PkXyhCARA2QbAzpC89+BvfMs8YO\nAUjUAMnGkN4tgpSqTVv7O07DofCHACRqgGRjSOs0PklP36bO3Fzc0a6BZdilYf8mBSBRAyQb\nQ5qfpL4uOubaXkmeuar40byPtVFWmR3mEIBEDZDsCSnj5fsf0w6Y2mO3dP63Ic1aPXmw+HoL\n1HUqyL/JnP4L5cBBXgESNUCyJaQ/r/TsqBsvWnG0ttLPrt13xLGkUaQfc/EESNQAyZaQBuj2\neN8gWnGCtldvV2Y3ZebB0MYBJGqAZEtIKTpIZfz/dovaxkRlnS6uz9WVY38PaRxAogZIdoR0\nLEYHib0vWvUF+Zcp6vzi0r45wT4PaSBAogZIdoTkKquHJN65vWHc/bOPuFxvaCt/H9I4gEQN\nkGwJqake0krKFnurKutelRnSOIBEDZBsCGn7f+J1jq6j2fi0orRuvY2hjQRI1ADJfpDe1O9q\nYH2pRyfe8/JT80Pc+w1I5ADJdpB+LaN3FPzr4pEESNQAyXaQntc7YlPMGkYJkKgBkt0gbWvm\nBSm03dmhBkjUAMlmkP6o4uWoe2h74UINkKgBkp0g7R95RZKXozZ7TRhFFyBRAyQbQdK+E6Hr\nPeMH8QqQqAGSjSC97Ouo5exhEzYbP44nQKIGSDaCNNjHUac60klC2F9/JQRI1ADJRpAe9BhK\nvaTK9ZvUb0YkfXlflfir3jV+OBcg0QMk+0B6R7fDrpx0PkP7DHht5XSR4eO5AIkeINkG0uoE\n/dO6xS7XLq/nebXN2BEOSNQAyTaQunu5qetynSnvtYT6mbtQAiRqgGQbSJd772n43pU9S5mp\nqZ4vFcHh6wIGSNQAya6QPnBln3qhFku+4yP1vPDQDeEGSNQAyS6Qjjb0hvSD8huy+4+7XGOV\nl0i/GjyeEiBRAySbQNpV19tRN+1XzeVWjxw4zYwndoBED5DsAenPit6OLt2hg2RegEQNkOwB\n6RZvR+z/XIBED5CoRTukN30c9ZcXAhI1QKIW3ZA+vNrrOHYsdrh8KG9AIgdI1KIa0rs+D0dJ\nL6jLAYkaIFGLZkhrUr0YXf2qe0c3IFEDJGpRDGmaz+NRTNEbRoBEDZCoRS+kjaV9IHkOdwJI\n1ACJWvRCmuLriLV0/1wYIFEDJGrRC2lcMUhsmnYRIFEDJGrRC+nD4pBu0i4CJGqARC16IWU2\nLwbJfZRiAaRt/x2zMOQfjPUTIFEDJItD2n+v/GZsUv240h3dpCZoFwWGNC9ZWq3JzshHByRq\ngGRhSJlzOtZTf9aoz5Qet7/4q3pYu2bun5QICOln9ecqbo78FgASNUCyMKTHvZ/SdT72SbfL\nmz9+wH1xQEiT1PVjIz8MKyBRAyTrQvrJ98XRZO/LA0J6TFt/S8Q3AZCoAZJ1Ib3uC6mL9+UB\nIb2qrl42I+KbAEjUAMm6kOb7QmrrfXlASOnqT8w+F/EtACRygGRdSN/6QhrmfXngvXbbbinF\nKkw24EB3gEQNkKwLqbuPo9p/ei47vEv8huyRbYYcLxKQqAGSZSHtjfViFHPj70UXfd8xltV9\nE59soAZI1KIR0jqfB6SqRQ9Iu6opC5YBEjFAohaFkHbV9n2JNMeVPqVp5aumHxutnr8GkIgB\nErWog7S5U5KvI/bconLKdGgP9XwiIBEDJGrRBumxYoqkZiRqM9qxuaoDEjFAohZdkLa38Oeo\nTX/33MPq5F+ARAyQqEUVpMxO/hzVa1PJPfvy0/Jpl1OARAyQqEUVpC+LK7r5Vd2vjCX+4vp+\n0pgP8cU+coBELVogpT/bvN4tLYs56pmp/02X/2orAxI1QKJW0pBy8oJUwPODreKn3B7+ntRJ\nvVg0F3v1xqIxwhkitAq56UPklcQQhaYPkV8CQ/AC08co8D9ErlmQzpwI0nl+LtgqfnqnmCDt\nYFxdtLOVm47JLFo752wYQ4RWLj9p+hgFpo9wguebPsTpXNOHOMOzTR8jK8vv4lNmQTLpqd3w\nYpDi1Em7GkVLrjvuXhtP7ajhqR21KHmNNKoYpOrqZPD7nkUvu9cGJGqARC1KIH3m66jqR8re\nuvJbXStvql9KXdbfvTYgUQMkalECyTXEi1GVoTtcHzVisa3XKBcmq0v7ulcGJGqARC1KIO3p\nqXeUpC7c5/552OvUxbPcawMSNUCiFiWQrvd6QKqsLDv40qjndyhzP5aRl7YvOvAjIFEDJGrR\nAcnnG0jKi6Ef5P115ZYol2+9u0nb8emuH3pffs3YI4BED5CoRQck78/Yxa2VFmU2Uh+c9njW\n+kb5EHjn44BEDpCoRQWkLV6OEl+Sl63Vzi3wrNZKXfIKIJEDJGpRAWm23lG1bcqypdrZmUVr\nHdUO5XAPIJEDJGrRAGlngh7SAG2h9ubRiqLVjmkfdrgPkMgBErUogPRJjN5RRfcvxY5QzuqP\nh99VXWMRIJEDJGr2h3TY+7diV7uXZ4wpz5KG7tOt+XMFeYW+2GtHD5Co2R/S016OYvSH7d5x\n3HvVHSO69HwtE5DoARI1+0Nq6g2JsgkgUQMkaraHNJ15Qzq+deq/Xk937R7Wqv2TBwNsA0jU\nAIma3SEN8nbEys+VD2t32dpa8pkrD/vfCJCoARI1m0Mq9tPl9dVdeNrX+Z70vxUgUQMkajaH\ndKePI5+jrLb3vxUgUQMkavaGtMLrLaT48pcwQDI0QKJma0gZlViAtPeWnvK/HSBRAyRqdoa0\n+/pAjthY5RcpGmFnQ4QBEjUbQ1qWqpNTVv/5huQpmXtGtG4/Bru/Iw2QqNkX0qEq+ocg/Yul\nqwI8EBUFSNQAiZp9IRXb8+3u5WPBNgUkaoBEzb6QXg/gqOLz76cH2RSQqAESNdtCyuwYcE8D\nu/Q78baARA2QqNkVUsbV/g2pr5UuPyLcGJCoARI1u0J63A+i2H5vT9VmPxRuDEjUAImaTSHt\n03+7PK7zDdWkScobLtcS9w4H4daARA2QqNkT0vHWOkfXfOVyHX1rzMzt0gWbtWUrhJsDEjVA\nomZPSHP1r4q8vgbbR1nW5XigLZUAiRogUbMlpCNX6CB11BaumvriFuk534AYxm7dId4ekKgB\nEjU7QjpUVb+LYa6y7Kh8FP2EqdLcni93BbsCQKIGSNRsCCm9ote+urrKwicpr43cARI1QKJm\nQ0gPeu/0Vn/DpbZ65h7SnQYkaoBEzYaQ4r0hXaIs1L4aewvpTgMSNUCiZjtIH9TwdsS6KIub\nqGd6TXsv2AftXIBED5Co2Q3ScubbP5Xlbyvz8vG+668NeqcBiRogUbMbpHbFIN2uXvBiRfeh\nT6ocCDYGIFEDJGp2g5RSDNIc7ZLjP7q/odQ8M8gYgEQNkKjZCtLvE1oWc9THc/Fi97J5QcYA\nJGqARM1GkNZeVUwRS5ilW2GDe+mQIGMAEjVAomYTSBnzH5/ge+yt8ow1/cJrrR7aJfcHGQOQ\nqAESNXtA+vUfxR+MSmf+7PuRur3aASLfCTIGIFEDJGr2gNS1uCNW033hu20qNZ6o/izSxjLy\nJX2DjQFI1ACJmi0g7fTjiDXTLnxFOaf9cuz2B9vf8r9gO+0AiRwgUbMFpI3+IN0tX/LHDwfK\nqWeXh3CnAYkaIFGzBaQjycUdxf3gcn3fyvP7E5NDuNOARA2QqNkB0h8P1y0Oqexvrj9r685P\nD+FOAxI1QKJmA0hbKhZnJDXI9ZzuXMKPIdxpQKIGSNRsAKm7X0fsGtdQ3bnnQrnTgEQNkKjZ\nAFKSf0gdXE9pD0Yjbh6yKqQ7DUjUAImaDSAl+of0nGuTug9icMh3GpCoARI1G0Aq9m5se/nk\nhuMu1zz5/deugX4FKXCARA2QqFkRUvr4hhVaLXIvOL4x1dtRy+OvDxr8hvKu6x9zpi4L404D\nEjVAomZFSP0UL8pRh49NrRdb65FYvaPE4N+ADRYgUQMkahaEtFoD89Iul+uJYi+Nmm6O/E4D\nEjVAomZBSNPcZip8uKOUr6NrjbjTgEQNkKhZENILRWoqLvB1VN+QOw1I1ACJmgUhbfP8ZMtA\nH0ex6w2504BEDZCoWRDSmalFcHz217GHjbnTgEQNkKhZEZJr9b2xzG8PGnOnAYkaIFGzJCSX\na7S6t8EX0mhj7jQgUQMkahaFdHR0Eou5ZYAvpK+NudOARA2QqFkUkkRp0wHXCh9H4wy604BE\nDZCoWRaS0qQYHaPGnxh1pwGJGiBRszake3SOehh2nwGJHCBRszSk9/TP62rtMexOAxI1QKJm\naUj6J3bBD0RMD5CoARI1K0P6t/euhoaG3WlAogZI1CwM6bjPPrsrDbvTgEQNkKhZGJLvpxsM\n+liDC5DoARI160L62MdR/X2G3WlAogZI1CwLabaXotYdH99v3J0GJGqARM2qkIZ7Pxx9bOid\nBiRqgETNopDe83bEvjH0TgMSNUCiZlFIV3o7anzc0DsNSNQAiZoFIe0f/c+7vB3VWWfsnQYk\naoBEzXqQ/vQ5Zv71T809YvCdBiRqgETNepCu9XaUnG74fQYkcoBEzWqQDj/ss5uhk/H3GZDI\nARI1q0G6x8cRu834+wxI5ACJmsUgbfV1xAaacKcBiRogUbMYpCXFIL1rwp0GJGqARM1ikD7x\ndRT6jx8RAiRqgETNYpBu9lI0YOiHJtxlQKIHSNSsBWmX13dibzDjDssBEjVAomYtSOu8HpA2\nmXGH5QCJGiBRMx3S2ZkD+k44ToS0Q++oqxn3VwmQqAESNdMhTR655/DkRwtokI7rntolbTPj\n/ioBEjVAomY2JFePvdKjUu8tNEjpHkcxkf/CZcAAiRogUTMb0vo+hdLp8A9okDy/jMTMuLPu\nAIkaIFEzG9Kqe+XTcXOlk13PSe3OFuV5ZndAuF6E5eeYee1KBfyC6WMUmj5CNi8wfYgL5g+R\nw/NMHyMvwBCGQRosn46VIX3dQmqTcO0iSKUjGxUha1S0cyBSSBuVp3bDlkgnZ/6QOn5SlPsI\nXDHCtSIu56y51y+Vy0+ZPkaB6SOc5PmmD3E61/QhzvJs08fIyvK7+LRRkE703C29MOq13X0+\nyKuTEniB5MJrJHp4jUTN9N3f04fvOfTM6EIipETFUVkT7qg+QKIGSNRMh3R+dlq/KZ6rC/on\nPCxukuF30jdAogZI1Kz1ESGXz+8jmRQgUQMkaoBkToBEDZAASRAgUQMkQBIESNQACZAEARI1\nQAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1\nQAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1\nQAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1\nQDKnH5/7rYRHNKUPnjt/sW+CEU1dcLFvgREdeO7ri30TSh7S4hYrS3hEUxrV4uTFvglG1Gbg\nxb4FRrS5xf8u9k0ApDADJAsFSPYNkCwUINk3QLJQjoSEUFQGSAgZECAhZECAhJABlSykszMH\n9J1wvESHNKK//zswbcwuzof3kOrnuRu+U4sX7Obb42782kNpmcX+a5QspMkj9xye/GhBiY5p\nQI+N2ZsxY2A2H7zU5XL97bkbvlOLF+zm2+Nu5Mofzfmt3yGL/dcoUUiuHnul/8PovaUkxzSg\nM9OOcJ7ZYxfvu1k5774bvtOLfDODFuTm2+VuyP373aB3p4RvUYlCWt+nUDod/kFJjmlUO3q6\ncnu8NGzwf9KL7obv9CLfwmAFu/k2uRty3w7JDXp3SvgmlSikVffKp+PmluSYBnXmkXn81N3T\nduyacM85993wnV7M20co2M23yd2QyhvyTfC7U8K3qWQhDZZPx9rgP5Vvhx94tVCdy+6/yn03\nfKcX68aFkuDm2+durBnsfgVknf8aJQppo/KoO2xJSY5pSL/cuaxo/pF33XfDd3qxbl1IBb75\n9rkbY98qmrXMf40ShXSi527pIbnX9pIc04h+v/NneXLg5VzOs/p+7b4bvtOLfTODFOzm2+Ru\nSPeh96/ccv81Snb39/Thew49M7qwRMeMvJz735d3uWafuXPW0SNT7ssuuhu+U2sX9Obb425I\n/7/W4zQn3J2SrWQhnZ+d1m/KieDrWatf3G8B7hmXNnDKMc/d8J1avGA33yZ3g6/tqSix1n8N\nfEQIIQMCJIQMCJAQnuleiwAABQdJREFUMiBAQsiAAAkhAwIkhAwIkBAyIEBCyIAAyUalpYSw\n8tEnGqekNhy5O+RR2jQIeRMESHYqFEg/VCzVa8rUAQmpy4KvW9RW+c9h1pRQbxcCJMuVI7gs\nBEjHKlfcJE931C53zO8Ifsd5CX8O4YZ/uRJpcZvyZVos5rxDpTz5bJvq+QUTaic0Xz08Tr9W\nuy7La7WUHkyuL5N49QL9ZrxwYq2EJksGSJAyhtZJqHr7Dv1mnTv+2jWlyoBMaXZFp9TExjML\nOX+Kvate+NXMTN1idYRi46xsl1Tl9t28O2OshfrUrmgL3ZUjQYBUEn3A+i5b+k+2jL/CVktn\nD8b8i/+H9f9iYZ3WXo8xXZo1nvU5/6Z0l89WP8Se123Gp7M7Vi9u1EBau221+WsWNblE/6sy\n3eq2/Gz/u6XSOP80ptNHX9zPRnN+ZcV8zwqexeoIvuMsj73+nQX1qx/d3Ytt/kOB5NnCc+VI\nFCCVRDNuzOX8dNxd3BX3gHT2eba1sGoT6f/u1zEvSNexT6TTFpdekE57lsvybFZYo7G0dnrp\nFH6aPSVduG9Kutdm30qn11fivGFtedPbSv9VGNtNt0LRYm0E33Ga/UNStyn+JT5E/nOQIem3\ncF85EgVIJVetTpzfdEkB560a8wPsMXlRI29IcdITPxd7OFvqdfa9Z7ODbLg81y6F51Wu+5Xv\nkaauS5JP743h6UxmyuezZedZL8/lnsXqCL7jZLJh2ppuSPot3FeOhAFSSXTysYZlExJYB84X\nsW/4fjaNb2Iz5Atu84ZUXTrZwrTe92y2iU2WL+8jrf1jPVax/3v5XpvVlU8lBD+ySfLcCjaP\nJ3XyXK5brIzgO84W9dq5B5J+i7pFFyBB+AcqiW6Mn7rxjx01JEhnkx/l02MO8Y3yaxPOb/eG\nVJfLu6AHb1ByeTbbqP6p95bXzv/6yaas/QXfzeS/9c1snDy3nM3nV6VmFV2uW6yu6jPOFjZW\nW9MNqdgWgBQs/AOVQOlshHSalyJB4mk1Cltey/ke9qR8SZPikP5mg4pttld98nW1e+3X2ULf\nzeS/9WPqpnPYKj6JzVIvXNdgo26xDpJnHJc6dyCzCFKxLQApWPgHKoF2sInS6WusrXT6GfuU\nLZAeV8q1kc5sZMUh8dbl5B8xWzg2z7NZQeUrpRdGO2JT+OY0+ajWe9XHM6/N5L/1ppXkx6Gb\nkk/zkzWTlspLt9eqekK3WAfJMw6/quI59UYOZXnqzgbfLQApWPgHKoFya9f8bO3j13cts+Yc\nz614WaJ87I6R7N5Vb1zR0Q+kb0s3efuLcaUH6zcbz3oteaVuqxSeUa7ZgtXvtUvYWWwz+W99\nWWyHjz6/g02Vzm6uybpN+E/fuOqb9Yv1kIrG4ctjWy2Y+49q6fwZNnGJAsl3C0AKFv6BSqLN\n7ZIveeDM0srVd3P+AOsnL7rwaOWkTj/emapfTf2j5d/fUCbxyqn5+s3yx1SLb/rpyNKcb+tT\nLb56r03FN1P+1ld1Skm45g1lqWvS1eXKNH1WeSe1aLEekmccvqJdUuXbdnF++Jqka9Q3ZH22\nAKRg4R/oonZ9jYt9C5AxAdJFavbt0suRU2Vvvti3AxkTIF2k3mW3fra4dezXF/t2IGMCpIvV\nouYpyR1W8pWsqNdC2DrMzZBZAdJF7uz2okI5OmiYmyGzAiSEDAiQEDIgQELIgAAJIQMCJIQM\nCJAQMiBAQsiA/h8WvfYvCnQfpwAAAABJRU5ErkJggg==",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#### check the results of WASP correction for RNA-seq.\n",
    "\n",
    "library(data.table)\n",
    "\n",
    "setwd(\"~/spivakov/HRJ_monocytes/eqtls/WASP/mono_34reps/total_counts\")\n",
    "#counts <- readRDS(\"featureCounts_mono34reps.Rds\")\n",
    "#print(counts$counts)\n",
    "#mycounts <- fread(\"./featureCounts_mono34reps.txt\", )\n",
    "#print(mycounts)\n",
    "\n",
    "\n",
    "### Check how the counts looked last time.\n",
    "\n",
    "orig <- fread(\"~/spivakov/HRJ_monocytes/eqtls/normalise_for_leo/submission_25thMay2022/eGenes_RNA_counts_untransformed.txt\")\n",
    "#print(head(orig))\n",
    "\n",
    "### So we need to do the following:\n",
    "### Change the gene name using the ensembl94 table\n",
    "### Get the gene location, also using the first ensembl94 table that I had with gene start/end\n",
    "### Change the sample names to the names of this table.\n",
    "\n",
    "genes_locs <- fread(\"~/spivakov/HRJ_monocytes/external_data/ensembl/ensembl94_hg38_geneStartEnd.txt\")\n",
    "genes_locs[, feature := paste(`Gene stable ID`, `HGNC symbol`, sep = \"_\")]\n",
    "genes_locs[, chr_pos := paste(`Chromosome/scaffold name`, `Gene start (bp)`, sep = \":\")]\n",
    "genes_locs[, hg38_location := paste(`chr_pos`, `Gene end (bp)`, sep = \":\")]\n",
    "genes_locs_join <- unique(genes_locs[, .(`Gene stable ID`, feature, hg38_location)])\n",
    "names(genes_locs_join)[1] = \"GeneID\"\n",
    "#print(head(genes_locs_join))\n",
    "\n",
    "### Join to the new results \n",
    "# wasp1 <- fread(\"./featureCounts_mono34reps.txt\")\n",
    "# then I was trying with results of multimapped\n",
    "# wasp2 <- fread(\"./featureCounts_mono34reps_multiFeatures.txt\")\n",
    "# now I am trying with results of GFT file - THIS IS THE ONE I USE\n",
    "wasp <- fread(\"./featureCounts_mono34reps_GTF.txt\")\n",
    "# now I am trying with results of multimapped to GTF file\n",
    "# wasp3 <- fread(\"./featureCounts_mono34reps_multiFeatures_GTF.txt\")\n",
    "\n",
    "#print(head(wasp))\n",
    "\n",
    "wasp_feat <- wasp[genes_locs_join, on = \"GeneID\", nomatch = NULL]\n",
    "nrow(wasp)\n",
    "nrow(wasp_feat) # from 56264 to 56373, probably because some ENSG IDs have more than one gene name. Will also compare with the old results.\n",
    "wasp_feat_inOld <- wasp_feat[feature %in% orig$feature]\n",
    "nrow(wasp_feat_inOld) # 4742 - probably because I filtered onto the number of reads.\n",
    "nrow(orig) # we had 4762 lines and now we have 4742, that seems OK\n",
    "\n",
    "### Get the average no. reads across the table\n",
    "mySamples1 <- names(orig[, 3:36])\n",
    "print(head(orig))\n",
    "orig[, avg_reads := rowMeans(.SD), .SDcols = c(mySamples1)]\n",
    "\n",
    "mySamples2 <- names(wasp_feat_inOld[, 2:35])\n",
    "wasp_feat_inOld[, avg_reads := rowMeans(.SD), .SDcols = c(mySamples2)]\n",
    "\n",
    "myfeats <- unique(orig$feature)\n",
    "myfeatsSampled <- sample(myfeats, 4762) # this is the number of genes anyway\n",
    "\n",
    "old_sampled <- orig[feature %in% myfeatsSampled]\n",
    "new_sampled <- wasp_feat_inOld[feature %in% myfeatsSampled]\n",
    "\n",
    "setnames(old_sampled, \"avg_reads\", \"avg_reads_noCorrection\")\n",
    "setnames(new_sampled, \"avg_reads\", \"avg_reads_WASPCorrection\")\n",
    "\n",
    "old_sampled_plot <- unique(old_sampled[, .(feature, avg_reads_noCorrection)])\n",
    "new_sampled_plot <- unique(new_sampled[, .(feature, avg_reads_WASPCorrection)])\n",
    "\n",
    "to_plot <- old_sampled_plot[new_sampled_plot, on = \"feature\", nomatch = NULL]\n",
    "\n",
    "print(to_plot)\n",
    "\n",
    "library(ggplot2)\n",
    "pdf(file = \"./RNA_WASP_comparison_with_no_correction.pdf\")\n",
    "p <- ggplot(to_plot, aes(x = avg_reads_noCorrection, y = avg_reads_WASPCorrection)) + geom_point() +\n",
    "       ggtitle(\"RNA-seq WASP correction\")\n",
    "print(p)\n",
    "dev.off()\n",
    "\n",
    "print(p)\n",
    "\n",
    "pearson <- cor(to_plot$avg_reads_noCorrection, to_plot$avg_reads_WASPCorrection, method = 'pearson')\n",
    "pearson\n",
    "spearman <- cor(to_plot$avg_reads_noCorrection, to_plot$avg_reads_WASPCorrection, method = 'spearman')\n",
    "spearman\n",
    "\n",
    "## Looks OK! but investigate why some go to zero (See next chunk)\n",
    "## When we allow for multiple overlaps, we don't get any more at zero. However, we get some that have MORE reads than previously.\n",
    "## Now with GFT it looks a lot better.\n",
    "## Look at the multiple overlaps GFT, out of interest.\n",
    "## Again, sometimes there were genes with too many reads.\n",
    "## So we stick with the GTF single overlaps method."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "56264"
      ],
      "text/latex": [
       "56264"
      ],
      "text/markdown": [
       "56264"
      ],
      "text/plain": [
       "[1] 56264"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "56273"
      ],
      "text/latex": [
       "56273"
      ],
      "text/markdown": [
       "56273"
      ],
      "text/plain": [
       "[1] 56273"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4742"
      ],
      "text/latex": [
       "4742"
      ],
      "text/markdown": [
       "4742"
      ],
      "text/plain": [
       "[1] 4742"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4762"
      ],
      "text/latex": [
       "4762"
      ],
      "text/markdown": [
       "4762"
      ],
      "text/plain": [
       "[1] 4762"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                   feature          hg38_location S025NM S025QG S025RE S025SC\n",
      "1:    ENSG00000213341_CHUK 10:100188298:100229619    802    551    579    506\n",
      "2: ENSG00000095485_CWF19L1 10:100232298:100267680    814    600    613    498\n",
      "3:  ENSG00000107593_PKD2L1 10:100288154:100330486      8      5      4      7\n",
      "4:    ENSG00000119906_SLF2 10:100912569:100965136    513    274    378    446\n",
      "5:  ENSG00000055950_MRPL43 10:100969458:100987515    233    183    240    297\n",
      "6:    ENSG00000107833_NPM3 10:101781325:101783413     46     35     55     41\n",
      "   S025TA S025U8 S025V6 S025W4 S025X2 S025Y0 S0260R S0262N S0263L S0264J S0265H\n",
      "1:    665    504    689    815    690    537    709    527    523    751    579\n",
      "2:    656    531    740   1049    542    716    616    531    720    758    576\n",
      "3:      5      9     18      6     13      1      6      7      9      8      9\n",
      "4:    574    495    647    612    497    469    485    444    358    323    240\n",
      "5:    302    244    367    394    392    367    372    199    327    193    211\n",
      "6:     41     55     24     64     49     49     58     30     47     19     44\n",
      "   S0267D S0268B S026A7 S026B5 S026C3 S026D1 S026FY S026GW S026HU S026JQ S026KO\n",
      "1:    787    750    780    921    507    469    553    907    845    654    736\n",
      "2:    847    708    899    912    544    455    687    601    765    773    755\n",
      "3:     14     11     11      3      2     22      9      2      6     17      0\n",
      "4:    403    337    519    387    399    353    392    493    542    564    284\n",
      "5:    261    216    257    225    213    209    271    362    306    298    235\n",
      "6:     39     35     54     32     40     33     38     54     31     55     36\n",
      "   S026MK S026NI S026S8 S026T6 S026XZ S026YX S0270N S0271L\n",
      "1:   1058    465    416   1265    754    579    859    545\n",
      "2:    939    578    508   1310    702    604    821    579\n",
      "3:      3      8     11     12      3     15      4      4\n",
      "4:    435    490    400    676    401    593    657    478\n",
      "5:    256    311    240    415    208    334    343    363\n",
      "6:     66     65     45     86     54     54     42     50\n",
      "                     feature avg_reads_noCorrection avg_reads_WASPCorrection\n",
      "   1:  ENSG00000088766_CRLS1              253.26471                240.64706\n",
      "   2: ENSG00000086475_SEPHS1              216.85294                178.26471\n",
      "   3: ENSG00000134571_MYBPC3              190.41176                158.23529\n",
      "   4:   ENSG00000198373_WWP2             1841.64706               1598.05882\n",
      "   5:   ENSG00000256043_CTSO              470.11765                392.14706\n",
      "  ---                                                                       \n",
      "4738: ENSG00000163374_YY1AP1             1615.91176               2224.88235\n",
      "4739: ENSG00000141542_RAB40B               24.76471                 25.76471\n",
      "4740:  ENSG00000182199_SHMT2              695.76471                441.44118\n",
      "4741:  ENSG00000276045_ORAI1              702.17647                462.85294\n",
      "4742:  ENSG00000110801_PSMD9              163.11765                405.41176\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3dCZzM9R/H8e/e2F2yjtyKyq1D\nEqKUblehFRUKHXKldJAcRZSIThIqCUVyRhFFjnKWY93WWru//m4Wa/f7/11z7sx3PjPz++3+\nZuf9ejz++/vNb3+/+c5P+/zPzG9mfsM4QijoWEHfAIQKQ4CEkAEBEkIGBEgIGRAgIWRAgISQ\nAQESQgYESAH1Kvu0oG8CslRGQ9rCtIpe/2yKfvkt/Vd3snRt5lxxdldg196MTdNm7mOvaTO9\n2KA8V7nj+ToJkQm1n9/pdINiKj32R2BjuvfVAvnHu1W/MebaUCHJeEgJyXKP3ZXEiv6iXmax\n/2q/skP6nFVnuwO69jHsMXV6oQi7UVtSla1xv8oZUaz2488nV2Rxix036L4qLGJigPvkWrln\nDLkaVLgyHlJ1beb8c6zSFeVyfdYkV11ih3RTwmLWP6Br/5eVvKJMF7MqEeqV7WJJV9yu8lSx\nqDnKNOdNVi7LcYNyPmcxhwhDnHOZ5G0/AySUN9Mg8SvF2Vblcu/H2CfqAhukdeyJnAols1w2\nm9uiZEz5B5Yos7lTGifG1Xj9tDJ/KLlU0fpfpLNbbetVY+oDtBfZOO1B3njW2f0qf2UNtXVz\ne7yS5nyD7nE8r7k87saiZVr+qs5OuDUhrvqL8pp8MPvpo3JFbBOXG2LfoL3yOLGp/hzJadu3\n2IIN910VV//b4P8JUShmHiR+HdugXO5xvGTxo8plG6Qn2S98EJvhvNVkVubZN59JivhKnn+C\nVenzxu2s3hnOT1Rmd7094OoX2B22FfuywepVVzzOkpWZB9i37le5lt2Q4/EGPcdG6HO5rVmN\nF7oksOnyHdWD7IbXRj7IKhzmfCQbFN/padvE+YY4NljUjd0+fo4GyXnbUaxvqaFzxpZgK437\nt0QhlHmQUiKjTimXu8lM2ikLdEj/Fbkml+9mjZ23qsf2yT9Ti9/O+Wx200n5L/dF9grnQ1lH\neXH61exO24rL2S3yz73sKV47SeaSVTTqhPtVXq7J2u7wdIPuYlP1uRns/mzOdxcrdla+bU0u\nykveVJ58jWIlVnP7xPmGOG0wV31op0Jy3nY0i/xLnp/Euhr1L4lCKrMgZc67jr2gXu7Kc5uz\nH7gd0lg2XP7ZhG1z2qpyxHFlcokrD8CUQwT8ZEwZzm9ka5X5txyQLiUqa05i3/B+7E/Ol7Hm\nea9yV03GqnT5eLvrDcr9jMVn6NfSgilS+PiB+3lTtlSZPRMbc0HG0EiZ1yfON8RpAydIrtu2\nUeb/Zk2C/RdEIZlZh7/lul7gGiS+O678KRuk3OqR8iMh/oXKzNaz7IYv9SMRCeyEOr2ZHcyJ\nZepz/qUOSPJzFPnxWKuIDL5EOaw+gI31cJVX5nWpKI9f+e2L3H7U7sFrWPRM25UkMNuxhNwi\n7JQ6U49tkjH0U2b1idMNcdrACZLbtm8osynspiD+8VDoZtLh7ztYrYP65a7yz+HsWRukpew+\nZfnZ+OLy3+aVO5V+5Re6RzFWZ5D8+O6CA+K606yIeh2bnCBNk58bXYq/mfPzcfL9Rm220/0q\n9fZ/9Vg8u+2SXXZUhcf/tv3uAouzzZ6xzd4t37uMZu8os9rE+YY4beAEyW3b95TZvbbD8ijM\nMumh3YVq2sMeHdKl2hFrdEht7H+fkznPVmdmyYvTPnvkKhY7l2exiLf0DpzS/1LXO0HKjCyZ\n86v6auw9kSdSWTXufpWOjtdVHn85Hf2wdYHF2mbP2GZbsGU2DNrE+YY4beACKe+2gBSumfUc\naSmrcka73FWZ/BFR8+I9CqQjUVd1VevAGrhteumzqJKXeQlmeybDr0Sz88p0nhMk3ohtfYOt\n4sqLs/O/ZH153qs8Y1v1E/a0R0g8UX/UJj+0K6bP1mF/u0JyviFOGzg/tPOwLSCFa6YdtevI\nntcud1UvPsuGtlYgDWYD9DVrys8sbB06pk5ayo/U7mPfq/P/k/9XSzvY0N8Z0ttsYrME5ajE\nVta3K1vO81zlHRFL9IsjlCdNniC1YMrbfPioe9bJd5MLldkT0UUvukFyviFOGzgdbPCwLSCF\na6ZBSisesZo7IJ0qH3uTDOlyOaa/YYi/53iLwGZ2t0LjbOWo//hsVu+sPL8mOpnzl1kHeXbP\nVc6QtrI2RdQDZLzcLdUT5c3cr3IEu1p9pTV3ifqijidIM9gt8l3dwZJFTvHprJFySKIf68Hd\nIDnfEKcNFqrH8lVIHrYFpHDNvNeRPmTXXXBAkv+fnMmQZjteW82MjT9lm+/CqvUe2rsqGyjP\nP86qvPRW+5ji6zk/VoY1eqVb4ovOkHjlSPaxOvNkJHuU8zxXmf04Y9c81PmBKixiBPcMKbcV\nq/Lck4nKAcDc1qzmkOF3shv+c4fkfEOcNtgXEfP0cxokD9sCUrhmHqScBooLOyTlkEA6v8vp\nHQ0d2STbbM4nTcrGVrp3vjo/pXFikWt77lHmd7e9qkidz7e5QHqeqa/ecv4NY19y7uEqf+5y\nfbHIxHrPb3a9QU5dfr9+kWLN1Idr2R/eUiyu5usneR5IzjfEaYN3Sxe5VX+LUN5tASlcC4XP\nI21xgYSQBQMkhAwIkBAyIEBCyIAACSEDCgVICFk+QELIgAAJIQMCJIQMCJAQMiBAQsiAAAkh\nAwIkhAwIkBAyIIMhnTvlo/PZPlcJuqyzpg9xMfu06WNcNn2EU9mXTB/izEXThzibfcH0Mc6f\n97jYfloDoyGdknx0jp/xtUrQZfm8FUF3if/P9DFyTB9B4tmmD3HikulDnOIXTB/j3DmPi+2n\n8gCkwAIkaoAESIIAiRogAZIgQKIGSAF1+YqPcniOr1WCLjcfhuCmD3ElP4bINX2InHwYIj92\nw/PfVLZZkE7/56Nz/KyvVYLuos9bEXTyPZLpY+SYPsJ/PNv0IU5eNn2I0/yC6WOcO+dx8Umz\nIOGhnXHhoR2xwvjQDpCMC5CIAZJJARI1QKIGSOYESNQACZAEARI1QAIkQYBEDZAASRAgUQMk\nQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMk\nQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMk\nQBIESN7a9dk7P2Y6XQYkQBIESF76sjhjrMkBxwJAAiRBgOS5v+OZUifHEkACJEGA5Lk3VUcs\n9qh9CSABkiBA8twLGiT2r30JIAGSIEDy3Puao5LH7UsACZAEAZLnDlVTIb3tWAJIgCQIkLy0\nviljxQY7Hf8GJEASBEhe27P2mPNFQAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARK1EIa0ff7aDNs8IJkTIFELWUipnRhjN63VLwGSOQEStZCF\n1F39Ntzqh7VLgGROgEQtVCHti9a+oP0z7SIgmRMgUQtVSL9rjthg7SIgmRMgUQtVSHsiNUgf\naRcByZwAiVqoQpI6qo4q7tcuAZI5ARK1kIW0/17ZUbXl+iVAMidAohaykCRpzZSf0mzzgGRO\ngEQthCE5ZwSkX1r/yfnZcZ06DMvwPgUkwwMkYiEC6eST7WVII/vtTR3ZO8frFJAMD5CIhQik\n0dOe/JNLrffL9z7tNnubApLxARKx0IC0ruclGdK69rnyfJ/Z3qbyj4wVckfP+Ogiz/K1StBd\nPm/6ENn8rOlj5Jo+whl+xfQhzmWbPsR5fsn0MS5e9Lj4LBnS2ae2cxnSsm7KhSGTvU3lHysb\nyG3weQeHUOHJ/pzGJ6QJH3EVUnflwuDJ3qbyj8PT5Q6c89ElftHXKkGXnWX6EFf4edPHyDV9\nhHM8x/QhLlwxfYgsftn0MS5d8rj4PBXSlmfOq5DWqw/hXpzrbWpbH8+RjAvPkYiFwnOkse07\nd+7c5rFRJ9qkyEra7vA2BSTjAyRioQBJ/at/YvlpPqbP3iNDB+Z6nQKS4QESsVCApCY/tOPn\nJyR3HHXC+xSQDA+QiIUMJD8CJOMCJGKAZFKARA2QqAGSOQESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAElQ+EBaM/Sl\nGRlBDAFIgCQobCC9GcsYu/VQ4EMAEiAJChdIS5ha18CHACRAEhQukHpqkBIDHwKQAElQuEDq\npEGKTA94CEACJEHhAuktDVKNwIcAJEASFC6QDlRVIX0b+BCABEiCwgWStOm+GHbtl0EMAUiA\nJChsIEnSsX1BDQFIgCQojCAFGSABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZI\ngCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZI\ngCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgBRQ2bm+4j7XCLr8GAK7\nQR4jH4bIhzE8d8UsSLhHMi7cIxErjPdIgGRcgEQMkEwKkKgBEjVAMidAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAomYZSBnfjZyyN+Ax\nAMmcAImaVSD9W58xVmpuoGMAkjkBEjWrQGrJlJJ2BTgGIJkTIFGzCKSdTGtCgGMAkjkBEjWL\nQPpdhzQkwDEAyZwAiZpFIB2M1SBNC3AMQDInQKJmEUhSf9XRzWkBjgFI5gRI1KwCKb2/fJ90\n/7ZAxwAkcwIkalaBJElpa/cFPgYgmRMgUbMOpKACJHMCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkimQZre\nrfO4Y4bvqVOARA2QqFkPUm575XMhdQ8bvquOAIkaIFGzHqTp2icVexm+q44AiRogUbMepPYa\npMqG76ojQKIGSNSsB+lBDVIZw3fVESBRAyRq1oP0pgbpXsN31REgUQMkataDdKqq4qjYWsN3\n1REgUQMkataDxHcnly1xzyrD99QpQKIGSNQsCAkvyBIDJGKAZFKARA2QqAGSOQESNUACJEGA\nRA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAEnryPDHk1/bxfnZ\ncZ06DMvwPgUkwwMkYqEA6fJTHxw9Nj75Ah/Zb2/qyN45XqeAZHiARCwUIJ2ad4Hzo60PSK33\ny/c+7TZ7mwKS8QESsVCApHTmk+cvr2ufK8/1me1tKv9I+0Hu8FkfXeRZvlYJussXTB8im58z\nfYxc00c4y3NMH+J8tulDXOCXTB/j4kWPi8/RIeU80vq1//Fl3ZT5IZO9TeUfKxvIbfDtEqFC\nk/05DeEe6cj2t587t6y7Mjt4srcpxz2S4eEeiVho3CPJXem8aL36EO7Fud6mtlXxHMm48ByJ\nWCg8R9raM4vz3C4LT7RJkZW03eFtCkjGB0jEQgHSuafePZI+uX0aH9Nn75GhA3O9TgHJ8ACJ\nWChA4oeGd04etI3z8xOSO4464X0KSIYHSMRCApKfAZJxhRCkP+4vWfqRzR5/BUiAJAiQnPu7\nuHKKtXK7Pf0OkABJECA510476WdPT78DJEASBEhK//yyT51W0yA19LQOIAGSIECSpK0tGIvq\nmirP1dYg3elpLUACJEGAJB27RdXTTZ59SYM02tNqgARIggBJmqXpidwlSUdvVeZaZnhaDZAA\nSRAgSaM1SGypPJ8+sctTX2R6XA2QAEkQIElTdUieXz6yB0iAJAiQpIOVVEctfKwGSIAkCJAk\naVll2VGDf32sBUiAJAiQ5I58+/4Cz0+MnAIkQBIESNTCEtLxx0pFaM8gAcmpI1OHTTvqugiQ\nqIUlpMei7+32jFphgfR1k3INxnt8fYPeKuVZdZU1LssAiVpYQio9K1BAFoU0Tr17fT6oa0ur\nrl5JjXTnhYBELSwhFTtUuCAdKqY9UP09mGv7UX+9ZLHzQkCiFpaQmi8sXJCW6AYmBHNtX+pX\n8pXzQkCiFpaQtt62ulBBWqEb+DiYa1utX8k654WARC0sId1RlRWrqlY4IB27WiVQZFtQV9da\nvZL2LssAiVpYQmp+9z16hQOSNDdOMfB+cFe3/4koFt3toMsyQKIWlpCCz2KQpA29Wj71c9BX\nmLrO7WUkQCIXppD+Wzj5i2VnCg8kswIkamEJKad/jPJQKH4sIPkIkKiFJaT32KNfLFn8+f1s\nBiCJK/yQMt6/PuaaN9OCHiIsIdV+Vpsm3wJI4go/pNftZ2QIrrCEFLdcmy4oCkjiCj2kPdHa\nq2drvK1ALSwhxf+gTecmAJK4Qg9pvv4y9EfBDhGWkO5ofkmZXGjaHJDEFXpIy3RI04IdIiwh\nLY6o0mvEkK6lolYAkrhCD+mYdkaGEnuCHSIsIfH5NZV/vfpLAnUESAZWsAcbFibIfwlxQd8h\nhSkkztM2bjoeMCNAMrICfh3pn9eTB24KfohwhRRkgGRceEGWmMUg1RjFa9gDJHGARC38IDUa\nzxvZAyRxgEQt/CAZEiAZFyARsx6kBju16Vy8RchHgEQtLCGxTeoke3gxQBIHSNTCEBJzhHsk\nHwEStTCEtPVD1lY9O2SPoamAJA6QqIUhJM7vTwkUECAZHyARsx4kfmyi/CNz+DFA8hEgUQtL\nSLvLxco/D7HyewFJHCBRC0tI7a7bqEx2XtcekMQBErWwhFTqS236eRlAEgdI1MISUpGZ2nQq\nXkfyESBRC0tIje/LUSbnGjYBJHGARC0sIS2JrPHiyMFdkyKXApI4QKIWlpD44vrK+xpuXByo\nI0AyMEAiZkFInGdu3iwFzAiQjAyQiFkR0oXfv5P/dQHJV4BELTwhjS3O2J/8je5XAEkcIFEL\nS0hTWJvPZEgzokcDkjhAohaWkG58jmfJkPjrtQFJHCBRC0tIcSs0SMtw7m8fARK1sISU8JMG\naUZxQBIHSNTCEtLdd19UIJ2sex8giQMkamEJaXV0tedZ964lYv4AJHGARC0sIfFfb1He2XDb\nb4E6IkKa3aZh8krD99UeIFEDJGp+v7NB2rzlBA88EqQ31ROsfGX4ztoCJGqARM0vSJduXRiE\nITKkv2JVSEnBfz+plwCJGiBR8+8eqcL4/IA0ST/n1zLD91YPkKgBEjX/IC2sMy+I99lRIU3U\nIS01fG/1AIkaIFHzD1Lz+iy6fFUlMyGt1xyVSDV8b/UAiRogUfMPUpM779EzE5L0sgppsuE7\nawuQqAESNWt+G8WXd9dstcjwfbUHSNQAiZoVj9oZvpPuARI1QKJmxaN2hu+ke4BEDZCoWfGo\nneE76R4gUQMkalY8amf4TroHSNQAiZolj9qZHSBRAyRq1jxqZ3KARA2QqPkL6b+Fk79YdgaQ\nfAVI1MISUk7/GOWl0vixAUPKzvUV97lG0OXHENgN8hj5MEQ+jOE5x+m2XCC9xx79Ysniz+9n\nMwKFhHsk48I9EjHr3SPVflabJuPLmH0ESNTCElLccm26AGcR8hEgUQtLSPE/aNO5CYAkDpCo\nhSWkO5pfUiYXmjYHJHGARC0sIS2OqNJrxJCupaJWAJI4QKIWlpD4/JrK4e/6SwJ1BEgGBkjE\nLAiJ87SNm44HzAiQjAyQiFkOUnqmOvkziPNxAZJxARIxq0FaWEL7OFKNqzcBko8AiVr4QUqJ\nr/KLOrO2YvlMQBIHSNTCD1Lfonv0ue3RwwBJHCBRCz9INTrbZ9vXByRxgEQt/CDFv2ufHZkI\nSOIAiVr4QSrm+OLYt64CJHFWgLSwcULFp/cENQQgUfMDUu129tkWNwGSOAtAWqieZvPGoL6K\nAJCo+QHp5agN+txPbCQgibMApLramZ/HBjMEIFHzA1JayaSZygf+LowvUlECJHEFDyk9UoPU\nOZghAImaPy/Iri7Fku5u2yyBVcILsr4qeEiZcRqkHsEMAUjU/HuL0KBakazojSNOBuwIkAzM\nx0O7Nhqk74MZApCo+fum1dzzgSMCJGPzAWlXFcXRc0ENAUjU/IJ0fv0fwXx/LCAZm6/D30dG\nJff8IbghAImaP5DGxTMW9cIlQCJkBUgGBEjU/IA0m107cEgz1heQCAEStfCD1OTac/LPp2NO\nAZLvAIla+EFKeFP5uYH9AUi+AyRq4QeJfa78PMaWAZLvAIlaGEKaovxMZ0sByXeARA2QAEkQ\nIFELQ0iv/ym3mI1XJoAkDpCohSEk5wBJHCBRCz9IbzkHSOIsBOnYjsyAhwAkavjqS3OyDKQ9\nnWJY8TfSAxwCkKj5BSlH/bH6m+2A5CurQMporj4SHxjgEIBEzR9Is6pd4Pzc7fJ/l36A5COr\nQJqjPaWN2RfYEIBEzQ9ISyPLpXL+Cnvmq4fZfEASZxVI7+gHh1YENgQgUfMD0n3V5KVXSjXL\n5VfqtgIkcVaB9JEO6a/AhgAkan5AKv2O/ONP9oX8c1h5QBJnFUi7k1RHDQMcApCo+QEpcpb8\nYww7KP+cEgNI4qwCSZpVQnZ0bYB3SIBEzg9IRabJPx6qpMx+WBKQxFkGkrRr3MufB3xuO0Ci\n5gekWr1lB0W7KrM9agOSOOtACipAouYHpIHFZu3txJRvj91TPODj34AUZOnj2rYanqrOAhIx\ni0HKrCo/2n5cnvmhZMJuQBJnFqT0psqxgxqHlHlAImYxSPx/4wbNVaZfX/9roI4AKcje1o5m\n91LmAYmY1SDZupjLzwGSOLMgtdAgVVfmAYmYVSHxDT3x/Ug+MgtSMw1SFWUekIhZE9L/JtRj\nrBkgiTMLUn8NUgdlHpCIWRBS7opOcazCaymBOgKkIDt4jeIoaZsyD0jELAcpdeS1LK6VegQc\nkMSZdvh79zPXV03erM4CEjGLQZr3UBSrP+E/CZAI5f8LsvM7NHlyjdFDABI1f87ZUHLg3/IE\nkCjlOyT1uHjstwYPAUjU/PlWc1b/3TRAopXfkDbHqgchSgf1jbF5AyRqfkA6/cnNLOqh748B\nEqH8hmT73NEyY4cAJGr+HbXb1CuRXcVmAZLP8hvSRB3SEmOHACRq/r6OdHZKQ8Yaf3EWkMTl\nN6Q/NUeJR4wdApCoBfDOhq0vlGDxgCQu3w82aK/UfmzwEIBEzQ9I6fa589MaA5K4fIeU+Vmz\nai3nGT0EIFHzA1LELYP/uBIoIEAyPrwgS8xikNqVYKxk8vTjgEQIkKiFHySevW74HdEs4pYh\nawO/YwIk4wIkYlaDpHR2Yb86yh0TIIkDJGphCknpYL8EfK2LjwCJWphCurzmzSYx7OrHAUkc\nIFELR0i7JrZOZAkPjtuWG6gjQDIwQCJmMUjdK7HYpm+uvhwwIkAyNkAiZjFIjDX5JjMoRYBk\naIBEzGKQFvW5gUXU778o4PfZAZKxARIxi0GSOzzlsdIsptnwtdmAJA6QqIUlJLncv8a2LsOK\nA5I44yFt3em+BJCIWRISz93+8eN4HclXRkOaUYmxWgtdlwESMQtCSv+qy9WMlX3yW0ASZzCk\nxeqHJBI2uCwEJGIWg5S1/OX6jEXePnxT4C8jAVJg6Wcq7uKyEJCIWQxSUcbKPPHt/wJHBEgB\nV1mD1MhlISARsxik24dvdLkr+lRd6X/vP5H82h7Oz47r1GFYhvcpIAXTjRqkh10WAhIxi0Fy\nj+1Qfg54bf+xsU9k8ZH99qaO7J3jdQpIwTRag/SVy0JAIhYKkM68e5TzzNZ7pNb75Xufdpu9\nTQEpqDIfVxy95LoQkIiFAiS1XW2kde2Vh319ZnubAlKQrXhn7Fq3RYBELFQgnXlhCl/WTZkb\nMtnbVP5xeLrcgXM+usQv+lol6LKzTB/iCj9v+hi5po9wjueYPsSFK6YPkcUvmz7GpUseF5/3\nA1Jqr09y+bLuyuzgyd6m8o+VDeQ2eL06hApf9oMDviFt7bxI/rlefQj34lxvU/lHxgq5o2d8\ndJFn+Vol6C6fN32IbH7W9DFyTR/hDL9i+hDnsk0f4jy/ZPoYFy96XOx4h7cvSP92Vr6igp9o\nkyI/AWq7w9vUthGeIxkXniMRC4XnSJd6fqesn8XH9Nl7ZOjAXK9TQDI8QCIWCpC2tlZbxM9P\nSO44St7M2xSQDA+QiIUCJH8DJOMCJGIWhHT+GOdZ09/bo8wvDeSjsoBkXIBEzHqQdpUdzbMb\nM1bkrwAMAZLBARIx60F6tN3qjVkAACAASURBVP4B/g2bvL/xI4DkVMa6n3a7LQIkamEJqexM\nzjvW53xmRUBytLoeY1FPH3NZBkjUwhJS7EqekzSI8+VxgGTvQBX1ndl9XBYCErWwhFT5C/4L\nW8X51AqAZG+89hGHuFTnhYBELSwhPVPu1arXXOEZ9TsCkr0B+vcg/+28EJCohSWkY7ezMn9x\nnnxVIC8hFVZIYzRHMYedFwIStbCExPlp5dSQm9J5oBVCSLtLq5CedFkISNTCFFKwFUJI0k+V\nZEcPutwhARK58IMU71QMIDmV+sNna9wWARK18IOUnJzcqUZM4/btbopo2A+QxAEStfCDJDe3\n7jFlsqvKT4Akzhekf7+etj3YMQCJmPUg1f1Om350IyCJ8wFpVFHG4l4PcgxAImY9SLErtOkc\nvLPBR2JIc7Uj5lODGwOQiFkPUoXH1Y+65rQpD0jixJAe1iA1Dm4MQCJmPUhvsbp9Ro7sXZO9\nBkjixJAaaJCuDW4MQCJmPUi5Y8orfwCl37wCSOLEkNpqkO4KbgxAImY9SDKlwxvW7885lwJI\n4sSQlsWpkOYEN0agkNaPe3cFcVVAohbYOxt+wbu/vZe25u9MX0ftJpdirMT4IAcKENLrsbLh\nThmkdQGJmp+QFnVp1rRp09sTS+cDpPQ1C1OM3FVHpkJ6J5GxWkt9vY6UunzZYeEKhAKDNEd7\nWDmCtDIgUfMP0iwWXYlViGctFpkPaen1jMW8SPs/Tj8zE9In6p9p0kHLvrPhEQ1SLdLKgETN\nP0gNHjrD43Zlf3pvICcQ8g/SrjLqf+8hxu6ulpmQrtP+Tl+zLKQ7tRtYhrQyIFHzD1KifE8U\nt4vzl18wHdJQ7b93yUxj91fNTEjR2u3uYFlIT/vzGhYgUfMPUonFMqaNnK8pZzok/b8322fs\n/qqZCam8drN7WxbSX4nqDZxPWhmQqPkHqVGHy7zGWM7nx5sO6XXtDzLBjCdJZkIapJ3AYYNl\nIUmLajFW/gvauoBEzT9I37L7ed+4vsMrNDEd0uZED+fmMSgzIaU/Kt/q+ImW/hjFP39THzAD\nEjU/D39/N5afuJexyptMhyTNKiv/RbZPM3Rv9cx9HWnVB1P+xeeRyIUnJLW0XZcDdeTP60gH\n53y+zshddVTg72wwJkAiZkVIF37/Tv7XDdhROLyzQQ2QqIUnpLHFGfuTv9Edb1r1ESBRC0tI\nU1ibz2RIM6JHA5I4QKIWlpBufI5nyZD467XDENKxbX4cigckamEJKW6FBmlZ0cINKWXJJmc0\nh/44LO19IoYVG0A+iAhI1MISUsJPGqQZxQszpPRnoxm7yX6mur3JESzyybvVl7Weow4BSNTC\nEtLdd19UIJ2se19hhvSSaqbKAe1S5oPMUeS/xCEAiVpYQlodXe151r1riZg/CjGktGIamnHa\nxZXMuQXEIQCJWlhC4r/eovw13fZboI5CAdJW3Uxf7eIXLpDcT03sLQ+QDgx/vP8qv2+rIEAi\nZkFInEubt5zggRcCkI7EamZGaRfnOTuqR32TWl5IP6rnaRjt/631GiARsxykS7cuDMJQqECS\nntA+5Ko/HUrTPqpXtaTyvOlP6hB5IO3TzncSu9bvW+s1QCJmOUi8wvhwgHT4XvkvvtwPtou/\nKd8RW21tyoeDPj9KHiIPpN76fdpb/t1UUYBEzHqQFtaZF8T77EIFkiStnDT7kONS6vSRX/n7\nLvQ8kO7WIb3i5xUJAiRi1oPUvD6LLl9VqXBDCr48kB7TIX1t3BiARMx6kJrceY8eIInLA+kb\nzVElAz/xC0jErAfJpU8DOXoX6pD2fjL0q3TCenmP2vVSHF29M+CR8wZIxCwOiQXy1eYhDmle\nkqyh5hbfK3p4HWlR/+cmHw90YE8BEjFAMqmAIaWUpp7KCu9soAZIYQjpU/2IwV8+1wQkaoAU\nhpBG6ZB+9rkmIFEDpDCENFtzFO375P6ARA2QwhDS8cYqpBd9rwlI1AApDCFJO9tFsiL9j3n4\nzbonG7Wd6bgISNQAKRwhSdLhjXkZHV0x/2v1LeMD7YsAiRoghSckD82pyFiE9uxptW0ZIFED\nJEDS2pTg+LzS27aFgEQNkJYG8m1jhRDSi04f/LN/oWRgkDb2uLvzT+S1AYmY9SBFx+slVHho\nZQCMCiWkdk6QltsWBgRJ+xAt7dtdJUAiZz1IfW5jtR95tC67o3PL4pFLAUnteYejZ+wLA4F0\nvJL2IdqNxPUBiZj1IC2vuk6ZrLv+L37qtjsASW1tUfXvv/b1zSY5PiMRCKQ1Osf3iesDEjHr\nQbr5a2069U7Ovw3oW/sKISRpWinGYnq5nhYlEEi2E3+9S1wfkIhZD1Lscm26tBjnPyYAkt7B\n76dvdVsUCKS0EhqklcT1AYmY9SBVfCxXnT5Vhmc/eCsgeS+ggw2fuj3R8hEgEbMepKGs7oCx\n7w26jfXhj7BZgOS9wA5/z21RseE48ofRAYmY9SDljFC+2JWVGHCJT/g2EEeAZGCARMx6kK7w\n3IMb1u8N+Pv6goQ0p2HiNf0Pef01OUCiBkjU/INU/qUtgRsKGtLX6lOIFsGfhycfIZnyrex6\ngETMepBuj2B13j1SQJAyK2oHtaYFs79q+QVpV6cSUbWmmzYGIBGzHiR++L2GLOKuqacLAtK/\n+sss/YLZX7V8gpR2o3p7Z5g1BiARsyAkuYNjbmVFHisASPv0jyq8Gsz+quUTpA+121vZrDEA\niZg1IXGeO7+69/eEmwdJ0j7nzX4NYne18glST/0udJ9JYwASMUtCurKqdwWW1LMgIG1Uzyr3\nehB7q5dPkAbop0ox64ADIBGzHqTs5b3KsmLJCy4H6ii4w997h7TvuTjwfbWXT5BWaJAeNmsM\nQCJmPUhJLPrBr88FrEju9H8+OsfP+lol6C76vBVBJ0P677+hiqNqu8waI8esK3bEs00f4uRl\n04c4zS+YPsa5cx4Xn/QM6Y6PJXV6LiVQSJev+CiH5/haJehy82EIrvz8a8gLn583bQxu2jU7\nhsg1fYicfBgiP3bD89+U4+vEPB5W+KVCoJDwFiHjwkM7YtZ7aMcXdWnWtGnT2xNLA5I4QKIW\nlpBmsehKrEI8a7EIkMQBErWwhNTgoTM8blf2p/cGcgIhQDI4QCJmPUiJ8j1R3C7OX34BkMQB\nErWwhFRisYxpI+drygGSODdIfz9zR7uvjB4DkIhZD1KjDpd5jbGczw/oxCfhC2l5EeUVpd4G\njwFIxKwH6Vt2P+8b13d4hSaAJM4VUk3q95P5FSARsx4k/t1YfuJexipvAiRxLpB26O9efcPY\nMQCJmAUhqaXtKqD32hlVfkPayoz6AIhLgETMqpCCKSwhZVbWIM03dgxAIgZIJpUPkC4cdT61\nxPeqo2SDxwAkYoBkUqZDOtq3GIt7+oBjwYpW190+5rjBowASMUAyKdMhPa3eA7UyeRRAIgZI\nJmU2JNvBheW+Vw0mQCIGSCZlCKTtc5Yf9fKruTqkjwwYRhAgEQMkkwoK0t6Phkw7JmX0jGKs\n8o+eV9E/Y85mBjEMIUAiBkgmFQykeUmykOs3D1allNzmcZ3j2jsZKhhwdmVRgEQMkEwqCEh7\ny6hGbkvS7nO8nNJojXJS2NKLAh6FFiARAySTCgLS58ylJ7ysdmTaWxPNOp2dPUAiBkgmFQSk\n0bqgROGbfrb8lYXPIxEDpDCClG6f0w/IRfVVJyW2eFr7h2ry86O5gEQLkMIFUuqAqyOqjtHf\n8ZNxhyroxeNdlaMJcz2tv7aYuorZz5AAiRwgmZR/kDq4fARiV/soVqT/Mfnh2zcLUz2u/7h2\np3VXsLfSZ4BEDJBMyi9IP2suYlJsC1L/ShetLzUy+Uso7AESMUAyKb8gfaAfXlhI3eBhbf2b\nA7hh/gVIxADJpPyCZDvgvYq6wTfa+u8HcMP8C5CIAZJJ+QVpdwnVRXX6hyAGxcrrP/ef/7fL\nzwCJGCCZlH8HG6YppwBK+sWPLTaMf+8vvI5EDJDCBZL09+CuI1J8r+YSThBJDZDCBlIgARI1\nQAIkQYBEDZAKI6R9094jH+YWBkjUAKkQQppdijHWdL8BQwASNUAqfJB2XKUe6e5gwBCARA2Q\nCh8k/TMSUQeDHwKQqAFS4YP0kv4mhr+DHwKQqAFS4YM0SXNULC34IQCJGiAVNKT09b8F+Bfv\nDdLh60Ufe/UrQKIGSAUMaU5lxkp+6NfO2vJ61G59Y8Zi+5PfVXf0/acHeHnrECBRA6SChaR/\nDvU7/3ZXS/CC7D+/pfw8dyftav69RrkFgz3+DpCoAVLBQnpSez5zm3+7qyV6Z8PsCozF9M2k\nXM1D2k3w+D18gEQNkAoWUnPtr7isf7urJYC0KUG92lGEa0mL1m5Cf0+/BCRqgFSwkNprf8V1\n/dtdLQEk7exArBLhWg7oB8t7ePolIFEDpIKF9KP2V/yeX3urJ4Ck+4zI8LqGo6rauh4PeAAS\nNUAqWEjSO8rn7XqSnsy4J4D0ooajAuVqtM+U33zM0+8AiRogFTAkafuUSev92VVHAkgfxag6\nhpGu5+u60SWf3O3xV4BEDZAKGlLgeYfURWUU3ZPyyE7J64m5AIkaIBVCSJ9pD+x6Bj8EIFED\npEIIST8l3XXBDwFI1ACpMEA6MvP9+U4P4+7SIF0d/BCARA2QQhRS2h/b7JCWVVIOu+2w/+55\nDdI9wY8CSNQAKTQhvZ3I2E2rtfmDlVQ3d9p/uauscrnIb8EPA0jUACkkIWkfObpae1vql/p7\nEzbZf/3nvUViGi42YBxAogZIIQlJfzeC9pGjMTqkJU4rHPf4+qrfARI1QAo1SOnjkzt9GKHR\neUxdMke7EEn82IQ/ARI1QAoxSGkNFTTaGxfYC+qidHUR6+r/lR3+YfIa4QqARA2QQgzSIOao\niH60YVtLxqK6ef7ivbwdnD72R+29fT9WkK+k1RHBuoBEDZAKHNLxXX7sqHSTZihJ/l/Cp/al\nu34hnw7yx6vlTRsqb63bWUq9qu6ClQGJGiAVMKSUJ2LZVUPE30LpXC0N0s0/j/nsYCDn/t5T\nRt3+AXl2lHZVsYK7MkCiBkgFCymzhfrH/DJ5T/XvSH5GCvAk+vq5uiL+laR++iPErd7XBiRq\ngJRfkH4e0HNS3mPS32t/yzH7qHu6PUl9BUl5aBYQpKG6Hvnp1XvaXFHB2cAAiRog5ROkl5W/\n2Vp73dfTTy/s+dQjHtvU7uryHTcrcwFBmqbLPSBJ+7R3RHg8WYMeIFEDpPyB9JPzCz9OfaxD\n2hDATgcEKa2eOl5vZX5lHcaie4heugUkaoCUP5B66Qes3T9TvrukuvyWQHY6sC8a23yXfH/0\nnPZw7vgfCzx/MtYWIFEDpPyB1Fl/jp/n//5nJsqLKwdyhxTwN/btXHWIuiogUQOk/IGkH2qu\nmXfNf8f0/4j6Wqpr+OpLaoBEzfKQUmuqkOYasrd6gEQNkKhZHpK0o2Px6HqzDNjVH+8qV3+Y\n+hQHkKgBEjXrQ5Iz4PuKbGeha6fMAhI1QKIWEpCMKKOc9mTre6lAIO005P8MXAMkYoBkYJv1\n151ekwoA0viyLOq+zUaPAUjEAMmfUrcJT1+8Q4f0pqRAypg5dJL+3vG0NRupp4P0IxdI2ovH\n1x82eAxAIhYakI4ObKtMzo7r1GFYhvepyZC2PRzJSgzzQGnN4OcmqQ+ramuQVsmzWftvlOeK\nT1cWjy/J2LU/BDKkMBdI5bWh3zV4DEAiFhKQ1jw1XoU0st/e1JG9c7xOzYWUVl/9Ux2R5xdv\nx8qLb1DehbBK/eajQcrCrNbq2olbJWmmOhcf0Ou6opwh2b7/5WmDxwAkYiEB6dfMPxVIUuv9\n8r1Pu83epiZD0s81nOD+lH6ltryVMr+jb8sntLue45Ha4nck6VZtTvQZvYByhpQeqw3yksFj\nABKxkIDEuQppXftc+Wef2d6mJkMaqP+f/l+el0e7AtvN7H/Z2mddWYsAxhTm8tAuWR0jTnyW\nB/8DJGKhBGlZN2V2yGRvU/nH/oly+y746DK/5GsVD43VPESmuy3vrZNJc1l6PlFb+uWFC/oz\npy4BjCnsCs9yXEhvoLzt9iOjx8g1+grzxnNMHyLriulDXOTZpo9x+bLn5X5D6q7MDp7sbSr/\nWNlAboPPqwuoQ9pXv7Z1X/6F5qRyrmPRufc69eqmLr3pIucfaiusNudm2cr5cdjE/eYOgSyZ\n/eAAFdJ69SHci3O9TeUf/9sgd+yUjy7w875W8dT0eJlDvb3uizO0gxDTHUv2VlYW3JzAIur2\nniGdOql8O3r0O4EMKSybnzb8Ot3LNX2EU/yK6UOczTZ9iHP8ouljZGV5XHzGX0gn2qTIT4Da\n7vA2ta1s2utIOz54/avjeRf/0zE+4rop2vyWb+Ydktppd0EzxisP767fuv9a5VK9o06bHP30\n5XF+nZ/IU3iLEDU8R9I6IS1vK0lZfEyfvUeGDsz1OjUbktcy9dPPZfSMYazMV/EapI5XqZOm\nPbWLrzrW36ic1bjEbHkubVvgowISNUDSeqa10gJ+fkJyx1HyZt6mBQbJ1nBVTLFoTU5D/ShE\ndW3SWF/pwPfT6qgLkvZsbxXFSo4I6MueJUCiB0gBVXCQtBOWMP29qw8yl4uNtHW+Ks1sTdSe\nXb0d4GiARA2QQgxSlAbkrjjlZ23bCYz1A+ED1FU2xNsdsWT99wG+axuQqAFSiEGqosnou6Jl\nqWo3R9vBKOchtr2ftL/DEeuoT/8ObDRAogZI5kDaMePjPw3fS6V3VRcJynsf3nMCs/zZG2/p\nr5//u5NjcYsR2jSSfAJK1wCJGiCZAul15ZHX88bvpyRl9o9lrOIcZbaiA0wF549QvKwti2Cx\nT6Rs1h7mPRDgaIBEDZDMgDRTf6Zvwp5K0u65y49K0q5dxyMckGY7r7BVOyb+9g7l1PxTiinP\npQJ9QQmQqAGSGZD0g2kNjdm51e+Pd/98xPfXyc+IkrRRKl33sHrC4w2fTNFPhz+/KmOx/fVD\n3tvff3W6h5d4aQESNUAyA5L+8k5VI3Yts5t8TbGvuixbXUS5eu1r++I3astekOfjhmnzx1b+\nuMeIsQGJHCCZAUk/VmbIpxr0QwoznZe11ZZdI/+vrP6LidqiOUYM6QiQqAGSGZBWq6/ysJ+M\n2LWbNSFJTV85JEnbJoyY+0GXZ/Rj4LGrZsyznUGhgbaolRFDOgIkaoBkBqQt6rGyBL/e5Lbo\njhJVeub52hdJqmw/olDzyMSizKXkPKsZ9LTMFiBRAyQzIOmv5HT2Yw8Wqlvckvc7Vu5yuHkq\nztURS3Ks1kxbkud7Y4ILkKgBkhmQ9A+s1vNjD+pqm4zL84uFDjfl3RyxYo7VtK/+s33PuVEB\nEjVAMgOSftSuEX0HjukvCj2hXTw84NrEht9p81PL2dyUcYfUzOkaPkpirNLMPFccXIBEDZDM\ngPSW9nc+kr4Dy3RIPdVLmS3VC9O032Vs0h8qNncQUt/fXdTlPCRpq9fSvxydGCBRAyQzIKXf\nofyd30N/HfR321GELupF7Uz57GrbG3/2qYcSauywH3io+nePa8q2Xku+/kADJGqAZAakrerp\nse6lf5zuAZuQ2C3KxVf0S6OaVrlL/SqYvX0bNhp4UPqjCWOJNySWTt6uLMS3UVADJGoWg/SQ\n5uAj8u13HONWT8wwxPmJ0PvOK+7d5Hh76sqXu3fqOeIf8iBK3z3esp+yxfGNf1OYAxI1QDID\nknZWLT+ORde0s5mqXFztDKnofs/bPKf9OtGfNzP0U7dYox7AqErYEJCoAZIZkPRnPB3It/9l\nm5o47Q5GvUuyvbt7vuu6h97u3Pdn+zvMGSvtBZqHfta2uFE7pF7U9zlTAYkaIJkBqYX2Bzsu\nbc0a2me805rqKt7RH3At6dXMfpe00GXVLRWUZYOdPr73FWkIpdf1LZpS7zEBiRogmQFpg/rY\nrvGUsoyVmULagYypvTo2b3hbxYiE5J3akttsTpJSpcOrttrX1JH+0soOif5UzHZucf0p2a0+\nNwAkaoBkBiRp81P1Gw3+SX1HT9xi8l5oZyaur92J6efFZ7FfHb9XfpBXcaW2zn79KyheHmSH\nRP8Oiu90R/oLxr4/NwtI1ADJFEjqyU/0+4z7qTuRWUHb4EP10vXahWprpHvUmfgD8sKPrtfP\nIsRe2Gc/0hdNf3OsdpPmTNA2/Mbn+oBEDZDMg6S/fa6GfeG+3nWqdfJ+Pp+9Ooxe6qXB2oU5\n0h59cR9JGme/F2KfSX+3tb2HdQb53yntzfrl75HvInuwPB8W9BggUQMk8yDdrf2ZN7ctS62l\nXCzhVdJR/exaA9VLx5VP78W+5jg811xKS7Q7uv24JGV9r1/4zv9/sN8/+HAjYTVAogZI5kGa\nof2Zf2lbpr/M+rDXjbQHXnGr9ItLho9dJ09W6FwavjNGn4sq11M5vVZWqnaWkxL049/+BkjU\nAMk8SNJrsfJ9yiv2ZfdrDEp73WjXdcqd0DtuSzMc90N6o7VfZJ2apnwbZew03/8+gQZI1ADJ\nREjS5smfb3Yse1hjUF6e/aFxyeovHc6zVdqkZ9/I+yrpgmg3SPortFmnpLW97u9l5ltXAYka\nIJkJybX3NQadJelbdaaFt3e7bZ38kctpWnd3bxDpDEn/XnG8aZUaIFELCUgZ6mupVVKkTP3I\ntZfHZCOVo3G9XBb96XKP1Oyb8T/OHvf9WUAiBkjUQgKSdPyD1ve+cVCSduog+nncdJ72S5c3\nfae43CPpb+Wru4XyTxNUgEQNkPIRkq39+ttRB9kWpDl9tDXzBu2X9V02acU8dHOGZHKARA2Q\nCgCSdLsGYYV2adltMbHNf7P9brSupLjLCYVS1O9tYdH6//SWUP5tggmQqAFSQUDaoJ62Wz8w\nvk45zT27yvY47RqbkuYu9zdHblVWmrbmg09edECK7pJC+ucJOECiBkgFAUlKeaPtM7bzsLbW\nUKjnD0ob3dHxbMjt1Fw/v/+lekLvyc6P7u4O9NthaQESNUAqEEjO6e9OvUWePVDTGUlb5bcZ\nHzS7ofUK5/W1dxrZmu/xOo0KkKgBUoFD0k/ufac8+7TLsYTWym+fVGddPhW+pobTSmOowwQU\nIFEDpAKHpJ8E74NFd5Z2fQvDu/IvF2uzFZ2eL/1ayXmlydRhAgqQqAFSgUM6fp8Cov08Bw7t\njKpxyjOiN/VFm+yrp17j7Ki0h9PuGxggUQOkAockSTP79v/e9lTJqf7yrwbo8+qzpINLlh6W\nPndeJfF7+iiBBEjUAMkCkJQO5HGkfk/ZMH1+prSlR7UoxqJKOq9xozHfy+c9QKIGSBaBdDQy\nD6TGkv3kxWzu6mJuv60WXemVI9q2v73a40Pa2Yr8DZCoAZJFIDl/D5LebfLSFO1ckyUO3uz2\ny2Qp0/bu7xGqq3/9HI8UIFEDJItA+r1phDukZpK0/D31iHjsl/tdfxXZ7ZD9YxQrtUUP+jce\nLUCiBkgWgJT5Scs6sXnukNiTx5Q3PRRp0qL7GinF9VfqO/N0SPpZWqNS6QOSAyRqgGQBSN3d\n7m60SdzvL2lT9Yx2Lu95uFbdTIeknwOcmXHgAZCoAVLBQ1ridk8UqT0vij9QVrvc6nP5DmiJ\n011W3Hp1Ox3SJG1hRTPedQdI1ACp4CENzvuoTq19lH225Z7+MfYLN+jngdAhHdOOQ3wpGCHg\nAIkaIFkXEnM64u38yK6Vvp3tqN2ux4tH1jDFESCRA6SCh+T+0M5X9+rbOZ385Ch5MP8CJGqA\nVPCQpDbeyDQv7mmp7RPqOIsQNUCiFtKQjldXfeR5HYmxAcf+aJ5nYbWD+naARA2QqIUipMX9\nn3x+snLc4BddiHJcrnyc80kZFtkOybEq9mW32d/CAEjUAIlaCELSTr0Q81ymNF8nMnLcKw8X\nZbH32dHcL0kNtLmqW260LYz4w3YNgEQNkKiFHqS5NhijpX/1V2Drrevi9ijuO0n/EF+jLV+8\n8oC+sJntKgCJGiBRsyakzf0f7bfBywrdbFiin93bT5+NY27VkqpqM9Xlh322F5JK2a4CkKgB\nEjVLQppTRMEx3fMKdzq4lPyrirsgvajMsnkXVrBdhQhSxgFft5AUIFEDJNMgpWqfGL/K84fB\nk5xo3FPRC6QkKc+dFGOdbFfhHdKu5CKs4ge+/9l8BkjUAMk0SAv0P/yvPf1+t8sdz01eIPWQ\nbsizrJj922C8Qjp2q7rmOC+/9iNAogZIpkGarf/lT/H0+33ONiLez8OlvPKj3KdHP1Iv6QfE\nlS/oq2A/aOcdkn4KyZLpXn5PD5CoAZJpkP7V//w9f1frrU5q6km93RwlrV1TX52peF8xxq4d\nqs7HLZ//4Q9OOLxCekm/ls1efk8PkKgBkmmQpFfVv+benlf4w/Hun4gi17y14emG1e0flKj4\n5u7NY20Xbh++Jl0aV4Kxsu5fp+QV0jD9evf5upU+AyRqgGQepIwJtYvWfPe4lzX+7dcy+ZvR\nVYto91v10192ukN6zeVcQfV2SdLhn3/Jc3oTr5A2FFG3a+nrRvoOkKgBknmQSP2ta+nm/kWx\nTj3geVPvR+0mKgf7qu+g3QJRgEQNkAoE0kH73Je2x3feHbGIPwY+8vwq5+33zPjsT9HrSJuG\n9fnYiBN0ARI1QMp/SPt7JLKrR+pn8/5GAMie8uwpdrxylpS6RasPTZM+SpQXdD2PdzYQAyRq\noQQpUz3XN+uqnfVnr+iuyKWiW6R31JkuK7WXaUcBEjFAomZBSCc/aH3va57eqPOjDqPcd+rF\nelRIbMIB7SACa6tNrgEkYoBEzXqQcloqf+uV3c6RdWzUPXfY32UXN3K7tOlBDye089LopfqM\n9hITKwJIxACJmtUgpb7bUPtj7+Sy+HhTVxpFh5dzuRydxAT93EefaalNagESMUCiZjFIe6vb\n/vjLOi/e/7y7Dbcz58f/pM/UuEaduLxl9fHPbGstTVSnUwGJGCBRsxikJ+x//UlOS0fYTq/l\n/fiC7Ux2P3yhrHvHGZlEogAAEYZJREFU/Kts4Ire8Faa/gbWqI+keVUZKzYMn0eiBkjU8hvS\n6f+ElbHLuH/uxEWStvBL+8KSdbxK0rrundLqdNKesb2Hti4e23CBcgX6J/vulmfT//j50H8X\nfdwKA5IhmT5Gjukj/MezTR/i5GXThzjNL5g+xrlzHhefNAvSpWxRl+xnRY1XPuHa6Ii6tKHd\nyaPVGa2q2vVlaRP9U0u9bcPkXBHeCiPK5aYPkZ0fQ+SaPsSVfBiC55g+Ro7nIS6bBUn8oMp2\nxscS7bVTLihfWC4NsT+gK7nlKncx3jrmfLWD1EVxv9ou46EdNTy0o2at50if6gxmfKXPbJSk\nR2w2ruq6XapNdJTkcrXpj8qLin1ovwxI1ACJmrUg2c4QtHKIPrNQmmazEbdUXmEWEVJftyte\nNW7yP45LgEQNkKhZC1Ka9g3lt2zWvp+FRXy2qYNOI2KilD62VYs7ozzLca2V8r7TpW3rP/iN\nx2EAiRogUbMWJGl1NZlB7b9aOUzYjuNVkTLuUqaU9zM0V65qqjr7hsvVZ/ylnlAFkKgBEjWL\nQZLS5n2yIF3y8D6F6IN5z8/gpcRN8hWlasclYp0/rz5GXnb774BED5CoWQ2SdC5nw/x/Snng\n8XgrDwvzVrx4yzXK9SzTL09yXPXH6oJKewGJHCBRsxykDcrX6MV7EFK1AQnSbv16PECqrC0Z\nCkjkAIma1SAdvNabkLgEiqOoCdNT1CtK1c7eELvJftVp+iqdAYkcIFGzGqRJDhNJ2vdINPZK\ny0tXaScN0t5YNNjpurV3rLK+gEQOkKhZDdKrTiLWKI/yGm37wU9IrJh2/v2fH6n/0Ezn6+6h\n/jZuNSCRAyRqVoM03uEhNj1j+Re/ZkoP+wuJDfR83al3yb8rOgFH7egBEjWrQUpxfCF5nXXq\nkql+O2KNU6Vj7z/W+ZMM92v/ccSEbRIg0QMkalaDJNVwukuakLZi9jbawTrXKv+pnl2/hZdz\nTAISNUCiZjVIs50/2xqrHLCO8WxFnP59fe94HgSQqAESNYtBWlM0EDZeu9PzKIBEDZCoWQxS\nJ0MdsUaeRwEkaoBEzWKQGmkAYksEKUh/PPi851EAiRogUbMYJNv76R4PEtIb6ltWK6Z4HgWQ\nqAESNYtB+k6H0HJStYgSncgfLHc+PVdsAqsyXlrfumyFTtu8jAJI1ACJmsUgSbod+clNqiRp\n37fHEn07qtTcPvvsvNW+dhqQqAESNatBekDT8Ix6ob12wfeHYm/K+MumTTljULOt4kEAiRog\nUbMapN/V49+ltK/6ukfH4fN7J2ZL0vKGUZH1h3xyh3q5obdv+9MCJGqARM1qkGYqJ3is/od0\naGyPlz+0QfIhKV77zFHqYUn6U1+0QDgIIFEDJGoWg7RWe0H24w3lfT6c04sc9bPTOezm6Esn\neR9CAiR6gETNYpC0TzqwG/x4g91Dztv/ri/8QTgKIFEDJGoWg3S/5oD8ZXxyV2tbHnzjwfaf\nZkrac6Q64q+BBSRqgETNYpC6+iFIr4q64W71baqtMrcr742ou148CiBRAyRqFoM02J1JUhFf\nkHqoG+rnkfxEylzx+RLxMTtAogdI1KwFKSXP6R+7dXBf4la09vYF/Ux4DQZ8sNf3TgMSNUCi\nZi1I0/NCiff1fOlldUvHux+SFvvcaUCiBkjUrAVpog80nnpA3bKFY0HFI752GpCoARI1a0Fa\nFwCkjtqWjnM9KO9zEAdI1ACJmrUgSaX9h/SVtuW6NpX1z5ezz33tNCBRAyRqFoP0jN+Oejg2\n3qwvwru/DQuQqFkLUqa/Z1WNcbn36a4u6+hzpwGJGiBRsxak5X4YeiGBscpzlK3+WbAhU938\naP8EVuz5wz53GpCoARI1a0H61Lcfe3sP/bxGebvqEeWEKQ21sxRLmTvznBTSQ4BEDZCoWQvS\nbD8gHZLXX/lKjw+7qJdqpvqx04BEDZCoWQtSqqdvGPNcdcn+UXS1L/3YaUCiBkjUrAVpp893\n1tmrdFha4Xz5LT92GpCoARI1a0GaRnbEWH+pv/PFz/zYaUCiBkjUrAXpcz8gNbR9ClCt8gE/\ndhqQqAESNWtBWu0XpAnajPJw8NoV/uw0IFEDJGrWgjTLD0gDpLT66syMXz+dL/5ErHuARA2Q\nqFkL0hSKoMhrlJ81j0jSzk7Fo2pN93+nAYkaIFGzFiTSu7+b7h9we5NBh7Qt/Lsr0gMkaoBE\nzVqQpG6+HSWsC3qnAYkaIFGzGKQ0H5+HrXhj9y3B7zQgUQMkahaDJIm/gGKGMTsNSNQAiZrV\nINUSQvrdmJ0GJGqARM1ikH4UP7IbZMxOAxI1QKJmMUi1xZB6GbPTgEQNkKhZC1JmpBjSeGN2\nGpCoARI1a0H6U+yo1lFjdhqQqAESNWtBWiFiFN3KxxfxkQMkaoBEzVqQhOdsuMewnQYkaoBE\nzVqQ1oogVTdspwGJGiBRsxYkSQTpdsN2GpCoARI1a0H6WwTpY8N2GpCoARI1a0EaL3DU27id\nBiRqgETNWpAe8abolg99fAmfXwESNUCiZi1IjbxBmmfoTgMSNUCiZi1I93lxFL3b0J0GJGqA\nRM1akJ7zAulVY3cakKgBEjVrQXrfA6KyEde86+vblf0MkKgBEjVrQfokr6PrJIMVSYBED5Co\nWQvSN3khdTVhpwGJGiBRsxakt/I4ivvFhJ0GJGqARM1akOq4O0qYZMZOAxI1QKJmLUjXuDl6\nZI8Z+wxI5ACJmrUgVXN1ZNBJg/IESNQAiZq1ICW4OOpuxg4rARI1QKJmOqSz4zp1GJZBhBTj\n7CjSjP1VAyRqgETNdEgj++1NHdk7hwYpyhlSphn7qwZI1ACJmtmQpNb75XuldptpkJxPWDzH\njN3VAiRqgETNbEjr2ufKP/vM9htSMzP2Vg+QqAESNbMhLeum/BwyWf6x5x25lCxRDkjRwvWC\n7MolM69dLYdfNH2MXNNHyOI5pg9x0fwhLvFs08fI9jKEYZC6Kz8HK5BWNpDbIFzbDikiuFER\nskb2gwPBQlqvPrR7ca7848xOuYyTomwnWo0QrhV0l86ae/1yl/kp08fIMX2Ek/yK6UOcvmz6\nEGd5luljXLjgcfFpoyCdaJMiPzFqu8N2mXYWoQwTHsQ6hedI1PAciZrph7/H9Nl7ZOjAXCKk\nyqqjq0zYUecAiRogUTMd0vkJyR1HOa7O55/wqrgNhu+ke4BEDZCoWestQsoN4mcM30n3AIka\nIFEDJHMCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAGSOW185598HtGUZr9zvqBvghGNnlrQt8CI\nDr2zsqBvQv5DmtVgaT6PaEr9G5ws6JtgRI2eKOhbYESbGnxU0DcBkAIMkCwUIIVugGShACl0\nAyQLFZaQECqUARJCBgRICBkQICFkQPkL6ey4Th2GZeTrkEb0v/efSH5tD+d9Wst1dOyG+9Ti\n+br5obEb21urLbLYf438hTSy397Ukb1z8nVMAxrw2v5jY5/I4t0XSpL0P8duuE8tnq+bHxq7\ncVl5a84/HY9Y7L9GvkKSWu+X/w+j3eb8HNOAzrx7lPPM1nt4h03qZdtuuE8L+Gb6zMfND5Xd\nUHpjps/dyedblK+Q1rXPlX/2mZ2fYxrVrjbS5dYTX+z+dpp9N9ynBXwLfeXr5ofIbij99sxl\nn7uTzzcpXyEt66b8HDI5P8c0qDMvTOGnnnx3155hT52z7Yb7tCBvHyFfNz9EdkMu+5lVvncn\nn29T/kLqrvwcHAL/qdxL7fVJrjaX9dgy2264TwvqxvmT4OaHzm782t32DMg6/zXyFdJ69V73\nxbn5OaYhbe28yD7/wkzbbrhPC+rW+ZX3mx86uzF4un3WMv818hXSiTYp8l1y2x35OaYR/dv5\nb2VyaNJlzi90WGnbDfdpQd9MH/m6+SGyG/I+tNvOLfdfI38Pf4/ps/fI0IG5+Tpm8F3q+Z1y\nyDXrTOfx6UdHPZ1l3w33qbXzefNDYzfk/19rfZoTdid/y19I5yckdxx1wvd61mqr7SXAvUOS\nnxh13LEb7lOL5+vmh8hu8DVtVCXW+q+BtwghZECAhJABARJCBgRICBkQICFkQICEkAEBEkIG\nBEgIGRAghVDJ8X6snP5KnfiEmv1S/B6lUQ2/N0GAFEr5A+mPpKi2o0Z3iktY5Htde1uUP4fx\no/y9XQiQLNclwe/8gHS8dNIGZbqrconjHkfwOM5E/DkEGv7l8qVZja5KbDCL86alspWLjcpf\nyRlWOe6W5X2inddqfOfiSrfKdyYtE4vcNNV5M547vFJc3bmdZEjHelSJu/rRXc6bNb9je4v4\nMp0y5dklzRKK1BmXy/mrbKb2y1/GZTot1kbIM87SxkXLPJrC72eMNdAe2tm3cLpyJAiQ8qPZ\nrMOihQ+wRfxjtly+eDjiJf42e+znGVVuc7mPubN+nfE/8VUxdy5Y/hx7z2kzPoY9vnxW7Rry\n2reX++LXr+uWdf5Wmbur3rrg4MyoZM7nRzT7/ueebCDntZKuOFZwLNZGcB9ncWTLb6ZWK5+e\n0pZt2qlCcmzhuHIkCpDyo7H3Xeb8dHQXLkX3ki++x7bkXl1X/r/7tcwF0j1snvyzwbUX5Z9t\nSlxwbJZboY68dlpMPD/NXpV/eWBUmstmv8k/W5bivGZlZdNHYv7LjbzbaQX7Yn0E93Hq3yCr\n2xA7kT+j/DkokJy3sF05EgVI+VelZpw/WDaH84Z1+CE2QFlU2xVStPzAT2LPZ8l9xn53bHaY\n9VHmGsfz7NJVf3E/09Q9RZWf3SJ4GlOY8i/YovOsreP3jsXaCO7jZLIX9TVtkJy3sF05EgZI\n+dHJATWLx8Wxppx/zVbxg+xdvoGNVX7xiCuk8vKPzUzvO8dmG9hI5fft5bU3XsOSHvv2istm\nVZWfMoKNbIQyt4RN4UWbOX7vtFgdwX2czdq1cwck5y2q2n+BBOEfKD+6L3b0+p27KsiQzhbr\nzcdEHOHrlecmnD/qCqkqVw5Bd/9TTXJstl77U2+nrH1l5aB6rMlF982Uv/VNbIgyt5h9wW9M\nuGD/vdNibVW3cTazwfqaNkh5tgAkX+EfKB9KY33ln9nxMiSeXCH31rs438sGKb+pmxfS/1jX\nPJvt1x583WRb+zM2w30z5W/9uLbp52wZH8HGa79cW2O902InSI5xJG3uUKYdUp4tAMlX+AfK\nh3ax4fLPT9nt8s8FbD6bKt+vlGgkX1jP8kLit5VQvsRsxuBsx2Y5pWvJT4x2RcbzTcnKWa33\na/dnLpspf+v1Sin3Qw8WO81PViy6UFm6o9LVJ5wWO0FyjMNvTDqn3cgeLFs72OC+BSD5Cv9A\n+dDlyhUXrHm5ZYvEX8/xy0nViyjn7ujHui378vo7PED6LabuVz8PienuvNmbrO3cj6s2jOfH\nStSfuvzbxnG782ym/K0vimz6/U+Ps9HyxU0V2d3D3u4QXX6T82JnSPZx+OLIhlMn31AujQ9l\nw+eqkNy3ACRf4R8oP9rUuFjZXmcWli6fwnkv1lFZdLF36aLNNnZOcF5N+6Plv9+bWKTW6CvO\nm115rVxsvfn9Yjjf1r5cbPm2G/Jupv6tL2sWH3fzl+pSacRNJRLrvaW+kmpf7AzJMQ5f0rho\n6Uf2cJ56c9GbtRdk3bYAJF/hH6hAa1mhoG8BMiZAKqAmPCo/HTlV/KGCvh3ImACpgJrJWi2Y\ndVvkyoK+HciYAKmg+vqW+GJNl/KlzN6nfmwd4GbIrACpgDu7w54/ZwcNcDNkVoCEkAEBEkIG\nBEgIGRAgIWRAgISQAQESQgYESAgZ0P8BhPIas/LuS0MAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.90116241022625"
      ],
      "text/latex": [
       "0.90116241022625"
      ],
      "text/markdown": [
       "0.90116241022625"
      ],
      "text/plain": [
       "[1] 0.9011624"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.919161012527925"
      ],
      "text/latex": [
       "0.919161012527925"
      ],
      "text/markdown": [
       "0.919161012527925"
      ],
      "text/plain": [
       "[1] 0.919161"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAADAFBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcICAgJCQkKCgoLCwsMDAwNDQ0ODg4PDw8QEBARERESEhITExMUFBQVFRUW\nFhYXFxcYGBgZGRkaGhobGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJyco\nKCgpKSkqKiorKyssLCwtLS0uLi4vLy8wMDAxMTEyMjIzMzM0NDQ1NTU2NjY3Nzc4ODg5OTk6\nOjo7Ozs8PDw9PT0+Pj4/Pz9AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJSUlKSkpLS0tM\nTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dYWFhZWVlaWlpbW1tcXFxdXV1e\nXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1ubm5vb29w\ncHBxcXFycnJzc3N0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiJiYmKioqLi4uMjIyNjY2Ojo6Pj4+QkJCRkZGSkpKTk5OU\nlJSVlZWWlpaXl5eYmJiZmZmampqbm5ucnJydnZ2enp6fn5+goKChoaGioqKjo6OkpKSlpaWm\npqanp6eoqKipqamqqqqrq6usrKytra2urq6vr6+wsLCxsbGysrKzs7O0tLS1tbW2tra3t7e4\nuLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLDw8PExMTFxcXGxsbHx8fIyMjJycnK\nysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV1dXW1tbX19fY2NjZ2dna2trb29vc\n3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn5+fo6Ojp6enq6urr6+vs7Ozt7e3u\n7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5+fn6+vr7+/v8/Pz9/f3+/v7////i\nsF19AAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nO3dCZzM9R/H8e/e2F2yjtyKyq1D\nEqKUblehFRUKHXKldJAcRZSIThIqCUVyRhFFjnKWY93WWru//m4Wa/f7/11z7sx3PjPz++3+\nZuf9ejz++/vNb3+/+c5P+/zPzG9mfsM4QijoWEHfAIQKQ4CEkAEBEkIGBEgIGRAgIWRAgISQ\nAQESQgYESAH1Kvu0oG8CslRGQ9rCtIpe/2yKfvkt/Vd3snRt5lxxdldg196MTdNm7mOvaTO9\n2KA8V7nj+ToJkQm1n9/pdINiKj32R2BjuvfVAvnHu1W/MebaUCHJeEgJyXKP3ZXEiv6iXmax\n/2q/skP6nFVnuwO69jHsMXV6oQi7UVtSla1xv8oZUaz2488nV2Rxix036L4qLGJigPvkWrln\nDLkaVLgyHlJ1beb8c6zSFeVyfdYkV11ih3RTwmLWP6Br/5eVvKJMF7MqEeqV7WJJV9yu8lSx\nqDnKNOdNVi7LcYNyPmcxhwhDnHOZ5G0/AySUN9Mg8SvF2Vblcu/H2CfqAhukdeyJnAols1w2\nm9uiZEz5B5Yos7lTGifG1Xj9tDJ/KLlU0fpfpLNbbetVY+oDtBfZOO1B3njW2f0qf2UNtXVz\ne7yS5nyD7nE8r7k87saiZVr+qs5OuDUhrvqL8pp8MPvpo3JFbBOXG2LfoL3yOLGp/hzJadu3\n2IIN910VV//b4P8JUShmHiR+HdugXO5xvGTxo8plG6Qn2S98EJvhvNVkVubZN59JivhKnn+C\nVenzxu2s3hnOT1Rmd7094OoX2B22FfuywepVVzzOkpWZB9i37le5lt2Q4/EGPcdG6HO5rVmN\nF7oksOnyHdWD7IbXRj7IKhzmfCQbFN/padvE+YY4NljUjd0+fo4GyXnbUaxvqaFzxpZgK437\nt0QhlHmQUiKjTimXu8lM2ikLdEj/Fbkml+9mjZ23qsf2yT9Ti9/O+Wx200n5L/dF9grnQ1lH\neXH61exO24rL2S3yz73sKV47SeaSVTTqhPtVXq7J2u7wdIPuYlP1uRns/mzOdxcrdla+bU0u\nykveVJ58jWIlVnP7xPmGOG0wV31op0Jy3nY0i/xLnp/Euhr1L4lCKrMgZc67jr2gXu7Kc5uz\nH7gd0lg2XP7ZhG1z2qpyxHFlcokrD8CUQwT8ZEwZzm9ka5X5txyQLiUqa05i3/B+7E/Ol7Hm\nea9yV03GqnT5eLvrDcr9jMVn6NfSgilS+PiB+3lTtlSZPRMbc0HG0EiZ1yfON8RpAydIrtu2\nUeb/Zk2C/RdEIZlZh7/lul7gGiS+O678KRuk3OqR8iMh/oXKzNaz7IYv9SMRCeyEOr2ZHcyJ\nZepz/qUOSPJzFPnxWKuIDL5EOaw+gI31cJVX5nWpKI9f+e2L3H7U7sFrWPRM25UkMNuxhNwi\n7JQ6U49tkjH0U2b1idMNcdrACZLbtm8osynspiD+8VDoZtLh7ztYrYP65a7yz+HsWRukpew+\nZfnZ+OLy3+aVO5V+5Re6RzFWZ5D8+O6CA+K606yIeh2bnCBNk58bXYq/mfPzcfL9Rm220/0q\n9fZ/9Vg8u+2SXXZUhcf/tv3uAouzzZ6xzd4t37uMZu8os9rE+YY4beAEyW3b95TZvbbD8ijM\nMumh3YVq2sMeHdKl2hFrdEht7H+fkznPVmdmyYvTPnvkKhY7l2exiLf0DpzS/1LXO0HKjCyZ\n86v6auw9kSdSWTXufpWOjtdVHn85Hf2wdYHF2mbP2GZbsGU2DNrE+YY4beACKe+2gBSumfUc\naSmrcka73FWZ/BFR8+I9CqQjUVd1VevAGrhteumzqJKXeQlmeybDr0Sz88p0nhMk3ohtfYOt\n4sqLs/O/ZH153qs8Y1v1E/a0R0g8UX/UJj+0K6bP1mF/u0JyviFOGzg/tPOwLSCFa6YdtevI\nntcud1UvPsuGtlYgDWYD9DVrys8sbB06pk5ayo/U7mPfq/P/k/9XSzvY0N8Z0ttsYrME5ajE\nVta3K1vO81zlHRFL9IsjlCdNniC1YMrbfPioe9bJd5MLldkT0UUvukFyviFOGzgdbPCwLSCF\na6ZBSisesZo7IJ0qH3uTDOlyOaa/YYi/53iLwGZ2t0LjbOWo//hsVu+sPL8mOpnzl1kHeXbP\nVc6QtrI2RdQDZLzcLdUT5c3cr3IEu1p9pTV3ifqijidIM9gt8l3dwZJFTvHprJFySKIf68Hd\nIDnfEKcNFqrH8lVIHrYFpHDNvNeRPmTXXXBAkv+fnMmQZjteW82MjT9lm+/CqvUe2rsqGyjP\nP86qvPRW+5ji6zk/VoY1eqVb4ovOkHjlSPaxOvNkJHuU8zxXmf04Y9c81PmBKixiBPcMKbcV\nq/Lck4nKAcDc1qzmkOF3shv+c4fkfEOcNtgXEfP0cxokD9sCUrhmHqScBooLOyTlkEA6v8vp\nHQ0d2STbbM4nTcrGVrp3vjo/pXFikWt77lHmd7e9qkidz7e5QHqeqa/ecv4NY19y7uEqf+5y\nfbHIxHrPb3a9QU5dfr9+kWLN1Idr2R/eUiyu5usneR5IzjfEaYN3Sxe5VX+LUN5tASlcC4XP\nI21xgYSQBQMkhAwIkBAyIEBCyIAACSEDCgVICFk+QELIgAAJIQMCJIQMCJAQMiBAQsiAAAkh\nAwIkhAwIkBAyIIMhnTvlo/PZPlcJuqyzpg9xMfu06WNcNn2EU9mXTB/izEXThzibfcH0Mc6f\n97jYfloDoyGdknx0jp/xtUrQZfm8FUF3if/P9DFyTB9B4tmmD3HikulDnOIXTB/j3DmPi+2n\n8gCkwAIkaoAESIIAiRogAZIgQKIGSAF1+YqPcniOr1WCLjcfhuCmD3ElP4bINX2InHwYIj92\nw/PfVLZZkE7/56Nz/KyvVYLuos9bEXTyPZLpY+SYPsJ/PNv0IU5eNn2I0/yC6WOcO+dx8Umz\nIOGhnXHhoR2xwvjQDpCMC5CIAZJJARI1QKIGSOYESNQACZAEARI1QAIkQYBEDZAASRAgUQMk\nQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMk\nQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMk\nQBIESN7a9dk7P2Y6XQYkQBIESF76sjhjrMkBxwJAAiRBgOS5v+OZUifHEkACJEGA5Lk3VUcs\n9qh9CSABkiBA8twLGiT2r30JIAGSIEDy3Puao5LH7UsACZAEAZLnDlVTIb3tWAJIgCQIkLy0\nviljxQY7Hf8GJEASBEhe27P2mPNFQAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBEDZAASRAgUQMkQBIESNQACZAEARI1QAIkQYBE\nDZAASRAgUQMkQBIESNQACZAEARK1EIa0ff7aDNs8IJkTIFELWUipnRhjN63VLwGSOQEStZCF\n1F39Ntzqh7VLgGROgEQtVCHti9a+oP0z7SIgmRMgUQtVSL9rjthg7SIgmRMgUQtVSHsiNUgf\naRcByZwAiVqoQpI6qo4q7tcuAZI5ARK1kIW0/17ZUbXl+iVAMidAohaykCRpzZSf0mzzgGRO\ngEQthCE5ZwSkX1r/yfnZcZ06DMvwPgUkwwMkYiEC6eST7WVII/vtTR3ZO8frFJAMD5CIhQik\n0dOe/JNLrffL9z7tNnubApLxARKx0IC0ruclGdK69rnyfJ/Z3qbyj4wVckfP+Ogiz/K1StBd\nPm/6ENn8rOlj5Jo+whl+xfQhzmWbPsR5fsn0MS5e9Lj4LBnS2ae2cxnSsm7KhSGTvU3lHysb\nyG3weQeHUOHJ/pzGJ6QJH3EVUnflwuDJ3qbyj8PT5Q6c89ElftHXKkGXnWX6EFf4edPHyDV9\nhHM8x/QhLlwxfYgsftn0MS5d8rj4PBXSlmfOq5DWqw/hXpzrbWpbH8+RjAvPkYiFwnOkse07\nd+7c5rFRJ9qkyEra7vA2BSTjAyRioQBJ/at/YvlpPqbP3iNDB+Z6nQKS4QESsVCApCY/tOPn\nJyR3HHXC+xSQDA+QiIUMJD8CJOMCJGKAZFKARA2QqAGSOQESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEAS\nBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAElQ+EBaM/Sl\nGRlBDAFIgCQobCC9GcsYu/VQ4EMAEiAJChdIS5ha18CHACRAEhQukHpqkBIDHwKQAElQuEDq\npEGKTA94CEACJEHhAuktDVKNwIcAJEASFC6QDlRVIX0b+BCABEiCwgWStOm+GHbtl0EMAUiA\nJChsIEnSsX1BDQFIgCQojCAFGSABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZI\ngCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZI\ngCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgBRQ2bm+4j7XCLr8GAK7\nQR4jH4bIhzE8d8UsSLhHMi7cIxErjPdIgGRcgEQMkEwKkKgBEjVAMidAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJ\nAiRqgARIggCJGiABkiBAogZIgCQIkKgBEiAJAiRqgARIggCJGiABkiBAomYZSBnfjZyyN+Ax\nAMmcAImaVSD9W58xVmpuoGMAkjkBEjWrQGrJlJJ2BTgGIJkTIFGzCKSdTGtCgGMAkjkBEjWL\nQPpdhzQkwDEAyZwAiZpFIB2M1SBNC3AMQDInQKJmEUhSf9XRzWkBjgFI5gRI1KwCKb2/fJ90\n/7ZAxwAkcwIkalaBJElpa/cFPgYgmRMgUbMOpKACJHMCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkimQZre\nrfO4Y4bvqVOARA2QqFkPUm575XMhdQ8bvquOAIkaIFGzHqTp2icVexm+q44AiRogUbMepPYa\npMqG76ojQKIGSNSsB+lBDVIZw3fVESBRAyRq1oP0pgbpXsN31REgUQMkataDdKqq4qjYWsN3\n1REgUQMkataDxHcnly1xzyrD99QpQKIGSNQsCAkvyBIDJGKAZFKARA2QqAGSOQESNUACJEGA\nRA2QAEkQIFEDJEASBEjUAAmQBAESNUACJEGARA2QAEkQIFEDJEASBEjUAEnryPDHk1/bxfnZ\ncZ06DMvwPgUkwwMkYqEA6fJTHxw9Nj75Ah/Zb2/qyN45XqeAZHiARCwUIJ2ad4Hzo60PSK33\ny/c+7TZ7mwKS8QESsVCApHTmk+cvr2ufK8/1me1tKv9I+0Hu8FkfXeRZvlYJussXTB8im58z\nfYxc00c4y3NMH+J8tulDXOCXTB/j4kWPi8/RIeU80vq1//Fl3ZT5IZO9TeUfKxvIbfDtEqFC\nk/05DeEe6cj2t587t6y7Mjt4srcpxz2S4eEeiVho3CPJXem8aL36EO7Fud6mtlXxHMm48ByJ\nWCg8R9raM4vz3C4LT7RJkZW03eFtCkjGB0jEQgHSuafePZI+uX0aH9Nn75GhA3O9TgHJ8ACJ\nWChA4oeGd04etI3z8xOSO4464X0KSIYHSMRCApKfAZJxhRCkP+4vWfqRzR5/BUiAJAiQnPu7\nuHKKtXK7Pf0OkABJECA510476WdPT78DJEASBEhK//yyT51W0yA19LQOIAGSIECSpK0tGIvq\nmirP1dYg3elpLUACJEGAJB27RdXTTZ59SYM02tNqgARIggBJmqXpidwlSUdvVeZaZnhaDZAA\nSRAgSaM1SGypPJ8+sctTX2R6XA2QAEkQIElTdUieXz6yB0iAJAiQpIOVVEctfKwGSIAkCJAk\naVll2VGDf32sBUiAJAiQ5I58+/4Cz0+MnAIkQBIESNTCEtLxx0pFaM8gAcmpI1OHTTvqugiQ\nqIUlpMei7+32jFphgfR1k3INxnt8fYPeKuVZdZU1LssAiVpYQio9K1BAFoU0Tr17fT6oa0ur\nrl5JjXTnhYBELSwhFTtUuCAdKqY9UP09mGv7UX+9ZLHzQkCiFpaQmi8sXJCW6AYmBHNtX+pX\n8pXzQkCiFpaQtt62ulBBWqEb+DiYa1utX8k654WARC0sId1RlRWrqlY4IB27WiVQZFtQV9da\nvZL2LssAiVpYQmp+9z16hQOSNDdOMfB+cFe3/4koFt3toMsyQKIWlpCCz2KQpA29Wj71c9BX\nmLrO7WUkQCIXppD+Wzj5i2VnCg8kswIkamEJKad/jPJQKH4sIPkIkKiFJaT32KNfLFn8+f1s\nBiCJK/yQMt6/PuaaN9OCHiIsIdV+Vpsm3wJI4go/pNftZ2QIrrCEFLdcmy4oCkjiCj2kPdHa\nq2drvK1ALSwhxf+gTecmAJK4Qg9pvv4y9EfBDhGWkO5ofkmZXGjaHJDEFXpIy3RI04IdIiwh\nLY6o0mvEkK6lolYAkrhCD+mYdkaGEnuCHSIsIfH5NZV/vfpLAnUESAZWsAcbFibIfwlxQd8h\nhSkkztM2bjoeMCNAMrICfh3pn9eTB24KfohwhRRkgGRceEGWmMUg1RjFa9gDJHGARC38IDUa\nzxvZAyRxgEQt/CAZEiAZFyARsx6kBju16Vy8RchHgEQtLCGxTeoke3gxQBIHSNTCEBJzhHsk\nHwEStTCEtPVD1lY9O2SPoamAJA6QqIUhJM7vTwkUECAZHyARsx4kfmyi/CNz+DFA8hEgUQtL\nSLvLxco/D7HyewFJHCBRC0tI7a7bqEx2XtcekMQBErWwhFTqS236eRlAEgdI1MISUpGZ2nQq\nXkfyESBRC0tIje/LUSbnGjYBJHGARC0sIS2JrPHiyMFdkyKXApI4QKIWlpD44vrK+xpuXByo\nI0AyMEAiZkFInGdu3iwFzAiQjAyQiFkR0oXfv5P/dQHJV4BELTwhjS3O2J/8je5XAEkcIFEL\nS0hTWJvPZEgzokcDkjhAohaWkG58jmfJkPjrtQFJHCBRC0tIcSs0SMtw7m8fARK1sISU8JMG\naUZxQBIHSNTCEtLdd19UIJ2sex8giQMkamEJaXV0tedZ964lYv4AJHGARC0sIfFfb1He2XDb\nb4E6IkKa3aZh8krD99UeIFEDJGp+v7NB2rzlBA88EqQ31ROsfGX4ztoCJGqARM0vSJduXRiE\nITKkv2JVSEnBfz+plwCJGiBR8+8eqcL4/IA0ST/n1zLD91YPkKgBEjX/IC2sMy+I99lRIU3U\nIS01fG/1AIkaIFHzD1Lz+iy6fFUlMyGt1xyVSDV8b/UAiRogUfMPUpM779EzE5L0sgppsuE7\nawuQqAESNWt+G8WXd9dstcjwfbUHSNQAiZoVj9oZvpPuARI1QKJmxaN2hu+ke4BEDZCoWfGo\nneE76R4gUQMkalY8amf4TroHSNQAiZolj9qZHSBRAyRq1jxqZ3KARA2QqPkL6b+Fk79YdgaQ\nfAVI1MISUk7/GOWl0vixAUPKzvUV97lG0OXHENgN8hj5MEQ+jOE5x+m2XCC9xx79Ysniz+9n\nMwKFhHsk48I9EjHr3SPVflabJuPLmH0ESNTCElLccm26AGcR8hEgUQtLSPE/aNO5CYAkDpCo\nhSWkO5pfUiYXmjYHJHGARC0sIS2OqNJrxJCupaJWAJI4QKIWlpD4/JrK4e/6SwJ1BEgGBkjE\nLAiJ87SNm44HzAiQjAyQiFkOUnqmOvkziPNxAZJxARIxq0FaWEL7OFKNqzcBko8AiVr4QUqJ\nr/KLOrO2YvlMQBIHSNTCD1Lfonv0ue3RwwBJHCBRCz9INTrbZ9vXByRxgEQt/CDFv2ufHZkI\nSOIAiVr4QSrm+OLYt64CJHFWgLSwcULFp/cENQQgUfMDUu129tkWNwGSOAtAWqieZvPGoL6K\nAJCo+QHp5agN+txPbCQgibMApLramZ/HBjMEIFHzA1JayaSZygf+LowvUlECJHEFDyk9UoPU\nOZghAImaPy/Iri7Fku5u2yyBVcILsr4qeEiZcRqkHsEMAUjU/HuL0KBakazojSNOBuwIkAzM\nx0O7Nhqk74MZApCo+fum1dzzgSMCJGPzAWlXFcXRc0ENAUjU/IJ0fv0fwXx/LCAZm6/D30dG\nJff8IbghAImaP5DGxTMW9cIlQCJkBUgGBEjU/IA0m107cEgz1heQCAEStfCD1OTac/LPp2NO\nAZLvAIla+EFKeFP5uYH9AUi+AyRq4QeJfa78PMaWAZLvAIlaGEKaovxMZ0sByXeARA2QAEkQ\nIFELQ0iv/ym3mI1XJoAkDpCohSEk5wBJHCBRCz9IbzkHSOIsBOnYjsyAhwAkavjqS3OyDKQ9\nnWJY8TfSAxwCkKj5BSlH/bH6m+2A5CurQMporj4SHxjgEIBEzR9Is6pd4Pzc7fJ/l36A5COr\nQJqjPaWN2RfYEIBEzQ9ISyPLpXL+Cnvmq4fZfEASZxVI7+gHh1YENgQgUfMD0n3V5KVXSjXL\n5VfqtgIkcVaB9JEO6a/AhgAkan5AKv2O/ONP9oX8c1h5QBJnFUi7k1RHDQMcApCo+QEpcpb8\nYww7KP+cEgNI4qwCSZpVQnZ0bYB3SIBEzg9IRabJPx6qpMx+WBKQxFkGkrRr3MufB3xuO0Ci\n5gekWr1lB0W7KrM9agOSOOtACipAouYHpIHFZu3txJRvj91TPODj34AUZOnj2rYanqrOAhIx\ni0HKrCo/2n5cnvmhZMJuQBJnFqT0psqxgxqHlHlAImYxSPx/4wbNVaZfX/9roI4AKcje1o5m\n91LmAYmY1SDZupjLzwGSOLMgtdAgVVfmAYmYVSHxDT3x/Ug+MgtSMw1SFWUekIhZE9L/JtRj\nrBkgiTMLUn8NUgdlHpCIWRBS7opOcazCaymBOgKkIDt4jeIoaZsyD0jELAcpdeS1LK6VegQc\nkMSZdvh79zPXV03erM4CEjGLQZr3UBSrP+E/CZAI5f8LsvM7NHlyjdFDABI1f87ZUHLg3/IE\nkCjlOyT1uHjstwYPAUjU/PlWc1b/3TRAopXfkDbHqgchSgf1jbF5AyRqfkA6/cnNLOqh748B\nEqH8hmT73NEyY4cAJGr+HbXb1CuRXcVmAZLP8hvSRB3SEmOHACRq/r6OdHZKQ8Yaf3EWkMTl\nN6Q/NUeJR4wdApCoBfDOhq0vlGDxgCQu3w82aK/UfmzwEIBEzQ9I6fa589MaA5K4fIeU+Vmz\nai3nGT0EIFHzA1LELYP/uBIoIEAyPrwgS8xikNqVYKxk8vTjgEQIkKiFHySevW74HdEs4pYh\nawO/YwIk4wIkYlaDpHR2Yb86yh0TIIkDJGphCknpYL8EfK2LjwCJWphCurzmzSYx7OrHAUkc\nIFELR0i7JrZOZAkPjtuWG6gjQDIwQCJmMUjdK7HYpm+uvhwwIkAyNkAiZjFIjDX5JjMoRYBk\naIBEzGKQFvW5gUXU778o4PfZAZKxARIxi0GSOzzlsdIsptnwtdmAJA6QqIUlJLncv8a2LsOK\nA5I44yFt3em+BJCIWRISz93+8eN4HclXRkOaUYmxWgtdlwESMQtCSv+qy9WMlX3yW0ASZzCk\nxeqHJBI2uCwEJGIWg5S1/OX6jEXePnxT4C8jAVJg6Wcq7uKyEJCIWQxSUcbKPPHt/wJHBEgB\nV1mD1MhlISARsxik24dvdLkr+lRd6X/vP5H82h7Oz47r1GFYhvcpIAXTjRqkh10WAhIxi0Fy\nj+1Qfg54bf+xsU9k8ZH99qaO7J3jdQpIwTRag/SVy0JAIhYKkM68e5TzzNZ7pNb75Xufdpu9\nTQEpqDIfVxy95LoQkIiFAiS1XW2kde2Vh319ZnubAlKQrXhn7Fq3RYBELFQgnXlhCl/WTZkb\nMtnbVP5xeLrcgXM+usQv+lol6LKzTB/iCj9v+hi5po9wjueYPsSFK6YPkcUvmz7GpUseF5/3\nA1Jqr09y+bLuyuzgyd6m8o+VDeQ2eL06hApf9oMDviFt7bxI/rlefQj34lxvU/lHxgq5o2d8\ndJFn+Vol6C6fN32IbH7W9DFyTR/hDL9i+hDnsk0f4jy/ZPoYFy96XOx4h7cvSP92Vr6igp9o\nkyI/AWq7w9vUthGeIxkXniMRC4XnSJd6fqesn8XH9Nl7ZOjAXK9TQDI8QCIWCpC2tlZbxM9P\nSO44St7M2xSQDA+QiIUCJH8DJOMCJGIWhHT+GOdZ09/bo8wvDeSjsoBkXIBEzHqQdpUdzbMb\nM1bkrwAMAZLBARIx60F6tN3qjVkAACAASURBVP4B/g2bvL/xI4DkVMa6n3a7LQIkamEJqexM\nzjvW53xmRUBytLoeY1FPH3NZBkjUwhJS7EqekzSI8+VxgGTvQBX1ndl9XBYCErWwhFT5C/4L\nW8X51AqAZG+89hGHuFTnhYBELSwhPVPu1arXXOEZ9TsCkr0B+vcg/+28EJCohSWkY7ezMn9x\nnnxVIC8hFVZIYzRHMYedFwIStbCExPlp5dSQm9J5oBVCSLtLq5CedFkISNTCFFKwFUJI0k+V\nZEcPutwhARK58IMU71QMIDmV+sNna9wWARK18IOUnJzcqUZM4/btbopo2A+QxAEStfCDJDe3\n7jFlsqvKT4Akzhekf7+etj3YMQCJmPUg1f1Om350IyCJ8wFpVFHG4l4PcgxAImY9SLErtOkc\nvLPBR2JIc7Uj5lODGwOQiFkPUoXH1Y+65rQpD0jixJAe1iA1Dm4MQCJmPUhvsbp9Ro7sXZO9\nBkjixJAaaJCuDW4MQCJmPUi5Y8orfwCl37wCSOLEkNpqkO4KbgxAImY9SDKlwxvW7885lwJI\n4sSQlsWpkOYEN0agkNaPe3cFcVVAohbYOxt+wbu/vZe25u9MX0ftJpdirMT4IAcKENLrsbLh\nThmkdQGJmp+QFnVp1rRp09sTS+cDpPQ1C1OM3FVHpkJ6J5GxWkt9vY6UunzZYeEKhAKDNEd7\nWDmCtDIgUfMP0iwWXYlViGctFpkPaen1jMW8SPs/Tj8zE9In6p9p0kHLvrPhEQ1SLdLKgETN\nP0gNHjrD43Zlf3pvICcQ8g/SrjLqf+8hxu6ulpmQrtP+Tl+zLKQ7tRtYhrQyIFHzD1KifE8U\nt4vzl18wHdJQ7b93yUxj91fNTEjR2u3uYFlIT/vzGhYgUfMPUonFMqaNnK8pZzok/b8322fs\n/qqZCam8drN7WxbSX4nqDZxPWhmQqPkHqVGHy7zGWM7nx5sO6XXtDzLBjCdJZkIapJ3AYYNl\nIUmLajFW/gvauoBEzT9I37L7ed+4vsMrNDEd0uZED+fmMSgzIaU/Kt/q+ImW/hjFP39THzAD\nEjU/D39/N5afuJexyptMhyTNKiv/RbZPM3Rv9cx9HWnVB1P+xeeRyIUnJLW0XZcDdeTP60gH\n53y+zshddVTg72wwJkAiZkVIF37/Tv7XDdhROLyzQQ2QqIUnpLHFGfuTv9Edb1r1ESBRC0tI\nU1ibz2RIM6JHA5I4QKIWlpBufI5nyZD467XDENKxbX4cigckamEJKW6FBmlZ0cINKWXJJmc0\nh/44LO19IoYVG0A+iAhI1MISUsJPGqQZxQszpPRnoxm7yX6mur3JESzyybvVl7Weow4BSNTC\nEtLdd19UIJ2se19hhvSSaqbKAe1S5oPMUeS/xCEAiVpYQlodXe151r1riZg/CjGktGIamnHa\nxZXMuQXEIQCJWlhC4r/eovw13fZboI5CAdJW3Uxf7eIXLpDcT03sLQ+QDgx/vP8qv2+rIEAi\nZkFInEubt5zggRcCkI7EamZGaRfnOTuqR32TWl5IP6rnaRjt/631GiARsxykS7cuDMJQqECS\nntA+5Ko/HUrTPqpXtaTyvOlP6hB5IO3TzncSu9bvW+s1QCJmOUi8wvhwgHT4XvkvvtwPtou/\nKd8RW21tyoeDPj9KHiIPpN76fdpb/t1UUYBEzHqQFtaZF8T77EIFkiStnDT7kONS6vSRX/n7\nLvQ8kO7WIb3i5xUJAiRi1oPUvD6LLl9VqXBDCr48kB7TIX1t3BiARMx6kJrceY8eIInLA+kb\nzVElAz/xC0jErAfJpU8DOXoX6pD2fjL0q3TCenmP2vVSHF29M+CR8wZIxCwOiQXy1eYhDmle\nkqyh5hbfK3p4HWlR/+cmHw90YE8BEjFAMqmAIaWUpp7KCu9soAZIYQjpU/2IwV8+1wQkaoAU\nhpBG6ZB+9rkmIFEDpDCENFtzFO375P6ARA2QwhDS8cYqpBd9rwlI1AApDCFJO9tFsiL9j3n4\nzbonG7Wd6bgISNQAKRwhSdLhjXkZHV0x/2v1LeMD7YsAiRoghSckD82pyFiE9uxptW0ZIFED\nJEDS2pTg+LzS27aFgEQNkJYG8m1jhRDSi04f/LN/oWRgkDb2uLvzT+S1AYmY9SBFx+slVHho\nZQCMCiWkdk6QltsWBgRJ+xAt7dtdJUAiZz1IfW5jtR95tC67o3PL4pFLAUnteYejZ+wLA4F0\nvJL2IdqNxPUBiZj1IC2vuk6ZrLv+L37qtjsASW1tUfXvv/b1zSY5PiMRCKQ1Osf3iesDEjHr\nQbr5a2069U7Ovw3oW/sKISRpWinGYnq5nhYlEEi2E3+9S1wfkIhZD1Lscm26tBjnPyYAkt7B\n76dvdVsUCKS0EhqklcT1AYmY9SBVfCxXnT5Vhmc/eCsgeS+ggw2fuj3R8hEgEbMepKGs7oCx\n7w26jfXhj7BZgOS9wA5/z21RseE48ofRAYmY9SDljFC+2JWVGHCJT/g2EEeAZGCARMx6kK7w\n3IMb1u8N+Pv6goQ0p2HiNf0Pef01OUCiBkjU/INU/qUtgRsKGtLX6lOIFsGfhycfIZnyrex6\ngETMepBuj2B13j1SQJAyK2oHtaYFs79q+QVpV6cSUbWmmzYGIBGzHiR++L2GLOKuqacLAtK/\n+sss/YLZX7V8gpR2o3p7Z5g1BiARsyAkuYNjbmVFHisASPv0jyq8Gsz+quUTpA+121vZrDEA\niZg1IXGeO7+69/eEmwdJ0j7nzX4NYne18glST/0udJ9JYwASMUtCurKqdwWW1LMgIG1Uzyr3\nehB7q5dPkAbop0ox64ADIBGzHqTs5b3KsmLJCy4H6ii4w997h7TvuTjwfbWXT5BWaJAeNmsM\nQCJmPUhJLPrBr88FrEju9H8+OsfP+lol6C76vBVBJ0P677+hiqNqu8waI8esK3bEs00f4uRl\n04c4zS+YPsa5cx4Xn/QM6Y6PJXV6LiVQSJev+CiH5/haJehy82EIrvz8a8gLn583bQxu2jU7\nhsg1fYicfBgiP3bD89+U4+vEPB5W+KVCoJDwFiHjwkM7YtZ7aMcXdWnWtGnT2xNLA5I4QKIW\nlpBmsehKrEI8a7EIkMQBErWwhNTgoTM8blf2p/cGcgIhQDI4QCJmPUiJ8j1R3C7OX34BkMQB\nErWwhFRisYxpI+drygGSODdIfz9zR7uvjB4DkIhZD1KjDpd5jbGczw/oxCfhC2l5EeUVpd4G\njwFIxKwH6Vt2P+8b13d4hSaAJM4VUk3q95P5FSARsx4k/t1YfuJexipvAiRxLpB26O9efcPY\nMQCJmAUhqaXtKqD32hlVfkPayoz6AIhLgETMqpCCKSwhZVbWIM03dgxAIgZIJpUPkC4cdT61\nxPeqo2SDxwAkYoBkUqZDOtq3GIt7+oBjwYpW190+5rjBowASMUAyKdMhPa3eA7UyeRRAIgZI\nJmU2JNvBheW+Vw0mQCIGSCZlCKTtc5Yf9fKruTqkjwwYRhAgEQMkkwoK0t6Phkw7JmX0jGKs\n8o+eV9E/Y85mBjEMIUAiBkgmFQykeUmykOs3D1allNzmcZ3j2jsZKhhwdmVRgEQMkEwqCEh7\ny6hGbkvS7nO8nNJojXJS2NKLAh6FFiARAySTCgLS58ylJ7ysdmTaWxPNOp2dPUAiBkgmFQSk\n0bqgROGbfrb8lYXPIxEDpDCClG6f0w/IRfVVJyW2eFr7h2ry86O5gEQLkMIFUuqAqyOqjtHf\n8ZNxhyroxeNdlaMJcz2tv7aYuorZz5AAiRwgmZR/kDq4fARiV/soVqT/Mfnh2zcLUz2u/7h2\np3VXsLfSZ4BEDJBMyi9IP2suYlJsC1L/ShetLzUy+Uso7AESMUAyKb8gfaAfXlhI3eBhbf2b\nA7hh/gVIxADJpPyCZDvgvYq6wTfa+u8HcMP8C5CIAZJJ+QVpdwnVRXX6hyAGxcrrP/ef/7fL\nzwCJGCCZlH8HG6YppwBK+sWPLTaMf+8vvI5EDJDCBZL09+CuI1J8r+YSThBJDZDCBlIgARI1\nQAIkQYBEDZAKI6R9094jH+YWBkjUAKkQQppdijHWdL8BQwASNUAqfJB2XKUe6e5gwBCARA2Q\nCh8k/TMSUQeDHwKQqAFS4YP0kv4mhr+DHwKQqAFS4YM0SXNULC34IQCJGiAVNKT09b8F+Bfv\nDdLh60Ufe/UrQKIGSAUMaU5lxkp+6NfO2vJ61G59Y8Zi+5PfVXf0/acHeHnrECBRA6SChaR/\nDvU7/3ZXS/CC7D+/pfw8dyftav69RrkFgz3+DpCoAVLBQnpSez5zm3+7qyV6Z8PsCozF9M2k\nXM1D2k3w+D18gEQNkAoWUnPtr7isf7urJYC0KUG92lGEa0mL1m5Cf0+/BCRqgFSwkNprf8V1\n/dtdLQEk7exArBLhWg7oB8t7ePolIFEDpIKF9KP2V/yeX3urJ4Ck+4zI8LqGo6rauh4PeAAS\nNUAqWEjSO8rn7XqSnsy4J4D0ooajAuVqtM+U33zM0+8AiRogFTAkafuUSev92VVHAkgfxag6\nhpGu5+u60SWf3O3xV4BEDZAKGlLgeYfURWUU3ZPyyE7J64m5AIkaIBVCSJ9pD+x6Bj8EIFED\npEIIST8l3XXBDwFI1ACpMEA6MvP9+U4P4+7SIF0d/BCARA2QQhRS2h/b7JCWVVIOu+2w/+55\nDdI9wY8CSNQAKTQhvZ3I2E2rtfmDlVQ3d9p/uauscrnIb8EPA0jUACkkIWkfObpae1vql/p7\nEzbZf/3nvUViGi42YBxAogZIIQlJfzeC9pGjMTqkJU4rHPf4+qrfARI1QAo1SOnjkzt9GKHR\neUxdMke7EEn82IQ/ARI1QAoxSGkNFTTaGxfYC+qidHUR6+r/lR3+YfIa4QqARA2QQgzSIOao\niH60YVtLxqK6ef7ivbwdnD72R+29fT9WkK+k1RHBuoBEDZAKHNLxXX7sqHSTZihJ/l/Cp/al\nu34hnw7yx6vlTRsqb63bWUq9qu6ClQGJGiAVMKSUJ2LZVUPE30LpXC0N0s0/j/nsYCDn/t5T\nRt3+AXl2lHZVsYK7MkCiBkgFCymzhfrH/DJ5T/XvSH5GCvAk+vq5uiL+laR++iPErd7XBiRq\ngJRfkH4e0HNS3mPS32t/yzH7qHu6PUl9BUl5aBYQpKG6Hvnp1XvaXFHB2cAAiRog5ROkl5W/\n2Vp73dfTTy/s+dQjHtvU7uryHTcrcwFBmqbLPSBJ+7R3RHg8WYMeIFEDpPyB9JPzCz9OfaxD\n2hDATgcEKa2eOl5vZX5lHcaie4heugUkaoCUP5B66Qes3T9TvrukuvyWQHY6sC8a23yXfH/0\nnPZw7vgfCzx/MtYWIFEDpPyB1Fl/jp/n//5nJsqLKwdyhxTwN/btXHWIuiogUQOk/IGkH2qu\nmXfNf8f0/4j6Wqpr+OpLaoBEzfKQUmuqkOYasrd6gEQNkKhZHpK0o2Px6HqzDNjVH+8qV3+Y\n+hQHkKgBEjXrQ5Iz4PuKbGeha6fMAhI1QKIWEpCMKKOc9mTre6lAIO005P8MXAMkYoBkYJv1\n151ekwoA0viyLOq+zUaPAUjEAMmfUrcJT1+8Q4f0pqRAypg5dJL+3vG0NRupp4P0IxdI2ovH\n1x82eAxAIhYakI4ObKtMzo7r1GFYhvepyZC2PRzJSgzzQGnN4OcmqQ+ramuQVsmzWftvlOeK\nT1cWjy/J2LU/BDKkMBdI5bWh3zV4DEAiFhKQ1jw1XoU0st/e1JG9c7xOzYWUVl/9Ux2R5xdv\nx8qLb1DehbBK/eajQcrCrNbq2olbJWmmOhcf0Ou6opwh2b7/5WmDxwAkYiEB6dfMPxVIUuv9\n8r1Pu83epiZD0s81nOD+lH6ltryVMr+jb8sntLue45Ha4nck6VZtTvQZvYByhpQeqw3yksFj\nABKxkIDEuQppXftc+Wef2d6mJkMaqP+f/l+el0e7AtvN7H/Z2mddWYsAxhTm8tAuWR0jTnyW\nB/8DJGKhBGlZN2V2yGRvU/nH/oly+y746DK/5GsVD43VPESmuy3vrZNJc1l6PlFb+uWFC/oz\npy4BjCnsCs9yXEhvoLzt9iOjx8g1+grzxnNMHyLriulDXOTZpo9x+bLn5X5D6q7MDp7sbSr/\nWNlAboPPqwuoQ9pXv7Z1X/6F5qRyrmPRufc69eqmLr3pIucfaiusNudm2cr5cdjE/eYOgSyZ\n/eAAFdJ69SHci3O9TeUf/9sgd+yUjy7w875W8dT0eJlDvb3uizO0gxDTHUv2VlYW3JzAIur2\nniGdOql8O3r0O4EMKSybnzb8Ot3LNX2EU/yK6UOczTZ9iHP8ouljZGV5XHzGX0gn2qTIT4Da\n7vA2ta1s2utIOz54/avjeRf/0zE+4rop2vyWb+Ydktppd0EzxisP767fuv9a5VK9o06bHP30\n5XF+nZ/IU3iLEDU8R9I6IS1vK0lZfEyfvUeGDsz1OjUbktcy9dPPZfSMYazMV/EapI5XqZOm\nPbWLrzrW36ic1bjEbHkubVvgowISNUDSeqa10gJ+fkJyx1HyZt6mBQbJ1nBVTLFoTU5D/ShE\ndW3SWF/pwPfT6qgLkvZsbxXFSo4I6MueJUCiB0gBVXCQtBOWMP29qw8yl4uNtHW+Ks1sTdSe\nXb0d4GiARA2QQgxSlAbkrjjlZ23bCYz1A+ED1FU2xNsdsWT99wG+axuQqAFSiEGqosnou6Jl\nqWo3R9vBKOchtr2ftL/DEeuoT/8ObDRAogZI5kDaMePjPw3fS6V3VRcJynsf3nMCs/zZG2/p\nr5//u5NjcYsR2jSSfAJK1wCJGiCZAul15ZHX88bvpyRl9o9lrOIcZbaiA0wF549QvKwti2Cx\nT6Rs1h7mPRDgaIBEDZDMgDRTf6Zvwp5K0u65y49K0q5dxyMckGY7r7BVOyb+9g7l1PxTiinP\npQJ9QQmQqAGSGZD0g2kNjdm51e+Pd/98xPfXyc+IkrRRKl33sHrC4w2fTNFPhz+/KmOx/fVD\n3tvff3W6h5d4aQESNUAyA5L+8k5VI3Yts5t8TbGvuixbXUS5eu1r++I3astekOfjhmnzx1b+\nuMeIsQGJHCCZAUk/VmbIpxr0QwoznZe11ZZdI/+vrP6LidqiOUYM6QiQqAGSGZBWq6/ysJ+M\n2LWbNSFJTV85JEnbJoyY+0GXZ/Rj4LGrZsyznUGhgbaolRFDOgIkaoBkBqQt6rGyBL/e5Lbo\njhJVeub52hdJqmw/olDzyMSizKXkPKsZ9LTMFiBRAyQzIOmv5HT2Yw8Wqlvckvc7Vu5yuHkq\nztURS3Ks1kxbkud7Y4ILkKgBkhmQ9A+s1vNjD+pqm4zL84uFDjfl3RyxYo7VtK/+s33PuVEB\nEjVAMgOSftSuEX0HjukvCj2hXTw84NrEht9p81PL2dyUcYfUzOkaPkpirNLMPFccXIBEDZDM\ngPSW9nc+kr4Dy3RIPdVLmS3VC9O032Vs0h8qNncQUt/fXdTlPCRpq9fSvxydGCBRAyQzIKXf\nofyd30N/HfR321GELupF7Uz57GrbG3/2qYcSauywH3io+nePa8q2Xku+/kADJGqAZAakrerp\nse6lf5zuAZuQ2C3KxVf0S6OaVrlL/SqYvX0bNhp4UPqjCWOJNySWTt6uLMS3UVADJGoWg/SQ\n5uAj8u13HONWT8wwxPmJ0PvOK+7d5Hh76sqXu3fqOeIf8iBK3z3esp+yxfGNf1OYAxI1QDID\nknZWLT+ORde0s5mqXFztDKnofs/bPKf9OtGfNzP0U7dYox7AqErYEJCoAZIZkPRnPB3It/9l\nm5o47Q5GvUuyvbt7vuu6h97u3Pdn+zvMGSvtBZqHfta2uFE7pF7U9zlTAYkaIJkBqYX2Bzsu\nbc0a2me805rqKt7RH3At6dXMfpe00GXVLRWUZYOdPr73FWkIpdf1LZpS7zEBiRogmQFpg/rY\nrvGUsoyVmULagYypvTo2b3hbxYiE5J3akttsTpJSpcOrttrX1JH+0soOif5UzHZucf0p2a0+\nNwAkaoBkBiRp81P1Gw3+SX1HT9xi8l5oZyaur92J6efFZ7FfHb9XfpBXcaW2zn79KyheHmSH\nRP8Oiu90R/oLxr4/NwtI1ADJFEjqyU/0+4z7qTuRWUHb4EP10vXahWprpHvUmfgD8sKPrtfP\nIsRe2Gc/0hdNf3OsdpPmTNA2/Mbn+oBEDZDMg6S/fa6GfeG+3nWqdfJ+Pp+9Ooxe6qXB2oU5\n0h59cR9JGme/F2KfSX+3tb2HdQb53yntzfrl75HvInuwPB8W9BggUQMk8yDdrf2ZN7ctS62l\nXCzhVdJR/exaA9VLx5VP78W+5jg811xKS7Q7uv24JGV9r1/4zv9/sN8/+HAjYTVAogZI5kGa\nof2Zf2lbpr/M+rDXjbQHXnGr9ItLho9dJ09W6FwavjNGn4sq11M5vVZWqnaWkxL049/+BkjU\nAMk8SNJrsfJ9yiv2ZfdrDEp73WjXdcqd0DtuSzMc90N6o7VfZJ2apnwbZew03/8+gQZI1ADJ\nREjS5smfb3Yse1hjUF6e/aFxyeovHc6zVdqkZ9/I+yrpgmg3SPortFmnpLW97u9l5ltXAYka\nIJkJybX3NQadJelbdaaFt3e7bZ38kctpWnd3bxDpDEn/XnG8aZUaIFELCUgZ6mupVVKkTP3I\ntZfHZCOVo3G9XBb96XKP1Oyb8T/OHvf9WUAiBkjUQgKSdPyD1ve+cVCSduog+nncdJ72S5c3\nfae43CPpb+Wru4XyTxNUgEQNkPIRkq39+ttRB9kWpDl9tDXzBu2X9V02acU8dHOGZHKARA2Q\nCgCSdLsGYYV2adltMbHNf7P9brSupLjLCYVS1O9tYdH6//SWUP5tggmQqAFSQUDaoJ62Wz8w\nvk45zT27yvY47RqbkuYu9zdHblVWmrbmg09edECK7pJC+ucJOECiBkgFAUlKeaPtM7bzsLbW\nUKjnD0ob3dHxbMjt1Fw/v/+lekLvyc6P7u4O9NthaQESNUAqEEjO6e9OvUWePVDTGUlb5bcZ\nHzS7ofUK5/W1dxrZmu/xOo0KkKgBUoFD0k/ufac8+7TLsYTWym+fVGddPhW+pobTSmOowwQU\nIFEDpAKHpJ8E74NFd5Z2fQvDu/IvF2uzFZ2eL/1ayXmlydRhAgqQqAFSgUM6fp8Cov08Bw7t\njKpxyjOiN/VFm+yrp17j7Ki0h9PuGxggUQOkAockSTP79v/e9lTJqf7yrwbo8+qzpINLlh6W\nPndeJfF7+iiBBEjUAMkCkJQO5HGkfk/ZMH1+prSlR7UoxqJKOq9xozHfy+c9QKIGSBaBdDQy\nD6TGkv3kxWzu6mJuv60WXemVI9q2v73a40Pa2Yr8DZCoAZJFIDl/D5LebfLSFO1ckyUO3uz2\ny2Qp0/bu7xGqq3/9HI8UIFEDJItA+r1phDukZpK0/D31iHjsl/tdfxXZ7ZD9YxQrtUUP+jce\nLUCiBkgWgJT5Scs6sXnukNiTx5Q3PRRp0qL7GinF9VfqO/N0SPpZWqNS6QOSAyRqgGQBSN3d\n7m60SdzvL2lT9Yx2Lu95uFbdTIeknwOcmXHgAZCoAVLBQ1ridk8UqT0vij9QVrvc6nP5DmiJ\n011W3Hp1Ox3SJG1hRTPedQdI1ACp4CENzvuoTq19lH225Z7+MfYLN+jngdAhHdOOQ3wpGCHg\nAIkaIFkXEnM64u38yK6Vvp3tqN2ux4tH1jDFESCRA6SCh+T+0M5X9+rbOZ385Ch5MP8CJGqA\nVPCQpDbeyDQv7mmp7RPqOIsQNUCiFtKQjldXfeR5HYmxAcf+aJ5nYbWD+naARA2QqIUipMX9\nn3x+snLc4BddiHJcrnyc80kZFtkOybEq9mW32d/CAEjUAIlaCELSTr0Q81ymNF8nMnLcKw8X\nZbH32dHcL0kNtLmqW260LYz4w3YNgEQNkKiFHqS5NhijpX/1V2Drrevi9ijuO0n/EF+jLV+8\n8oC+sJntKgCJGiBRsyakzf0f7bfBywrdbFiin93bT5+NY27VkqpqM9Xlh322F5JK2a4CkKgB\nEjVLQppTRMEx3fMKdzq4lPyrirsgvajMsnkXVrBdhQhSxgFft5AUIFEDJNMgpWqfGL/K84fB\nk5xo3FPRC6QkKc+dFGOdbFfhHdKu5CKs4ge+/9l8BkjUAMk0SAv0P/yvPf1+t8sdz01eIPWQ\nbsizrJj922C8Qjp2q7rmOC+/9iNAogZIpkGarf/lT/H0+33ONiLez8OlvPKj3KdHP1Iv6QfE\nlS/oq2A/aOcdkn4KyZLpXn5PD5CoAZJpkP7V//w9f1frrU5q6km93RwlrV1TX52peF8xxq4d\nqs7HLZ//4Q9OOLxCekm/ls1efk8PkKgBkmmQpFfVv+benlf4w/Hun4gi17y14emG1e0flKj4\n5u7NY20Xbh++Jl0aV4Kxsu5fp+QV0jD9evf5upU+AyRqgGQepIwJtYvWfPe4lzX+7dcy+ZvR\nVYto91v10192ukN6zeVcQfV2SdLhn3/Jc3oTr5A2FFG3a+nrRvoOkKgBknmQSP2ta+nm/kWx\nTj3geVPvR+0mKgf7qu+g3QJRgEQNkAoE0kH73Je2x3feHbGIPwY+8vwq5+33zPjsT9HrSJuG\n9fnYiBN0ARI1QMp/SPt7JLKrR+pn8/5GAMie8uwpdrxylpS6RasPTZM+SpQXdD2PdzYQAyRq\noQQpUz3XN+uqnfVnr+iuyKWiW6R31JkuK7WXaUcBEjFAomZBSCc/aH3va57eqPOjDqPcd+rF\nelRIbMIB7SACa6tNrgEkYoBEzXqQcloqf+uV3c6RdWzUPXfY32UXN3K7tOlBDye089LopfqM\n9hITKwJIxACJmtUgpb7bUPtj7+Sy+HhTVxpFh5dzuRydxAT93EefaalNagESMUCiZjFIe6vb\n/vjLOi/e/7y7Dbcz58f/pM/UuEaduLxl9fHPbGstTVSnUwGJGCBRsxikJ+x//UlOS0fYTq/l\n/fiC7Ux2P3yhrHvHGZlEogAAEYZJREFU/Kts4Ire8Faa/gbWqI+keVUZKzYMn0eiBkjU8hvS\n6f+ElbHLuH/uxEWStvBL+8KSdbxK0rrundLqdNKesb2Hti4e23CBcgX6J/vulmfT//j50H8X\nfdwKA5IhmT5Gjukj/MezTR/i5GXThzjNL5g+xrlzHhefNAvSpWxRl+xnRY1XPuHa6Ii6tKHd\nyaPVGa2q2vVlaRP9U0u9bcPkXBHeCiPK5aYPkZ0fQ+SaPsSVfBiC55g+Ro7nIS6bBUn8oMp2\nxscS7bVTLihfWC4NsT+gK7nlKncx3jrmfLWD1EVxv9ou46EdNTy0o2at50if6gxmfKXPbJSk\nR2w2ruq6XapNdJTkcrXpj8qLin1ovwxI1ACJmrUg2c4QtHKIPrNQmmazEbdUXmEWEVJftyte\nNW7yP45LgEQNkKhZC1Ka9g3lt2zWvp+FRXy2qYNOI2KilD62VYs7ozzLca2V8r7TpW3rP/iN\nx2EAiRogUbMWJGl1NZlB7b9aOUzYjuNVkTLuUqaU9zM0V65qqjr7hsvVZ/ylnlAFkKgBEjWL\nQZLS5n2yIF3y8D6F6IN5z8/gpcRN8hWlasclYp0/rz5GXnb774BED5CoWQ2SdC5nw/x/Snng\n8XgrDwvzVrx4yzXK9SzTL09yXPXH6oJKewGJHCBRsxykDcrX6MV7EFK1AQnSbv16PECqrC0Z\nCkjkAIma1SAdvNabkLgEiqOoCdNT1CtK1c7eELvJftVp+iqdAYkcIFGzGqRJDhNJ2vdINPZK\ny0tXaScN0t5YNNjpurV3rLK+gEQOkKhZDdKrTiLWKI/yGm37wU9IrJh2/v2fH6n/0Ezn6+6h\n/jZuNSCRAyRqVoM03uEhNj1j+Re/ZkoP+wuJDfR83al3yb8rOgFH7egBEjWrQUpxfCF5nXXq\nkql+O2KNU6Vj7z/W+ZMM92v/ccSEbRIg0QMkalaDJNVwukuakLZi9jbawTrXKv+pnl2/hZdz\nTAISNUCiZjVIs50/2xqrHLCO8WxFnP59fe94HgSQqAESNYtBWlM0EDZeu9PzKIBEDZCoWQxS\nJ0MdsUaeRwEkaoBEzWKQGmkAYksEKUh/PPi851EAiRogUbMYJNv76R4PEtIb6ltWK6Z4HgWQ\nqAESNYtB+k6H0HJStYgSncgfLHc+PVdsAqsyXlrfumyFTtu8jAJI1ACJmsUgSbod+clNqiRp\n37fHEn07qtTcPvvsvNW+dhqQqAESNatBekDT8Ix6ob12wfeHYm/K+MumTTljULOt4kEAiRog\nUbMapN/V49+ltK/6ukfH4fN7J2ZL0vKGUZH1h3xyh3q5obdv+9MCJGqARM1qkGYqJ3is/od0\naGyPlz+0QfIhKV77zFHqYUn6U1+0QDgIIFEDJGoWg7RWe0H24w3lfT6c04sc9bPTOezm6Esn\neR9CAiR6gETNYpC0TzqwG/x4g91Dztv/ri/8QTgKIFEDJGoWg3S/5oD8ZXxyV2tbHnzjwfaf\nZkrac6Q64q+BBSRqgETNYpC6+iFIr4q64W71baqtMrcr742ou148CiBRAyRqFoM02J1JUhFf\nkHqoG+rnkfxEylzx+RLxMTtAogdI1KwFKSXP6R+7dXBf4la09vYF/Ux4DQZ8sNf3TgMSNUCi\nZi1I0/NCiff1fOlldUvHux+SFvvcaUCiBkjUrAVpog80nnpA3bKFY0HFI752GpCoARI1a0Fa\nFwCkjtqWjnM9KO9zEAdI1ACJmrUgSaX9h/SVtuW6NpX1z5ezz33tNCBRAyRqFoP0jN+Oejg2\n3qwvwru/DQuQqFkLUqa/Z1WNcbn36a4u6+hzpwGJGiBRsxak5X4YeiGBscpzlK3+WbAhU938\naP8EVuz5wz53GpCoARI1a0H61Lcfe3sP/bxGebvqEeWEKQ21sxRLmTvznBTSQ4BEDZCoWQvS\nbD8gHZLXX/lKjw+7qJdqpvqx04BEDZCoWQtSqqdvGPNcdcn+UXS1L/3YaUCiBkjUrAVpp893\n1tmrdFha4Xz5LT92GpCoARI1a0GaRnbEWH+pv/PFz/zYaUCiBkjUrAXpcz8gNbR9ClCt8gE/\ndhqQqAESNWtBWu0XpAnajPJw8NoV/uw0IFEDJGrWgjTLD0gDpLT66syMXz+dL/5ErHuARA2Q\nqFkL0hSKoMhrlJ81j0jSzk7Fo2pN93+nAYkaIFGzFiTSu7+b7h9we5NBh7Qt/Lsr0gMkaoBE\nzVqQpG6+HSWsC3qnAYkaIFGzGKQ0H5+HrXhj9y3B7zQgUQMkahaDJIm/gGKGMTsNSNQAiZrV\nINUSQvrdmJ0GJGqARM1ikH4UP7IbZMxOAxI1QKJmMUi1xZB6GbPTgEQNkKhZC1JmpBjSeGN2\nGpCoARI1a0H6U+yo1lFjdhqQqAESNWtBWiFiFN3KxxfxkQMkaoBEzVqQhOdsuMewnQYkaoBE\nzVqQ1oogVTdspwGJGiBRsxYkSQTpdsN2GpCoARI1a0H6WwTpY8N2GpCoARI1a0EaL3DU27id\nBiRqgETNWpAe8abolg99fAmfXwESNUCiZi1IjbxBmmfoTgMSNUCiZi1I93lxFL3b0J0GJGqA\nRM1akJ7zAulVY3cakKgBEjVrQXrfA6KyEde86+vblf0MkKgBEjVrQfokr6PrJIMVSYBED5Co\nWQvSN3khdTVhpwGJGiBRsxakt/I4ivvFhJ0GJGqARM1akOq4O0qYZMZOAxI1QKJmLUjXuDl6\nZI8Z+wxI5ACJmrUgVXN1ZNBJg/IESNQAiZq1ICW4OOpuxg4rARI1QKJmOqSz4zp1GJZBhBTj\n7CjSjP1VAyRqgETNdEgj++1NHdk7hwYpyhlSphn7qwZI1ACJmtmQpNb75XuldptpkJxPWDzH\njN3VAiRqgETNbEjr2ufKP/vM9htSMzP2Vg+QqAESNbMhLeum/BwyWf6x5x25lCxRDkjRwvWC\n7MolM69dLYdfNH2MXNNHyOI5pg9x0fwhLvFs08fI9jKEYZC6Kz8HK5BWNpDbIFzbDikiuFER\nskb2gwPBQlqvPrR7ca7848xOuYyTomwnWo0QrhV0l86ae/1yl/kp08fIMX2Ek/yK6UOcvmz6\nEGd5luljXLjgcfFpoyCdaJMiPzFqu8N2mXYWoQwTHsQ6hedI1PAciZrph7/H9Nl7ZOjAXCKk\nyqqjq0zYUecAiRogUTMd0vkJyR1HOa7O55/wqrgNhu+ke4BEDZCoWestQsoN4mcM30n3AIka\nIFEDJHMCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQ\nqAESIAkCJGqABEiCAIkaIAGSIECiBkiAJAiQqAGSOW185598HtGUZr9zvqBvghGNnlrQt8CI\nDr2zsqBvQv5DmtVgaT6PaEr9G5ws6JtgRI2eKOhbYESbGnxU0DcBkAIMkCwUIIVugGShACl0\nAyQLFZaQECqUARJCBgRICBkQICFkQPkL6ey4Th2GZeTrkEb0v/efSH5tD+d9Wst1dOyG+9Ti\n+br5obEb21urLbLYf438hTSy397Ukb1z8nVMAxrw2v5jY5/I4t0XSpL0P8duuE8tnq+bHxq7\ncVl5a84/HY9Y7L9GvkKSWu+X/w+j3eb8HNOAzrx7lPPM1nt4h03qZdtuuE8L+Gb6zMfND5Xd\nUHpjps/dyedblK+Q1rXPlX/2mZ2fYxrVrjbS5dYTX+z+dpp9N9ynBXwLfeXr5ofIbij99sxl\nn7uTzzcpXyEt66b8HDI5P8c0qDMvTOGnnnx3155hT52z7Yb7tCBvHyFfNz9EdkMu+5lVvncn\nn29T/kLqrvwcHAL/qdxL7fVJrjaX9dgy2264TwvqxvmT4OaHzm782t32DMg6/zXyFdJ69V73\nxbn5OaYhbe28yD7/wkzbbrhPC+rW+ZX3mx86uzF4un3WMv818hXSiTYp8l1y2x35OaYR/dv5\nb2VyaNJlzi90WGnbDfdpQd9MH/m6+SGyG/I+tNvOLfdfI38Pf4/ps/fI0IG5+Tpm8F3q+Z1y\nyDXrTOfx6UdHPZ1l3w33qbXzefNDYzfk/19rfZoTdid/y19I5yckdxx1wvd61mqr7SXAvUOS\nnxh13LEb7lOL5+vmh8hu8DVtVCXW+q+BtwghZECAhJABARJCBgRICBkQICFkQICEkAEBEkIG\nBEgIGRAghVDJ8X6snP5KnfiEmv1S/B6lUQ2/N0GAFEr5A+mPpKi2o0Z3iktY5Htde1uUP4fx\no/y9XQiQLNclwe/8gHS8dNIGZbqrconjHkfwOM5E/DkEGv7l8qVZja5KbDCL86alspWLjcpf\nyRlWOe6W5X2inddqfOfiSrfKdyYtE4vcNNV5M547vFJc3bmdZEjHelSJu/rRXc6bNb9je4v4\nMp0y5dklzRKK1BmXy/mrbKb2y1/GZTot1kbIM87SxkXLPJrC72eMNdAe2tm3cLpyJAiQ8qPZ\nrMOihQ+wRfxjtly+eDjiJf42e+znGVVuc7mPubN+nfE/8VUxdy5Y/hx7z2kzPoY9vnxW7Rry\n2reX++LXr+uWdf5Wmbur3rrg4MyoZM7nRzT7/ueebCDntZKuOFZwLNZGcB9ncWTLb6ZWK5+e\n0pZt2qlCcmzhuHIkCpDyo7H3Xeb8dHQXLkX3ki++x7bkXl1X/r/7tcwF0j1snvyzwbUX5Z9t\nSlxwbJZboY68dlpMPD/NXpV/eWBUmstmv8k/W5bivGZlZdNHYv7LjbzbaQX7Yn0E93Hq3yCr\n2xA7kT+j/DkokJy3sF05EgVI+VelZpw/WDaH84Z1+CE2QFlU2xVStPzAT2LPZ8l9xn53bHaY\n9VHmGsfz7NJVf3E/09Q9RZWf3SJ4GlOY8i/YovOsreP3jsXaCO7jZLIX9TVtkJy3sF05EgZI\n+dHJATWLx8Wxppx/zVbxg+xdvoGNVX7xiCuk8vKPzUzvO8dmG9hI5fft5bU3XsOSHvv2istm\nVZWfMoKNbIQyt4RN4UWbOX7vtFgdwX2czdq1cwck5y2q2n+BBOEfKD+6L3b0+p27KsiQzhbr\nzcdEHOHrlecmnD/qCqkqVw5Bd/9TTXJstl77U2+nrH1l5aB6rMlF982Uv/VNbIgyt5h9wW9M\nuGD/vdNibVW3cTazwfqaNkh5tgAkX+EfKB9KY33ln9nxMiSeXCH31rs438sGKb+pmxfS/1jX\nPJvt1x583WRb+zM2w30z5W/9uLbp52wZH8HGa79cW2O902InSI5xJG3uUKYdUp4tAMlX+AfK\nh3ax4fLPT9nt8s8FbD6bKt+vlGgkX1jP8kLit5VQvsRsxuBsx2Y5pWvJT4x2RcbzTcnKWa33\na/dnLpspf+v1Sin3Qw8WO81PViy6UFm6o9LVJ5wWO0FyjMNvTDqn3cgeLFs72OC+BSD5Cv9A\n+dDlyhUXrHm5ZYvEX8/xy0nViyjn7ujHui378vo7PED6LabuVz8PienuvNmbrO3cj6s2jOfH\nStSfuvzbxnG782ym/K0vimz6/U+Ps9HyxU0V2d3D3u4QXX6T82JnSPZx+OLIhlMn31AujQ9l\nw+eqkNy3ACRf4R8oP9rUuFjZXmcWli6fwnkv1lFZdLF36aLNNnZOcF5N+6Plv9+bWKTW6CvO\nm115rVxsvfn9Yjjf1r5cbPm2G/Jupv6tL2sWH3fzl+pSacRNJRLrvaW+kmpf7AzJMQ5f0rho\n6Uf2cJ56c9GbtRdk3bYAJF/hH6hAa1mhoG8BMiZAKqAmPCo/HTlV/KGCvh3ImACpgJrJWi2Y\ndVvkyoK+HciYAKmg+vqW+GJNl/KlzN6nfmwd4GbIrACpgDu7w54/ZwcNcDNkVoCEkAEBEkIG\nBEgIGRAgIWRAgISQAQESQgYESAgZ0P8BhPIas/LuS0MAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>avg_reads_noCorrection</th><th scope=col>avg_reads_WASPCorrection</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000064726_BTBD1  </td><td>1069.5882</td><td>1192.2647</td></tr>\n",
       "\t<tr><td>ENSG00000148925_BTBD10 </td><td> 105.1176</td><td> 171.0882</td></tr>\n",
       "\t<tr><td>ENSG00000176595_KBTBD11</td><td>1363.9412</td><td> 839.2059</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3  3\n",
       "\\begin{tabular}{lll}\n",
       " feature & avg\\_reads\\_noCorrection & avg\\_reads\\_WASPCorrection\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000064726\\_BTBD1   & 1069.5882 & 1192.2647\\\\\n",
       "\t ENSG00000148925\\_BTBD10  &  105.1176 &  171.0882\\\\\n",
       "\t ENSG00000176595\\_KBTBD11 & 1363.9412 &  839.2059\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3  3\n",
       "\n",
       "| feature &lt;chr&gt; | avg_reads_noCorrection &lt;dbl&gt; | avg_reads_WASPCorrection &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000064726_BTBD1   | 1069.5882 | 1192.2647 |\n",
       "| ENSG00000148925_BTBD10  |  105.1176 |  171.0882 |\n",
       "| ENSG00000176595_KBTBD11 | 1363.9412 |  839.2059 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature                 avg_reads_noCorrection avg_reads_WASPCorrection\n",
       "1 ENSG00000064726_BTBD1   1069.5882              1192.2647               \n",
       "2 ENSG00000148925_BTBD10   105.1176               171.0882               \n",
       "3 ENSG00000176595_KBTBD11 1363.9412               839.2059               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>avg_reads_noCorrection</th><th scope=col>avg_reads_WASPCorrection</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000255423_EBLN2</td><td>152.3824</td><td>102.1765</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  3\n",
       "\\begin{tabular}{lll}\n",
       " feature & avg\\_reads\\_noCorrection & avg\\_reads\\_WASPCorrection\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000255423\\_EBLN2 & 152.3824 & 102.1765\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  3\n",
       "\n",
       "| feature &lt;chr&gt; | avg_reads_noCorrection &lt;dbl&gt; | avg_reads_WASPCorrection &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000255423_EBLN2 | 152.3824 | 102.1765 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature               avg_reads_noCorrection avg_reads_WASPCorrection\n",
       "1 ENSG00000255423_EBLN2 152.3824               102.1765                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>avg_reads_noCorrection</th><th scope=col>avg_reads_WASPCorrection</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000175279_CENPS</td><td>1.529412</td><td>6.5</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  3\n",
       "\\begin{tabular}{lll}\n",
       " feature & avg\\_reads\\_noCorrection & avg\\_reads\\_WASPCorrection\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000175279\\_CENPS & 1.529412 & 6.5\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  3\n",
       "\n",
       "| feature &lt;chr&gt; | avg_reads_noCorrection &lt;dbl&gt; | avg_reads_WASPCorrection &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000175279_CENPS | 1.529412 | 6.5 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature               avg_reads_noCorrection avg_reads_WASPCorrection\n",
       "1 ENSG00000175279_CENPS 1.529412               6.5                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>avg_reads_noCorrection</th><th scope=col>avg_reads_WASPCorrection</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000204577_LILRB3</td><td>3480.118</td><td>2780.618</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  3\n",
       "\\begin{tabular}{lll}\n",
       " feature & avg\\_reads\\_noCorrection & avg\\_reads\\_WASPCorrection\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000204577\\_LILRB3 & 3480.118 & 2780.618\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  3\n",
       "\n",
       "| feature &lt;chr&gt; | avg_reads_noCorrection &lt;dbl&gt; | avg_reads_WASPCorrection &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000204577_LILRB3 | 3480.118 | 2780.618 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature                avg_reads_noCorrection avg_reads_WASPCorrection\n",
       "1 ENSG00000204577_LILRB3 3480.118               2780.618                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>avg_reads_noCorrection</th><th scope=col>avg_reads_WASPCorrection</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000164442_CITED2</td><td>1992</td><td>1263.382</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  3\n",
       "\\begin{tabular}{lll}\n",
       " feature & avg\\_reads\\_noCorrection & avg\\_reads\\_WASPCorrection\\\\\n",
       " <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000164442\\_CITED2 & 1992 & 1263.382\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  3\n",
       "\n",
       "| feature &lt;chr&gt; | avg_reads_noCorrection &lt;dbl&gt; | avg_reads_WASPCorrection &lt;dbl&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000164442_CITED2 | 1992 | 1263.382 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature                avg_reads_noCorrection avg_reads_WASPCorrection\n",
       "1 ENSG00000164442_CITED2 1992                   1263.382                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Checking the genes that went to zero before allowing for multimapping:\n",
    "to_plot[feature %like% \"BTBD1\"]\n",
    "to_plot[feature %like% \"EBLN2\"]\n",
    "to_plot[feature %like% \"CENPS\"]\n",
    "to_plot[feature %like% \"LILRB3\"]\n",
    "to_plot[feature %like% \"CITED2\"]\n",
    "\n",
    "### It is better, but how should we use the featureCounts settings?\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 14  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000064726</td><td>ENSG00000064726_BTBD1      </td><td>15:83016422:83067354             </td></tr>\n",
       "\t<tr><td>ENSG00000234438</td><td>ENSG00000234438_KBTBD13    </td><td>15:65076816:65079939             </td></tr>\n",
       "\t<tr><td>ENSG00000273645</td><td>ENSG00000273645_KBTBD11    </td><td>CHR_HSCHR8_8_CTG1:1973878:2006936</td></tr>\n",
       "\t<tr><td>ENSG00000187715</td><td>ENSG00000187715_KBTBD12    </td><td>3:127915232:127987671            </td></tr>\n",
       "\t<tr><td>ENSG00000219433</td><td>ENSG00000219433_BTBD10P2   </td><td>6:149794763:149795746            </td></tr>\n",
       "\t<tr><td>ENSG00000257675</td><td>ENSG00000257675_BTBD10P1   </td><td>12:52748776:52750127             </td></tr>\n",
       "\t<tr><td>ENSG00000148925</td><td>ENSG00000148925_BTBD10     </td><td>11:13388001:13463297             </td></tr>\n",
       "\t<tr><td>ENSG00000253696</td><td>ENSG00000253696_KBTBD11-OT1</td><td>8:1971397:1976478                </td></tr>\n",
       "\t<tr><td>ENSG00000176595</td><td>ENSG00000176595_KBTBD11    </td><td>8:1973878:2006936                </td></tr>\n",
       "\t<tr><td>ENSG00000204347</td><td>ENSG00000204347_BTBD17     </td><td>17:74356416:74361946             </td></tr>\n",
       "\t<tr><td>ENSG00000233436</td><td>ENSG00000233436_BTBD18     </td><td>11:57743514:57751781             </td></tr>\n",
       "\t<tr><td>ENSG00000138152</td><td>ENSG00000138152_BTBD16     </td><td>10:122271306:122338162           </td></tr>\n",
       "\t<tr><td>ENSG00000222009</td><td>ENSG00000222009_BTBD19     </td><td>1:44808482:44815585              </td></tr>\n",
       "\t<tr><td>ENSG00000151136</td><td>ENSG00000151136_BTBD11     </td><td>12:107318413:107659642           </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 14  3\n",
       "\\begin{tabular}{lll}\n",
       " GeneID & feature & hg38\\_location\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000064726 & ENSG00000064726\\_BTBD1       & 15:83016422:83067354             \\\\\n",
       "\t ENSG00000234438 & ENSG00000234438\\_KBTBD13     & 15:65076816:65079939             \\\\\n",
       "\t ENSG00000273645 & ENSG00000273645\\_KBTBD11     & CHR\\_HSCHR8\\_8\\_CTG1:1973878:2006936\\\\\n",
       "\t ENSG00000187715 & ENSG00000187715\\_KBTBD12     & 3:127915232:127987671            \\\\\n",
       "\t ENSG00000219433 & ENSG00000219433\\_BTBD10P2    & 6:149794763:149795746            \\\\\n",
       "\t ENSG00000257675 & ENSG00000257675\\_BTBD10P1    & 12:52748776:52750127             \\\\\n",
       "\t ENSG00000148925 & ENSG00000148925\\_BTBD10      & 11:13388001:13463297             \\\\\n",
       "\t ENSG00000253696 & ENSG00000253696\\_KBTBD11-OT1 & 8:1971397:1976478                \\\\\n",
       "\t ENSG00000176595 & ENSG00000176595\\_KBTBD11     & 8:1973878:2006936                \\\\\n",
       "\t ENSG00000204347 & ENSG00000204347\\_BTBD17      & 17:74356416:74361946             \\\\\n",
       "\t ENSG00000233436 & ENSG00000233436\\_BTBD18      & 11:57743514:57751781             \\\\\n",
       "\t ENSG00000138152 & ENSG00000138152\\_BTBD16      & 10:122271306:122338162           \\\\\n",
       "\t ENSG00000222009 & ENSG00000222009\\_BTBD19      & 1:44808482:44815585              \\\\\n",
       "\t ENSG00000151136 & ENSG00000151136\\_BTBD11      & 12:107318413:107659642           \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 14  3\n",
       "\n",
       "| GeneID &lt;chr&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000064726 | ENSG00000064726_BTBD1       | 15:83016422:83067354              |\n",
       "| ENSG00000234438 | ENSG00000234438_KBTBD13     | 15:65076816:65079939              |\n",
       "| ENSG00000273645 | ENSG00000273645_KBTBD11     | CHR_HSCHR8_8_CTG1:1973878:2006936 |\n",
       "| ENSG00000187715 | ENSG00000187715_KBTBD12     | 3:127915232:127987671             |\n",
       "| ENSG00000219433 | ENSG00000219433_BTBD10P2    | 6:149794763:149795746             |\n",
       "| ENSG00000257675 | ENSG00000257675_BTBD10P1    | 12:52748776:52750127              |\n",
       "| ENSG00000148925 | ENSG00000148925_BTBD10      | 11:13388001:13463297              |\n",
       "| ENSG00000253696 | ENSG00000253696_KBTBD11-OT1 | 8:1971397:1976478                 |\n",
       "| ENSG00000176595 | ENSG00000176595_KBTBD11     | 8:1973878:2006936                 |\n",
       "| ENSG00000204347 | ENSG00000204347_BTBD17      | 17:74356416:74361946              |\n",
       "| ENSG00000233436 | ENSG00000233436_BTBD18      | 11:57743514:57751781              |\n",
       "| ENSG00000138152 | ENSG00000138152_BTBD16      | 10:122271306:122338162            |\n",
       "| ENSG00000222009 | ENSG00000222009_BTBD19      | 1:44808482:44815585               |\n",
       "| ENSG00000151136 | ENSG00000151136_BTBD11      | 12:107318413:107659642            |\n",
       "\n"
      ],
      "text/plain": [
       "   GeneID          feature                    \n",
       "1  ENSG00000064726 ENSG00000064726_BTBD1      \n",
       "2  ENSG00000234438 ENSG00000234438_KBTBD13    \n",
       "3  ENSG00000273645 ENSG00000273645_KBTBD11    \n",
       "4  ENSG00000187715 ENSG00000187715_KBTBD12    \n",
       "5  ENSG00000219433 ENSG00000219433_BTBD10P2   \n",
       "6  ENSG00000257675 ENSG00000257675_BTBD10P1   \n",
       "7  ENSG00000148925 ENSG00000148925_BTBD10     \n",
       "8  ENSG00000253696 ENSG00000253696_KBTBD11-OT1\n",
       "9  ENSG00000176595 ENSG00000176595_KBTBD11    \n",
       "10 ENSG00000204347 ENSG00000204347_BTBD17     \n",
       "11 ENSG00000233436 ENSG00000233436_BTBD18     \n",
       "12 ENSG00000138152 ENSG00000138152_BTBD16     \n",
       "13 ENSG00000222009 ENSG00000222009_BTBD19     \n",
       "14 ENSG00000151136 ENSG00000151136_BTBD11     \n",
       "   hg38_location                    \n",
       "1  15:83016422:83067354             \n",
       "2  15:65076816:65079939             \n",
       "3  CHR_HSCHR8_8_CTG1:1973878:2006936\n",
       "4  3:127915232:127987671            \n",
       "5  6:149794763:149795746            \n",
       "6  12:52748776:52750127             \n",
       "7  11:13388001:13463297             \n",
       "8  8:1971397:1976478                \n",
       "9  8:1973878:2006936                \n",
       "10 17:74356416:74361946             \n",
       "11 11:57743514:57751781             \n",
       "12 10:122271306:122338162           \n",
       "13 1:44808482:44815585              \n",
       "14 12:107318413:107659642           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 5  7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene stable ID</th><th scope=col>Transcript stable ID</th><th scope=col>HGNC symbol</th><th scope=col>Transcript start (bp)</th><th scope=col>Transcript end (bp)</th><th scope=col>Chromosome/scaffold name</th><th scope=col>Strand</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000064726</td><td>ENST00000261721</td><td>BTBD1</td><td>83016422</td><td>83067354</td><td>15</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000064726</td><td>ENST00000379403</td><td>BTBD1</td><td>83017961</td><td>83067151</td><td>15</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000064726</td><td>ENST00000559652</td><td>BTBD1</td><td>83018740</td><td>83056479</td><td>15</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000064726</td><td>ENST00000558344</td><td>BTBD1</td><td>83020746</td><td>83030766</td><td>15</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000064726</td><td>ENST00000560015</td><td>BTBD1</td><td>83029543</td><td>83041821</td><td>15</td><td>-1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5  7\n",
       "\\begin{tabular}{lllllll}\n",
       " Gene stable ID & Transcript stable ID & HGNC symbol & Transcript start (bp) & Transcript end (bp) & Chromosome/scaffold name & Strand\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ENSG00000064726 & ENST00000261721 & BTBD1 & 83016422 & 83067354 & 15 & -1\\\\\n",
       "\t ENSG00000064726 & ENST00000379403 & BTBD1 & 83017961 & 83067151 & 15 & -1\\\\\n",
       "\t ENSG00000064726 & ENST00000559652 & BTBD1 & 83018740 & 83056479 & 15 & -1\\\\\n",
       "\t ENSG00000064726 & ENST00000558344 & BTBD1 & 83020746 & 83030766 & 15 & -1\\\\\n",
       "\t ENSG00000064726 & ENST00000560015 & BTBD1 & 83029543 & 83041821 & 15 & -1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5  7\n",
       "\n",
       "| Gene stable ID &lt;chr&gt; | Transcript stable ID &lt;chr&gt; | HGNC symbol &lt;chr&gt; | Transcript start (bp) &lt;int&gt; | Transcript end (bp) &lt;int&gt; | Chromosome/scaffold name &lt;chr&gt; | Strand &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| ENSG00000064726 | ENST00000261721 | BTBD1 | 83016422 | 83067354 | 15 | -1 |\n",
       "| ENSG00000064726 | ENST00000379403 | BTBD1 | 83017961 | 83067151 | 15 | -1 |\n",
       "| ENSG00000064726 | ENST00000559652 | BTBD1 | 83018740 | 83056479 | 15 | -1 |\n",
       "| ENSG00000064726 | ENST00000558344 | BTBD1 | 83020746 | 83030766 | 15 | -1 |\n",
       "| ENSG00000064726 | ENST00000560015 | BTBD1 | 83029543 | 83041821 | 15 | -1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene stable ID  Transcript stable ID HGNC symbol Transcript start (bp)\n",
       "1 ENSG00000064726 ENST00000261721      BTBD1       83016422             \n",
       "2 ENSG00000064726 ENST00000379403      BTBD1       83017961             \n",
       "3 ENSG00000064726 ENST00000559652      BTBD1       83018740             \n",
       "4 ENSG00000064726 ENST00000558344      BTBD1       83020746             \n",
       "5 ENSG00000064726 ENST00000560015      BTBD1       83029543             \n",
       "  Transcript end (bp) Chromosome/scaffold name Strand\n",
       "1 83067354            15                       -1    \n",
       "2 83067151            15                       -1    \n",
       "3 83056479            15                       -1    \n",
       "4 83030766            15                       -1    \n",
       "5 83041821            15                       -1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>hg38_location</th><th scope=col>S025NM</th><th scope=col>S025QG</th><th scope=col>S025RE</th><th scope=col>S025SC</th><th scope=col>S025TA</th><th scope=col>S025U8</th><th scope=col>S025V6</th><th scope=col>S025W4</th><th scope=col></th><th scope=col>S026KO</th><th scope=col>S026MK</th><th scope=col>S026NI</th><th scope=col>S026S8</th><th scope=col>S026T6</th><th scope=col>S026XZ</th><th scope=col>S026YX</th><th scope=col>S0270N</th><th scope=col>S0271L</th><th scope=col>avg_reads</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000148925_BTBD10 </td><td>11:13388001:13463297</td><td> 132</td><td> 46</td><td>  81</td><td>  74</td><td> 117</td><td> 90</td><td> 130</td><td> 129</td><td></td><td>  82</td><td> 116</td><td> 131</td><td>  95</td><td> 160</td><td> 58</td><td> 169</td><td> 113</td><td> 90</td><td> 105.1176</td></tr>\n",
       "\t<tr><td>ENSG00000064726_BTBD1  </td><td>15:83016422:83067354</td><td>1382</td><td>730</td><td>1042</td><td>1028</td><td>1220</td><td>877</td><td>1153</td><td>1369</td><td></td><td>1062</td><td>1592</td><td> 970</td><td> 661</td><td>1966</td><td>963</td><td>1022</td><td>1372</td><td>834</td><td>1069.5882</td></tr>\n",
       "\t<tr><td>ENSG00000176595_KBTBD11</td><td>8:1973878:2006936   </td><td> 638</td><td>727</td><td> 849</td><td> 951</td><td>1050</td><td>585</td><td> 549</td><td>1360</td><td></td><td>1643</td><td>2532</td><td>1481</td><td>1092</td><td>2969</td><td>856</td><td> 701</td><td>1935</td><td>913</td><td>1363.9412</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " feature & hg38\\_location & S025NM & S025QG & S025RE & S025SC & S025TA & S025U8 & S025V6 & S025W4 &  & S026KO & S026MK & S026NI & S026S8 & S026T6 & S026XZ & S026YX & S0270N & S0271L & avg\\_reads\\\\\n",
       " <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000148925\\_BTBD10  & 11:13388001:13463297 &  132 &  46 &   81 &   74 &  117 &  90 &  130 &  129 &  &   82 &  116 &  131 &   95 &  160 &  58 &  169 &  113 &  90 &  105.1176\\\\\n",
       "\t ENSG00000064726\\_BTBD1   & 15:83016422:83067354 & 1382 & 730 & 1042 & 1028 & 1220 & 877 & 1153 & 1369 &  & 1062 & 1592 &  970 &  661 & 1966 & 963 & 1022 & 1372 & 834 & 1069.5882\\\\\n",
       "\t ENSG00000176595\\_KBTBD11 & 8:1973878:2006936    &  638 & 727 &  849 &  951 & 1050 & 585 &  549 & 1360 &  & 1643 & 2532 & 1481 & 1092 & 2969 & 856 &  701 & 1935 & 913 & 1363.9412\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3  37\n",
       "\n",
       "| feature &lt;chr&gt; | hg38_location &lt;chr&gt; | S025NM &lt;int&gt; | S025QG &lt;int&gt; | S025RE &lt;int&gt; | S025SC &lt;int&gt; | S025TA &lt;int&gt; | S025U8 &lt;int&gt; | S025V6 &lt;int&gt; | S025W4 &lt;int&gt; |   | S026KO &lt;int&gt; | S026MK &lt;int&gt; | S026NI &lt;int&gt; | S026S8 &lt;int&gt; | S026T6 &lt;int&gt; | S026XZ &lt;int&gt; | S026YX &lt;int&gt; | S0270N &lt;int&gt; | S0271L &lt;int&gt; | avg_reads &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000148925_BTBD10  | 11:13388001:13463297 |  132 |  46 |   81 |   74 |  117 |  90 |  130 |  129 |  |   82 |  116 |  131 |   95 |  160 |  58 |  169 |  113 |  90 |  105.1176 |\n",
       "| ENSG00000064726_BTBD1   | 15:83016422:83067354 | 1382 | 730 | 1042 | 1028 | 1220 | 877 | 1153 | 1369 |  | 1062 | 1592 |  970 |  661 | 1966 | 963 | 1022 | 1372 | 834 | 1069.5882 |\n",
       "| ENSG00000176595_KBTBD11 | 8:1973878:2006936    |  638 | 727 |  849 |  951 | 1050 | 585 |  549 | 1360 |  | 1643 | 2532 | 1481 | 1092 | 2969 | 856 |  701 | 1935 | 913 | 1363.9412 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature                 hg38_location        S025NM S025QG S025RE S025SC\n",
       "1 ENSG00000148925_BTBD10  11:13388001:13463297  132    46      81     74  \n",
       "2 ENSG00000064726_BTBD1   15:83016422:83067354 1382   730    1042   1028  \n",
       "3 ENSG00000176595_KBTBD11 8:1973878:2006936     638   727     849    951  \n",
       "  S025TA S025U8 S025V6 S025W4  S026KO S026MK S026NI S026S8 S026T6 S026XZ\n",
       "1  117    90     130    129      82    116    131     95    160    58   \n",
       "2 1220   877    1153   1369    1062   1592    970    661   1966   963   \n",
       "3 1050   585     549   1360    1643   2532   1481   1092   2969   856   \n",
       "  S026YX S0270N S0271L avg_reads\n",
       "1  169    113    90     105.1176\n",
       "2 1022   1372   834    1069.5882\n",
       "3  701   1935   913    1363.9412"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 13  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>S025NM-01.keep.merge.unique.sort.bam</th><th scope=col>S025QG-02.keep.merge.unique.sort.bam</th><th scope=col>S025RE-03.keep.merge.unique.sort.bam</th><th scope=col>S025SC-04.keep.merge.unique.sort.bam</th><th scope=col>S025TA-05.keep.merge.unique.sort.bam</th><th scope=col>S025U8-06.keep.merge.unique.sort.bam</th><th scope=col>S025V6-07.keep.merge.unique.sort.bam</th><th scope=col>S025W4-08.keep.merge.unique.sort.bam</th><th scope=col>S025X2-09.keep.merge.unique.sort.bam</th><th scope=col></th><th scope=col>S026MK-27.keep.merge.unique.sort.bam</th><th scope=col>S026NI-28.keep.merge.unique.sort.bam</th><th scope=col>S026S8-29.keep.merge.unique.sort.bam</th><th scope=col>S026T6-30.keep.merge.unique.sort.bam</th><th scope=col>S026XZ-31.keep.merge.unique.sort.bam</th><th scope=col>S026YX-32.keep.merge.unique.sort.bam</th><th scope=col>S0270N-33.keep.merge.unique.sort.bam</th><th scope=col>S0271L-34.keep.merge.unique.sort.bam</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000064726</td><td>1715</td><td>890</td><td>1252</td><td>1185</td><td>1350</td><td>1196</td><td>1351</td><td>1708</td><td>1164</td><td></td><td>1515</td><td>951</td><td>743</td><td>1233</td><td>1441</td><td>1145</td><td>1592</td><td>1075</td><td>ENSG00000064726_BTBD1      </td><td>15:83016422:83067354  </td></tr>\n",
       "\t<tr><td>ENSG00000234438</td><td>   0</td><td>  0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   1</td><td></td><td>   0</td><td>  0</td><td>  0</td><td>   0</td><td>   1</td><td>   1</td><td>   0</td><td>   0</td><td>ENSG00000234438_KBTBD13    </td><td>15:65076816:65079939  </td></tr>\n",
       "\t<tr><td>ENSG00000187715</td><td>   2</td><td>  0</td><td>   1</td><td>   1</td><td>   1</td><td>   1</td><td>   4</td><td>   1</td><td>   1</td><td></td><td>   2</td><td>  0</td><td>  2</td><td>   1</td><td>   3</td><td>   1</td><td>   1</td><td>   0</td><td>ENSG00000187715_KBTBD12    </td><td>3:127915232:127987671 </td></tr>\n",
       "\t<tr><td>ENSG00000219433</td><td>   2</td><td>  0</td><td>   1</td><td>   1</td><td>   1</td><td>   1</td><td>   0</td><td>   2</td><td>   2</td><td></td><td>   4</td><td>  0</td><td>  0</td><td>   2</td><td>   3</td><td>   1</td><td>   2</td><td>   3</td><td>ENSG00000219433_BTBD10P2   </td><td>6:149794763:149795746 </td></tr>\n",
       "\t<tr><td>ENSG00000257675</td><td>   0</td><td>  0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   1</td><td>   0</td><td>   0</td><td></td><td>   0</td><td>  0</td><td>  0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>ENSG00000257675_BTBD10P1   </td><td>12:52748776:52750127  </td></tr>\n",
       "\t<tr><td>ENSG00000148925</td><td> 251</td><td>102</td><td> 132</td><td> 158</td><td> 178</td><td> 158</td><td> 231</td><td> 246</td><td> 259</td><td></td><td> 194</td><td>141</td><td>148</td><td> 140</td><td> 152</td><td> 217</td><td> 171</td><td> 180</td><td>ENSG00000148925_BTBD10     </td><td>11:13388001:13463297  </td></tr>\n",
       "\t<tr><td>ENSG00000253696</td><td>   2</td><td>  2</td><td>   3</td><td>   2</td><td>   5</td><td>   1</td><td>   1</td><td>   7</td><td>   2</td><td></td><td>   6</td><td>  1</td><td>  1</td><td>   6</td><td>   0</td><td>   3</td><td>   6</td><td>   1</td><td>ENSG00000253696_KBTBD11-OT1</td><td>8:1971397:1976478     </td></tr>\n",
       "\t<tr><td>ENSG00000176595</td><td> 405</td><td>519</td><td> 513</td><td> 601</td><td> 709</td><td> 440</td><td> 387</td><td> 898</td><td>1320</td><td></td><td>1334</td><td>868</td><td>741</td><td> 947</td><td> 585</td><td> 505</td><td>1293</td><td> 680</td><td>ENSG00000176595_KBTBD11    </td><td>8:1973878:2006936     </td></tr>\n",
       "\t<tr><td>ENSG00000204347</td><td>   0</td><td>  0</td><td>   0</td><td>   0</td><td>   1</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td></td><td>   0</td><td>  0</td><td>  0</td><td>   1</td><td>   1</td><td>   0</td><td>   1</td><td>   0</td><td>ENSG00000204347_BTBD17     </td><td>17:74356416:74361946  </td></tr>\n",
       "\t<tr><td>ENSG00000233436</td><td>  15</td><td> 12</td><td>  12</td><td>  12</td><td>   6</td><td>   9</td><td>   7</td><td>  23</td><td>   7</td><td></td><td>  11</td><td> 14</td><td> 10</td><td>   5</td><td>   8</td><td>   5</td><td>   8</td><td>  17</td><td>ENSG00000233436_BTBD18     </td><td>11:57743514:57751781  </td></tr>\n",
       "\t<tr><td>ENSG00000138152</td><td>   0</td><td>  0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td></td><td>   0</td><td>  0</td><td>  0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   1</td><td>ENSG00000138152_BTBD16     </td><td>10:122271306:122338162</td></tr>\n",
       "\t<tr><td>ENSG00000222009</td><td> 277</td><td>411</td><td> 421</td><td> 314</td><td> 184</td><td> 266</td><td> 717</td><td> 279</td><td> 869</td><td></td><td> 214</td><td>320</td><td>785</td><td> 330</td><td> 127</td><td> 384</td><td> 126</td><td> 567</td><td>ENSG00000222009_BTBD19     </td><td>1:44808482:44815585   </td></tr>\n",
       "\t<tr><td>ENSG00000151136</td><td> 487</td><td>476</td><td> 503</td><td> 399</td><td> 405</td><td> 564</td><td> 806</td><td> 434</td><td> 526</td><td></td><td> 446</td><td>556</td><td>371</td><td> 403</td><td> 518</td><td> 400</td><td> 531</td><td> 446</td><td>ENSG00000151136_BTBD11     </td><td>12:107318413:107659642</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 13  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " GeneID & S025NM-01.keep.merge.unique.sort.bam & S025QG-02.keep.merge.unique.sort.bam & S025RE-03.keep.merge.unique.sort.bam & S025SC-04.keep.merge.unique.sort.bam & S025TA-05.keep.merge.unique.sort.bam & S025U8-06.keep.merge.unique.sort.bam & S025V6-07.keep.merge.unique.sort.bam & S025W4-08.keep.merge.unique.sort.bam & S025X2-09.keep.merge.unique.sort.bam &  & S026MK-27.keep.merge.unique.sort.bam & S026NI-28.keep.merge.unique.sort.bam & S026S8-29.keep.merge.unique.sort.bam & S026T6-30.keep.merge.unique.sort.bam & S026XZ-31.keep.merge.unique.sort.bam & S026YX-32.keep.merge.unique.sort.bam & S0270N-33.keep.merge.unique.sort.bam & S0271L-34.keep.merge.unique.sort.bam & feature & hg38\\_location\\\\\n",
       " <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000064726 & 1715 & 890 & 1252 & 1185 & 1350 & 1196 & 1351 & 1708 & 1164 &  & 1515 & 951 & 743 & 1233 & 1441 & 1145 & 1592 & 1075 & ENSG00000064726\\_BTBD1       & 15:83016422:83067354  \\\\\n",
       "\t ENSG00000234438 &    0 &   0 &    0 &    0 &    0 &    0 &    0 &    0 &    1 &  &    0 &   0 &   0 &    0 &    1 &    1 &    0 &    0 & ENSG00000234438\\_KBTBD13     & 15:65076816:65079939  \\\\\n",
       "\t ENSG00000187715 &    2 &   0 &    1 &    1 &    1 &    1 &    4 &    1 &    1 &  &    2 &   0 &   2 &    1 &    3 &    1 &    1 &    0 & ENSG00000187715\\_KBTBD12     & 3:127915232:127987671 \\\\\n",
       "\t ENSG00000219433 &    2 &   0 &    1 &    1 &    1 &    1 &    0 &    2 &    2 &  &    4 &   0 &   0 &    2 &    3 &    1 &    2 &    3 & ENSG00000219433\\_BTBD10P2    & 6:149794763:149795746 \\\\\n",
       "\t ENSG00000257675 &    0 &   0 &    0 &    0 &    0 &    0 &    1 &    0 &    0 &  &    0 &   0 &   0 &    0 &    0 &    0 &    0 &    0 & ENSG00000257675\\_BTBD10P1    & 12:52748776:52750127  \\\\\n",
       "\t ENSG00000148925 &  251 & 102 &  132 &  158 &  178 &  158 &  231 &  246 &  259 &  &  194 & 141 & 148 &  140 &  152 &  217 &  171 &  180 & ENSG00000148925\\_BTBD10      & 11:13388001:13463297  \\\\\n",
       "\t ENSG00000253696 &    2 &   2 &    3 &    2 &    5 &    1 &    1 &    7 &    2 &  &    6 &   1 &   1 &    6 &    0 &    3 &    6 &    1 & ENSG00000253696\\_KBTBD11-OT1 & 8:1971397:1976478     \\\\\n",
       "\t ENSG00000176595 &  405 & 519 &  513 &  601 &  709 &  440 &  387 &  898 & 1320 &  & 1334 & 868 & 741 &  947 &  585 &  505 & 1293 &  680 & ENSG00000176595\\_KBTBD11     & 8:1973878:2006936     \\\\\n",
       "\t ENSG00000204347 &    0 &   0 &    0 &    0 &    1 &    0 &    0 &    0 &    0 &  &    0 &   0 &   0 &    1 &    1 &    0 &    1 &    0 & ENSG00000204347\\_BTBD17      & 17:74356416:74361946  \\\\\n",
       "\t ENSG00000233436 &   15 &  12 &   12 &   12 &    6 &    9 &    7 &   23 &    7 &  &   11 &  14 &  10 &    5 &    8 &    5 &    8 &   17 & ENSG00000233436\\_BTBD18      & 11:57743514:57751781  \\\\\n",
       "\t ENSG00000138152 &    0 &   0 &    0 &    0 &    0 &    0 &    0 &    0 &    0 &  &    0 &   0 &   0 &    0 &    0 &    0 &    0 &    1 & ENSG00000138152\\_BTBD16      & 10:122271306:122338162\\\\\n",
       "\t ENSG00000222009 &  277 & 411 &  421 &  314 &  184 &  266 &  717 &  279 &  869 &  &  214 & 320 & 785 &  330 &  127 &  384 &  126 &  567 & ENSG00000222009\\_BTBD19      & 1:44808482:44815585   \\\\\n",
       "\t ENSG00000151136 &  487 & 476 &  503 &  399 &  405 &  564 &  806 &  434 &  526 &  &  446 & 556 & 371 &  403 &  518 &  400 &  531 &  446 & ENSG00000151136\\_BTBD11      & 12:107318413:107659642\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 13  37\n",
       "\n",
       "| GeneID &lt;chr&gt; | S025NM-01.keep.merge.unique.sort.bam &lt;int&gt; | S025QG-02.keep.merge.unique.sort.bam &lt;int&gt; | S025RE-03.keep.merge.unique.sort.bam &lt;int&gt; | S025SC-04.keep.merge.unique.sort.bam &lt;int&gt; | S025TA-05.keep.merge.unique.sort.bam &lt;int&gt; | S025U8-06.keep.merge.unique.sort.bam &lt;int&gt; | S025V6-07.keep.merge.unique.sort.bam &lt;int&gt; | S025W4-08.keep.merge.unique.sort.bam &lt;int&gt; | S025X2-09.keep.merge.unique.sort.bam &lt;int&gt; |   | S026MK-27.keep.merge.unique.sort.bam &lt;int&gt; | S026NI-28.keep.merge.unique.sort.bam &lt;int&gt; | S026S8-29.keep.merge.unique.sort.bam &lt;int&gt; | S026T6-30.keep.merge.unique.sort.bam &lt;int&gt; | S026XZ-31.keep.merge.unique.sort.bam &lt;int&gt; | S026YX-32.keep.merge.unique.sort.bam &lt;int&gt; | S0270N-33.keep.merge.unique.sort.bam &lt;int&gt; | S0271L-34.keep.merge.unique.sort.bam &lt;int&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000064726 | 1715 | 890 | 1252 | 1185 | 1350 | 1196 | 1351 | 1708 | 1164 |  | 1515 | 951 | 743 | 1233 | 1441 | 1145 | 1592 | 1075 | ENSG00000064726_BTBD1       | 15:83016422:83067354   |\n",
       "| ENSG00000234438 |    0 |   0 |    0 |    0 |    0 |    0 |    0 |    0 |    1 |  |    0 |   0 |   0 |    0 |    1 |    1 |    0 |    0 | ENSG00000234438_KBTBD13     | 15:65076816:65079939   |\n",
       "| ENSG00000187715 |    2 |   0 |    1 |    1 |    1 |    1 |    4 |    1 |    1 |  |    2 |   0 |   2 |    1 |    3 |    1 |    1 |    0 | ENSG00000187715_KBTBD12     | 3:127915232:127987671  |\n",
       "| ENSG00000219433 |    2 |   0 |    1 |    1 |    1 |    1 |    0 |    2 |    2 |  |    4 |   0 |   0 |    2 |    3 |    1 |    2 |    3 | ENSG00000219433_BTBD10P2    | 6:149794763:149795746  |\n",
       "| ENSG00000257675 |    0 |   0 |    0 |    0 |    0 |    0 |    1 |    0 |    0 |  |    0 |   0 |   0 |    0 |    0 |    0 |    0 |    0 | ENSG00000257675_BTBD10P1    | 12:52748776:52750127   |\n",
       "| ENSG00000148925 |  251 | 102 |  132 |  158 |  178 |  158 |  231 |  246 |  259 |  |  194 | 141 | 148 |  140 |  152 |  217 |  171 |  180 | ENSG00000148925_BTBD10      | 11:13388001:13463297   |\n",
       "| ENSG00000253696 |    2 |   2 |    3 |    2 |    5 |    1 |    1 |    7 |    2 |  |    6 |   1 |   1 |    6 |    0 |    3 |    6 |    1 | ENSG00000253696_KBTBD11-OT1 | 8:1971397:1976478      |\n",
       "| ENSG00000176595 |  405 | 519 |  513 |  601 |  709 |  440 |  387 |  898 | 1320 |  | 1334 | 868 | 741 |  947 |  585 |  505 | 1293 |  680 | ENSG00000176595_KBTBD11     | 8:1973878:2006936      |\n",
       "| ENSG00000204347 |    0 |   0 |    0 |    0 |    1 |    0 |    0 |    0 |    0 |  |    0 |   0 |   0 |    1 |    1 |    0 |    1 |    0 | ENSG00000204347_BTBD17      | 17:74356416:74361946   |\n",
       "| ENSG00000233436 |   15 |  12 |   12 |   12 |    6 |    9 |    7 |   23 |    7 |  |   11 |  14 |  10 |    5 |    8 |    5 |    8 |   17 | ENSG00000233436_BTBD18      | 11:57743514:57751781   |\n",
       "| ENSG00000138152 |    0 |   0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |  |    0 |   0 |   0 |    0 |    0 |    0 |    0 |    1 | ENSG00000138152_BTBD16      | 10:122271306:122338162 |\n",
       "| ENSG00000222009 |  277 | 411 |  421 |  314 |  184 |  266 |  717 |  279 |  869 |  |  214 | 320 | 785 |  330 |  127 |  384 |  126 |  567 | ENSG00000222009_BTBD19      | 1:44808482:44815585    |\n",
       "| ENSG00000151136 |  487 | 476 |  503 |  399 |  405 |  564 |  806 |  434 |  526 |  |  446 | 556 | 371 |  403 |  518 |  400 |  531 |  446 | ENSG00000151136_BTBD11      | 12:107318413:107659642 |\n",
       "\n"
      ],
      "text/plain": [
       "   GeneID          S025NM-01.keep.merge.unique.sort.bam\n",
       "1  ENSG00000064726 1715                                \n",
       "2  ENSG00000234438    0                                \n",
       "3  ENSG00000187715    2                                \n",
       "4  ENSG00000219433    2                                \n",
       "5  ENSG00000257675    0                                \n",
       "6  ENSG00000148925  251                                \n",
       "7  ENSG00000253696    2                                \n",
       "8  ENSG00000176595  405                                \n",
       "9  ENSG00000204347    0                                \n",
       "10 ENSG00000233436   15                                \n",
       "11 ENSG00000138152    0                                \n",
       "12 ENSG00000222009  277                                \n",
       "13 ENSG00000151136  487                                \n",
       "   S025QG-02.keep.merge.unique.sort.bam S025RE-03.keep.merge.unique.sort.bam\n",
       "1  890                                  1252                                \n",
       "2    0                                     0                                \n",
       "3    0                                     1                                \n",
       "4    0                                     1                                \n",
       "5    0                                     0                                \n",
       "6  102                                   132                                \n",
       "7    2                                     3                                \n",
       "8  519                                   513                                \n",
       "9    0                                     0                                \n",
       "10  12                                    12                                \n",
       "11   0                                     0                                \n",
       "12 411                                   421                                \n",
       "13 476                                   503                                \n",
       "   S025SC-04.keep.merge.unique.sort.bam S025TA-05.keep.merge.unique.sort.bam\n",
       "1  1185                                 1350                                \n",
       "2     0                                    0                                \n",
       "3     1                                    1                                \n",
       "4     1                                    1                                \n",
       "5     0                                    0                                \n",
       "6   158                                  178                                \n",
       "7     2                                    5                                \n",
       "8   601                                  709                                \n",
       "9     0                                    1                                \n",
       "10   12                                    6                                \n",
       "11    0                                    0                                \n",
       "12  314                                  184                                \n",
       "13  399                                  405                                \n",
       "   S025U8-06.keep.merge.unique.sort.bam S025V6-07.keep.merge.unique.sort.bam\n",
       "1  1196                                 1351                                \n",
       "2     0                                    0                                \n",
       "3     1                                    4                                \n",
       "4     1                                    0                                \n",
       "5     0                                    1                                \n",
       "6   158                                  231                                \n",
       "7     1                                    1                                \n",
       "8   440                                  387                                \n",
       "9     0                                    0                                \n",
       "10    9                                    7                                \n",
       "11    0                                    0                                \n",
       "12  266                                  717                                \n",
       "13  564                                  806                                \n",
       "   S025W4-08.keep.merge.unique.sort.bam S025X2-09.keep.merge.unique.sort.bam \n",
       "1  1708                                 1164                                 \n",
       "2     0                                    1                                 \n",
       "3     1                                    1                                 \n",
       "4     2                                    2                                 \n",
       "5     0                                    0                                 \n",
       "6   246                                  259                                 \n",
       "7     7                                    2                                 \n",
       "8   898                                 1320                                 \n",
       "9     0                                    0                                 \n",
       "10   23                                    7                                 \n",
       "11    0                                    0                                 \n",
       "12  279                                  869                                 \n",
       "13  434                                  526                                 \n",
       "   S026MK-27.keep.merge.unique.sort.bam S026NI-28.keep.merge.unique.sort.bam\n",
       "1  1515                                 951                                 \n",
       "2     0                                   0                                 \n",
       "3     2                                   0                                 \n",
       "4     4                                   0                                 \n",
       "5     0                                   0                                 \n",
       "6   194                                 141                                 \n",
       "7     6                                   1                                 \n",
       "8  1334                                 868                                 \n",
       "9     0                                   0                                 \n",
       "10   11                                  14                                 \n",
       "11    0                                   0                                 \n",
       "12  214                                 320                                 \n",
       "13  446                                 556                                 \n",
       "   S026S8-29.keep.merge.unique.sort.bam S026T6-30.keep.merge.unique.sort.bam\n",
       "1  743                                  1233                                \n",
       "2    0                                     0                                \n",
       "3    2                                     1                                \n",
       "4    0                                     2                                \n",
       "5    0                                     0                                \n",
       "6  148                                   140                                \n",
       "7    1                                     6                                \n",
       "8  741                                   947                                \n",
       "9    0                                     1                                \n",
       "10  10                                     5                                \n",
       "11   0                                     0                                \n",
       "12 785                                   330                                \n",
       "13 371                                   403                                \n",
       "   S026XZ-31.keep.merge.unique.sort.bam S026YX-32.keep.merge.unique.sort.bam\n",
       "1  1441                                 1145                                \n",
       "2     1                                    1                                \n",
       "3     3                                    1                                \n",
       "4     3                                    1                                \n",
       "5     0                                    0                                \n",
       "6   152                                  217                                \n",
       "7     0                                    3                                \n",
       "8   585                                  505                                \n",
       "9     1                                    0                                \n",
       "10    8                                    5                                \n",
       "11    0                                    0                                \n",
       "12  127                                  384                                \n",
       "13  518                                  400                                \n",
       "   S0270N-33.keep.merge.unique.sort.bam S0271L-34.keep.merge.unique.sort.bam\n",
       "1  1592                                 1075                                \n",
       "2     0                                    0                                \n",
       "3     1                                    0                                \n",
       "4     2                                    3                                \n",
       "5     0                                    0                                \n",
       "6   171                                  180                                \n",
       "7     6                                    1                                \n",
       "8  1293                                  680                                \n",
       "9     1                                    0                                \n",
       "10    8                                   17                                \n",
       "11    0                                    1                                \n",
       "12  126                                  567                                \n",
       "13  531                                  446                                \n",
       "   feature                     hg38_location         \n",
       "1  ENSG00000064726_BTBD1       15:83016422:83067354  \n",
       "2  ENSG00000234438_KBTBD13     15:65076816:65079939  \n",
       "3  ENSG00000187715_KBTBD12     3:127915232:127987671 \n",
       "4  ENSG00000219433_BTBD10P2    6:149794763:149795746 \n",
       "5  ENSG00000257675_BTBD10P1    12:52748776:52750127  \n",
       "6  ENSG00000148925_BTBD10      11:13388001:13463297  \n",
       "7  ENSG00000253696_KBTBD11-OT1 8:1971397:1976478     \n",
       "8  ENSG00000176595_KBTBD11     8:1973878:2006936     \n",
       "9  ENSG00000204347_BTBD17      17:74356416:74361946  \n",
       "10 ENSG00000233436_BTBD18      11:57743514:57751781  \n",
       "11 ENSG00000138152_BTBD16      10:122271306:122338162\n",
       "12 ENSG00000222009_BTBD19      1:44808482:44815585   \n",
       "13 ENSG00000151136_BTBD11      12:107318413:107659642"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>S025NM-01.keep.merge.unique.sort.bam</th><th scope=col>S025QG-02.keep.merge.unique.sort.bam</th><th scope=col>S025RE-03.keep.merge.unique.sort.bam</th><th scope=col>S025SC-04.keep.merge.unique.sort.bam</th><th scope=col>S025TA-05.keep.merge.unique.sort.bam</th><th scope=col>S025U8-06.keep.merge.unique.sort.bam</th><th scope=col>S025V6-07.keep.merge.unique.sort.bam</th><th scope=col>S025W4-08.keep.merge.unique.sort.bam</th><th scope=col>S025X2-09.keep.merge.unique.sort.bam</th><th scope=col></th><th scope=col>S026MK-27.keep.merge.unique.sort.bam</th><th scope=col>S026NI-28.keep.merge.unique.sort.bam</th><th scope=col>S026S8-29.keep.merge.unique.sort.bam</th><th scope=col>S026T6-30.keep.merge.unique.sort.bam</th><th scope=col>S026XZ-31.keep.merge.unique.sort.bam</th><th scope=col>S026YX-32.keep.merge.unique.sort.bam</th><th scope=col>S0270N-33.keep.merge.unique.sort.bam</th><th scope=col>S0271L-34.keep.merge.unique.sort.bam</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000064726</td><td>1715</td><td>890</td><td>1252</td><td>1185</td><td>1350</td><td>1196</td><td>1351</td><td>1708</td><td>1164</td><td></td><td>1515</td><td>951</td><td>743</td><td>1233</td><td>1441</td><td>1145</td><td>1592</td><td>1075</td><td>ENSG00000064726_BTBD1</td><td>15:83016422:83067354</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " GeneID & S025NM-01.keep.merge.unique.sort.bam & S025QG-02.keep.merge.unique.sort.bam & S025RE-03.keep.merge.unique.sort.bam & S025SC-04.keep.merge.unique.sort.bam & S025TA-05.keep.merge.unique.sort.bam & S025U8-06.keep.merge.unique.sort.bam & S025V6-07.keep.merge.unique.sort.bam & S025W4-08.keep.merge.unique.sort.bam & S025X2-09.keep.merge.unique.sort.bam &  & S026MK-27.keep.merge.unique.sort.bam & S026NI-28.keep.merge.unique.sort.bam & S026S8-29.keep.merge.unique.sort.bam & S026T6-30.keep.merge.unique.sort.bam & S026XZ-31.keep.merge.unique.sort.bam & S026YX-32.keep.merge.unique.sort.bam & S0270N-33.keep.merge.unique.sort.bam & S0271L-34.keep.merge.unique.sort.bam & feature & hg38\\_location\\\\\n",
       " <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000064726 & 1715 & 890 & 1252 & 1185 & 1350 & 1196 & 1351 & 1708 & 1164 &  & 1515 & 951 & 743 & 1233 & 1441 & 1145 & 1592 & 1075 & ENSG00000064726\\_BTBD1 & 15:83016422:83067354\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| GeneID &lt;chr&gt; | S025NM-01.keep.merge.unique.sort.bam &lt;int&gt; | S025QG-02.keep.merge.unique.sort.bam &lt;int&gt; | S025RE-03.keep.merge.unique.sort.bam &lt;int&gt; | S025SC-04.keep.merge.unique.sort.bam &lt;int&gt; | S025TA-05.keep.merge.unique.sort.bam &lt;int&gt; | S025U8-06.keep.merge.unique.sort.bam &lt;int&gt; | S025V6-07.keep.merge.unique.sort.bam &lt;int&gt; | S025W4-08.keep.merge.unique.sort.bam &lt;int&gt; | S025X2-09.keep.merge.unique.sort.bam &lt;int&gt; |   | S026MK-27.keep.merge.unique.sort.bam &lt;int&gt; | S026NI-28.keep.merge.unique.sort.bam &lt;int&gt; | S026S8-29.keep.merge.unique.sort.bam &lt;int&gt; | S026T6-30.keep.merge.unique.sort.bam &lt;int&gt; | S026XZ-31.keep.merge.unique.sort.bam &lt;int&gt; | S026YX-32.keep.merge.unique.sort.bam &lt;int&gt; | S0270N-33.keep.merge.unique.sort.bam &lt;int&gt; | S0271L-34.keep.merge.unique.sort.bam &lt;int&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000064726 | 1715 | 890 | 1252 | 1185 | 1350 | 1196 | 1351 | 1708 | 1164 |  | 1515 | 951 | 743 | 1233 | 1441 | 1145 | 1592 | 1075 | ENSG00000064726_BTBD1 | 15:83016422:83067354 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          S025NM-01.keep.merge.unique.sort.bam\n",
       "1 ENSG00000064726 1715                                \n",
       "  S025QG-02.keep.merge.unique.sort.bam S025RE-03.keep.merge.unique.sort.bam\n",
       "1 890                                  1252                                \n",
       "  S025SC-04.keep.merge.unique.sort.bam S025TA-05.keep.merge.unique.sort.bam\n",
       "1 1185                                 1350                                \n",
       "  S025U8-06.keep.merge.unique.sort.bam S025V6-07.keep.merge.unique.sort.bam\n",
       "1 1196                                 1351                                \n",
       "  S025W4-08.keep.merge.unique.sort.bam S025X2-09.keep.merge.unique.sort.bam \n",
       "1 1708                                 1164                                 \n",
       "  S026MK-27.keep.merge.unique.sort.bam S026NI-28.keep.merge.unique.sort.bam\n",
       "1 1515                                 951                                 \n",
       "  S026S8-29.keep.merge.unique.sort.bam S026T6-30.keep.merge.unique.sort.bam\n",
       "1 743                                  1233                                \n",
       "  S026XZ-31.keep.merge.unique.sort.bam S026YX-32.keep.merge.unique.sort.bam\n",
       "1 1441                                 1145                                \n",
       "  S0270N-33.keep.merge.unique.sort.bam S0271L-34.keep.merge.unique.sort.bam\n",
       "1 1592                                 1075                                \n",
       "  feature               hg38_location       \n",
       "1 ENSG00000064726_BTBD1 15:83016422:83067354"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000255423</td><td>ENSG00000255423_EBLN2</td><td>3:73061659:73063337</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  3\n",
       "\\begin{tabular}{lll}\n",
       " GeneID & feature & hg38\\_location\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000255423 & ENSG00000255423\\_EBLN2 & 3:73061659:73063337\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  3\n",
       "\n",
       "| GeneID &lt;chr&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000255423 | ENSG00000255423_EBLN2 | 3:73061659:73063337 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          feature               hg38_location      \n",
       "1 ENSG00000255423 ENSG00000255423_EBLN2 3:73061659:73063337"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene stable ID</th><th scope=col>Transcript stable ID</th><th scope=col>HGNC symbol</th><th scope=col>Transcript start (bp)</th><th scope=col>Transcript end (bp)</th><th scope=col>Chromosome/scaffold name</th><th scope=col>Strand</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000255423</td><td>ENST00000533473</td><td>EBLN2</td><td>73061659</td><td>73063337</td><td>3</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  7\n",
       "\\begin{tabular}{lllllll}\n",
       " Gene stable ID & Transcript stable ID & HGNC symbol & Transcript start (bp) & Transcript end (bp) & Chromosome/scaffold name & Strand\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ENSG00000255423 & ENST00000533473 & EBLN2 & 73061659 & 73063337 & 3 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  7\n",
       "\n",
       "| Gene stable ID &lt;chr&gt; | Transcript stable ID &lt;chr&gt; | HGNC symbol &lt;chr&gt; | Transcript start (bp) &lt;int&gt; | Transcript end (bp) &lt;int&gt; | Chromosome/scaffold name &lt;chr&gt; | Strand &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| ENSG00000255423 | ENST00000533473 | EBLN2 | 73061659 | 73063337 | 3 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene stable ID  Transcript stable ID HGNC symbol Transcript start (bp)\n",
       "1 ENSG00000255423 ENST00000533473      EBLN2       73061659             \n",
       "  Transcript end (bp) Chromosome/scaffold name Strand\n",
       "1 73063337            3                        1     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>hg38_location</th><th scope=col>S025NM</th><th scope=col>S025QG</th><th scope=col>S025RE</th><th scope=col>S025SC</th><th scope=col>S025TA</th><th scope=col>S025U8</th><th scope=col>S025V6</th><th scope=col>S025W4</th><th scope=col></th><th scope=col>S026KO</th><th scope=col>S026MK</th><th scope=col>S026NI</th><th scope=col>S026S8</th><th scope=col>S026T6</th><th scope=col>S026XZ</th><th scope=col>S026YX</th><th scope=col>S0270N</th><th scope=col>S0271L</th><th scope=col>avg_reads</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000255423_EBLN2</td><td>3:73061659:73063337</td><td>187</td><td>104</td><td>140</td><td>110</td><td>148</td><td>163</td><td>196</td><td>152</td><td></td><td>113</td><td>253</td><td>109</td><td>88</td><td>268</td><td>154</td><td>128</td><td>178</td><td>145</td><td>152.3824</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " feature & hg38\\_location & S025NM & S025QG & S025RE & S025SC & S025TA & S025U8 & S025V6 & S025W4 &  & S026KO & S026MK & S026NI & S026S8 & S026T6 & S026XZ & S026YX & S0270N & S0271L & avg\\_reads\\\\\n",
       " <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000255423\\_EBLN2 & 3:73061659:73063337 & 187 & 104 & 140 & 110 & 148 & 163 & 196 & 152 &  & 113 & 253 & 109 & 88 & 268 & 154 & 128 & 178 & 145 & 152.3824\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| feature &lt;chr&gt; | hg38_location &lt;chr&gt; | S025NM &lt;int&gt; | S025QG &lt;int&gt; | S025RE &lt;int&gt; | S025SC &lt;int&gt; | S025TA &lt;int&gt; | S025U8 &lt;int&gt; | S025V6 &lt;int&gt; | S025W4 &lt;int&gt; |   | S026KO &lt;int&gt; | S026MK &lt;int&gt; | S026NI &lt;int&gt; | S026S8 &lt;int&gt; | S026T6 &lt;int&gt; | S026XZ &lt;int&gt; | S026YX &lt;int&gt; | S0270N &lt;int&gt; | S0271L &lt;int&gt; | avg_reads &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000255423_EBLN2 | 3:73061659:73063337 | 187 | 104 | 140 | 110 | 148 | 163 | 196 | 152 |  | 113 | 253 | 109 | 88 | 268 | 154 | 128 | 178 | 145 | 152.3824 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature               hg38_location       S025NM S025QG S025RE S025SC S025TA\n",
       "1 ENSG00000255423_EBLN2 3:73061659:73063337 187    104    140    110    148   \n",
       "  S025U8 S025V6 S025W4  S026KO S026MK S026NI S026S8 S026T6 S026XZ S026YX\n",
       "1 163    196    152     113    253    109    88     268    154    128   \n",
       "  S0270N S0271L avg_reads\n",
       "1 178    145    152.3824 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>S025NM-01.keep.merge.unique.sort.bam</th><th scope=col>S025QG-02.keep.merge.unique.sort.bam</th><th scope=col>S025RE-03.keep.merge.unique.sort.bam</th><th scope=col>S025SC-04.keep.merge.unique.sort.bam</th><th scope=col>S025TA-05.keep.merge.unique.sort.bam</th><th scope=col>S025U8-06.keep.merge.unique.sort.bam</th><th scope=col>S025V6-07.keep.merge.unique.sort.bam</th><th scope=col>S025W4-08.keep.merge.unique.sort.bam</th><th scope=col>S025X2-09.keep.merge.unique.sort.bam</th><th scope=col></th><th scope=col>S026MK-27.keep.merge.unique.sort.bam</th><th scope=col>S026NI-28.keep.merge.unique.sort.bam</th><th scope=col>S026S8-29.keep.merge.unique.sort.bam</th><th scope=col>S026T6-30.keep.merge.unique.sort.bam</th><th scope=col>S026XZ-31.keep.merge.unique.sort.bam</th><th scope=col>S026YX-32.keep.merge.unique.sort.bam</th><th scope=col>S0270N-33.keep.merge.unique.sort.bam</th><th scope=col>S0271L-34.keep.merge.unique.sort.bam</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000255423</td><td>127</td><td>78</td><td>96</td><td>77</td><td>103</td><td>120</td><td>141</td><td>109</td><td>120</td><td></td><td>146</td><td>72</td><td>64</td><td>92</td><td>117</td><td>97</td><td>123</td><td>113</td><td>ENSG00000255423_EBLN2</td><td>3:73061659:73063337</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " GeneID & S025NM-01.keep.merge.unique.sort.bam & S025QG-02.keep.merge.unique.sort.bam & S025RE-03.keep.merge.unique.sort.bam & S025SC-04.keep.merge.unique.sort.bam & S025TA-05.keep.merge.unique.sort.bam & S025U8-06.keep.merge.unique.sort.bam & S025V6-07.keep.merge.unique.sort.bam & S025W4-08.keep.merge.unique.sort.bam & S025X2-09.keep.merge.unique.sort.bam &  & S026MK-27.keep.merge.unique.sort.bam & S026NI-28.keep.merge.unique.sort.bam & S026S8-29.keep.merge.unique.sort.bam & S026T6-30.keep.merge.unique.sort.bam & S026XZ-31.keep.merge.unique.sort.bam & S026YX-32.keep.merge.unique.sort.bam & S0270N-33.keep.merge.unique.sort.bam & S0271L-34.keep.merge.unique.sort.bam & feature & hg38\\_location\\\\\n",
       " <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000255423 & 127 & 78 & 96 & 77 & 103 & 120 & 141 & 109 & 120 &  & 146 & 72 & 64 & 92 & 117 & 97 & 123 & 113 & ENSG00000255423\\_EBLN2 & 3:73061659:73063337\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| GeneID &lt;chr&gt; | S025NM-01.keep.merge.unique.sort.bam &lt;int&gt; | S025QG-02.keep.merge.unique.sort.bam &lt;int&gt; | S025RE-03.keep.merge.unique.sort.bam &lt;int&gt; | S025SC-04.keep.merge.unique.sort.bam &lt;int&gt; | S025TA-05.keep.merge.unique.sort.bam &lt;int&gt; | S025U8-06.keep.merge.unique.sort.bam &lt;int&gt; | S025V6-07.keep.merge.unique.sort.bam &lt;int&gt; | S025W4-08.keep.merge.unique.sort.bam &lt;int&gt; | S025X2-09.keep.merge.unique.sort.bam &lt;int&gt; |   | S026MK-27.keep.merge.unique.sort.bam &lt;int&gt; | S026NI-28.keep.merge.unique.sort.bam &lt;int&gt; | S026S8-29.keep.merge.unique.sort.bam &lt;int&gt; | S026T6-30.keep.merge.unique.sort.bam &lt;int&gt; | S026XZ-31.keep.merge.unique.sort.bam &lt;int&gt; | S026YX-32.keep.merge.unique.sort.bam &lt;int&gt; | S0270N-33.keep.merge.unique.sort.bam &lt;int&gt; | S0271L-34.keep.merge.unique.sort.bam &lt;int&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000255423 | 127 | 78 | 96 | 77 | 103 | 120 | 141 | 109 | 120 |  | 146 | 72 | 64 | 92 | 117 | 97 | 123 | 113 | ENSG00000255423_EBLN2 | 3:73061659:73063337 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          S025NM-01.keep.merge.unique.sort.bam\n",
       "1 ENSG00000255423 127                                 \n",
       "  S025QG-02.keep.merge.unique.sort.bam S025RE-03.keep.merge.unique.sort.bam\n",
       "1 78                                   96                                  \n",
       "  S025SC-04.keep.merge.unique.sort.bam S025TA-05.keep.merge.unique.sort.bam\n",
       "1 77                                   103                                 \n",
       "  S025U8-06.keep.merge.unique.sort.bam S025V6-07.keep.merge.unique.sort.bam\n",
       "1 120                                  141                                 \n",
       "  S025W4-08.keep.merge.unique.sort.bam S025X2-09.keep.merge.unique.sort.bam \n",
       "1 109                                  120                                  \n",
       "  S026MK-27.keep.merge.unique.sort.bam S026NI-28.keep.merge.unique.sort.bam\n",
       "1 146                                  72                                  \n",
       "  S026S8-29.keep.merge.unique.sort.bam S026T6-30.keep.merge.unique.sort.bam\n",
       "1 64                                   92                                  \n",
       "  S026XZ-31.keep.merge.unique.sort.bam S026YX-32.keep.merge.unique.sort.bam\n",
       "1 117                                  97                                  \n",
       "  S0270N-33.keep.merge.unique.sort.bam S0271L-34.keep.merge.unique.sort.bam\n",
       "1 123                                  113                                 \n",
       "  feature               hg38_location      \n",
       "1 ENSG00000255423_EBLN2 3:73061659:73063337"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 2  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000251503</td><td>ENSG00000251503_CENPS-CORT</td><td>1:10430102:10452153</td></tr>\n",
       "\t<tr><td>ENSG00000175279</td><td>ENSG00000175279_CENPS     </td><td>1:10430443:10442809</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 2  3\n",
       "\\begin{tabular}{lll}\n",
       " GeneID & feature & hg38\\_location\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000251503 & ENSG00000251503\\_CENPS-CORT & 1:10430102:10452153\\\\\n",
       "\t ENSG00000175279 & ENSG00000175279\\_CENPS      & 1:10430443:10442809\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 2  3\n",
       "\n",
       "| GeneID &lt;chr&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000251503 | ENSG00000251503_CENPS-CORT | 1:10430102:10452153 |\n",
       "| ENSG00000175279 | ENSG00000175279_CENPS      | 1:10430443:10442809 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          feature                    hg38_location      \n",
       "1 ENSG00000251503 ENSG00000251503_CENPS-CORT 1:10430102:10452153\n",
       "2 ENSG00000175279 ENSG00000175279_CENPS      1:10430443:10442809"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 5  7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene stable ID</th><th scope=col>Transcript stable ID</th><th scope=col>HGNC symbol</th><th scope=col>Transcript start (bp)</th><th scope=col>Transcript end (bp)</th><th scope=col>Chromosome/scaffold name</th><th scope=col>Strand</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000175279</td><td>ENST00000309048</td><td>CENPS</td><td>10430443</td><td>10442809</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>ENSG00000175279</td><td>ENST00000602486</td><td>CENPS</td><td>10430512</td><td>10431130</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>ENSG00000175279</td><td>ENST00000477755</td><td>CENPS</td><td>10430783</td><td>10442334</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>ENSG00000175279</td><td>ENST00000462462</td><td>CENPS</td><td>10431123</td><td>10442502</td><td>1</td><td>1</td></tr>\n",
       "\t<tr><td>ENSG00000175279</td><td>ENST00000464507</td><td>CENPS</td><td>10440123</td><td>10442657</td><td>1</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5  7\n",
       "\\begin{tabular}{lllllll}\n",
       " Gene stable ID & Transcript stable ID & HGNC symbol & Transcript start (bp) & Transcript end (bp) & Chromosome/scaffold name & Strand\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ENSG00000175279 & ENST00000309048 & CENPS & 10430443 & 10442809 & 1 & 1\\\\\n",
       "\t ENSG00000175279 & ENST00000602486 & CENPS & 10430512 & 10431130 & 1 & 1\\\\\n",
       "\t ENSG00000175279 & ENST00000477755 & CENPS & 10430783 & 10442334 & 1 & 1\\\\\n",
       "\t ENSG00000175279 & ENST00000462462 & CENPS & 10431123 & 10442502 & 1 & 1\\\\\n",
       "\t ENSG00000175279 & ENST00000464507 & CENPS & 10440123 & 10442657 & 1 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5  7\n",
       "\n",
       "| Gene stable ID &lt;chr&gt; | Transcript stable ID &lt;chr&gt; | HGNC symbol &lt;chr&gt; | Transcript start (bp) &lt;int&gt; | Transcript end (bp) &lt;int&gt; | Chromosome/scaffold name &lt;chr&gt; | Strand &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| ENSG00000175279 | ENST00000309048 | CENPS | 10430443 | 10442809 | 1 | 1 |\n",
       "| ENSG00000175279 | ENST00000602486 | CENPS | 10430512 | 10431130 | 1 | 1 |\n",
       "| ENSG00000175279 | ENST00000477755 | CENPS | 10430783 | 10442334 | 1 | 1 |\n",
       "| ENSG00000175279 | ENST00000462462 | CENPS | 10431123 | 10442502 | 1 | 1 |\n",
       "| ENSG00000175279 | ENST00000464507 | CENPS | 10440123 | 10442657 | 1 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene stable ID  Transcript stable ID HGNC symbol Transcript start (bp)\n",
       "1 ENSG00000175279 ENST00000309048      CENPS       10430443             \n",
       "2 ENSG00000175279 ENST00000602486      CENPS       10430512             \n",
       "3 ENSG00000175279 ENST00000477755      CENPS       10430783             \n",
       "4 ENSG00000175279 ENST00000462462      CENPS       10431123             \n",
       "5 ENSG00000175279 ENST00000464507      CENPS       10440123             \n",
       "  Transcript end (bp) Chromosome/scaffold name Strand\n",
       "1 10442809            1                        1     \n",
       "2 10431130            1                        1     \n",
       "3 10442334            1                        1     \n",
       "4 10442502            1                        1     \n",
       "5 10442657            1                        1     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>hg38_location</th><th scope=col>S025NM</th><th scope=col>S025QG</th><th scope=col>S025RE</th><th scope=col>S025SC</th><th scope=col>S025TA</th><th scope=col>S025U8</th><th scope=col>S025V6</th><th scope=col>S025W4</th><th scope=col></th><th scope=col>S026KO</th><th scope=col>S026MK</th><th scope=col>S026NI</th><th scope=col>S026S8</th><th scope=col>S026T6</th><th scope=col>S026XZ</th><th scope=col>S026YX</th><th scope=col>S0270N</th><th scope=col>S0271L</th><th scope=col>avg_reads</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000175279_CENPS</td><td>1:10430443:10442809</td><td>0</td><td>0</td><td>0</td><td>1</td><td>7</td><td>0</td><td>0</td><td>0</td><td></td><td>1</td><td>0</td><td>0</td><td>0</td><td>3</td><td>2</td><td>1</td><td>4</td><td>1</td><td>1.529412</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " feature & hg38\\_location & S025NM & S025QG & S025RE & S025SC & S025TA & S025U8 & S025V6 & S025W4 &  & S026KO & S026MK & S026NI & S026S8 & S026T6 & S026XZ & S026YX & S0270N & S0271L & avg\\_reads\\\\\n",
       " <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000175279\\_CENPS & 1:10430443:10442809 & 0 & 0 & 0 & 1 & 7 & 0 & 0 & 0 &  & 1 & 0 & 0 & 0 & 3 & 2 & 1 & 4 & 1 & 1.529412\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| feature &lt;chr&gt; | hg38_location &lt;chr&gt; | S025NM &lt;int&gt; | S025QG &lt;int&gt; | S025RE &lt;int&gt; | S025SC &lt;int&gt; | S025TA &lt;int&gt; | S025U8 &lt;int&gt; | S025V6 &lt;int&gt; | S025W4 &lt;int&gt; |   | S026KO &lt;int&gt; | S026MK &lt;int&gt; | S026NI &lt;int&gt; | S026S8 &lt;int&gt; | S026T6 &lt;int&gt; | S026XZ &lt;int&gt; | S026YX &lt;int&gt; | S0270N &lt;int&gt; | S0271L &lt;int&gt; | avg_reads &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000175279_CENPS | 1:10430443:10442809 | 0 | 0 | 0 | 1 | 7 | 0 | 0 | 0 |  | 1 | 0 | 0 | 0 | 3 | 2 | 1 | 4 | 1 | 1.529412 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature               hg38_location       S025NM S025QG S025RE S025SC S025TA\n",
       "1 ENSG00000175279_CENPS 1:10430443:10442809 0      0      0      1      7     \n",
       "  S025U8 S025V6 S025W4  S026KO S026MK S026NI S026S8 S026T6 S026XZ S026YX\n",
       "1 0      0      0       1      0      0      0      3      2      1     \n",
       "  S0270N S0271L avg_reads\n",
       "1 4      1      1.529412 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 2  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>S025NM-01.keep.merge.unique.sort.bam</th><th scope=col>S025QG-02.keep.merge.unique.sort.bam</th><th scope=col>S025RE-03.keep.merge.unique.sort.bam</th><th scope=col>S025SC-04.keep.merge.unique.sort.bam</th><th scope=col>S025TA-05.keep.merge.unique.sort.bam</th><th scope=col>S025U8-06.keep.merge.unique.sort.bam</th><th scope=col>S025V6-07.keep.merge.unique.sort.bam</th><th scope=col>S025W4-08.keep.merge.unique.sort.bam</th><th scope=col>S025X2-09.keep.merge.unique.sort.bam</th><th scope=col></th><th scope=col>S026MK-27.keep.merge.unique.sort.bam</th><th scope=col>S026NI-28.keep.merge.unique.sort.bam</th><th scope=col>S026S8-29.keep.merge.unique.sort.bam</th><th scope=col>S026T6-30.keep.merge.unique.sort.bam</th><th scope=col>S026XZ-31.keep.merge.unique.sort.bam</th><th scope=col>S026YX-32.keep.merge.unique.sort.bam</th><th scope=col>S0270N-33.keep.merge.unique.sort.bam</th><th scope=col>S0271L-34.keep.merge.unique.sort.bam</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000251503</td><td>12</td><td>7</td><td>5</td><td>12</td><td>9</td><td>13</td><td>20</td><td>11</td><td>16</td><td></td><td>6</td><td>8</td><td>6</td><td>11</td><td>11</td><td>13</td><td>15</td><td>12</td><td>ENSG00000251503_CENPS-CORT</td><td>1:10430102:10452153</td></tr>\n",
       "\t<tr><td>ENSG00000175279</td><td> 4</td><td>4</td><td>3</td><td> 8</td><td>7</td><td> 7</td><td>15</td><td> 6</td><td>10</td><td></td><td>2</td><td>2</td><td>3</td><td> 8</td><td> 6</td><td>12</td><td> 9</td><td> 8</td><td>ENSG00000175279_CENPS     </td><td>1:10430443:10442809</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 2  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " GeneID & S025NM-01.keep.merge.unique.sort.bam & S025QG-02.keep.merge.unique.sort.bam & S025RE-03.keep.merge.unique.sort.bam & S025SC-04.keep.merge.unique.sort.bam & S025TA-05.keep.merge.unique.sort.bam & S025U8-06.keep.merge.unique.sort.bam & S025V6-07.keep.merge.unique.sort.bam & S025W4-08.keep.merge.unique.sort.bam & S025X2-09.keep.merge.unique.sort.bam &  & S026MK-27.keep.merge.unique.sort.bam & S026NI-28.keep.merge.unique.sort.bam & S026S8-29.keep.merge.unique.sort.bam & S026T6-30.keep.merge.unique.sort.bam & S026XZ-31.keep.merge.unique.sort.bam & S026YX-32.keep.merge.unique.sort.bam & S0270N-33.keep.merge.unique.sort.bam & S0271L-34.keep.merge.unique.sort.bam & feature & hg38\\_location\\\\\n",
       " <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000251503 & 12 & 7 & 5 & 12 & 9 & 13 & 20 & 11 & 16 &  & 6 & 8 & 6 & 11 & 11 & 13 & 15 & 12 & ENSG00000251503\\_CENPS-CORT & 1:10430102:10452153\\\\\n",
       "\t ENSG00000175279 &  4 & 4 & 3 &  8 & 7 &  7 & 15 &  6 & 10 &  & 2 & 2 & 3 &  8 &  6 & 12 &  9 &  8 & ENSG00000175279\\_CENPS      & 1:10430443:10442809\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 2  37\n",
       "\n",
       "| GeneID &lt;chr&gt; | S025NM-01.keep.merge.unique.sort.bam &lt;int&gt; | S025QG-02.keep.merge.unique.sort.bam &lt;int&gt; | S025RE-03.keep.merge.unique.sort.bam &lt;int&gt; | S025SC-04.keep.merge.unique.sort.bam &lt;int&gt; | S025TA-05.keep.merge.unique.sort.bam &lt;int&gt; | S025U8-06.keep.merge.unique.sort.bam &lt;int&gt; | S025V6-07.keep.merge.unique.sort.bam &lt;int&gt; | S025W4-08.keep.merge.unique.sort.bam &lt;int&gt; | S025X2-09.keep.merge.unique.sort.bam &lt;int&gt; |   | S026MK-27.keep.merge.unique.sort.bam &lt;int&gt; | S026NI-28.keep.merge.unique.sort.bam &lt;int&gt; | S026S8-29.keep.merge.unique.sort.bam &lt;int&gt; | S026T6-30.keep.merge.unique.sort.bam &lt;int&gt; | S026XZ-31.keep.merge.unique.sort.bam &lt;int&gt; | S026YX-32.keep.merge.unique.sort.bam &lt;int&gt; | S0270N-33.keep.merge.unique.sort.bam &lt;int&gt; | S0271L-34.keep.merge.unique.sort.bam &lt;int&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000251503 | 12 | 7 | 5 | 12 | 9 | 13 | 20 | 11 | 16 |  | 6 | 8 | 6 | 11 | 11 | 13 | 15 | 12 | ENSG00000251503_CENPS-CORT | 1:10430102:10452153 |\n",
       "| ENSG00000175279 |  4 | 4 | 3 |  8 | 7 |  7 | 15 |  6 | 10 |  | 2 | 2 | 3 |  8 |  6 | 12 |  9 |  8 | ENSG00000175279_CENPS      | 1:10430443:10442809 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          S025NM-01.keep.merge.unique.sort.bam\n",
       "1 ENSG00000251503 12                                  \n",
       "2 ENSG00000175279  4                                  \n",
       "  S025QG-02.keep.merge.unique.sort.bam S025RE-03.keep.merge.unique.sort.bam\n",
       "1 7                                    5                                   \n",
       "2 4                                    3                                   \n",
       "  S025SC-04.keep.merge.unique.sort.bam S025TA-05.keep.merge.unique.sort.bam\n",
       "1 12                                   9                                   \n",
       "2  8                                   7                                   \n",
       "  S025U8-06.keep.merge.unique.sort.bam S025V6-07.keep.merge.unique.sort.bam\n",
       "1 13                                   20                                  \n",
       "2  7                                   15                                  \n",
       "  S025W4-08.keep.merge.unique.sort.bam S025X2-09.keep.merge.unique.sort.bam \n",
       "1 11                                   16                                   \n",
       "2  6                                   10                                   \n",
       "  S026MK-27.keep.merge.unique.sort.bam S026NI-28.keep.merge.unique.sort.bam\n",
       "1 6                                    8                                   \n",
       "2 2                                    2                                   \n",
       "  S026S8-29.keep.merge.unique.sort.bam S026T6-30.keep.merge.unique.sort.bam\n",
       "1 6                                    11                                  \n",
       "2 3                                     8                                  \n",
       "  S026XZ-31.keep.merge.unique.sort.bam S026YX-32.keep.merge.unique.sort.bam\n",
       "1 11                                   13                                  \n",
       "2  6                                   12                                  \n",
       "  S0270N-33.keep.merge.unique.sort.bam S0271L-34.keep.merge.unique.sort.bam\n",
       "1 15                                   12                                  \n",
       "2  9                                    8                                  \n",
       "  feature                    hg38_location      \n",
       "1 ENSG00000251503_CENPS-CORT 1:10430102:10452153\n",
       "2 ENSG00000175279_CENPS      1:10430443:10442809"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 4  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENSG00000274587_LILRB3</td><td>CHR_HSCHR19_4_CTG3_1:54217271:54223706      </td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENSG00000277816_LILRB3</td><td>CHR_HSCHR19LRC_COX1_CTG3_1:54217107:54223520</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENSG00000275019_LILRB3</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1:54217293:54223789</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENSG00000204577_LILRB3</td><td>19:54216278:54223506                        </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 4  3\n",
       "\\begin{tabular}{lll}\n",
       " GeneID & feature & hg38\\_location\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000274587 & ENSG00000274587\\_LILRB3 & CHR\\_HSCHR19\\_4\\_CTG3\\_1:54217271:54223706      \\\\\n",
       "\t ENSG00000277816 & ENSG00000277816\\_LILRB3 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1:54217107:54223520\\\\\n",
       "\t ENSG00000275019 & ENSG00000275019\\_LILRB3 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1:54217293:54223789\\\\\n",
       "\t ENSG00000204577 & ENSG00000204577\\_LILRB3 & 19:54216278:54223506                        \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 4  3\n",
       "\n",
       "| GeneID &lt;chr&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000274587 | ENSG00000274587_LILRB3 | CHR_HSCHR19_4_CTG3_1:54217271:54223706       |\n",
       "| ENSG00000277816 | ENSG00000277816_LILRB3 | CHR_HSCHR19LRC_COX1_CTG3_1:54217107:54223520 |\n",
       "| ENSG00000275019 | ENSG00000275019_LILRB3 | CHR_HSCHR19LRC_PGF1_CTG3_1:54217293:54223789 |\n",
       "| ENSG00000204577 | ENSG00000204577_LILRB3 | 19:54216278:54223506                         |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          feature               \n",
       "1 ENSG00000274587 ENSG00000274587_LILRB3\n",
       "2 ENSG00000277816 ENSG00000277816_LILRB3\n",
       "3 ENSG00000275019 ENSG00000275019_LILRB3\n",
       "4 ENSG00000204577 ENSG00000204577_LILRB3\n",
       "  hg38_location                               \n",
       "1 CHR_HSCHR19_4_CTG3_1:54217271:54223706      \n",
       "2 CHR_HSCHR19LRC_COX1_CTG3_1:54217107:54223520\n",
       "3 CHR_HSCHR19LRC_PGF1_CTG3_1:54217293:54223789\n",
       "4 19:54216278:54223506                        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 33  7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene stable ID</th><th scope=col>Transcript stable ID</th><th scope=col>HGNC symbol</th><th scope=col>Transcript start (bp)</th><th scope=col>Transcript end (bp)</th><th scope=col>Chromosome/scaffold name</th><th scope=col>Strand</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000610675</td><td>LILRB3</td><td>54217271</td><td>54223597</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000611086</td><td>LILRB3</td><td>54217481</td><td>54223675</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000621660</td><td>LILRB3</td><td>54217544</td><td>54223597</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000613710</td><td>LILRB3</td><td>54217544</td><td>54223597</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000622425</td><td>LILRB3</td><td>54217712</td><td>54218314</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000614962</td><td>LILRB3</td><td>54219211</td><td>54219971</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000616782</td><td>LILRB3</td><td>54220458</td><td>54223626</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000274587</td><td>ENST00000622778</td><td>LILRB3</td><td>54222897</td><td>54223706</td><td>CHR_HSCHR19_4_CTG3_1      </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000615764</td><td>LILRB3</td><td>54217361</td><td>54223411</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000613833</td><td>LILRB3</td><td>54217343</td><td>54223462</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000622537</td><td>LILRB3</td><td>54217361</td><td>54223411</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000614546</td><td>LILRB3</td><td>54217361</td><td>54223411</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000619617</td><td>LILRB3</td><td>54217107</td><td>54223411</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000620460</td><td>LILRB3</td><td>54217361</td><td>54223411</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000621620</td><td>LILRB3</td><td>54221723</td><td>54223520</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000277816</td><td>ENST00000619090</td><td>LILRB3</td><td>54219028</td><td>54219788</td><td>CHR_HSCHR19LRC_COX1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000616493</td><td>LILRB3</td><td>54217293</td><td>54223601</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000613698</td><td>LILRB3</td><td>54217311</td><td>54223789</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000612057</td><td>LILRB3</td><td>54217487</td><td>54223601</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000622504</td><td>LILRB3</td><td>54217487</td><td>54223601</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000615387</td><td>LILRB3</td><td>54217547</td><td>54223601</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000620730</td><td>LILRB3</td><td>54217547</td><td>54223601</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000614249</td><td>LILRB3</td><td>54219214</td><td>54219974</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000275019</td><td>ENST00000611751</td><td>LILRB3</td><td>54221913</td><td>54223710</td><td>CHR_HSCHR19LRC_PGF1_CTG3_1</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000391750</td><td>LILRB3</td><td>54216278</td><td>54223506</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000414379</td><td>LILRB3</td><td>54216674</td><td>54222978</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000346401</td><td>LILRB3</td><td>54216884</td><td>54223039</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000245620</td><td>LILRB3</td><td>54216928</td><td>54222978</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000460208</td><td>LILRB3</td><td>54217096</td><td>54217698</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000469273</td><td>LILRB3</td><td>54217120</td><td>54220790</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000436504</td><td>LILRB3</td><td>54218595</td><td>54219355</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000468668</td><td>LILRB3</td><td>54221978</td><td>54223087</td><td>19                        </td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000204577</td><td>ENST00000445347</td><td>LILRB3</td><td>54222011</td><td>54223087</td><td>19                        </td><td>-1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 33  7\n",
       "\\begin{tabular}{lllllll}\n",
       " Gene stable ID & Transcript stable ID & HGNC symbol & Transcript start (bp) & Transcript end (bp) & Chromosome/scaffold name & Strand\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ENSG00000274587 & ENST00000610675 & LILRB3 & 54217271 & 54223597 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000274587 & ENST00000611086 & LILRB3 & 54217481 & 54223675 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000274587 & ENST00000621660 & LILRB3 & 54217544 & 54223597 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000274587 & ENST00000613710 & LILRB3 & 54217544 & 54223597 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000274587 & ENST00000622425 & LILRB3 & 54217712 & 54218314 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000274587 & ENST00000614962 & LILRB3 & 54219211 & 54219971 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000274587 & ENST00000616782 & LILRB3 & 54220458 & 54223626 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000274587 & ENST00000622778 & LILRB3 & 54222897 & 54223706 & CHR\\_HSCHR19\\_4\\_CTG3\\_1       & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000615764 & LILRB3 & 54217361 & 54223411 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000613833 & LILRB3 & 54217343 & 54223462 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000622537 & LILRB3 & 54217361 & 54223411 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000614546 & LILRB3 & 54217361 & 54223411 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000619617 & LILRB3 & 54217107 & 54223411 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000620460 & LILRB3 & 54217361 & 54223411 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000621620 & LILRB3 & 54221723 & 54223520 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000277816 & ENST00000619090 & LILRB3 & 54219028 & 54219788 & CHR\\_HSCHR19LRC\\_COX1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000616493 & LILRB3 & 54217293 & 54223601 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000613698 & LILRB3 & 54217311 & 54223789 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000612057 & LILRB3 & 54217487 & 54223601 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000622504 & LILRB3 & 54217487 & 54223601 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000615387 & LILRB3 & 54217547 & 54223601 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000620730 & LILRB3 & 54217547 & 54223601 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000614249 & LILRB3 & 54219214 & 54219974 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000275019 & ENST00000611751 & LILRB3 & 54221913 & 54223710 & CHR\\_HSCHR19LRC\\_PGF1\\_CTG3\\_1 & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000391750 & LILRB3 & 54216278 & 54223506 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000414379 & LILRB3 & 54216674 & 54222978 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000346401 & LILRB3 & 54216884 & 54223039 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000245620 & LILRB3 & 54216928 & 54222978 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000460208 & LILRB3 & 54217096 & 54217698 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000469273 & LILRB3 & 54217120 & 54220790 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000436504 & LILRB3 & 54218595 & 54219355 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000468668 & LILRB3 & 54221978 & 54223087 & 19                         & -1\\\\\n",
       "\t ENSG00000204577 & ENST00000445347 & LILRB3 & 54222011 & 54223087 & 19                         & -1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 33  7\n",
       "\n",
       "| Gene stable ID &lt;chr&gt; | Transcript stable ID &lt;chr&gt; | HGNC symbol &lt;chr&gt; | Transcript start (bp) &lt;int&gt; | Transcript end (bp) &lt;int&gt; | Chromosome/scaffold name &lt;chr&gt; | Strand &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| ENSG00000274587 | ENST00000610675 | LILRB3 | 54217271 | 54223597 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000274587 | ENST00000611086 | LILRB3 | 54217481 | 54223675 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000274587 | ENST00000621660 | LILRB3 | 54217544 | 54223597 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000274587 | ENST00000613710 | LILRB3 | 54217544 | 54223597 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000274587 | ENST00000622425 | LILRB3 | 54217712 | 54218314 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000274587 | ENST00000614962 | LILRB3 | 54219211 | 54219971 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000274587 | ENST00000616782 | LILRB3 | 54220458 | 54223626 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000274587 | ENST00000622778 | LILRB3 | 54222897 | 54223706 | CHR_HSCHR19_4_CTG3_1       | -1 |\n",
       "| ENSG00000277816 | ENST00000615764 | LILRB3 | 54217361 | 54223411 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000277816 | ENST00000613833 | LILRB3 | 54217343 | 54223462 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000277816 | ENST00000622537 | LILRB3 | 54217361 | 54223411 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000277816 | ENST00000614546 | LILRB3 | 54217361 | 54223411 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000277816 | ENST00000619617 | LILRB3 | 54217107 | 54223411 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000277816 | ENST00000620460 | LILRB3 | 54217361 | 54223411 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000277816 | ENST00000621620 | LILRB3 | 54221723 | 54223520 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000277816 | ENST00000619090 | LILRB3 | 54219028 | 54219788 | CHR_HSCHR19LRC_COX1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000616493 | LILRB3 | 54217293 | 54223601 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000613698 | LILRB3 | 54217311 | 54223789 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000612057 | LILRB3 | 54217487 | 54223601 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000622504 | LILRB3 | 54217487 | 54223601 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000615387 | LILRB3 | 54217547 | 54223601 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000620730 | LILRB3 | 54217547 | 54223601 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000614249 | LILRB3 | 54219214 | 54219974 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000275019 | ENST00000611751 | LILRB3 | 54221913 | 54223710 | CHR_HSCHR19LRC_PGF1_CTG3_1 | -1 |\n",
       "| ENSG00000204577 | ENST00000391750 | LILRB3 | 54216278 | 54223506 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000414379 | LILRB3 | 54216674 | 54222978 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000346401 | LILRB3 | 54216884 | 54223039 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000245620 | LILRB3 | 54216928 | 54222978 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000460208 | LILRB3 | 54217096 | 54217698 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000469273 | LILRB3 | 54217120 | 54220790 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000436504 | LILRB3 | 54218595 | 54219355 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000468668 | LILRB3 | 54221978 | 54223087 | 19                         | -1 |\n",
       "| ENSG00000204577 | ENST00000445347 | LILRB3 | 54222011 | 54223087 | 19                         | -1 |\n",
       "\n"
      ],
      "text/plain": [
       "   Gene stable ID  Transcript stable ID HGNC symbol Transcript start (bp)\n",
       "1  ENSG00000274587 ENST00000610675      LILRB3      54217271             \n",
       "2  ENSG00000274587 ENST00000611086      LILRB3      54217481             \n",
       "3  ENSG00000274587 ENST00000621660      LILRB3      54217544             \n",
       "4  ENSG00000274587 ENST00000613710      LILRB3      54217544             \n",
       "5  ENSG00000274587 ENST00000622425      LILRB3      54217712             \n",
       "6  ENSG00000274587 ENST00000614962      LILRB3      54219211             \n",
       "7  ENSG00000274587 ENST00000616782      LILRB3      54220458             \n",
       "8  ENSG00000274587 ENST00000622778      LILRB3      54222897             \n",
       "9  ENSG00000277816 ENST00000615764      LILRB3      54217361             \n",
       "10 ENSG00000277816 ENST00000613833      LILRB3      54217343             \n",
       "11 ENSG00000277816 ENST00000622537      LILRB3      54217361             \n",
       "12 ENSG00000277816 ENST00000614546      LILRB3      54217361             \n",
       "13 ENSG00000277816 ENST00000619617      LILRB3      54217107             \n",
       "14 ENSG00000277816 ENST00000620460      LILRB3      54217361             \n",
       "15 ENSG00000277816 ENST00000621620      LILRB3      54221723             \n",
       "16 ENSG00000277816 ENST00000619090      LILRB3      54219028             \n",
       "17 ENSG00000275019 ENST00000616493      LILRB3      54217293             \n",
       "18 ENSG00000275019 ENST00000613698      LILRB3      54217311             \n",
       "19 ENSG00000275019 ENST00000612057      LILRB3      54217487             \n",
       "20 ENSG00000275019 ENST00000622504      LILRB3      54217487             \n",
       "21 ENSG00000275019 ENST00000615387      LILRB3      54217547             \n",
       "22 ENSG00000275019 ENST00000620730      LILRB3      54217547             \n",
       "23 ENSG00000275019 ENST00000614249      LILRB3      54219214             \n",
       "24 ENSG00000275019 ENST00000611751      LILRB3      54221913             \n",
       "25 ENSG00000204577 ENST00000391750      LILRB3      54216278             \n",
       "26 ENSG00000204577 ENST00000414379      LILRB3      54216674             \n",
       "27 ENSG00000204577 ENST00000346401      LILRB3      54216884             \n",
       "28 ENSG00000204577 ENST00000245620      LILRB3      54216928             \n",
       "29 ENSG00000204577 ENST00000460208      LILRB3      54217096             \n",
       "30 ENSG00000204577 ENST00000469273      LILRB3      54217120             \n",
       "31 ENSG00000204577 ENST00000436504      LILRB3      54218595             \n",
       "32 ENSG00000204577 ENST00000468668      LILRB3      54221978             \n",
       "33 ENSG00000204577 ENST00000445347      LILRB3      54222011             \n",
       "   Transcript end (bp) Chromosome/scaffold name   Strand\n",
       "1  54223597            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "2  54223675            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "3  54223597            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "4  54223597            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "5  54218314            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "6  54219971            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "7  54223626            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "8  54223706            CHR_HSCHR19_4_CTG3_1       -1    \n",
       "9  54223411            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "10 54223462            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "11 54223411            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "12 54223411            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "13 54223411            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "14 54223411            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "15 54223520            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "16 54219788            CHR_HSCHR19LRC_COX1_CTG3_1 -1    \n",
       "17 54223601            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "18 54223789            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "19 54223601            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "20 54223601            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "21 54223601            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "22 54223601            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "23 54219974            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "24 54223710            CHR_HSCHR19LRC_PGF1_CTG3_1 -1    \n",
       "25 54223506            19                         -1    \n",
       "26 54222978            19                         -1    \n",
       "27 54223039            19                         -1    \n",
       "28 54222978            19                         -1    \n",
       "29 54217698            19                         -1    \n",
       "30 54220790            19                         -1    \n",
       "31 54219355            19                         -1    \n",
       "32 54223087            19                         -1    \n",
       "33 54223087            19                         -1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>hg38_location</th><th scope=col>S025NM</th><th scope=col>S025QG</th><th scope=col>S025RE</th><th scope=col>S025SC</th><th scope=col>S025TA</th><th scope=col>S025U8</th><th scope=col>S025V6</th><th scope=col>S025W4</th><th scope=col></th><th scope=col>S026KO</th><th scope=col>S026MK</th><th scope=col>S026NI</th><th scope=col>S026S8</th><th scope=col>S026T6</th><th scope=col>S026XZ</th><th scope=col>S026YX</th><th scope=col>S0270N</th><th scope=col>S0271L</th><th scope=col>avg_reads</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000204577_LILRB3</td><td>19:54216278:54223506</td><td>3169</td><td>3223</td><td>3127</td><td>3431</td><td>4586</td><td>4368</td><td>4174</td><td>5938</td><td></td><td>1732</td><td>2834</td><td>3425</td><td>2512</td><td>3566</td><td>2106</td><td>6458</td><td>4155</td><td>2800</td><td>3480.118</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " feature & hg38\\_location & S025NM & S025QG & S025RE & S025SC & S025TA & S025U8 & S025V6 & S025W4 &  & S026KO & S026MK & S026NI & S026S8 & S026T6 & S026XZ & S026YX & S0270N & S0271L & avg\\_reads\\\\\n",
       " <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000204577\\_LILRB3 & 19:54216278:54223506 & 3169 & 3223 & 3127 & 3431 & 4586 & 4368 & 4174 & 5938 &  & 1732 & 2834 & 3425 & 2512 & 3566 & 2106 & 6458 & 4155 & 2800 & 3480.118\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| feature &lt;chr&gt; | hg38_location &lt;chr&gt; | S025NM &lt;int&gt; | S025QG &lt;int&gt; | S025RE &lt;int&gt; | S025SC &lt;int&gt; | S025TA &lt;int&gt; | S025U8 &lt;int&gt; | S025V6 &lt;int&gt; | S025W4 &lt;int&gt; |   | S026KO &lt;int&gt; | S026MK &lt;int&gt; | S026NI &lt;int&gt; | S026S8 &lt;int&gt; | S026T6 &lt;int&gt; | S026XZ &lt;int&gt; | S026YX &lt;int&gt; | S0270N &lt;int&gt; | S0271L &lt;int&gt; | avg_reads &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000204577_LILRB3 | 19:54216278:54223506 | 3169 | 3223 | 3127 | 3431 | 4586 | 4368 | 4174 | 5938 |  | 1732 | 2834 | 3425 | 2512 | 3566 | 2106 | 6458 | 4155 | 2800 | 3480.118 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature                hg38_location        S025NM S025QG S025RE S025SC\n",
       "1 ENSG00000204577_LILRB3 19:54216278:54223506 3169   3223   3127   3431  \n",
       "  S025TA S025U8 S025V6 S025W4  S026KO S026MK S026NI S026S8 S026T6 S026XZ\n",
       "1 4586   4368   4174   5938    1732   2834   3425   2512   3566   2106  \n",
       "  S026YX S0270N S0271L avg_reads\n",
       "1 6458   4155   2800   3480.118 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>S025NM-01.keep.merge.unique.sort.bam</th><th scope=col>S025QG-02.keep.merge.unique.sort.bam</th><th scope=col>S025RE-03.keep.merge.unique.sort.bam</th><th scope=col>S025SC-04.keep.merge.unique.sort.bam</th><th scope=col>S025TA-05.keep.merge.unique.sort.bam</th><th scope=col>S025U8-06.keep.merge.unique.sort.bam</th><th scope=col>S025V6-07.keep.merge.unique.sort.bam</th><th scope=col>S025W4-08.keep.merge.unique.sort.bam</th><th scope=col>S025X2-09.keep.merge.unique.sort.bam</th><th scope=col></th><th scope=col>S026MK-27.keep.merge.unique.sort.bam</th><th scope=col>S026NI-28.keep.merge.unique.sort.bam</th><th scope=col>S026S8-29.keep.merge.unique.sort.bam</th><th scope=col>S026T6-30.keep.merge.unique.sort.bam</th><th scope=col>S026XZ-31.keep.merge.unique.sort.bam</th><th scope=col>S026YX-32.keep.merge.unique.sort.bam</th><th scope=col>S0270N-33.keep.merge.unique.sort.bam</th><th scope=col>S0271L-34.keep.merge.unique.sort.bam</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000204577</td><td>2510</td><td>2648</td><td>2331</td><td>2606</td><td>3862</td><td>3231</td><td>3771</td><td>4522</td><td>4101</td><td></td><td>1728</td><td>2806</td><td>2352</td><td>1964</td><td>2088</td><td>5311</td><td>3262</td><td>2588</td><td>ENSG00000204577_LILRB3</td><td>19:54216278:54223506</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " GeneID & S025NM-01.keep.merge.unique.sort.bam & S025QG-02.keep.merge.unique.sort.bam & S025RE-03.keep.merge.unique.sort.bam & S025SC-04.keep.merge.unique.sort.bam & S025TA-05.keep.merge.unique.sort.bam & S025U8-06.keep.merge.unique.sort.bam & S025V6-07.keep.merge.unique.sort.bam & S025W4-08.keep.merge.unique.sort.bam & S025X2-09.keep.merge.unique.sort.bam &  & S026MK-27.keep.merge.unique.sort.bam & S026NI-28.keep.merge.unique.sort.bam & S026S8-29.keep.merge.unique.sort.bam & S026T6-30.keep.merge.unique.sort.bam & S026XZ-31.keep.merge.unique.sort.bam & S026YX-32.keep.merge.unique.sort.bam & S0270N-33.keep.merge.unique.sort.bam & S0271L-34.keep.merge.unique.sort.bam & feature & hg38\\_location\\\\\n",
       " <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000204577 & 2510 & 2648 & 2331 & 2606 & 3862 & 3231 & 3771 & 4522 & 4101 &  & 1728 & 2806 & 2352 & 1964 & 2088 & 5311 & 3262 & 2588 & ENSG00000204577\\_LILRB3 & 19:54216278:54223506\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| GeneID &lt;chr&gt; | S025NM-01.keep.merge.unique.sort.bam &lt;int&gt; | S025QG-02.keep.merge.unique.sort.bam &lt;int&gt; | S025RE-03.keep.merge.unique.sort.bam &lt;int&gt; | S025SC-04.keep.merge.unique.sort.bam &lt;int&gt; | S025TA-05.keep.merge.unique.sort.bam &lt;int&gt; | S025U8-06.keep.merge.unique.sort.bam &lt;int&gt; | S025V6-07.keep.merge.unique.sort.bam &lt;int&gt; | S025W4-08.keep.merge.unique.sort.bam &lt;int&gt; | S025X2-09.keep.merge.unique.sort.bam &lt;int&gt; |   | S026MK-27.keep.merge.unique.sort.bam &lt;int&gt; | S026NI-28.keep.merge.unique.sort.bam &lt;int&gt; | S026S8-29.keep.merge.unique.sort.bam &lt;int&gt; | S026T6-30.keep.merge.unique.sort.bam &lt;int&gt; | S026XZ-31.keep.merge.unique.sort.bam &lt;int&gt; | S026YX-32.keep.merge.unique.sort.bam &lt;int&gt; | S0270N-33.keep.merge.unique.sort.bam &lt;int&gt; | S0271L-34.keep.merge.unique.sort.bam &lt;int&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000204577 | 2510 | 2648 | 2331 | 2606 | 3862 | 3231 | 3771 | 4522 | 4101 |  | 1728 | 2806 | 2352 | 1964 | 2088 | 5311 | 3262 | 2588 | ENSG00000204577_LILRB3 | 19:54216278:54223506 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          S025NM-01.keep.merge.unique.sort.bam\n",
       "1 ENSG00000204577 2510                                \n",
       "  S025QG-02.keep.merge.unique.sort.bam S025RE-03.keep.merge.unique.sort.bam\n",
       "1 2648                                 2331                                \n",
       "  S025SC-04.keep.merge.unique.sort.bam S025TA-05.keep.merge.unique.sort.bam\n",
       "1 2606                                 3862                                \n",
       "  S025U8-06.keep.merge.unique.sort.bam S025V6-07.keep.merge.unique.sort.bam\n",
       "1 3231                                 3771                                \n",
       "  S025W4-08.keep.merge.unique.sort.bam S025X2-09.keep.merge.unique.sort.bam \n",
       "1 4522                                 4101                                 \n",
       "  S026MK-27.keep.merge.unique.sort.bam S026NI-28.keep.merge.unique.sort.bam\n",
       "1 1728                                 2806                                \n",
       "  S026S8-29.keep.merge.unique.sort.bam S026T6-30.keep.merge.unique.sort.bam\n",
       "1 2352                                 1964                                \n",
       "  S026XZ-31.keep.merge.unique.sort.bam S026YX-32.keep.merge.unique.sort.bam\n",
       "1 2088                                 5311                                \n",
       "  S0270N-33.keep.merge.unique.sort.bam S0271L-34.keep.merge.unique.sort.bam\n",
       "1 3262                                 2588                                \n",
       "  feature                hg38_location       \n",
       "1 ENSG00000204577_LILRB3 19:54216278:54223506"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000164442</td><td>ENSG00000164442_CITED2</td><td>6:139371807:139374620</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  3\n",
       "\\begin{tabular}{lll}\n",
       " GeneID & feature & hg38\\_location\\\\\n",
       " <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000164442 & ENSG00000164442\\_CITED2 & 6:139371807:139374620\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  3\n",
       "\n",
       "| GeneID &lt;chr&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| ENSG00000164442 | ENSG00000164442_CITED2 | 6:139371807:139374620 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          feature                hg38_location        \n",
       "1 ENSG00000164442 ENSG00000164442_CITED2 6:139371807:139374620"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 4  7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Gene stable ID</th><th scope=col>Transcript stable ID</th><th scope=col>HGNC symbol</th><th scope=col>Transcript start (bp)</th><th scope=col>Transcript end (bp)</th><th scope=col>Chromosome/scaffold name</th><th scope=col>Strand</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000164442</td><td>ENST00000367651</td><td>CITED2</td><td>139371807</td><td>139374620</td><td>6</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000164442</td><td>ENST00000537332</td><td>CITED2</td><td>139372255</td><td>139374213</td><td>6</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000164442</td><td>ENST00000536159</td><td>CITED2</td><td>139372255</td><td>139374362</td><td>6</td><td>-1</td></tr>\n",
       "\t<tr><td>ENSG00000164442</td><td>ENST00000618718</td><td>CITED2</td><td>139373074</td><td>139374605</td><td>6</td><td>-1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 4  7\n",
       "\\begin{tabular}{lllllll}\n",
       " Gene stable ID & Transcript stable ID & HGNC symbol & Transcript start (bp) & Transcript end (bp) & Chromosome/scaffold name & Strand\\\\\n",
       " <chr> & <chr> & <chr> & <int> & <int> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t ENSG00000164442 & ENST00000367651 & CITED2 & 139371807 & 139374620 & 6 & -1\\\\\n",
       "\t ENSG00000164442 & ENST00000537332 & CITED2 & 139372255 & 139374213 & 6 & -1\\\\\n",
       "\t ENSG00000164442 & ENST00000536159 & CITED2 & 139372255 & 139374362 & 6 & -1\\\\\n",
       "\t ENSG00000164442 & ENST00000618718 & CITED2 & 139373074 & 139374605 & 6 & -1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 4  7\n",
       "\n",
       "| Gene stable ID &lt;chr&gt; | Transcript stable ID &lt;chr&gt; | HGNC symbol &lt;chr&gt; | Transcript start (bp) &lt;int&gt; | Transcript end (bp) &lt;int&gt; | Chromosome/scaffold name &lt;chr&gt; | Strand &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| ENSG00000164442 | ENST00000367651 | CITED2 | 139371807 | 139374620 | 6 | -1 |\n",
       "| ENSG00000164442 | ENST00000537332 | CITED2 | 139372255 | 139374213 | 6 | -1 |\n",
       "| ENSG00000164442 | ENST00000536159 | CITED2 | 139372255 | 139374362 | 6 | -1 |\n",
       "| ENSG00000164442 | ENST00000618718 | CITED2 | 139373074 | 139374605 | 6 | -1 |\n",
       "\n"
      ],
      "text/plain": [
       "  Gene stable ID  Transcript stable ID HGNC symbol Transcript start (bp)\n",
       "1 ENSG00000164442 ENST00000367651      CITED2      139371807            \n",
       "2 ENSG00000164442 ENST00000537332      CITED2      139372255            \n",
       "3 ENSG00000164442 ENST00000536159      CITED2      139372255            \n",
       "4 ENSG00000164442 ENST00000618718      CITED2      139373074            \n",
       "  Transcript end (bp) Chromosome/scaffold name Strand\n",
       "1 139374620           6                        -1    \n",
       "2 139374213           6                        -1    \n",
       "3 139374362           6                        -1    \n",
       "4 139374605           6                        -1    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>feature</th><th scope=col>hg38_location</th><th scope=col>S025NM</th><th scope=col>S025QG</th><th scope=col>S025RE</th><th scope=col>S025SC</th><th scope=col>S025TA</th><th scope=col>S025U8</th><th scope=col>S025V6</th><th scope=col>S025W4</th><th scope=col></th><th scope=col>S026KO</th><th scope=col>S026MK</th><th scope=col>S026NI</th><th scope=col>S026S8</th><th scope=col>S026T6</th><th scope=col>S026XZ</th><th scope=col>S026YX</th><th scope=col>S0270N</th><th scope=col>S0271L</th><th scope=col>avg_reads</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000164442_CITED2</td><td>6:139371807:139374620</td><td>1559</td><td>3605</td><td>1881</td><td>1754</td><td>1799</td><td>1525</td><td>3660</td><td>1322</td><td></td><td>897</td><td>1466</td><td>1743</td><td>3530</td><td>2690</td><td>854</td><td>1887</td><td>1201</td><td>2270</td><td>1992</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " feature & hg38\\_location & S025NM & S025QG & S025RE & S025SC & S025TA & S025U8 & S025V6 & S025W4 &  & S026KO & S026MK & S026NI & S026S8 & S026T6 & S026XZ & S026YX & S0270N & S0271L & avg\\_reads\\\\\n",
       " <chr> & <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <dbl>\\\\\n",
       "\\hline\n",
       "\t ENSG00000164442\\_CITED2 & 6:139371807:139374620 & 1559 & 3605 & 1881 & 1754 & 1799 & 1525 & 3660 & 1322 &  & 897 & 1466 & 1743 & 3530 & 2690 & 854 & 1887 & 1201 & 2270 & 1992\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| feature &lt;chr&gt; | hg38_location &lt;chr&gt; | S025NM &lt;int&gt; | S025QG &lt;int&gt; | S025RE &lt;int&gt; | S025SC &lt;int&gt; | S025TA &lt;int&gt; | S025U8 &lt;int&gt; | S025V6 &lt;int&gt; | S025W4 &lt;int&gt; |   | S026KO &lt;int&gt; | S026MK &lt;int&gt; | S026NI &lt;int&gt; | S026S8 &lt;int&gt; | S026T6 &lt;int&gt; | S026XZ &lt;int&gt; | S026YX &lt;int&gt; | S0270N &lt;int&gt; | S0271L &lt;int&gt; | avg_reads &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000164442_CITED2 | 6:139371807:139374620 | 1559 | 3605 | 1881 | 1754 | 1799 | 1525 | 3660 | 1322 |  | 897 | 1466 | 1743 | 3530 | 2690 | 854 | 1887 | 1201 | 2270 | 1992 |\n",
       "\n"
      ],
      "text/plain": [
       "  feature                hg38_location         S025NM S025QG S025RE S025SC\n",
       "1 ENSG00000164442_CITED2 6:139371807:139374620 1559   3605   1881   1754  \n",
       "  S025TA S025U8 S025V6 S025W4  S026KO S026MK S026NI S026S8 S026T6 S026XZ\n",
       "1 1799   1525   3660   1322    897    1466   1743   3530   2690   854   \n",
       "  S026YX S0270N S0271L avg_reads\n",
       "1 1887   1201   2270   1992     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1  37</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>GeneID</th><th scope=col>S025NM-01.keep.merge.unique.sort.bam</th><th scope=col>S025QG-02.keep.merge.unique.sort.bam</th><th scope=col>S025RE-03.keep.merge.unique.sort.bam</th><th scope=col>S025SC-04.keep.merge.unique.sort.bam</th><th scope=col>S025TA-05.keep.merge.unique.sort.bam</th><th scope=col>S025U8-06.keep.merge.unique.sort.bam</th><th scope=col>S025V6-07.keep.merge.unique.sort.bam</th><th scope=col>S025W4-08.keep.merge.unique.sort.bam</th><th scope=col>S025X2-09.keep.merge.unique.sort.bam</th><th scope=col></th><th scope=col>S026MK-27.keep.merge.unique.sort.bam</th><th scope=col>S026NI-28.keep.merge.unique.sort.bam</th><th scope=col>S026S8-29.keep.merge.unique.sort.bam</th><th scope=col>S026T6-30.keep.merge.unique.sort.bam</th><th scope=col>S026XZ-31.keep.merge.unique.sort.bam</th><th scope=col>S026YX-32.keep.merge.unique.sort.bam</th><th scope=col>S0270N-33.keep.merge.unique.sort.bam</th><th scope=col>S0271L-34.keep.merge.unique.sort.bam</th><th scope=col>feature</th><th scope=col>hg38_location</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ENSG00000164442</td><td>1046</td><td>2604</td><td>1150</td><td>1082</td><td>1184</td><td>1052</td><td>2350</td><td>851</td><td>4067</td><td></td><td>766</td><td>991</td><td>2276</td><td>1166</td><td>584</td><td>1314</td><td>851</td><td>1550</td><td>ENSG00000164442_CITED2</td><td>6:139371807:139374620</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1  37\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " GeneID & S025NM-01.keep.merge.unique.sort.bam & S025QG-02.keep.merge.unique.sort.bam & S025RE-03.keep.merge.unique.sort.bam & S025SC-04.keep.merge.unique.sort.bam & S025TA-05.keep.merge.unique.sort.bam & S025U8-06.keep.merge.unique.sort.bam & S025V6-07.keep.merge.unique.sort.bam & S025W4-08.keep.merge.unique.sort.bam & S025X2-09.keep.merge.unique.sort.bam &  & S026MK-27.keep.merge.unique.sort.bam & S026NI-28.keep.merge.unique.sort.bam & S026S8-29.keep.merge.unique.sort.bam & S026T6-30.keep.merge.unique.sort.bam & S026XZ-31.keep.merge.unique.sort.bam & S026YX-32.keep.merge.unique.sort.bam & S0270N-33.keep.merge.unique.sort.bam & S0271L-34.keep.merge.unique.sort.bam & feature & hg38\\_location\\\\\n",
       " <chr> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> &  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ENSG00000164442 & 1046 & 2604 & 1150 & 1082 & 1184 & 1052 & 2350 & 851 & 4067 &  & 766 & 991 & 2276 & 1166 & 584 & 1314 & 851 & 1550 & ENSG00000164442\\_CITED2 & 6:139371807:139374620\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1  37\n",
       "\n",
       "| GeneID &lt;chr&gt; | S025NM-01.keep.merge.unique.sort.bam &lt;int&gt; | S025QG-02.keep.merge.unique.sort.bam &lt;int&gt; | S025RE-03.keep.merge.unique.sort.bam &lt;int&gt; | S025SC-04.keep.merge.unique.sort.bam &lt;int&gt; | S025TA-05.keep.merge.unique.sort.bam &lt;int&gt; | S025U8-06.keep.merge.unique.sort.bam &lt;int&gt; | S025V6-07.keep.merge.unique.sort.bam &lt;int&gt; | S025W4-08.keep.merge.unique.sort.bam &lt;int&gt; | S025X2-09.keep.merge.unique.sort.bam &lt;int&gt; |   | S026MK-27.keep.merge.unique.sort.bam &lt;int&gt; | S026NI-28.keep.merge.unique.sort.bam &lt;int&gt; | S026S8-29.keep.merge.unique.sort.bam &lt;int&gt; | S026T6-30.keep.merge.unique.sort.bam &lt;int&gt; | S026XZ-31.keep.merge.unique.sort.bam &lt;int&gt; | S026YX-32.keep.merge.unique.sort.bam &lt;int&gt; | S0270N-33.keep.merge.unique.sort.bam &lt;int&gt; | S0271L-34.keep.merge.unique.sort.bam &lt;int&gt; | feature &lt;chr&gt; | hg38_location &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| ENSG00000164442 | 1046 | 2604 | 1150 | 1082 | 1184 | 1052 | 2350 | 851 | 4067 |  | 766 | 991 | 2276 | 1166 | 584 | 1314 | 851 | 1550 | ENSG00000164442_CITED2 | 6:139371807:139374620 |\n",
       "\n"
      ],
      "text/plain": [
       "  GeneID          S025NM-01.keep.merge.unique.sort.bam\n",
       "1 ENSG00000164442 1046                                \n",
       "  S025QG-02.keep.merge.unique.sort.bam S025RE-03.keep.merge.unique.sort.bam\n",
       "1 2604                                 1150                                \n",
       "  S025SC-04.keep.merge.unique.sort.bam S025TA-05.keep.merge.unique.sort.bam\n",
       "1 1082                                 1184                                \n",
       "  S025U8-06.keep.merge.unique.sort.bam S025V6-07.keep.merge.unique.sort.bam\n",
       "1 1052                                 2350                                \n",
       "  S025W4-08.keep.merge.unique.sort.bam S025X2-09.keep.merge.unique.sort.bam \n",
       "1 851                                  4067                                 \n",
       "  S026MK-27.keep.merge.unique.sort.bam S026NI-28.keep.merge.unique.sort.bam\n",
       "1 766                                  991                                 \n",
       "  S026S8-29.keep.merge.unique.sort.bam S026T6-30.keep.merge.unique.sort.bam\n",
       "1 2276                                 1166                                \n",
       "  S026XZ-31.keep.merge.unique.sort.bam S026YX-32.keep.merge.unique.sort.bam\n",
       "1 584                                  1314                                \n",
       "  S0270N-33.keep.merge.unique.sort.bam S0271L-34.keep.merge.unique.sort.bam\n",
       "1 851                                  1550                                \n",
       "  feature                hg38_location        \n",
       "1 ENSG00000164442_CITED2 6:139371807:139374620"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#print(to_plot[avg_reads_WASPCorrection == 0])\n",
    "genes_transcripts <- fread(\"~/spivakov/HRJ_monocytes/external_data/ensembl/ensembl94_transcript_coords_hg38_strand.txt\")\n",
    "#print(genes_transcripts)\n",
    "\n",
    "### BTBD1: same ENSGID; the counts all genuinely go to zero\n",
    "genes_locs_join[feature %like% \"BTBD1\"]\n",
    "genes_transcripts[`HGNC symbol` == \"BTBD1\"]\n",
    "orig[feature %like% \"BTBD1\"]\n",
    "wasp_feat[feature %like% \"BTBD1\"]\n",
    "wasp_feat[feature %like% \"ENSG00000064726\"] # this is the ENSG ID\n",
    "\n",
    "### EBLN2: same ENSGID; the counts all genuinely go to zero\n",
    "genes_locs_join[feature %like% \"EBLN2\"]\n",
    "genes_transcripts[`HGNC symbol` == \"EBLN2\"]\n",
    "orig[feature %like% \"EBLN2\"]\n",
    "wasp_feat[feature %like% \"EBLN2\"]\n",
    "\n",
    "### CENPS: same ENSGID; the counts all genuinely go to zero\n",
    "genes_locs_join[feature %like% \"CENPS\"]\n",
    "genes_transcripts[`HGNC symbol` == \"CENPS\"]\n",
    "orig[feature %like% \"CENPS\"]\n",
    "wasp_feat[feature %like% \"CENPS\"]\n",
    "\n",
    "### LILRB3: same ENSGID; the counts all genuinely go to zero\n",
    "genes_locs_join[feature %like% \"LILRB3\"]\n",
    "genes_transcripts[`HGNC symbol` == \"LILRB3\"]\n",
    "orig[feature %like% \"LILRB3\"]\n",
    "wasp_feat[feature %like% \"LILRB3\"]\n",
    "\n",
    "### CITED2: same ENSGID; the counts all genuinely go to zero\n",
    "genes_locs_join[feature %like% \"CITED2\"]\n",
    "genes_transcripts[`HGNC symbol` == \"CITED2\"]\n",
    "orig[feature %like% \"CITED2\"]\n",
    "wasp_feat[feature %like% \"CITED2\"]\n",
    "\n",
    "### Now check these regions in WASP reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for featureCounts {Rsubread}\"><tr><td>featureCounts {Rsubread}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Count Reads by Genomic Features</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>Assign mapped sequencing reads to specified genomic features.</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "featureCounts(files,\n",
       "\n",
       "    # annotation\n",
       "    annot.inbuilt = \"mm10\",\n",
       "    annot.ext = NULL,\n",
       "    isGTFAnnotationFile = FALSE,\n",
       "    GTF.featureType = \"exon\",\n",
       "    GTF.attrType = \"gene_id\",\n",
       "    GTF.attrType.extra = NULL,\n",
       "    chrAliases = NULL,\n",
       "    \n",
       "    # level of summarization\n",
       "    useMetaFeatures = TRUE,\n",
       "    \n",
       "    # overlap between reads and features\n",
       "    allowMultiOverlap = FALSE,\n",
       "    minOverlap = 1,\n",
       "    fracOverlap = 0,\n",
       "    fracOverlapFeature = 0,\n",
       "    largestOverlap = FALSE,\n",
       "    nonOverlap = NULL,\n",
       "    nonOverlapFeature = NULL,\n",
       "\n",
       "    # Read shift, extension and reduction\n",
       "    readShiftType = \"upstream\",\n",
       "    readShiftSize = 0,\n",
       "    readExtension5 = 0,\n",
       "    readExtension3 = 0,\n",
       "    read2pos = NULL,\n",
       "    \n",
       "    # multi-mapping reads\n",
       "    countMultiMappingReads = TRUE,\n",
       "\n",
       "    # fractional counting\n",
       "    fraction = FALSE,\n",
       "\n",
       "    # long reads\n",
       "    isLongRead = FALSE,\n",
       "\n",
       "    # read filtering\n",
       "    minMQS = 0,\n",
       "    splitOnly = FALSE,\n",
       "    nonSplitOnly = FALSE,\n",
       "    primaryOnly = FALSE,\n",
       "    ignoreDup = FALSE,\n",
       "    \n",
       "    # strandness\n",
       "    strandSpecific = 0,\n",
       "    \n",
       "    # exon-exon junctions\n",
       "    juncCounts = FALSE,\n",
       "    genome = NULL,\n",
       "    \n",
       "    # parameters specific to paired end reads\n",
       "    isPairedEnd = FALSE,\n",
       "    countReadPairs = TRUE,\n",
       "    requireBothEndsMapped = FALSE,\n",
       "    checkFragLength = FALSE,\n",
       "    minFragLength = 50,\n",
       "    maxFragLength = 600,\n",
       "    countChimericFragments = TRUE,    \n",
       "    autosort = TRUE,\n",
       "    \n",
       "    # number of CPU threads\n",
       "    nthreads = 1,\n",
       "\n",
       "    # read group\n",
       "    byReadGroup = FALSE,\n",
       "\n",
       "    # report assignment result for each read\n",
       "    reportReads = NULL,\n",
       "    reportReadsPath = NULL,\n",
       "\n",
       "    # miscellaneous\n",
       "    maxMOp = 10,\n",
       "    tmpDir = \".\",\n",
       "    verbose = FALSE)\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>files</code></td>\n",
       "<td>\n",
       "<p> a character vector giving names of input files containing read mapping results. The files can be in either SAM format or BAM format. The file format is automatically detected by the function.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>annot.inbuilt</code></td>\n",
       "<td>\n",
       "<p> a character string specifying an in-built annotation used for read summarization. It has four possible values including <code>\"mm10\"</code>, <code>\"mm9\"</code>, <code>\"hg38\"</code> and <code>\"hg19\"</code>, corresponding to the NCBI RefSeq annotations for genomes &lsquo;mm10&rsquo;, &lsquo;mm9&rsquo;, &lsquo;hg38&rsquo; and &lsquo;hg19&rsquo;, respectively. <code>\"mm10\"</code> by default. The in-built annotation has a SAF format (see below).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>annot.ext</code></td>\n",
       "<td>\n",
       "<p> A character string giving name of a user-provided annotation file or a data frame including user-provided annotation data. If the annotation is in GTF format, it can only be provided as a file. If it is in SAF format, it can be provided as a file or a data frame. See below for more details about SAF format annotation. If an annotation file is provided, it can be uncompressed or gzip compressed. Note that <code>annot.ext</code> will override <code>annot.inbuilt</code> if both provided.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>isGTFAnnotationFile</code></td>\n",
       "<td>\n",
       "<p> logical indicating whether the annotation provided via the <code>annot.ext</code> argument is in GTF format or not. <code>FALSE</code> by default. This option is only applicable when <code>annot.ext</code> is not <code>NULL</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>GTF.featureType</code></td>\n",
       "<td>\n",
       "<p> a character string or a vector of character strings giving the feature type or types used to select rows in the GTF annotation which will be used for read summarization. <code>\"exon\"</code> by default. This argument is only applicable when <code>isGTFAnnotationFile</code> is <code>TRUE</code>. Feature types can be found in the third column of a GTF annotation.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>GTF.attrType</code></td>\n",
       "<td>\n",
       "<p> a character string giving the attribute type in the GTF annotation which will be used to group features (eg. exons) into meta-features (eg. genes). <code>\"gene_id\"</code> by default. This argument is only applicable when <code>isGTFAnnotationFile</code> is <code>TRUE</code>. Attributes can be found in the ninth column of a GTF annotation.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>GTF.attrType.extra</code></td>\n",
       "<td>\n",
       "<p> a character vector specifying extra GTF attribute types that will also be included in the counting output. These attribute types will not be used to group features. <code>NULL</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>chrAliases</code></td>\n",
       "<td>\n",
       "<p> a character string giving the name of a chromosome name alias file. This should be a two-column comma-delimited text file. Chromosome name aliases included in this file are used to match chr names in annotation with those in the reads. First column in the file should include chr names in the annotation and second column should include chr names in the reads. Chr names are case sensitive. No column header should be included in the file.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>useMetaFeatures</code></td>\n",
       "<td>\n",
       "<p> logical indicating whether the read summarization should be performed at the feature level (eg. exons) or meta-feature level (eg. genes). If <code>TRUE</code>, features in the annotation (each row is a feature) will be grouped into meta-features, using their values in the <code>GeneID</code> column in the SAF-format annotation file or using the <code>GTF.attrType</code> attribute in the GTF-format annotation file, and then reads will be assiged to the meta-features instead of the features. See below for more details.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>allowMultiOverlap</code></td>\n",
       "<td>\n",
       "<p> logical indicating if a read is allowed to be assigned to more than one feature (or meta-feature) if it is found to overlap with more than one feature (or meta-feature). <code>FALSE</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>minOverlap</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the minimum number of overlapped bases required for assigning a read to a feature (or a meta-feature). For assignment of read pairs (fragments), number of overlapping bases from each read in the same pair will be summed. If a negative value is provided, then a gap of up to specified size will be allowed between read and the feature that the read is assigned to. <code>1</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fracOverlap</code></td>\n",
       "<td>\n",
       " <p><code>numeric</code> giving minimum fraction of overlapping bases in a read that is required for read assignment. Value should be within range [0,1]. <code>0</code> by default. Number of overlapping bases is counted from both reads if paired end. Both this option and <code>minOverlap</code> option need to be satisfied before a read can be assigned.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fracOverlapFeature</code></td>\n",
       "<td>\n",
       " <p><code>numeric</code> giving minimum fraction of bases included in a feature that is required for overlapping with a read or a read pair. Value should be within range [0,1]. <code>0</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>largestOverlap</code></td>\n",
       "<td>\n",
       "<p> If <code>TRUE</code>, a read (or read pair) will be assigned to the feature (or meta-feature) that has the largest number of overlapping bases, if the read (or read pair) overlaps with multiple features (or meta-features).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>nonOverlap</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the maximum number of bases in a read (or a read pair) that are allowed not to overlap the assigned feature. <code>NULL</code> by default (ie. no limit is set).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>nonOverlapFeature</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the maximum number of non-overlapping bases allowed in a feature during read assignment. <code>NULL</code> by default (ie. no limit is set).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>readShiftType</code></td>\n",
       "<td>\n",
       "<p> a character string indicating how reads should be shifted. It has four possible values including <code>upstream</code>, <code>downstream</code>, <code>left</code> and <code>right</code>. Read shifting is performed before read extension (<code>readExtension5</code> and <code>readExtension3</code>) and reduction (<code>read2pos</code>).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>readShiftSize</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> specifying the number of bases the reads will be shifted by. <code>0</code> by default. Negative value is not allowed.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>readExtension5</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the number of bases extended upstream from 5' end of each read. <code>0</code> by default. Read extension is performed after read shifting but before read reduction. Negative value is not allowed.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>readExtension3</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the number of bases extended downstream from 3' end of each read. <code>0</code> by default. Negative value is not allowed.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>read2pos</code></td>\n",
       "<td>\n",
       "<p> Specifying whether each read should be reduced to its 5' most base or 3' most base. It has three possible values: <code>NULL</code>, <code>5</code> (denoting 5' most base) and <code>3</code> (denoting 3' most base). Default value is <code>NULL</code>, ie. no read reduction will be performed. If a read is reduced to a single base, only that base will be considered for the read assignment. Read reduction is performed after read shifting and extension.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>countMultiMappingReads</code></td>\n",
       "<td>\n",
       "<p> logical indicating if multi-mapping reads/fragments should be counted, <code>TRUE</code> by default. &lsquo;NH&rsquo; tag is used to located multi-mapping reads in the input BAM/SAM files.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>fraction</code></td>\n",
       "<td>\n",
       "<p> logical indicating if fractional counts are produced for multi-mapping reads and/or multi-overlapping reads. <code>FALSE</code> by default. See below for more details.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>isLongRead</code></td>\n",
       "<td>\n",
       "<p> logical indicating if input data contain long reads. This option should be set to <code>TRUE</code> if counting Nanopore or PacBio long reads.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>minMQS</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the minimum mapping quality score a read must satisfy in order to be counted. For paired-end reads, at least one end should satisfy this criteria. <code>0</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>splitOnly</code></td>\n",
       "<td>\n",
       "<p> logical indicating whether only split alignments (their CIGAR strings contain letter 'N') should be included for summarization. <code>FALSE</code> by default. Example split alignments are exon-spanning reads from RNA-seq data. <code>useMetaFeatures</code> should be set to <code>FALSE</code> and <code>allowMultiOverlap</code> should be set to <code>TRUE</code>, if the purpose of summarization is to assign exon-spanning reads to all their overlapping exons.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>nonSplitOnly</code></td>\n",
       "<td>\n",
       "<p> logical indicating whether only non-split alignments (their CIGAR strings do not contain letter 'N') should be included for summarization. <code>FALSE</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>primaryOnly</code></td>\n",
       "<td>\n",
       "<p> logical indicating if only primary alignments should be counted. Primary and secondary alignments are identified using bit 0x100 in the Flag field of SAM/BAM files. If <code>TRUE</code>, all primary alignments in a dataset will be counted no matter they are from multi-mapping reads or not (ie. <code>countMultiMappingReads</code> is ignored).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>ignoreDup</code></td>\n",
       "<td>\n",
       "<p> logical indicating whether reads marked as duplicates should be ignored. <code>FALSE</code> by default. Read duplicates are identified using bit Ox400 in the FLAG field in SAM/BAM files. The whole fragment (read pair) will be ignored if paired end.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>strandSpecific</code></td>\n",
       "<td>\n",
       "<p> an integer vector indicating if strand-specific read counting should be performed. Length of the vector should be either <code>1</code> (meaning that the value is applied to all input files), or equal to the total number of input files provided. Each vector element should have one of the following three values: <code>0</code> (unstranded), <code>1</code> (stranded) and <code>2</code> (reversely stranded). Default value of this parameter is <code>0</code> (ie. unstranded read counting is performed for all input files).</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>juncCounts</code></td>\n",
       "<td>\n",
       "<p> logical indicating if number of reads supporting each exon-exon junction will be reported. Junctions are identified from those exon-spanning reads in input data. <code>FALSE</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>genome</code></td>\n",
       "<td>\n",
       "<p> a character string giving the name of a FASTA-format file that includes the reference sequences used in read mapping that produced the provided SAM/BAM files. <code>NULL</code> by default. This argument should only be used when <code>juncCounts</code> is <code>TRUE</code>. Note that providing reference sequences is optional when <code>juncCounts</code> is set to <code>TRUE</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>isPairedEnd</code></td>\n",
       "<td>\n",
       "<p> A logical scalar or a logical vector, indicating whether libraries contain paired-end reads or not. <code>FALSE</code> by default. For any library that contains paired-end reads, the 'countReadPairs' parameter controls if read pairs or reads should be counted.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>countReadPairs</code></td>\n",
       "<td>\n",
       "<p> A logical scalar or a logical vector, indicating if read pairs, instead of reads, should be counted for paired-end read data. <code>TRUE</code> by default. This parameter is only applicable for paired-end data. For single-end data, it is ignored.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>requireBothEndsMapped</code></td>\n",
       "<td>\n",
       "<p> logical indicating if both ends from the same fragment are required to be successfully aligned before the fragment can be assigned to a feature or meta-feature. This parameter is only appliable when <code>isPairedEnd</code> is <code>TRUE</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>checkFragLength</code></td>\n",
       "<td>\n",
       "<p> logical indicating if the two ends from the same fragment are required to satisify the fragment length criteria before the fragment can be assigned to a feature or meta-feature. This parameter is only appliable when <code>isPairedEnd</code> is <code>TRUE</code>. The fragment length criteria are specified via <code>minFragLength</code> and <code>maxFragLength</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>minFragLength</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the minimum fragment length for paired-end reads. <code>50</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>maxFragLength</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the maximum fragment length for paired-end reads. <code>600</code> by default. <code>minFragLength</code> and <code>maxFragLength</code> are only applicable when <code>isPairedEnd</code> is <code>TRUE</code>. Note that when a fragment spans two or more exons, the observed fragment length might be much bigger than the nominal fragment length.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>countChimericFragments</code></td>\n",
       "<td>\n",
       "<p> logical indicating whether a chimeric fragment, which has its two reads mapped to different chromosomes, should be counted or not. <code>TRUE</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>autosort</code></td>\n",
       "<td>\n",
       "<p> logical specifying if the automatic read sorting is enabled. This option is only applicable for paired-end reads. If <code>TRUE</code>, reads will be automatically sorted by their names if reads from the same pair are found not to be located next to each other in the input. No read sorting will be performed if there are no such reads found.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>nthreads</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the number of threads used for running this function. <code>1</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>byReadGroup</code></td>\n",
       "<td>\n",
       "<p> logical indicating if read counting will be performed for each individual read group. <code>FALSE</code> by default.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>reportReads</code></td>\n",
       "<td>\n",
       "<p> output detailed read assignment results for each read (or fragment if paired end). The detailed assignment results can be saved in three different formats including <code>CORE</code>, <code>SAM</code> and <code>BAM</code> (note that these values are case sensitive). Default value of this option is <code>NULL</code>, indicating not to report assignment results for reads. <code>CORE</code> format represents a tab-delimited text file that contains four columns including read name, status (assigned or the reason if not assigned), number of targets and target list. A target is a feature or a meta-feature. Items in the target lists is separated by comma. If a read is not assigned, its number of targets will be set as -1. For each input file, a text file is generated and its name is the input file name added with &lsquo;.featureCounts&rsquo;. When <code>SAM</code> or <code>BAM</code> format is specified, the detailed assignment results will be saved to SAM and BAM format files. Names of generated files are the input file names added with &lsquo;.featureCounts.sam&rsquo; or &lsquo;.featureCounts.bam&rsquo;. Three tags are used to describe read assignment results: XS, XN and XT. Tag XS gives the assignment status. Tag XN gives number of targets. Tag XT gives comma separated target list.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>reportReadsPath</code></td>\n",
       "<td>\n",
       "<p> a character string specifying the directory where files including detailed assignment results are saved. If <code>NULL</code>, the results will be saved to the current working directory.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>maxMOp</code></td>\n",
       "<td>\n",
       " <p><code>integer</code> giving the maximum number of &lsquo;M&rsquo; operations (matches or mis-matches) allowed in a CIGAR string. <code>10</code> by default. Both &lsquo;X&rsquo; and &lsquo;=&rsquo; operations are treated as &lsquo;M&rsquo; and adjacent &lsquo;M&rsquo; operations are merged in the CIGAR string.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>tmpDir</code></td>\n",
       "<td>\n",
       "<p> a character string specifying the directory under which intermediate files are saved (later removed). By default, current working directory is used.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>verbose</code></td>\n",
       "<td>\n",
       "<p> logical indicating if verbose information for debugging will be generated. This may include information such as unmatched chromosomes/contigs between reads and annotation.</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p><code>featureCounts</code> is a general-purpose read summarization function that can assign mapped reads from genomic DNA and RNA sequencing to genomic features or meta-features.\n",
       "</p>\n",
       "<p>The function takes as input a set of SAM or BAM files containing read mapping results.\n",
       "The files might be generated by <code>align</code> or <code>subjunc</code> or any suitable aligner.\n",
       "</p>\n",
       "<p><code>featureCounts</code> accepts two annotation formats to specify the genomic features: SAF (Simplified Annotation Format) or GTF/GFF.\n",
       "The GTF/GFF format is specified at <a href=\"https://genome.ucsc.edu/FAQ/FAQformat.html\">https://genome.ucsc.edu/FAQ/FAQformat.html</a>.\n",
       "SAF is a Simplified Annotation Format with five columns.\n",
       "It can be either a data.frame or a tab-delimited text file in the following style:\n",
       "</p>\n",
       "<pre>\n",
       "GeneID\t\tChr\tStart\tEnd\tStrand\n",
       "497097\t\tchr1\t3204563\t3207049\t-\n",
       "497097\t\tchr1\t3411783\t3411982\t-\n",
       "497097\t\tchr1\t3660633\t3661579\t-\n",
       "100503874\tchr1\t3637390\t3640590\t-\n",
       "100503874\tchr1\t3648928\t3648985\t-\n",
       "100038431\tchr1\t3670236\t3671869\t-\n",
       "...\n",
       "</pre>\n",
       "<p>SAF annotation includes the following five required columns: <code>GeneID</code>, <code>Chr</code>, <code>Start</code>, <code>End</code> and <code>Strand</code>.\n",
       "The <code>GeneID</code> column includes identifiers of features.\n",
       "These identifiers can be integer or character string.\n",
       "The <code>Chr</code> column includes chromosome names of features and these names should match the chromosome names includes in the provided SAM/BAM files.\n",
       "The <code>Start</code> and <code>End</code> columns include start and end coordinates of features, respectively.\n",
       "Both start and end coordinates are inclusive.\n",
       "The <code>Strand</code> column indicates the strand of features (<code>\"+\"</code> or <code>\"-\"</code>).\n",
       "Column names in a SAF annotation should be the same as those shown in the above example (case insensitive).\n",
       "Columns can be in any order.\n",
       "Extra columns are allowed to be added into the annotation.\n",
       "</p>\n",
       "<p>In-built annotations, which were generated based on NCBI RefSeq gene annotations, are provided to faciliate convenient read summarization.\n",
       "We provide in-built annotations for the following genomes: <code>\"hg38\"</code>, <code>\"hg19\"</code>, <code>\"mm10\"</code> and <code>\"mm9\"</code>.\n",
       "The content of in-built annotations can be accessed via the <code>getInBuiltAnnotation</code> function.\n",
       "These annotations have a SAF format.\n",
       "</p>\n",
       "<p>The in-built annotations are a modified version of NCBI RefSeq gene annotations.\n",
       "We downloaded the RefSeq gene annotations from NCBI ftp server (eg. RefSeq annotation for <code>mm10</code> was downloaded from <a href=\"ftp://ftp.ncbi.nlm.nih.gov/genomes/M_musculus/ARCHIVE/BUILD.38.1/mapview/seq_gene.md.gz\">ftp://ftp.ncbi.nlm.nih.gov/genomes/M_musculus/ARCHIVE/BUILD.38.1/mapview/seq_gene.md.gz</a>).\n",
       "We then used these annotations to create our in-built annotations.\n",
       "For each gene, we used its CDS (coding DNA sequence) and UTR (untranslated) regions provided in the original annotation to construct a list of exons, by merging and concatenating all CDSs and UTRs belonging to the same gene.\n",
       "Exons within each gene include all chromosomal bases included in the original CDS and UTR regions, but they include each base only once (they might be included multiple times in the original CDSs and UTRs).\n",
       "Also, exons within the same gene do not overlap with each other.\n",
       "</p>\n",
       "<p>Users may provide an external annotation for read summarization via the <code>annot.ext</code> argument.\n",
       "If the external annotation is in SAF format, it can be provdied as either a <code>data.frame</code> or a tab-delimited text file with proper column names included.\n",
       "If it is in GTF/GFF format, it should be provided as a file only (and <code>isGTFAnnotationFile</code> should be set to <code>TRUE</code>).\n",
       "</p>\n",
       "<p><code>featureCounts</code> function uses the <code>GTF.attrType</code> attribute in a GTF/GFF annotation to group features to form meta-features when performing read summarization at meta-feature level.\n",
       "</p>\n",
       "<p>The argument <code>useMetaFeatures</code> specifies whether read summarization should be performed at feature level or at meta-feature level.\n",
       "A feature represents a continuous genomic region and a meta-feature is a group of features.\n",
       "For instance, an exon is a feature and a gene comprising one or more exons is a meta-feature.\n",
       "To assign reads to meta-features, <code>featureCounts</code> firstly groups into meta-features the features that have the same gene identifiers.\n",
       "<code>featureCounts</code> looks for gene identifiers in <code>GeneID</code> column of a SAF annotation or by using <code>GTF.attrType</code> attribute in a GTF/GFF annotation.\n",
       "Then for each read <code>featureCounts</code> searches for meta-features that have at least one feature that overlaps with the read.\n",
       "A read might be found to overlap with more than one feature within the same meta-feature (eg. an exon-spanning read overlaps with more than one exon from the same gene), however this read will still be counted only once for the meta-feature.\n",
       "</p>\n",
       "<p>RNA-seq reads are often summarized to meta-features to produce read counts for genes.\n",
       "Further downstream analysis can then be carried out to discover differentially expressed genes.\n",
       "Feature-level summarization of RNA-seq data often yields exon-level read counts, which is useful for investigating alternative splicing of genes.\n",
       "</p>\n",
       "<p><code>featureCounts</code> provides multiple options to count multi-mapping reads (reads mapping to more than one location in the reference genome).\n",
       "Users can choose to ignore such reads by setting <code>countMultiMappingReads</code> to <code>FALSE</code>, or fully count every alignment reported for a multi-mapping read by setting <code>countMultiMappingReads</code> to <code>TRUE</code> (each alignment carries <code>1</code> count), or count each alignment fractionally by setting both <code>countMultiMappingReads</code> and <code>fraction</code> to <code>TRUE</code> (each alignment carries <code>1/x</code> count where <code>x</code> is the total number of alignments reported for the read).\n",
       "</p>\n",
       "<p><code>featureCounts</code> also provides multiple options to count multi-overlapping reads (reads overlapping with more than one meta-feature when conducting meta-feature-level summarization or overlapping with more than one feature when conducting feature-level summarization).\n",
       "Users can choose to ignore such reads by setting <code>allowMultiOverlap</code> to <code>FALSE</code>, or fully count them for each overlapping meta-feature/feature by setting <code>allowMultiOverlap</code> to <code>TRUE</code> (each overlapping meta-feature/feature receives a count of 1 from a read), or assign a fractional count to each overlapping meta-feature/feature by setting both <code>allowMultiOverlap</code> and <code>fraction</code> to <code>TRUE</code> (each overlapping meta-feature/feature receives a count of <code>1/y</code> from a read where <code>y</code> is the total number of meta-features/features overlapping with the read).\n",
       "</p>\n",
       "<p>If a read is both multi-mapping and multi-overlapping, then each overlapping meta-feature/feature will receive a fractional count of <code>1/(x*y)</code> if <code>countMultiMappingReads</code>, <code>allowMultiOverlap</code> and <code>fraction</code> are all set to <code>TRUE</code>.\n",
       "</p>\n",
       "<p>When <code>isPairedEnd</code> is <code>TRUE</code>, fragments (pairs of reads) instead of reads will be counted.\n",
       "<code>featureCounts</code> function checks if reads from the same pair are adjacent to each other (this could happen when reads were for example sorted by their mapping locations), and it automatically reorders those reads that belong to the same pair but are not adjacent to each other in the input read file.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       " \n",
       "<p>A list with the following components:\n",
       "</p>\n",
       "<table summary=\"R valueblock\">\n",
       "<tr valign=\"top\"><td><code>counts</code></td>\n",
       "<td>\n",
       "<p> a data matrix containing read counts for each feature or meta-feature for each library.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>counts_junction (optional)</code></td>\n",
       "<td>\n",
       "<p> a data frame including primary and secondary genes overlapping an exon junction, position information for the left splice site (&lsquo;Site1&rsquo;) and the right splice site (&lsquo;Site2&rsquo;) of a junction (including chromosome name, coordinate and strand) and number of supporting reads for each junction in each library. Both primary and secondary genes overlap at least one of the two splice sites of a junction. Secondary genes do not overlap more splice sites than the primary gene. When the primary and secondary genes overlap same number of splice sites, the gene with the smallest leftmost base position is selected as the primary gene. For each junction, no more than one primary gene is reported but there might be more than one secondary genes reported. If a junction does not overlap any genes, it has a missing value in the fields of primary gene and secondary gene. Note that this data frame is only generated when <code>juncCounts</code> is <code>TRUE</code>.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>annotation</code></td>\n",
       "<td>\n",
       "<p> a data frame with six columns including <code>GeneID</code>, <code>Chr</code>, <code>Start</code>, <code>End</code> and <code>Length</code>. When read summarization was performed at feature level, each row in the data frame is a feature and columns in the data frame give the annotation information for the features. When read summarization was performed at meta-feature level, each row in the data frame is a meta-feature and columns in the data frame give the annotation information for the features included in each meta feature except the <code>Length</code> column. For each meta-feature, the <code>Length</code> column gives the total length of genomic regions covered by features included in that meta-feature.   Note that this length will be less than the sum of lengths of features included in the meta-feature when there are features overlapping with each other. Also note the <code>GeneID</code> column gives Entrez gene identifiers when the in-built annotations are used.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>targets</code></td>\n",
       "<td>\n",
       "<p> a character vector giving sample information.</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>stat</code></td>\n",
       "<td>\n",
       "<p> a data frame giving number of successfully assigned reads in each library and also number of unassigned reads falling in each filtering category (unmapped, read type, singleton, mapping quality, chimera, fragment length, duplicate, multi-mapping, secondary alignment, split alignment, no features, overlapping length and assignment ambiguity). See Users Guide for more details.</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Author(s)</h3>\n",
       "\n",
       "<p>Wei Shi and Yang Liao</p>\n",
       "\n",
       "\n",
       "<h3>References</h3>\n",
       "\n",
       "<p>Yang Liao, Gordon K Smyth and Wei Shi (2019).\n",
       "The R package Rsubread is easier, faster, cheaper and better for alignment and quantification of RNA sequencing reads.\n",
       "<em>Nucleic Acids Research</em>, 47(8):e47.\n",
       "<a href=\"http://www.ncbi.nlm.nih.gov/pubmed/30783653\">http://www.ncbi.nlm.nih.gov/pubmed/30783653</a>\n",
       "</p>\n",
       "<p>Yang Liao, Gordon K Smyth and Wei Shi (2014).\n",
       "featureCounts: an efficient general-purpose program for assigning sequence reads to genomic features.\n",
       "<em>Bioinformatics</em>, 30(7):923-30.\n",
       "<a href=\"http://www.ncbi.nlm.nih.gov/pubmed/24227677\">http://www.ncbi.nlm.nih.gov/pubmed/24227677</a>\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>See Also</h3>\n",
       "\n",
       "<p><code>getInBuiltAnnotation</code></p>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>## Not run: \n",
       "# Summarize SAM format single-end reads using built-in RefSeq annotation for mouse genome mm10:\n",
       "featureCounts(files=\"mapping_results_SE.sam\",annot.inbuilt=\"mm10\")\n",
       "\n",
       "# Summarize single-end reads using a user-provided GTF annotation file:\n",
       "featureCounts(files=\"mapping_results_SE.sam\",annot.ext=\"annotation.gtf\",\n",
       "isGTFAnnotationFile=TRUE,GTF.featureType=\"exon\",GTF.attrType=\"gene_id\")\n",
       "\n",
       "# Summarize single-end reads using 5 threads:\n",
       "featureCounts(files=\"mapping_results_SE.sam\",nthreads=5)\n",
       "\n",
       "# Summarize BAM format single-end read data:\n",
       "featureCounts(files=\"mapping_results_SE.bam\")\n",
       "\n",
       "# Perform strand-specific read counting (strandSpecific=2 if reversely stranded):\n",
       "featureCounts(files=\"mapping_results_SE.bam\",strandSpecific=1)\n",
       "\n",
       "# Summarize paired-end reads and counting fragments (instead of reads):\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE)\n",
       "\n",
       "# Count fragments satisfying the fragment length criteria, eg. [50bp, 600bp]:\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,\n",
       "checkFragLength=TRUE,minFragLength=50,maxFragLength=600)\n",
       "\n",
       "# Count fragments that have both ends successfully aligned without checking the fragment length:\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,requireBothEndsMapped=TRUE)\n",
       "\n",
       "# Exclude chimeric fragments from fragment counting:\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,countChimericFragments=FALSE)\n",
       "\n",
       "## End(Not run)</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>Rsubread</em> version 2.8.2 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{featureCounts}{Count Reads by Genomic Features}{featureCounts}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "Assign mapped sequencing reads to specified genomic features.\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "featureCounts(files,\n",
       "\n",
       "    # annotation\n",
       "    annot.inbuilt = \"mm10\",\n",
       "    annot.ext = NULL,\n",
       "    isGTFAnnotationFile = FALSE,\n",
       "    GTF.featureType = \"exon\",\n",
       "    GTF.attrType = \"gene_id\",\n",
       "    GTF.attrType.extra = NULL,\n",
       "    chrAliases = NULL,\n",
       "    \n",
       "    # level of summarization\n",
       "    useMetaFeatures = TRUE,\n",
       "    \n",
       "    # overlap between reads and features\n",
       "    allowMultiOverlap = FALSE,\n",
       "    minOverlap = 1,\n",
       "    fracOverlap = 0,\n",
       "    fracOverlapFeature = 0,\n",
       "    largestOverlap = FALSE,\n",
       "    nonOverlap = NULL,\n",
       "    nonOverlapFeature = NULL,\n",
       "\n",
       "    # Read shift, extension and reduction\n",
       "    readShiftType = \"upstream\",\n",
       "    readShiftSize = 0,\n",
       "    readExtension5 = 0,\n",
       "    readExtension3 = 0,\n",
       "    read2pos = NULL,\n",
       "    \n",
       "    # multi-mapping reads\n",
       "    countMultiMappingReads = TRUE,\n",
       "\n",
       "    # fractional counting\n",
       "    fraction = FALSE,\n",
       "\n",
       "    # long reads\n",
       "    isLongRead = FALSE,\n",
       "\n",
       "    # read filtering\n",
       "    minMQS = 0,\n",
       "    splitOnly = FALSE,\n",
       "    nonSplitOnly = FALSE,\n",
       "    primaryOnly = FALSE,\n",
       "    ignoreDup = FALSE,\n",
       "    \n",
       "    # strandness\n",
       "    strandSpecific = 0,\n",
       "    \n",
       "    # exon-exon junctions\n",
       "    juncCounts = FALSE,\n",
       "    genome = NULL,\n",
       "    \n",
       "    # parameters specific to paired end reads\n",
       "    isPairedEnd = FALSE,\n",
       "    countReadPairs = TRUE,\n",
       "    requireBothEndsMapped = FALSE,\n",
       "    checkFragLength = FALSE,\n",
       "    minFragLength = 50,\n",
       "    maxFragLength = 600,\n",
       "    countChimericFragments = TRUE,    \n",
       "    autosort = TRUE,\n",
       "    \n",
       "    # number of CPU threads\n",
       "    nthreads = 1,\n",
       "\n",
       "    # read group\n",
       "    byReadGroup = FALSE,\n",
       "\n",
       "    # report assignment result for each read\n",
       "    reportReads = NULL,\n",
       "    reportReadsPath = NULL,\n",
       "\n",
       "    # miscellaneous\n",
       "    maxMOp = 10,\n",
       "    tmpDir = \".\",\n",
       "    verbose = FALSE)\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{files}]  a character vector giving names of input files containing read mapping results. The files can be in either SAM format or BAM format. The file format is automatically detected by the function.\n",
       "\\item[\\code{annot.inbuilt}]  a character string specifying an in-built annotation used for read summarization. It has four possible values including \\code{\"mm10\"}, \\code{\"mm9\"}, \\code{\"hg38\"} and \\code{\"hg19\"}, corresponding to the NCBI RefSeq annotations for genomes `mm10', `mm9', `hg38' and `hg19', respectively. \\code{\"mm10\"} by default. The in-built annotation has a SAF format (see below).\n",
       "\\item[\\code{annot.ext}]  A character string giving name of a user-provided annotation file or a data frame including user-provided annotation data. If the annotation is in GTF format, it can only be provided as a file. If it is in SAF format, it can be provided as a file or a data frame. See below for more details about SAF format annotation. If an annotation file is provided, it can be uncompressed or gzip compressed. Note that \\code{annot.ext} will override \\code{annot.inbuilt} if both provided.\n",
       "\\item[\\code{isGTFAnnotationFile}]  logical indicating whether the annotation provided via the \\code{annot.ext} argument is in GTF format or not. \\code{FALSE} by default. This option is only applicable when \\code{annot.ext} is not \\code{NULL}.\n",
       "\\item[\\code{GTF.featureType}]  a character string or a vector of character strings giving the feature type or types used to select rows in the GTF annotation which will be used for read summarization. \\code{\"exon\"} by default. This argument is only applicable when \\code{isGTFAnnotationFile} is \\code{TRUE}. Feature types can be found in the third column of a GTF annotation.\n",
       "\\item[\\code{GTF.attrType}]  a character string giving the attribute type in the GTF annotation which will be used to group features (eg. exons) into meta-features (eg. genes). \\code{\"gene\\_id\"} by default. This argument is only applicable when \\code{isGTFAnnotationFile} is \\code{TRUE}. Attributes can be found in the ninth column of a GTF annotation.\n",
       "\\item[\\code{GTF.attrType.extra}]  a character vector specifying extra GTF attribute types that will also be included in the counting output. These attribute types will not be used to group features. \\code{NULL} by default.\n",
       "\\item[\\code{chrAliases}]  a character string giving the name of a chromosome name alias file. This should be a two-column comma-delimited text file. Chromosome name aliases included in this file are used to match chr names in annotation with those in the reads. First column in the file should include chr names in the annotation and second column should include chr names in the reads. Chr names are case sensitive. No column header should be included in the file.\n",
       "\\item[\\code{useMetaFeatures}]  logical indicating whether the read summarization should be performed at the feature level (eg. exons) or meta-feature level (eg. genes). If \\code{TRUE}, features in the annotation (each row is a feature) will be grouped into meta-features, using their values in the \\code{GeneID} column in the SAF-format annotation file or using the \\code{GTF.attrType} attribute in the GTF-format annotation file, and then reads will be assiged to the meta-features instead of the features. See below for more details.\n",
       "\\item[\\code{allowMultiOverlap}]  logical indicating if a read is allowed to be assigned to more than one feature (or meta-feature) if it is found to overlap with more than one feature (or meta-feature). \\code{FALSE} by default.\n",
       "\\item[\\code{minOverlap}]  \\code{integer} giving the minimum number of overlapped bases required for assigning a read to a feature (or a meta-feature). For assignment of read pairs (fragments), number of overlapping bases from each read in the same pair will be summed. If a negative value is provided, then a gap of up to specified size will be allowed between read and the feature that the read is assigned to. \\code{1} by default.\n",
       "\\item[\\code{fracOverlap}]  \\code{numeric} giving minimum fraction of overlapping bases in a read that is required for read assignment. Value should be within range [0,1]. \\code{0} by default. Number of overlapping bases is counted from both reads if paired end. Both this option and \\code{minOverlap} option need to be satisfied before a read can be assigned.\n",
       "\\item[\\code{fracOverlapFeature}]  \\code{numeric} giving minimum fraction of bases included in a feature that is required for overlapping with a read or a read pair. Value should be within range [0,1]. \\code{0} by default.\n",
       "\\item[\\code{largestOverlap}]  If \\code{TRUE}, a read (or read pair) will be assigned to the feature (or meta-feature) that has the largest number of overlapping bases, if the read (or read pair) overlaps with multiple features (or meta-features).\n",
       "\\item[\\code{nonOverlap}]  \\code{integer} giving the maximum number of bases in a read (or a read pair) that are allowed not to overlap the assigned feature. \\code{NULL} by default (ie. no limit is set).\n",
       "\\item[\\code{nonOverlapFeature}]  \\code{integer} giving the maximum number of non-overlapping bases allowed in a feature during read assignment. \\code{NULL} by default (ie. no limit is set).\n",
       "\\item[\\code{readShiftType}]  a character string indicating how reads should be shifted. It has four possible values including \\code{upstream}, \\code{downstream}, \\code{left} and \\code{right}. Read shifting is performed before read extension (\\code{readExtension5} and \\code{readExtension3}) and reduction (\\code{read2pos}).\n",
       "\\item[\\code{readShiftSize}]  \\code{integer} specifying the number of bases the reads will be shifted by. \\code{0} by default. Negative value is not allowed.\n",
       "\\item[\\code{readExtension5}]  \\code{integer} giving the number of bases extended upstream from 5' end of each read. \\code{0} by default. Read extension is performed after read shifting but before read reduction. Negative value is not allowed.\n",
       "\\item[\\code{readExtension3}]  \\code{integer} giving the number of bases extended downstream from 3' end of each read. \\code{0} by default. Negative value is not allowed.\n",
       "\\item[\\code{read2pos}]  Specifying whether each read should be reduced to its 5' most base or 3' most base. It has three possible values: \\code{NULL}, \\code{5} (denoting 5' most base) and \\code{3} (denoting 3' most base). Default value is \\code{NULL}, ie. no read reduction will be performed. If a read is reduced to a single base, only that base will be considered for the read assignment. Read reduction is performed after read shifting and extension.\n",
       "\\item[\\code{countMultiMappingReads}]  logical indicating if multi-mapping reads/fragments should be counted, \\code{TRUE} by default. `NH' tag is used to located multi-mapping reads in the input BAM/SAM files.\n",
       "\\item[\\code{fraction}]  logical indicating if fractional counts are produced for multi-mapping reads and/or multi-overlapping reads. \\code{FALSE} by default. See below for more details.\n",
       "\\item[\\code{isLongRead}]  logical indicating if input data contain long reads. This option should be set to \\code{TRUE} if counting Nanopore or PacBio long reads.\n",
       "\\item[\\code{minMQS}]  \\code{integer} giving the minimum mapping quality score a read must satisfy in order to be counted. For paired-end reads, at least one end should satisfy this criteria. \\code{0} by default.\n",
       "\\item[\\code{splitOnly}]  logical indicating whether only split alignments (their CIGAR strings contain letter 'N') should be included for summarization. \\code{FALSE} by default. Example split alignments are exon-spanning reads from RNA-seq data. \\code{useMetaFeatures} should be set to \\code{FALSE} and \\code{allowMultiOverlap} should be set to \\code{TRUE}, if the purpose of summarization is to assign exon-spanning reads to all their overlapping exons.\n",
       "\\item[\\code{nonSplitOnly}]  logical indicating whether only non-split alignments (their CIGAR strings do not contain letter 'N') should be included for summarization. \\code{FALSE} by default.\n",
       "\\item[\\code{primaryOnly}]  logical indicating if only primary alignments should be counted. Primary and secondary alignments are identified using bit 0x100 in the Flag field of SAM/BAM files. If \\code{TRUE}, all primary alignments in a dataset will be counted no matter they are from multi-mapping reads or not (ie. \\code{countMultiMappingReads} is ignored).\n",
       "\\item[\\code{ignoreDup}]  logical indicating whether reads marked as duplicates should be ignored. \\code{FALSE} by default. Read duplicates are identified using bit Ox400 in the FLAG field in SAM/BAM files. The whole fragment (read pair) will be ignored if paired end.\n",
       "\\item[\\code{strandSpecific}]  an integer vector indicating if strand-specific read counting should be performed. Length of the vector should be either \\code{1} (meaning that the value is applied to all input files), or equal to the total number of input files provided. Each vector element should have one of the following three values: \\code{0} (unstranded), \\code{1} (stranded) and \\code{2} (reversely stranded). Default value of this parameter is \\code{0} (ie. unstranded read counting is performed for all input files).\n",
       "\\item[\\code{juncCounts}]  logical indicating if number of reads supporting each exon-exon junction will be reported. Junctions are identified from those exon-spanning reads in input data. \\code{FALSE} by default.\n",
       "\\item[\\code{genome}]  a character string giving the name of a FASTA-format file that includes the reference sequences used in read mapping that produced the provided SAM/BAM files. \\code{NULL} by default. This argument should only be used when \\code{juncCounts} is \\code{TRUE}. Note that providing reference sequences is optional when \\code{juncCounts} is set to \\code{TRUE}.\n",
       "\\item[\\code{isPairedEnd}]  A logical scalar or a logical vector, indicating whether libraries contain paired-end reads or not. \\code{FALSE} by default. For any library that contains paired-end reads, the 'countReadPairs' parameter controls if read pairs or reads should be counted.\n",
       "\\item[\\code{countReadPairs}]  A logical scalar or a logical vector, indicating if read pairs, instead of reads, should be counted for paired-end read data. \\code{TRUE} by default. This parameter is only applicable for paired-end data. For single-end data, it is ignored.\n",
       "\\item[\\code{requireBothEndsMapped}]  logical indicating if both ends from the same fragment are required to be successfully aligned before the fragment can be assigned to a feature or meta-feature. This parameter is only appliable when \\code{isPairedEnd} is \\code{TRUE}.\n",
       "\\item[\\code{checkFragLength}]  logical indicating if the two ends from the same fragment are required to satisify the fragment length criteria before the fragment can be assigned to a feature or meta-feature. This parameter is only appliable when \\code{isPairedEnd} is \\code{TRUE}. The fragment length criteria are specified via \\code{minFragLength} and \\code{maxFragLength}.\n",
       "\\item[\\code{minFragLength}]  \\code{integer} giving the minimum fragment length for paired-end reads. \\code{50} by default.\n",
       "\\item[\\code{maxFragLength}]  \\code{integer} giving the maximum fragment length for paired-end reads. \\code{600} by default. \\code{minFragLength} and \\code{maxFragLength} are only applicable when \\code{isPairedEnd} is \\code{TRUE}. Note that when a fragment spans two or more exons, the observed fragment length might be much bigger than the nominal fragment length.\n",
       "\\item[\\code{countChimericFragments}]  logical indicating whether a chimeric fragment, which has its two reads mapped to different chromosomes, should be counted or not. \\code{TRUE} by default.\n",
       "\\item[\\code{autosort}]  logical specifying if the automatic read sorting is enabled. This option is only applicable for paired-end reads. If \\code{TRUE}, reads will be automatically sorted by their names if reads from the same pair are found not to be located next to each other in the input. No read sorting will be performed if there are no such reads found.\n",
       "\\item[\\code{nthreads}]  \\code{integer} giving the number of threads used for running this function. \\code{1} by default.\n",
       "\\item[\\code{byReadGroup}]  logical indicating if read counting will be performed for each individual read group. \\code{FALSE} by default.\n",
       "\\item[\\code{reportReads}]  output detailed read assignment results for each read (or fragment if paired end). The detailed assignment results can be saved in three different formats including \\code{CORE}, \\code{SAM} and \\code{BAM} (note that these values are case sensitive). Default value of this option is \\code{NULL}, indicating not to report assignment results for reads. \\code{CORE} format represents a tab-delimited text file that contains four columns including read name, status (assigned or the reason if not assigned), number of targets and target list. A target is a feature or a meta-feature. Items in the target lists is separated by comma. If a read is not assigned, its number of targets will be set as -1. For each input file, a text file is generated and its name is the input file name added with `.featureCounts'. When \\code{SAM} or \\code{BAM} format is specified, the detailed assignment results will be saved to SAM and BAM format files. Names of generated files are the input file names added with `.featureCounts.sam' or `.featureCounts.bam'. Three tags are used to describe read assignment results: XS, XN and XT. Tag XS gives the assignment status. Tag XN gives number of targets. Tag XT gives comma separated target list.\n",
       "\\item[\\code{reportReadsPath}]  a character string specifying the directory where files including detailed assignment results are saved. If \\code{NULL}, the results will be saved to the current working directory.\n",
       "\\item[\\code{maxMOp}]  \\code{integer} giving the maximum number of `M' operations (matches or mis-matches) allowed in a CIGAR string. \\code{10} by default. Both `X' and `=' operations are treated as `M' and adjacent `M' operations are merged in the CIGAR string.\n",
       "\\item[\\code{tmpDir}]  a character string specifying the directory under which intermediate files are saved (later removed). By default, current working directory is used.\n",
       "\\item[\\code{verbose}]  logical indicating if verbose information for debugging will be generated. This may include information such as unmatched chromosomes/contigs between reads and annotation.\n",
       "\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "\\code{featureCounts} is a general-purpose read summarization function that can assign mapped reads from genomic DNA and RNA sequencing to genomic features or meta-features.\n",
       "\n",
       "The function takes as input a set of SAM or BAM files containing read mapping results.\n",
       "The files might be generated by \\code{\\LinkA{align}{align}} or \\code{\\LinkA{subjunc}{subjunc}} or any suitable aligner.\n",
       "\n",
       "\\code{featureCounts} accepts two annotation formats to specify the genomic features: SAF (Simplified Annotation Format) or GTF/GFF.\n",
       "The GTF/GFF format is specified at \\url{https://genome.ucsc.edu/FAQ/FAQformat.html}.\n",
       "SAF is a Simplified Annotation Format with five columns.\n",
       "It can be either a data.frame or a tab-delimited text file in the following style:\n",
       "\n",
       "\\begin{alltt}\n",
       "GeneID\t\tChr\tStart\tEnd\tStrand\n",
       "497097\t\tchr1\t3204563\t3207049\t-\n",
       "497097\t\tchr1\t3411783\t3411982\t-\n",
       "497097\t\tchr1\t3660633\t3661579\t-\n",
       "100503874\tchr1\t3637390\t3640590\t-\n",
       "100503874\tchr1\t3648928\t3648985\t-\n",
       "100038431\tchr1\t3670236\t3671869\t-\n",
       "...\n",
       "\\end{alltt}\n",
       "\n",
       "\n",
       "SAF annotation includes the following five required columns: \\code{GeneID}, \\code{Chr}, \\code{Start}, \\code{End} and \\code{Strand}.\n",
       "The \\code{GeneID} column includes identifiers of features.\n",
       "These identifiers can be integer or character string.\n",
       "The \\code{Chr} column includes chromosome names of features and these names should match the chromosome names includes in the provided SAM/BAM files.\n",
       "The \\code{Start} and \\code{End} columns include start and end coordinates of features, respectively.\n",
       "Both start and end coordinates are inclusive.\n",
       "The \\code{Strand} column indicates the strand of features (\\code{\"+\"} or \\code{\"-\"}).\n",
       "Column names in a SAF annotation should be the same as those shown in the above example (case insensitive).\n",
       "Columns can be in any order.\n",
       "Extra columns are allowed to be added into the annotation.\n",
       "\n",
       "In-built annotations, which were generated based on NCBI RefSeq gene annotations, are provided to faciliate convenient read summarization.\n",
       "We provide in-built annotations for the following genomes: \\code{\"hg38\"}, \\code{\"hg19\"}, \\code{\"mm10\"} and \\code{\"mm9\"}.\n",
       "The content of in-built annotations can be accessed via the \\code{\\LinkA{getInBuiltAnnotation}{getInBuiltAnnotation}} function.\n",
       "These annotations have a SAF format.\n",
       "\n",
       "The in-built annotations are a modified version of NCBI RefSeq gene annotations.\n",
       "We downloaded the RefSeq gene annotations from NCBI ftp server (eg. RefSeq annotation for \\code{mm10} was downloaded from \\url{ftp://ftp.ncbi.nlm.nih.gov/genomes/M_musculus/ARCHIVE/BUILD.38.1/mapview/seq_gene.md.gz}).\n",
       "We then used these annotations to create our in-built annotations.\n",
       "For each gene, we used its CDS (coding DNA sequence) and UTR (untranslated) regions provided in the original annotation to construct a list of exons, by merging and concatenating all CDSs and UTRs belonging to the same gene.\n",
       "Exons within each gene include all chromosomal bases included in the original CDS and UTR regions, but they include each base only once (they might be included multiple times in the original CDSs and UTRs).\n",
       "Also, exons within the same gene do not overlap with each other.\n",
       "\n",
       "Users may provide an external annotation for read summarization via the \\code{annot.ext} argument.\n",
       "If the external annotation is in SAF format, it can be provdied as either a \\code{data.frame} or a tab-delimited text file with proper column names included.\n",
       "If it is in GTF/GFF format, it should be provided as a file only (and \\code{isGTFAnnotationFile} should be set to \\code{TRUE}).\n",
       "\n",
       "\\code{featureCounts} function uses the \\code{GTF.attrType} attribute in a GTF/GFF annotation to group features to form meta-features when performing read summarization at meta-feature level.\n",
       "\n",
       "The argument \\code{useMetaFeatures} specifies whether read summarization should be performed at feature level or at meta-feature level.\n",
       "A feature represents a continuous genomic region and a meta-feature is a group of features.\n",
       "For instance, an exon is a feature and a gene comprising one or more exons is a meta-feature.\n",
       "To assign reads to meta-features, \\code{featureCounts} firstly groups into meta-features the features that have the same gene identifiers.\n",
       "\\code{featureCounts} looks for gene identifiers in \\code{GeneID} column of a SAF annotation or by using \\code{GTF.attrType} attribute in a GTF/GFF annotation.\n",
       "Then for each read \\code{featureCounts} searches for meta-features that have at least one feature that overlaps with the read.\n",
       "A read might be found to overlap with more than one feature within the same meta-feature (eg. an exon-spanning read overlaps with more than one exon from the same gene), however this read will still be counted only once for the meta-feature.\n",
       "\n",
       "RNA-seq reads are often summarized to meta-features to produce read counts for genes.\n",
       "Further downstream analysis can then be carried out to discover differentially expressed genes.\n",
       "Feature-level summarization of RNA-seq data often yields exon-level read counts, which is useful for investigating alternative splicing of genes.\n",
       "\n",
       "\\code{featureCounts} provides multiple options to count multi-mapping reads (reads mapping to more than one location in the reference genome).\n",
       "Users can choose to ignore such reads by setting \\code{countMultiMappingReads} to \\code{FALSE}, or fully count every alignment reported for a multi-mapping read by setting \\code{countMultiMappingReads} to \\code{TRUE} (each alignment carries \\code{1} count), or count each alignment fractionally by setting both \\code{countMultiMappingReads} and \\code{fraction} to \\code{TRUE} (each alignment carries \\code{1/x} count where \\code{x} is the total number of alignments reported for the read).\n",
       "\n",
       "\\code{featureCounts} also provides multiple options to count multi-overlapping reads (reads overlapping with more than one meta-feature when conducting meta-feature-level summarization or overlapping with more than one feature when conducting feature-level summarization).\n",
       "Users can choose to ignore such reads by setting \\code{allowMultiOverlap} to \\code{FALSE}, or fully count them for each overlapping meta-feature/feature by setting \\code{allowMultiOverlap} to \\code{TRUE} (each overlapping meta-feature/feature receives a count of 1 from a read), or assign a fractional count to each overlapping meta-feature/feature by setting both \\code{allowMultiOverlap} and \\code{fraction} to \\code{TRUE} (each overlapping meta-feature/feature receives a count of \\code{1/y} from a read where \\code{y} is the total number of meta-features/features overlapping with the read).\n",
       "\n",
       "If a read is both multi-mapping and multi-overlapping, then each overlapping meta-feature/feature will receive a fractional count of \\code{1/(x*y)} if \\code{countMultiMappingReads}, \\code{allowMultiOverlap} and \\code{fraction} are all set to \\code{TRUE}.\n",
       "\n",
       "When \\code{isPairedEnd} is \\code{TRUE}, fragments (pairs of reads) instead of reads will be counted.\n",
       "\\code{featureCounts} function checks if reads from the same pair are adjacent to each other (this could happen when reads were for example sorted by their mapping locations), and it automatically reorders those reads that belong to the same pair but are not adjacent to each other in the input read file.\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Value}\n",
       " \n",
       "A list with the following components:\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{counts}]  a data matrix containing read counts for each feature or meta-feature for each library.\n",
       "\\item[\\code{counts\\_junction (optional)}]  a data frame including primary and secondary genes overlapping an exon junction, position information for the left splice site (`Site1') and the right splice site (`Site2') of a junction (including chromosome name, coordinate and strand) and number of supporting reads for each junction in each library. Both primary and secondary genes overlap at least one of the two splice sites of a junction. Secondary genes do not overlap more splice sites than the primary gene. When the primary and secondary genes overlap same number of splice sites, the gene with the smallest leftmost base position is selected as the primary gene. For each junction, no more than one primary gene is reported but there might be more than one secondary genes reported. If a junction does not overlap any genes, it has a missing value in the fields of primary gene and secondary gene. Note that this data frame is only generated when \\code{juncCounts} is \\code{TRUE}.\n",
       "\\item[\\code{annotation}]  a data frame with six columns including \\code{GeneID}, \\code{Chr}, \\code{Start}, \\code{End} and \\code{Length}. When read summarization was performed at feature level, each row in the data frame is a feature and columns in the data frame give the annotation information for the features. When read summarization was performed at meta-feature level, each row in the data frame is a meta-feature and columns in the data frame give the annotation information for the features included in each meta feature except the \\code{Length} column. For each meta-feature, the \\code{Length} column gives the total length of genomic regions covered by features included in that meta-feature.   Note that this length will be less than the sum of lengths of features included in the meta-feature when there are features overlapping with each other. Also note the \\code{GeneID} column gives Entrez gene identifiers when the in-built annotations are used.\n",
       "\\item[\\code{targets}]  a character vector giving sample information.\n",
       "\\item[\\code{stat}]  a data frame giving number of successfully assigned reads in each library and also number of unassigned reads falling in each filtering category (unmapped, read type, singleton, mapping quality, chimera, fragment length, duplicate, multi-mapping, secondary alignment, split alignment, no features, overlapping length and assignment ambiguity). See Users Guide for more details.\n",
       "\\end{ldescription}\n",
       "\\end{Value}\n",
       "%\n",
       "\\begin{Author}\\relax\n",
       "Wei Shi and Yang Liao\n",
       "\\end{Author}\n",
       "%\n",
       "\\begin{References}\\relax\n",
       "Yang Liao, Gordon K Smyth and Wei Shi (2019).\n",
       "The R package Rsubread is easier, faster, cheaper and better for alignment and quantification of RNA sequencing reads.\n",
       "\\emph{Nucleic Acids Research}, 47(8):e47.\n",
       "\\url{http://www.ncbi.nlm.nih.gov/pubmed/30783653}\n",
       "\n",
       "Yang Liao, Gordon K Smyth and Wei Shi (2014).\n",
       "featureCounts: an efficient general-purpose program for assigning sequence reads to genomic features.\n",
       "\\emph{Bioinformatics}, 30(7):923-30.\n",
       "\\url{http://www.ncbi.nlm.nih.gov/pubmed/24227677}\n",
       "\\end{References}\n",
       "%\n",
       "\\begin{SeeAlso}\\relax\n",
       "\\code{\\LinkA{getInBuiltAnnotation}{getInBuiltAnnotation}}\n",
       "\\end{SeeAlso}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "## Not run: \n",
       "# Summarize SAM format single-end reads using built-in RefSeq annotation for mouse genome mm10:\n",
       "featureCounts(files=\"mapping_results_SE.sam\",annot.inbuilt=\"mm10\")\n",
       "\n",
       "# Summarize single-end reads using a user-provided GTF annotation file:\n",
       "featureCounts(files=\"mapping_results_SE.sam\",annot.ext=\"annotation.gtf\",\n",
       "isGTFAnnotationFile=TRUE,GTF.featureType=\"exon\",GTF.attrType=\"gene_id\")\n",
       "\n",
       "# Summarize single-end reads using 5 threads:\n",
       "featureCounts(files=\"mapping_results_SE.sam\",nthreads=5)\n",
       "\n",
       "# Summarize BAM format single-end read data:\n",
       "featureCounts(files=\"mapping_results_SE.bam\")\n",
       "\n",
       "# Perform strand-specific read counting (strandSpecific=2 if reversely stranded):\n",
       "featureCounts(files=\"mapping_results_SE.bam\",strandSpecific=1)\n",
       "\n",
       "# Summarize paired-end reads and counting fragments (instead of reads):\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE)\n",
       "\n",
       "# Count fragments satisfying the fragment length criteria, eg. [50bp, 600bp]:\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,\n",
       "checkFragLength=TRUE,minFragLength=50,maxFragLength=600)\n",
       "\n",
       "# Count fragments that have both ends successfully aligned without checking the fragment length:\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,requireBothEndsMapped=TRUE)\n",
       "\n",
       "# Exclude chimeric fragments from fragment counting:\n",
       "featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,countChimericFragments=FALSE)\n",
       "\n",
       "## End(Not run)\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "featureCounts             package:Rsubread             R Documentation\n",
       "\n",
       "_\bC_\bo_\bu_\bn_\bt _\bR_\be_\ba_\bd_\bs _\bb_\by _\bG_\be_\bn_\bo_\bm_\bi_\bc _\bF_\be_\ba_\bt_\bu_\br_\be_\bs\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     Assign mapped sequencing reads to specified genomic features.\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     featureCounts(files,\n",
       "     \n",
       "         # annotation\n",
       "         annot.inbuilt = \"mm10\",\n",
       "         annot.ext = NULL,\n",
       "         isGTFAnnotationFile = FALSE,\n",
       "         GTF.featureType = \"exon\",\n",
       "         GTF.attrType = \"gene_id\",\n",
       "         GTF.attrType.extra = NULL,\n",
       "         chrAliases = NULL,\n",
       "         \n",
       "         # level of summarization\n",
       "         useMetaFeatures = TRUE,\n",
       "         \n",
       "         # overlap between reads and features\n",
       "         allowMultiOverlap = FALSE,\n",
       "         minOverlap = 1,\n",
       "         fracOverlap = 0,\n",
       "         fracOverlapFeature = 0,\n",
       "         largestOverlap = FALSE,\n",
       "         nonOverlap = NULL,\n",
       "         nonOverlapFeature = NULL,\n",
       "     \n",
       "         # Read shift, extension and reduction\n",
       "         readShiftType = \"upstream\",\n",
       "         readShiftSize = 0,\n",
       "         readExtension5 = 0,\n",
       "         readExtension3 = 0,\n",
       "         read2pos = NULL,\n",
       "         \n",
       "         # multi-mapping reads\n",
       "         countMultiMappingReads = TRUE,\n",
       "     \n",
       "         # fractional counting\n",
       "         fraction = FALSE,\n",
       "     \n",
       "         # long reads\n",
       "         isLongRead = FALSE,\n",
       "     \n",
       "         # read filtering\n",
       "         minMQS = 0,\n",
       "         splitOnly = FALSE,\n",
       "         nonSplitOnly = FALSE,\n",
       "         primaryOnly = FALSE,\n",
       "         ignoreDup = FALSE,\n",
       "         \n",
       "         # strandness\n",
       "         strandSpecific = 0,\n",
       "         \n",
       "         # exon-exon junctions\n",
       "         juncCounts = FALSE,\n",
       "         genome = NULL,\n",
       "         \n",
       "         # parameters specific to paired end reads\n",
       "         isPairedEnd = FALSE,\n",
       "         countReadPairs = TRUE,\n",
       "         requireBothEndsMapped = FALSE,\n",
       "         checkFragLength = FALSE,\n",
       "         minFragLength = 50,\n",
       "         maxFragLength = 600,\n",
       "         countChimericFragments = TRUE,    \n",
       "         autosort = TRUE,\n",
       "         \n",
       "         # number of CPU threads\n",
       "         nthreads = 1,\n",
       "     \n",
       "         # read group\n",
       "         byReadGroup = FALSE,\n",
       "     \n",
       "         # report assignment result for each read\n",
       "         reportReads = NULL,\n",
       "         reportReadsPath = NULL,\n",
       "     \n",
       "         # miscellaneous\n",
       "         maxMOp = 10,\n",
       "         tmpDir = \".\",\n",
       "         verbose = FALSE)\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "   files: a character vector giving names of input files containing\n",
       "          read mapping results. The files can be in either SAM format\n",
       "          or BAM format. The file format is automatically detected by\n",
       "          the function.\n",
       "\n",
       "annot.inbuilt: a character string specifying an in-built annotation\n",
       "          used for read summarization. It has four possible values\n",
       "          including \"mm10\", \"mm9\", \"hg38\" and \"hg19\",\n",
       "          corresponding to the NCBI RefSeq annotations for genomes\n",
       "          `mm10', `mm9', `hg38' and `hg19', respectively. \"mm10\" by\n",
       "          default. The in-built annotation has a SAF format (see\n",
       "          below).\n",
       "\n",
       "annot.ext: A character string giving name of a user-provided annotation\n",
       "          file or a data frame including user-provided annotation data.\n",
       "          If the annotation is in GTF format, it can only be provided\n",
       "          as a file. If it is in SAF format, it can be provided as a\n",
       "          file or a data frame. See below for more details about SAF\n",
       "          format annotation. If an annotation file is provided, it can\n",
       "          be uncompressed or gzip compressed. Note that annot.ext\n",
       "          will override annot.inbuilt if both provided.\n",
       "\n",
       "isGTFAnnotationFile: logical indicating whether the annotation provided\n",
       "          via the annot.ext argument is in GTF format or not. FALSE\n",
       "          by default. This option is only applicable when annot.ext\n",
       "          is not NULL.\n",
       "\n",
       "GTF.featureType: a character string or a vector of character strings\n",
       "          giving the feature type or types used to select rows in the\n",
       "          GTF annotation which will be used for read summarization.\n",
       "          \"exon\" by default. This argument is only applicable when\n",
       "          isGTFAnnotationFile is TRUE. Feature types can be found\n",
       "          in the third column of a GTF annotation.\n",
       "\n",
       "GTF.attrType: a character string giving the attribute type in the GTF\n",
       "          annotation which will be used to group features (eg. exons)\n",
       "          into meta-features (eg. genes). \"gene_id\" by default. This\n",
       "          argument is only applicable when isGTFAnnotationFile is\n",
       "          TRUE. Attributes can be found in the ninth column of a GTF\n",
       "          annotation.\n",
       "\n",
       "GTF.attrType.extra: a character vector specifying extra GTF attribute\n",
       "          types that will also be included in the counting output.\n",
       "          These attribute types will not be used to group features.\n",
       "          NULL by default.\n",
       "\n",
       "chrAliases: a character string giving the name of a chromosome name\n",
       "          alias file. This should be a two-column comma-delimited text\n",
       "          file. Chromosome name aliases included in this file are used\n",
       "          to match chr names in annotation with those in the reads.\n",
       "          First column in the file should include chr names in the\n",
       "          annotation and second column should include chr names in the\n",
       "          reads. Chr names are case sensitive. No column header should\n",
       "          be included in the file.\n",
       "\n",
       "useMetaFeatures: logical indicating whether the read summarization\n",
       "          should be performed at the feature level (eg. exons) or\n",
       "          meta-feature level (eg. genes). If TRUE, features in the\n",
       "          annotation (each row is a feature) will be grouped into\n",
       "          meta-features, using their values in the GeneID column in\n",
       "          the SAF-format annotation file or using the GTF.attrType\n",
       "          attribute in the GTF-format annotation file, and then reads\n",
       "          will be assiged to the meta-features instead of the features.\n",
       "          See below for more details.\n",
       "\n",
       "allowMultiOverlap: logical indicating if a read is allowed to be\n",
       "          assigned to more than one feature (or meta-feature) if it is\n",
       "          found to overlap with more than one feature (or\n",
       "          meta-feature). FALSE by default.\n",
       "\n",
       "minOverlap: integer giving the minimum number of overlapped bases\n",
       "          required for assigning a read to a feature (or a\n",
       "          meta-feature). For assignment of read pairs (fragments),\n",
       "          number of overlapping bases from each read in the same pair\n",
       "          will be summed. If a negative value is provided, then a gap\n",
       "          of up to specified size will be allowed between read and the\n",
       "          feature that the read is assigned to. 1 by default.\n",
       "\n",
       "fracOverlap: numeric giving minimum fraction of overlapping bases in\n",
       "          a read that is required for read assignment. Value should be\n",
       "          within range [0,1]. 0 by default. Number of overlapping\n",
       "          bases is counted from both reads if paired end. Both this\n",
       "          option and minOverlap option need to be satisfied before a\n",
       "          read can be assigned.\n",
       "\n",
       "fracOverlapFeature: numeric giving minimum fraction of bases included\n",
       "          in a feature that is required for overlapping with a read or\n",
       "          a read pair. Value should be within range [0,1]. 0 by\n",
       "          default.\n",
       "\n",
       "largestOverlap: If TRUE, a read (or read pair) will be assigned to\n",
       "          the feature (or meta-feature) that has the largest number of\n",
       "          overlapping bases, if the read (or read pair) overlaps with\n",
       "          multiple features (or meta-features).\n",
       "\n",
       "nonOverlap: integer giving the maximum number of bases in a read (or\n",
       "          a read pair) that are allowed not to overlap the assigned\n",
       "          feature. NULL by default (ie. no limit is set).\n",
       "\n",
       "nonOverlapFeature: integer giving the maximum number of\n",
       "          non-overlapping bases allowed in a feature during read\n",
       "          assignment. NULL by default (ie. no limit is set).\n",
       "\n",
       "readShiftType: a character string indicating how reads should be\n",
       "          shifted. It has four possible values including upstream,\n",
       "          downstream, left and right. Read shifting is performed\n",
       "          before read extension (readExtension5 and readExtension3)\n",
       "          and reduction (read2pos).\n",
       "\n",
       "readShiftSize: integer specifying the number of bases the reads will\n",
       "          be shifted by. 0 by default. Negative value is not allowed.\n",
       "\n",
       "readExtension5: integer giving the number of bases extended upstream\n",
       "          from 5' end of each read. 0 by default. Read extension is\n",
       "          performed after read shifting but before read reduction.\n",
       "          Negative value is not allowed.\n",
       "\n",
       "readExtension3: integer giving the number of bases extended\n",
       "          downstream from 3' end of each read. 0 by default. Negative\n",
       "          value is not allowed.\n",
       "\n",
       "read2pos: Specifying whether each read should be reduced to its 5' most\n",
       "          base or 3' most base. It has three possible values: NULL,\n",
       "          5 (denoting 5' most base) and 3 (denoting 3' most base).\n",
       "          Default value is NULL, ie. no read reduction will be\n",
       "          performed. If a read is reduced to a single base, only that\n",
       "          base will be considered for the read assignment. Read\n",
       "          reduction is performed after read shifting and extension.\n",
       "\n",
       "countMultiMappingReads: logical indicating if multi-mapping\n",
       "          reads/fragments should be counted, TRUE by default. `NH'\n",
       "          tag is used to located multi-mapping reads in the input\n",
       "          BAM/SAM files.\n",
       "\n",
       "fraction: logical indicating if fractional counts are produced for\n",
       "          multi-mapping reads and/or multi-overlapping reads. FALSE\n",
       "          by default. See below for more details.\n",
       "\n",
       "isLongRead: logical indicating if input data contain long reads. This\n",
       "          option should be set to TRUE if counting Nanopore or PacBio\n",
       "          long reads.\n",
       "\n",
       "  minMQS: integer giving the minimum mapping quality score a read\n",
       "          must satisfy in order to be counted. For paired-end reads, at\n",
       "          least one end should satisfy this criteria. 0 by default.\n",
       "\n",
       "splitOnly: logical indicating whether only split alignments (their\n",
       "          CIGAR strings contain letter 'N') should be included for\n",
       "          summarization. FALSE by default. Example split alignments\n",
       "          are exon-spanning reads from RNA-seq data. useMetaFeatures\n",
       "          should be set to FALSE and allowMultiOverlap should be\n",
       "          set to TRUE, if the purpose of summarization is to assign\n",
       "          exon-spanning reads to all their overlapping exons.\n",
       "\n",
       "nonSplitOnly: logical indicating whether only non-split alignments\n",
       "          (their CIGAR strings do not contain letter 'N') should be\n",
       "          included for summarization. FALSE by default.\n",
       "\n",
       "primaryOnly: logical indicating if only primary alignments should be\n",
       "          counted. Primary and secondary alignments are identified\n",
       "          using bit 0x100 in the Flag field of SAM/BAM files. If\n",
       "          TRUE, all primary alignments in a dataset will be counted\n",
       "          no matter they are from multi-mapping reads or not (ie.\n",
       "          countMultiMappingReads is ignored).\n",
       "\n",
       "ignoreDup: logical indicating whether reads marked as duplicates should\n",
       "          be ignored. FALSE by default. Read duplicates are\n",
       "          identified using bit Ox400 in the FLAG field in SAM/BAM\n",
       "          files. The whole fragment (read pair) will be ignored if\n",
       "          paired end.\n",
       "\n",
       "strandSpecific: an integer vector indicating if strand-specific read\n",
       "          counting should be performed. Length of the vector should be\n",
       "          either 1 (meaning that the value is applied to all input\n",
       "          files), or equal to the total number of input files provided.\n",
       "          Each vector element should have one of the following three\n",
       "          values: 0 (unstranded), 1 (stranded) and 2 (reversely\n",
       "          stranded). Default value of this parameter is 0 (ie.\n",
       "          unstranded read counting is performed for all input files).\n",
       "\n",
       "juncCounts: logical indicating if number of reads supporting each\n",
       "          exon-exon junction will be reported. Junctions are identified\n",
       "          from those exon-spanning reads in input data. FALSE by\n",
       "          default.\n",
       "\n",
       "  genome: a character string giving the name of a FASTA-format file\n",
       "          that includes the reference sequences used in read mapping\n",
       "          that produced the provided SAM/BAM files. NULL by default.\n",
       "          This argument should only be used when juncCounts is\n",
       "          TRUE. Note that providing reference sequences is optional\n",
       "          when juncCounts is set to TRUE.\n",
       "\n",
       "isPairedEnd: A logical scalar or a logical vector, indicating whether\n",
       "          libraries contain paired-end reads or not. FALSE by\n",
       "          default. For any library that contains paired-end reads, the\n",
       "          'countReadPairs' parameter controls if read pairs or reads\n",
       "          should be counted.\n",
       "\n",
       "countReadPairs: A logical scalar or a logical vector, indicating if\n",
       "          read pairs, instead of reads, should be counted for\n",
       "          paired-end read data. TRUE by default. This parameter is\n",
       "          only applicable for paired-end data. For single-end data, it\n",
       "          is ignored.\n",
       "\n",
       "requireBothEndsMapped: logical indicating if both ends from the same\n",
       "          fragment are required to be successfully aligned before the\n",
       "          fragment can be assigned to a feature or meta-feature. This\n",
       "          parameter is only appliable when isPairedEnd is TRUE.\n",
       "\n",
       "checkFragLength: logical indicating if the two ends from the same\n",
       "          fragment are required to satisify the fragment length\n",
       "          criteria before the fragment can be assigned to a feature or\n",
       "          meta-feature. This parameter is only appliable when\n",
       "          isPairedEnd is TRUE. The fragment length criteria are\n",
       "          specified via minFragLength and maxFragLength.\n",
       "\n",
       "minFragLength: integer giving the minimum fragment length for\n",
       "          paired-end reads. 50 by default.\n",
       "\n",
       "maxFragLength: integer giving the maximum fragment length for\n",
       "          paired-end reads. 600 by default. minFragLength and\n",
       "          maxFragLength are only applicable when isPairedEnd is\n",
       "          TRUE. Note that when a fragment spans two or more exons,\n",
       "          the observed fragment length might be much bigger than the\n",
       "          nominal fragment length.\n",
       "\n",
       "countChimericFragments: logical indicating whether a chimeric fragment,\n",
       "          which has its two reads mapped to different chromosomes,\n",
       "          should be counted or not. TRUE by default.\n",
       "\n",
       "autosort: logical specifying if the automatic read sorting is enabled.\n",
       "          This option is only applicable for paired-end reads. If\n",
       "          TRUE, reads will be automatically sorted by their names if\n",
       "          reads from the same pair are found not to be located next to\n",
       "          each other in the input. No read sorting will be performed if\n",
       "          there are no such reads found.\n",
       "\n",
       "nthreads: integer giving the number of threads used for running this\n",
       "          function. 1 by default.\n",
       "\n",
       "byReadGroup: logical indicating if read counting will be performed for\n",
       "          each individual read group. FALSE by default.\n",
       "\n",
       "reportReads: output detailed read assignment results for each read (or\n",
       "          fragment if paired end). The detailed assignment results can\n",
       "          be saved in three different formats including CORE, SAM\n",
       "          and BAM (note that these values are case sensitive).\n",
       "          Default value of this option is NULL, indicating not to\n",
       "          report assignment results for reads. CORE format represents\n",
       "          a tab-delimited text file that contains four columns\n",
       "          including read name, status (assigned or the reason if not\n",
       "          assigned), number of targets and target list. A target is a\n",
       "          feature or a meta-feature. Items in the target lists is\n",
       "          separated by comma. If a read is not assigned, its number of\n",
       "          targets will be set as -1. For each input file, a text file\n",
       "          is generated and its name is the input file name added with\n",
       "          `.featureCounts'. When SAM or BAM format is specified,\n",
       "          the detailed assignment results will be saved to SAM and BAM\n",
       "          format files. Names of generated files are the input file\n",
       "          names added with `.featureCounts.sam' or\n",
       "          `.featureCounts.bam'. Three tags are used to describe read\n",
       "          assignment results: XS, XN and XT. Tag XS gives the\n",
       "          assignment status. Tag XN gives number of targets. Tag XT\n",
       "          gives comma separated target list.\n",
       "\n",
       "reportReadsPath: a character string specifying the directory where\n",
       "          files including detailed assignment results are saved. If\n",
       "          NULL, the results will be saved to the current working\n",
       "          directory.\n",
       "\n",
       "  maxMOp: integer giving the maximum number of `M' operations\n",
       "          (matches or mis-matches) allowed in a CIGAR string. 10 by\n",
       "          default. Both `X' and `=' operations are treated as `M' and\n",
       "          adjacent `M' operations are merged in the CIGAR string.\n",
       "\n",
       "  tmpDir: a character string specifying the directory under which\n",
       "          intermediate files are saved (later removed). By default,\n",
       "          current working directory is used.\n",
       "\n",
       " verbose: logical indicating if verbose information for debugging will\n",
       "          be generated. This may include information such as unmatched\n",
       "          chromosomes/contigs between reads and annotation.\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     featureCounts is a general-purpose read summarization function\n",
       "     that can assign mapped reads from genomic DNA and RNA sequencing\n",
       "     to genomic features or meta-features.\n",
       "\n",
       "     The function takes as input a set of SAM or BAM files containing\n",
       "     read mapping results. The files might be generated by align or\n",
       "     subjunc or any suitable aligner.\n",
       "\n",
       "     featureCounts accepts two annotation formats to specify the\n",
       "     genomic features: SAF (Simplified Annotation Format) or GTF/GFF.\n",
       "     The GTF/GFF format is specified at <URL:\n",
       "     https://genome.ucsc.edu/FAQ/FAQformat.html>. SAF is a Simplified\n",
       "     Annotation Format with five columns. It can be either a data.frame\n",
       "     or a tab-delimited text file in the following style:\n",
       "\n",
       "     GeneID          Chr     Start   End     Strand\n",
       "     497097          chr1    3204563 3207049 -\n",
       "     497097          chr1    3411783 3411982 -\n",
       "     497097          chr1    3660633 3661579 -\n",
       "     100503874       chr1    3637390 3640590 -\n",
       "     100503874       chr1    3648928 3648985 -\n",
       "     100038431       chr1    3670236 3671869 -\n",
       "     ...\n",
       "     \n",
       "     SAF annotation includes the following five required columns:\n",
       "     GeneID, Chr, Start, End and Strand. The GeneID column\n",
       "     includes identifiers of features. These identifiers can be integer\n",
       "     or character string. The Chr column includes chromosome names of\n",
       "     features and these names should match the chromosome names\n",
       "     includes in the provided SAM/BAM files. The Start and End\n",
       "     columns include start and end coordinates of features,\n",
       "     respectively. Both start and end coordinates are inclusive. The\n",
       "     Strand column indicates the strand of features (\"+\" or \"-\").\n",
       "     Column names in a SAF annotation should be the same as those shown\n",
       "     in the above example (case insensitive). Columns can be in any\n",
       "     order. Extra columns are allowed to be added into the annotation.\n",
       "\n",
       "     In-built annotations, which were generated based on NCBI RefSeq\n",
       "     gene annotations, are provided to faciliate convenient read\n",
       "     summarization. We provide in-built annotations for the following\n",
       "     genomes: \"hg38\", \"hg19\", \"mm10\" and \"mm9\". The content of\n",
       "     in-built annotations can be accessed via the\n",
       "     getInBuiltAnnotation function. These annotations have a SAF\n",
       "     format.\n",
       "\n",
       "     The in-built annotations are a modified version of NCBI RefSeq\n",
       "     gene annotations. We downloaded the RefSeq gene annotations from\n",
       "     NCBI ftp server (eg. RefSeq annotation for mm10 was downloaded\n",
       "     from <URL:\n",
       "     ftp://ftp.ncbi.nlm.nih.gov/genomes/M_musculus/ARCHIVE/BUILD.38.1/mapview/seq_gene.md.gz>).\n",
       "     We then used these annotations to create our in-built annotations.\n",
       "     For each gene, we used its CDS (coding DNA sequence) and UTR\n",
       "     (untranslated) regions provided in the original annotation to\n",
       "     construct a list of exons, by merging and concatenating all CDSs\n",
       "     and UTRs belonging to the same gene. Exons within each gene\n",
       "     include all chromosomal bases included in the original CDS and UTR\n",
       "     regions, but they include each base only once (they might be\n",
       "     included multiple times in the original CDSs and UTRs). Also,\n",
       "     exons within the same gene do not overlap with each other.\n",
       "\n",
       "     Users may provide an external annotation for read summarization\n",
       "     via the annot.ext argument. If the external annotation is in SAF\n",
       "     format, it can be provdied as either a data.frame or a\n",
       "     tab-delimited text file with proper column names included. If it\n",
       "     is in GTF/GFF format, it should be provided as a file only (and\n",
       "     isGTFAnnotationFile should be set to TRUE).\n",
       "\n",
       "     featureCounts function uses the GTF.attrType attribute in a\n",
       "     GTF/GFF annotation to group features to form meta-features when\n",
       "     performing read summarization at meta-feature level.\n",
       "\n",
       "     The argument useMetaFeatures specifies whether read\n",
       "     summarization should be performed at feature level or at\n",
       "     meta-feature level. A feature represents a continuous genomic\n",
       "     region and a meta-feature is a group of features. For instance, an\n",
       "     exon is a feature and a gene comprising one or more exons is a\n",
       "     meta-feature. To assign reads to meta-features, featureCounts\n",
       "     firstly groups into meta-features the features that have the same\n",
       "     gene identifiers. featureCounts looks for gene identifiers in\n",
       "     GeneID column of a SAF annotation or by using GTF.attrType\n",
       "     attribute in a GTF/GFF annotation. Then for each read\n",
       "     featureCounts searches for meta-features that have at least one\n",
       "     feature that overlaps with the read. A read might be found to\n",
       "     overlap with more than one feature within the same meta-feature\n",
       "     (eg. an exon-spanning read overlaps with more than one exon from\n",
       "     the same gene), however this read will still be counted only once\n",
       "     for the meta-feature.\n",
       "\n",
       "     RNA-seq reads are often summarized to meta-features to produce\n",
       "     read counts for genes. Further downstream analysis can then be\n",
       "     carried out to discover differentially expressed genes.\n",
       "     Feature-level summarization of RNA-seq data often yields\n",
       "     exon-level read counts, which is useful for investigating\n",
       "     alternative splicing of genes.\n",
       "\n",
       "     featureCounts provides multiple options to count multi-mapping\n",
       "     reads (reads mapping to more than one location in the reference\n",
       "     genome). Users can choose to ignore such reads by setting\n",
       "     countMultiMappingReads to FALSE, or fully count every\n",
       "     alignment reported for a multi-mapping read by setting\n",
       "     countMultiMappingReads to TRUE (each alignment carries 1\n",
       "     count), or count each alignment fractionally by setting both\n",
       "     countMultiMappingReads and fraction to TRUE (each alignment\n",
       "     carries 1/x count where x is the total number of alignments\n",
       "     reported for the read).\n",
       "\n",
       "     featureCounts also provides multiple options to count\n",
       "     multi-overlapping reads (reads overlapping with more than one\n",
       "     meta-feature when conducting meta-feature-level summarization or\n",
       "     overlapping with more than one feature when conducting\n",
       "     feature-level summarization). Users can choose to ignore such\n",
       "     reads by setting allowMultiOverlap to FALSE, or fully count\n",
       "     them for each overlapping meta-feature/feature by setting\n",
       "     allowMultiOverlap to TRUE (each overlapping\n",
       "     meta-feature/feature receives a count of 1 from a read), or assign\n",
       "     a fractional count to each overlapping meta-feature/feature by\n",
       "     setting both allowMultiOverlap and fraction to TRUE (each\n",
       "     overlapping meta-feature/feature receives a count of 1/y from a\n",
       "     read where y is the total number of meta-features/features\n",
       "     overlapping with the read).\n",
       "\n",
       "     If a read is both multi-mapping and multi-overlapping, then each\n",
       "     overlapping meta-feature/feature will receive a fractional count\n",
       "     of 1/(x*y) if countMultiMappingReads, allowMultiOverlap and\n",
       "     fraction are all set to TRUE.\n",
       "\n",
       "     When isPairedEnd is TRUE, fragments (pairs of reads) instead\n",
       "     of reads will be counted. featureCounts function checks if reads\n",
       "     from the same pair are adjacent to each other (this could happen\n",
       "     when reads were for example sorted by their mapping locations),\n",
       "     and it automatically reorders those reads that belong to the same\n",
       "     pair but are not adjacent to each other in the input read file.\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "     A list with the following components:\n",
       "\n",
       "  counts: a data matrix containing read counts for each feature or\n",
       "          meta-feature for each library.\n",
       "\n",
       "counts_junction (optional): a data frame including primary and\n",
       "          secondary genes overlapping an exon junction, position\n",
       "          information for the left splice site (`Site1') and the right\n",
       "          splice site (`Site2') of a junction (including chromosome\n",
       "          name, coordinate and strand) and number of supporting reads\n",
       "          for each junction in each library. Both primary and secondary\n",
       "          genes overlap at least one of the two splice sites of a\n",
       "          junction. Secondary genes do not overlap more splice sites\n",
       "          than the primary gene. When the primary and secondary genes\n",
       "          overlap same number of splice sites, the gene with the\n",
       "          smallest leftmost base position is selected as the primary\n",
       "          gene. For each junction, no more than one primary gene is\n",
       "          reported but there might be more than one secondary genes\n",
       "          reported. If a junction does not overlap any genes, it has a\n",
       "          missing value in the fields of primary gene and secondary\n",
       "          gene. Note that this data frame is only generated when\n",
       "          juncCounts is TRUE.\n",
       "\n",
       "annotation: a data frame with six columns including GeneID, Chr,\n",
       "          Start, End and Length. When read summarization was\n",
       "          performed at feature level, each row in the data frame is a\n",
       "          feature and columns in the data frame give the annotation\n",
       "          information for the features. When read summarization was\n",
       "          performed at meta-feature level, each row in the data frame\n",
       "          is a meta-feature and columns in the data frame give the\n",
       "          annotation information for the features included in each meta\n",
       "          feature except the Length column. For each meta-feature,\n",
       "          the Length column gives the total length of genomic regions\n",
       "          covered by features included in that meta-feature.  Note that\n",
       "          this length will be less than the sum of lengths of features\n",
       "          included in the meta-feature when there are features\n",
       "          overlapping with each other. Also note the GeneID column\n",
       "          gives Entrez gene identifiers when the in-built annotations\n",
       "          are used.\n",
       "\n",
       " targets: a character vector giving sample information.\n",
       "\n",
       "    stat: a data frame giving number of successfully assigned reads in\n",
       "          each library and also number of unassigned reads falling in\n",
       "          each filtering category (unmapped, read type, singleton,\n",
       "          mapping quality, chimera, fragment length, duplicate,\n",
       "          multi-mapping, secondary alignment, split alignment, no\n",
       "          features, overlapping length and assignment ambiguity). See\n",
       "          Users Guide for more details.\n",
       "\n",
       "_\bA_\bu_\bt_\bh_\bo_\br(_\bs):\n",
       "\n",
       "     Wei Shi and Yang Liao\n",
       "\n",
       "_\bR_\be_\bf_\be_\br_\be_\bn_\bc_\be_\bs:\n",
       "\n",
       "     Yang Liao, Gordon K Smyth and Wei Shi (2019). The R package\n",
       "     Rsubread is easier, faster, cheaper and better for alignment and\n",
       "     quantification of RNA sequencing reads. _Nucleic Acids Research_,\n",
       "     47(8):e47. <URL: http://www.ncbi.nlm.nih.gov/pubmed/30783653>\n",
       "\n",
       "     Yang Liao, Gordon K Smyth and Wei Shi (2014). featureCounts: an\n",
       "     efficient general-purpose program for assigning sequence reads to\n",
       "     genomic features. _Bioinformatics_, 30(7):923-30. <URL:\n",
       "     http://www.ncbi.nlm.nih.gov/pubmed/24227677>\n",
       "\n",
       "_\bS_\be_\be _\bA_\bl_\bs_\bo:\n",
       "\n",
       "     getInBuiltAnnotation\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     ## Not run:\n",
       "     \n",
       "     # Summarize SAM format single-end reads using built-in RefSeq annotation for mouse genome mm10:\n",
       "     featureCounts(files=\"mapping_results_SE.sam\",annot.inbuilt=\"mm10\")\n",
       "     \n",
       "     # Summarize single-end reads using a user-provided GTF annotation file:\n",
       "     featureCounts(files=\"mapping_results_SE.sam\",annot.ext=\"annotation.gtf\",\n",
       "     isGTFAnnotationFile=TRUE,GTF.featureType=\"exon\",GTF.attrType=\"gene_id\")\n",
       "     \n",
       "     # Summarize single-end reads using 5 threads:\n",
       "     featureCounts(files=\"mapping_results_SE.sam\",nthreads=5)\n",
       "     \n",
       "     # Summarize BAM format single-end read data:\n",
       "     featureCounts(files=\"mapping_results_SE.bam\")\n",
       "     \n",
       "     # Perform strand-specific read counting (strandSpecific=2 if reversely stranded):\n",
       "     featureCounts(files=\"mapping_results_SE.bam\",strandSpecific=1)\n",
       "     \n",
       "     # Summarize paired-end reads and counting fragments (instead of reads):\n",
       "     featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE)\n",
       "     \n",
       "     # Count fragments satisfying the fragment length criteria, eg. [50bp, 600bp]:\n",
       "     featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,\n",
       "     checkFragLength=TRUE,minFragLength=50,maxFragLength=600)\n",
       "     \n",
       "     # Count fragments that have both ends successfully aligned without checking the fragment length:\n",
       "     featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,requireBothEndsMapped=TRUE)\n",
       "     \n",
       "     # Exclude chimeric fragments from fragment counting:\n",
       "     featureCounts(files=\"mapping_results_PE.bam\",isPairedEnd=TRUE,countChimericFragments=FALSE)\n",
       "     ## End(Not run)\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(Rsubread)\n",
    "?featureCounts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:DT_DPLYR]",
   "language": "R",
   "name": "conda-env-DT_DPLYR-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
