{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: ‘plotgardener’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    c\n",
      "\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Loading required package: GenomicFeatures\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(\"plotgardener\")\n",
    "library(\"data.table\")\n",
    "library(\"org.Hs.eg.db\")\n",
    "library(\"TxDb.Hsapiens.UCSC.hg38.knownGene\")\n",
    "library(png)\n",
    "options(bedtools.path = \"~/anaconda3/bin/\")\n",
    "library(bedtoolsr)\n",
    "\n",
    "## In this notebook, code for making genomic regions plots showing CHiC interactions and ATAC-seq signal tracks \n",
    "## Mostly using the PlotGardener package: https://phanstiellab.github.io/plotgardener/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "loops <- fread(\"~/HRJ_monocytes/CHiC/chicago/combinations_output/DpnII/combined_34reps_DpnII_binsize1500_maxL150K_mergeWeights/data/combined_34reps_DpnII_binsize1500_maxL150K_mergeWeights.bedpe\")\n",
    "# intersect loops with proxies so we can restrict loops to just incldue them\n",
    "\n",
    "proxies1 <- fread(\"~/HRJ_monocytes/findmotifs/find_new_dpnII/snps_went_into_loopingQTL_analysis.DpnII.marked.REFALTseqs.PURGED.DISTAL.txt\")\n",
    "proxies1[, chrom := paste0(\"chr\", Chr)]\n",
    "proxies1[, hg38Proxy_pos0 := hg38Proxy_pos-1]\n",
    "proxies_small <- proxies1[, .(chrom, hg38Proxy_pos0, hg38Proxy_pos, hg38Proxy_ID, hg38SNP_ID)]\n",
    "setkey(loops, V1, V2, V3)\n",
    "loops1 <- foverlaps(proxies_small, loops, by.x = c(\"chrom\", \"hg38Proxy_pos0\", \"hg38Proxy_pos\"), nomatch = NULL)\n",
    "setnames(loops1, c(\"chrom\", \"V2\", \"V3\", \"V4\", \"V5\", \"V6\"), c(\"chrom1\", \"start1\", \"end1\", \"chrom2\", \"start2\", \"end2\"))\n",
    "setkey(loops, V4, V5, V6)\n",
    "loops2 <- foverlaps(proxies_small, loops, by.x = c(\"chrom\", \"hg38Proxy_pos0\", \"hg38Proxy_pos\"), nomatch = NULL)\n",
    "setnames(loops2, c(\"chrom\", \"V1\", \"V2\", \"V3\", \"V5\", \"V6\"), c(\"chrom2\", \"chrom1\", \"start1\", \"end1\", \"start2\", \"end2\"))\n",
    "loops2_edit <- loops2[, .(chrom1, start1, end1, chrom2, start2, end2, hg38Proxy_pos0, hg38Proxy_pos, hg38Proxy_ID, hg38SNP_ID)]\n",
    "loops_proxies <- rbind(loops1, loops2_edit)\n",
    "\n",
    "## OR: we can instead look at eQTLs and only their interactions with eGenes\n",
    "eqtl_inters <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/consensus_DpnII_level_proxy_eGene_interactions_DpnIIFilt_withGeneTypes.txt\")\n",
    "sig_eqtl_inters <- eqtl_inters[score >= 5]\n",
    "sig_eqtl_inters <- sig_eqtl_inters[, c(\"Chr\", \"Proxy_dpnStart\", \"Proxy_dpnEnd\", \"DpnID\", \"Gene\", \"hg19SNP_ID\", \"hg19Proxy_ID\", \"hg38SNP_ID\")]\n",
    "# have to get other end coordinates\n",
    "dpnII <- fread(\"~/spivakov/live/Design/Human_eQTL_CHiC_DpnII_hg38/hg38_dpnII.rmap\")\n",
    "setkey(dpnII, V4)\n",
    "sig_eqtl_inters_dpn <- sig_eqtl_inters[dpnII, on = c(DpnID = \"V4\"), nomatch = NULL]\n",
    "setnames(sig_eqtl_inters_dpn, c(\"V2\", \"V3\"), c(\"TSS_dpnStart\", \"TSS_dpnEnd\"))\n",
    "sig_eqtl_inters_dpn[, chrom1 := paste0(\"chr\", Chr)]\n",
    "sig_eqtl_inters_dpn[, chrom2 := paste0(\"chr\", V1)]\n",
    "eQTL_eGene_score5 <- sig_eqtl_inters_dpn[, .(chrom1, Proxy_dpnStart, Proxy_dpnEnd, chrom2, TSS_dpnStart, TSS_dpnEnd, Gene, hg19SNP_ID, hg19Proxy_ID, hg38SNP_ID)]\n",
    "\n",
    "\n",
    "# need to make proxy_sets only the ones that we keep!\n",
    "proxy_sets <- fread(\"~/HRJ_monocytes/Browser_tracks/proxy_setshg38.bed\")\n",
    "eqtls <- fread(\"~/HRJ_monocytes/Browser_tracks/eqtlshg38.bed\")\n",
    "#proxies <- fread(\"~/HRJ_monocytes/Browser_tracks/proxieshg38.bed\")\n",
    "proxies <- fread(\"~/HRJ_monocytes/findmotifs/find_new_dpnII/snps_went_into_loopingQTL_analysis.DpnII.marked.REFALTseqs.PURGED.DISTAL.txt\")\n",
    "proxies[, chrom := paste0(\"chr\", Chr)]\n",
    "proxies_plot <- proxies[, .(chrom, hg38Proxy_pos, hg38Proxy_end, hg38SNP_ID)]\n",
    "\n",
    "# Load in DpnII baits\n",
    "baits <- fread(\"~/spivakov/live/Design/Human_eQTL_CHiC_DpnII_hg38/eCHiC_grch38.baitmap\")\n",
    "baits[, chrom := paste0(\"chr\", V1)]\n",
    "baits <- baits[, .(chrom, V2, V3, V4, V5)]\n",
    "names(baits) = c(\"chrom\", \"DpnStart\", \"DpnEnd\", \"DpnID\", \"DpnName\")\n",
    "\n",
    "# also make an overlap with genes\n",
    "proxies1[, TSS1 := TSS - 500]\n",
    "proxies1[, TSS2 := TSS + 500]\n",
    "genes_small <- proxies1[, .(chrom, TSS1, TSS2, rep_Gene, Gene, ENSG_ID)]\n",
    "setkey(loops, V1, V2, V3)\n",
    "loopsG1 <- foverlaps(genes_small, loops, by.x = c(\"chrom\", \"TSS1\", \"TSS2\"), nomatch = NULL)\n",
    "setnames(loopsG1, c(\"chrom\", \"V2\", \"V3\", \"V4\", \"V5\", \"V6\"), c(\"chrom1\", \"start1\", \"end1\", \"chrom2\", \"start2\", \"end2\"))\n",
    "setkey(loops, V4, V5, V6)\n",
    "loopsG2 <- foverlaps(genes_small, loops, by.x = c(\"chrom\", \"TSS1\", \"TSS2\"), nomatch = NULL)\n",
    "setnames(loopsG2, c(\"chrom\", \"V1\", \"V2\", \"V3\", \"V5\", \"V6\"), c(\"chrom2\", \"chrom1\", \"start1\", \"end1\", \"start2\", \"end2\"))\n",
    "loopsG2_edit <- loopsG2[, .(chrom1, start1, end1, chrom2, start2, end2, TSS1, TSS2, rep_Gene, Gene, ENSG_ID)]\n",
    "loops_genes <- rbind(loopsG1, loopsG2_edit)\n",
    "\n",
    "## bring in the regions that were tested in BaseQTL. for each proxy SNP, we can filter this file and merge the gene end to make one PE contact.\n",
    "regions <- fread(\"~/HRJ_monocytes/AS_CHiC/BaseQTL/scripts/bedpe_for_chic_egenes_with_Gene_number.txt\")\n",
    "regions[, chrom1 := paste0(\"chr\", V1)]\n",
    "regions[, chrom2 := paste0(\"chr\", V4)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "### Read in the BaseQTL results, in order to plot the contact QTLs\n",
    "\n",
    "BaseRes <- fread(\"~/HRJ_monocytes/BaseQTL/findings_round2_dpnIICorrection/BaseQTL_CHiC_Filt_on_Dpn_Distal_Rhat_AI_rsids.txt\")\n",
    "BaseBed <- BaseRes[Signif.99 == \"yes\", .(chrom, hg38Proxy_pos, hg38Proxy_end, Gene, hg19SNP_ID, BaseQTL_SNP_hg19, rsid, log_mean_aFC)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "### Bring in information on ATAC, H3K27ac and H3K4me1\n",
    "# ATAC, prior to filtering for nearest ATAC peak within 5kb\n",
    "atac <- fread(\"~/HRJ_monocytes/BaseQTL/findings_round2_dpnIICorrection/AT_CHiC_with_LD_SNPs_and_MikBetas_Signif99.txt\")\n",
    "# also for 27ac and me1\n",
    "histones <- fread(\"~/HRJ_monocytes/BaseQTL/findings_round2_dpnIICorrection/blueprint_table_27ac_me1_sig99_LD.txt\")\n",
    "\n",
    "# make bed files for the plots. keep the name of the baseQTL ID in each case and whether the direction goes up or down.\n",
    "atac[, Chr := tstrsplit(BaseQTL_SNP_hg19, split = \":\", keep = 1)]\n",
    "atac[, chrom := paste0(\"chr\", Chr)]\n",
    "atac_plot <- unique(atac[!is.na(at_featStart), .(chrom, at_featStart, at_featEnd, BaseQTL_SNP_hg19, chic_log_mean_aFC, at_log_mean_aFC)])\n",
    "\n",
    "histones[, c(\"h3k27ac_start\", \"h3k27ac_end\") := tstrsplit(h3k27ac_phenotypeID, split = \":\", keep = c(2,3), type.convert = T)]\n",
    "histones[, c(\"h3k4me1_start\", \"h3k4me1_end\") := tstrsplit(h3k4me1_phenotypeID, split = \":\", keep = c(2,3), type.convert = T)]\n",
    "histones[, chrom := paste0(\"chr\", Chr)]\n",
    "\n",
    "h3k27ac <- unique(histones[!is.na(h3k27ac_phenotypeID), .(chrom, h3k27ac_start, h3k27ac_end, BaseQTL_hg19Proxy_ID, chic_log_mean_aFC, h3k27ac_beta)])\n",
    "h3k4me1 <- unique(histones[!is.na(h3k4me1_phenotypeID), .(chrom, h3k4me1_start, h3k4me1_end, BaseQTL_hg19Proxy_ID, chic_log_mean_aFC, h3k4me1_beta)])\n",
    "\n",
    "#h3k27ac\n",
    "#h3k4me1\n",
    "\n",
    "## will have to take note of the beta direction and then make unique each time"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# below: THBS1 locus showing alleleic fold change\n",
    "# Now trying with the consensus ATAC-seq data: pileups from 34 reps in Genrich."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "arches[arches3]\n",
      "\n",
      "arches[arches4]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## THBS1 locus showing alleleic fold change\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/thbs1_plotGardener_aFC_arches_V02.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"15:39312634:C:T\"\n",
    "BaseSNP <- \"15:39606026:C:T\"\n",
    "log2aFC <- 0.6\n",
    "myGene = \"THBS1\"\n",
    "mychrom = \"chr15\"\n",
    "mystart = 39286285\n",
    "myend = 39615829\n",
    "###\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "#loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "# look at all loops!\n",
    "loops_eqtl <- copy(loops)\n",
    "\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "\n",
    "# merge regions if right nect to each other! \n",
    "regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "regions_keep_LHS_merge <- regions_keep_LHS[1:2, ]\n",
    "#LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1)\n",
    "RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1)\n",
    "setnames(RHS_merge, \"V1\", \"chrom2\")\n",
    "both_regions <- cbind(regions_keep_LHS_merge, RHS_merge)\n",
    "#both_regions\n",
    "\n",
    "###\n",
    "aFC <- round(2^log2aFC, digits = 2)\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 10, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.3, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 2.8, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.7)\n",
    "\n",
    "\n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/THBS1_DpnII_2070430.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(\n",
    "    data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 5.5,\n",
    "    linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 5.5, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "## text for loops\n",
    "plotText(label = \"THBS1 CHi-C \\nsignal, N reads\", fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 5.7, height = 0.5\n",
    ")\n",
    "\n",
    "###\n",
    "\n",
    "### plot pairs and make the height equal to the score\n",
    "## REF\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 3.7, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T, style = \"3D\")\n",
    "}\n",
    "\n",
    "## ALT\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 3.7, height = (1*aFC), archHeight = (1*aFC), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T, style = \"3D\")\n",
    "}\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 2.5, y = 4, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: ~\", aFC), x = 2.5, y = 4.6, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"Allelic \\nfold change\", x = 1.3, y = 4, height = 0.3, fontcolor = \"gray40\", fontsize = 7, rot = 0, params=params_text)\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 1.2, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead eQTL\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 1.2, height = 0.5,\n",
    ")\n",
    "\n",
    "### Plot the contact QTLs\n",
    "contact <- BaseBed[Gene == myGene]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTLs\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 1.8, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD variants\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.3, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "#plotRanges( \n",
    "#    data = baits[DpnName %like% myeQTL], params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.7, alpha = 0.4\n",
    "#)\n",
    "\n",
    "\n",
    "## plot ATAC - now using the proper 34 reps consensus data\n",
    "ATACPval <- paste0(\"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus_34reps/consensus_34reps_Pval_\", mychrom, \".bw\")\n",
    "PvalRegion <- readBigwig(file = ATACPval,\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATACPvalPlot <- plotSignal(\n",
    "    data = PvalRegion,\n",
    "    params = params_main, height = 1, y = 0,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.2, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATACPvalPlot,\n",
    "    at = c(0,68), label = T, # top pileup value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Below: THBS1, showing local H3K27ac and the ATAC peak that is affected. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 3 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>h3k4me1_start</th><th scope=col>h3k4me1_end</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr15</td><td>39601134</td><td>39611421</td></tr>\n",
       "\t<tr><td>chr15</td><td>39614513</td><td>39633565</td></tr>\n",
       "\t<tr><td>chr15</td><td>39648224</td><td>39652822</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 3 × 3\n",
       "\\begin{tabular}{lll}\n",
       " chrom & h3k4me1\\_start & h3k4me1\\_end\\\\\n",
       " <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr15 & 39601134 & 39611421\\\\\n",
       "\t chr15 & 39614513 & 39633565\\\\\n",
       "\t chr15 & 39648224 & 39652822\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 3 × 3\n",
       "\n",
       "| chrom &lt;chr&gt; | h3k4me1_start &lt;int&gt; | h3k4me1_end &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| chr15 | 39601134 | 39611421 |\n",
       "| chr15 | 39614513 | 39633565 |\n",
       "| chr15 | 39648224 | 39652822 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom h3k4me1_start h3k4me1_end\n",
       "1 chr15 39601134      39611421   \n",
       "2 chr15 39614513      39633565   \n",
       "3 chr15 39648224      39652822   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 5 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>h3k4me1_start</th><th scope=col>h3k4me1_end</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr15</td><td>39463593</td><td>39466771</td></tr>\n",
       "\t<tr><td>chr15</td><td>39472057</td><td>39477427</td></tr>\n",
       "\t<tr><td>chr15</td><td>39539690</td><td>39547161</td></tr>\n",
       "\t<tr><td>chr15</td><td>39552305</td><td>39563308</td></tr>\n",
       "\t<tr><td>chr15</td><td>39486078</td><td>39490241</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 5 × 3\n",
       "\\begin{tabular}{lll}\n",
       " chrom & h3k4me1\\_start & h3k4me1\\_end\\\\\n",
       " <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr15 & 39463593 & 39466771\\\\\n",
       "\t chr15 & 39472057 & 39477427\\\\\n",
       "\t chr15 & 39539690 & 39547161\\\\\n",
       "\t chr15 & 39552305 & 39563308\\\\\n",
       "\t chr15 & 39486078 & 39490241\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 5 × 3\n",
       "\n",
       "| chrom &lt;chr&gt; | h3k4me1_start &lt;int&gt; | h3k4me1_end &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| chr15 | 39463593 | 39466771 |\n",
       "| chr15 | 39472057 | 39477427 |\n",
       "| chr15 | 39539690 | 39547161 |\n",
       "| chr15 | 39552305 | 39563308 |\n",
       "| chr15 | 39486078 | 39490241 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom h3k4me1_start h3k4me1_end\n",
       "1 chr15 39463593      39466771   \n",
       "2 chr15 39472057      39477427   \n",
       "3 chr15 39539690      39547161   \n",
       "4 chr15 39552305      39563308   \n",
       "5 chr15 39486078      39490241   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "arches[arches3]\n",
      "\n",
      "arches[arches4]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges4]\n",
      "\n",
      "text[text2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "29551183"
      ],
      "text/latex": [
       "29551183"
      ],
      "text/markdown": [
       "29551183"
      ],
      "text/plain": [
       "[1] 29551183"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "21663557"
      ],
      "text/latex": [
       "21663557"
      ],
      "text/markdown": [
       "21663557"
      ],
      "text/plain": [
       "[1] 21663557"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.733085947862053"
      ],
      "text/latex": [
       "0.733085947862053"
      ],
      "text/markdown": [
       "0.733085947862053"
      ],
      "text/plain": [
       "[1] 0.7330859"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "signal[signal1]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "segments[segments1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# THBS1, showing local histone and ATAC peaks that are affected. \n",
    "pdf(file = \"~/HRJ_monocytes/paper/thbs1_plotGardener_crossPhenotypes_V04.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"15:39312634:C:T\"\n",
    "BaseSNP1 <- \"15:39598843:A:G\"\n",
    "BaseSNP2 <- \"15:39602380:A:G\"\n",
    "BaseSNP3 <- \"15:39606026:C:T\"\n",
    "BaseSNP4 <- \"15:39606089:A:T\"\n",
    "BaseSNP5 <- \"15:39607442:A:G\"\n",
    "BaseSNP6 <- \"15:39607630:A:G\"\n",
    "BaseSNP7 <- \"15:39607665:A:G\"\n",
    "log2aFC <- 0.6\n",
    "myGene = \"THBS1\"\n",
    "mychrom = \"chr15\"\n",
    "mystart = 39286285\n",
    "myend = 39615829\n",
    "###\n",
    "\n",
    "\n",
    "### prepare ATAC peaks: now trying with pileups\n",
    "myATAC <- unique(atac_plot[BaseQTL_SNP_hg19 == BaseSNP1 | BaseQTL_SNP_hg19 == BaseSNP2 |\n",
    "                           BaseQTL_SNP_hg19 == BaseSNP3 | BaseQTL_SNP_hg19 == BaseSNP4 |\n",
    "                           BaseQTL_SNP_hg19 == BaseSNP5 | BaseQTL_SNP_hg19 == BaseSNP6 |\n",
    "                           BaseQTL_SNP_hg19 == BaseSNP7])\n",
    "#myATAC[at_log_mean_aFC < 0] # none\n",
    "atacUP <- unique(myATAC[, .(chrom, at_featStart, at_featEnd)])\n",
    "\n",
    "### prepare h3k27ac peaks\n",
    "myH3K27 <- unique(h3k27ac[BaseQTL_hg19Proxy_ID == BaseSNP1 | BaseQTL_hg19Proxy_ID == BaseSNP2 |\n",
    "                           BaseQTL_hg19Proxy_ID == BaseSNP3 | BaseQTL_hg19Proxy_ID == BaseSNP4 |\n",
    "                           BaseQTL_hg19Proxy_ID == BaseSNP5 | BaseQTL_hg19Proxy_ID == BaseSNP6 |\n",
    "                           BaseQTL_hg19Proxy_ID == BaseSNP7])\n",
    "#myH3K27[h3k27ac_beta < 0] # none\n",
    "h3k27acUPhg19 <- unique(myH3K27[, .(chrom, h3k27ac_start, h3k27ac_end)])\n",
    "# convert to hg38\n",
    "h3k27acUP <- data.table(chrom = \"chr15\", h3k27ac_start = 39312773, \n",
    "                        h3k27ac_end = 39314688)\n",
    "\n",
    "### prepare h3k4me1 peaks\n",
    "myH3K4 <- unique(h3k4me1[BaseQTL_hg19Proxy_ID == BaseSNP1 | BaseQTL_hg19Proxy_ID == BaseSNP2 |\n",
    "                           BaseQTL_hg19Proxy_ID == BaseSNP3 | BaseQTL_hg19Proxy_ID == BaseSNP4 |\n",
    "                           BaseQTL_hg19Proxy_ID == BaseSNP5 | BaseQTL_hg19Proxy_ID == BaseSNP6 |\n",
    "                           BaseQTL_hg19Proxy_ID == BaseSNP7])\n",
    "#myH3K4[h3k4me1_beta < 0] \n",
    "h3k4me1UPhg19 <- unique(myH3K4[h3k4me1_beta > 0, .(chrom, h3k4me1_start, h3k4me1_end)])\n",
    "h3k4me1DOhg19 <- unique(myH3K4[h3k4me1_beta < 0, .(chrom, h3k4me1_start, h3k4me1_end)])\n",
    "\n",
    "# convert to hg38\n",
    "h3k4me1UPhg19\n",
    "h3k4me1DOhg19\n",
    "\n",
    "h3k4me1UP <- data.table(chrom = c(\"chr15\", \"chr15\", \"chr15\"), start = c(39308933, 39322312, 39356023), \n",
    "                        end = c(39319220, 39341364, 39360621)) \n",
    "h3k4me1DO <- data.table(chrom = c(\"chr15\", \"chr15\", \"chr15\", \"chr15\", \"chr15\"), \n",
    "                        start = c(39171392, 39179856, 39247489, 39260104, 39193877), \n",
    "                        end = c(39174570, 39185226, 39254960, 39271107, 39198040))\n",
    "\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "#loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "# look at all loops!\n",
    "loops_eqtl <- copy(loops)\n",
    "\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene & V7 %like% BaseSNP3] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "\n",
    "# merge regions if right nect to each other! \n",
    "regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "regions_keep_LHS_merge <- regions_keep_LHS[1:2, ]\n",
    "#LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1)\n",
    "RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1)\n",
    "setnames(RHS_merge, \"V1\", \"chrom2\")\n",
    "both_regions <- cbind(regions_keep_LHS_merge, RHS_merge)\n",
    "#both_regions\n",
    "\n",
    "###\n",
    "aFC <- round(2^log2aFC, digits = 2)\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 11, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 3.8, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 4, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.8)\n",
    "\n",
    "### plot affected atac peak(s)\n",
    "plotRanges( \n",
    "    data = atacUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 2.6, height = 0.3\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Affected ATAC\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 2.7, height = 0.5,\n",
    ")\n",
    "\n",
    "### plot affected h3k27ac peaks\n",
    "plotRanges( \n",
    "    data = h3k27acUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3, height = 0.3\n",
    ")\n",
    "# text for H3K27ac\n",
    "plotText(\n",
    "    label = \"Affected H3K27ac\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.1, height = 0.5,\n",
    ")\n",
    "# h3k4me1 can go at 2.2\n",
    "plotRanges( \n",
    "    data = h3k4me1UP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3.4, height = 0.3\n",
    ")\n",
    "# text \n",
    "plotText(\n",
    "    label = \"Affected H3K4me1\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.5, height = 0.5,\n",
    ")\n",
    "# There is no h3k4me1 down\n",
    "#plotRanges( \n",
    "#    data = h3k4me1DO, params = params_main, linecolor=\"fill\", fill = \"blue\", collapse = T,\n",
    "#    y = 3.2, height = 0.3\n",
    "#)\n",
    "# text\n",
    "#plotText(\n",
    "#    label = \"Affected H3K4me1 DOWN\", \n",
    "#    fontsize = 7, fontcolor = \"blue\", rot = 0, params = params_text, y = 3.2, height = 0.5,\n",
    "#)\n",
    "\n",
    "\n",
    "###\n",
    "\n",
    "### plot pairs and make the height equal to the score\n",
    "## REF\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T, style = \"3D\")\n",
    "}\n",
    "\n",
    "## ALT\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5, height = (1*aFC), archHeight = (1*aFC), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T, style = \"3D\")\n",
    "}\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 2.5, y = 5.3, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: ~\", aFC), x = 2.5, y = 6, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"CHi-C allelic \\nfold change\", x = 1.3, y = 5.3, height = 0.3, fontcolor = \"gray40\", fontsize = 7, rot = 0, params=params_text)\n",
    "\n",
    "\n",
    "### Plot the contact QTLs\n",
    "contact <- BaseBed[Gene == myGene]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTLs\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.2, height = 0.5,\n",
    ")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# The below pileups look good.\n",
    "\n",
    "ATAC_REF_Pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/THBS1/REF_Genrich_merged_THBS1_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "ATAC_ALT_Pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/THBS1/ALT_Genrich_merged_THBS1_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "# scale ALT and REF.\n",
    "totalREF <- sum(ATAC_REF_Pileup$score)\n",
    "totalALT <- sum(ATAC_ALT_Pileup$score)\n",
    "totalREF\n",
    "totalALT\n",
    "ratio <- totalALT/totalREF\n",
    "ratio\n",
    "ATAC_REF_Pileup$score <- ATAC_REF_Pileup$score*ratio\n",
    "\n",
    "top_alt <- max(ATAC_ALT_Pileup$score)\n",
    "# then height of REF = max(ALT)\n",
    "# and scale accordingly\n",
    "ATAC_REF_Pileupplot <- plotSignal(\n",
    "    data = ATAC_REF_Pileup,\n",
    "    params = params_main, height = 1, y = 0,\n",
    "    linecolor = \"#0072B2\", range=c(0,2000)\n",
    ")\n",
    "\n",
    "ATAC_ALT_Pileupplot <- plotSignal(\n",
    "    data = ATAC_ALT_Pileup,\n",
    "    params = params_main, height = 1, y = 1,\n",
    "    linecolor = \"#b2182b\", range=c(0,2000)\n",
    ")\n",
    "# plot points at 1000 intervals\n",
    "plotSegments(x0=c(2,2), y0=c(0.5, 1.5), x1=12, y1=c(0.5, 1.5), \n",
    "             params = params_main, linecolor = c(\"gray80\", \"gray80\")\n",
    ")\n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, REF\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#0072B2\",\n",
    "    rot = 0, params = params_text, y = 0.4, height = 0.5\n",
    ") \n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, ALT\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#b2182b\",\n",
    "    rot = 0, params = params_text, y = 1.4, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_REF_Pileupplot,\n",
    "    at = NULL, label = T, # top pileup value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "annoYaxis(\n",
    "    ATAC_ALT_Pileupplot,\n",
    "    at = NULL, label = T, # top pileup value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>at_featStart</th><th scope=col>at_featEnd</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr15</td><td>39312360</td><td>39318460</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " chrom & at\\_featStart & at\\_featEnd\\\\\n",
       " <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr15 & 39312360 & 39318460\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 3\n",
       "\n",
       "| chrom &lt;chr&gt; | at_featStart &lt;int&gt; | at_featEnd &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| chr15 | 39312360 | 39318460 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom at_featStart at_featEnd\n",
       "1 chr15 39312360     39318460  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr15</td><td>68.09358</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr15 & 68.09358\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr15 | 68.09358 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr15    68.09358"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr15</td><td>149.4578</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr15 & 149.4578\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr15 | 149.4578 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr15    149.4578"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2.19488899298987"
      ],
      "text/latex": [
       "2.19488899298987"
      ],
      "text/markdown": [
       "2.19488899298987"
      ],
      "text/plain": [
       "[1] 2.194889"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### chekc the change in ATAC signal.\n",
    "atacUP\n",
    "ATAC_REF_Pileup_dt <- as.data.table(ATAC_REF_Pileup)\n",
    "region_ref <- ATAC_REF_Pileup_dt[start >= 39312360 & end <= 39318460, c(\"seqnames\", \"score\")]\n",
    "ref <- region_ref[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "ref # 68.09358\n",
    "\n",
    "ATAC_ALT_Pileup_dt <- as.data.table(ATAC_ALT_Pileup)\n",
    "region_alt <- ATAC_ALT_Pileup_dt[start >= 39312360 & end <= 39318460, c(\"seqnames\", \"score\")]\n",
    "alt <- region_alt[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "alt # 149.4578\n",
    "\n",
    "alt$score/ref$score\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>at_featStart</th><th scope=col>at_featEnd</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr15</td><td>39312360</td><td>39318460</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " chrom & at\\_featStart & at\\_featEnd\\\\\n",
       " <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr15 & 39312360 & 39318460\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 3\n",
       "\n",
       "| chrom &lt;chr&gt; | at_featStart &lt;int&gt; | at_featEnd &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| chr15 | 39312360 | 39318460 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom at_featStart at_featEnd\n",
       "1 chr15 39312360     39318460  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## plot ATAC: Pileups\n",
    "\n",
    "ATACPileup <- paste0(\"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus_34reps/consensus_34reps_Pileup_\", mychrom, \".bw\")\n",
    "PileupRegion <- readBigwig(file = ATACPileup,\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATACPileupPlot <- plotSignal(\n",
    "    data = PileupRegion,\n",
    "    params = params_main, height = 1, y = 0,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(Read pileup)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.2, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATACPileupPlot,\n",
    "    at = NULL, label = T, # top pileup value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "# Height of this difRegion should be around 0.4\n",
    "difRegion <- readBigwig(file = ATACPileup,\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = atacUP$at_featStart,\n",
    "                        chromend = atacUP$at_featEnd)\n",
    "\n",
    "ATACdifRegionALT <- plotSignal(\n",
    "    data = difRegion,\n",
    "    params = params_main, height = 0.4, y = 0.6,\n",
    "    linecolor = \"#b2182b\", \n",
    "    fill = \"#b2182b\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches3]\n",
      "\n",
      "arches[arches4]\n",
      "\n",
      "yaxis[yaxis3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis4]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## TFPT and OSCAR locus showing alleleic fold change, flipped arches\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/TFPT_plotGardener_aFC_arches_flipped.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"19:54045772:T:C\"\n",
    "BaseSNP <- \"19:54551115:G:T\"\n",
    "log2aFC1 <- -0.6040825\n",
    "log2aFC2 <- -0.3432409\n",
    "myGene1 = \"TFPT\"\n",
    "myGene2 = \"OSCAR\"\n",
    "mychrom = \"chr19\"\n",
    "mystart = 54036654\n",
    "myend = 54131092 \n",
    "\n",
    "###\n",
    "\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene1 & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all1 <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "# using regions_keep_all; only one promoter tested\n",
    "regions_keep <- regions[V7 %like% myGene2 & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all2 <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "# merge regions if right nect to each other! \n",
    "#regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "#regions_keep_LHS_merge <- regions_keep_LHS[1:2, ]\n",
    "#LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1)\n",
    "#RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1)\n",
    "#setnames(RHS_merge, \"V1\", \"chrom2\")\n",
    "#both_regions <- cbind(regions_keep_LHS_merge, RHS_merge)\n",
    "#both_regions\n",
    "\n",
    "###\n",
    "aFC1 <- round(2^log2aFC1, digits = 2)\n",
    "aFC2 <- round(2^log2aFC2, digits = 2)\n",
    "\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 10, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.3, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = c(myGene1, myGene2), \"color\" = \"chartreuse4\")\n",
    "\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 4, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 4.3, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 7.3)\n",
    "\n",
    "\n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/TFPT_DpnII_3013211.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(\n",
    "    data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 6.2,\n",
    "    linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 6.2, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "### text for loops\n",
    "plotText(label = \"TFPT CHi-C \\nsignal, N reads\", fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 6.3, height = 0.5\n",
    ")\n",
    "\n",
    "###\n",
    "\n",
    "########### TFPT\n",
    "### plot pairs and make the height equal to the score\n",
    "\n",
    "## REF\n",
    "for(n in 1:nrow(regions_keep_all1)) {\n",
    "    myplot <- plotPairsArches(data = regions_keep_all1[n, ], params = params_main, y = 0.8, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = F, style = \"3D\", \n",
    "                             baseline = T)\n",
    "}\n",
    "## ALT\n",
    "for(n in 1:nrow(regions_keep_all1)) {\n",
    "    plotPairsArches(data = regions_keep_all1[n, ], params = params_main, y = 1.8-aFC1, height = (1*aFC1), archHeight = (1*aFC), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = F, style = \"3D\", \n",
    "                   baseline = T)\n",
    "}\n",
    "# range for REF - test\n",
    "annoYaxis(\n",
    "    myplot,\n",
    "    at = NULL, label = T, \n",
    "    main = TRUE, axisLine = T, params = params_main, y = 3.7, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 3.1, y = 0.8, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: \", aFC1), x = 3.1, y = 1.2, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"Allelic fold \\nchange (CI 0.99)\", x = 1.3, y = 0.9, height = 0.3, fontsize = 7, rot = 0, params=params_text)\n",
    "#plotSegments(x0 = 2, x1 = 12, y0 = 4.7, y1 = 4.7, linecolor = \"grey\", lwd = 0.5, params = params_main)\n",
    "\n",
    "\n",
    "######### OSCAR\n",
    "for(n in 1:nrow(regions_keep_all2)) {\n",
    "    myplot2 <- plotPairsArches(data = regions_keep_all2[n, ], params = params_main, y = 2, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = F, style = \"3D\", alpha = 0.15, \n",
    "                              baseline = T)\n",
    "}\n",
    "## ALT\n",
    "for(n in 1:nrow(regions_keep_all2)) {\n",
    "    plotPairsArches(data = regions_keep_all2[n, ], params = params_main, y = 3-aFC1, height = (1*aFC1), archHeight = (1*aFC1), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = F, style = \"3D\", alpha = 0.15, \n",
    "                   baseline = T)\n",
    "}\n",
    "## REF\n",
    "# range for REF - test\n",
    "annoYaxis(\n",
    "    myplot2,\n",
    "    at = NULL, label = T, \n",
    "    main = TRUE, axisLine = T, params = params_main, y = 5, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "#plotSegments(x0 = 2, x1 = 12, y0 = 6, y1 = 6, linecolor = \"grey\", lwd = 0.5, params = params_main)\n",
    "\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 3, y = 2.2, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: \", aFC2), x = 3.1, y = 2.6, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"Allelic fold \\nchange (CI 0.90)\", x = 1.3, y = 2.2, height = 0.3, fontsize = 7, rot = 0, params=params_text)\n",
    "\n",
    "\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3.2, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead eQTL\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.2, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 3.7, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD SNPs\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 3.7, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "#plotRanges( \n",
    "#    data = baits[DpnName %like% myeQTL], params = params_main, fill = \"gray\", collapse = T, y = 0, height = 8, alpha = 0.4\n",
    "#)\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_main, height = 1, y = 5,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 5, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0, 12), label = c(0, 12), # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 5, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches3]\n",
      "\n",
      "arches[arches4]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# TFPT and OSCAR locus showing alleleic fold change, regular arches\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/TFPT_plotGardener_aFC_arches_V02.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"19:54045772:T:C\"\n",
    "BaseSNP <- \"19:54551115:G:T\"\n",
    "log2aFC1 <- -0.6040825\n",
    "log2aFC2 <- -0.3432409\n",
    "myGene1 = \"TFPT\"\n",
    "myGene2 = \"OSCAR\"\n",
    "mychrom = \"chr19\"\n",
    "mystart = 54036654\n",
    "myend = 54131092 \n",
    "\n",
    "###\n",
    "\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene1 & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all1 <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "# using regions_keep_all; only one promoter tested\n",
    "regions_keep <- regions[V7 %like% myGene2 & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all2 <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "# merge regions if right nect to each other! \n",
    "#regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "#regions_keep_LHS_merge <- regions_keep_LHS[1:2, ]\n",
    "#LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1)\n",
    "#RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1)\n",
    "#setnames(RHS_merge, \"V1\", \"chrom2\")\n",
    "#both_regions <- cbind(regions_keep_LHS_merge, RHS_merge)\n",
    "#both_regions\n",
    "\n",
    "###\n",
    "aFC1 <- round(2^log2aFC1, digits = 2)\n",
    "aFC2 <- round(2^log2aFC2, digits = 2)\n",
    "\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 10, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.3, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = c(myGene1, myGene2), \"color\" = \"chartreuse4\")\n",
    "\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 3, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 3.3, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 7.6)\n",
    "\n",
    "\n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/TFPT_DpnII_3013211.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 6.3,linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 6.3, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "### text for loops\n",
    "plotText(label = \"TFPT CHi-C \\nsignal, N reads\", fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 6.6, height = 0.5\n",
    ")\n",
    "\n",
    "########### TFPT\n",
    "### plot pairs and make the height equal to the score\n",
    "\n",
    "## REF\n",
    "for(n in 1:nrow(regions_keep_all1)) {\n",
    "    myplot <- plotPairsArches(data = regions_keep_all1[n, ], params = params_main, y = 4.1, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T, style = \"3D\", \n",
    "                             baseline = F)\n",
    "}\n",
    "## ALT\n",
    "for(n in 1:nrow(regions_keep_all1)) {\n",
    "    plotPairsArches(data = regions_keep_all1[n, ], params = params_main, y = 4.1, height = (1*aFC1), archHeight = (1*aFC1), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T, style = \"3D\", \n",
    "                   baseline = F)\n",
    "}\n",
    "\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 3.1, y = 4.2, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: \", aFC1), x = 3.1, y = 4.7, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"Allelic \\nfold change\", x = 1.3, fontcolor = \"gray40\", y = 4.3, height = 0.3, fontsize = 7, rot = 0, params=params_text)\n",
    "#plotSegments(x0 = 2, x1 = 12, y0 = 4.7, y1 = 4.7, linecolor = \"grey\", lwd = 0.5, params = params_main)\n",
    "\n",
    "######### OSCAR\n",
    "for(n in 1:nrow(regions_keep_all2)) {\n",
    "    myplot2 <- plotPairsArches(data = regions_keep_all2[n, ], params = params_main, y = 5.2, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T, style = \"3D\", alpha = 0.15, \n",
    "                              baseline = F)\n",
    "}\n",
    "## ALT\n",
    "for(n in 1:nrow(regions_keep_all2)) {\n",
    "    plotPairsArches(data = regions_keep_all2[n, ], params = params_main, y = 5.2, height = (1*aFC1), archHeight = (1*aFC1), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T, style = \"3D\", alpha = 0.15, \n",
    "                   baseline = F)\n",
    "}\n",
    "\n",
    "#plotSegments(x0 = 2, x1 = 12, y0 = 6, y1 = 6, linecolor = \"grey\", lwd = 0.5, params = params_main)\n",
    "\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 3, y = 5.3, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: \", aFC2), x = 3.1, y = 5.7, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"Allelic fold \\nchange (NS)\", x = 1.3, fontcolor = \"gray40\", y = 5.3, height = 0.3, fontsize = 7, rot = 0, params=params_text)\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead eQTL\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 1.8, height = 0.5,\n",
    ")\n",
    "\n",
    "### Plot the contact QTL, TFPT - would it be the same SNP? Yes, would be the same one\n",
    "contact <- BaseBed[Gene == myGene1]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTL\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 2.3, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.7, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD variants\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.8, height = 0.5\n",
    ")\n",
    "\n",
    "## plot ATAC\n",
    "## plot ATAC - now using the proper 34 reps consensus data\n",
    "ATACPval <- paste0(\"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus_34reps/consensus_34reps_Pval_\", mychrom, \".bw\")\n",
    "PvalRegion <- readBigwig(file = ATACPval,\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATACPvalPlot <- plotSignal(\n",
    "    data = PvalRegion,\n",
    "    params = params_main, height = 1, y = 0.5,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.8, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATACPvalPlot,\n",
    "    at = c(0,60), label = T, # top P value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0.5, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "dev.off()\n",
    "#regions_keep_all1\n",
    "#regions_keep_all2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "arches[arches3]\n",
      "\n",
      "arches[arches4]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis3]\n",
      "\n",
      "yaxis[yaxis4]\n",
      "\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAIAAAByhViMAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4nOzdd2BT1R4H8JPRmaZ7b9oCbSll76VYkCk4EBVUEBRF1AcIiIiPIT5RwAGKgANFZCgoIo/ZsgQKpS2lpYPSvdLdZjRp1n1/HN41thAKNyQxfj9/JffenPs7996c/HLHOTyGYQgAAAAA/P3xLR0AAAAAAJgGEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjsAAAAAGwEEjsAAAAAG4HEDgAAAMBGILEDAAAAsBFI7AAAAABsBBI7AAAAABuBxA4AAADARiCxAwAAALARSOwAAAAAbAQSOwAAAAAbgcQOAAAAwEYgsQMAAACwEUjswAS2bt0aHx/fu3fvBx98sLi4mBCyb9++Pn36xMbGTpw4sbGxsYNTAAAAgAsewzCWjgH+3tLT0x955JGrV696eHisWLEiPz9//fr1vXr1SktLCwgIePvtt1Uq1eLFi+84ZcOGDZauCgAAwN+b0NIBwN9et27d0tPTPTw8CCGdOnVKT08/ePBgQkJCQEAAIeSFF15ISEiIiYm545T2iZ1Op5NKpe0nuri4ODo6mqNuAAAAfyu4FAtc2dvbe3t7E0JkMtmGDRvmzZtXUVERFBRE5wYHB1dWVpaXl99xSvuTxytXrvRsx8fHZ8CAAeaqHAAAwN8JztiBaZSWlj722GPz5s0bNWrU+fPn2SxNr9fz+Xw+n3/HKTwer02ZS5cunTFjRpuJy5Yt0+v197EmAAAAf1tI7MAEcnJyJk+e/PnnnyckJBBCQkJCkpKS6KyysrKQkJCOTGlfrJOTU0RERJuJrq6ucrn8ftUEAADg7wyXYoErpVI5efLkHTt20KyOEDJx4sSkpKTKykpCyLZt255++umOTLFgFQAAAGwDztgBV3v27CkrK3vhhRfoW29v71OnTq1bt27kyJE8Hi8uLm779u0ikeiOUyxaCQAAAFuA7k7gb2bOnDlyuXznzp2WDgQAAMDq4FIsAAAAgI1AYgcAAABgI5DYAQAAANgIJHYAAAAANgKJHQAAAICNQGIHAAAAYCOQ2FmX5ubmBQsWhIeHOzg4BAYGzp49WyKRWDAejUazdOlSgUDQt29fC4ZB1dXVLVy4cO/evXv27AkPD3/88cczMzMtFUxhYeGsWbOioqLo8BhPPfVUVlaWpYIxtGDBAh6PN3v2bIusffv27bxbee+99ywSD3XkyJHhw4eLxWJPT89Ro0adOXPGImE4OjrecuPweLzi4mLzx1NcXDxr1qzOnTs7OTl16tTpySefzMjIMH8YbDAzZswIDAy0t7cPCwtbuHChTCYzZwDG2zrzt8zG4zFny2x8XbRZjomJcXZ2tnizDBQ6KLYiKpVq5MiRaWlpjz/+eK9evQoKCr777rukpKSUlBQvLy/zx5OTkzN9+vT8/Hzzr7q9+vr6AQMGFBUVhYaGBgcH9+rVa/fu3b///ntSUtKQIUPMHExqauoDDzygVquffPLJyMjIGzdu7N2799dffz158uSgQYPMHIyhy5cvf/bZZxYMoKmpiRDy9NNPh4aGGk43/z5i7dix47nnnuvcufO//vWv1tbWHTt2jB49OikpafDgwWaOZNGiRRqNps3EPXv2SCQSV1dXMweTmZk5ZMgQe3v7efPmRUVFlZaWfvHFF/369Tty5MjIkSPNHExRUVH//v3r6+unTJkSFxeXkpKyYcOG8+fPnzlzxs7OzgwBGG/rzN8yG4/HnC2z8XWxzfKECROmTJlSXFxswWYZ/sSA1diwYQMhZO3ateyUXbt2EUIWLFhg/mCam5udnJz69u2bn5/v4ODQp08f88dgaN68eYSQzz///KWXXnrmmWcYhvnll18IIWPHjjV/MHTAjNOnT7NT9u/fTwiZOHGi+YNhaTSanj17xsfHE0JmzZplkRj+/e9/E0JSUlIssvb2amtrxWJxr169FAoFnVJQUCAWi1955RXLBkYlJyfz+fz333/f/KueOnUqISQpKYmdcvXqVULIiBEjzB/MU089RQj56quv2ClLliyh33czrP2ObZ2ZW2bj8ZizZb7juthmmZ1iwWYZWEjsrEifPn3EYrFSqTScGBkZ6efnp9frzRxMfX39woUL1Wo1wzDWkNjNnz//4Ycf1mg0bGKn1+udnZ3DwsLMH8yyZcuWLl1qOEWr1drZ2cXFxZk/GNYHH3zA5/OPHj1qwcTujTfeIITk5+dbZO3trV+/nhBy5MgRw4nm/zbdkkaj6datW0xMDP2WmdmAAQMIISqVynCiWCwODw83fzCurq5BQUGG+6W5udnZ2XngwIFmWPsd2zozt8zG4zFny3zHdbHNMjvFgs0ysJDYtVVSUvLss8/6+vo6OjrGxsauW7eObXanTZtGCGlubn7llVe8vb2dnJwGDRqUmpra0tLyxhtv+Pv7i0SiwYMHX758+R7Wq1Kp+Hz+gw8+2Gb6888/TwgpKCjgWjEOrCGxY7GJnUqlsrOzs5LAioqKCCGTJk2yVAA3btxwcnJ6/fXXGxsbLZjY0cO1traWYZjq6uqamhqLhMEaO3ask5MT/eFRqVTNzc2WjcfQxx9/TAg5fvy4RdY+Y8YMQsjVq1fZKbW1tRY51yKXywkhw4cPbzM9Pj7e3t5eq9WaM5j2bZ1lW2bjba85W+YOrsuqmuV/LDw88RdVVVUDBw78+eefn3zyyffeey8iIuLNN9988cUX6VwHBwdCyLRp0zw8PPbv37927drLly8//vjjzz33nFar3b1798aNG7OyssaNG6dWq+921SUlJXq9PiwsrM10OqWwsJBz5WzNli1bNBoNzbYtSCqVnjx58rHHHnNxcVm2bJmlwpgzZ463t/eaNWssFQDV3NxMCPnkk0+8vLz8/Px8fX0jIyN/+OEHS8WTnZ0dERGRmZk5dOhQJycnNze38PDwb775xlLxsJqbm1euXDlu3LiEhASLBLB48WJ3d/fp06efOHGivLw8OTl56tSpzs7O9GK6OTk5OQmFwrq6uvbT1Wp1VVWVmeNpAy3zXbGSZvkfDondX7z77rtVVVW//vrrxo0bFy5cePDgwfHjx3/33Xf0gUcej0cICQoKWrNmzbBhw1577bXx48cXFxdrNJpNmzaNGDFi5syZzz33XE1NTUpKyt2umj4C5uLi0ma6WCwmhEilUhNUz4acPn160aJFgwYNevXVVy0Yhru7u5ub28iRI6Ojo9PT0/v162eRMLZv356YmPjFF1+0P37MjD488eOPP7722mvbt29/++236+rqnn322S1btlgknvr6erlcPnbs2IEDB+7du/ezzz5jGGbWrFmbN2+2SDysDRs2NDU1rVq1ylIBxMTE/PHHH2q1etSoUSEhIYMGDcrPzz9+/Di9RGtOfD6/X79+ubm5hk9TFhUVpaamEkLo+TwLQsvccVbSLAMSu7/Yt29faGjo6NGj2SmfffZZUlKSv78/O+WJJ55gX0dFRd1yigmfhGcYhvw/pwSqpKTk4Ycf7t69+8GDB+3t7S0YySuvvDJr1qyhQ4fu3r175syZFvn7XlNTs3DhwqlTp06YMMH8a29j+fLlP//889WrV1esWPH888+vWbPmwoULjo6OS5cubW1tNX88Go2mpKRk/fr169ate+KJJ1577bULFy64uLgsX75cq9WaPx5KqVRu2rRpyJAhffr0sVQMOTk548ePb2lp+fjjj3/77bcvv/zS3d19zJgxJ06cMH8wixcv1uv1kyZN+vXXX/Pz8/ft25eQkEAfrKbXSawQWuY2du3aZSXNMqC7kz9JJJLGxsY2TW1ERERERIThlKCgIPa1SCS65ZT2nRrckZubG7nV/z86hc4FhmEuX76clpY2fvz43bt3W/wE1X/+8x/64syZM2PGjHn00UfT0tIEAoE5Y3jjjTcYhvn000/NudLbad9TRmxs7Lhx4/bv33/16lXzn9EUiURardbwr1dgYOCYMWN+/vnnnJyc7t27mzke6qeffmpoaHjppZcssnZq1qxZdXV12dnZbMc0zzzzTLdu3WbMmFFYWGjmH+bJkydv3LhxyZIljz76KCHExcVl1apVqamphYWFnp6e5oykPbTMd8QwzIoVK1atWmUlzTLgjN2f6Dl/Jycn44u171fJJD0thYWFCYVCeg++IXoSiJ4I/IdjGGb27NlpaWnR0dG//fabVTUfw4cPHzdu3NWrV69fv27O9R4+fHj37t1Lly7VaDTl5eXl5eWVlZWEkJaWlvLyciu5TuTr60ssdE2tU6dOhBA+/y8NHY3HzP3fGtq7d69AILDgGVa5XH7hwoW+ffsadjcoFotHjBhRUVGRl5dn/pDmzZsnkUhOnTp15syZysrK+fPnZ2dnBwQEWDxzQstsHG2WV61aNX/+fGtrlv+xkNj9iR6Rlrqlw87Orl+/fqmpqQqFgp2o0+lOnToVFhbWprvXf6b58+d/8803AwcO7N27d5ufanOSSCQ9e/akz8QZopdmDHefGSQmJhJCFi9eHPJ/3bp1I4Ts2rUrJCTk/fffN2cwcrl88+bNO3fubDM9Ozub/P9mczMbPHiwTqdLS0sznFhQUEAICQwMNH88hBCNRnPy5MnevXtb8FyUUqkkhKhUqjbTW1paCCH38OwXdzqdjmaWw4YNE4vFJSUlV65cGTVqlPkjaQMts3G0Wf7oo482bNhgwWYZDGE3/Mnf39/Dw+PatWv0F5rKy8vbtGnTtWvXzBDAjBkzWlpa1q5dy07ZsmVLZWXlCy+8YIa1W7n9+/d/+umnCxYsoB3wWpC/v79cLt+9e/elS5fYidevXz927JiLiwvNq8xm1qxZB/9q9+7dhJDRo0cfPHiQdmlhNs7Ozu+///5LL71k+H05dOjQmTNnevbs2eaWBvOYMWMGj8dbtmwZe4ff5cuXjx8/HhsbGx4ebv54CCFZWVktLS2WugpM+fj4REREpKen5+bmshMbGhpOnz4tFovNfAwTQpYsWeLk5MQ+c6bX6+fPn88wzNy5c80cyS2hZb4dtll+8803LR0L/An32P3Fo48++s033+zZs4f2hE4IWbFixe7du9PT082w9pkzZ+7YsWP16tVXrlzp06dPbm7unj17evTosXDhQjOsvY3Tp08fPnyYvtZqtRUVFW+99RZ9u2jRIvMPcbZ48WIayaVLl9RqNRsMIWTJkiUeHh7mDGbbtm2jR48eNmzY448/HhERUVFR8dNPPykUio0bN97xUr5pxcTExMTEGE6hz6WGhISY/0ofn8//7LPPpkyZMnDgwKlTpwYFBWVnZ+/fv9/FxWXr1q1mDobq3bv3ggUL1q9f379//wkTJtTX1+/YsUMgEGzcuNEi8RBC6MV6S6WVrPXr1z/++ONDhw59+eWXo6KiJBLJV199VV9fv3nzZkdHRzMHM3369E2bNo0ePfr555/38vI6ePBgSkrKm2++aZ5HdO/Y1pm5ZTYeT1ZWltla5jtuGbZZNmyQKfM3y/AnC/WfZ6XKy8v9/PwcHBxeffXVjz76iP40Pvfcc3TurFmzyF971ad9Pp09e5adsm3bNkLIrl277i0AuVy+ePHisLAwe3v74OBg2t8slxrdM/axgPYsMq6AkWO4qKjI/PGkpaU99dRTnTp1sre3d3d3f+ihh3777Tfzh9GeZTsoZhjm7NmzjzzySFBQkJ2dXUBAwPTp0/Py8iwVDMMwer1+y5YtPXr0cHBwcHNzGzt2bHJysgXj+fzzzwkhn3zyiQVjoP74449Jkyb5+PgIBAIPD49Ro0YdOnTIUsGcP3/+4Ycf9vLycnR07NWrl+HwYvdbR9o6c7bMxuMxZ8t8x3VZW7MMFI8xum/+gYqLi5cvX37s2DGpVBoRETFz5szXX3+dPiM2e/bsr7/+Oj8/n71hdsWKFStXrjx79uzQoUPplK+++urFF1/ctWsXe84PTGvOnDlyubz9jVwAAACAxA7+ZpDYAQAA3A4enrBGOp1uxYoV9Mk1a7B161aL9IBwSxUVFbRHD2tQXFxswbu12tBoNCtWrLBIP8C3tHnzZvr8qTU4duzYsWPHLB3FTQUFBRYf+oLV2tq6YsWKe+h68z7ZuHFjcXGxpaO46b///S998Nwa5OXlWepe1faUSuWKFSt0Op2lA4FbQGJnjerq6lauXGk9rdvHH3988uRJS0dxU2FhofVsmXPnzhk+K2dZEolk5cqV5eXllg7kpnXr1p0+fdrSUdy0a9euXbt2WTqKm06fPr1u3TpLR3FTeXn5ypUrTThYDkdr1649d+6cpaO4aefOnXv37rV0FDedPHny448/tnQUNxUXF69cubL9CL9gDZDYAQAAANgIJHYAAAAANgKJnTWiPb9zHw/q4MGDCxYs4B6PRCLJz8/nWEhZWVlCQoJer+dYTmNjo0lGyho3bhz34b+Kiopo9yIcLVu2bM+ePRwLoTdlch86Zf/+/bR7Ko5qa2u532NXVFQ0evRo7sHk5OTk5ORwL2f06NHtR5e6WwUFBbW1tdyDWbx48f79+zkWQg8Y7nf07tmzZ9myZRwLIYQ0NjZy38LXr18fN24c92Byc3O5HzZ6vT4hIaGsrIxjOfn5+Sa5Yr5gwYKDBw9yLIQ2whYZpATuCImdNaK9q4hEIo7lXLt27cKFC9zjcXV1DQgI4FhIeXl5YmIi93u0nZ2dTdKB6okTJ0pKSjgW4ufn5+zszD2YS5cuZWVlcSyEbhbu8WRlZV28eJFjIYQQFxcXPz8/joWUlpYeP36c+8P7AQEB3I9hhmGOHz9eWlrKsRw/Pz+TDKl58eJF7ocNPWC4f6eysrIMx2LhEg/3w6akpOTEiRPcg/H39+d+2Gg0msTERO43vwYEBLi6unIshBBy4cIF7mMp0Z8n+lMF1gYjT1gvgUBg6RBucnZ2tp6hnR0cHLif9jMVZ2dnBwcHS0dxE4/HI+0GvLcgJycn6zls3N3dLR3Cn1xcXMw8QokR9IChB481cHBwMMmfJZNwd3e3nmPYxcXFeraM9fw8QXvW8hsAxhkO163VarVa7S1ntXnLMIxhDqRWqw3fGvmgTqczPLXW5nyJkQ9qNBrDB+DbjzLO0uv1hqfx1Wq14VqMV6rjSxr2/dEmtjYMP9gmttbWViNrNJx1V7vGMDa9Xt/BSrXZNUa2MLmbXWNkC9/VYWNYfeOHjZFd0/HDxviuabNPO1iLNrumTfXbMLJr2vQ7Y/yw6eCuaVP9jn8XzHPYGG6oju+aNt8aI4eN8V1jpFLtd03H25COHzZtGrR7O2yMrIL8deN0fNe0OWw63toYb9DAaiGxs0a08aKDfhJCDhw40LdvX3buG2+8YXgLVFxc3NGjR+nrGzdueHp6srfLpKen37hxg11y6tSpGzZsoK91Op2Pj09mZiZ9e/bsWcMx2v/973+/9NJL7NuysrKMjAz6ura21sPDo7q6mr7dvXv3sGHD2CVffvnl5cuXs2+joqLYDi9o53Nsg/LJJ59MmTKFXXLSpEl0qCVafU9PT/YGODpkO7tkXV2d4W1t/fv3/+WXX+jr8vJyDw8Pdrt99913CQkJ7JIzZsxYs2YN+1ar1bLdpqSmpvr7+7OzPvjgg2effZZ9O2bMmK+//pq+lslkHh4e7DXca9euGT7wP3/+fMPBsOPj49mRFgsKCjw9PVtaWujbLVu2GI7lmpOTc/nyZfqaYRhfX192g587d85wXNGVK1fOnj2bffvggw/+8MMP9DXd7zKZjL7du3fvkCFD2CXnzp1reAtUly5d2C5ssrOzvb292V+XtLQ0w7HhH3300U2bNtHXarXay8uLnZuYmBgdHc0uuXTp0ldffZV9W1VVxR5glZWVHh4eDQ0N9O33338/cuRIdskXXnhh9erV7NvQ0FD2FoI2dyZ9+OGH06ZNY9+OGzeODuJHCFEoFB4eHuztWYcOHerZsye7ZHp6uuGIzz179jx06BB9XVRU5OHhoVAo6Ntt27YZ3p41bdq0Dz/80DAGNqQLFy6Ehoay01evXm04KvzIkSO///57+rqhocHDw4PtfzEzM7Oqqopd8tVXX126dCn7Njo6mu07LTc318vLi/1V3rRp06OPPsoumZubm5aWRl9rNBpvb+/s7Gz69uTJk126dGGXXLZs2dy5c9m3Q4YMYXvxoAcM22j88MMPDz74ILvk7NmzV65cyb4NDw9n+yLJyMjw9fVlv9GXL182vB1twoQJW7Zsoa9bWlo8PT3ZGy4PHz4cHx/PLvnmm2/Onz+ffVtXV8deKCwpKfHw8GAP6a+//nrMmDHsks8+++wHH3zAvvX3909NTaWvi4uLDTOSNWvWzJgxg32bkJDw3Xff0ddNTU0eHh7sddJffvmlf//+7JLp6ensFiaExMbGHj9+nL6+fv26p6cnm2h+/vnnkyZNYpecMmXKJ598Ql/TTcTu/dOnT7NjFxFCli9f/vLLL7Nvhw0btnv3bvq6urraw8ODvR0zIyPD8Ovw0ksv0TEtqYiIiLNnz9LXmZmZPj4+bNq3YcOGqVOnskveuHGD/S4olUpPT0/2Z+Lo0aNxcXHskosXL37jjTfYt3379j1w4AC73Ui7fy9gJXAp1hrRq3vsxSOpVGr4uIBUKjW8IcZwrkwmUyqVarWaXujp2bOnYZdvUqmUbSL1er1cLmc/2GYVMpnM8G1QUBDbELe0tKhUKjZBaR+b4cUCw7mBgYHE4IpP+w+yb9VqtVKpZENts6SXl5fhjd6GcxUKhUqlUiqVdNO1XwVbJiFEKBSyCZPx6hvOValUKpWKfUAhNjbWcNRtqVQqFApv+UG5XK5UKltbW+n2abPG6Ojo3r1709d6vd4wgI7HRne6WCy+5QelUqnhDTFtDhuFQqHVau3s7AghPXv2rK+vv+WSGo2mpaXldrumTWz+/v7sjwS7a24Xm+GuMZwbEhJCDHR817RZRY8ePQzLabNr6GfpbUPGYzMMqeO7RqlUqlQqNneMi4sz/CMhlUoNL6C32TUtLS0ajYbuuzZr7NKlC5u8arVahULRwV1jOJceMOyl4Y7vGjpLr9fTq3K9e/c2PC1kuGRra6tSqbzdrpHJZIZJmJeXF/tHjt01NMiOxxYeHm74Tez4rmmzivj4eMNLsW12DW1saXPdPjb2LW30aAN4y+rf7oNtGtv4+PigoCDDJQ1vuWuza+RyObtr2qwiMjKSPWyMN7YymczwLJ3hXNrGWs+NKGAIiZ0t4/P5JrkTQiAQWM8tOHw+33puI+PxeCYJRiAQWM89K6YKhs/nW89hY/gzb3E4bIywtsPGeo4cHo9nqvbceppQuB+wdwEAAABsBBI7AAAAABuBxI6r5ubmBQsWhIeHOzg4BAYGzp4923pGXQQAAIB/FGu5e+BvSqVSjRw5Mi0t7fHHH+/Vq1dBQcF3332XlJSUkpJieE89AAAAgBkgseNk8+bNaWlpa9euZfsfGT169NNPP/3++++vX7/esrEBAADAPw0uxXKyc+dOsVj8+uuvs1OeeuqpyMjInTt3ch8E6W9k69at8fHxvXv3Pn36NH0eft++fX369FEoFKtXr6bdzp06dYoQ0rNnz4kTJ9IpBQUFIpHI0dHR398fXV8CAABwh8Tu3rW2tqanp/ft27fNMItDhw6trq7mPo7130Vra+vq1atPnz6dlpbm4+OTnJwskUjmzZv3+++/i0SisLCw1atXSyQSegrzypUr3bt3X716dXV1dXl5+blz51QqVUREBNsdMQAAANwzJHb3rqSkRK/Xh4WFtZlOpxQWFloiKAuwt7dPT0/38FH/AHIAACAASURBVPAghIhEIo1Gc/DgwYSEBDp49qhRo/bv33/w4EF28IwXXnhh//79Fy5ccHBwoP1krl69mu0gFAAAAO4Z7x91xdC0UlNT+/btO2/evI0bNxpOX7du3aJFi/bt2/fYY4/Nnj2bDtzEMIxEIvHz8+Pz+a2trVKp1MfH53Yla7Xa6OhokUhEu6NUq9XV1dXswDI1NTVxcXHBwcH0bXNzs06noyNMqNXq8vLyhIQE2i24RqMRCoUtLS15eXmEkKqqKkdHx759+7q5uTEM09zc7OLi0trampeX19LSUlNTEx4eHhkZ6ebmplQq9Xq9SCRSKBR5eXk8Hk8ul4tEosjISLFYTHs85/P5CoXi8uXLTU1NtC/+sLCwy5cvx8bGymQynU4XERFx9OhRHx+fmpoad3d3sVjcv39/OvbigQMHxGKxTCajYz/o9fri4uIJEybQc58Mw5SWllZWVjY3NysUCoZhDDssZRhGKBSyQzmVlpYOHjzY19eXliOVSgUCQX5+Pvl/P+kjRoxwc3MjhCgUCj6fr9fr6dag8cfFxbm5uWm1Wrlc7u7uTuvb0NCgVqv9/f3p1pDJZPb29g4ODgqFIjs7u6SkJCQkxM7OLjIy0tnZ2d7enn6JFArFH3/8QQihezYyMpIQ4uzsTIdzaG5uPn78eHh4OO0aNDAwUCQSsaNEyOXyq1evsqd+XV1d2a7ztVptZWUlO3BZfX19dHQ0O/iBTCZTq9V0iDM6dER4eDhNstVqtUqlEggEtL41NTV8Pn/AgAF0azQ3Nzs7O6vV6ry8vNbW1oqKik6dOkVFRdGtQbukplujsrLS2dnZ3d09MjLS1dW1ublZLBYLBAKFQpGWllZbW0t3YmRkJN0UdGgNhUKRn58vl8vpW1pxetjQWpw7d449hn18fLy9vdnBD0pLS8vKyuhblUoVFxfH9rCvVCobGxvZB88rKir69etHR3GghzSPx6NDV8nl8vr6+pEjR9L60uGVVCoV3RolJSVeXl49evRwc3PT6XQymczNzY1+U5qbm+VyeVBQEN37crlcIBA4OTmx3wWlUuno6BgZGSkSiehhQ+t7/vx5rVbr5+dHtwafz6eHDd1NR48epYcNISQkJMRwMC6JRHL16lVvb2/61t3dnV1Sp9OVl5fX1dXRr0BTU1NkZCT7Z5JWmf6HZBimqKjowQcfpI9t0TFChEIhrW9dXZ1erx80aBDdGlKp1NHRUaPR5OXl0cNGo9F07drVzc2ttbVVrVaLxWJa3+rqaqFQ6OXlRbdGU1OTi4uLUChUKBQZGRmVlZV0KEK6NdiezBUKRVJSkouLC10dXYYeNoSQhoaG06dPs7Xw9fX19PRkR6xRKpWXL1+mo4BoNBovL6+IiAh62LS2ttbV1bHjGUokkp49e7IjOtBBrujWUCqVIpGoU6dONADaoBFC6NYoLy93dXXt1auXm5sbbTRcXV2VSmVeXp5cLm9oaAgNDaX1pcOq0qM9Ly+vpKTE29ubtoQuLi70sKHflIsXL7a0tNBgIiMjBQKBUCik32i5XJ6fn69SqejBEBISYm9vTwOjX8zU1FR62BBCPDw8goKC6EAjer2+rKyM7gK6VXv06MF+FxQKhUKhYAdDKy4uHjZsGG18aIMmFArplZDGxkaVSjV06FC6UoZhtFot/e7Ttjc4ODgmJsbNzY0eNm5ubrS+tbW1dIRDujXoYWNvb69QKLKysuzt7U+ePNlmbBiwBnh4wvRoe00buOnTp7MjD165ciU+Pp7P5yuVyuLi4piYmNuVIJVKz58/HxcXR5uJlpaWioqKrl270rmVlZV8Pp+mMoSQ69evh4WF0V9BvV6fmZkZHx9PY2hqapLJZKGhoTSY4uJiV1dXDw8POvfatWtdu3a1s7Pj8Xitra35+flxcXF0WPHa2lqNRhMYGMjj8Xg8XnFxsaenp6urK03LsrKy4uLi6KyEhITKysouXbo0NDRs3rx5+PDhAwYMuHTpklqtnjRpUufOnUNDQ996662hQ4cqlcrIyMjw8HC1Wn3w4EEvLy+RSLRkyRJCiFwuX7hwIdvuEEKysrK6deuWm5t78eJFnU4nl8vZWXl5ec7OzvSDhJDc3Nzg4GD6q6DX669du9a9e3f6MyCVSqurq6Oiomh9y8vLnZycvL296dYoKioKDg4WCoUMw2g0mvz8/NjYWFqp+vp6uVweFhZGt0ZhYaGPj49YLObxeJMnT2b3I8MwCoWirq6OZpk8Hq9///48Ho+eqmQYJi8vLyQkhO4ahmE6d+7MDmzV3NxcV1fXqVMn+raiomLYsGHsT3tmZmbXrl1pa65WqwsKCrp160Zn1dbWqlQqdmShgoICPz8/2tbTn3Z2nLSWlpbKysrOnTvT+paXl9vZ2bH5RG5ubqdOnRwdHXk8nlarvXbtWo8ePWh9GxoaWlpagoOD6dYoLCz09PR0d3enc7OysmJiYuhP+JgxY9jYGIaprq7W6XS0+jwej/4K0hGZ6JFJDxu6x/v06dO5c2caallZmYODA1v9nJycZ599lv62aTSasrIytlINDQ0KhYLNCQoKCry9vdn8mK6CZg8tLS0lJSXR0dG0vjQXDAgIoAHk5OTQbw09pOlhQ+tLd01kZCStb1lZmUgk8vT0ZLcGzbrogPQFBQX0i8zj8YYMGaJSqejvHMMwN27c8PPzYwekioiI6NGjB127TCarr69n/5xUVVWNHj2a/WnPzs4ODw+nOYFOp8vJyWEH9GtqampoaGC3Rnl5uUgkYsdczsjIiI2NpdVvbW0tKiqKiYmha6yurtZoNEFBQXRr3Lhxg/614PF49LChhyLDMHK5vLq6OjIykta3rKzM0dHRx8eHbo2cnByawfN4vAkTJmRnZ9PYGIapq6tjj0wej9e9e3dvb282k8jMzIyNjaVfTJVKFR8fzzaAVVVVDMOwY6yVlpY+9NBDNLGjh0337t3pLJlMVltbyw60WlJS4uLiQv/G0O3WpUsX9rApLy8PDw+n9a2vr1cqlSEhIXRr5Ofn+/n5icVi9pDu1q0b+2emrKyMHjb03ziPx/Pz86NbIzs7OyIiwtHRkeZGOTk5dNA8Ho/3wAMPNDQ00BSWYZiSkhJXV1ea9PN4vKKiotDQUPqNVqlUhYWF7CDLdXV1I0aMYP/kXL9+PSAggFafxhYfH0/DlkqlNOmkS1ZWVtrb27M/BFlZWVFRUbT6Wq02Ly+P/bo1NDRIpVLaoNGvm1gs9vDwoHMzMjLowc8wjEqlKi0t7dKlC62vRCLR6/WBgYF0a1y/fp02tjwer1+/fhkZGYajFIIVYeBe0dNCzz33XJvp77zzDiEkMTHxnkuWSCSTJk3Kzs7mFiBTW1t79uxZjoUwDHP8+HGZTGZkAdqkHj9+nL79+uuvp02bRl/n5uZGRUXRKb/88gs7ZebMmbTZZRjm8OHD9JeyI6ZOnTp+/Ph7rIkBGgxHra2thw4d4l7OxYsXKyoquJdjkkpJJJLz589zL+fIkSMtLS3cyzFJpXJycnJycqwkmJaWliNHjnAv5/z58xKJhHs5JqlURUXFxYsXuZdz6NCh1tZW7uWYpFJZWVnXr1+3kmBkMhnbxnJx9uxZeiqOi+zs7EmTJpnk8AOTwz129y4sLEwoFLZ/SIJeC2D/Vt6brKwsLh+nXFxc2D/3XERFRbV5QMSQUqmcPHnyjh07EhIS6JSJEycmJSVVVlYSQrZt2/b000/TKfTPJZ2yePHi0tLS9PR0Qsjy5csHDhzYwWCcnJyMBNNx7Pj0XAiFQvafNxchISHsKUkuTFIpV1dX9qwAF+zpE45MUilfX1/23IbFg7G3t+/SpQv3ckJDQw2Hgb9nJqmUm5ubSa7KRUdHm2R4VpNUys/Pz8g9M2YOxtHRkePPChUeHs6eSObCJD9ScD/gHjtOBg8enJGRUVNTQ8+cE0J0Ol1oaKidnR296e2WGIYpLS2lN/3cUl1d3YABA7Kzs41crrUe27dvnzt3Lr0GQQjx9vY+derUjz/+uGrVKh6PFxcXt337dpFI1H7KvHnztm3bRgjx9fVNTU3t4O/unDlz5HL5zp0772OVAADg9nJycmJjY+mN45aOBdpCYsfJ1q1b58yZs3z58lWrVtEpX3zxxauvvrpy5cp33333dp86derUgw8+eMfC//jjjyFDhpgsVluBxA4AwLKQ2FkzPDzBycyZM3fs2LF69eorV6706dMnNzd3z549PXr0WLhwoZFPjRgxori4+I5n7Dw9Pe9DyAAAAGCzkNhxYmdnd+TIkVWrVu3Zs+fo0aO+vr6vvfbaypUr2Suzt8Tj8dr3fmdIJBLRx8e40+l09Ck5ixdiwnJMcprZqiql1+vpY2jWEIypyrGqYGhvF9y/VlZVKasK5uZd27a1hfHFNMJUP1JgctgxXIlEorVr1xYXF7e2tpaVlX366af0EXeOxo8fz72Q+vr65ORk7uWcOnXKJB0IHzp0iHshtHc6KwlGo9EcO3aMezmpqals32xcmKRSNTU1KSkp3MtJTEw0yUhxJqlUfn4+fYzdGoJRqVSJiYncy0lJSampqeFejkkqJZFIUlNTuZdz7NgxjUbDvRyTVCo3N5d2i2gNwSgUCjowI0fJycn19fXcyzHJjxTcDzhjZwFSqfSnn34ycilWKpXy+XwjC3QQ/Q/NsRATlkPPmnAPhnshxHTB2OQWtslKcS+EYAvf/2BsslJWFYypysEZO6uFxM4Cbty48eGHH2q12tstoNPpevbsKZfLzRkVAAAA/N0hsbOA3r1705Ftbqe6unrOnDkm6dgMAAAA/jlwKhUAAADARuCMnZX6G4080XEm6X7dycnJyFVsMwdjwpEnrGcIAVsdeYJ7IQQjT9yem5sb96dHifWNPGE9wWDkCeggdFBsATdu3HjnnXeM3E6rUqkOHjx4+fLlPn36mDOwvwV0UAwAYFnooNia4YydBTg6Onp5eRk57aRUKgkhJulqCAAAAP45kNhZQHBw8Oeff25kgerq6h07djg4OJgtJAAAALABeHjCSplw5AkrKcSE5Zhq5AnuhZiqHL1eb3uVsqpg9Hq9SfoSs6pKWVUwDMPY3hbGF9MI9GNntbBjrBRGnrgdjDxhBEaeuB2MPGEERp64HYw8YQRGnrBauBRrARcuXJg4caKRf7d6vf6BBx5obGzkuCJr66kcQwjc73JQKSPBcC+EYAvf/2BsslJWFQxGnrB5SOzMRKtnwt87EeMnPj5nYO/evXsu+SanmRBCKpU8PwdGwCc+DsyyaLWOIa+mOwTaacjVHeyYs0qN7t0jeT9frRLyeXwemRTnv2pMtKPQ9F8qeatW/PbhGD8XPo/HMESj0z/RI3DN2GillhH/+2QXH5FhdwZp84frGUb89mHD6eNjfNc/0q19yU1NTT4+PoGBge+88w4hZPbs2f/617/orE2bNr322msajcYk3QoAAAD8k+Gn1Ex+uyYR2QvOFzfcqFNEeYtOLHqEThcu+v3yklHBbjd7idufWeVdmFPcpAyyc2Wzpak7UgU8XvqC4e5OdlKVdubuK3N+yvju6V73KdRLbwxzcRASQqQq7bBN57r5i2lPWanzh9PpLHmr9pbT22tubg4JCdmwYcPkyZMNp1+7dm337t3Ozs4mrQEAAMA/FBI7M9l8vmT2gLD0iuatySUfTog1vtiFgurM2q50Skal9HRBfenyBDdHO0KIq6Pwu6d7avV/OZHeqNQ892N6i1onbdX2CHT98ol4IZ+3+XzxzpQSbavSLTv588e6R3mL5u3PvFTaFO7pHOzmmF0tO/LSQOMxuzoKh0d65VTLBnDudrSpqan9CGkqlWrWrFnffvttv379uK4AAAAAkNiZR36d4mxh/cr+YjdV61t/VE32kTfW1fr7+xNCGIbJzMysduITQkpl2jMFdUviBDyx7DfnCI2eIYRcqWjuHuBKszqq/emxA1kSsYPw4Kz+hJBvLpXWylurpK3bU8qSXuzf2FB/qVEwb3/mwgciTxfUpy8cIeTzpu1ME3bg9ogqqepwTs3Gx+LCBPbk+DUuW6C5ubmxsfHTTz9ds2ZNZGTkRx99FBISsnjx4pkzZ8bExNzuU+vXr1+zZk2biTKZzMPDg0swFEaeuB2MPGEERp64HYw8cb+DwcgT0EEYecIcFv52rayx5fC8kXK5nMz4giTvJrlnbs6b/yv5ajaR1RFCyIgXiIsXOfQRIcTplW+f7sQbE+58tk5wRCLcNtJtxIgRbIEXLlwoLy9n3+pdvN+5wvQKchvVxWdynP+NzNStl6sOVAgDnBhCiEBoJ3RyeapX4I06xdYpPS5cuLAzvepCveCtaDUhxNfXly2Z3mMX6szwCKPS8dR6MjvW6cPnRtHpXXxEKpWK9qvczVX3Yiet2NNn7H9l7D12SqXyAc/WCQE3O142LLm0tHTz5s0xMTFOTk4HDhxIS0t7/vnnL168uH//fkKIi4vLkSNHqqqq2BrRz5aXl1+4cIEQcv36dfYxru+++04gEBg+DNhmaxiuF3OtZO6qfZv2F/0x0atfvEuo9USFuZiLufc219fXFyNPWC0kdvedSqsPWnnMX+wodhASQiQyVbin86m5g+lc4aLfi99JCHZzNFxMo9Gk5RX26xp26c2E7GpZv0/OFi9L8HG5efKjRa37o6jhRp1iw+lCQsiHE2Me6x6g1TMXSxpP5Nd9lVxy/OVBx/JqL5Y27pzWmw1jbdKNgnrF1ik9CCHfXCrdf1Xy++z+bUKlCZzs/bEuDsLyZlX/T84mvTIo2tfFcPrtlu/4BqmoqIiOjp44cWJOTg7thDklJaVfv3579uwJCwu748djY2OVSmVRUVHH1wgWtzP10IcndzzYue8nk960dCwAwBWGFLNmuBR73+1Jrwj1cEpfcPPfj1SlDVl9PLdGHu3rcrvFqqur/UMn5CzYm1sjj/UTj+nqO21n2vfP9PIXOzQqNbP3ZHiJ7LZO6TF3SDj97IEsichekNDFZ0gnz6uV0rTy5mERnmtO5Ncr1F4i+zOF9ZlVsvhA1x2p5Vo9I+DxDl6rph/MksgivZyd7G4xdlmwm+P746Kn7UxLfn0o941w4MCBnTt37tq1SyAQHDhwoG/fvj/++CM718XF5Y8//sBTsTZMZO8s4PNN0ukDAAAYgZ/S+27z+ZJ/DYtg37o6Cmf2C/3yfMknk7sZWYyvVT3a2ZUu9uP03quPXx/82R8MIY5C/rN9g5c8+Jc7LeICxC/uvbr6eD5DmFB3p8lx/s72gpUPdx2z7aKLvUDPkM1PdI/2ddl31aPPhjM+Lvbd/MUFdS2EkAe+OJ/48qAegbe+TWdGv5D9mVXvHM57e2QEIaTPx2cM76H56skePW/zwfbGjRt3/Pjx+Ph4R0dHX1/fb775poMfvCVT9QVvktF4TVKOXq/n8Xjcb1GyqkoZFnI0L5kwJNDVx1LB0JySe89bVruFLV4O7R3NxrawzX8xuUA/dlYLl2ItoKamZvv27Ua2vFwuz8jIeO+99+Lj47msqL6+Pjc3d8iQIW2m/3y1avulst9n93/7v7mvD+vkL77DoLSJiYkDBw4UiURcgiGE/Pbbb4888gjHQqKiopRKZUVFhTUEo9FoTpw4MXbsWI7lpKSkBAcHBwQEcCzHJJWqqakpLCwcOPAOD03f0bFjx4YPH+7o6EgI6b1+GiHMkz1HvfXQzLstxySVysvLI4R07dqVYzkmCUalUp05c2b06NEcy0lOTo6IiOD+XIhJKlVVVVVeXs79CffDhw8nJCTY2dndeVGjTFKp7Oxse3t77o8smCQYhUKRnJz80EMPcSzn3Llz0dHRXl5eXArJyclZsmTJtm3bcCnWCuGMnQXU1tYmJiYauSylVqs9PDy4D8d0xx7GewS63jGr60g5HWRVV+Jssi94q62Ur9hTZH/nI+2WMPLE/S7Hlio1esvcaN/wzx5dbFWVsqpgMPKEzUNiZwHdunU7evSokQWqq6vnzJkjFovvUwBPxAc8ER9ACJnaM/A+rQLAkL1AaCfgeg4G4I6alHKJrMHSUQBYEjJuALi/GlqkJY1VUpXC0oGA7dPr9cUNlZaOAsCSkNgBwP3FEIYQ4i1yl8jqLR0L2KzXf1l3tuiKvdDOjo8rUfCPhi+ABWRlZc2ePZv29HtLGo1GoVDIZDKOK3JxcQkPD+dYCCEkKiqK3gLPkUm6XxcIBCZ5pAsjT9zOfRp5okJak1dzL70PYuSJ28HIE4bOFKSqtWo7gVDA4xOMPHF7GHnC5iGxswB/f//HH3/cyO20crn8vffec3Jy4rgiR0fH4OBgjoUQQkySHRJCTNIqmSqxM0kwfD4/IiLizsvdCffnYSmTVMrJySkoKIh7OZ06dTJ8+0jsiHPFGfdQjkkq5enpyb0QYrpjuM3GuTcm2U3ERJUSiUTcH5wnhNzzF4ohjLvItUWtEtk7ERNVytvbm3shxETBCIVCkzTFJvlRIIQUFBSYpBwwOSR2FuDt7b1o0SIjC1RXV7/33nvosBcAoONOXU9R6zROersPk75bPPJ5S4cDYBm4x85M5vx0dWtyyb19Nrmksef603dc7OerVRO+umRkgctlTcM2net4gR239FDOmhP5JiwQAKDjbtSXEYb4uXoRQhQa1c60w606taWDArAMnBMyh58yKmvkrS8N/HMgVJlMZuQeu+bmZlN1EWTYyXjfEPez89p2Vny3hdzS++Ni+n1yJqGL94BQDy7ldBBGnrivwZiqHKsKBiNP3O9yLDvyRK2skRAidnAmhGj1ekKITqsjnLcNvphGoB87q4UdYw6rjl1fltCZENKo1Ez8+lKf/xx2feUrz6f+7ent4+np6Tlqpue8bzznfes59V06petLa91e3vbMryWjtyTXyv/yv3Pz+eIHvzg/dtvFh7cm36hTEELm7c/s/8nZp3akJuXXGS55rqih50dJz25LevL71AGfnr1U2nTLE3VtCvyjqKHvx2forN3pFZO/TTlX1BC75si/9mew5dBaPLT5Qr9Pzs7em6HVMzweWTIy6o4n7Q4dOsRxSxJC1Gq1RqPhXo5JgtFoNMeOHeNeTmpqqkQi4V6OSSpVU1OTkpLCvZzExETunWwTE1UqPz8/P98EJ5VNEoxKpUpMTOReTkpKSk1NDfdyTFIpiUSSmprKvZxjx47dwxfcyc6BEGIn+PNUxSc/f8U9mNzcXJPcSWaSLaxQKE6dOsW9nOTk5Pp6EzyfPn78eO6FwP2AM3b3XVmTslre2ifYnRByIEsidhBeWjw6MzPgl/y+g5dPr2nRrUhu2jnWx47PW5ncY8wbT8d42n2fUV9/dOvaeWu+ytdvPl88uuvNETbTypu3p5Sdf22ogM/bn1k1b3/mwgcizxQ2pC0YLuTznv0x3XC9Aj7vWk3Lv3u5PZrQZ39m1fwD19Y/EsvOZQgz6su5ro5R5c39zr82dNWxLYfzGodszBU7pNUpBkz6ZkGAq/cDkc8zDDP/wLr85oG/DQzuGuhFy5nZP7CiudxPfNLfTnC5LHDN8WtvPTRlTLTvjN1XNDq9neC2/xYw8sT9LsfKK9WiUTWrFG6Od3eL/V1VKjE/ZcGB9RO7DX9v7Nw2wdzVSk0SzO3gsLnf5fAIz55vgg6xsYWNwBk7q4XE7r4rblCGeTjTc/kPRHqtOZH/9M4ro7r4zBsb6+Ni/+nZwodinAf260EI+a0PIYToGeZY9ZUjfg8tSJJIWvlDwv98mu9sUX15k+qhLy8QQnR6RqHWpVc0DwrzEPJ5hJCEzt4/ZVQZrjrcw9HHkUcI6RfinlklNZylaFU6uTtcLle0aloe+vICIfEOAibITSdvTQ/zCDjwwgZCyO70irqWJgehnYOdnCHNhHjRcv4oTClqCA33nD42OmB0F/dPz361+vi298bOdbYTVMvVwW4m6BgFbFJZo2Tevg92TFt9/1bxR3E64ZFaeeP9WwVYKR7JlNy4+ZJHeOQO10+lKvnUHUvfemjmiIje9z84APNBYmcCGo3m3Xff/fDDD3v16nX58uX2C7ANTLinc86SBy+WNJ7Ir+u94fTxlwcxDNH/9b/T7vTKozca+1YnffjGyH3lQolMZVAO74Eor53T/myG1ibdoL2/EtK2HEKI7v+TdHqmzV0iTUrZY/EjlZqGBqXi1NyH2eld/vMz+1qrZyqba6f36ZtRoTpw7XR0YIROzzBEf702o2jZi9ckrSfy64Zuunhw9vORXo6EENOcEgGb89+cPwghldIaQkh5kwmuOBvRxTuMz+NnVF6/r2sB63RXJ8YyKvMrm2t/uHwIiR3YGCR2XOXk5EyfPt3I7Tvhnk4ljUr6+kCWRGQvGBLiUp2eFSQQbN53LMBRfzDffof+uqOAbCkU9nBn6tU8pxaNgNE1qXT7M2v7hrixRQ2L8FxzIr9eofYS2Z8prM+sksUHuu5ILdfqGQGP93t2dZtVlza3liscCCFnCht6Bf1ZTrWsXqlpndRtjJ2gYObuQom0xd/VmRYo4GkaWrQMQ3g8cjinvEkpH95pxPsn0vdln1swUnumsCHcg+doF3+5rCWhi8+QTp5XK6XZElXPQC9Zq1ap0fm52BOAv/J18RTZO/q5ePEIz9Huvp/QdXNwCXY3TXfE8Heh1KgIIQK+QKfXk45deXd1FBFCegR1vd+xAZgZEjtOpFJpnz59unXrlpaWdru+xUPcnXxd7NPKm3sHu8UFiF/ce7WxqfladhFPVpd37lueTq3q9vDMqiGEIYKGkrTz2/UOLvKH3rB3GyE9WfHe2F6z92Z4Ot/MlnoFua18uOuYbRfFDkKdntn8RPdoX5f9Vz36fnwm2M2pm7+4zfXWLt6i3yqFX315oVGp2Talh1p38+/sifxLLg7OHs6u03r3+igpafgXp4Pd3GmBXyU3dw9wHb01ubO3qFJa7i1yFzuKIj0ca6Qx/T85zueL19zktwAAIABJREFUXhwoSsqXv594Y/XxfIYwoe5Ok+P8CSFHcmtGRnkbucGOYOSJ27PtkScEPL6dwG7BA9NFDk4XS+66t/q7rRSfx6NjDxjCyBNG2MDIEzvTjhium8fjdbAv3wvFV+cNefJ2czHyhBEYecJqIbHjRKvVzp079z//+Y+dnbEbdd8d3WX18eu/zOwX6SVKemUQIYSQMYQQQv7TbtmPCCHV1dX+/v6fZ2fHxPhW/nsUIeSD8TF09suDw14eHGb4gW1P9rjdeu2F/B9nDjaccmXhiFatJqMiaXC4+/Sd6YQQf9emIDfF11PfZZfZ+FgXP7Fnq1Yz6stP4/zdVxwtqlN26h+aGeRW/fXUdwvry3emfXPkxTc8nG/+oqg0reeLrq1Navr88e5GNgJp1/36vn373n//faVSGRkZ+f3333t4GOsqhYWRJ4yw5pEn7tndVkqt11yrbjt2GUaeMMIGRp7oHRRd1lRdKa39/wReYGBgRz5Yr2gyMhcjTxiBkSesFp5q4cTT03PdunXGszpCyJQegb4uDvfcQbFpHc077+/q/cvMdT9Me++Hae/tm/FRTnVRUUPl7RZb8fCcUA9/drEIr+Ah4T2WHtpYp2gihEhVirf/+/lb/81+tHuA8U7s2pBIJPPmzfv999+zs7O7d+++evV9vKEe/jl2pv5X2arR6DR6xooewQYz4BGi0d3sJ6Uj9/s2qeSEkBB3//sZFIAFILEzky1T4g07KDaDgWEeVxaOaD9975Xj0/uMZd+K7J0mxz3w05Xjt1uMlmO42AcTXosLiHrux3fHbnvtuR+Xx/pHnJr7JO2or+MOHjyYkJBAz1S98MIL+/fvv9sKArTXySswNiDc0lGAZbB31jGM/qvkX4wv/OvVkzxCTHH1GMC64FLs/fXll1+mp6cTQhiGSU1N7dmzp1AolMvlxcXFRu66UCqVAQEBZWVlDg4OdIqjo6PhlQWJRNLS0sK+NTJXr9c7Ozsbzl330LyWlpbCwkL2s4aDKh6etkFR21RY27Rq0AuEkMrKysDAQNpT+eKRz0skEvrBcYH9xgX2u9uo5HI52/VRbm4ueyEpODi4srKyoKCAvUeHfvbEiRM//fQTIaSlpYUdqKOiooLeuXUPW8NwLq3UvX3WcK5er6eVuud9RAixt7cPCgpiq3/PUel0utraWo41IoTQSnGpkaOjo5+fH3vRXK/XFxYWNjY2qlSqwsLCuyq5srLSsKNj45/laYmXs/vNlRJ9WUWFrlVD/n83vZOTE5casd+Fe/us4VzD74IJv90W/C4oFApCCD2GOX4XDK8VdvCzDQ0NGs3NJuLYnE0v/fR+ekVunw3TtHr9T5NXCvnC9p/tJPa/4uiqUqpom3bLkulhw+PxTPhd4LKd7+q7cF+/3QT92FkxJHb3l0KhaGxsJIQwDCOXyxsbG4VCoUKhkMlkdPotqVSqESNGqNVqpfLm47Rt7ko2nGVkrkwmq6ys7NWr1z181nDuqVOnBg4cKBKJ7uGzhnPPnTvXt29f+lan07F3JdNxewwbLPrZ1tZWupUMEzuhUOjs7MyxRoSQQ4cOPfLIIxxrJJfLr169Srfwve0j6vr16wKBgL3T7p6jOnToUM+ePbnUSKlUNjc319TUdO7cmUuNeDxeYmLi8OHD6W8AIUSpVMqUigZls1KpvKuST5061aNHj9vNbfNZvV7P/tr02/AswzA7xiwjhFRWVpJ2tzrdw3amh829fZYll8svXbrEVuqet3N+fn5YWJjhD7AFvwtVVVUNDQ2RkZFcakQISUlJeeyxx9j7Wzr4WY1Go2V0hJBhkb39xN6NLU06vd6OxyeEqWyq9XFyb/9ZnU7HMIxer6cl3LLk8vJyoVDo7+/P8bvAHjb38Fn2tUKhSE5O7tq16z181nBuXl5eYGCgn58flxoRjDxhxZDY3V8LFy68h09VV1fPmTMnMjIyJibmlgsYf2iRnVtXV6fVatvcpt3Bzxpieyq/h88aznVzc+vWrRt9Gx8fn5SURF+XlZWFhoays1jjx49v33Y89dRTcrn8rtZ7y+m0yyuONfL39y8vL28f+d2WrFAoDDtouOeo9Ho9xxoRQqqrq3k8XvtK3W3J169fZyvF5/O7deumuHGkStF4tyU7Ozvfcgvf8rOiosRW7c0brewEAiFfSD9Lz5dw/y6wPaVx2c5BQUFeXl63q1THS5ZKpW0eEbDgd0EoFNrb23P/LpSUlNzDd+Gi/IZDrR0hZO7gJwghzvbO0laljhBCSGx0jK/LLR6dcXd3t7OzE4lExmO2t7fv3LntfSZ3u60yMjLu+bMshmFcXV07/l243dzGxsaoqKg2z4XcbVQ5OTk4Y2e1sGPAMiZOnJiUlETPo2zbtu3pp5+2dERA1Fr16UITDPd5O9N6jTHnHU18Hq9Vq9GadyC70gaJQq2883JwP/F4PD7h0Qdo6luMPfcKYHuQ2IFl+Pj4rFu3buTIkTExMSUlJUuWLLF0REDWnPj29f0fnSpOs3QgJlAprVWoW9U6TWNLsznXO/Gb+eO2vW7ONYJxp2+kt5mSW1M85fslWr3OIvEA3G+4FGsBly9ffvrpp42MfqPT6Xr27NnUZON/NJ955plnnnnG0lHAn2J8ww8Q0qSUhzq4WzqWe9eqURNCLpVk07dCvgl6Pew4IZ+v0rSac43Qnu7/rau7o7iTZ9sO7b44t/d6TUn/kG46RlfRXGP26ADuLyR2nJw+ffrw4cP0tVarraioeOutt+jbRYsWeXl53fJT0dHRb7/9tkajuV2xUqn0yy+/5N6nvIuLi0k6tIyKimJvgefCJN2vOzk5sQ9SWDwYGxt54sC104SQQE/fUC9TjjzRXpNS5uLgRJ9VvKO7rVSguw8pIT9lHDWcaLaRJxyFDmJHJ+PLYOQJI+5t5Ik2NDqtnkcIQ5hbdWlH++A8mH1G3qpsUEjbL8DCyBNGYOQJq4XEjpMLFy6sXbuWfSuRSNi3s2fPvl1i5+LiMnPmTCPFVldXL1q0iPutqY6OjibpZNwk2SExUffr9vb2bC8wFg/Gxkae8HHxvF5bQvte4R7MLQdXYAiz+fxPX57fF+Dmc+TFjR0p564qVatoFPIFhCHZ1cWG080z8kR5c7Vc3WJ/p1QAI08YYZIv1Mm5W8Zte61KWt9+cDlCSLRvRE51kYu9U6hHQN5fj5M2MPKEERh5wmohsePkrbfeYk/RWdzqY9ti/COeiH+InXK1Kv+9Y1/tfX6tkU8Zkre2vJ/47YnryZf+tYNOOXH94lcXf23VqoPd/daMfVUoEAz6dEaEVxCPx2MYxsnOYcGI6X1DYj89u8vZzvHFgY+avlZgRh7O4lv+EHK0J/2YtFVB/t+Z3Hcphwgh0b7hJl8RISS9PE/mpSAW6nWWPs2p1pngjDJwxOfxHYR2tzyJKOTz7YX2YgeRr4uH8cQO4O8IiZ0FMAxTXFxs+Eh/G3V1dXdb5rG85IYWqWFWdw8WHfx0QuzQ43nJN8NQNP0n8ds9z33gLXL/7OzurRf2zR36JCFk5/Q1znaOhJCU0mv/+nX9yblbXh/21DM7lg0M6949wAR/TMHG+Iu9IjyDCCE6oieEhHsF5EiKnIUmuLjfnrfI/fkBE9859Dn77TqRf3FKj1H3Y123I29tkbUqxA4mOH0FHUcfhmhW3ewLqVbeqNFraUsF8I+CxM4Czpw588ADDxhfhs/nNzQ0dLzMLRf2rRrzSvvpbPq4I/W/x/OSne0cCCFvj5oV6u7/QdL2tPJcXxeP3sExqWXZnz/+1rpH/mU4wubpgtQBYd29Re46nW5y9wde2vseTexY/UK7qXWaOkVTgKv3zP6PbEv+5bNHFxkJku21nyMjOXHHmSoYk5RDu2jmfosS92DUOs23Kb8ldO7PvVI0GK1eezD7jLfIgxCiVKsJIZXSOkLIpbJrd1VOx9fLI7wX+k3+OuXXUA//almDq4OY/L+rNu63N3QwGN6dzhma5LCxqmOY9nZpti3cnoez+8uDpwwKi6dvg9x8SholHIOxni+mCcsxVTDox85qYcdYwIgRIxobGxtuLy8vb/z48R2/K0giq69vaY71b3vXTnNzMx0H5mpV/u60I1unLPtyyrKHowf/58S3ySWZl0qzfpy+ZuNji5OLr9KWS2T/l5u+a+SNfi4ehJBTp0658Bxr5Y1tMqoT1y96Orv6ib0IIUM69bhYmmW8B4FDhw51sEZGSKVSqdTY/c4dZJJgNBrNsWPHuJeTmpoqkXD9ESKmqJSHk2uY2C8lJYV7MImJiSqVqlpazxDiaGdPCBnZue8j3YY3t8gIIWqtuoPl3K5SeqIft+31XzJPslO0eq1aqyGEDI3oIbJ3+uKJpQ5COzuBgBCS/z/2zjsuimuL42e2wwJLFxCQZgEb1qgYNMZYYn3J08QY042mGGOJRhMTzYvR+BKNxhJNVWOJJsauIaigqAiiggoo0tsusJSFXbbOvD9G1w2wIzIDuw/P9w8/MvfOmXOGucvZO/eeX1ZWVlYWy4gYnHkotFrtqVOn2NtJTk4uK+NgUycnQcnl8pQUDoogxsTEMGwsY4AHMLF7tPlHgRYIgiDZfQfMzMzkZCUZJ3dYrVbHxcWxt5OYmKhUKtnbQeUJuwVn7GyDqytTOQm9Xv9QX4ZKasr9XDwbTxKY87BrxbcGBHaXCMUA8Fhgj6/jfs1Q5PbtGE5XghjVdVBcdhOfyATc3VFGURRJmngEj87/Xtr1CY8gSIrylXlueuZD3r2kUCIQKdU1HZyt5qMMFV6aDyfTdcCdM5z4w5Ud9kFJhCIxX8hVUG/+8UWRqowAWH9vKjddngsAfB7Py8mtmXasBVWuUhbXlP15/fS/ej5BH1kbv7taW6vU1IwLH3rhvZ8bONPCMJrnzEPR/h4bDp3hyk6dqZ4kyRpt7bbEA6O7DbZsOpFx3mAy3Ckv1Jse8O0C7zADOGNnt2Bi1264m9X9di1m5+VjADBv2HRhU++DKKAIAiiK4t17v2BtfPo4eyQV3H1fVqau8nG+u8l3x/TPcOVKu+SvjAucVKwwk1mRrzPoBDx+qMfdjXhzhj33ecyP1Zpa9saTCjMAoKim3HwkQ5ENAK2x/6OZpBRl2OrSjzjfJ/7ZoLJJX+dQpa/eZDIFuHZo0FkqcVAbtEaK7OXXtai6HBCkfYGJnQ2orq7ev38/w3cm+lWjydTcwuh+Mq9S1d2Pp+ciRz0XOYr+/9mMuy/U+nbstv9arM6oFwtECbnX+nbsFurp//ftSyRFEgTx961L/KZyu2Gh/b5N+K28rgoADmecGxM+hMEHtb5ea9R7SGXN9BmxQ1wkUgchB6VkzBiMBoB//L0dHtJ/l/uJq5pb7I17OrnxCF5v3/s6nnweHwCMpjbVELNEqa4WCARGo3Huwf/++NyntnLjUeP0nWSNQdvVK8jyYH9Z6OTJkxceXifkN/wz5yxyVBLVJqCAstmjgiCtByZ2NiAvL2/btm0MiZ3BYAgODlar1c006OPs4e4oy1DkhndoujhWD9+w5/qMenP/SgeBSMAXfDTy9Q7O7jG3Ep/fudTX2SOiQ3C2skipqZm57z8URRlMxmd+Wegidvpl2vIFw2e8se8/GrW6e8ewucNfaLLaJ8353NSBgd3buMo/wi0SodhBxOVcrKNQUqfXmFpTrbVWpzH/30vqBgAk1fAb0arYn85kJW94Ym7ruXHXASc3V4lTRV315cIMCqgHbqFAOOFc9hUA6ODyEKUKZY5OyrqaXn5djmdcAICUoozX9q54Z+jUNwc901peIkhbgYmdDYiMjGRenK5QKKKioh6qpvyswc9uu3hg3eQFlgcHBvfc6ny3zN6L/Z5+sd/T5qY6nWZAQMR/xr7FI3jfntsb7NHRw1F24JWvGph9Ojzq6fCovLw8f39/ugJ76sK9ja9OAfVz0uGlI19jdhKVJ6xhJ8oTAKDR15+4c2l85yj2zpzT3NYY6gEgwNWrQRMJZGG1wkiSgmYs07EWVIYil6TIy0Xp5iOOQsnT4VGvDpxoPlKv1y48vM5J7FCjVWtEHGiDNv8Oq7R1Molzk02oPMFAC5QnunUIhutnGii53QuKTC/LMZLGhjInJCEWiL2d3fUmw1u/fxHmEQgASfk3GiR2ZuUJI2nk8Xi8lm43ROUJpC3BxY92ysNuxRrVdZC7o8vvaf/YasegPOEockhX5D6/Y8mre5dnVRS+MmACg/GgoCDmj9pvz/02ovOABxaxQ+UJa/j6+nJStZ99UFX1tXqTvthYxcbI6aykkd+9dST7PElRAFB9r7QYjcFkJAjQGfWTfnn/879/MD5oPq/JoEggj944K+DxXST379uR9HNXi//xktdAmkwUaSBJAEgu4+D9b/Pv8OXCdGtNHCpPODg8QL6sOXClPMGJgEpISEgLVuULePxAVx/LI3RQmWWFqcVZ+1P/8cGoqFU+22tE4txfKmqrAOBWRWHMrUt8gufSqPSgp6cnvdFt8PpXRn/37sN61cAZlnCoPMGJRCQqT9gtmNi1H5aNmtn8AsU8glg68rV9L3/58/PLN/zrA5mE1Re49x5/HmUn7A25qvxayUOnMvQaO5Ki3j/09QMrkhhJ8lTW/bnnny8d6v3VtL1X/lp0ZD29NNNDKhvRecCmZ/6hzqIzGYEgAKC4qmJ/auyVRhsODt2MI4HMLMtluPSG+D3ZlUVigcjydaeno+vTEUMtu9FzSDqjngDiYl4qg8EjN+Of3DJb9c8ctAWQ95ZYbDn/B0tTSPMJdPNdNmpm4+P0ThqexX6aWp26RqvWGvUA8K9eI54OjxISPHldOUlRl4ub3vtSb9DqTUZdswv0IIhtwVexNuDWrVvz5s1jqNWk1+sBoPlr7BCkMWN/mEuS5Ll3f7Sc02omn/31PQB1IS9teFh/hm6rT/20PzV2XvSLJGV67bFJB2/EA1A/JR0yT8IJePx1kxY0OGvPiytV2rrRW9/VGLQAoNbXW7Z+dGLz0Ztnf0g8XFBVIuDxz737o2NTy/7oc/kW7/7yqkoU6koxX2jZLf7t74dvftNEmjylrgKCaQHo/rTTFerqzLL8gYHdGbo9EDFfSAvuZVUUao16iUDExhrCjEavHbzhFReJlK6A3Rh6weXftxKfi7yrPkJ/FRkSfLeOsZAvkNcqAUAiFDtLpCqtxkXi2MgIBf9czdkGlKmrbpbceaLzgLa8KNI+wMSuTaGMptgBT4k6+UdGRpIkGXY+1aWsEgAcVGqtsyNFEDonx5tPPaaprXs1Q171yVdw/P6CNpNWd2vNt6VH/yYEfCB4PmOe6LboXd6DXkq2uMg47apzl9BBv31/V0JArTkRNnDsnSSB9O4H37V5y6qSrwJAXXaeNCiA4PMd/P0G7d3GuTMNfUPliWY44yAUq3X1OpMeoFmJXa1eTc9J8Ii7f8k2nt/XPzDCSdTw7xxNRnnuH2mnAWD9uV0kRT3X96nSWiUAKOruKqZIhZLlo2c3ea6LxOn4zPXDN80CAn64dPAJi/SxvK4SAEpVZQBgJE1ZFQU9OoQ2uMNn7iTvTz0l4PEFfIF5hmz16V+MJqP6nwutZA53p6KdxdIcZTFD+HfKCwDgTPZl5sSO4dddp6v/5OQWPo8/o//TO5KPAVA35dn9/MMf1k7zsatn2CbKE3RFdLVeZyAbVi2h7Xg7u8trKwqrSn+79ncnN59BnXpq9Fo+jz/g3m/5eskdABDzRcFufullOR8cXTe979ghQZH06k96YOZXlgCAk7jlb70f9g5XaWpGb32XJE27X/yiu8/9xR529dhgHTu7BX8xbYo85gzf0VGdlv7RzNmrV69+49yJqbcuTb11ieDzJsQfmXrr0oyUM6tXr34rajjf0VGXfludW2A+N2XWAnVuYXTs7yMuHI8+uVeTV5j6wQrmyymVysTERDauViZfVecWxMXFNTl9GLnuP08kHH0i4SjB5w058MsTCUcZsjpoVH79jz/+6NevX0RExIQJE6qqmruoC5UnGLAMykUkhWZoW9Ecvhk/dMMb8lqlRq8F4BEE8AleVnnBdwlNv0/89y+Lpm1fQgvQ0anVkPWv6Yw6IAj6Nau31HWS76C+Pla3CLg5yiI7dhbxBRmKHMvjCpWSBzyDySjiCwGIC7mpjX9TV4ozSYo0kSYKoEZbO2P3JwDgInL0dnafH/1Cg853vwYYTJWamgv51635E+rhL+QLy2ofUJH/2LFjt8rzmlwXWFhVSlEAFMyLnk4fcRI3nRNzpTxxOP7klrP72Nsx3+EzWSmRX0/bnfJXC4xwpTxx+OTRaTs+zKsqYejz5/XTfb+e9s6BL7+O2wkAJGkMkPk06EMH9eNzn4oFInld5RexP87av3LyTwtisy4BgHkbxNjwoQCwZ8bKrVM/dhQ75FQUzDmwZkP8LrqVVp5YenwTAYTeqH/gelBrPPDT5ovTP5+8lQgA+1Nje381bfjm2SRpgn8q76HyBNJMMLFrU/K3/9Zp+rM+Y0bk/7qfoVv5/iNZHWROQweau6lu3lJevBy5fqXQxRkABM5OkRu+6PH5EsuzDDWqpJfeuTjl9XNjnktd8AllNFEUpf0r/sKzr156YXbi82/SaeL1pSvPjX0+5c0FNz9dkzhtVnNcbY3y63K5/N133z169Gh6enrPnj3/85//NNNIiz0prJZfzEut02uGbHjtp0uH22UteHNQWqNeqa4GgNHb3tGT+gfmK3uv/g1AdXDxchZL6TodFAABRJhPwJ3ywgadMxV5WRUFFMCATj1e6DfKz9XT3LRu8vwJ4Y8DwNxh0yNlwcxBbX/hP090GWAkydjbiQO+eXnw+ldGfjc7v6rU39Wng7P7tqnLBALB94kHdSZdgxMDXH34BJ8iCLVOTVFAr8bzcfHs5h3UxGUIiiCIcaFDKYAdyYeb9ERv1KcrcgDgVlm+NW9JID86vvFKzZ2p2z+c+NP7DVpNFPn8rx8BWC8IZAH7X7eRNP565dhnV3Z+l3TgSjHbTSHmx+ZSwQ2Kor4+u+PwzbMM/QuqSnZcaZiptCCoC3nXVNq6r+J2Dvzmpf7fvJhWnH305rkNOUfTy/Ke37F0yo7Fs//4wrL/7pSTwza90fur55f/tc1EUQm5V49lJAABFEBNfcPFkfcHuMUCu4JquYgv7OwVYD4S6NZBJBBKRQ4uEkdXiVOZugYA0hR3twUUquSvHv8it7KEAkprNDy55c16o67e2PCBpDGS5JGbZwHgWPrZoRvfKK+7L/bN8Gnz/p//fXzjG79d+WvpsW9TS25//vcPABRdWIoA+CZ+V++vns9TFt8ovWOiTHb1aYMzdnYLvoptO9Q5+crElL6b17hG9rj85vxuH77HEwqb7KZKSbvd03/yuJGFK9fT3WpuZrp06yx0ub/Fwfw+1Iz85GmBk3Tgjk0AULDnT52yUn07S3fmwoi4wwSfV3o89vrSlaGzX1FevDzs798JAf/KO4t5gqYn5Bu4yh/IwV79Bhw5cmTkyJH0NrrXXntt5MiRa9eubea59UZt44Maozat+PagTr3UxnqpoImXJhN/nEdSlIvYQa2vX39uNw9gxVe/EQTlKHbUGvR/v/Wd270dJMl5N5b8tclRKPl6/LzO3oFmC3JVxVdndgzo1GNMt8ESgchoMhIUYSSN5epKdwcZv6kafvMOrj19J0kqdvxjxpe+96p+1GrVztbXvd2qyJu174sAZ6+M8nyKgFlD/vXvniPLNdXTdy6lKBgS3BuAl6HIGuDf4/GQvj06BDs4ir0cPIwkmVKQfqQ8aecv52vqa0wUpSeNQBBGk2nA2pcAQMDj0++tCAIoCgi4++aXpMiXB0y4VZ4LAH39uqweP2fk5tnlmmoCgALq0+PfWfpGELxvJy2Yf+wbAPjtxS+6+YQAwOInXjuReeHQjbPB7j4jQgdEB/cL8vAd0y3qVOHfjL9GAIAlT7z6V8bFBYe/AQA9gNqgJQCiQ/p8MGIGAJiMBgrg8+wDpeeMjwX3IIzUwZvnYrMu6IwmABAKBFumLP0+4cBV+e0+616gTJSbQxPlRS7O3Q4A+Xdyf0w7lJh/Y/CGlzUGA9BzawAEgIPQQWPQAlAdnFwLqxXP/PzBD8995O7oWl5XdTD9THGVMjE3zcnB0dvJ7XzuNdpmaU35iO/echaJ/917lEQo2nxuf41eBUC5iKVn3vmeIAgRn683mV789eMfpn7EI3g9/boAQE193aXCG64S555eIcnVOaXJR6dGjlAbdeml2VFBkQKeQGOodxQ6lNdVejm577568nTW5beH/PtK8a1/9Rzu4SgDACNJVmtUBAGjvnvbeK+47qt7Pt0weUGVrvZc7vWzdy6vHPMWEODn6iUhJIGevrX1KorgldVWLji8tk6n/fn5FWGeHekTb5fnC3l8E0lqSR0ArIjdduDaaQLAaDJ9cmLLj5cOHHrtm7tPrK7u23P7xoYP2XPlrwt519V6DUmR6+P3dHTz0ei0JlLfx7+rXlNvNFIdQgMJAayP3zulzxN9fcP/zkrclfIXj+C9PHCMt9Szm3fI+bzUXZePz4l+7pXdnzZILmbs+YgCIAAIIOoN2ttl+QAQ+dU0kVBsNOpM/8xFxAKx1qgzmIz+rt5F1WVbn1tq7Rk7+vq6J797x2gyAICJNCXmpXWU3deiGNNtyJhud0uv9/cPP3yznAAitfhWv7XTGytfV9fXDfrmZQLg2T6jiipLkwpuABAjuwy8UZqtqFWK+IJ6o/7jE98BkAAwdtuc96OnkRQxsecwACipKRMKRARB/uvnD9wcnE0kVavT1NTfl2AxmUwv7f6EAuATPJIie/t27ertvy/tFFAw6ecFAODn4vVawAhz/0xFbpCHr0QgAYCUwpvBHh0dRBIHgaRWW2egjFVqVUe3Dgaj0VksNZEmjV7rLJGaSBOfx9eS+pK6ck9PT/pET6lbJ3e/Uk0R0FPJAAAgAElEQVT5hrjfJnSP7h8Qcfhm7Pr4/bU6DUVRfB6fpEigKILH7+zhv+/lL63dZ8R+ILiSUESY0Wq1CwLDpQbjf6uLAGCzZ9jeuvKz2hq69aBPxBvlWRUmAwC85uzjwRdUPjt60KBBHTbtgvFPwsDeTjezhOcuDz22m+6flJSkUCjMxt3d3aOiojSFxYnPzZT1CK8L9FGFBICz1HAynh+bIPL3lclklIk0aer9Jo/NS0qp//cYAICLV+Bymvun86Oi7hctu2t5/3GoroGZ09zd3fUfrtaPejxq3tsSgncibCB8uwLEIsvrAsBR/14jk/9OK8xv7JWl5UuXLpm36x88eNDLy2v16tX0zXF2dv7jjz/Mi8zoc1etWrV0aROf1xIfWZcFY8w/Eg+eJrnfhQIQ8HmmB4gTNGmSzoia6EEBN4VoCSAYSkA35/y73tA/ERDq5HenlumVlqXnEoHw352iI52Cvr51oFJfK+KLjBRZb9Td6/UPxwjgre03s4O3t/lIZWWlVns/4RaJREaj0cvLi17N07iV/rsCAP9J3JFZXSgRCAmK0hj1YU6+74ZP9Pb0BoAUxa21V5vQaCEAenuHdXLq8FzXEXuuxxwpukQf7+rs/063CWbLltc1GAxb807macqs3QoewevvEZpUkcVwu+7+gqw/cJ889nI390D6uh9f3V6trzM7bO332oyn19ytQV8CgGrm6c2/opgv0JtMJNATYc1zsKUDgEfwSIp04IvCXTqp9LV31PRqBOIp/z4X5Rl1xvoG/WUCqYDH04Kxh0fwUwH9P0/aIeDxXuo0kkcQ3Zz94Z/PVXFxsYODA/3b/zX/tFxbMyP0yW/SDxgo0xMBfWb2GG/tmZwX/61CU20+7iSUvNV5vFJf2zeg2/tnNxpN5P0bQgFBNLw7jW/GvSON7yQBQBF3v2nd/Vs8pmO/kZ6Rll7958buinpV40+G+xdqZPgBvzPCMgJrvj0YAfCOz1jXoUNDxTbE5mBi10aQOt3xHtGEzBkkYoqijJXVQh8v8Udz6Nb6l+dJ1n1KuLuCwVA/51PKWVpRVxsQEEDUqh0D/IYc+KX2dva5Mc+NTP5b5HF385epXluZdEWdW5CzdTsAhC9b4Pv0SMpoqrqaVnH2Yv6uPwbv+yHv2N/F5xNH7/vJ7MadjT+q8wp7f7UcAAr2/Ck//vfAnZsbuxoTOULi7SlwkgKAtqxC5+Qw7MguOrGz3Dxhhk7sJL4PGOEHDx6cPHky/f8VK1ZoNJovv/wSADQajbu7u+WHLE15eXlqasMSFZmZmR07dvzXv9hWV7F0psXo9frY2Ninn376wV0ZSUpK8vf39/PzY2mHk6AUCkVOTs7gwYMf3JWRv/76Kzo6mn2hNU6CyszMBAD21aQ5caa+vv7s2bOjR49maefixYshISHs/7JyElRJSUlRUdHAgQNZ2jl+/PjIkSNFIra7iTkJ6ubNmyKRqHPnzg/u2vrO1NXVJSYmjhw5kqWdhISEbt26WX7/aQEZGRlLlizZunUrJnZ2CL6KbSOKD5107hQwLPZ3+kdjbd3f/UYOCgp1CgsGgKME8fjjj0t8OxTuO5QTFNht16aoqKgjm3Z29g/4u9/Iuju5zl1CvZ8YeuWdRX02rBJ7expqVKnzPxG6ufb+annQK8/TNuUnT/MdHbyiB7sP6KNKv12Tlt7h8ceKtu3QV1WL3FyViZdrM7JcIroU/X6EMpoIPk8RE0efWJuZ5RgUyJeIza46dPS1dPWvPiNMxXLwZ5t2WJZfDwgIOH36NP3/wsLCgICAxv29vLwaf4r1798flSeswUlQLi4ugYGBD+73ILp06cL+zzNwFJS3xeQiGzhxBpUnGGiB8kSTcBKUWXnCHpxB5QmkmeCMXRuRMO6FTi8/FzB1kvnIzU9WA0D3zz4Eixkvupto2CAfH5/09PTw8HBzN1Kvv732u+I/jwEFPLHI/98Tw959neDfX76qzitMW/gpZSIpinLo6Nv7q+V8B0n+jn0Fu/8QOEkpE9nzy2VOYcFpHyyvvnZD5OHu3DVUk1c4cOfmvyKGDt7/o0v3rsyudl383omwgdKQIPOn95A/t4u9PKDZM3aWlJeX9+7d+/Lly35+fgsXLnR0dPzss89aeHMRBEGQNiQjIyMiIkIul+OMnR2CiZ0NMBgM58+fZ5h2qqqqmjp1Kp3YtZ4bpUdjCn87OHDn5sxV64Nfny72ZjUz3wJ279792WefEQTRo0ePX375hRNNLQRBEKS1wcTOnsFXsTbg0qVLI0eONJkeIEleWVnJ3IErXCK6tn1WBwAvvPDCCy80rDqGIAiCIEiLwTo0NmDo0KFGo5Gyjlwu5/F47u7u7K/FkD76jh9F75zwmzTGWp8HGuHKmeZDURQnJejsKiiSJLmS02BvhCs7duUMSZLt77GxK2dwYLa2M1zZ4coZrGNnt+Avxk7hpKg3G+UJS6wpTzwsnIg9ZGdn0zsc7cEZe1aeaDFlZWXJycns7Zw6darxTucWwElQWVlZWVlMpUza0hmulCeSk5PLyqwWcGk+nATFlfJETEwMg4528+EkKFp5wk6cQeUJpJngq1g7hZMvQ3ari4DOtJIdDIrBGfZGAO9w6zvTLoOyK2dQeaLdg4mdDZDL5d9++y3DfDg9PabX69vQKQRBEARB/u/BxM4GVFdXX7lyhWFXrF6vd3Nzw8QOQRAEQZCHAhM7G9CtW7cTJ04wdFAoFLNmzeKkhiSCIAiCII8OmNjZKZwU9XZycjJrs7IhLCxMIpGwt8NJ+XUvLy9UnrAGKk9YA5UnGEDlCWug8gQDqDxht2CB4jYi7YMVst4RnV6c0pzOCoXCrDwBAFUpqWmLPxsW+wfzWeaCw9Y6VKfevPnJl1GHdjTTYPPJ+OIbgdSx89w3uTKIIAiC2C1YoNiewRm7tqDkyF+6ikpzVpeamjpz5kyGfVL0bv/a2lpu3XDt3T3q0A5ubdKEL5l7dsxzno8PcuvbqzXsIwiCIAjSHDCxawtur/0u8pvPAcBQo7o6Z4lWVbvI4Fjv6Vo6YhDFIzyuZrjcygOK0nq5lT7xGPB5brEXaz3qaj7+b6K7a99NayxN5W3/reTwSb5YTJFkz1UfS4MDry9dWX31umMnf5GrzLJnZfLVGx+v8owaWF9Uqikq6blyKUWSjSfqGhjUlVfcWLY6+q99AFB88ETJoROhb7/awI40pNPVOUtM9VpjbZ1L9669vvyUEPDD3n09a/22gds3tv7tRBAEQRCkaTCxa3XqS+S68grXXhEAID95WuAkjd6xCQAK9vzpPWKoVl527cTF6KS/eULhtfc/Hhs13DWy+w2DYNCZo9fWHyV/O5q3fa/XsCG0qZrr6YW/HRx6ZBfB55Uej72+dGXo7FcqEy9Hx/xOCPhX3/3Q8roEj6dKv9Xvx2+k/n6lx2NvfvplxKcfNPCtscEu82Y16EPbGfjLtxLfDrSdwOn/FjhJB96LQqeslHTw8n5i6LW5H5EGI09o9aEymUx8Pp/l/aSLMLEvocSJM1zZIUmSIAj2S5TsKii7coaeIG9nj41dOYMDs7Wd4coOV85gHTu7BX8xbKmoqFiwYEF4eLijo2NQUNCzzz57/fp1yw71hcWO/n5AEADgMWRA9bUbKW8uKNj1u8+o4ZIOXpVJV936R/KEQgCI/OZzr2GDBc5OQBBru/STf7a2IuGSruK+Yqwy8Yq2VHFxymsXnnklZ+sOfUVlzY0Mt36RhIAPAJ7Rgxv4JvH3u5afCwCukT1UGU2U3W9ssMkYCS8Pk4uT2U7jKABA4CTlOzjoK5gKmqPyhDVQeYIBVJ6wBipPMIDKE9ZA5Yl2D87YsUKpVD722GO5ubnjx4+fMmVKXl7e3r17jx49evr06aioqPv97n3hcwzo+MTZI1VX04pizmQ8+UzPn77RaNQ6bX1VVZW5b/nRvytOnTsd5jPik/mCv85qyyoszBAeQwb03fSl+cidjT/e3/7SaNEedU/okDKZoKkvnY0NViZfvX/6vRLKDeyYo6g4ezH+qX8P3veDU1gwAMCDNuLYVfl1u3IGg2IAlSda2w4G9Ug5g8oT7R5M7FixfPnynJycTZs2vf322/SRZ5555l//+tfKlSuPHz9OH3EI6FhfVEL/X37yNN/RIZ3SD/94wRLXgFVRw4pN+sWu/sM2f6WlyHdc/K7p6zz5wiCBxDS8v7KgkDwWK+vd3Xw590F9s9Zv1VdVi9xclYmXazOyXCK6FP1+hDKaCD5P8Xd8A/d0xaXi0jIAqExMkfVooipHY4OeQx/TV1YDRQFBVCbe/f5NlSvr8wqdeoTTdugovKIHuw/oo0q/XZOW7hQWbKxTm7RakacHpzcYQRAEQZCHABM7VgiFwtGjR7/55v0yH5MmTXJ0dExPTzcfcfDzEXl61FxPl/WMcO7WOW3hp0IT+dfIZ3me7s/MeZUQi9SH/joQm0AQhDC00yvvvELV1ik+/aoko1j/074eH85JXfCpyO3urghZj/CuH7x7adosgZOUMpE9v1zmFBbsdjz27JipDr4dnLuGNXjf6hAUWH/or4v7jxuqVb2/Xk7qG77gaMJg5xCPQf0Sn39TGhwolDlry8oBgOfbIf/7nXdKy2g7AmfntIWfZq3bSlGUQ0dfn7FPAkDZmQTPoY8xLLBDEARBEKS1wT/DrFi7dm2DI3q93mAweHp6Wh7sMn/27bVbB/y8XhoUMPj3nxpa6dcPPvvHAUXP7rNmzVq1fIF3ePhT184AQPhH8+imTi9N7fTSVMvOvb9aYdU/AV86e8bgoUPNB+gtsW79epv3xjY22OfbVZY/VqWkUnxet5VLLWtaNoyCou5s/LHnqo+teoIgCIIgSOuD78g5ZuvWrQaDYfr06ZYH/SaMFnu65/+6v/l2OCnqzefzOVGeEIvFzMoTmas3+D498oFF7LhSnuCkJKa9KU/IZLIH93sQqDxhDW9vb07EJ1B5whoymSwgIIC9HXtTnvDy8rITZ1B5AmkmqDzBJfHx8aNGjerXr19cXBz9J+3y5cu5ubl0640bN7p3704QhE6ny8/PZ/hkr6mpeeutt5KSksxJg0AgEAqF5g5Go9Fy4xi2Yiu2Yiu2Ymubtd65cweVJ+wWfBXLGXv27Hn11Vd79ux55MgR80TF+vXrL1y4AAAURZWUlPj4+PD5fJ1OV11dzTAeTCZTnz59UlJSioqK6CPu7u6W22yvXr1qWR0DW7EVW7EVW7G1zVrd3d0BsVdwxo4DKIpavnz5Z599Nm7cuL1797Kf5W6gFYsgCIIg9gNqxdozuMaOLRRFvfHGG5999tm8efMOHz7MydoF4K5EkOleLTqbG+HKDkVRnBSFsqugyHuVAu3BGa7s2JUzJEm2v8fGrpzBgdnaznBlhytnsI6d3YK/GLbMmzfvp59++u9//7t27VoOH3ROinorlcrExET2duLi4tRqNXs7qDxhDVSeYACVJ6yByhMMoPKENVB5ot2Da+xYceDAgfXr18+fP3/hwoXNPys3N/fLL79k+CJYX1/P4/Hq6+tZumdvlcrtqvy6XTmDQTGAyhOtbQeDeqScQeWJdg8mdqxYtGgRABiNxg8//LBB0+LFi93c3Jo8i1aVfuDQYq88jSAIgiDIIwUmdqygZ+k3bNjQuGn27NnWErugoKAtW7YwmFUoFLNmzWIuHYcgCIIgCNIATOxYgXuKEQRBEASxHzCxs1M4Kert5OTEifJEWFgYJ9OHXClPGI1GO3GGQ+UJ+5EQaK/KE+yNACpPWEcmk3GyesTelCfsxxlUnkCaCdaxswFJSUnjx49nyE5Ikqypqblw4cLgwYPb0jEEQRAEeSBYx86ewRk7G9CjR48tW7Yw7JOqqamZOXMmJ+KhCIIgCII8OmBiZwMcHR2fffZZhg4KhQJwVyyCIAiCIA8J1qGxU1B5whpY4L61neHKjl05g8oTrW0HB2ZrO8OVHVSeaPfgL8YGkCSZmpqaYp20tDRUnrAGKk8wgMoT1kDlCQZQecIaqDzBACpP2C34KtYGnDt3bvjw4cx9Jk2aVFlZyfJC9lap3K7Kr9uVMxgUA6g80dp2MKhHyhlUnmj3YGJnA4YNG1ZTU8MwH15eXr5o0SJ3d/e29ApBEARBkP93MLGzDczVp/R6fZt5giAIgiBIuwGnUhEEQRAEQdoJOGNnA5RK5c8//8zwKrauru7GjRvs9y6h8kRrO4PKEwyg8oQ1UHmCAVSesAYqTyDNBJUnbEBaWtrMmTMZ8jaDwZCWlnbp0qWBAwe2pWMIgiAI8kBQecKewRk7G9CrV69Lly4xdFAoFD4+Ps7Ozm3mEoIgCIIg7QBcY4cgCIIgCNJOwMTOTkHlCWtggfvWdoYrO3blDCpPtLYdHJit7QxXdlB5ot2Dr2LbCL1eP23aNJVKBQBqtbqkpKRz587MndPT0zUaDX1EKpVaLtLPysqiTTG3qtVquVzeq1evFpxr2RoXFzdo0CCpVNqCcy1bz50717t3bzYRAUBZWZlAIHjqqafYRAQAx44dmzhxIsuIqqqqMjIy6JXRLY4IAEpLS/v16+fr68smIjqo8PBwNhGpVKra2try8vKQkBA2EUml0oKCgujoaHrbDZv7vHv37oiICDYRwT395ZCQEJZjgX5sWEZ08+bNK1eumINq8X3OyckJDAx87LHH2EQEHI2FwsLCqqoqercWmyfn1q1bU6ZMEQqFbCICjsZCaWkpn8/39vZmORZu375NPzZsIlKr1cePHw8JCWETEQBkZ2f7+Ph4e3uziQhQecKOwcSujRAIBAMGDKiurgYAtVrt4eHRvXt3a53VarVCofD09DT/pReLxZYdPD096aHF3FpdXV1XV+fl5dWCcy1bzZXKW3CuZatEImEZEQA0lhtqmVf07ALLiEQiUX5+Ph1UiyMCAJVKZTkx0GKvSJJkGZFUKhWLxVqt1tfXl01EYrE4Pz/fHBQbr0QikfmxaVlEAKDT6QCA/VgwT0qxicjd3d3R0ZH9WCgvL3d1dWUZEXA0FjQajclkYj8WLB8bNl5xMhbq6+sFAgH7sWCpgthiryiKcnBwYD8W6D8uLMdCWVkZztjZLZjYtRE8Hu/DDz9sZmeFQjFr1ixvb28/P78mO7i5ubm5uVk73dwqEokqKys9PDxacG5rtDo6OrKMCADUanWDAs42jEgqlUql0iaDeijLRUVFbebzA1v5fH5tbW3joGzllVgstvbYNN8yPd9gJ2PB1dXVycmJ/VjIz89vkNjZ8Mmpr6/X6XTsx0KDykq2HQtVVVUikch+xoJEImE/FnJycry9vVmOBU5EipFWAjNuBEEQBEGQdgImdgiCIAiCIO0EfBVrAxISEoYNG8a8gyw0NLSyspLlhVB5orWdQeUJBlB5whqoPMEAKk9YA5UnkGaCyhM2wGg0nj9/vvEmADNVVVVTp05NT08PDw9vS8cQBEEQ5IGg8oQ9gzN2NkAgEAwbNoyhA12aAUEQBEEQ5KHANXYIgiAIgiDtBJyxswE6nS4hIYHhJXhVVRWHyhN8Pt8ejHBlhy6qx/7+2FVQJEkSBMF+iZJdBWVXztBLWtvZY2NXzuDAbG1nuLLDlTNYx85uwV+MDUhJSZnKyMyZM8eNG1dVVcXyQkqlMjExkb3DcXFxarWavZ1jx46xN5KdnW1Z7dO2zhgMhpiYGPZ2UlJS5HI5ezucBHXu3LktW7Y02bRv37709PRm2jl16pRWq2XvDydBZWVlZWVl2YkzWq321KlT7O0kJydzUk6Mk6DkcnlKSgp7OzExMQzrj5sPJ0FlZmZmZ2fbiTNqtTouLo69ncTERKVSyd4OKk/YLThjZwOGDBnCPK7oAsUM5SKbiVkxwk7scCIl2S6dsaugKioqiouLm2xKT0/39va2FPhiwK6C4mqXGD42re1MuwzKrpzhyg7O2Nkt+ItBEARBEARpJ2BihyAIgiAI0k7AV7E2oLi4eOPGjQyT4fSCNlq2HEEQBEEQpJlgYmcD1Gp1Tk4OQ2Kn1WrT09Pr6+tnzZq1YcMGsVjcsgtlZ2cnJiYOHTqUoY9erz906NC1a9cAgKKoyMjIiRMnSiSS7du306uGFQqFm5tbfHy8p6fn7Nmz33vvvRa7hMoT1rAr5QmJRMKJYgQqT1gDlScYQOUJa6DyBNJMMLGzAV26dPntt98YOigUCh8fH6lUyvJCQqHwgVJgmzdvdnJyWrZsmUQi0el0e/fu3bBhwwcffPDyyy/THd56663FixfTOzlYTiJy8qnk6urK3ghw5AyPxwsJCWFvx9fXl70R4CgooVAoFArZ2wkODmZvBDgKyt3dnb0R4MgZPp/Pyc3p2LEjeyPAUVBSqZT9RxYAcDKggKOgPD092RsBjpwRCAScSET6+/uzNwIAnOwXRloDTOz+z4iPj798+bJQKKQoatq0ad7e3vv378/OzhYIBAKB4PXXX3d2dt6zZ09eXp6np2fjz9kGp9fV1RUVFX3xxRf0zIpYLJ4+ffqyZcsyMzObqWam0Wh+/vlnvV5fX1/v7+//4osv4lYpBEEQBLEVmNixJScnZ+XKlfHx8cXFxb6+vgMHDvz44485mXhvTEFBwYULFxYvXszj8a5evbp379433nhDKpUuXLhQIBDs378/Pj4+NDQ0Kyvr448/5vF4P/30E/PpvXr1CggIsHxfJhAIgoODCwoKmpnYXbt2TSwWv/POOwBw/vz52tpamUzGbdQIgiAIgjQTTOxYkZKSMnz4cL1eP3Xq1NDQ0Dt37uzbt+/gwYNnzpwZPHiwtbOuXr06bdo0hiKcJpOJx+PV1tY2OJ6VlVVdXb1u3ToAIElSp9M5ODgAwLZt24RCoUKhCA0NLSgoCAkJoafNwsPDL1++zHA6j8drcqlfg7rkJpPJmqtdu3Y9ceLEtm3bwsPDIyMjnZ2drfUELHBvHXsrcM9JmSt7u8OAyhOtaQcHZms7w5UdVJ5o92Bix4pFixbR1cCjo6PpI88+++wzzzyzatWqw4cPWzsrJCRk4cKFDCUrVSrV2bNnHR0dGxwnCKJLly6vv/66+UhSUlJqaurChQuFQuHx48dramos+9fW1lZXVzOcnpOTc/DgQZ1OZ94MQZJkXl7e8OHDLe1cvHhxxIgRTS6g8fDwWLFiRW5ubkZGxsqVK99//30fHx9rcR07dmzixInWWptJdna2Xq9vZo1cBjhxxmAwxMbGjh07lqWdlJQUf39/9ivtOAmqrq5Oo9GwNAIAp06dio6OfuAqzwfCSVC07ETXrl3twRmtVnv27NlRo0axtJOcnBwSEsJ+XwgnQcnl8qKiogEDBrC0ExMTM3LkSParPDkJKjMzUyQSsV8hx4kzarU6MTHxySefZGknMTGxW7duHh4eLO2g8oTdgokdKwYPHvzYY4+ZszoAmDhxolAozM3NZThLJpO98cYbDB0UCkVCQkLjL1WdO3c+fvy4Wq2WSqVZWVnFxcVGo9HPz08oFKrV6itXrgQFBXXs2DExMZH+opmZmWk59dL49GHDhnXs2HHnzp0zZswQi8Umk2n//v2enp4NtuwxVCpPTU0ViUTh4eGhoaFFRUUFBQUMiZ1dlV+3K2fsLSj2RqCdBoWPTWs70y6DsitnUHmi3YOJHSs+//zzBkcKCwsNBkNoaChXl1i5cqX5/wsXLpw4ceL69eslEglJktOnT5dKpZs2bVq7dq1UKp08efKOHTu6d+8eEhKycuVKNzc3Dw8Pyxm7gICABqcTBPH222///vvvn376qUQiMZlM3bt3f/fdd5vv0ssvv3zo0CFaCdHd3T0yMpKrwBEEQRAEeVgwseMMlUqVkpKyYMECJyenjz76iD5YW1tbXl5O/1+j0Zjfrlr+vzEVFRUAIBQKt27d2qApOjracoIQAJYsWWL+/5o1awCgT58+ZjuZmZnMpzs4OMyYMQMAdu7cKRAInn/++QZX3LJlS2xsLP1/sVjc2KX58+dbCwRBEARBkLYEEztucHV1pde3TZs2bd++feY1GRMnToyLi2uBwddffz0hIeHOnTv0j+7u7lFRUebWpKQkhUJh/tFaq1arra6uJgiiOec+88wzW7duXbJkSWBgoLnQUcuua621sLDwyJEjLC3X1NQ4Ojp2796dK6/YtJaWlhYVFdFBsbGsVqsti0vZMCKFQlFeXu7k5HTkyJHGrW5ubgUFBc2MFyxg41VFRYX5sWlxvPTUtVKpbMs7aa31ypUrOTk57MeCQqFQqVSjR4+2eURJSUm5ubl1dXV0HzaWy8rKwALbjoWqqioej5eZmcnSMlcRlZaWsh8Lcrm8tLTUz8+PTURcFYZEWgOCq6UnjzhLliwpLy+/devW+fPno6Kitm/fTpfZrKmpUSqVdB+VSkWXiacoqra2lqFkfEVFxQsvvHDkyJFm1hyxhlarraioYF+OMi8vz9/fn30F9jt37rBfhlxdXW00GtkXDuXEGXqvCfuSqqWlpS4uLuzru3ISVH19fWVlJfvit7m5uYGBgez333ESVGVlJXBRppgTZ0wmU0FBAfsaxcXFxe7u7vTWeDZwEpRarVapVOw3AOXk5AQFBbFfv8VJUBUVFQKBgH1RdE6cMRqNRUVF7GsUFxUVeXp6stzVlJGRMWHChPPnz3fo0IGlPwjnYGLHMWfPnh0zZkznzp2vXLnS4j9ptPJEeno6y8QOQRAEQTgnIyMjIiJCLpdjYmeH4K4WjomOjn766afT0tJu375ta18QBEEQBHm0wDV2LUcul48ZM6Z3797bt2+3PE5PgqrVamsnqlSq/fv3M1T9ValUwNEOeQRBEARBHh0wsWs5Pj4+dXV1e/fufeeddwYOHEgfvH37dkxMjJOTk+Xq/gZkZWV9+eWXDImdNeWJFmBXlcqxwL01sMB9a9tB5YnWtoMDs7Wd4coOKk+0e/AXw4rvv/+eJMnHH3/8hRde+Pjjj1999b/b4c4AACAASURBVNW+ffvW1dWtWrWKYUVzv379bt++nW2dS5cujRs3jr3oqlKpTExMZGkEAOLi4hgmIJsPXe6OJdnZ2Q1quNjQGYPBEBMTw95OSkqKXC5nb4eToMrKypKTk9nbOXXqlFarZW+Hk6CysrJo8Ql7cEar1Z46dYq9neTk5AZ7SFsGJ0HJ5fKUlBT2dmJiYhjkFpsPJ0FlZmZmZ2fbiTO0yhF7O4mJieYtfWxA5Qm7BWfsWPHEE08kJSWtWbMmMTHxjz/+cHR0HDRo0Ny5cydMmMDSMidfhuytUrldlV+3K2cwKAZQeaK17WBQj5QzqDzR7sHEji19+vTZs2ePrb1AEARBEATBxM4W3LlzZ9myZQxr7LRaLY/H40SIHUEY0CvLcz5f0OW/P/EsihTWJ8YLE07B4ME2dAxBEARpGZjY2QCxWOzu7m40Gq11qK+vV6lUONGNtDaaW9c1uXf0SoWkw/1yxPUp53l5t2zoFYIgCNJiMLGzAQEBAZs2bWLooFAooqKiWFYGBwAnJyf2ZcoBICwsjL0zANCjRw/2Rry8vBhy4jZ2RiAQdOvWjb2dgIAABiWS5sNNUEIB+22AANClSxeRSMTeDidBeXt7szcCHDkjEom6dOnC3k5gYKD9PDYymYyTx6Zbt27sRW6Ao6A6dOhgP85IJBL28hUAEBQU5OTkxN7OjRs32BtBWgNUnrBHUHkCaRuqL5zOWfNxxNb9ljN2OSsXqdIuR/522oaOIQhiz6DyhD2DL/sQBEEQBEHaCfgq1gYkJiaOHz+eYQM83VRVVdWGTiEIgiAI8n8PJnY2oFdE+O+zpjvk3QIejwJCExBa2WsIZVEKvKamZtasWa6uro3PrTxzwv2Jsc28UOWZE7LoUcxFxkmdtmT399WJ8QQARVGujz3u+8KbfAfH/G9XqjOvA4C2uEDs05Hg80VePiEfrrr2/JORe0/xJFbLLzOABe6tYW8F7jnB3u4woPJEa9rBgdnaznBlB5Un2j34i7EBpd+uDHRy6L/t98d2Hh+wdV+QzKm/ImuKBRMmTBg3blzjTxOKJEt2b2vmVSiSLPr1uwcqT+Ss+tBQWR6+bnv3rb9HbPjVpFFnfzYfKLLTnI8iNu2N2LSX4PHKxk7rtObHsOXftCRaC1B5whp2pTyh0+s4qaeKyhPWQOUJBlB5whqoPIE0E5yxa2vq8+7U3bza44c/+Y5OAMB3lAbNXUaZTABQdnhv1fnTfAdHvV7vyaPI4rzMn75y7tnPWF2pyc0KnLWwIuaQoVJ5Z8W8kMVflOz5QZ2RRgiFPKEoaP5ygYtr2eG9VediKaAcgzoHzF5YsGm1qbpKtP97sl9f8wRb+ckD1QmnCZEISDJg1kKjqlqTc7vH9wd4YgkA8CQOgW8tuvnW1Nq0FOfeA8w+M1QqN9XV5n3zGanXmjRqh6DOgW8tIqx/F7Sr8ut25YxdBQUc7aeyq6BQeYIBDOqRcgaVJ9o9mNi1NZqc25LAkLRbWQ2HelEe8ecuavZSEIo052PHpqcTPH593p3QD1cLPbwqYg6XHd3v/+q7Ncnnwz5dZ6qrFUidu3yxmRAIi37+tvzEAVm/IRWxR8LX/kIIhPkbVtZev+I3/c3qpATdv98wZ3Wa7FuVp453+XIbweNVX4wr3Pa1bODjjiFd6KyOhhAIpV26a3JuWSZ2DFQnneU5OIZ+/F8AUMYeNaqqhW4enN0sBEEQBEEeBkzs2hqCz6+tqRnQt2+D4y+G+QU7Ofxn8BAA8HMU73+qX01NjczbV+jhBQBCdw9jbY25M18qBQJy1nzME4q0JQVO4b3qMtKcuvUkBEIA6PTeRwBgUDZ8R1OXnqpXlmUtmwMAQJKkrt7a2hGC39wHw7lHX/n+7bn//di590DXQdEClybWBSIIgiAI0jZgYtfWOIZ0EVSWVZcUkRJH+gil12mz0vVFefrignmn5wJA2a30itWLXGUysHytaTF5XnkutiYpofPKTTyRWL7vF0O1EoCiyAfNrhOEc4++QfOXmw+ob90s2bWN1NabZ/Uok0mdleE59plmhiPy9o3YuEd9+2ZtanLGvFc6r1gv8e/UzHMRBEEQBOEWfEfe1kgCgl36Dirf+l8nIN3c3FyEgupfNhjTkrz6DTLcSZdJHd3c3IQFd27W6po4meCRBj1QlLG6UhIQzBOJjXWq6sQ4ymhwCu9dez2F1NYDQMHm1dUXzgDBo4yGoE730yyniN6q1GR65q/u5rXy439Iu0RIAkPyN62mT6RMxqKf1os7+Dr3+MeEYnBwsDXliZpL5+puXHEK7+X7/OvSLhGaHCYpKq6UJzgpiWlvyhMymYy9HVSesIa3tzcn4hOoPGENmUwWEBDA3o69KU94eXnZiTOoPIE0E1SesAGUwVC67+fKszFAUTyRyH34mA7PzCB4vLIjv1VfOMMTS/RG45Nrfzh19DAc3B6xYRcA1Fw+X3b4t7Dl39z64A0AKnD2osJtX/MkEr6zzOPJ8QUbvwiY/YFeUaI8c4IgCIfgzoFvLwaCR3cOWbJa5Hk3E6o4+WdF7BG+gyNFkoGzP5AEBJs0dcU/b1RdvcSTOFAmo0vfwR1festy1d3VZ4b2+P6A0MMbAEht/bXnnxT7BZr/7nea81HJrq0USQJFibx8At9ebHkuYueg8gSCIC0AlSfsGUzs2giKojZt2qRWqwFArVZnZ2f36tXLWue6urqNGzceOnTI/P1MIpG4u7ubO1RXV2s0GvOPnLQWbFpNCIXOU1/n3DK22m2r6frl8i2rzYkd3Vr93Wpd5vUO3+yyT5+xFVux1eatCoUCEzu7BdfYtREGgyE2Ntac2BUVFTEUoNLr9b6+vsXFxeb95C4uLpbDrKSkpLKy0vwjJ61+L72Vu+ajig/f1EYOMfXo32bXxVYbtjpJpCLPDqJ7G5npVqGnP9+7Micnxz59xlZsxVabtwqFQkDsFZyxs0cUCoWfn9+NGzfCw8NZmrKrSuVY4N4aWOC+te2g8kRr28GB2drOcGWHEyMZGRk9evQoKSnBGTs7BDdP2CmcFPVWKpUPVJ5oDnFxcfRcI0tQecIadqU8UVZWlpyczN4OKk9YA5UnGEDlCWug8gTSTPBVrJ3CSVFve6tUblfl1+3KGQyKAVSeaG07GNQj5QwqT7R7MLGzATdu3Jg/fz7D0NLr9W5ubrW1tW3pFYIgCIIg/+9gYmcDOnTo8NRTTzF8h6urq7t+/bqDg0NbeoUgCIIgyP87mNjZAC8vrw8++IChg0KhmDVrFidVOhEEQRAEeXTA1MFO4aSot5OTU1BQEHs7YWFh1pQnHgqulCeMRqOdOMOh8oT9SAi4uLgEBgayt2NvyhPsjQAqT1hHJpNxIlhib8oT9uMMKk8gzQTLndiG2tpahuykvLy8a9eu6enp7MudIAiCIAi3oPKEPYMzdjbg7Nmzw4YNe2A3y4KQyP8Rc+bM6d+//8svv9yCcxcvXuzg4LB8+fKHuly/fv1eeeWVFlwuPj5+3759H3300e7du0tKStauXQsA69atKywspP/fgLlz5+r1+pCQkMLCwg0bNjT/Qunp6Zs2bZo5c+aff/6p0+lWr17dZLdvvvkmPz9/3bp1DY7Pnz/fz89v4cKFzb8igiDIowkmdjYgKirqypUrDJsnKisrR40aZVn4G/k/Qq/XX7lypWWJXU1NTU1NzcNeLi0trQXXAoCkpCSj0ZiVlZWdnW1+ILOyskwmU5P96bp0OTk5D1u+ITU11Wg0pqamKhQKhm5ZWVlNzmTTKnwPdUUEQZBHE0zsbACfz+/Tpw9DB4VCwVWJIPupVM6VHSxw39rOcAWHd5i9EVSeaG07ODBb2xmu7HDlDNaxs1vwF2OnoPKENVB5ggFOguIKVJ6wBipPMIDKE9ZA5QmkmeCMnQ2or68/efIkw+aJmpoaHo/Hfl+LvVUqt6vy63bljF0FxRV2FRQqTzCAQT1SzqDyRLsHEzsbkJqa+uabbzK8YCJJcvjw4dXV1eYjOp3uvffe27Bhg1gstjzi6+tL/z8gIGD8+PFsqlTk5+fv27ePucCeJYmJiYMGDWrx5RAEQRAE4RxM7GzAoEGDysvLGTrQBYrd3NweaGrJkiVisZiiqOTk5HXr1s2fPz8gIKAFLlEU1alTp+ZndSRJHjp0CBM7BEEQBLErMLFrDxAEMXDgwJKSkpMnT86cOZM+uGbNmsGDB9PLfk+cOKFUKl988cX9+/dnZ2cLBAKBQPD6668rlcrffvtNJpO5u7v3799/165dy5YtA4AG3aqqqtavXx8eHq5SqYqKiqZNm3bu3LmampoNGzbMmDFj9+7der2+vr7e39//xRdfxPl5BEEQBLEV+DeYS+bPn08QxBtvvMHeVAuKenfu3LmoqMj84+DBg2/evEkrT1y6dGnIkCEajUYqlS5cuHDhwoUdO3aMj4/n8/kFBQXPPvvs1KlTzSc27hYYGFhSUjJixIhXXnll2LBhp0+fnjRpkkQiee+99zIyMsRi8bx585YuXRoaGlpbW8vgIVfKE5yUxLQ35QmZTMbeDidBcYW9KU9wIj6ByhPWkMlkLXtd0AB7U57w8vKyE2dQeQJpJjhjxxmXL19uZsnWgoKC1atXM6yxq6+vz87ObsGOQqFQaP5///79//zzTy8vr8LCQgAICQmhF8xu27ZNKBQqFIrQ0FAAcHNza/DJ5eDg0KBb7969PTw8XF1dAUAmk1nukO3ateuJEye2bdsWHh4eGRnp7OzM4B4nn0q0G+zhxBkejxcSEsLeDr1Wkj2cBMUVwcHBnNjhJCiuqkJy4gyfz+fk5nTs2JG9EeAoKKlUKpVK2dvhZEABR0F5enqyNwIcOSMQCDiRiPT392dvBACwtKTdgokdNxiNxpkzZ3bv3r05pWINBkNVVRVDYkendA+7cSkjI8Ny2Ds4OERERKSlpRUUFAwZMgQAkpOTU1NTFy5cKBQKjx8/ThfCtcwFaZrsZln3yNIxDw+PFStW5ObmZmRkrFy58v333/fx8XkotxEEQRAE4QpM7Ljh66+/TktLO3HixOjRox/YOTQ0dM+ePQwdFAqFj48PPXPWHCiKSklJuXjx4ocffmh5fPDgwefOnSssLFy8eDEAqFQqPz8/oVCoVquvXLli7ctfc7oRBGE0GimKSktLE4lE4eHhoaGhRUVFBQUFmNghCIIgiK3AxI4DsrOzV6xY8e677w4cOJArm01uQVi5cqX5/3PmzAGAVatW0TmWj4/PggULGrxUjYiI+PXXXwMCAuiFOAMHDty0adPatWulUunkyZN37NjR5Lquxt0ar+NxcXHx8fH54osvpkyZcvToUbr8pru7e2RkJENQWODeGvZW4J4TUHmite3YlTM4MFvbGa7soPJEuwcTOw6YNWuWp6fnypUrGWoOPywNinqLxeKtW7c26NP4SAMqKyvHjx8fFRVF/+ji4rJkyRJz65o1awBgwoQJ9I8hISH0ltjG3U6dOmXWX+/Vq1evXr0AYOnSpfSR+fPnNzOoY8eOTZw4sZmdrZGdna3X6yMiIlja4cQZg8EQGxs7duxYlnZSUlL8/f3Zr7TjJCiuOHXqVHR0tEQiYWmHK+UJAOjatSt7Z9jfYa1We/bs2VGjRrG0k5ycHBISwn5TCCdByeXyoqKiAQMGsLQTExMzcuTIxktEHhZOgsrMzBSJROxXyHHijFqtTkxMfPLJJ1naSUxM7Natm4eHB0s7qDxht2Bix5Zffvnl1KlTR44ccXJysiwpTPPqq6+ePXsWACiKKi0t9fHx4fF4Op2upqaG4ePYZDKNGDEiOTm5uLiYPuLi4mI5HXj16lVLTRhrrfX19UqlUigUtuBcy1ZzpfIWnGvZWlxcHBsbyyYiAKiqqhKJRJaJXcu8oqdwWEZUVlZWUFBAB2XZ6uzs/FCWq6qq6IXwFEUJBIKHOpcgCPOeFZIkH+pc+ou7TCYTCAR6vZ5u5fF4JpMpNja28bkEQdCzMiRJJiUlMd8rywL39LLRqqoq2oK1c3k8HkEQDVrpoo8EQTS+z01et8lW+t+amhqWY8EsIcDmyUlNTc3Pz2c/FoqLi9VqteVfehuOhfz8/NraWnpJbosjAgC5XG65irfFXj3sWGiytaKigs/n5+XlsYnIxcXFUnmixV5RFCWXy82PTcsiAoCioqLKysoOHTqwicjZ2Rln7OwWgiulnUeTsrKy8PDwp556au/evQBQXV3t5ub2+uuv//DDD3SHhISE9PR0+v83btzo3r07QRBarbagoICh3oFKpTp//vwnn3xiLqXB5/MtK0cYDAbL2UFrrUqlMisrKyoqqgXnWrbGxsYOGjTIycmpBedath46dMj8Ja9lEQFAdna2yWTq3bs3m4gA4ODBg5MnT2YZkUajiY+Pf+qpp9hEBABXr14NCgry8/NjExEd1Lhx49hEZDQay8vL8/LyBgwYwCYiPp9/5syZ6Ohoeqkom/t84MAByznRlnlFz9h169aN5VigHxuWEalUqnPnzo0YMYJNRHBvxs5yk6MNx0JRUVFJSUnfvn3ZRAQAsbGxo0ePNnew7VjIzMwUCoWhoaEsx8Lx48fpx4ZNRHV1defPn4+OjmYTEQAkJiZ27tzZ29ubTUTZ2dlLlizZunUrJ8WnEG7BGTtWzJ07l6Ko9evXW+swdOjQoUOHPqxZhUKRkJAgkUis7Z8QCoUMryrMrRKJRCgUNqgl1sxzW6NVIBCwjAgAxGKxXq9vM5+ZWx0cHKwF9VCWGxTusmFEQqFQLBbTobXxda218ng8ho1EzbRMj4J2NhYa97RhRNYem4e13GD5l23HgkgkEolE9jMW+Hw++7EgFAolEgmHYwGxNzCxazknTpzYu3fvmjVr6G+rAKBSqQBAo9EUFRW5uLhwUjsUQRAEQRCkmWBi13JOnToFAIsWLVq0aJHl8T179uzZs2fx4sWrV69u8kSKovLz8y1XXTSgoqKCk6LeTk5OnBS0DAsLY78EHrhTnuBkk4q9KU/Yj4SAi4tLYGAgezv2pjzB3gig8oR1ZDIZ+92jYH/KE/bjDCpPIM0E19i1nIyMjAalt9Vq9fPPPz9q1Kg5c+aEhYVZ+5MfHx8/fPjwB9pPSEgwb2hFEARBEDshIyMjIiJCLpfjGjs7BGfsWk54eHh4eLjlEXpXbEBAwPjx4xlOHDZsWEFBgcFgsNahoqLiscce40oECUEQBEGQRwRM7GwDs1o2J5KLCIIgCII8amBixyWurq5cvdrmqkSQXVUqxwL31sAC961tB5UnWtsODszWdoYrO6g80e7BX4wNqK6u/v7777dZZ+fOnePGjWOvpKRUKhMTE9k7HBcXp1ar2dvhREIgOzs7MzPTTpwxGAwxMTHs7aSkpMjlcvZ2OAmqrKwsOTmZvZ1Tp07RdYlZwpXyBF3Kzh6c0Wq19NYrliQnJ5eVlbG3w0lQcrk8JSWFvZ2YmBiGZSrNh5OgMjMzGyyktqEzarU6Li6OvZ3ExETLasMtBpUn7BacsbMBeXl527ZtY5jbMxgMwcHB7HMpy9L/lsZPnz5948YNWgOgU6dOY8eOZX7526SdFsCwEbj5tEtnMCgGuAqKvRHAO9z6zrTLoOzKGa7s4Iyd3YKJnQ2IjIxknhFRKBSzZs1qpTJ4e/fuFQgE77zzjkQiMRqNsbGxu3fvnjlzZmtcC0EQBEGQtgQTu0eL0tLS/Pz8RYsW0QXGBALB6NGjze98L126dP36daFQSFHUhAkTtFrt4cOHQ0JCsrKy0tPTJ0+eHBgY2KCPo6Pj77//rtfrdTqdr6/vpEmT8GscgiAIgtgKTOweLUpLS318fCzLxhIEQVfgLCkpuXLlyqxZs3g83s2bN48cOTJ69Gi5XD59+nQ+ny+RSC5cuCAQCBr06dWrl0gkmjFjBgCkpKSo1WqzID2CIAiCIG0MJnY24Pbt28uWLWNY5aDVatPT09mvsWusPMHn882rPRQKxY4dOwBAr9e/9tpreXl5KpXqp59+AgCSJOn1y66uri4uLmFhYWq1+vbt2437BAcHx8XF7dmzJywsLCIignmtHnP5dZPJFBcXJ5PJ+vfvz9ANlScYQOUJa6DyBAOoPGENVJ5gAJUn7BZUnrABJSUl3377LcOdV6vVGzduvHr1amRkJLeXrqio2Lx588KFCx0dHc0HN2/ePGnSpIKCgsLCwqlTp5qPl5aW7tu3b+7cuQCQmZl54cKF8PDwBn0AgCTJwsLC7Ozsy5cvv/rqq15eXi3zTa/Xr1ixIiAgYPbs2S2zgCAIgrQBqDxhz+CMnQ3w8/NbtWoVQweFQrFx40axWMz5pT09Pbt06bJv374pU6ZIpVKKovLy8mpqakQiUadOnc6cOaPRaBwdHfPy8uRyeadOnRqc3riPTCYTiUShoaGdOnWSy+UlJSUtTuwQBEEQBGEJJnaPHFOmTImPj9+2bRtJkiaTqWPHji+++CKdjY0cOfKXX34Ri8UkSU6aNKlxIT0/P78GfQQCwZ9//nn69GkAkMlkERERNggJQRAEQRAAwFex9olCofDz87tx40YDLdoWYFeVypu0k5mZuXPnzib7L1u2TCKRNDiIBe5b2xmu7NiVM6g80dp2cGC2tjNc2eHESEZGRo8ePUpKSvBVrB2ClSlswIULF0QiEWEdHx+fcePGVVVVsbzQ/4XyhFgs9rwHAAgEAvOPTX6eovIEA6g8YQ1UnmAAlSesgcoTDKDyhN2Cr2JtQP/+/U+ePMlQi7yqqmrXrl1ubm4sL2RvlcqbDDk4OHjevHlwb/OEr68v8+YJrAXPAAbF4Ax7I4B3uPWdaZdB2ZUzqDzR7sHEzgaIRKIRI0YwdFAoFLt27WozfxAEQRAEaR9gxo0gCIIgCNJOwBm7tiMlJYVeNmcymcrKynx9fa31ZL+67v8RkUi0cuVKW3uBIAiCIP/HYGLXRmi12hEjRqhUqmb2Hzt2bEJCwp07d+gf3d3do6KizK1JSUkKhcL8o7VWk8mk1WrPnz/fgnMtW8PCwujdqS0417JVrVYfOXKETUQAoNfrnZ2du3fvziYiuFcLnmVEcrm8traWDqrFEQGAQCBgHxEdFMuIFAqF0WjU6XQVFRVsInJ3d7dUnmDjlVAoND82LYsIAHQ6HQAolUqWY8EsIcAmotTUVKVSyX4sqNVqnU5nuS3RhmOhuLhYr9fTfdg8OQBgKfZg27Gg1Wp5PF5mZibLsWCpPNFiryQSicFgYD8W1Gp1RUWFl5cXm4jc3d1RecJuwXIn9ohCofDx8UlPT2df7uTR5MaNG+fPn6coysfHZ/Lkya13IaVSeeDAgdra2ieffLJ3796WTTk5ObGxsYGBgWPGjLF2enZ2dkJCwpQpUyyFQOrq6rZv3+7l5dVA4YNmy5YtdXV1kyZNaqxJtXnzZpVKJRaLZ86cyUYy6OrVq2fOnHnppZfofcqINdKK4vcmfjHnqc2+slDL4xRFHrq6McJvSBef++J41Zqys7f2j4+czSP4AECSxpVHn/sfe+cZH0d1LvwzdXvRFvXeLVlykXs3LmAbjCEYAqYTWm4gIQk3Lzc4yU3B3JvQEjq5odk0G4JxwUUusmRbli3JkmX1Xla7q+19ZnZm3g9jlkVlLWnX1mLm/9OH3TlnHj1n6rPnPGVd0cNzMi5dHv3W1ie2z4uTpb15fx0A4Lmv7ihv/zcMIJUs8cOHu8JUlaI8m99ISIzJfv2eCASuBnPP2xlCTPzOAxcjIu3Wf6hZ4P/osT4RNuFSaZUde/5n/z1JMbmv3l1V1fX13w89IhLI/37XGREegfJZkcVms3355ZerV69OTk6OlMzGxsby8vJp06YtW7YssHFgYOCrr76iKApFUYFAsGnTJrVaPZaEAwcO6HS6W2+9ValUAgDKy8tramrkcvkdd9wR/IDi4CtPRDO8jx3PNcjhw4e5CmkRSdsRgvLy8u7ubrPZPDLpyZEjR3p6eioqKkLsXlpa2traOixJRGVlpU6nq6urG9nfarX29/fbbLaR/25wcHBgYMDpdJpMpvLy8kmN5hKHDh0ym82HDx8OR8gPgfcrtp7rPrSz6m/Dthsdva8f/fnOc9/Zfqp996tHftZvaeW+nu89Vt76xXsVWwMdPq58zk3YO0313NcTbTtphqIY0mDvDl/VXWdf9JCOdkNN+KKC0ZlbDY7uHnNjRKTV9h5yERY34Xj3xLOT2P2N40+5CXur/qzLa/nn8aeNzv4eU+Oe869FRLfIcvLkyba2Ni6ve6Q4dOhQb2/vsWPHgjeeOHGiv7/fYDAMDAx0dnaGzpZy8uTJjo6O06dPc1+PHTtmNBrb29tra2sjqCfPVYA37HiuQa7aPHSIfzQeHbg+wyp8hNgx0HNk9oRgIWHmVuAUiEiChmsblmUAADQ7/EAxgAm0fruRpYM3cl+ZoD40/Z3kbZE9+gREAgBAuHl2h+MDZASleSmC+8Aw/knsznxzC5A0yXxz/BhmePmcaOBK3GKczGFPj/E/WwKM1Ip/FHzv4JdipwC9Xv+Pf/xjZMGuAG63+/XXX6+trS0uLg7zf0VPpvJIyWFZds+ePaGfNQ0NDV6vl/s8d+7cMP8jR2DyL1jgsBnBvLy8wOe+vj6PxwNB0Jw5c8AYNDc3O53O5OTk4EiagYEBnU4XLK2lpYX7TJJkV1cXAEAsFgf75AEA3G53Y+OliROVShVOxd7Ozk6KomJiYrKzsyctJDTccfvjH/84nushSipPNA+eeXLHAu7zDUU/AQBUdu6xuQ1JqvyipCXBPT2k/UTLzhhJfFbsLG6Lj3Q1DJQDANI1RTgqAgC4CduAtVWASdLUhaTf2226ENg9SZUPABiwtoBvHs658fMmrTaHydVvcekABOakrwtTVDBeyn2x/wQAYE7GhMUyLFPTfXDUJqUoNlaRPlGBrfoq7kNRyso2/Vkf5Zpr/S8AwIMPPjhRUQG492P4TJKKeQAAIABJREFUlScYhrFarbt37w7eKJVKk5KSwpQcoLu7myAIBEGCb1udTud0OgNfNRpNRkbGWBKqq6sZhgk8Pdrb27mXVGxs7Mikqm63+//+7//27dvHL8VGIbxhNwU0NTU99dRTIQw7kiQVCsXWrVvDNErMZnNzc3OwD+zkOHLkyIIFCyQSSZhyvvrqq40bN4YppL29/d133w1TCE80cM899+Tn51+2W0Qum5aWFvBdy3tCuAn7Lf9QhqkDz1WGM+x4rhxPPvkkb9hFIXxU7BQwbdq0AwcOhOhgMBgeffTRcPzfOaItU3lUlRDgmXLGeT1Ew2VDT2pxkIfnGibE3ATP1ML72PHw8EwN36OSRDAcAVcEHp5riYj45/BcCfgZO56o4MKFC5999hn3WS6XP/300yE6Jycnh57O7OzsJMlLbt1jLfbp9fr4+Pjxa9jc3DxMIMuy3AJfgGCPGZPJRBAEBEEhlv96e3s9Ho9Wq0UQRCgUcskCTSaTyWQKljYwMMB99vv9XDYpoVCYnp4eLMrn83V3d3OfpVKpQqEY/9CGodfraZqWSqVh5mIYGhpSq9WjWm/cwbxyPnwRRypQPr/58P/buYb7Oj9zAwCgUXfK6bPGylIztEXBnT2k60J/mVQQk6C45M9EUJ5eazMAQCtNRmGMYYHLa3P7rRgs0EgTKZowuXWB3WNEsQAAq3cIgEuzjAny0f2ifASBYRgyDvvYSdhchBUAkBM7e2SrzWZXKidzwRB+X6+lkRNLUhRFkRLxeB02GJbpGDo/apMYkylEE061M+i4lBQmTV04aOsgaR/39c4775yoqAB9fX0oioZIJj9Ozp49m56eXlpaGrxRKBSGSD4yEpqmPR6PTCYbtdVoNFIUBcNwsLYWiyXgbQwAUCgUCQkJFotFKpUGckwGaG1tZRgm8PQYHBzkJssVCsXI563P5ws8EnmiDd6wmwLq6+sffvjhEPPYFEVlZGQEO71e87AsG1hxu+zS27Jly4aFDgzjxRdfNJvN3Od77rln1D5ffvll+Cnudu3axeUCUCqVw4zRt99+u6enJ4QCAIA333zT4/HMnDlTLBYnJycnJiYCAA4fPsxlJfjpT386rL/JZHrppZcAAAqFYpjY/v7+N954g/tcXFy8YcOGSQ/queeec7vdqampW7ZsmbQQAMDBgweXLVsmEonCERI9zE5bfejXbPBl88uPlzQMnCzJuOGptW8F9xy0d973Ttbs9NXP3vRZYOPeujf/fvjx5247kKYp9Hq9H+x78fPuZ9M0ha/feynZzfN7txxt/ggA8Ol/GAAAa1+Av7HrwPuPdI6q0unTpzMzM8fj5PTe6d99dPJPAAKv3TtKHrtJ3wud5obH3i0CALx2b7VOp+vv7583b/JxHqc69vzh3xsBACvz7vz5DW9dtv8w7nk7w+DoBgD8z+bS/9x5Xa+56WzMcw8u+UtwcuCJAkEQjuM5OTmTlsDR3t6+cuXKlStXcl/37t17+vTplJSU+++/f/xCXC5XZWXl6tWrR2196aWXTCYTgiDBz43t27c3NTUFvmZmZt52220VFRX5+fkjs1Ru3boVBD09/vjHP3JpvRcuXLh06dJhnZuamk6cODF+5XmuJrxhNwUkJyfffvvtfv+YXjsul+vjjz8emRNyokil0mHzOpMjUHkiTEI8YYuLi8cZAqzVakMcOo7xrPGF87gPEFiMGBk3x20J7drF9UEQJCUlRS6XB28cNRAvxL8LXhYJc4mTExX+Omlw5YlwiMiZio2NDV8IGK4MBABAoOEHCgYwAAD6bnIRGIIBABAEAwBwHE9JSQXd3zmPKPKdpzHEgst6BaampgYum9AIWBwAMJbESR9hIfj2/CoUijCjR0WYgPuAopO5bKBvTgSCIAFNwlxDj4uLCy6DMWmGHeEQ93gIhEJhiEnuUaUN28jd1Onp6SFWPAI3fuDZNdajgK88EbXwht0UoFKpfvWrX4XoYDAY/vznP4fvwSAUCiOS2Twi1iGI0NIblxU9NCtWrKisrAQAqFSqK6rMkiVLhoaGrFbr8uXLR+pw5MiR0Ku9K1asKC8vnzVrVvBzds6cOU1NTaO+sGNiYuLj4z0ez6pVq4Y1JSQkxMXFOZ1OHMcXLVo02QEBAMDKlSvLysquu+66cIQAAEIkVpgQETlTIa6ECRGszF0Ln91x+s83z35iWJ9YeeqDS7cVJX9nkmN+5o0PLn0uKSYHAIAgyNq5d5T3fnjTzG8nVzbP/c/y1i+Uokt5auZkrKnuOQoAJBeNuVo3/mQZt8395UdVz2tko/ef9BFOVOeqZYkohAMAJBJJmIHzM1PXiHGZn6XvWfyHSez+k+XbXjjwE40kWSHS3rvwv984/nMMFq4vejgclSJVfGXYEV6wYIFOpxv50AgNiqIhHsXLly8/c+ZMVtZ3iqAsXbrUYrFwq0MikWjhwoUAgLFeCvPnz9fpdIE51yVLltTW1srl8mE1dQJ0dHRMSH+eqwaf7iQa4UuKXRts27bN6/VyP5pxHE9PT1+/fj2XESp0k8vlGibq97//PY7jI5vmzJlzyy239PX17d+/f2hoCMOwGTNmXH/99eFn3uL5nnLH6/EKsfb1e2tQGOO2vHrkZyJM+tCy5wN9Pj/34o7Tf3pkxd9uKHpo2O6Ta+IBADidzkOHDnV2dnKGVGpq6vr167kfosH3u0KhyM3NXbFixbBps5MnTx49enTdunXBmS8vu2NdXd2ePXvmzp17/fXXX51hcvAlxaIZfsZuarDb7SE8yex2+9VUhufKcdttt3FLzB6PZ8+ePTt27PjZz3522aY77rhjrIXpkU0kSb733ntr166dN2+e3W5/6623tFptSUnJFRwVT3Tjo9yfnNl298LfjdrKAnZf3VtPrHl919kXhplok2vi4fjss88UCsWTTz4pEAgIgtizZ8/27duH3e9+v99oNB45cuS111577LHHAkFOLMtWVVXdfPPNFRUVw1Kah9hx//79FouF883l4QnwvUk3cC1RVlamVCpVY5OXlwfDsMViCf9/RSTVUKTyFUVETnCYxZQrM345YrF4zZo1g4ODI612sVi8atWqUZvGg9/vX7du3fz58yEIkslkGRkZXFBtOFx7lw3DMN/Hy2ZyQn6y7H8+P/fiWCVca7oPKyVxK/PvJP3eVv25YDkjm0LsNYzv6Y0ZGoZhxr+oZTAYcnNzBQIBAEAgEGzcuHFk4BSKoomJiXfffbdSqTxy5Ehge3t7u1QqLS4upihqYGBgpHBuxzvvvDN4x+Li4rvvvnui3tiROsLfo3RFPzT4EzMFLF269Pz58+fG5tChQxs2bAjfK8hsNnOuZmFy/Phxt9sdvpx9+/aFL6SjoyMiYfYRUYaiqEOHDo2zM/eGGNV1knNDnpxXpVgsDvzE3717d1dXV5hOaUajcViptMlx5MgRn88XvpyInKm2tra2trYoUcbn8wW/1CfN2bNnjUbjyO0Jysw75v3mpYM/YUfUsQUA7Dn/+obiRwAA64of3lP3Bgga1MimEHsNQ6/XV1ePEnI7UQ4dOkRR1OX7XY6InKnm5ubxe5LNmjVrz549ZWVlfX19NE3jOB6YkOPCSwNAEDRz5szgC/LMmTNcnaG5c+eeOXNmVPlut7usrCx4x8m5UFdWVgaSBoRDOKH3PFcUfil2CoBheCx3VA6DwRCRH0PXauWJ76Mybrf70KFDaWlpI+PR3G73xYsXExISAk27du36/PPPAx2Kiopuu+027vOnn3766aefBppuuOGGQCYCr9fb0tJSXFw8zIF6olyrl034QkCUXcNjyWFZdvPcp483f7q79rVN343tGHL2XdSd+q8bPwEArCm89/5/5jy64gVuUKM2SQXK0E1XbVAT5eqfqXXr1mVkZNTV1Z05c8bj8WRnZ69atWqsHHhKpTKQYc5ut/f29v74xz8GAMyaNeuFF15Yv379yEQEnDLBO06OSB1hfsYuauENu7B47733HnjggZHb//SnPz377LNXXx+eaCNgogkEgqysrOBcqcFNSqUy2Pc54H43krHc74xG444dO5RKJf8zmgcAgMDoL6//5292rl6UfXPw9r3n33QT9jveuBSsTVCe0osfAJA6VtOm2U+GbuIJJj8/n0tgbrFYKioq3nnnnV//+tejLpXabLZA5HtVVZXP59u2bRv3laKo2tpaLoI19I48PKPCG3ZhYbPZAAB33nlnampq8PbFixeH2Mvtdh84cOCywRMR+cXJM7WEMNGCm6qqqsJJW6jT6bZv375x40Y+FzxPgNz4OddPf/CVw48lx+RyW/wMdeDC//1jy5ms2Jnclpqew28c/cWGmL+M1bRp9pMhmq7+oKIWl8vV0dERWIpRqVTr16+vqqoymUzD3g4AAJZla2pqCgoKAAA0TZ87d+7xxx8PzO21t7fv27dvLMMusCMPz1jwhl1YcIbdL3/5y2FxTKG5ePHiT3/60xB+JAzDrFixwuFwREBFnmsdv9//8ccfb9iwIT8/nzfseIJ5YMmfH35v+qCtY3HOLQCA8tZdMpEqYJ8BAGamXmf3mnTCi+Wt3lGb6vvKzG7dWE3FKRPLxHZts3v3bpPJtHjxYqFQSBDEyZMnxWLxsGwgLMuazebDhw+73W4uj93FixdFIlHwim1mZqbb7e7q6grOBMmyrMViaWho8Pv9E02Ax/NDgzfswoIz7MaTMjeYefPmcRU/x8JgMCxevHiiYkfyPao8MX7GU3niqimDouhYtWgnRHDlCQDAZ599tnPnzuAOd91111hJDVtbWy0WC1dpl2XZ2trazMzM++67b9LKyOXykXMMk+AHUHlikuA4npubG76cy1aeEGDin69585ldl1b599S+ft20u4I7wBCyLG9z/1Ble+2BUZv21r1pcvaP1RRs2IVfeYIjPz//ShR7mBzjrzwhlUoffvjho0ePvvjii1zkRGpq6k9+8hMuSBZF0YDrhUQiyc/P/+lPf8pV26usrBzmcg3DcFFRUVVVFWfYBe+YlJR0yy23BMr0/fd//zfDMDRNQxB06tSp1NTUhx66fCaa0JUnxg9feSJq4RMUh8X999///vvvDw0NaTQao9EIQZBWqw1fLJ+gmIeHh4cnauETFEczfFRLWHDOcC+//LJarY6Li4uNjc3Kytq+fftU68XDw8PDw8PzQ4Rfig0Lbin2o48+euKJJzIyMlpbW1999dV77rnH7XY/+uijAIBXX331woULAACGYc6dOzd79mwURZ1OZ2dnZ4iMJ16vNzk5uaOjIxA/IZFIghdV+/r6gj3w+Fa+lW/lW/lWvvWqtQKeKIY37MJi69atP/vZz66//vqAy8KWLVtKSkqeeeaZ+++/n/OumByRShEUqVzwk8udO1KZ8McVwQT30TMoLsF9+C5KkcopH5FBRfAIhy8kUj4n0XbZhK8JiNCguCMc/jUcwUGFLyRSg4qqeyEi1x7g89hFMyxPpLn11lsBAFVVVZOWoNfrb7rppsbGxjA1MZlMFRUVYQphWba0tNTlcoUvZ/fu3eELaWtru3jxYpQoQ5Lk/v37w5dTVVWl0+nClxORQRkMhtOnT4cv5+DBg16vN3w5ERlUc3Nzc3NzlCjj9XoPHjwYvpzTp08bDIbw5URkUDqdLpyHXoD9+/eTJBm+nIgM6uLFi21tbVGijMvlKi0tDV9ORUWFyWQKU0hjY+NNN92k1+vD14cn4vAzdpGHi79zuVzhCOErT/yglOEHFYKomrHjj/CVVuaaHFRUKRMpOfyMXdTCn5jJ43K53njjjR07dgzb3tjYCABIS0ubCqV4eHh4eHh4frjwM3aTRywWP/fccxaLZebMmYWFhdzGffv2nThxYubMmZmZmWPtWFVVddNNN102QTEXmcHDw8PDw8PDM054w27ywDD897//ffPmzQsWLLjjjjuSkpIaGxu/+OILqVT69ttvh9ixsLDw9ddfD11SbO/evXxBQB4eHh4eHp4JwRt2YXHLLbccP378r3/964EDB4xGo0ajueuuu7Zu3Ro6rbxEIvnRj34UooPBYHj++efD92DgK09caWWuUOWJSRORQfGVJ0Lw/ao8MU4iMqgfeOWJq6CMUCjMzs4OXw5feeKah688EY3wlSd4eHh4eKIWvvJENMPP2E0BDMNcuHAhxLSTxWK5mvrw8PDw8PDwXBvwht0UUF5evmLFist2uxLmHcMwzzzzTEJCwi9+8QtuS2dn544dO7Zu3Rr4MHKvQFNPT89nn3329NNPR1yxYFyEX/ZfX0+Lk8IQxLKAopnbZiT+ZV2+m/TL/uvrXK0keMWn5qllDMsO275hWuwLGwtHSrb7qJ990bCrTuf9nw1XdAjfCxrOXex86T8AAIJb/7D21hWRWEbjmTCtjT0XP39jqrWYGPPv/UViWnxERLEsqD3b2HPwg4hIu2osfeQ/NXGqqdaCh2d0eMNuCli+fLnNZgsRPDE0NDRt2jSVKgIPjmEZz+vq6gQCQUdHh9FoHL/XUUDVtLS0cKy6CaVfr/r5UqkABQA4fP6lr54sjJfdXBgHADj3i6USHAl2QHQRfgBA9VPLuP4huOOD6rtLknfW6SaqTAgiIodhGAiCrnKC+7zC7CpsUUzXwfr3P/fK5BvXzIK/USAig4q2IwwikXkrsoMyWN1VXx1VIpAiYzLrWWwkqpVMSI6uZRCTFmjiYkYVwrLshI4ww4KjFQ31+8oLE8VirXyiyoQmUnKCcTh95uqO1OW3amJHOQIhiKp7IVLK8Hnsohb+xEwNCoUiZmwUCsWGDRGYUjKbzZWVlcFbysrKlixZMmPGjPLy8hA7lpWVvfDCC3//+99feeUVo9FYU1PDvRc7Ozv/9Kc/cX1KS0u3bdu2bdu2Dz/8cJzVe/bt2zeJUciF6LIsdZPByX3t6upqbm4ez47K3x5o0DuDt3x275wbC+LCUWYYFEUdOnQofDnV1dV6vT58ORMaFCYSrHvqXr84Mdt+6sKH/957qo2iGQCA0Wg8e/Zs+MocOXLE5/OFLyciZ6qtra2trS1KlPH5fEeOHCH8zLEDp8UDVarsJATHJvFnc9ophp7cvsF/+iHDeLpZXaRkyDJ/7QJcOEqxRL1eX11dPf6DQPqZrytbLxw4WazVy5LUgf9iNA9BKHLVBhX6z+X1eAgv99lO+k0X+7IXrl+2eTWYoMkYkcvG7XYfP348fDmVlZVmszl8ORF5SfFcCXjDLkq5EpUnjEZjW1vbwoULly1bdvr06bGc/Hp7e0+dOvXUU089+eSTy5Yt++STT0ZG2PT09Jw8efLpp59+5plnGIZpbW0djz6TS78+6PB93WRcnKEadVAhqP/18jztd4pVy4XfTuldk7ngJzqohBn5OT9+wI3Gp+orz3z69e5z3W6S/r4P6nuhDMOypTVdQ42dmumpAAlnYikiZ+ryfTyk39oykLj45qSi0eN5J3SEXQS951xX26GTM8Td4vSRSwdXaVDjl2JweNm2gcKZK+fffgOMYRMVE1XXMF954pqHX4qdAiwWy7vvvhsieIIrRxapgu4BysrKZs2aJZPJZDKZRCKpra2dO3fuyG5tbW02m+2ll14CADAMQxDEyLin9vb2rKwsLhHAfffdF1k9ORb94yQMARdBeyn6jzfkrcuP5ZZcN+3qAgAI9hoBACuz1W/8qJjrX/LSicDKyyML0n65PDM1RnQlFLvGmLZhhbl30HD865z2g9W7pT5mZUnCKPMxPJGlycYQtWfnwK2oJmOqdbk8fobp77Wk00Th6gUTnawaidFFHrrQbyurLGYbBXnRXqGHZUGf1S0ZNCVnzC+4eRUi5p8qPNEOb9hNAf39/Z999lkIu42iqIyMDI/HE8F/SlHU6dOn5XL5888/z309ceLEqIYdBEG5ubkPPfRQYMunn346sltEfoOG4NQTi6UCtN/um/dyeWC6DgDw5W0ZKOsPlPoIMB4fO56RCDSq4lXzajoqqd4msml/E4qaF81KRCL8o4InmEaDq7nLeYv+hGhuzlTrcnlYlu0ZciVbLAW33i+I1YQprW3IXd5qpM9UT3fVYEXpEXeDiyw0yw7afcmUPy2pJH3tUlw1Mdc6Hp4pgX8RTgHFxcVnzpwJ0cFgMDz66KMymSyC//TcuXMqlerZZ5/lvvp8vt/85jej+nXl5OTs37/f7XZLJJK2traBgYGRfbKzs48ePUoQhEAg2L59e0FBwezZsyOobYBkhfC59flbdtRUPrlkovv2Wr0JcgGG8OsFl0E9Z0Zu7+bGvZ/GWC2CjkONAFRnZid3W+enKeHofu9+H+m2eKrajNPM/eLMeEgYgQTOV5oBm0frcGQUX6eeMzMcOQwLTndb/n3RVNzfnGE6hRWnQ2hU35uEn+6x+hJJT4rXn7Jprjg5cao14uEZF7xhF6VEJKl3cOWJsrKyVatWBZqEQuHixYtPnDgxZ86cYXulpKRs3LjxlVdeEQqFDMNs2bIFw7CWlpbgPmlpaStXrvzrX//K9Z8xY8Z49Jlc+vX756Z8cWHw2a9btq7JAQDc9mUfAADZa+Ra/3n7jJmJoyffL/5bWcUTi6fHX7KPDU5i1ZunGZYlaWb6X4+LINmmSWjzXa6NyhMwjsfOm2GvO9TH0IzHVdh1mIXhzxq0Zg95XY5GjE0ygI6vPDESnd13tM0k6OhMhzrQxHDTj0vEEgyLwBFWKhVjNQ05fRI/meB0xc2fiYhClZ8JXXnCQ9GHW4Yq+10LzN3q7iOC4gyAjf72kcsVMBQBgy/EoMaDw0v22zx5ONCYPal3/0KWM2bt7/HAV57guZrwlSeiEb7yBM/Vx1BaNnDs84FWvQuFYJGoIXe9a/rMWXGilTmaFCXvVxQBjC5iX6OB6DfknnpfMCMLEkf7dJ3NQ/pJX4rBmjBnffzaFZOW02v1Hm83dTvp2YYOyfmv8OnpUT12ltU7fBjrT8EhYcdg3E0PxJSM64frDwq+8kQ0w8/YTQFNTU1PPfVUCB87kiTBNyEUPDxXB/X8EmLITIHjvf0yFdM3s+NriwI/Dwp9fuOMRPnc1BiYX5UNA6OL2N9odNrJvMFOLC02qi0bAAAAboLqt7nn4ow6fb56fsnkhNAMe6bHuqvRkqkWzzd14LW78aKMaB67n2H6LB4VDrKlON3Ur127hbfqeL538IbdFBATE1NSUhLCsHO73SdOnBAI+OBEnqsHKpPGzC72NJfNTLDXmrPVTFfs+a+EMFQLpjVZzP1238psjUo84UQPPAAAg5P4usloJcG0gVahuQYpTJ1qjS6Dl6J7zO7ZaqGodUC5ZDqmmIy/r9lNHms3Vet989NilI31+Ll/40VRPVfnJqg+qyddgedKUXd9r3rlZtWCSVq0PDxTCL8UG40YDIbExMSGhobwl2KjKlN5RORMIsH9lVMmUnKmpPLEqBgOlznO7tWZnOeZ3Hh3gxgB/jk3l8dOYxF4ugafk6IsTpSPU8toO8JgiipP9Nt9h1uG7AxaYOrFyj8SzM4EKMqyDBS2J1lEhIyUQ/jpziHndK1YMzAkLVwTt3bFeFKcBN+YDMueH3DU9NvMFFKUIEfPVWHn9+CF47LqrtCgLtvb4PQNOX2zEyUpOGI7361esVm9ZD4LQJTcmBGUExEhTU1N06dP1+l0/FJsFBLVQUk/ZK5Q5YnJcfz4cbfbHb6ciKRf7+joGGfliaugzPe68sSoqBfOITTTUYwsppt1cfMJBkbPfbXO1lKshBtM5Hu1xt0X9BYPNR5RfOUJAEC7yX2gyeAGWCHtxI59iOUmARSlaToip9tsNkfkCAdHvpM03WVyTdMIk0gfljBHvWjuOBPXBSpPmN3k7gbDJw0mXCwtSZIjp09h9fvGP1c3ODgYkVRKo4bzjwpJ0x0mlxDQ12Up04So7XyXavlm9ZL5EAw3Nzd3dHSErwxfeYLnasIbdlHKlag8MeVyoir9elQpEz2DQqUS2YzpjINIy9HmxQi6U5dRLOQ99WV6c93yWBQAUNbr+qJ+8GyvjWYuo3D0DGqqlKnpt79+ZhAIJPliiD5fg6TFwkrJN/qEr8slSREQ8Y0Mkqa7hlw5MXi2BHe161UlM1DZeMMnWZb1M+yZHuu/L+hb7f7lOXGJAshfdgJvPoQXpU8wscvVqzxh9RDtRmeqHJ2dopAzjKWmU7X8ds3S+RAMg2v0GuYrT1zz8D52VwmfzxcXF+dwOMbZ/6GHHqqoqGhvb+e+qlSqxYsXB1qrqqoMBkPg61itPp/PZrNBEDSJfa9Ea19f3549e8KUbLfbxWJxcILiKRzR4OBgf38/N6hwJLvd7uTk5GgYkcFg8Hq9PmUW0dgsl+kK0tZewFZP6z02ULmX7u5KT0tup0WHB+SHL7AFMuaB6xcGynuMlAyCCEcrk8kUuGwmPV6bzQYAMJvNV+dInj5TdazVUO+VZatwV7d+sLExBXSh+ZdOscVicblcA/393FdcINBqtYF9zWaTz/vtPFyIVq/PR5JUYqJoEvuObNUbh7rMLi3k8xPgQvkQUrgqPzdr/EfjbHN3vYlyK7wZCjxZLcM9qP/0KZHuNFacbrZbx6+Vj/jOHGQ4Iwrdahgy9VtcqN8bJwSsH/QYUbkN0qz+sXrRXAiGufFarVYYhpubm8O8NoJHFM51NTg4GP69oNfrBwcHExMTwxnRsEHxRBW8j93Vo7q62mq1AgBomjYajQkJCWP1tFqtO3bs2LZtW5g+diaTqbm5ecmSCaf2HUZpaemCBQvCT3305ZdfbtoUbvK4trY2kiRHVp6YEmVIkiwtLV2/fn2YcqqqqpKTkxMTw82AGpFBGQyGjtbWNJOTHDjtdxLt2oX1Traw65CDYsRZi4puuq7Gi54Z9OAoQlDMmkzZonRVcBHeAAcPHly2bJlIFG6qlIgMilu+Dz/p4HiUcRL+wy1D5wy+klS1UoD4T50SNB3AizICNWFpmuaWW/Y2AAAgAElEQVT8aMNUxmQakkqlQmG4R7ivry8uIbHL7MpWCgripa5WPaItSbxxDTy++C2blzrdbf3qokGJ+pfPzENhiDUa6TOVInsdlpMAJjivo9MNxMfHw3C4TmB9fX0pKSljtdo95IDdk6nE82KlYgwhLW5rfXfs+vtU82YFLz1fvHgRx/GcnHALhETkGna5XJWVlatXrw5TTkVFRX5+vkYTVhGRpqamZ5555q233uJ97KIQfsbu6lFSMt4AK4PBsGPHjiuqDA9PCCAU1SyeZzjsxaQtGb3lcN4N5+D1szq/dvVWndtFzVi9MK0wuV7n6HZCFX2eAbtverxsVrIC54t8ANBj9ZZ1mHtd7JLsWByG6OpzeP1ebEZmwKqLQkia7TS5clWCaXFS0uQE0kLN4nnjseoIP1M7YG8YdBgIeEF6DOVxojDEdnQwDQ1iqgnNS7oKyk8UkqZ1Nq8cZeclSZMVQgiCiCGHraE//pafKGcVTbV2PDwRAPnDH/4w1TrwDMftdj///PN33XVX8MLBu+++++WXX2ZlZSkU38mo3tnZ+eqrry5fvjx4Y01Nzeeff75gwQIcx0ctadDX1/fyyy+vWLFiPPrAMKxSqcL3qEBRNPwJfARBxGKxWCyOBmUgCEJRNCYm3AqSMAwrFIrw6zRE6gjjOB6TEA9g2HKqTJ6TIGypURaWVKIZGeYWQJu7K8ok8rhFc3Npmu6xEX4IbTB69TY3DAGtVBCY70AQJHouGxiGJRJJ+NOHIZTxM+zpbuu7NUapRFScrERhmGlowMo/xqanwaLvnFkIAjAMh3+6IQDhuCDMI+wm/AMOojhBNi1Wyvooa223ZsX10uyM0HvRDHth0FnaOnR6wJOoVuTFywUogiEw3tSIlb4nUFNoknpy+kAQLMAF44zYCAEMQ8MzRrGsyUX0WtwZCrwwQaaR4BAEefrNjiZdwh2PK2eMsggAw7BMJhMKQ5XcGA+RuoZRFFUqleHLUSqVKBrWtI7JZHrllVceeuihiBSx4Iks/IzdFECSZEVFRQh3WqvVOjIUq6am5rnnnptQAVmhUBjsuTVpAnXJwiQi9XDCf65xREQZGIYzM8OqNcQRYl1+QkRkUCKRKCkpCQCgmJ5Pmm63nf5COS0JVO28bsU9h4S3F5p6NXh9//4dJr1l8W2rszXis722PgYeIuHtdUMzB52zkxU5WgkMQRkZlzEOxklEBhUpl6CxlDG6iOPt5lqjb36GRi0VAADY1lbs6Ad4YSokGW4WQBAckdehKOyfNw4v2Wt1L0hXZaolgGGtdb0xS29VFheE2IVh2bYhd3W/vX6IyI6Vr8gVc9lABCyL1l/AGr7GitJh2eQN6EgZCjLZd37Qeil6wOrWCOElaYp4mQAAAFjW2a5nJdOT779TmpU+qpAw1ysDROQaRlE0Io/iiLwUAAARiRfmuRLwht0UcO7cuRtuuIGiLpMzgnPI49ixYwdFUf/85z9vu+22lpaW6upq7sfoXXfdFbzLxx9/3N3drdFoJBLJMGmlpaVnz54FACQnJwf22rlzZ3d3t9PpvP/++zMzM3fu3NnR0YGiKIqiDz30kMVi+eijj/Ly8hwOR39//5133pmVlfXJJ5+0t7crlcqcnJzW1tYnnniirKzs3LlzGIaxLHvnnXdGqignTzSgXjyXsjt8PRXK6SnQ8Q83Xf9AnVZKNLLJ6EVr69HD/3Rkr1l8a3HG2V7r1+32BIXQQED/rNaXxItKkpXZWkn0rj5GDj/DVvfZLgw67Qy6PDcO5RK5tbdjX7+D5SVBiuF3YvRgcvn0du+CFHmKUggAsDX3i/NXaJctGGu2jGHZdpOnpt9eY/Cmq6Ur8mLQb6qRsMYh5txZweBpbGYmhEfXa8VPM3qH1+Yli2LFWWoxhsAAAJZhHRf7kNg5scsXChN4LzGea4rougN/ICxatIgrGjYWXK3Y4AW+LVu2nDp16tFHH9Xr9ceOHfv973+P4/jJkyc/+eSTG2+8kevT1NTU1tb27LPPwjD8r3/9K1hgT0/PyZMnf/vb36Io+v7777e2tkqlUqPR+Mgjj2zevPnYsWNHjhyJj4+XSCS//vWvURTduXNnWVnZjBkz+vv7H3/8caVSWV5efvToUZIkW1pauH/x8ssvYxjW29t76tSp3/zmNzAM19bWfvLJJ08++eSVOGg8UwKMYdrlCwe/9vjdTfJpSezBfy26+bHTgtkd5+Bc2QU30VT5ua+3c9qiNfMK4mUnuyzn9N5MrXyIYt46NzgnXjQzSZGjkSDXbjGyXqv3ZJel3kRMT4pJl12amWM6O7F9b6E5SbB6FC+IqIBlB2weEcysyFBopQIAgLvXhKhnaZcvhEdbIKYZts3kPj9gr9Z709WylXnxWMCfkmWZ5mampUVMNiJF6eHn8o0kLDC5fUaHN1MpmJ/0bYgPQ/qt9b2irKXa65bgSkVoGTw83zt4wy5KGctvpqOjIy8vj/POyc/P37VrV6Cpt7c3MzOT23HatGlcslAuyXh7e3tWVhbnVHHfffcBAPr6+tRqNbcCqFar6+rqOPejt99+G8Mwg8GQlZXFNSmVSpqmFQqF2+3u7e3Nzs7mspaXlJTU19e3tbXZbLaXXnoJAMAwDEEQIQbFV54Yi+ipPDFSDqZUaBbP63nnuHJ6qiI/yb77zZWbHqlYvqCuHC2wV82K1VmO177frp+zbuHavIT8WG9Nv73DQRcmKE0k9a8aQ7FGUJwoz4+V4ugkT1l0Vp7gQkFLu5ypKvHy3PjA3BXT1o7tfwvNSUS0oay6Kaw8QTNMr8WjEoDCeLlMgAIAvEN2V7sh5cG7cdVwb1GSZhr1zga9s36ISFdLV+bF42jQufB66epqvHYPmpOAJEZm3itSlSdsHsLg9MWLkEWp8niZMHBvUU6vrb5XNntd7MoliOgyznPRfGNOuTJ8HruohT8xUcp4knqHSFXDNQVXnhjp0hd8b7Mse/bs2bq6uocffvjhhx+ePXt2cJ/jx4/7fL5hFhX3AYKg3NzcX/3qV7/61a+efvrpZ599NoTCfOWJsYieyhMAAKPRyK3aBxCnJife/ri9sQ+VCRUFyd4v31ktJtPWLavWLPX1mzXTkue6znR8dfTfh2qdhP/movh12XLa565v6ShOlHtgwccNph01/ae6LQ6ff6oGFcHKEx6Srui07KobrDESS7K1hYnKb626piZ0/1tobhKiDTUPNIWVJ7yUv33ImSCGZyUpOKvO7/Y1HKyMv+0RScZ3KtjafdSpbuuO6v7PGi1eSHBdfnxunDzYqmP7+/1HjghbDuEzMxC13OvzmS0RqGcQfuUJN0F1GJ09nR3FseI5qcoE+bdWnVdvs1R3qlbeEX/9dZe16gAAfOWJEPCVJ6IWfsZuCtDpdP/4xz9CmGVutxuG4VFnv3Jyck6cOEFRFIZhDQ0NwQmWkpKSKisruZ+Y9fX1ICjDeHZ29tGjRwmCEAgE27dvLygoCI635XA4HImJiRiGud3umpqaYC/dgKqJiYk1NTXcv6ipqYFhOCcnZ//+/W63WyKRtLW1DQwMhIi0jar061GlTPQPSlE0jXJsMR/9RDM3S1GUYt39zuIb7o25ednRw+LZF79UZicUQd3OI2dO9W1onV1YmBxzU2G8qbPpbI8lKUY6J03j8lGHOp0X9c5UpaggXpasEI1/EiRSgwpfiJeim22MqXagx83mxcnnJwZZBgzDnD+PVnyMT0uBYy7v/j8llSesHmLA5pkRK87WSmAIAtyi5Pkeb86iQEwoy4J+u/ei3tln8+q8IEMtXRknGj5l5aeY8+exk58JUrRoUdoln7xIpkSdpCgfResdXinC5KqEdg+bqQ4KLmFZZ4cBiAsT79gkL8gdrx5Rf2NOoRx+xi5q4Q27KcDlcnV2doa4tXw+HwzDfv8o0xvp6ekrVqx46aWXcBxHEGTLli1cSn0AQGFhYW1t7V/+8peYmJjExMTgUolpaWkrV67861//CgBISUmZMWOGTqcbJnnevHmvvfbaiy++KJFINm3a9MEHHwzLqwIAmD59enV19XPPPadSqVJTU3U6XUpKysaNG1955RWhUMgwzJYtWyZ3THiiH/WCObTbY63eHzMnI2Zmhung9mkrNsdtWlSZpNbVN+SxzfLi9DmdRwYp58HerKSMeCECHpsd22knj3QYExWimSkqkmbqTM5DnQNz4kX5cdI8rVSMR2BJ6CrgJPwXBp3NBucFUrlGJFkSLw42dVi/n6mqwmu+wqaHFRB65WAYZsDmkSDsklR5gvySPcr6Gcv5HlnJetZjBwC4SbrF6Go2Oqv1vgSFKF0Tk5GIjRTFDuqY8+cFvRXYjHRIEi2D9VG00em1e6lCrShdJZbgyPmgJxxD0raLfXjSfM3SBaIILRnz8EQtvGE3BeTm5n766achOhgMhkcffXRYZOtrr73GfVi1atWqVasC21Uq1datWwEAEATdc889wbuYTKbA59WrVwenLE9JSfn973/PfS4uLi4uLgYAPPPMM4EO//u//wsAuOmmm7ivBQUF8+bN83q9ubm59913HwzDX375Jeeit2zZsmXLlo179DzfVyAE1q5cTBOkte6oalaGek6m9eQXcpd7w6K5x9QxNcfx6S1VkoKUZEtTfFWFnrrlFCkQdFpy4xX3zYzrtXqOtRqSYsS5cYrCRGW/1bOr0ZIhs6coRTlacVqMOGoDLAYdRMOgo9viGfCCLK2sWCtIVH43yYjbxVSewduO4DMmWhH1KuGl/L0WT7IUmRYrkwguPfNZhrFe6JVMW6lZsdiw++CBZmOf1dvjYlLV0uvylQJ0FIObJXxMXR1+eheSpEZnZoHoOGU+ijY6vHYfVaARLkqRyQTDX2qk1WVr6JfPWa9dthCVRm+QMg9PpOANO54JIBQKe3t7jx49KhKJRCLR/fffP9Ua8VxVYAyLW7VUT1G2htMxRSnquZm2hmO4x7tuyfya2LVle6VF575W5yfjxRkJzfsXQWnCBPlZRmB02RYmiW/MVVo81Ik2Q5xMmB0rz9LKzG6iweIp7dJP1+BZGkmWWhIvE0RJVKWXotuG3M1GV+WAJ04uyNAoM5JwAIDhu+qxRiNdVSUyncNmZIDRjKEphmWNTp/RRcyIFQWWX7nttov93ti5A5nTD10YOuqQz7P4k5WKFUljVJtgWba3h2m4KNCdjp6JOjdBmVyEw0cVaEQLU2Sj1LVjWVf3kLtrKG7jAzElxRMtbsbD8z2FrzwRjYxaeWIScNntR608MVE5XAkBCIKKioqWL1++aNGiefPmTTR1Pl95YiyisPLEWJcNjGGixASvgfL2tIriFMJ4Jalr9/R709WSvGUlnUhCZ4dbae8TFaTgHp3i4plYUUxqZuKgH60ddIkRkKEU4BBtd3k6rIRKgmdopOkaicsPLhg8pe0Wk8PjIv0YDElwNGCEXM3KE6Sf6TR7qnptJ7us5f1uTCCckaxMVUtF3ywZc3IA+CbNx/nzYrIRy04AEymndnUqT/goutvskqOgJEmaEnMpjTDLAgfh76rpafLnHNBMM8IihUycpxVnJ2rEY+SfY61W5swZrPwjgcSD5iRC+Cjrs9ygEATFsNFbJzCocVSecPqoAasHYagEGVaSJE9RigQjYq4hP0t1mRHVrIRNt8sLcidXyoKvPDEWfOWJaAaKiBMlz4Q4f/78ww8/HOLIUxRVX19fWVk5f/78q6kYD884IS1Ww6Ey2lKrmJYEACCGHLaL/TFLbpHMKznVNlT99clpQyfVxWmwWEh16n3yGXB6OsjPt/ihAZvHYPcmSSAvRZN+1gfjObHSJIUYhiHCT+sdvkGbx+IhZ8cJU5WidJU4SSG8Cqu0LoLus3m7LR6dw9dioeIVoqQYsUo85vQh6/Gw1dVY3V40JxFRTaAYzFWCBUaXz+j0FmpFORoJhsAMy1q9lMVNWTykvnHQIykSLV0am5446pLrt2IIH9PYxPb0CJ31eGY8mOrMwwzL2jyE2UWqBCBWJkxWCsXY6Pp7jQ5ns04+b4N2yXxUxlsekaepqamgoECv18fF8T6LUQdv2E0BNptt586dIY68w+F4+umnL1y4MH369Kup2FWD9dOlc9fIcrMWfPoOt8Xv9nydPW9dexUquTQVd/6prdaztQAAV0e3JD0FQhBRcuKCT94eJopyuBr+68+6vYc3dFcPa+rZvrP7Xx9BCIrKpTNf/rM4Jan9tX8NfL4HQjF5YV7x88+Op8w5z1gQJovxyAnWVifLTQAA0F7S3tiPJy9QzZtllKsrD1UaGzryJd2y3CTG4aE69WTiAjg/H8rMYmF4yOUz2L39Ng/GUk6v3wvQ7FhZQbxcKsQAAARFG10+vd1ncvsKVXiyUpSsEKbEiEa6T4WDh6T1TmLQ4Ruw+2xeqstJx8qE8QqxRnKZ5WCmp5u9cEFoPIvmJkGCcCeoIo6boAZs3jgxnKkWywWo1UtZPJTNSw26/VIBLjeaWcVMwZKlkDpkLVc/zXS0M60tAt0ZLDMOVk6xbUT5GbObsHiIJCkaLxMkyIUjp+g4GD/tbNXDymL59HxF0TSIX369MvCGXTTDG3bRCFd5orGxcdq0aVOtyxVhcH9p019e9g3qlx/5gsueNdKwC7A3uXj12cNjlf2pvPPR5Ntuqvvl7zb01ARvtzc0Vd37sxXH/o0p5C1/e93d1ZN2z+a6X/1h+eGdiFjUsHWbQKPJ+fnDV2iAPxAIo8l4tJyx1cvzEgAAgGVdPSZXp1G9crN0zsyzTQO9x0+LBqs0Rem4QkwbbP6eITJnJZSfD6emcktjDi9lcHpb9M6OITsEI1qZKEMrTVKIJDgqwREUhoacviGnb8hNpEthhRBLVgoT5cIEhVA48XTHTsJv9VAWDznkIofchIug223+GIlALRFoZULZSA+tEbAeN3O+Hj/7BZIWiyaqwi9UH1n8DGOwe01uMisGlwtQB+H3UbSZYKU4KhPhUgHKdBs8okJkwSIohI8Hy7K9vXRTo6CjDE2LRWIVUzlMFrgIyuImHD4qKwZPVIhipTg8tj6k1WVv0kmmr9Isni/QRKY6MM+o8IZdNMMHT0SAAwcOPPfcc7W1tRiGlZSUbN26Nfwo0UilCIqqTOUBOT3vf5q25Uf2hqae7TsLtv5q/BIO5C1Y9OUH0rzswPGZ8/YLo/44keVlLy/dhSnkAABxapKjocnZ0hFTUoyIRQCApJvXX/jtXyJi2F3zlSdCIIjVxF631HgMsjfVKvITAQRJ07VCrdxevddnHJpRVJD74C2VBxP6j34iT4tT5yfhGgUyUEt/dYSctgaeVgAlJclFmFyE5cTKGTapx+xuMdgu9JpP+iGJCM/TiHEEFmGIAEXS1DKKZrvdZJ3J4fSZchSYWoIlK4VJClGCXCjBET/D0gzLsCxJM4SfIWnWS9JeinaTtJPwO3yU3Uv5GbbV7pcJMIUQU4gFGpUgIxEb7yFnGLajg25uFBjP4jOzIHG47nGRrTxB+mmd3auzeRQ4rBIivXZSLGBlAkwmFWqUCNfV36n3SKaPatUxDAPDMGBZtr+faWnGmw4LEtVISTY0EcfBS3nswr6GuUFRNGP1kFYPoRFAqXI8IVmqFIWaHGVo2tWmB7Ii9eoVqpIZEIr8kG/Mq6MMn8cuauENu3D58MMP77333pycnF/84hcEQXz44Ydr1649evTookWLQuzV19dHUdRYrSaTKSJJvc1mc3Nz8+LFi8OUc/z48QULFgxLvzIJ9u3bt3HjRndnj7myevbr/6ucOf3cI7/M/39PwuN2uF5+9N/9Tkd/c3NBQQG3BZVJKYdzZE8Yw7j6SH6Xu/Ot9wt+9zQml3a89i/SYsVjlIP7Dtt7+8McDgCAoqjS0tJ169aFKae6ujo5OZlLHxMO3BEOU4jRaOzs7FywYMF4OgtiNXGrlxtKIdvFc4rCZAiCUIlAXZLh1bXUvPJJ1vV3X79ybmdx9sVDp7rLKmIKUpWpsUiiCuk/R+86RBatg/PzoYQEAEEwBDI0kgyNhKDoPpun1+xqNbqkQkwuFrjs1hiVhqIZP8P6GYiB8bNmRmz1uNodbsIvQ/wCFJELUZkAkQrQfheAYQBDEI5AQhwRIKgIh8U45va5JDhyQ2HSJF7RjH6QbWjAm0qFGXGDKiw5bKuOpmmj0ZCQkDhpCQzLeilaZzDBmMBOMhY3gSGoVibQygRSAZqCIsEGFsuy/jadTz4TWbwYGs2F/0J9fbFWyzQ1YY0HBfEx6JxsMHHPeq/P5/G41WrNpAfFqdrWM4BJFW6CzlDiWQnSOBmOXc6+9JkcjmadrHitakGJMO6S2RqRe6G5uRnH8ezs7DDlREQZt9tdWVkZnOtqclRWVubn56tDr8WPA77yRNTCG3ZhYTKZ/uM//mPWrFkVFRVcnOZjjz02c+bM7du3hzDsysrKQpRn4Lj55pstFkuY6kVbpnIu/Xr3B58lrF+Fq2PUi+biMcrBvYeTblk/TgmipATQ5mLp8Vam8g4Mnn3w5xkP3qVdvhAAkP3EQ1X3/QwVi7XLF7GRSE4RnUf4KiuDq2Pir19hOAzZ6s4oi1IhBAIQJEpSYQWxft2ZnjcPa5f/aN2PV1+oTrUc+LC7X6aZlizNiEOT1HBfJd3wNVWwFsrJhVNSuGwUAgzJ1sqyNLIiNzFgdesdXr3Vo1Ay8XKhVIgDABiWoWnAApZmAcsCL0W5fX6njzR6KNJGxUkQCYbGiHG5CJcJMakQFWMoAEDvdwEAJmrVsRYL09iIV++G42OwOdkARdi+UX5FTIKJnm2SpgmK8VG0l/L7KIbw0w6CJUkWwJRKghckxsTJxogyYVh/S79XPQdZvBgaEVAJMQzT2ys9dxZ1t8LxMWhJDjTpCInwLmA36be5CbuXoj3u3ERNgkIeeoqOgyH9jrZBJGZm7Ia1yuJCKOimjqpiD1GlDF954pqHN+zC4oMPPnA6ndu2bQtk38jMzLTb7aGn7pctW9bV1RXiVjeZTFx1hwirGwUwBNH36ZfCWE3FhrsAALSP6Pnws/EbdhPC1dZZdf+TRdt+q122kNuSds/taffcDgAwHC6DtOH+YOUJgCkV8TdcZziCWWqOx8xIh3EEAAChsCw3AUqPczWWWo5/nrPqDvD4rxuqGt1VB4YS1Nr0WGlWPJqqRXXV9FeHidzr4MxsOCMNoBgAAIKAVirQSgV+mkHsehnGGs22VhLSyIQamUgcFEUhxhH1N1PJDMu6fZSLoPrdlNvs8pJUrBhBYSAV4KTHLcZgVuQVc957l5sEYi0WprkFO/M5ppFjkVh7HT8MyxB+lqRokmYIP034GR/lZ1jgJFkMRYQYjKOYAMHkKENDpEYujI+RomO8Ylm/39/Y70tehCxaAg0LDvVTTFcP09EmaD0Wg1D43AJojAjTK4qX9Nt9lN1DqoVQvBgrjpXrWUOORnzZZBwsy3r7Lc4Og2LejeqFJdz0PA8PD+ANuzApLS0ViUTc3DhBEARByOXyyzpkQBAUXIl1JOEvekYtA7sPiJISlpfu4r76na7DJatd7V1jxUYMwzswyPrp0H0cTW3SzFSWBVX3Pzn71W3KWUXcdp/eePaBJxd98R6MYZ3vfMgsmBXOQHiGgcqk8TesHBLglnNfK2emoeJLEceoVKAsSiEdalfdfsLkLFh2iy/17rambl/tGWOiVhOvkKfFIskadLDef/g4mbIISk2Ds7LBNxUCUATWipGZmRoP5R+0eQftXr3JYqcgtUyokX7HwgMAwBAkE+Ey0SUjjGUB4ac9hN9DUmYS8bnpZpfJT7PxEgRDIBGOijFEiCEiDBWgsBBHBAiCIwA1D7GtbVjNV7hKhszKhCXh5jAbFZoFhJ+maMZPs36GIf0MRbMUTZM0QzOsg2BhGOAogiIIjiBCIYajiAaBIQB8FO0kKBHCasUogBClBBvTqvOR5MVeMm8NsnAhCMrExjpdbEc729eL955CE9Xw3Fx2cOAqW3Ve0m/3UnYvqRZCMQIsJ1EaK8VxFAYAGMcxrUrZPPZmnTBzcdJdm2W5WdEWxcLDM7Xwhl1YNDY2ZmZmXrhw4Yknnjh16hTLsmlpab/73e8efPDBqVYtSul5/9PMR76te4bKpKk/3tTzwad5v3kSAHBi7e2BR/Sif78vGDGpVnbdLcl//zOUfMkXjRgyn978EMswDEUdX7EJU8gX7/7g1KZ7Fu/+0FZ30acbPP/UVq4nrlIu+uK9uDXLT6y5jWXZ+OtXGoqyrvRgf2ggQmHcmhWwQGAp/0JZlBzchMvF+Iw0yu1ztx3z6e3506/zSJcM1h11Wm1tWo1GJVYmqfEkNWrqoM+e9XWVwLFakJYOJyQEqgWIMTRLK8vSyjykX2e/ZOFZSaCSClUSgVyEj/xBBUFAiCFCDFEBgRylAQByucJPMz6KJvw0QdFOn9/nICnaS1A0S3ilel2abTB5oJxRS9nkOFiMIz4/QrhgGEZgCIYgCAAnyVjcPgiCRo/NZAHNXpqJpxnAsCwLABfVQTPf/vlp2m4newk7gAACQSgCowiEwjCKIDiOYSiiVkDwd0MrWMB6SL+boMQoiBVjKjGGIbCJGHNdmHF4qaY+cvZGeN48zmEOYhh6cBD0dDMGo9BSgyaoobk54YcFTAAWuAjK4aOcBKUWQDECNCdGqpXiYyUuGRWGoBwdBlhRrFqxRDlrOhJ26mAenmsPPt1JWMhkMrVa7fP57r777gULFgwODv7tb3/r7e19/fXXH3/8cQBARUVFY2Mj17mhoaGwsBCCIJ/P19vbm5ubO5ZYh8PxzjvvfPLJJ/n5+dwWrhhAoANFUX7/t35mY7USBGE2m1NSUiaxb3Brd3d3cnIyiqKT2De4tbW1NTMzM5wRAQDsdjvLssFxBpPTqr29PTs7O8wRkSTZ29vLzb9OekQAgKGhIbVaHZipnbRW7e3taWlp4YzI7/f7fALrc54AACAASURBVD6r1ZqQkDDJEbGs/XyDed+HbJw4oTCDc8ShaTrge8CQtG/QShociGq6x+q06moccRqDQqGUCmIkOAbDwOVljA7aZKeyV0Bp6UNCkSo1JfCPuGofAAAPSRkcxKDdY3H5hnxsjESgFAuUUqEgaBWPYTjjCpAEAQAQCAUw/F03LL8fMerxgX7UPCQ3nmE0Skojo2GEZlmaZjibjGFZlgUsYBkWoigSRTFuXyb44QkBCEBBBh8LWIDAl0IqERiCYRiBIBiGEAiCAKD9pEgkQrjOEAg24xiWAUGCGZb1UIyb8MsFkEKIyQUI8o0x5vV6BQIcw749C9x4GYON6tSTS++Ei4tRHGedLrani+npxdqPwWo5khADSYTgUgEMBADgdDpkMnngWH1znMGwYxW61U9RlJ8SCkXDWv0M4/CSDi/p8vljRZBChKkleKzs21oRwdcGAMBiscTGxgYszkAr62c8fSZPv0W9cFPM3Jncr74rfS+YzWYURRUKRTh3N4Igvb29gQiMSd/dfr+/u7s7KSkpnBEBAHQ6nVqtFovF4Yyoo6PjpptuOnnyJJ/uJArhDbuwEAqFBEFs3759y5Yt3BadTpeXlycQCPR6PYqiDzzwwIkTJwAALMsODg7Gx8fDMEwQhN1uj42NHUssTdMxMTFPPfVUYuKluDm5XD5v3rxAh9raWrPZHPjKt/Kt0dYqdnqSvX7GVifPS4BguK+vz+VyBVpFIlF6Whpp9XgGrc62QaMNJz2d9jiVThMnEuIqqSA1PpZlGHbI7h+0DqIZHlxAxcX542JZDBeJRMGeDH19fTaH0+qlLT6/3cfYYWGcSqUUC5RSgUyA2WxWwkcEOmM4xoVtQgSBGvW+znZgGIwbOudXSBiVDFJKpNJvy0i4PW4/FfQyQxGpRHo1W0mG8VEsxUJJMWKlCJMJUBiCrFbLqCPisFrMZNugDc0lc3IojUbqsCfQDGu1CkzVSHyME6EJPzXmvqElT6QVxTChTOHy+V0E5aVoCeXAGEKOwTIMwBAYeQaHXxvfbXU6nYzJ4x90kImz5DMKF958Y6A1Sq72H2CrTCbj89hFLbxhFxZqtdput7vdbkFQDYPNmzfv2rWrvr6+qKhocmKv+QTFPD8ESJPFePykr7NCOT0FEeFjdWNp1jtkd1zsd3QYAE4RsXKXUmFGRFKVQimXCDCEcfsYo402OcikhUAbCyUlQAkJED5K1RAWALuXGnL6jA6v3UsZvIxchCvEArkIkwow2OdDbVbYakZMQ4jLqTBUgRgpGyMFcilAoshJi2IYL+n3kLQUA1IcVYgwEYqMK6CX9JPN/T5tCZyaBjxuxjgk0J1BtHIkVgmJr3yRFRa4ScpD0m7C7yapeDEqF6IxYkwpwqRBlX8nKJP1Gh3uLoMwY4l8er58Wu7EUuvxXDH4BMXRDO9jFxYZGRnnz58fFvXNTcU5nZFJi8DD8z0F16gSblxrqlCYyr9QFCYLtfJRu0EIJI5XiuOVsUvy7E0Djja9WHdeKwIuVZqBBIhELJbLJHKJQB2P+drprnq6xkWlLAIaNZSYCMcnAJHoW1EAKEWYUoTlaiSEy2032a0Wo6vPTFrtBqtdCrGpthpUIUFVMihGyiZlR5XTPc0wXor2kH6KZrRiTKsQyIUIMu6MErTBRpzvoDRFkI9AKz5G1XJEK4fmXNkxMgzjoWjOmPOQVJwYkeFoslqgEEmUQizMIr8+o93VOfT/2Xvv8Ciu6/H7TN1e1DtFoleDMR1M/JVxA1x+DjaJS1wwTl7Hdhwbtzh2cMc2cezEjdhxSZy4YFOMCyCiUAVCoiOBkJBQQSttbzM79f1j8LJI2tVKs0ILvp+Hh2d37r1nzpmZOzp7yzmaQVPTL7/cMnYkSgCIQMQJcuxUMX369IqKisrKyilTpoQP1tbWAkB4FrV3XNiZJ9SgBGFSf32SyqgLNsC9VpP1f7M0GWmeg1XcscOmomws+tgYrqFSLhpkGZHnqy9gW1xa1/EsE8HK9kAT62cEXpKEtCItSdA6imzZJB3lRX8oZB6F6fVYSgpmTQGjEQQBRAEYVhYEHCD75I4cDYVpaUlHszlEgKKCgwYHBMnDSnRI0IqSliY1JEGRPRivk0HCIAF9MyyHlySWE1lB5AQxTUfmmWiThqDjC7IoSSJ4Wcnt5482h/g0nUakrA4ihccHDeuRP9ejNBghQQxyQjAkBHkxJIg5etKiIQuMGovWYNaS3QYT7hZJkjin31/XrimYkvZ/xZaxIwldb3ZIoI7Z18qgOHZJC5qKVUVlZeWkSZMuu+yy9evXK7Oxe/bsmTJlyogRIw4fPhytVW1t7R/+8AdRjBq2g2VZAHjmmWcmTpyoRr1EZZ4oKSlJSOaJtWvXqg+/fvz4cY7jwpkn+leZRGWeKC8vT0jmiYQY1aPMEzHYsGHD7NmztVotALCnbPYd5UzN/yyj8imTrtu2QjAUqLezrW7Je8IN7uwh+X5O8Ac4jhOCnKQBoAmcxkFmJTEgSH5BZCXBPAg36fAUI55iwEx6XEOBhgr/SfZ6vQBgNpslWWY4McgJQU4IcqIgST5O1lCEhiJoktCQBB3Tz3O5nCkpagNMipLocHtpnYHlRSMFNIEbaNKoITTx+HOyLAdZ0cNI7oC3udXkbiRETs4aSIwswrOjLtuNTVNTY35+QZdFsiyzgsTyAsMJDCeygpilJ2gCN2spk4Y0awmjhlT2i3g8HpfLFTuQU7eWsTZP1dbKwVNvsAwrsowd2TlzdPwkpC8cOXIkIZknEqJMojJPbN++XX3miaqqqkcffXTlypVoKjYJQSN2qpg4ceJDDz302muvTZ48ed68eQ6H45NPPiEI4s0334zRiqKolJSUGI4dwzBer1f9z8Rki1SeVOHXk0qZC94obU5W7vy5jl3pjg3/NgzONAxKj/14k3qNZVSecXCGvyGlaff+4DGvIUuXmmEUZAhyQoATg5xoC0mUCdOk6mmC0GEAfr/od0ktvFQnCtnDcZMeM+lwkw436YHA4MeNpjiGGTSk4ccAeLwoMbzIcALDiyzHeVnJy8kUgdMkTpMETRI0iVNn9qFCr6+LBBLHyyFB5ASREyTgQukWY45Ro6eI7ke5REnyMZKfkb2M5AkQbcdwLUEYSJPko8wiDJuCF+QrgZ17x5m7LQMriCFe/PF/iROlbD2uIfEcA2lM0Rg0hJEmuwxQouaZkUWZbXX5G+zawumhkebs+XP1KR2TZPSUpOoLSaUMyjxxwYMcO7W88sorw4YNe+utt1577TWtVnvppZc+/fTTkTOznRkwYMBbb70Vo4LNZluyZIkWhWhCXEDgNJ0xa6qhIM+xq9JRvsMyMrfboTtCR1tG5FK+VnNKbqDZGTxYo83QmjO0Fi0ly8AKIsMr/3hHSCJxjE4l6QyalDGKaZZdgtQkSKzIpw/BdBoZF2S9RgIK02uxCL+EInCKwM3a016RIEksryTvElhBCrE8I8teTqYIjCQImsSDIVHD8RSOkySOx9zRIIEkiDInSJwo8YIkyJKVJowUrtPTNA5epz/XHLWDy6Ik+xk5GJICrORlZCZEOmpxLYHpScJIEaNTMUkU291eY45cOFjbq5yzkizzosQJUkgQHYwo2v0hUeJFKVuPkzieQhN6I22gCD1NGDUkqW61XCw1OCHQ5GCaXYYRczLnFZlGDDlUUqJmoA6BQCDHTi0Yht1zzz333HNPfyuCQJwH6AcVaLIynOWZjo3/0RekGgtjrbo7DYWbhmQbBmcwzZnBJqfj4BFtikaXpdPpCB1FgJJhQhRZQVKSqHKS5MFk0oSTVi2FYUSoBWdF2sdgIUmqIoXUIkxL4ToNpqdBpyF0NOg04QSpJI4bNbhRAwCnl+rzosQJYkgQOVHmeNEXYhhGE5RBSQ5BEQRJYCSBY3A6Ip0MIEmyJIMoSRYNThG4UYvrKEpD4NofN7eKouiNsE/mBDkYkllOZkJSkJOZkMzypLMW1+CUlsT0JGEl8ZzU00v7JFly+0WXXxw8kdfpNDFHtgRJEkRZECVBkgVR4iSJFyRelHhREiQ5x0DgGGagcJMUHJGm0dGElsT1FEH3JGJwr+G8TLDRztq8lklX5cwoMg0pxBKRvhmBQCDHDoFAnFMInTZj9jRj0SDnrkr7rv8ah2TrMrveMBsJThCGAemGgjR2SFawyek6so8y0dp0DW2lMRzTkoSWJEALACDJMidKIUHmRIkTJJ7AeC3uIkkMMC1FE3wTzkmYX8JCEsZLPC/xKUUYQWAaktBSuJYGmsQ1FFAERpEYTcoEQRE4SeA6GUSNFPTgWVZ9SJCMmh+nLFnRI0iiJJloDAOMIrCgiJE4pqUIUZRxWeYkSQzxjCTjooiLIi6IEOK5drvXxmK8ACEel4F21QJF4Fqc0hCYlsBTSTwnJXKThihLIIEcDAntHi51qDg2VzSb/U4XH+RpEf8xp4V0Vn4LWZZlOcdAKFoZSUJD4hqS1BCElsI1JK4hcZrAAQC31w9OO0eDZLIos22eYJODyplkGDkx+/ohutzspNqejECc7yDHrh8oKyubP39+jDV2kiSlp6e7XC6VJzIajWrWMocZMmRIQuaFx4wZo15IRkZGZAz0/lWGJMlwdhA1FBQUmM3dOzfdkhCjzGbzgAED1MsZNmxYZPD6DujycnLnX+EeVOCtqmEq95iG5VDGrp+xszaYY5g206LNtJiGZjPNzqDNEzhQo0mjtRk6Qnt6vAfHFD/vTCNZhhw9KcgyRlCCJImiLIbXGEmyxLXLvChzkuyT3KcEEGRMkGRBBlEOWAYDQKoWx0iCIAmCwLOYoMHHGQGTCZzAMYrASRyjSJwmMZrAcMA4ThBkmQ9Jok8QBFHgRJEXRUkmHLUAAASGUzgQmBEEGtPiFI6ZcIwi8FyTMxSx7EkWgT37/cDzlkBQ0OVDYZE2OwtwHEDCdYRWi5Ok0hAjMIrAQVGJJDCawGkSpwic6m4iVeUWfgWdThd73STvY5gWN3PKZRg5J3XOQNPwIZTF1LnaiBEjSDIBf5gS0heysrKSRxmtVqt+GwcADBo0yGg0dl+vOw4dOqReCKIvQLti+wGGYb7//vsY3onH41m8ePHhw4fV7/1EIJIc3utzVxywb/5Ml5NiKszENT3YBCAJUqjNE2h2BGsOkkZKm6bVpNDdz+3GhyyBLEiSIMmCJIkgC7KsZBGTZJBBErpaDo9huHJ2AsMwwAgcIzCcwIAEnCRwEo87qEiEGoLI2n28O2gYfrF+YH7vYn/0I5IgsjZPsNlF507SZKYbhww2DCxAs67nOyhAcTKDHLtkBGWeQPzUYFps7r0HPbvW6fNTjYMyMKpnwyS8n2VbXMFWT6i5mrbQmjSaNtNYny35PzfIosg5AyGXXztgjGFQAWVJwJjuOUMW5ZDdy7a6WYffNL7YUDjQWDS4yyE6xPkIcuySGTQVi0Ag+h9dbpYuN8syZoR7/yHbju+NBWn6/DScjvcFRRm11LAc49BszpkXPOXm7F5//XGNldakaigTed4t4ZIliXMGQk6/dsDolBE5mrSU88UEWZI4V4CxeULtXsPIOebJeTmDB2izM88X/RGICwDk2PUDoigeOHAgRmQjp9OJMk9EA2We6GtlEiWnF0L0A/N1BbnmUcM9h6vbd3yvy0k1DEjDNGSctxvDME2aSZNmkgWJbc9lbV7WFfDW1dBWmrZQlJnGVc/SyrKs/jbFkCPxIufyc+6gNn9UytAcOiM1xukkSUrIi0K9HFmQGLuXa/eyTr9h2Czj6JyswQN0eTm9mHJNqmcYdcwYoDh2SQu6Mf3A9u3bJ06cOCk6c+fOveaaa5xOp8oTORyOsrIy9QqXlpYGAgH1ctavX69eSG1tbXV1dZIow/P8hg0b1MupqKhobW1VLychRrW1tZWXl6uXU1JSouRQ6REYjhuHDM5bcOWAOx+hC2bYd9bsXbeFc/Xs8cNIXJeTknLRwIzZw9MvvVSTO6a9zXBqd7O31sfYWZHrfbTY5ubmXrcNI4pi59stsFywxemvtRGWopRpxZaLRmsy02K7FPX19QnJSX3w4MHeNRQZLtjsdO1raNta7fRmtuiHDrzj4YKfX5sxZ7p+YH7vFtJt2LCB5/ne6RNJQvpCdXW1kiIyGZQJBAKlpaXq5ZSVlTkcDvVyrrnmGvVCEH0BGrHrB2bPnu31emNsnmhvb1+6dGlqqtrMRckWqTypwq8nlTLIqI5gmH5gvn5gvnXC2KP/+JgNFniP7tYXpGqzrHhPoqzhBKHLtuqyrWwqyXsydJg2ZPe7DlYRWlJjpWkzRRp7NlGbqEXJYTmyJPNehnMHCOsgOnuAZUIWZe3BWrpE3akeVJZk3h0IOXwhh08IcKYJcy1Ts/UD8pySwJw6pR+Yr16ZC68vJJUyKPPEBQ9y7PoHkynWImKO486ZJghEMqPLzZJHD80rvtx3bFjgeH37thJNpkWfbaVTDD1btkVgVKrRkp0NALxvYMju55w+xsOEjtdQRpIyUpSJJg1EQqZZ40QMhjhPkPexmpzhxhFp2qwMXKs5Z2fvEbIkCV425PJz7iDvCeiGzdYWZlhnZutysyirRamDtbT0r5IIBEIBOXYIBCLZIY2GlInjrONHB6dMDNTWs6dsnh07tJlmbaaFMut6ujCfMukokw4GZ0iiyLkKeFeQc/oZD8sfP04ZKNJAkgaKMpAY2SdOnshyIU9QaPHy+qF09iDjmFQ6NSUJN/BKoih4mJA7wLuDvDeoGzpLk51iHJ+uzc7UZmWieCUIRNKCHLt+oL29/cMPP4wxOO/3+wEgIWF4EYgLBowgDAPzDQPzRZYN1o8LNDQx7Q7XwR3aDIs2w0RZDD2NYIcThDbdrE03A4DEC5yrgHMHeU+Q9bHe4zWkliANJGmkSD1BaAk1vpcsy2KQ432M4GfJtELckiUSadaJ4/AeRnXpU2RJ5v2M6GM5D8N7A0KAM4yaQ+VaTBela7MzNRnpON2DEIMIBKK/SKLXyk8Hm822cePGGKscOI5rbm5mGEbliVDmib5WBmWeiEHszBPx09koQqs1jRhqGjFUZNhg44TgyaZQm8O15X90ioFONdIpBtKo7TCpGnvxAwDgFKnktIDT694G894g7wlyXoZz8nzbcVxDkAZSJ+l5L49rCYLuboGRLIshXghwYjAkMCE6cyidZTKOsFIpViAJyelU79WlpqbqdLpeN5dCPO8P8T7G4ifsu2rEAKcbNpM0GY2DUjXpqXR6mibVCnGvo7JYLAmZyEaZJ6KBMk8g4gQFKE5GUIBiBKJHyILI2tqYFht7ysZ7vMzxbZRZT1kNtEVHGrSEVtVQkySIgo/hfSzvZwUvKwZDodZjQGCUhsR1BKHBSS2Ja3CCxgFkMcQLDC8ynMhwVEYRodPSFjNlMZEmUz9OX8q8wDOcGOREhuMDITHIiUxIFiTDqEspi4VOtdApVspqoVOteCJ8ccQFDwpQnMygETsEAnHeg5GELi9Hl5cDAFIoxLZND7U7uHZHyOlybS/FSIIyaUmDltBRhJ4mdRpC14O8FDhJ0ClGOuX0IIcsy2JwCO9nOUeA8wY4dyBgY/hgSHSfAJA1GXmUyUJZjdp0A5ViIY1ajCIIDQVE324hlCVZ4gSRE2RekEKCGOKlEC+yvMhyIsvLoqwfPovQaqk8o95soswm0mykzGbSZMDQ3kYE4sICOXYIBOKCAtdo9AV5+oI8AABZ5r2X824P53Lzbq/g8/N+P+cSg2VbMYokaAKnSVxDYgROUCTgGEbg2NkemCxIsiSDJMmiJPKCFBJkThR5XuJE/fBZ2hTcONpAmkyk0UBoaJwiZVkWg6wQCEhsSGBYjmFkRgwe3QoAGInjJIERBEbhOEngFInhgBE4huOAY4BhOIl3uRFEFiVZkkGWQZIBQOIFSZRkXpJEUflfEkQQJADQD5+FkQRp0FFpGtKgJwx60qAnjQbSoCf0erRIDoH4KYAcu3MEx3ELFy5UwvwGAoGmpqbhw4fHqDxq1Kh9+/aFw0iazeZx48aFKxw5ciQyfHGMUkmSrFZr79qGS8ORynvRNrLUbreHQx/12iJZls1m8/jx49VYBD+GX1dpkdPpDEft77VFAGAymcaNGxdeotRrrURRPHr0qEqL4MdUBGosMpvNo0ePDge4V3OdDx065Ha7e2lRVZVSKssyEGAZmD1u3DiRYQXfz0Q2dOzIEY/HE65sMBgie2Vtba3H4wHqrFLlsSEo6nhT48kQi2mUiUvZrMHHjRvVUSsDBWCSBdE85JcjiopEjpNCnCxJtbW1Lpcr3Be6Pm9srQAAQJIkk8k0YuRIQkPjGg2h0x47ccLpCV8rwUzCuOFnVmX1aV9QXlbKM6zmyTEajRdddFHv2ia8LyhLlTAMuxD6QkJ7N0VRKI5d0oIcu3MERVHFxcVhx662tjay23TA7/dXVVXl5eUVFhYqRzrsXcjNzbVareGv0Uo9Hk99fX2Hdbtxto0sLS0tnTp1qsFg6EXbyNIDBw7Mnj1bjUUA0NjYGAqFVFoEAOvXr1+wYIFKi4xG4+7du6dPn67GIgA4duxYa2trTk6OGosUo2bPnq3GIqvV6nQ6m5ubx44dq8YirVZbUlIye/ZspZqa61xdXa1c4V5bBAANDQ0AkJ+fDwCETkvotAAwyGwIBoORbU0RUcEHZaR2Ll27du2CBQsAYIDV1KE0tlaaCMk5ZkPd9u1Tp06N/7xdlh48eNBSWGgsGnTmvAX51rTUyLbxXKuE9AVBENra2kaNGtWLtpGlu3btGj16NEVRvWib8L5QV1dHUVRBQYHKvqBcYZUWBQKBhoaGCRMmqLEIAPbv3z9o0KAOa+N6qpXNZkOZJ5IW5NidIzAMu+++++KsbLPZlixZkpGRkZub22UFq9Ua2c2ildI07XA4OmSwiLNtJOFI5b1oG1mq1WpVWgQAgUCgQwDn3mmlhJtRaZFer9fr9V0a1SPJTU1NkduYeq2VMkCrxiKr1UoQhM/n62xUTyVHBrhXoxVJktEem/gle71eAFDfF8JRitRYZDablZ9JvWgbWdrQ0NBhM3U/9oVgMMiyrPq+oNFokqcvuFwumqbV94XI4Fa91kqWZY1Go74v1NXVZWRkqOwLNpsNjdglLejGIBAIBAKBQFwgoBG7/sHv98dIdB25wgaBQCAQCAQiTpBj1w9s3bo1vNQsGtdee23kYlUEAoFAIBCIbkGOXT8wffr0ysrKGCnFnE7nr3/96w5rIHpBf2WekGX5qaeeWrNmDU3Ter1+5cqV2dnZGRkZgwcP/sMf/gAAd99994MPPrh8+fJ//vOfFEWNHz/+rbfeivMUKPNEDH4imSd6QWZmpnohkCBlaJoeNmyYejkDBgxInscGZZ7oa2VQ5glEnKDME8nI+Z55YuvWrS+99NLq1aspilq+fHl5efmrr776s5/9rK6uLlxny5YtixcvrqysNBgMDzzwQFZW1hNPPNGPOiMQCAQiTlDmiWQGbZ5AJJ5Zs2atX7+eoiie5xsbGwcOHOh2uy0WS2Sdw4cPKyFUAODmm2/++uuv+0lZBAKBQCAuHNBUbD/Asux3330XYz5R2Txxvg+mPv7442+++ea0adPWrFmzZ88el8s1b948m81WVFT0yiuvXHzxxcuXL7fb7WlpaatWrWppaeksoaWlZfv27R0OulyucePGhWOAIRAIBAKBCIMcu35g3759ixcvjrHGTokMHhlhvNeEk0aceyEvvvjismXLnnrqqdtuu23FihVPPvnkjTfemJKS8vLLL99yyy3/+9//HnvssQULFhgMhrlz53a5wO5f//rXiy++2OFgMBgcPnz4/v37e2nMjyTkyiRKjiRJGIapX6KUVEYllTJKd1MfeSupjEoqZZSwhX13hQOH/xyo+guAUiRpB95gvuS1XsjpEahjxgDFsUta0I3pB6ZOnWq3253ROXr06DXXXJOSkqLyRA6Ho6ysTL3CpaWlSs6MODl8+HB5eTkAUBR1zz33bNy4ccCAAYsXL966dSsA3HLLLZWVlQCwZMmSHTt2bNy4ceTIkeEcG5E88sgjnS/O9ddfX1BQoN6o9evXqxfC8/yGDRvUy6moqGhtbVUvJyFGtbW1KbdPJSUlJSzLqpeTEKNqampqamqSRBmWZUtKStTLKS8vb2trUy8nIUa1trZWVFSol7Nhw4YuQ0Fx7WWiv0H014n+OtFfH2r+IbachBhVXV1dW1urXk5ClAkEAqWlperllJWVhZNVqgFlnkhakGOXpCTkx1Bk6P9zKaempubOO+/0+/0A8MMPP4wbN27NmjULFy5UZp/XrFkzadKk5ubmyZMnB4NBQRBef/31O+64I35lemFCZ2KMmMZPf13haCCjflLKIKNikFRGJZUyiZKDRuySFjQVi0g811133YEDByZPnkxRVGpq6t///vfCwsKNGzc++OCDzz//fGZm5gcffJCXlzdv3ryJEydKknTttdcuWrSov7VGIBAIBOK8Bzl2/UBjY+NLL70UY/MEwzAAkJBprP7ij3/84x//+MfII3/961+Li4uvu+66GHUQCAQCgUCoATl2/UAoFHI6naIoRqvAsiyO4wkZvUcgEAgEAvHTATl2/cCQIUP+/e9/x6hgs9lmzJih1+tVnqi/Mk9EIyHh13U6Hco8EQ2UeSIaKPNEDFDmiWigzBMxQJknkhaUeUIVWq02FAp1WXTixIleO1Xne+aJPmXJkiV+v/9f//pXfyuCQCDOHa7Sm9j6z8NfSevojOuQY9FvoMwTyQwasVPFI4880nln/meffdba2pqQX9IIBAKBQCAQ8YMcO1U8++yzHY7s2rXrlVdeee6551JTU6O1qqioWLRoUYw1dkqRkn8CgUAgfmpwpzb7D74MIAEARhosMz7ob40QiPMG5NglEkEQ7rrrruHDhz/88MMxqg0dOvTRRx+NnHP+OAAAIABJREFUdOxICF1F339cvKJKvAEAvF6vbcfSTN+nAFMBACSh7ctBpHVk6tyNZwmK43iHIOOyyPj2/pGt/xLDSQBcM+Ba04RlIAmt/zJl/9KHUacXXni2300YB2nyrvSW/y7tqq3xCMGI7hfhJSrieULWDyRVLHgU4L6v5aDME30tp9eZJ6SQw752oiwELDPe1w64VlEmZNsSajkT+ts4vr6nYlHH7GtlUBy7pAXdmETy17/+9fDhw2+88QZFUTGqmc3mu+66654I7rjjDlybPtxac8eNU+65555bb7116NCh4bcJ27gWIw1c2w7BezxSTrfH7Sf3dMg84S69SfQeT1+wN+OGmvT55aKv1rNjSTQ9qfRJaVdthU6ZJ3okJJKEhF/3er1er1e9HJR5Ihoo80RfK4MyT0QihVxi4KQUcoj+ExA980RPQZknooEyT1zwIMcuYXg8nj/96U9XX311cXFxrwTglsl/8exYDLIEAJG/EYNH39YPu1s74NrgsfciG3R7XGz4KHJwi3fuD7X+zzrzQ5y2AgBGma0zPzJP+Us0hfj2Mvvai+DsSOU9FRJJosKvqxcCF2gseGTUT0qZ88UotnGdfd0k+7pJju9mSVw3KbDPF6POX2VQ5okLHjQVmzBWrFjhdruXLVsWedDj8YR/G3m9XmVHhSzLPp8vcncFJgY1AtfMj6Kx9JPbnmnVzAsXCd4azrbVOvtTyn2Jq3ShaeJzGE7HeZzZfCMU3HBGlHMflTIWoy1nzksZMQCZ98dvZjQhCAQCEYn/wPOi/yQA8M5K3nF6ME9i25TfhAgEoo9Ajl1iYBjmr3/964wZMy6++OLI49ddd108g+cGDex+FUYXFWVZ4ds/fnf9c88uf2BCQ0PDoXXrCoIfUMTk2opjM2bMwTVpbP0qXeGi3bt30ydepIjJ5RvLAGC8pFeOA0Dw6DsBw+z1G8sAYLSg9Rz9dDtOz5gxAwAAJ2VZ2L17t81mC586NTV1+uTxAGBfdzFgGMuygiCQksumvd5Xpy1kA+kReu7evZs/eSCTbd+xbp3S9rTkH0s7SO5Q2tjYuG7dumilsduGS9va2jrEsYu/bcJLT5061dTUtK5XVyOyNBAI5OfnJ4NFNpuNYRiv12u321VKhgjUaGW328OPTa/tdbvdAOBwOM7llYxWWllZWVdXp74v2Gw2r9d7xRVX9LtFu3fvPnHihN/vV+oopf5Dr8hcxx1gmzdvNmS0R7Y9cOBADgAAHD582Fa7rsvJ5S1btmQzLZGPlOitcf/v5mO6h/rUXpfLheN4dXV1kvSFU6dOqe8Lra2tp06dys3NVWNRjN2BiH4HxbFLDB9//PHtt9/+0Ucf3XbbbZHHfT5fe3u78jkYDIZjDkd+BmXEbtcsdnoFAJCnPhVafygptxcXzx30s5fbPs/DddkYZQIAiWkljIPSriyVRTae40KgJSinFd60VzmL4D5i/+aSzBvrcW2GckQWglzbNjpjetebJ3KLPTvvTV+wr3L1/bn8NziOmyctJy0juhSiyZ3b7VVavXp1ZEqx3nHzzTf7/f5vvvlGpZyEKMNx3KZNm66++mqVcnbv3p2fn5+bm6tSTkKMstlsdXV106ZNUynnhx9+mD17tk6nUyknIUZVV1cDgPpo0glRhmGYLVu2RDpkvWPnzp2FhYXqA4n1ziiJc/P2PQCA01YqfVJLS0tTU9PkyZPDFVo/tXZ27DJuOEqazwrOLHiPt381FADMk/9sGPXgt99+W1xcHDrygn/fn8J10udX+A++HBnHDgAIfV7mwqbEGtWBw4cP0zQ9dOhQlXISoozf7y8rK+vtUp8zbNu2bcSIEenp6d1XjU5VVdXjjz/+7rvvojh2SQgasUsMn3/+OUEQ8+bN63DcZDKZTKZum8u8v20PUVhYCABQ+GTrmnWF5noAYE98RhgGpC/Y+2M1r+3zAsFTzbfviuc442uTVhcJnmrSMgIASOsoTd6V7i2/tM76GNdlS5zLs/1uXJNGZ0yPrV7qRQ+l568IR2DvUkg8jh3KPBENlHkiBijzRDT6N/OE/8ALgUOvKJ+zfuFKbOaJrsO+9wSUeSIaKPPEBQ9y7BIAz/P//e9/J06cmKDRaQwb88pQ+6UiQODo24ZRD54poMz6oXcEj77DtZfFc1xnyuSH3xU8+o558uvKEevsT/37n7Wvnw4gY4RWV3SrceyjssDEVqhDCo0uhcRjWELeSjRNazQa9XISogyO46fdcXXk5OSoFwIJMkqn0+Xl5amXM3jwYPVCIEFGJWrmKCHKEASRkIuTkNsEvTZKjvh9JfEGg9VgMKhXJiEdChJ0p1QOa4VJiDIkSSYkRWTkqg81JGS/MKIvQI5dAjh06FAwGBw7dmyc9T0ezxdffNFpn9QL8N7pza1er/eRR+DgwcWDx3TcbRp20Xp3HCM0ponPmSY+d9ZBypjzq7Nm5C0z/q58SF+wr7PMLoUgEAhEH8Ec/7DDPCwCgYgGcuwSwLFjx6DTsFYM6urqXn311RixmpTYxZGh4xAIBOIni8S297cKCMR5A3LsEoAS0CT+xS4TJkxQlnJHw2az5ebmJmT1TFJFKkeZJ6KBAtz3tRyUeaKv5fQ680RfKJMoOahjxgDFsUta0I1JAL/5zW9kWX7ggQcSKDMhQb0dDkeHzBO9o0PmiV6DMk9EA2WeiAHKPBGNCyPzRAfizzwhy5LEuSTO1XnvLaDME9FBmScueNCIXZKSkB9DyRapHGWe6Gs5yKgYyqgXAugK970y8cuRmFO2T0/viUm9/DtN3pVnlSaTUUmlDMo8ccGDHLt+4NixY0899VSMrsWyLI7jaI0dAoFAxEOo+QeZ9wEAlTmD0KuNB4lAnNcgx64fMBgMhYWFMRy7QCDQ2NiYkPhJCAQCkTx4dz2gLZivH/GbxIoNHHk9cOR1ADCOexJt2Ef8xEGuQz+Ql5f34osvxqhgs9mWLFmSkGhtCATiPEUWQ3z7DsAoOmtmf+uSMELN33Nt24M1H4SPyJL6aMSRJGDSE4E4r0GOXZKSkKDeRqMxIQEthwwZotVq1ctBmSeigTJPxOAnm3mCbfjSveUWAEhfsI9KHd+5Qr9knhA8VYLrEAAInmORx+PPPCHzPt4RdZsFyjwRDZR5AhEnaPFj/8Pb9zi+m9XhYPxbsfj2Mvvai7os0hACsanAV/FY+Ihv3zP+/T2epxg0aFDntxtb/6VzU8cUarFBmSeikZOTk5Co/cmWeSIhURUSlXkiIcknzmnmCYnv+OFs8vLy1KfihR4a5d31gKt0oat0YajprJ2eBoMhIQlUCgsLE7IqP1GZJ6xWa5Iok8DMEwn5oY4yTyQtaMSuH9ixY8f8+fM7rbE781dH2ULlcrnUnwvXpjP1X2gHL+ryF/8ZZAkw5OUjEIhukGWxv1VAIBCxQI7duUbiXMMCz+/620BcCoSoohbrAzr+WI7r9dqsdwEgx/03fehgULK8/Z/KfNtjvOPPnp2/1uRcJjKtgnO/eerf6Mzp3vLfc23bMVyDERrr7H91d0LcMvkvnh2L068p6+y68fbdnl0PELocwjjQPPnPweq3mfrPMUILsmSe+jfSPKTDuXBthqfsPt6+mzQVYppUxRz31ttkISjzXiplvGXaO4CjhwqBuBCQeb9z01WRuaTNU97oR30QCEQ8oL/B55rQyTWkxjLk5vUAEKz5YHj+ONFv8ew0FRcXh1o2eXfXZNxQZWt3zDmUCxgOGCG4DqRc9jWhzwseWxmseoO0jsTplLSrtmA47S3/ffDo25rcubHOJ4uagnnB2o8CVW8aRnUKoYzTgnNvynX/JExFvKMyePzD9Gt2AEawDV95d91nvfTfyrkkmQhULg0efZvKmMbZtmTMrwScdG+9VTEHp0zW/1unmCOF2nFd1OkYlHkiGijAfV/LQZkneiFH4lycbVvkEcETNWVOn2aeYE+uToicnoI6ZgxQHLukBd2Ycw2dPYe3l7tKFwaPrdQWzI90g3hnJZ01C3AKAKtnxyoHCeMgQp8HAIQuRwo5cMoCGOb+789dpTdxpzaLTKy4806nU9lnYJnyRuDwq2KgsXMdXJ9PmIoAgLNtlYJNjh/+z/H9nMCRP0tMW/hcNV/+H9u8SWTaeOdeOmOaMianySmObU5nUOaJaKDMEzFAmSeioSbzhBhs9pT9xrNjiafs/9uw7pPAkddPfYid+hBr+7LQVXKtzPe4i/Vp5glZjPdBCjV959mxxLNjyeHPr/HsWMLZtnSu49//rFKnwzLBzqDMEzFAmSeSFjRid64hjINS5h2sLH1fe2hbyu4n69NfJiR/js+3b9OmdN8xSmwv37TJ5XLJYZ8bi7hHssyc+A97cm3aVaUYofPvf05kYnkD4ZEtXJdjHPekp+w3dPolHepgxI+raDGMzp4TObfL1H2qnKviv9szLVtAdIIsA8g/CpcUczKuq+Lsu7iWTfZ1E1PnbiQtUXeJohQCfS0HGRVDGfVC4Ly9wlLI6dp8bXhS1Th2KWBksPpt5ave8KgYPG2X6D8h+k+IwRbS0sV+W7b+S9HXtaOTJI8N79zHO/cBQAZA8BhghJbOmt2hjm//s8p+FJn3avKvYeo+ZU+uBsCMY5dSaRcnUJkw8T82Mu/zlP1/YUdWP+weTW5xYpVBmScueJBjd65hT645XH1i2lW/lyTp7/fB+j13nmyHl2+Hy2+5/PKL4PfXwjXPrpVk2PhSJs/rOzeXGBtpHYUROinkZBu+otInRZYKrkOEuQgjutgopx++hKn9J9v0rTa/662sdNYs//7npZAD16Rxti2C66Ascsq5CMknNK8lMidTqeOY2k9AEgAnQk3fKOZgpEGTW0xnzuBdB3hHZQzHDoFAJATv7gcF9+Fsj0d26NixD2gH3hC7vsTaIidVeXsFlTGlF+cNNX/Xi1b9jiwEZIkDAAwjMcrUoZQ9+TVb/yUA0JnTOzh28SOFHEzNB7IsYRiuG3onrknrnRwx0MjUfhL+ShoHhx07BCJOkGN3rqFSxgyS/tL2zUwAmTAMuPPP7wmuA56d98ryPpBF97Y7WiceFMis77adGEp3sSNdV7iIKfmP4/vLcG2aaeJz7u1345oz22kd389JvaIkygZYzDL93fa1E6I5dlTqBNOEPzk3XomRJpBFy7S3cU2qcq5cr0hPeIqt/K0m70o6c5r9m0m4Pp+yjuZdB6mUMe7ti/37n1XM0Q64LjGXCdFPeHf9VmrekstxTOYTuqJb+1sdRNewDV+LgZNaADkAXNukbh27nhI4/Jp+5G9xOiWxYvsHWbT9J0sWAgAAGJH585Ox6nIeGSQM12CkHgAI9gTO8bzj9MS08kO3y4acbat3z1LlM2EqSvgdQSDiBzl25xrCVJR25ebII1TG1PQF+wBA5v109hzrzH/Y2uywbbakK6JSL8q47nQQSE3BPE3BPABIn7cr3DbrphYAMF38kvJVP/weQpd1RjRpODH4m3BEMtI6Ouc27qxTR8gHAP3we/XD742soJxr76ZNeTlTrTe1AEDnF1YHcxDnNZxtK3gP0AC8cx9y7LrFt+fR4LGVp79gmOnil/XD7u5XjTojO0sWiL4TPWoTPPZ3wjhIV3RbH+l0LpFl8bRXBwCyyNR8AHLXE6PB6re8ux8EAMCpzBvrZSFgOTgfAOz7T1fQD1tsHPsYYBhhGABYh/0HZyY3OdtWKWRXPmtyLyeM3ccs5FpLBe8xAJCCp3pgGwLRFcixSyIwysQ7KuxVb0iy1ow1szl39lQCmTIe12WHv6LME32tzDnLPBE8+q7/0PLwV8PwXxvGPNy5WkKMShTRMk9IjI2p/RhwWj/8XozoPtZ0MmeeEDzVEncm3qToOx6PkDgzT4Rxb1mkLGnFCF3aFSV45I+3bpGEUOM3sasMGDAAoL7j0R4uw4o/80RsEpV5Ihq+vU9FK5I454+feJn3dc51Fjy2UvHjzZOWawcvklgbAABGUqnjIqspWWsVDKMeNE/+c7fPsHvbnaK/G+cbZZ5AxAly7M4RkiQtX77c7XYDQCAQqK+vHz16dFcVTQBXBgKBf/7z6Oq5IXNLi3JUo9GkpZ1ZtOFyuRjmTHCpcKlu8E0dSnEcdzgc8bSNURr2DnvRNrJUr9e39NCizqWdvbreaaW8IlVaxDCMVqtVjOq1RUppZOaJzqWk64Doqwsf8Tbv9KR2cSWHDBmi0qLwhkS/38+re3IikytElmItX8ChpQBAZ81UljTFlpyWlhZ+bHphUWRpl30B49rg5AcgSwRBKIkc6OyfafKu6FJy+C/r6bZRtv3G1srr9WoorOX4TgAADNdYi9LS0ju3xdxu5YjgPeMvumzHQrSIiV1HCe7ivCld/GDw+/0y4Qx7YTk5OVxbfcc6DeulULx7J91uNyvxANBlX4gTj8fjaWmxWCwJXJUfCAT8p05F8zcZhgm2tADDdK7g9XpZxhutYdDv8HwzE2MalK/U9NXGKL98A34f73BEOmRRno2uBxEZhgkvCSRJ0mKxqO8LOI47nU6V7ytAmSeSGOTYnSMEQSgvL1fidAQCgZaWlhiRIDiOS09Pt9vt4d9VBoMhspvZ7fbIkB8/qdJgMMhxZ00oJ4NWfV3aIYoMwzCuU6f64rxaQVBeCoFAwNve3lPJOs8GvXM1AIgEob/oft3gmzu31btdSpImwV2lHHE2nXILp03UeLcY5GZNRoY2/2oyZax6izqU6l3fhPeSe+x2licl0pJy8k0AEAH8AACg9VRr8q6IR3Iqy3b5B10pxQWX3r0OZDlE04ahs7UFCwCAay11nzgAp1Zrvf9VKrtGrExLu7tDWwDQu92ds1m53W63eCpLFLsMRNaFzl05doJtk2ivCA+Wcm07xLMTvwKA7CwLOMu6OkkXOJ1ODxP1WsWJ1+sNnDrVu7bRCAQCnlOncqOUivat4ta5NHNY+SqLZ4boXC6X3+eMNswbDAYJgQnfArezzZib3mVNxtPory9Ntc4LD06H7xEmegHAYMqIYW8wGIz8Gv/Trnd+peePkxH50HweD8fz/oxfiXRu7LbxlEZTGJEMIMfuHEHT9KpVq+KsbLPZsrOzR40aNXLkyC4rDB06NEbzC7s0PT3d7/cnm1Z9Xeo5+y9sampq4cVdb9/rgWRZkniPzPvC+wSHDh1qr9LxDABAVlaW+exZ5ngke8o+CJ7crRzhbBPCjl1kW+b4QXcTAIAS4xoA9AADrz9CWkYCgH3NnbzrgO8kSGy7+ZJXe2bRjwSPvhM6tRkAMnBqyNS3MNoSLmr9ZEI4loQGQANgmb7S09V6+njO63JbWV/UUubEZ+6q06kavM5VIIYUq7XiWXN8+TmpndtCxIWKxFz/mIXUC5InPFEaavzGFWjEcNI85c0udO4qzyzJHIt89QcOvRrFyngpLCzEtRkqhRQUFBhG9XJTajQyMzOLJk5sjTJnSAhOQnCGv8r8mXs5cOBAWcq2Rwl9mJqayjho8cdLW1BQIItMlzV1ng1wcANjfEc/fIlyRLlHbP3nrtKbAECTdyWf+Zwscl02T7UalLl+jNBjhCb+vmBfdw/vqAxGRDmkACiAvLE/1w3u4iL3tJdVVVXFqI/oX5Bjl6QkajIiqSKVo8wT0ThnAe4Dh1cwdZ+ePmnIIfrrASDlsq+0A66Xea+z5Frhx5EbzrbVV/GYJu8KOvtnXevMnHKVLpQFhkodT1pHi0yrFDjJ2fecUcZ/MtSySflMpU/C6VjJ1Dv/YZM5tzL7rBiFa9JO+2eyxJz4TBZ8GK7RFi7C8C6W8bENX4VaNiqfjWMfI+mxEZknOj88PXicwldYYttk3n9mVX4Us8409Ne7ShfGf6JoCO4jHY94jynr7g2jftfrKBvq6dPME+eewJE/Q+Ky4opCx/kZiTs9ABZq/j7U/H1UNareCFS9AQC4Nl0/bIl2+G8pQ09WWHZFqK1MdB8AjNQNWohRpyeFuPadoZNrADDdkNuUn1jdguLYJS3IsUtSEhLU2+FwVFdXz5gxQ6Wc0tLSqVOnqh9+X79+/YIFC1QK8Xq9Pl9XgyT9oQzPc/vW/X7k0HzlqybvCir1ol7IqaioyM/Pz8mJlbcjHqIZxTv3hpo3AIaxdf/hnXs7lEqsAwBExsa1lp5pYi/n7eWCtyaaYyf6G5TQaLyj63wDoaZvlGCHAGAY87B50isAwDZ8FTz+UefKnp334pSRypgaPhKseT9Y8374K5V6kWXa28Gaf8iCP+ybmoLNpHkoYRxEKZG3Jd5b8bjM+yIzXwneGvs3lyizbKS5B1sWOhM89t6hvdsnXvUUocu0fZ7f5WDYT5bW1tampqZLLukYAr2nbNiwobg4YZHb2JNf8+7erPEPHn03UToAwOHDh6d2uaY6biTW7j/wfFULPXneH9XIEX0n2sqeokPHAQDXpGgHXC+FnNypzYHqt7jW/wIARuqN46PuL4kEZZ5IWpBj1w8cOHBg8eLFYpTlzwDA8/zgwYPVuy/JFqn8wkshIDFt+d6/+n50bETfccv0leFSmfNIIQdgBGEcGFtOT6+wLHGir66z5LBRvsonIlfc8/Y93e65ixMx2KysjRM8PZiLCVa9yRx7HwAkwd+lP8S3lwGA4K3pFEXiNLLgZ0+uCR57L/Kgr/JJACCMgzJvPCGx7YLnaODwax0aSiFHeO2UyLRAJyTWHo8JEuf27FhSAGBf8wVGaBLo1UmMTRmbJAwFgFOJEqsgnyvvM9neNgpioLHLVIoqCR59T47YWcI79kLM4fbEWdRRjuA95vh2FkbqM64/Ei3GXiS+yifODHHLEgAEa/7u2/NojFNEA43YJS3IsesH8vLyfv7zn8dw7Px+/8GDB/X6LjJPIJIEwXtc5j1CwBajjrNkPmfbCgApl63W5FzWOeR9r2Hrv2DrvwCAtCs3Y5RZcQt416HBwTLvrhLDmEeD1e9EhuGIjRhoiJaLk2346tRHpHnS8kD1W4Q+T5N/NXvis85jft0iiyFZ7D6EhehviFYkhVxce5S1/LIIAO1fDZM4d+dCz457ImpKnaN4+Cqf6FYxpfGPZ2OiLajqHZ6y3ygfTJNeNo5ZmkDJAOA/+FJiBSIAQGLOijbX7SNEy/F2xtgU2P/k+G4TRmgNox/W5F0BABJjk9g2AJCFYDyOXRdIZ/0lYk+uZer+I0shy7R3NLmXJ0JrxLkGOXb9QFpa2sMPdxGELIzNZluyZEkyLDTpU1atWvXCCy8wDFNUVPTxxx+npPR5mHu24SvP9tPbD9PJ2wF6mydDltpXj+p2zEYWTu9oc22+DjA888YThGGAGGji23cCABAabd7VgJMAQHJNmG0fy6UAgOhvYBq+BEkAANIynDAO1A//NUTxipwl14Ishk+UCRCo2hCo+muPrPHvf05ioueSl0VfxeOyxIm+2i5Tqp8bpJAjcqa4A4KnWhL80UrDhC/U2WAdRilCzRvs30y2zvr4HOfHi6Je0slE9JRcdhXvqOh1vrIIZGX9gywwysypGOi0vyZu3Ftvx7Yvls9eShheWSF6jwNy7M5PkGOH6B9aW1vvu+++ysrKnJycJ5544tlnn12xYkWcbXHeoazK52xbBE81oc8lraPEQHPw6NsgCdqBNxjHnfkBLcti4OBLEu8DANFXFx7HGsy/7SnjO0QWBQCMMsu8L/yXXmLb2ZNrcE26FHKI3og9ctG9OomxsY3rACSJbT9zVJY8O+7V5F/Ft21nTnymHEu7crOygi2l9RXwl3X+Ua+8ZP0HXox2rsh9fGpgaj+KMaKm5NlMWiTO5fh+juIK94qOY3iyyPD2ctF/AiU+RiSKxHYirm0717ZdpRBZCMgQe/dPFwSPvuurfMIiCFMHJCBaMqIvQI5d/+D1emNMxXo8noQE9U7mzBPr1q0rLi5WtgvceeedxcXFnR07SZI8Hk+HgwRBEJ4y54aoPyU7LLqPBi6HgtV/64H2cRBq3ebatlh0H+LtXUwahpq/CzV/F7mL0/W/XwKhBQBtf+cRimeeNGmReb/Mdz9c11ME1oMFI1xzvuOj2BmRD3IRTUSuZ263xAXObt4zowTWLUcq/KNKPRLSO3jGYdDo87KMkfr3NHeFyAW4YPuwwiyJc0lcjx2OJEdg3RB5cxP9xAqMA8Qz64ZlUe3aSpH3c50eJwAI2bZKIScGMMzcMfYhIknAErhMFREn//vf/+bMmdNttW3btqnf0Jq0PPPMMyzLvvTSSwDAsqzZbA6FQh3ifTz99NPLli3r3Hb4AF3Jn4f3kWKSZQru2RXPQcSFjWSZ0vlg7Megc5MePTYJb94LIZ1ldts82nl7euoOci6wHqfy5vZUuHr5sW+rIAgffOv8zcuVWVlqw68gEg4asesHZs2atW/fvhgJT51O59y5c1NTU6NVuADAsDM/KiRJwnG8cxS3Rx55pHPwjmAwWFhYmJeXdy60RCAQCEQnqqqqlt006jcv97ceiK5Ajl0/gOP4+PHjY1Sw2WLttbwwKCgo2Lx5s/K5sbGxoKCgcx2j0XhxlPwKCAQCgUAgOoPi0CQpCcw8kSRCOsiZP3/+5s2blYTWK1euXLRoUZxCZFlOSAi6vjCq10iSlKh0GuqFJEpOUikjSdKF99gklTKoY/a1MomSkyhlUBy7pAXdmH4gGAyuWrXqi+isW7fummuuUf9CcTgcZWXxpvGOQWlpaSCQgLXM69evD3/OyMh49dVXL7vsspEjRzY0NDz66KMxGkZSW1tbXV3dfb2eKNNreJ7fsGGDejkVFRWtra3q5STEqLa2tvLycvVySkpKWLbr8Hg9IiFG1dTU1NRESfx5zpVhWbakpES9nPLy8ra26HFq4iYhRrW2tlZUdJ2DpEds2LCB5xMQVDkhRlVXV9fW1iaJMoFAoLS0VL2csrIyh8PRfb3uQJknkhY0FdsPHDp06Ne//nWMNXaSJM2ZM6fzhtCekmyx4DuwZtApAAAgAElEQVT8oP/FL37xi1/8IkmU6R1JfoX7V5lkM0q9EEBXuO+VuSCNSiplEiUHjdglLcix6wcmT54c+3e2EqDYao2VNB2BQCAQCASiA8jjRiAQCAQCgbhAQI6dWurr6++6666hQ4fqdLrBgwcvXLhw//79/a0UAoFAIBCInyIoQLEqDh48OGPGDJqm77vvviFDhpw8efKtt95qa2v7/vvvL7vssmit6uvrX3755RirLhiG2bFjxxdffDFhwgQ16rEsa7fb8/Pz1QgBgPr6+vz8fJJUO3F//PjxIUPUZqFxu92///3v1Svj9XrNZrNKIbIs+3w+9XKCwSBN00lilCAIoVDIYDColOPz+YxGY+fwhD0lIUaFQiEA0Gg0yaCMLMt+v99kMqmUEwgENBpN8jw2HMfp9Xr1yphMpiR5bFiWxXGcpunuq/a9MpIkBQKBhDw2Wq1WZS5yt9u9devWvXv3ogDFSQhaY6eK559/3ufzbd68+Wc/+5lyZP78+ePGjVu2bFkMxy6etau1tbXqfW6tVqveqwOAhOQlAwD1Xh0AWK1WrVYbY+tJnKh/zwIAhmEJkaP+z6FCQpQhSVK9rwAA6v8CKSTEKPUunUKiHpuEXBz1zrdCUj02CVEmUXISkkoREqQMjuNJ9dicOtXPiRAR0UAjdqqYOnXqrl27WJaN/LNhNpvT0tJOnDjRa7E2my07O/vIkSMjR45MhJoIBAKBQCSMqqqqUaNGtba2ohG7JAStsVOF4ngdO3YmF7Ldbvf5fMghQyAQCAQCce5Bjp0qli5darVab7nllk2bNjU1NZWVld100016vf7pp59WKfmCzzzRa1CA+75WJlFykkoZlHmir+WgjtnXyiRKDso8ccGD1tipYuTIkdu2bbvxxhsvv/xy5UhBQcHGjRunTJmifL311lt37NgBALIst7S0ZGdnEwQRCoXcbneMEWxRFH/1q19t3779+PHjypHU1NQZM2aEK5SXl0cmKohWyrKs2+0eNWpUL9pGlpaWlk6dOtVgMPSibWTprl27Bg4cqMYiAPB6vTqd7oYbblBjEQCsX79+wYIFKi1qaWlpbm5WEt322iIACAQCl156aU5OjhqLFKNycnLUWNTa2sowjM/ny8zMVGNRampqIBCYPXu2skpJzXX+8MMPMzMz1VgEAG63GwAKCwtV9gXlsVFp0Y4dO/bv3x9e/9rr69zW1pafn3/FFVeosQgS1Bfq6uoCgUBGRoYaixSjbrvtNoqi1FgECeoLLpeLIAiz2ayyLzgcDuWxUWNRIBD4+uuvLRaLGosAoLW11Wq15ubmqrEoNTUVZZ5IWtAaO1VUVVVdddVVsiz/7ne/Kyoqamlp+dvf/lZfX//VV18VFxcDwJ49e8KL7Q4dOjR69GgMw0KhUENDw7Bhw6KJ9Xg833777VNPPTVixAjlCEmS4TcdAAiCEJl1J1qpw+GoqamZOXNmL9pGlm7atGnq1KlGo7EXbSNLV69eHX4X9M4iAKitrf3+++8//PBD5SvHccFgcM+ePQUFBUOHDgUADMMWL1784IMPhts+88wz//3vf/V6vSAIb7zxRlVV1bJly3w+n8lkCoVCwWAw7EC/9957Dz74IM/zyjJw5byrV69evnw5y7JFRUWffPJJSkqKUtntdqekpHz22Wfz588HgMcee6yhoeGLL76oqKi499579+zZE/+12rdv38CBA3Nzc3t9j5TPq1evnjdvnpp7xPN8e3t7fX39JZdc0ut7pJRu3rx59uzZOp1OjUUA8PXXX1955ZVqLAIAJZ/YyJEjVfaF1atXX3fddT1te/jw4UsuueTNN9+87777AMDn823ZsiW8vyqyrSRJs2bN2rFjx/vvv68kUI4huby8vKioKC8vT41FAKAYpbJ3NzY2trS0TJw4sRdtI0tLSkrmzp0b3ojav32hurqaoqiioiKVfWH9+vXKYxOj7eeff75s2TJlax3P88FgsKKioqCgYPjw4UrNW2+9dcKECbNmzVK+rlix4j//+Q9N0+PHj3/rrbe0Wm1FRcU999zDcZzRaPzwww8HDx68bNmyNWvW0DSt1+vfeeedwsJCACgrKxs6dOiOHTtefvllhmGKioo+/vhjk8mkaOX3+zMzM51Op/Ki+93vfnfixIl///vfu3fv/u1vf7tr1y5F5+PHjz/++OPvvvsuWmOXhKARO1Xcdddddrv9yJEjAwYMUI784he/GD169K9+9au6ujqapidNmjRp0iSl6Oc//3mcYm022zfffKPVapU/ip2JvQEtXKrVaimKinzjxN+2j0pVWgQAGo3mqquuevjhh5WvTz31lOKfFRQUHDlypHPb8vLyzz77rLq6miCId99998knn1y7du3ChQuVP2ZKc0Wrw4cPf/HFF5EbVEmStNvtDz30UGVlZU5OzhNPPPHss8+uWLEiXAoABEHodLonnniirq5u1apV4avdo2vVIfRA/94jjUZDUVTnO9VfWmEYFu2xiV+y4ij0V1/oULNzX5BlWQnwsWLFCiUVMk3TXVodKZmiqC4lnwOLOpdGe2x6KrnDBF//9gWapru8EX1x3oULFy5cuFD5HPlaO3TokHLQ7/eXlZUpymzZsuXjjz+urKw0GAwPPPDAihUrHn/88VtuueXNN98sLi5+//3316xZM23atL1791ZWVlIUtXz58meeeeaLL74AAIqivF7vAw880OG1pmilTNQqPeWJJ544fvz4qlWraJrWaDQ4jsfoiYjkAc2R9x6/379z585JkyaFvToAMJlMl156aXNz89GjR/tRt58IR48e/fbbbx966CG32x05QxFJWlpaKBTy+XwA4HA4wtOdkc0BgGXZu+66a+XKlR0CaK1bt664uFhpdeedd3711VedT7Fs2bL9+/crrz/liCiKS5YsKS4uHj9+vPIbF/GT4pNPPhk+fLher58yZUp5eblysLm5+ZJLLjEYDLfffrssy7t378YwbOnSpZmZmf/6178AoKqq6umnn37xxRf7VXdEPxPPa+3w4cPK8hgAuPnmm7/++us9e/ZotVplpuiuu+5aunTprFmz1q9fT1EUz/ONjY3hZTAA8MMPP6h8rR04cCDhhiMSBRqx6z0MwwAAy7IdjgeDQQDgOC5aQ1mWDx48GDka3wGn05kgHS9w/vCHP/zpT38iSdLj8bhcrnnz5tlstqKioldeeUVZ9wYAw4YNe/DBB4cMGTJgwACGYbZs2dK5OQAsXbr0jjvu6Lydubm5OTzVlZ+f39LSEh5cUfjqq6/WrVvX2NgYGcX0yJEj//73v8eMGfPxxx8/9NBD27dv76MrgEhCKisrb7/99vvvv//BBx+8/fbbb7rpJmWkZO3ate+9995HH320cuXK8ePHK8sS1q9f/9VXXw0fPlwUxdtvv/2+++6bPn16f1uA6E+ivdaeeeaZcJ2LL754+fLldrs9LS1t1apVLS0tJ06cKCgouP/++w8dOpSSkvLaa68p8Ucff/zxN998c9q0aWvWrAk3P3XqVOzX2quvvrpixYoYr7WXX345ckQDkVzICBUUFhbSNF1VVRU+4nA40tLSTCYTwzDRWpWWlnZ7X4qKirZt26ZSPYZhGhsbVQqRZfnEiRM8z6uXU1NTo16Iy+Vqb2+XZfnYsWODBg1Stq01NDS89957Ho9HluWXXnpp9uzZ4fp79uwZNWqU0+mUZfn1119fsGCBcnzjxo3h5uvXr7/++uuV4waDIdLYZ555ZunSpcpnJdB/uEgZBVy0aNEdd9xx0003KaJkWd65c2dhYaHyub6+3mQydWtUS0uL3+/v1fU4i4Rc4WAw2NTUpF5OXV2dIAjq5STEKIfD4XA4zo0yypDbvn37wkf27t0LAC+88IIsy8oI7vPPP68cfO6555Q6zz333NixY0Oh0MGDBwHgk08+6fZETU1NwWCwt6acISFX2O/3K/6BSmpra0VRVC8nIUa1t7e7XK5zqUyM19qsWbNOnDgRrvnOO+9MmzatuLh4+fLlhYWFH330UWZmpvLC/8tf/nLppZeGa3Ic9+ijj/6///f/lK+NjY1/+MMfYr/WlixZEvu1ZjAYioqKWltbe3E1EH0NcuxU8fXXX+M4npaW9uSTT/7jH/948cUXi4qKAODtt9+O3dDtdjujo0zjHjly5NxYcZ7y7LPPht9NkTQ1NRmNxvDX5cuX33PPPcrnkydPms3mzs0XLVp00UUXTZkyZcqUKTiOT5kypb6+Xil6//33f/nLXyqfq6urhwwZEpasvAF9Ph/HcbNnz37yySeV4zt37hw6dKjyuaGhIXxGxE+E559/HgD27t0ry3IwGGxvb6+oqACAN998U/7RyXv77beVD8pBWZbHjBmjrFQLLwfs9jWCuPCI87UWibJcpKSkZObMmcqR5uZmi8Vy6NCh3bt3K0dqa2sjX0TqX2tGoxEAkGOXnKA1dqq47rrrtmzZMnPmzPfee+/uu+9WfjmtX7/+3nvvjd3QYrGkRCfaugpEJJs2bbrkkkuUz2vWrFm4cKGy7HfNmjXKhpVDhw4xDDNy5Mjdu3crE987d+4cNWpU5+affvrp3r17y8rKlLXJ27ZtGzhw4MGDB1mWnT9//ubNm1taWgBg5cqVykbFDlAUtWrVqk8//fSjjz5SjtTV1dXW1gLAtm3bVCb8RZx3XH755RiGffDBBw0NDXPnzh01alSMdRdh1q5de/DgwX379n355ZcA8Pzzz9988819rywiuYjztdbc3Dx58uRgMCgIwuuvv37HHXfMnDmzpaVFGRTYuHHj+PHja2pq7rzzTr/fDwA//PDDuHHjACBRrzUUhD+ZQWvs1DJjxozIeD+Ic0ZDQ0N4kcfVV1+9cePGcePGabXazMzMDz74AADmzJlTUlIyb9688vLymTNnKlOi77//fufmXTJr1qxt27aNGTPm1VdfveyyyzAMGzNmTDjMSgfS09PXrl07Z86cgQMHUhQ1ceLEV199tbGxsbW19b333kuk2Yik55JLLnn//fdfeOGFDz744KKLLtq0aVM8kXsHDx6sfFDyIOfk5Fit1r5VFJF8xPlaGz9+/Lx58yZOnChJ0rXXXrto0SIMwz799NM77rgjFAoZjcaVK1cOGzbswIEDkydPpigqNTX173//OyTutfb444/feOON5+qqIHoGimPXDzidzn/84x8x0tj7/f4XXnhh//79Y8aMUXkuURQ7hNLoLyGJknN6qDmOoOdPPPHE/fffn52d3XfKJEqOJEkYhnXYkNtfyiRKTlIpo7hW6mPlJ5VRSaVM/B3zHCiTKDlJ2DGfeuqpGK+1c6ZMVVXVmDFjWlpaUBy7JARNxfYDjY2Nn3322RfRWbt27TXXXKPsrlWDw+EoKytTr3BpaWkgEFAvZ/369eqF1NbWKoG+umX8+PExXn8JUYbn+Q0bNqiXU1FRERnYvdckxKi2trZwhA41lJSUdN4z3gsSYlRNTY0SozgZlGFZtqSkRL2c8vLytrY29XISYlRra6uylFAlGzZsiGfmulv+//buP7aq+v4f+OFSCsVSoFoUkF9KEAV/bP6acxHjH26JMWYuy7bERaNjJM4s0Q2d7i/9w8zE+Y8mm5nZYpRkzpn94c9MsyEKWkRkChRRKUKB3lCwve2lv9vPH/f7bTq2e6w9p71nbx6Pv+g9t+/7eveeU54995z3K5VJ7dmzp/ThYxaKKRaLGzdujP+1NhbvvvvusWPHktej80Rm+Si2Ai6++OKtW7fGPCGfz69bt27WrFkJX6j0N3TCQVIcJ5VWkmMv5gc/+EF2ipmccUwqppjkg0R+whNfTJCTSrGY+F9rYx8neT16xWaWNwYAIBCCHQBAIHwUWwF79uy5++67Y26e6Ovrmzt3buk2dQCAMRLsKmDOnDlf//rXS6sT/VfFYvHVV18d3ctlfGpra0tdZRJavnz5jBkzko+T/CbfKIoaGhpiMvEkF1NVVbVy5crk4yxatKiuri75OKlMqq6uLpVmQStWrEi+D0cpTWrevHnJB4lSKqa6unrFihXJx1m8eHF2dpvZs2cnv3s0iqKVK1eWuvwllMqkzjzzzOwUM2PGjOXLlycfZ+nSpaXlhRPauXNn8kGYCJY7yaJ8Pn/WWWft3r3bIpAAZE1TU9MFF1zQ2tpquZMMco0dAEAgBDsAgEAIdhXw9ttvV1VVTSnvrLPOyuVyx48fT/5aMVfyTfIgaY0zPDycyqJQmZrU0NBQKhdFZGpSmSpmaGgovN0mU8U4MCe6mLTGSasY69hlljemAq688so33njj9fL+8pe/3HDDDfX19Qlf6BTvPDEJxeg8EUPniXJ0noih80Q5pc4TycfReSJ47oqtgGnTpl177bUxT8jn8xs2bEj+QtaC/18pxqRi6Dwx0eOY1ClVjM4TwfPGAAAEwhm7ybN169ZCoRBF0cDAQGtr69lnn13umV988cUk1gUABEKwmyQ9PT3f+c53RhLblClfsoLgHXfcsXnz5k8//bT0ZX19/dVXXz2ydevWrfl8fuTLclt7enra29unTJkyju+diK0HDx588cUXE47c0dExc+bMVatWZWFGR44caWlpKU0qycjFYnF00K/gjPL5fHd3d6FQaGtrSzhyNEqSqtra2kZ2m3HPt729PYqiY8eOZeFY2L59+759+5IfC/l8vlAofPvb3674jLZu3drc3NzV1VV6TpKRT7pqsLLHwhdffJHL5fbs2ZORY+HIkSPJj4XW1tYjR44sWLAgyYySXwLOxLFAcRbl8/mrr776xRdfTLhAcU9PT1tbW8ypwTHav3//2WefnXwF9k8//TT5yunt7e0DAwNnnHFGFooZGhrav3//Oeeck3CcI0eO1NXVnXbaaQnHSWVS3d3dx48fX7hwYcJxmpubFy9ePHXq1ITjpDKp0j3myf83SqWYwcHBAwcOLFu2LOE4hw4dqq+vr6mpSThOKpMqFouFQmH+/PkJx9m3b9/SpUuTX7+VyqTa2tqqqqrmzJmThWIGBgZaWlqSNxNqaWk544wzEjYTampquvHGGzdv3myB4gwS7LJI5wkAMkvniSxz8wQAQCBcY1cBhw8ffuKJJ2JugC8tGtfX1zeJRQEA//MEuwro7OzcsWNHzCKcfX19uVwulVU6BwcHk1/klMogaY1TWoQp+SU4mZrU0NBQqelIFopJa5xMFVP6Oyqw3SZTxTgwJ7qYtMZJqxjr2GWWN6YCzjvvvFdeeeVLO08kv5Re54mJLkbniRg6T5Sj80QMnSfK0XmCMXLGLqNS+WMoayuVZ2r59UwVY1IxdJ6Y6HFM6pQqRueJ4HljAAAC4YxdBezYsWPt2rUxfzP19/cvW7as1KYCAGCMBLsKWLp06U9/+tOYYFcoFN5+++3k19gBAKcUwa4C5syZs3bt2pgn5PP53//+98lvXKqtrU2+THkURcuXL0+4THnJ6tWrkw/S0NAwMDCQkWKqqqpWrlyZfJxFixbV1dUlHyeVSdXV1S1evDj5OCtWrKiurk4+TiqTmjdvXvJBopSKqa6uXrFiRfJxFi9enJ3dZvbs2cnvHo2iaOXKlcmb3EQpTerMM8/MTjEzZsxI3r4iiqKlS5fW1tYmH2fnzp3JB2Ei6DyRRTpPAJBZOk9kmTN2ldHe3h4TqTs6OiazGAAgDIJdBbz55pvXXnvtlz6t1LYcAGCMBLsKuOaaa/bt2xdzxq6tre2qq66qr69P/lqZWqncAvflWOB+osfReWKix3FgTnQxaY2j80TwvDEVMGXKlGXLlp1T3pIlS1JZ1FvniYkuRueJGDpPlKPzRAydJ8rReYIxEuyS2r9//2233bZgwYLq6uolS5b84he/6OzsTD6szhOnVDEmFSNTk8pUMSYVI1OTylQxOk8Ez0exiTQ3N19xxRXHjh37/ve/v3r16vfee++xxx7bsmXLpk2bpk2bVu67urq6nn/++Zi/SktLE6fyuwAAOHUIdok88MADbW1tTz311B133FF65Fe/+tUjjzzyhz/84c477yz3XXv27Hn44Ydjctvg4OAll1yi8wQA8JUIdom88sorCxcuvP3220ceeeCBBx5//PFnnnkmJthddtll8Zf75PP5devWzZkzJ81aAYDQCXbjVywWC4XCJZdcMvqeqbq6uuXLl2/fvj3hnUepLOqt88REF6PzRAydJ8rReSKGzhPl6DzBGOk8MX5DQ0PTp09fsWLFrl27Rj/+jW98o7Gx8eDBg2efffb4RtZ5AoDM0nkiy5yxG79cLnf55Zc3NjZ+9NFHF154YenB5ubm0j3/XV1dURT99re/bWxsjKJoeHj4ww8/XLVqVVVVVbFYPHjwYMyZnp6enkWLFn322Wcj1+HNnDlz2bJlI084cODA6HtvbbXVVltttXXStkZkmGCXyL333vvd7373pptueuyxx1atWvXhhx/ee++9ixcv3rdv3/Tp06MoOu200+bOnRtF0fDwcG1t7dy5c6uqqqZNm9be3l56/L/q7u7u6+urrq6uqakpPVIabcToTbbaaquttto6mVtTWaKSCeKj2KSeeOKJ++6778SJE1EU1dbWPvTQQ++///6GDRva29tnz549vjHz+fyCBQt27tyZ/KPYTK1UboH7cixwP9Hj6Dwx0eM4MCe6mLTGSWWQpqam1atXHz582EexGWSBwaTuuuuu1tbWjRs3btq06fDhw3fffffu3bvnz58/7lRXovNEOTpPxNB5ohydJ2LoPFGOzhMxdJ7ILB/FJjU4ODhr1qw1a9aUvvz888937Njx4x//OOGwOk+cUsWYVIy0JpV8kMhPeOKLCXJSmSpG54ngeWMSue+++2pqakZObwwNDd19993Dw8Mxi9hFUfTuu+82NDTUl3feeedFUfTFF19MxhwAgFA4Y5fILbfc8sQTT1x//fW33nrr6aef/uKLL7733nu//OUvr7zyypjvuvjii5988snBwcFyT+jo6HjppZcsUAwAfCWCXSIXXnjhG2+88eCDDz777LPFYvH8888f3V6snJqamptvvjnmCfl8/qWXXkpltU8A4NQh2CV11VVXvfbaa6kPq/NEOTpPxNB5ohydJ2LoPFGOzhMxdJ7ILMudVMDg4OBHH30U81Hs8ePHr7/+ep0nAMggnSeyzBm7Cti8efPIXbQxjh8/PgnFAADBEOwq4JprrikUCjGfJx49evS8886rr6+fzKoAgP91gl1lzJo1K2ZrX19fWksEZWel8rTGscD9RBeT1jiZKkbniYkex4E50cWkNU5axVjHLrO8MRml80Q5Ok/E0HmiHJ0nYug8UY7OEzF0nsgsZ+wqoK2t7U9/+lPMWuRdXV25XC75vZ9ZW6k8U8uvZ6oYk4qh88REj2NSp1QxOk8ET7CrgNbW1hdeeCEmt/X39y9btqy7u3syqwIA/tcJdhWwevXq+E9I8/n8unXr4q/DAwA4iVOpAACBcMYuo3SeKEfniRg6T5Sj80QMnSfK0Xkihs4TmaXzRAXs2rXrnnvuibmctq+vb9OmTY2NjVdcccVkFgYAX0rniSxzxq4CGhoarrjiiphb+ovF4qZNm1I5SQYAnDqcscuifD5/1lln6RULQAY5Y5dlbp7IqBQ7T2RkkLTGGR4eTmVRqExNamhoKJU/sTI1qUwVMzQ0FN5uk6liHJgTXUxa46RVjHXsMssbk1E6T5Sj80QMnSfK0Xkihs4T5eg8EUPnicxyjd0k6enpaWho6OrqGuPz165du3HjxpHbjubNm7dmzZqRre+8805LS8vIl+W29vb2dnZ2DgwMjON7R28dWal8HN87euu+ffuef/75JDOKoqirq6u2tvaCCy5IMqPo/68Fn3BGBw4cyOfzpbd13DOKoqi/v3/hwoUJZ1SaVMIZtbS09PT0FIvFgwcPJpnRvHnzRi9wn6SqfD4/stuMb0ZRFHV2dkZR1NramvBYGDkplWRGW7du/fjjjzs6OpLMKIqiY8eOHTt2bPT/rxU8Fj755JPu7u79+/cnmVEURR0dHaNPklX2WCgUCrlc7oMPPkh4LIw+lznuqoaHh1taWpIfC21tbc3NzYsWLUoyo3nz5jljl13DTJampqZt27Zt27Zt69atL7300rby/v73v9900027d+9O+IpHjx596623klf++uuvd3Z2Jh/nb3/7W/JB9u7du3PnzowU09vb+/LLLycfp7Gx8dChQ8nHSWVSra2tW7ZsST7Oa6+9duLEieTjpDKppqampqamjBRz4sSJ1157Lfk4W7ZsaW1tTT5OKpM6dOhQY2Nj8nFefvnl3t7e5OOkMqmdO3fu3bs3I8V0dna+/vrrycd56623jh49mnCQ3bt333TTTansfqTOGbvJM/YFz/L5/IRWAgAEyalUAIBAOGNXAb29va+++mrMBcIdHR2lTxsTvpDOExNdjM4TMXSeKEfniRg6T5Sj8wRjZB27CnjnnXduvPHGmKUBhoaGOjo6Nm/e/M1vfnMyCwOAL2Uduyxzxq4Crrrqqra2tpgnlBYonjt37qSVBAAEwDV2AACBEOwySueJcoYtcD/BxaQ1TqaK0XliosdxYE50MWmNo/NE8LwxFdDS0vKzn/1sXXnr16+/4YYbent7E76QzhMTXYzOEzF0nihH54kYOk+Uo/MEY+Qauwro6ek5duxYzF+3PT09uVwu+d9VpbUKEw6S4jip/EEfZDEmFSOtSSUfJPITnvhigpxUpopJaxxn7DJLsKuA5cuX//nPf455Qj6fX7du3cyZMyetJAAgABI3AEAgBDsAgED4KLYCtm/f/qMf/SimfcLg4GBVVVVHR0fCF9J5YqKL0Xkihs4T5eg8EUPniXJ0nmCMdJ6ogEKh8Pzzz8fcG1EoFNavX//RRx+l8usAAFKk80SWOWNXAXV1dXfccUfME/L5/Pr166dOnTppJQEAAXCN3X/R399///33T5069bLLLvvPrR0dHffcc5ybkpEAAAjQSURBVM/SpUunT5++YMGCn/zkJ6msQAYAkJAzdidramq65ZZbyq1l2tPTc911123fvv173/ve1772tc8+++zpp5/+xz/+8d57751++uljfInh4eEDBw7EfBTb1taWYueJ5Gf+UhkkrXFKizAl//lkalJDQ0NTpkxJfolSpiaVqWJKC4kFtttkqhgH5kQXk9Y4aRVjHbvM8sb8m0KhcOmll+Zyue3bt0+bNu0/n/C73/1u+/btjzzyyF//+tdf//rXf/zjH5955pnm5uaHH3547K/y5ptvLl269NzyrrzyyhtuuOH48eMJp6PzxEQXo/NEDJ0nytF5IobOE+XoPMEYCXb/ZmBg4M4779yyZUu5m482bNgwa9asn//85yOP/PCHPzz33HM3bNgw9ttQ1qxZs3///s/Ka2xszOVy9fX1CaeTtZXKM7X8eqaKMakYmZpUpooxqRiZmlSmitF5Ing+iv039fX1jz76aLmtvb29H3zwwZo1a05a++Nb3/rW008/3dzcfM4554zlVaZMmbJkyZKYJ5x22mljLBgAYITE/RV8/vnnQ0ND/5nJSo/s27evEkUBAPw/1rEra8aMGatXr962bdvII++///5ll1121113Pf7446Of+eijj65fv/6FF164+eabTxpk48aNe/fuLf17x44dF110US6X6+7u3r9///nnn1/upQuFwpYtWy688ML58+eXHjnpAt6TzqWX2zowMNDb21tbWzuO7x29tVAo1NbW5nK5cXzv6K3t7e2zZ89OMqMoinp7e4eHh0c30h1fVe3t7XPmzEk4o8HBwa6urtIiseOeURRFJ06cmD59+shlneOuqvQTTjKj4eHh/v7+/v7+mTNnJplRabeZNWtW6fOaJD/n9vb20cvwjq+q0tV+NTU1CY+F0m6TcEaDg4OdnZ0jkxr3z/nEiRPV1dWjV4Gu4LHQ19dX2m2SzCiKos7OzpHdZtwzilI6Frq7u3O53PTp0xMeCx0dHaXdJsmMhoaGurq6Rq8tPL6qisVi6bdNkhm1trZ+8MEHTz75pHXsMshHsSko7fGjD4MRzz777D//+c/Sc0prOeZyud7e3kKh0NDQEDPg+eefv2vXrt27d5ceKRaLH3/88cgTzj333JGEFLO19Cupra1tHN87euuZZ57Z1tY2ODg4ju8dvXXhwoVdXV1JZhRF0bRp03p6ev71r38lmVEURQsWLDh8+HDCGc2ePbu2trY0qXHPKIqi4eHh3bt39/b2JplRaVI1NTUJZzR16tTq6uru7u4kMyoWi+3t7UePHi1dXZTk53zppZeO/k9lfFWVQnN7e3vCY6G02ySc0fLly+fPnz9yQ9K4f841NTXt7e27du1KMqMopWOhvr6+dGwmmVEURVOnTt22bdvIO17ZY6G6unp4eLi/vz/hsdDZ2VnabZLMaOrUqRdffHE0yviqqqmp6e3t7ezsTHgsrFixIpXGJ6ROsPsKSnt2oVA46fHSI6P3+xFPPfXUJBQGABC5xu4rWbJkSVVVVXNz80mPl66uS6WLHwDAuAl2X8G0adMuv/zy999/f/SiboODgxs3blyyZEkqfdMBAMZNsPtqbrvtthMnTjzyyCMjjzz55JOHDx++/fbbK1gVAEDkrtiTvPnmm6+++mrp348++mhDQ8Ott95a+nL9+vWnn356f3//dddd9/bbb994442XXnrpnj17nnvuuYsuumjz5s0WnwMAKkuw+ze/+c1v7r///v+66ZNPPildRVcsFh966KHnnnvuyJEj8+bNu/nmmx988MGRW9kBACpFsAMACIRr7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgRDsAAACIdgBAARCsAMACIRgBwAQCMEOACAQgh0AQCAEOwCAQAh2AACBEOwAAAIh2AEABEKwAwAIhGAHABAIwQ4AIBCCHQBAIAQ7AIBACHYAAIEQ7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgRDsAAACIdgBAARCsAMACIRgBwAQCMEOACAQgh0AQCAEOwCAQAh2AACBEOwAAAIh2AEABEKwAwAIhGAHABAIwQ4AIBCCHQBAIAQ7AIBACHYAAIEQ7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgRDsAAACIdgBAARCsAMACIRgBwAQCMEOACAQgh0AQCAEOwCAQAh2AACBEOwAAAIh2AEABEKwAwAIhGAHABAIwQ4AIBCCHQBAIAQ7AIBACHYAAIEQ7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgRDsAAACIdgBAARCsAMACIRgBwAQCMEOACAQgh0AQCAEOwCAQAh2AACBEOwAAAIh2AEABEKwAwAIhGAHABAIwQ4AIBCCHQBAIAQ7AIBACHYAAIEQ7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgRDsAAACIdgBAARCsAMACIRgBwAQCMEOACAQgh0AQCAEOwCAQAh2AACBEOwAAAIh2AEABEKwAwAIhGAHABAIwQ4AIBCCHQBAIAQ7AIBACHYAAIEQ7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgRDsAAACIdgBAARCsAMACIRgBwAQCMEOACAQgh0AQCAEOwCAQAh2AACBEOwAAAIh2AEABEKwAwAIhGAHABAIwQ4AIBCCHQBAIAQ7AIBACHYAAIEQ7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgRDsAAACIdgBAARCsAMACIRgBwAQCMEOACAQgh0AQCAEOwCAQAh2AACBEOwAAAIh2AEABEKwAwAIhGAHABAIwQ4AIBCCHQBAIAQ7AIBACHYAAIEQ7AAAAiHYAQAEQrADAAiEYAcAEAjBDgAgEIIdAEAgBDsAgEAIdgAAgfg/QMi+IhF6nBgAAAAASUVORK5CYII=",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "### NAAA locus, different genotypes\n",
    "\n",
    "#pdf(file = \"~/HRJ_monocytes/paper/NAAA_plotGardener_aFC_V02.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL1 <- \"4:75897880:A:G\"\n",
    "myeQTL2 <- \"4:75896259:G:A\"\n",
    "BaseSNP= \"4:76816143:G:A\"\n",
    "log2aFC <- 0.3708849\n",
    "myGene = \"NAAA\"\n",
    "mychrom = \"chr4\"\n",
    "mystart = 75874000\n",
    "myend = 75963000\n",
    "###\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL1 | hg38SNP_ID == myeQTL2]\n",
    "### restrict eQTLs to the ones we are interested in\n",
    "myeqtls <- eqtls[V4 %like% myeQTL1 | V4 %like% myeQTL2]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL1 | hg38SNP_ID == myeQTL2]\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "\n",
    "# merge regions if right nect to each other! \n",
    "regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "regions_keep_LHS_merge <- regions_keep_LHS[1:2, ]\n",
    "#LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1)\n",
    "RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1)\n",
    "setnames(RHS_merge, \"V1\", \"chrom2\")\n",
    "both_regions <- cbind(regions_keep_LHS_merge, RHS_merge)\n",
    "#both_regions\n",
    "\n",
    "###\n",
    "aFC <- round(2^log2aFC, digits = 2)\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 10, default.units = \"cm\", showGuides = T)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.3, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 4, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 4.2, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Kb\", fontsize = 7, y = 7.9)\n",
    "\n",
    "\n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/NAAA_DpnII_4605248.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(\n",
    "    data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 6.7,\n",
    "    linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 6.7, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "## text for loops\n",
    "plotText(label = \"NAAA CHi-C \\nsignal, N reads\", fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 6.9, height = 0.5\n",
    ")\n",
    "\n",
    "###\n",
    "\n",
    "### plot pairs and make the height equal to the score\n",
    "### HERE for NAAA using the \"both_regions\" object where I have merged the frags on the right that were adjacent.\n",
    "## REF\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5.2, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T)\n",
    "}\n",
    "\n",
    "## ALT\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5.2, height = (1*aFC), archHeight = (1*aFC), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T)\n",
    "}\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 2.5, y = 5.2, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: \", aFC), x = 2.5, y = 5.8, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"Allelic \\nfold change\", x = 1.3, y = 5.2, height = 0.3, fontsize = 7, fontcolor = \"gray40\", rot = 0, params=params_text)\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"gray40\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead eQTLs\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.8, height = 0.5,\n",
    ")\n",
    "\n",
    "### Plot the contact QTL, TFPT - would it be the same SNP? Yes, would be the same one\n",
    "contact <- BaseBed[Gene == myGene]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3.2, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTLs\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.3, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 3.7, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD variants\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 3.8, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "#plotRanges( \n",
    "#    data = baits[DpnName %like% myeQTL1 | DpnName %like% myeQTL2], params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.7, alpha = 0.4\n",
    "#)\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "## plot ATAC - now using the proper 34 reps consensus data\n",
    "ATACPval <- paste0(\"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus_34reps/consensus_34reps_Pval_\", mychrom, \".bw\")\n",
    "PvalRegion <- readBigwig(file = ATACPval,\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATACPvalPlot <- plotSignal(\n",
    "    data = PvalRegion,\n",
    "    params = params_main, height = 1, y = 0.5,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.8, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATACPvalPlot,\n",
    "    at = c(0,54), label = T, # top P value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0.5, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, REF\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#0072B2\",\n",
    "    rot = 0, params = params_text, y = 0.4, height = 0.5\n",
    ") \n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, ALT\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#b2182b\",\n",
    "    rot = 0, params = params_text, y = 1.4, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_REF_Pileupplot,\n",
    "    at = NULL, label = T, # top pileup value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "annoYaxis(\n",
    "    ATAC_ALT_Pileupplot,\n",
    "    at = NULL, label = T, # top pileup value\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "#dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "arches[arches3]\n",
      "\n",
      "arches[arches4]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges4]\n",
      "\n",
      "text[text2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "3282670"
      ],
      "text/latex": [
       "3282670"
      ],
      "text/markdown": [
       "3282670"
      ],
      "text/plain": [
       "[1] 3282670"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1767716"
      ],
      "text/latex": [
       "1767716"
      ],
      "text/markdown": [
       "1767716"
      ],
      "text/plain": [
       "[1] 1767716"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.53849945318902"
      ],
      "text/latex": [
       "0.53849945318902"
      ],
      "text/markdown": [
       "0.53849945318902"
      ],
      "text/plain": [
       "[1] 0.5384995"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1420"
      ],
      "text/latex": [
       "1420"
      ],
      "text/markdown": [
       "1420"
      ],
      "text/plain": [
       "[1] 1420"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1079.1529041908"
      ],
      "text/latex": [
       "1079.1529041908"
      ],
      "text/markdown": [
       "1079.1529041908"
      ],
      "text/plain": [
       "[1] 1079.153"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "signal[signal1]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "segments[segments1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# NAAA, showing local methylation and the ATAC peak that is affected. For the SNP 4:76817412:G:A\n",
    "pdf(file = \"~/HRJ_monocytes/paper/NAAA_plotGardener_crossPhenotypes_V04.pdf\")\n",
    "\n",
    "### THIS ONE HAS THE CORRECT CODE FOR ATAC\n",
    "\n",
    "### set parameters\n",
    "myeQTL1 <- \"4:75897880:A:G\" #Note our proxy is associated with both eQTLs\n",
    "myeQTL2 <- \"4:75896259:G:A\"\n",
    "BaseSNP= \"4:76817412:G:A\"\n",
    "log2aFC <- 0.2695567\n",
    "myGene = \"NAAA\"\n",
    "mychrom = \"chr4\"\n",
    "mystart = 75884000\n",
    "myend = 75973000\n",
    "###\n",
    "\n",
    "###\n",
    "aFC <- round(2^log2aFC, digits = 2)\n",
    "###\n",
    "\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL1 | hg38SNP_ID == myeQTL2]\n",
    "### restrict eQTLs to the ones we are interested in\n",
    "myeqtls <- eqtls[V4 %like% myeQTL1 | V4 %like% myeQTL2]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL1 | hg38SNP_ID == myeQTL2]\n",
    "myproxies_mycontactQTL <- myproxies[hg38Proxy_pos == 75896259]\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "\n",
    "# merge regions if right nect to each other! \n",
    "regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "regions_keep_LHS_merge <- regions_keep_LHS[1:2, ]\n",
    "#LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1)\n",
    "RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1)\n",
    "setnames(RHS_merge, \"V1\", \"chrom2\")\n",
    "both_regions <- cbind(regions_keep_LHS_merge, RHS_merge)\n",
    "#both_regions\n",
    "\n",
    "### Get the ATAC peak that is affected.\n",
    "#mypeakStart <- 75899810\n",
    "#mypeakEnd <- 75902380\n",
    "#log2aFC_at <- 0.3901208\n",
    "### \n",
    "### prepare ATAC peaks\n",
    "myATAC <- unique(atac_plot[BaseQTL_SNP_hg19 == BaseSNP])\n",
    "#myATAC[at_log_mean_aFC < 0] # none\n",
    "atacUP <- unique(myATAC[, .(chrom, at_featStart, at_featEnd)])\n",
    "\n",
    "### Get the methylation peak that is affected - NOT PLOTTING NOW\n",
    "methStart <- 75902508\n",
    "methEnd <- 75902509\n",
    "###\n",
    "\n",
    "### prepare h3k27ac peaks\n",
    "myH3K27 <- unique(h3k27ac[BaseQTL_hg19Proxy_ID == BaseSNP])\n",
    "#myH3K27[h3k27ac_beta < 0] # none\n",
    "h3k27acUPhg19 <- unique(myH3K27[, .(chrom, h3k27ac_start, h3k27ac_end)])\n",
    "# convert to hg38\n",
    "h3k27acUP <- data.table(chrom = c(\"chr4\", \"chr4\", \"chr4\"), \n",
    "                        h3k27ac_start = c(75900658, 75912570, 75932155), \n",
    "                        h3k27ac_end = c(75910105, 75932117, 75954921))\n",
    " #                       h3k27ac_end = 39314688)\n",
    "\n",
    "### prepare h3k4me1 peaks\n",
    "myH3K4 <- unique(h3k4me1[BaseQTL_hg19Proxy_ID == BaseSNP])\n",
    "#myH3K4[h3k4me1_beta < 0] # none\n",
    "#myH3K4[h3k4me1_beta > 0] \n",
    "\n",
    "h3k4me1UPhg19 <- unique(myH3K4[h3k4me1_beta > 0, .(chrom, h3k4me1_start, h3k4me1_end)])\n",
    "#h3k4me1DOhg19 <- unique(myH3K4[h3k4me1_beta < 0, .(chrom, h3k4me1_start, h3k4me1_end)])\n",
    "\n",
    "# convert to hg38\n",
    "h3k4me1UP <- data.table(chrom = c(\"chr4\", \"chr4\"), start = c(75898590, 75914051), \n",
    "                        end = c(75913068, 75916835)) \n",
    "\n",
    "\n",
    "##############################################\n",
    "\n",
    "pageCreate(width = 12, height = 8, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "##############################################\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.3, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##############################################\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 3.8, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 4, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.8)\n",
    "\n",
    "### plot affected atac peak(s)\n",
    "plotRanges( \n",
    "    data = atacUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 2.6, height = 0.3\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Affected ATAC\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 2.7, height = 0.5,\n",
    ")\n",
    "\n",
    "### plot affected h3k27ac peaks\n",
    "plotRanges( \n",
    "    data = h3k27acUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3, height = 0.3\n",
    ")\n",
    "# text for H3K27ac\n",
    "plotText(\n",
    "    label = \"Affected H3K27ac\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.1, height = 0.5,\n",
    ")\n",
    "# h3k4me1 can go at 2.2\n",
    "plotRanges( \n",
    "    data = h3k4me1UP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3.4, height = 0.3\n",
    ")\n",
    "\n",
    "# text \n",
    "plotText(\n",
    "    label = \"Affected H3K4me1\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.5, height = 0.5,\n",
    ")\n",
    "# There is no h3k4me1 down\n",
    "#plotRanges( \n",
    "#    data = h3k4me1DO, params = params_main, linecolor=\"fill\", fill = \"blue\", collapse = T,\n",
    "#    y = 3.2, height = 0.3\n",
    "#)\n",
    "# text\n",
    "#plotText(\n",
    "#    label = \"Affected H3K4me1 DOWN\", \n",
    "#    fontsize = 7, fontcolor = \"blue\", rot = 0, params = params_text, y = 3.2, height = 0.5,\n",
    "#)\n",
    "\n",
    "\n",
    "###\n",
    "\n",
    "### plot pairs and make the height equal to the score\n",
    "## REF\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T, style = \"3D\")\n",
    "}\n",
    "\n",
    "## ALT\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5, height = (1*aFC), archHeight = (1*aFC), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T, style = \"3D\")\n",
    "}\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 2.5, y = 5.3, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: ~\", aFC), x = 2.5, y = 6, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"CHi-C allelic \\nfold change\", x = 1.3, y = 5.3, height = 0.3, fontcolor = \"gray40\", fontsize = 7, rot = 0, params=params_text)\n",
    "\n",
    "### Plot the contact QTLs\n",
    "contact <- BaseBed[Gene == myGene]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTLs\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.2, height = 0.5,\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "#plotRanges( \n",
    "#    data = baits[DpnName %like% myeQTL], params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.7, alpha = 0.4\n",
    "#)\n",
    "\n",
    "\n",
    "ATACREF_pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/NAAA/REF_Genrich_merged_NAAA_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "ATACALT_pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/NAAA/ALT_Genrich_merged_NAAA_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "# scale ALT and REF.\n",
    "totalREF <- sum(ATACREF_pileup$score)\n",
    "totalALT <- sum(ATACALT_pileup$score)\n",
    "totalREF\n",
    "totalALT\n",
    "ratio <- totalALT/totalREF\n",
    "ratio\n",
    "ATACREF_pileup$score <- ATACREF_pileup$score*ratio\n",
    "# then height of REF = max(ALT)\n",
    "# and scale accordingly\n",
    "altmax <- max(ATACALT_pileup$score)\n",
    "refmax <- max(ATACREF_pileup$score)\n",
    "altmax\n",
    "refmax\n",
    "\n",
    "ATACPvalPlot_REF <- plotSignal(\n",
    "    data = ATACREF_pileup,\n",
    "    params = params_main, height = 1, y = 0,\n",
    "    linecolor = \"#0072B2\", range=c(0,1500) # need to try to show scale\n",
    ")\n",
    "ATACPvalPlot_ALT <- plotSignal(\n",
    "    data = ATACALT_pileup,\n",
    "    params = params_main, height = 1, y = 1,\n",
    "    linecolor = \"#b2182b\", range=c(0,1500)\n",
    ")\n",
    "\n",
    "# plot points at 500 intervals\n",
    "plotSegments(x0=c(2,2), y0=c(0.333, 0.666, 1.333, 1.666), x1=12, y1=c(0.333,0.666, 1.333, 1.666), \n",
    "             params = params_main, linecolor = c(\"gray80\", \"gray80\", \"gray80\", \"gray80\")\n",
    ")\n",
    "             \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATACPvalPlot_REF,\n",
    "    at = NULL, label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "annoYaxis(\n",
    "    ATACPvalPlot_ALT,\n",
    "    at = NULL, label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, REF\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#0072B2\",\n",
    "    rot = 0, params = params_text, y = 0.4, height = 0.5\n",
    ") \n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, ALT\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#b2182b\",\n",
    "    rot = 0, params = params_text, y = 1.4, height = 0.5\n",
    ") \n",
    "\n",
    "#atacUP\n",
    "#REFdt <- as.data.table(ATACREF_pileup)\n",
    "#REFdt_region <- REFdt[start >= 75906970 & end <= 75908720]\n",
    "#ALTdt <- as.data.table(ATACALT_pileup)\n",
    "#ALTdt_region <- ALTdt[start >= 75906970 & end <= 75908720]\n",
    "#max(REFdt_region$score)\n",
    "#max(ALTdt_region$score)\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr4</td><td>219.9288</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr4 & 219.9288\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr4 | 219.9288 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr4     219.9288"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr4</td><td>296.7347</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr4 & 296.7347\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr4 | 296.7347 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr4     296.7347"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.34923082301319"
      ],
      "text/latex": [
       "1.34923082301319"
      ],
      "text/markdown": [
       "1.34923082301319"
      ],
      "text/plain": [
       "[1] 1.349231"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>at_featStart</th><th scope=col>at_featEnd</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr4</td><td>75906970</td><td>75908720</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 3\n",
       "\\begin{tabular}{lll}\n",
       " chrom & at\\_featStart & at\\_featEnd\\\\\n",
       " <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr4 & 75906970 & 75908720\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 3\n",
       "\n",
       "| chrom &lt;chr&gt; | at_featStart &lt;int&gt; | at_featEnd &lt;int&gt; |\n",
       "|---|---|---|\n",
       "| chr4 | 75906970 | 75908720 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom at_featStart at_featEnd\n",
       "1 chr4  75906970     75908720  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr4</td><td>147.5804</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr4 & 147.5804\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr4 | 147.5804 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr4     147.5804"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr4</td><td>190.3282</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr4 & 190.3282\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr4 | 190.3282 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr4     190.3282"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1.28965813237108"
      ],
      "text/latex": [
       "1.28965813237108"
      ],
      "text/markdown": [
       "1.28965813237108"
      ],
      "text/plain": [
       "[1] 1.289658"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### chekc the change in ATAC signal.\n",
    "atacUP1 <- atacUP[1, ]\n",
    "atacUP2 <- atacUP[2, ]\n",
    "ATAC_REF_Pileup_dt <- as.data.table(ATACREF_pileup)\n",
    "region_ref <- ATAC_REF_Pileup_dt[start >= 75899810 & end <= 75902380, c(\"seqnames\", \"score\")]\n",
    "ref <- region_ref[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "ref # 219.9\n",
    "\n",
    "ATAC_ALT_Pileup_dt <- as.data.table(ATACALT_pileup)\n",
    "region_alt <- ATAC_ALT_Pileup_dt[start >= 75899810 & end <= 75902380, c(\"seqnames\", \"score\")]\n",
    "alt <- region_alt[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "alt # 296.7\n",
    "\n",
    "alt$score/ref$score\n",
    "\n",
    "\n",
    "\n",
    "atacUP2\n",
    "ATAC_REF_Pileup_dt <- as.data.table(ATACREF_pileup)\n",
    "region_ref <- ATAC_REF_Pileup_dt[start >= 75906970 & end <= 75908720, c(\"seqnames\", \"score\")]\n",
    "ref <- region_ref[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "ref # 219.9\n",
    "\n",
    "ATAC_ALT_Pileup_dt <- as.data.table(ATACALT_pileup)\n",
    "region_alt <- ATAC_ALT_Pileup_dt[start >= 75906970 & end <= 75908720, c(\"seqnames\", \"score\")]\n",
    "alt <- region_alt[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "alt # 296.7\n",
    "\n",
    "alt$score/ref$score\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 0 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>at_featStart</th><th scope=col>at_featEnd</th><th scope=col>BaseQTL_SNP_hg19</th><th scope=col>chic_log_mean_aFC</th><th scope=col>at_log_mean_aFC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 0 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " chrom & at\\_featStart & at\\_featEnd & BaseQTL\\_SNP\\_hg19 & chic\\_log\\_mean\\_aFC & at\\_log\\_mean\\_aFC\\\\\n",
       " <chr> & <int> & <int> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 0 × 6\n",
       "\n",
       "| chrom &lt;chr&gt; | at_featStart &lt;int&gt; | at_featEnd &lt;int&gt; | BaseQTL_SNP_hg19 &lt;chr&gt; | chic_log_mean_aFC &lt;dbl&gt; | at_log_mean_aFC &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     chrom at_featStart at_featEnd BaseQTL_SNP_hg19 chic_log_mean_aFC\n",
       "     at_log_mean_aFC"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 0 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>h3k27ac_start</th><th scope=col>h3k27ac_end</th><th scope=col>BaseQTL_hg19Proxy_ID</th><th scope=col>chic_log_mean_aFC</th><th scope=col>h3k27ac_beta</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 0 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " chrom & h3k27ac\\_start & h3k27ac\\_end & BaseQTL\\_hg19Proxy\\_ID & chic\\_log\\_mean\\_aFC & h3k27ac\\_beta\\\\\n",
       " <chr> & <int> & <int> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 0 × 6\n",
       "\n",
       "| chrom &lt;chr&gt; | h3k27ac_start &lt;int&gt; | h3k27ac_end &lt;int&gt; | BaseQTL_hg19Proxy_ID &lt;chr&gt; | chic_log_mean_aFC &lt;dbl&gt; | h3k27ac_beta &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     chrom h3k27ac_start h3k27ac_end BaseQTL_hg19Proxy_ID chic_log_mean_aFC\n",
       "     h3k27ac_beta"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "arches[arches3]\n",
      "\n",
      "arches[arches4]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges4]\n",
      "\n",
      "text[text2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "21336240"
      ],
      "text/latex": [
       "21336240"
      ],
      "text/markdown": [
       "21336240"
      ],
      "text/plain": [
       "[1] 21336240"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "1135910"
      ],
      "text/latex": [
       "1135910"
      ],
      "text/markdown": [
       "1135910"
      ],
      "text/plain": [
       "[1] 1135910"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "0.053238527500628"
      ],
      "text/latex": [
       "0.053238527500628"
      ],
      "text/markdown": [
       "0.053238527500628"
      ],
      "text/plain": [
       "[1] 0.05323853"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "signal[signal1]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### SVIL locus, different genotypes and histone modification locations\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/SVIL_plotGardener_crossPhenos_V03.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"10:29803371:C:T\"\n",
    "myGene = \"SVIL\"\n",
    "BaseSNP <- \"10:30092300:C:T\"\n",
    "mychrom = \"chr10\"\n",
    "mystart = 29500000\n",
    "myend = 29920000\n",
    "log2aFC = 0.9863521\n",
    "###\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "### restrict eQTL locus to my (distal) one\n",
    "proxy_sets_thisone <- proxy_sets[V4 == myeQTL]\n",
    "###\n",
    "### restrict eQTLs to the ones we are interested in\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "myproxies_mycontactQTL <- myproxies[hg38Proxy_pos == 29803371]\n",
    "\n",
    "### Get the ATAC peak that is affected.\n",
    "mypeakStart <- 29802946\n",
    "mypeakEnd <- 29805396\n",
    "### \n",
    "\n",
    "### Get the H3K27ac peaks that are affected\n",
    "start27ac1 <- 29802175\n",
    "start27ac2 <- 29811026\n",
    "start27ac3 <- 29750500\n",
    "end27ac1 <- 29805744\n",
    "end27ac2 <- 29813167\n",
    "end27ac3 <- 29752865\n",
    "###\n",
    "\n",
    "\n",
    "### Get the H3K4me1 peaks that are affected\n",
    "startme1_1 <- 29798790\n",
    "startme1_2 <- 29825092\n",
    "startme1_3 <- 29847491\n",
    "startme1_4 <- 29856821\n",
    "endme1_1 <- 29814327\n",
    "endme1_2 <- 29830969\n",
    "endme1_3 <- 29852391\n",
    "endme1_4 <- 29862027\n",
    "###\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V5, V6)]\n",
    "regions_keep_all <- cbind(regions_keep_RHS, regions_keep_LHS)\n",
    "both_regions <- copy(regions_keep_all)\n",
    "#regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "#LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1000000)\n",
    "#RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1000000)\n",
    "#both_regions <- cbind(RHS_merge, LHS_merge) ### have to put them the other way rounbd for this one\n",
    "###\n",
    "aFC <- round(2^log2aFC, digits = 2)\n",
    "###\n",
    "\n",
    "### prepare ATAC peaks\n",
    "myATAC <- unique(atac_plot[BaseQTL_SNP_hg19 == BaseSNP])\n",
    "myATAC[at_log_mean_aFC < 0] # none\n",
    "atacUP <- unique(myATAC[, .(chrom, at_featStart, at_featEnd)])\n",
    "\n",
    "\n",
    "### prepare h3k27ac peaks\n",
    "myH3K27 <- unique(h3k27ac[BaseQTL_hg19Proxy_ID == BaseSNP])\n",
    "myH3K27[h3k27ac_beta < 0] # none\n",
    "h3k27acUPhg19 <- unique(myH3K27[, .(chrom, h3k27ac_start, h3k27ac_end)])\n",
    "# convert to hg38\n",
    "h3k27acUP <- data.table(chrom = c(\"chr10\", \"chr10\", \"chr10\"), \n",
    "                        h3k27ac_start = c(start27ac1, start27ac2, start27ac3), \n",
    "                        h3k27ac_end = c(end27ac1, end27ac2, end27ac3))\n",
    "\n",
    "### prepare h3k4me1 peaks\n",
    "myH3K4 <- unique(h3k4me1[BaseQTL_hg19Proxy_ID == BaseSNP])\n",
    "#myH3K4[h3k4me1_beta < 0] # none\n",
    "#myH3K4[h3k4me1_beta > 0] \n",
    "\n",
    "h3k4me1UPhg19 <- unique(myH3K4[h3k4me1_beta > 0, .(chrom, h3k4me1_start, h3k4me1_end)])\n",
    "#h3k4me1DOhg19 <- unique(myH3K4[h3k4me1_beta < 0, .(chrom, h3k4me1_start, h3k4me1_end)])\n",
    "\n",
    "# convert to hg38\n",
    "h3k4me1UP <- data.table(chrom = c(\"chr10\", \"chr10\", \"chr10\", \"chr10\"), \n",
    "                        start = c(startme1_1, startme1_2, startme1_3, startme1_4), \n",
    "                        end = c(endme1_1, endme1_2, endme1_3, endme1_4)) \n",
    "\n",
    "\n",
    "######################################\n",
    "\n",
    "\n",
    "pageCreate(width = 12, height = 8, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 3.8, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 4, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 7.2)\n",
    "\n",
    "### plot affected atac peak(s)\n",
    "plotRanges( \n",
    "    data = atacUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 2.6, height = 0.3\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Affected ATAC\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 2.7, height = 0.5,\n",
    ")\n",
    "\n",
    "### plot affected h3k27ac peaks\n",
    "plotRanges( \n",
    "    data = h3k27acUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3, height = 0.3\n",
    ")\n",
    "# text for H3K27ac\n",
    "plotText(\n",
    "    label = \"Affected H3K27ac\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.1, height = 0.5,\n",
    ")\n",
    "# h3k4me1 can go at 2.2\n",
    "plotRanges( \n",
    "    data = h3k4me1UP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3.4, height = 0.3\n",
    ")\n",
    "\n",
    "# text \n",
    "plotText(\n",
    "    label = \"Affected H3K4me1\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.5, height = 0.5,\n",
    ")\n",
    "# There is no h3k4me1 down\n",
    "#plotRanges( \n",
    "#    data = h3k4me1DO, params = params_main, linecolor=\"fill\", fill = \"blue\", collapse = T,\n",
    "#    y = 3.2, height = 0.3\n",
    "#)\n",
    "# text\n",
    "#plotText(\n",
    "#    label = \"Affected H3K4me1 DOWN\", \n",
    "#    fontsize = 7, fontcolor = \"blue\", rot = 0, params = params_text, y = 3.2, height = 0.5,\n",
    "#)\n",
    "\n",
    "\n",
    "###\n",
    "\n",
    "### plot pairs and make the height equal to the score\n",
    "## REF\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T, style = \"3D\")\n",
    "}\n",
    "\n",
    "## ALT\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 5, height = (1*aFC), archHeight = (1*aFC), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T, style = \"3D\")\n",
    "}\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 2.5, y = 5.3, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: ~\", aFC), x = 2.5, y = 6, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"CHi-C allelic \\nfold change\", x = 1.3, y = 5.3, height = 0.3, fontcolor = \"gray40\", fontsize = 7, rot = 0, params=params_text)\n",
    "\n",
    "### Plot the contact QTLs\n",
    "contact <- BaseBed[Gene == myGene]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTLs\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.2, height = 0.5,\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "#plotRanges( \n",
    "#    data = baits[DpnName %like% myeQTL], params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.7, alpha = 0.4\n",
    "#)\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "\n",
    "ATACREF_pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/SVIL/REF_Genrich_merged_SVIL_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "ATACALT_pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/SVIL/ALT_Genrich_merged_SVIL_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "# scale ALT and REF.\n",
    "totalREF <- sum(ATACREF_pileup$score)\n",
    "totalALT <- sum(ATACALT_pileup$score)\n",
    "totalREF\n",
    "totalALT\n",
    "ratio <- totalALT/totalREF\n",
    "ratio\n",
    "ATACREF_pileup$score <- ATACREF_pileup$score*ratio\n",
    "# then height of REF = max(ALT)\n",
    "# and scale accordingly\n",
    "altmax <- max(ATACALT_pileup$score)\n",
    "\n",
    "ATACPvalPlot_REF <- plotSignal(\n",
    "    data = ATACREF_pileup,\n",
    "    params = params_main, height = 1, y = 0,\n",
    "    linecolor = \"#0072B2\", range=c(0,altmax)\n",
    ")\n",
    "ATACPvalPlot_ALT <- plotSignal(\n",
    "    data = ATACALT_pileup,\n",
    "    params = params_main, height = 1, y = 1,\n",
    "    linecolor = \"#b2182b\"\n",
    ")\n",
    "\n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATACPvalPlot_REF,\n",
    "    at = NULL, label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "annoYaxis(\n",
    "    ATACPvalPlot_ALT,\n",
    "    at = NULL, label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "plotText(\n",
    "    label = \"ATAC-seq, REF\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#0072B2\",\n",
    "    rot = 0, params = params_text, y = 0.4, height = 0.5\n",
    ") \n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, ALT\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#b2182b\",\n",
    "    rot = 0, params = params_text, y = 1.4, height = 0.5\n",
    ") \n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 0 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>at_featStart</th><th scope=col>at_featEnd</th><th scope=col>BaseQTL_SNP_hg19</th><th scope=col>chic_log_mean_aFC</th><th scope=col>at_log_mean_aFC</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 0 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " chrom & at\\_featStart & at\\_featEnd & BaseQTL\\_SNP\\_hg19 & chic\\_log\\_mean\\_aFC & at\\_log\\_mean\\_aFC\\\\\n",
       " <chr> & <int> & <int> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 0 × 6\n",
       "\n",
       "| chrom &lt;chr&gt; | at_featStart &lt;int&gt; | at_featEnd &lt;int&gt; | BaseQTL_SNP_hg19 &lt;chr&gt; | chic_log_mean_aFC &lt;dbl&gt; | at_log_mean_aFC &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     chrom at_featStart at_featEnd BaseQTL_SNP_hg19 chic_log_mean_aFC\n",
       "     at_log_mean_aFC"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>h3k4me1_start</th><th scope=col>h3k4me1_end</th><th scope=col>BaseQTL_hg19Proxy_ID</th><th scope=col>chic_log_mean_aFC</th><th scope=col>h3k4me1_beta</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr14</td><td>90546933</td><td>90565013</td><td>14:90560777:T:C</td><td>0.5761226</td><td>1.312785</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " chrom & h3k4me1\\_start & h3k4me1\\_end & BaseQTL\\_hg19Proxy\\_ID & chic\\_log\\_mean\\_aFC & h3k4me1\\_beta\\\\\n",
       " <chr> & <int> & <int> & <chr> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr14 & 90546933 & 90565013 & 14:90560777:T:C & 0.5761226 & 1.312785\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 6\n",
       "\n",
       "| chrom &lt;chr&gt; | h3k4me1_start &lt;int&gt; | h3k4me1_end &lt;int&gt; | BaseQTL_hg19Proxy_ID &lt;chr&gt; | chic_log_mean_aFC &lt;dbl&gt; | h3k4me1_beta &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| chr14 | 90546933 | 90565013 | 14:90560777:T:C | 0.5761226 | 1.312785 |\n",
       "\n"
      ],
      "text/plain": [
       "  chrom h3k4me1_start h3k4me1_end BaseQTL_hg19Proxy_ID chic_log_mean_aFC\n",
       "1 chr14 90546933      90565013    14:90560777:T:C      0.5761226        \n",
       "  h3k4me1_beta\n",
       "1 1.312785    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges4]\n",
      "\n",
      "text[text2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "559534"
      ],
      "text/latex": [
       "559534"
      ],
      "text/markdown": [
       "559534"
      ],
      "text/plain": [
       "[1] 559534"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "2863269"
      ],
      "text/latex": [
       "2863269"
      ],
      "text/markdown": [
       "2863269"
      ],
      "text/plain": [
       "[1] 2863269"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "5.1172386307177"
      ],
      "text/latex": [
       "5.1172386307177"
      ],
      "text/markdown": [
       "5.1172386307177"
      ],
      "text/plain": [
       "[1] 5.117239"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "signal[signal1]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "segments[segments1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "text[text2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### KCNK13 locus, different genotypes and histone modification locations\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/KCNK13_plotGardener_crossPhenos_V03.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"14:90106140:T:C\"\n",
    "myGene = \"KCNK13\"\n",
    "BaseSNP <- \"14:90560777:T:C\"\n",
    "mychrom = \"chr14\"\n",
    "mystart = 90035331\n",
    "myend = 90201804\n",
    "log2aFC = 0.5761226\n",
    "###\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "### restrict eQTL locus to my (distal) one\n",
    "proxy_sets_thisone <- proxy_sets[V4 == myeQTL]\n",
    "###\n",
    "\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "myproxies_mycontactQTL <- myproxies[hg38Proxy_pos == 90094433]\n",
    "\n",
    "### prepare ATAC peaks\n",
    "myATAC <- unique(atac_plot[BaseQTL_SNP_hg19 == BaseSNP])\n",
    "myATAC[at_log_mean_aFC < 0] # none\n",
    "atacUP <- unique(myATAC[, .(chrom, at_featStart, at_featEnd)])\n",
    "\n",
    "### Get the H3K27ac peaks that are affected\n",
    "start27ac1 <- 90059597\n",
    "start27ac2 <- 90086206\n",
    "start27ac3 <- 90092863\n",
    "end27ac1 <- 90064335\n",
    "end27ac2 <- 90089211\n",
    "end27ac3 <- 90094840\n",
    "###\n",
    "### prepare h3k27ac peaks\n",
    "myH3K27 <- unique(h3k27ac[BaseQTL_hg19Proxy_ID == BaseSNP])\n",
    "#myH3K27[h3k27ac_beta < 0] # none\n",
    "h3k27acUPhg19 <- unique(myH3K27[, .(chrom, h3k27ac_start, h3k27ac_end)])\n",
    "# convert to hg38\n",
    "h3k27acUP <- data.table(chrom = c(mychrom, mychrom, mychrom), \n",
    "                        h3k27ac_start = c(start27ac1, start27ac2, start27ac3), \n",
    "                        h3k27ac_end = c(end27ac1, end27ac2, end27ac3))\n",
    "\n",
    "### Get the H3K4me1 peaks that are affected\n",
    "startme1_1 <- 90080589\n",
    "endme1_1 <- 90098669\n",
    "###\n",
    "\n",
    "### prepare h3k4me1 peaks\n",
    "myH3K4 <- unique(h3k4me1[BaseQTL_hg19Proxy_ID == BaseSNP])\n",
    "#myH3K4[h3k4me1_beta < 0]# none\n",
    "h3k4me1UPhg19 <- unique(myH3K4[h3k4me1_beta > 0, .(chrom, h3k4me1_start, h3k4me1_end)])\n",
    "myH3K4\n",
    "# convert to hg38\n",
    "h3k4me1UP <- data.table(chrom = mychrom, start = startme1_1, end =  endme1_1)\n",
    "\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep <- regions[V7 %like% myGene & V7 %like% BaseSNP] # check which proxy!\n",
    "regions_keep_LHS <- regions_keep[, .(chrom1, V5, V6)] # NOTE : for this, had to swap RHS and LHS\n",
    "regions_keep_RHS <- regions_keep[, .(chrom2, V2, V3)]\n",
    "regions_keep_all <- cbind(regions_keep_LHS, regions_keep_RHS)\n",
    "both_regions <- copy(regions_keep_all)\n",
    "# merge regions if right nect to each other! \n",
    "#regions_keep_RHS_Sort <- setorder(regions_keep_RHS, cols = \"V5\")\n",
    "#regions_keep_LHS_merge <- regions_keep_LHS[1:2, ]\n",
    "##LHS_merge <- bedtoolsr::bt.merge(regions_keep_LHS, d = 1)\n",
    "#RHS_merge <- bedtoolsr::bt.merge(regions_keep_RHS_Sort, d = 1)\n",
    "#setnames(RHS_merge, \"V1\", \"chrom2\")\n",
    "#both_regions <- cbind(regions_keep_LHS_merge, RHS_merge)\n",
    "#both_regions\n",
    "\n",
    "\n",
    "aFC <- round(2^log2aFC, digits = 2)\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 8, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "\n",
    "\n",
    "###\n",
    "aFC <- round(2^log2aFC, digits = 2)\n",
    "###\n",
    "\n",
    "\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 3.8, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 4, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.3)\n",
    "\n",
    "### plot affected atac peak(s)\n",
    "plotRanges( \n",
    "    data = atacUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 2.6, height = 0.3\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Affected ATAC\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 2.7, height = 0.5,\n",
    ")\n",
    "\n",
    "### plot affected h3k27ac peaks\n",
    "plotRanges( \n",
    "    data = h3k27acUP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3, height = 0.3\n",
    ")\n",
    "# text for H3K27ac\n",
    "plotText(\n",
    "    label = \"Affected H3K27ac\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.1, height = 0.5,\n",
    ")\n",
    "# h3k4me1 can go at 2.2\n",
    "plotRanges( \n",
    "    data = h3k4me1UP, params = params_main, linecolor=\"fill\", fill = \"tomato3\", collapse = T,\n",
    "    y = 3.4, height = 0.3\n",
    ")\n",
    "# text \n",
    "plotText(\n",
    "    label = \"Affected H3K4me1\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 3.5, height = 0.5,\n",
    ")\n",
    "# There is no h3k4me1 down\n",
    "#plotRanges( \n",
    "#    data = h3k4me1DO, params = params_main, linecolor=\"fill\", fill = \"blue\", collapse = T,\n",
    "#    y = 3.2, height = 0.3\n",
    "#)\n",
    "# text\n",
    "#plotText(\n",
    "#    label = \"Affected H3K4me1 DOWN\", \n",
    "#    fontsize = 7, fontcolor = \"blue\", rot = 0, params = params_text, y = 3.2, height = 0.5,\n",
    "#)\n",
    "\n",
    "\n",
    "###\n",
    "\n",
    "### plot pairs and make the height equal to the score\n",
    "## REF\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 4.5, height = 1, archHeight = 1, linecolor = \"#0072B2\", fill = \"#56B4E9\", flip = T)\n",
    "}\n",
    "\n",
    "## ALT\n",
    "for(n in 1:nrow(both_regions)) {\n",
    "    plotPairsArches(data = both_regions[n, ], params = params_main, y = 4.5, height = (1*aFC), archHeight = (1*aFC), linecolor = \"#b2182b\", fill = \"#D55E00\", flip = T)\n",
    "}\n",
    "# Text to say the fold change\n",
    "plotText(label = \"REF: 1\", x = 2.5, y = 4.8, height = 0.3, fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params=params_text)\n",
    "plotText(label = paste0(\"ALT: ~\", aFC), x = 2.5, y = 5.5, height = 0.3, fontsize = 7, fontcolor = \"#b2182b\", rot = 0, params=params_text)\n",
    "plotText(label = \"CHi-C allelic \\nfold change\", x = 1.3, y = 4.8, height = 0.3, fontcolor = \"gray40\", fontsize = 7, rot = 0, params=params_text)\n",
    "\n",
    "\n",
    "### Plot the contact QTLs\n",
    "contact <- BaseBed[Gene == myGene]\n",
    "plotRanges( \n",
    "    data = contact, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for contact QTLs\n",
    "plotText(\n",
    "    label = \"Contact eQTLs\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.2, height = 0.5,\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "#plotRanges( \n",
    "#    data = baits[DpnName %like% myeQTL], params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.7, alpha = 0.4\n",
    "#)\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "\n",
    "\n",
    "ATACREF_pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/KCNK13/REF_Genrich_merged_KCNK13_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "ATACALT_pileup <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/merge_genos/KCNK13/ALT_Genrich_merged_KCNK13_locus_Pileup.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "# scale ALT and REF.\n",
    "totalREF <- sum(ATACREF_pileup$score)\n",
    "totalALT <- sum(ATACALT_pileup$score)\n",
    "totalREF\n",
    "totalALT\n",
    "ratio <- totalALT/totalREF\n",
    "ratio\n",
    "ATACREF_pileup$score <- ATACREF_pileup$score*ratio\n",
    "# then height of REF = max(ALT)\n",
    "# and scale accordingly\n",
    "altmax <- max(ATACALT_pileup$score)\n",
    "\n",
    "ATACPvalPlot_REF <- plotSignal(\n",
    "    data = ATACREF_pileup,\n",
    "    params = params_main, height = 1, y = 0,\n",
    "    linecolor = \"#0072B2\", range=c(0,3000) # instead of altmax\n",
    ")\n",
    "ATACPvalPlot_ALT <- plotSignal(\n",
    "    data = ATACALT_pileup,\n",
    "    params = params_main, height = 1, y = 1,\n",
    "    linecolor = \"#b2182b\", range=c(0,3000)\n",
    ")\n",
    "\n",
    "# plot points at 500 intervals\n",
    "plotSegments(x0=c(2,2), y0=c(0.333, 0.666, 1.333, 1.666), x1=12, y1=c(0.333,0.666, 1.333, 1.666), \n",
    "             params = params_main, linecolor = c(\"gray80\", \"gray80\", \"gray80\", \"gray80\")\n",
    ")\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATACPvalPlot_REF,\n",
    "    at = NULL, label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 0, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "annoYaxis(\n",
    "    ATACPvalPlot_ALT,\n",
    "    at = NULL, label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, REF\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#0072B2\",\n",
    "    rot = 0, params = params_text, y = 0.4, height = 0.5\n",
    ") \n",
    "\n",
    "plotText(\n",
    "    label = \"ATAC-seq, ALT\\n(scaled pileups)\", fontsize = 7, fontcolor = \"#b2182b\",\n",
    "    rot = 0, params = params_text, y = 1.4, height = 0.5\n",
    ") \n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr14</td><td>69.08272</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr14 & 69.08272\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr14 | 69.08272 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr14    69.08272"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 1 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>seqnames</th><th scope=col>score</th></tr>\n",
       "\t<tr><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr14</td><td>313.1702</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 1 × 2\n",
       "\\begin{tabular}{ll}\n",
       " seqnames & score\\\\\n",
       " <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\t chr14 & 313.1702\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 1 × 2\n",
       "\n",
       "| seqnames &lt;fct&gt; | score &lt;dbl&gt; |\n",
       "|---|---|\n",
       "| chr14 | 313.1702 |\n",
       "\n"
      ],
      "text/plain": [
       "  seqnames score   \n",
       "1 chr14    313.1702"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "4.53326436379435"
      ],
      "text/latex": [
       "4.53326436379435"
      ],
      "text/markdown": [
       "4.53326436379435"
      ],
      "text/plain": [
       "[1] 4.533264"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#atacUP\n",
    "ATAC_REF_Pileup_dt <- as.data.table(ATACREF_pileup)\n",
    "region_ref <- ATAC_REF_Pileup_dt[start >= 90093840 & end <= 90095070, c(\"seqnames\", \"score\")]\n",
    "ref <- region_ref[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "ref # 69.1\n",
    "\n",
    "ATAC_ALT_Pileup_dt <- as.data.table(ATACALT_pileup)\n",
    "region_alt <- ATAC_ALT_Pileup_dt[start >= 90093840 & end <= 90095070, c(\"seqnames\", \"score\")]\n",
    "alt <- region_alt[, lapply(.SD, mean), by=\"seqnames\"]\n",
    "alt # 313.17\n",
    "\n",
    "alt$score/ref$score\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "### MAPK14 locus\n",
    "#pdf(file = \"~/HRJ_monocytes/paper/MAPK14_plotGardenerV02.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"6:36116033:A:T\"\n",
    "myGene = \"MAPK14\"\n",
    "mychrom = \"chr6\"\n",
    "mystart = 36010000\n",
    "myend = 36130000\n",
    "###\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "### restrict eQTLs to mine\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "\n",
    "pageCreate(width = 12, height = 7, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 2.8, x = 1.5, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.2)\n",
    "\n",
    "        \n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/MAPK14_DpnII_5395592.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(\n",
    "    data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 3.5,\n",
    "    linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 5.5, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"MAPK14 CHi-C \\nsignal, N reads\", \n",
    "    fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 3.7, height = 0.5\n",
    ")\n",
    "\n",
    "###\n",
    "\n",
    "## plot loops\n",
    "archplot <- plotPairsArches(\n",
    "    data = loops_eqtl2, params = params_main, y = 4.5, height = 1.5, archHeight = \"arch\", fill = \"#0072B2\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"eGene-eQTL \\nsignificant \\n contacts\", \n",
    "    fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params = params_text, y = 4.8, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead \\neQTLs\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 1.6, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"black\", fill = \"black\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD SNPs\", \n",
    "    fontsize = 7, fontcolor = \"black\", rot = 0, params = params_text, y = 2.3, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "plotRanges( \n",
    "    data = baits, params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.2, alpha = 0.4\n",
    ")\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_main, height = 1, y = 0.5,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.7, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0, 12), label = c(0, 12), # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 275075 × 5</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>chrom</th><th scope=col>position0</th><th scope=col>position</th><th scope=col>hg19_ID</th><th scope=col>set</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 1</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 2</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 3</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 4</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 5</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 6</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 7</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 8</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td> 9</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>10</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>11</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>12</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>13</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>14</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>15</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>16</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>17</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>18</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>19</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>20</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780159</td><td>102780160</td><td>10:104539917:T:G</td><td>21</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 1</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 2</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 3</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 4</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 5</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 6</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 7</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 8</td></tr>\n",
       "\t<tr><td>chr10</td><td>102780703</td><td>102780704</td><td>10:104540461:C:T</td><td> 9</td></tr>\n",
       "\t<tr><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><td>chr9</td><td>93461753</td><td>93461754</td><td>9:96224036:T:C</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93461753</td><td>93461754</td><td>9:96224036:T:C</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93462018</td><td>93462019</td><td>9:96224301:C:A</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93462018</td><td>93462019</td><td>9:96224301:C:A</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93463195</td><td>93463196</td><td>9:96225478:A:T</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93463195</td><td>93463196</td><td>9:96225478:A:T</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93464228</td><td>93464229</td><td>9:96226511:A:T</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93464228</td><td>93464229</td><td>9:96226511:A:T</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93464817</td><td>93464818</td><td>9:96227100:G:A</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93464817</td><td>93464818</td><td>9:96227100:G:A</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93466539</td><td>93466540</td><td>9:96228822:C:T</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93466539</td><td>93466540</td><td>9:96228822:C:T</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93466614</td><td>93466615</td><td>9:96228897:T:A</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93466614</td><td>93466615</td><td>9:96228897:T:A</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93466669</td><td>93466670</td><td>9:96228952:G:A</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93466669</td><td>93466670</td><td>9:96228952:G:A</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93467279</td><td>93467280</td><td>9:96229562:T:C</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93467279</td><td>93467280</td><td>9:96229562:T:C</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93467911</td><td>93467912</td><td>9:96230194:G:A</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93467911</td><td>93467912</td><td>9:96230194:G:A</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93468888</td><td>93468889</td><td>9:96231171:T:C</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93468888</td><td>93468889</td><td>9:96231171:T:C</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93468958</td><td>93468959</td><td>9:96231241:C:T</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93468958</td><td>93468959</td><td>9:96231241:C:T</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93470119</td><td>93470120</td><td>9:96232402:A:G</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93470119</td><td>93470120</td><td>9:96232402:A:G</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93470459</td><td>93470460</td><td>9:96232742:T:C</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93470459</td><td>93470460</td><td>9:96232742:T:C</td><td>8971</td></tr>\n",
       "\t<tr><td>chr9</td><td>93470591</td><td>93470592</td><td>9:96232874:G:T</td><td>8970</td></tr>\n",
       "\t<tr><td>chr9</td><td>93470591</td><td>93470592</td><td>9:96232874:G:T</td><td>8971</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 275075 × 5\n",
       "\\begin{tabular}{lllll}\n",
       " chrom & position0 & position & hg19\\_ID & set\\\\\n",
       " <chr> & <dbl> & <int> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  1\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  2\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  3\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  4\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  5\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  6\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  7\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  8\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G &  9\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 10\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 11\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 12\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 13\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 14\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 15\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 16\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 17\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 18\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 19\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 20\\\\\n",
       "\t chr10 & 102780159 & 102780160 & 10:104539917:T:G & 21\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  1\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  2\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  3\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  4\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  5\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  6\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  7\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  8\\\\\n",
       "\t chr10 & 102780703 & 102780704 & 10:104540461:C:T &  9\\\\\n",
       "\t ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t chr9 & 93461753 & 93461754 & 9:96224036:T:C & 8970\\\\\n",
       "\t chr9 & 93461753 & 93461754 & 9:96224036:T:C & 8971\\\\\n",
       "\t chr9 & 93462018 & 93462019 & 9:96224301:C:A & 8970\\\\\n",
       "\t chr9 & 93462018 & 93462019 & 9:96224301:C:A & 8971\\\\\n",
       "\t chr9 & 93463195 & 93463196 & 9:96225478:A:T & 8970\\\\\n",
       "\t chr9 & 93463195 & 93463196 & 9:96225478:A:T & 8971\\\\\n",
       "\t chr9 & 93464228 & 93464229 & 9:96226511:A:T & 8970\\\\\n",
       "\t chr9 & 93464228 & 93464229 & 9:96226511:A:T & 8971\\\\\n",
       "\t chr9 & 93464817 & 93464818 & 9:96227100:G:A & 8970\\\\\n",
       "\t chr9 & 93464817 & 93464818 & 9:96227100:G:A & 8971\\\\\n",
       "\t chr9 & 93466539 & 93466540 & 9:96228822:C:T & 8970\\\\\n",
       "\t chr9 & 93466539 & 93466540 & 9:96228822:C:T & 8971\\\\\n",
       "\t chr9 & 93466614 & 93466615 & 9:96228897:T:A & 8970\\\\\n",
       "\t chr9 & 93466614 & 93466615 & 9:96228897:T:A & 8971\\\\\n",
       "\t chr9 & 93466669 & 93466670 & 9:96228952:G:A & 8970\\\\\n",
       "\t chr9 & 93466669 & 93466670 & 9:96228952:G:A & 8971\\\\\n",
       "\t chr9 & 93467279 & 93467280 & 9:96229562:T:C & 8970\\\\\n",
       "\t chr9 & 93467279 & 93467280 & 9:96229562:T:C & 8971\\\\\n",
       "\t chr9 & 93467911 & 93467912 & 9:96230194:G:A & 8970\\\\\n",
       "\t chr9 & 93467911 & 93467912 & 9:96230194:G:A & 8971\\\\\n",
       "\t chr9 & 93468888 & 93468889 & 9:96231171:T:C & 8970\\\\\n",
       "\t chr9 & 93468888 & 93468889 & 9:96231171:T:C & 8971\\\\\n",
       "\t chr9 & 93468958 & 93468959 & 9:96231241:C:T & 8970\\\\\n",
       "\t chr9 & 93468958 & 93468959 & 9:96231241:C:T & 8971\\\\\n",
       "\t chr9 & 93470119 & 93470120 & 9:96232402:A:G & 8970\\\\\n",
       "\t chr9 & 93470119 & 93470120 & 9:96232402:A:G & 8971\\\\\n",
       "\t chr9 & 93470459 & 93470460 & 9:96232742:T:C & 8970\\\\\n",
       "\t chr9 & 93470459 & 93470460 & 9:96232742:T:C & 8971\\\\\n",
       "\t chr9 & 93470591 & 93470592 & 9:96232874:G:T & 8970\\\\\n",
       "\t chr9 & 93470591 & 93470592 & 9:96232874:G:T & 8971\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 275075 × 5\n",
       "\n",
       "| chrom &lt;chr&gt; | position0 &lt;dbl&gt; | position &lt;int&gt; | hg19_ID &lt;chr&gt; | set &lt;int&gt; |\n",
       "|---|---|---|---|---|\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  1 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  2 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  3 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  4 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  5 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  6 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  7 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  8 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G |  9 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 10 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 11 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 12 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 13 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 14 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 15 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 16 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 17 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 18 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 19 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 20 |\n",
       "| chr10 | 102780159 | 102780160 | 10:104539917:T:G | 21 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  1 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  2 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  3 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  4 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  5 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  6 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  7 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  8 |\n",
       "| chr10 | 102780703 | 102780704 | 10:104540461:C:T |  9 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| chr9 | 93461753 | 93461754 | 9:96224036:T:C | 8970 |\n",
       "| chr9 | 93461753 | 93461754 | 9:96224036:T:C | 8971 |\n",
       "| chr9 | 93462018 | 93462019 | 9:96224301:C:A | 8970 |\n",
       "| chr9 | 93462018 | 93462019 | 9:96224301:C:A | 8971 |\n",
       "| chr9 | 93463195 | 93463196 | 9:96225478:A:T | 8970 |\n",
       "| chr9 | 93463195 | 93463196 | 9:96225478:A:T | 8971 |\n",
       "| chr9 | 93464228 | 93464229 | 9:96226511:A:T | 8970 |\n",
       "| chr9 | 93464228 | 93464229 | 9:96226511:A:T | 8971 |\n",
       "| chr9 | 93464817 | 93464818 | 9:96227100:G:A | 8970 |\n",
       "| chr9 | 93464817 | 93464818 | 9:96227100:G:A | 8971 |\n",
       "| chr9 | 93466539 | 93466540 | 9:96228822:C:T | 8970 |\n",
       "| chr9 | 93466539 | 93466540 | 9:96228822:C:T | 8971 |\n",
       "| chr9 | 93466614 | 93466615 | 9:96228897:T:A | 8970 |\n",
       "| chr9 | 93466614 | 93466615 | 9:96228897:T:A | 8971 |\n",
       "| chr9 | 93466669 | 93466670 | 9:96228952:G:A | 8970 |\n",
       "| chr9 | 93466669 | 93466670 | 9:96228952:G:A | 8971 |\n",
       "| chr9 | 93467279 | 93467280 | 9:96229562:T:C | 8970 |\n",
       "| chr9 | 93467279 | 93467280 | 9:96229562:T:C | 8971 |\n",
       "| chr9 | 93467911 | 93467912 | 9:96230194:G:A | 8970 |\n",
       "| chr9 | 93467911 | 93467912 | 9:96230194:G:A | 8971 |\n",
       "| chr9 | 93468888 | 93468889 | 9:96231171:T:C | 8970 |\n",
       "| chr9 | 93468888 | 93468889 | 9:96231171:T:C | 8971 |\n",
       "| chr9 | 93468958 | 93468959 | 9:96231241:C:T | 8970 |\n",
       "| chr9 | 93468958 | 93468959 | 9:96231241:C:T | 8971 |\n",
       "| chr9 | 93470119 | 93470120 | 9:96232402:A:G | 8970 |\n",
       "| chr9 | 93470119 | 93470120 | 9:96232402:A:G | 8971 |\n",
       "| chr9 | 93470459 | 93470460 | 9:96232742:T:C | 8970 |\n",
       "| chr9 | 93470459 | 93470460 | 9:96232742:T:C | 8971 |\n",
       "| chr9 | 93470591 | 93470592 | 9:96232874:G:T | 8970 |\n",
       "| chr9 | 93470591 | 93470592 | 9:96232874:G:T | 8971 |\n",
       "\n"
      ],
      "text/plain": [
       "       chrom position0 position  hg19_ID          set \n",
       "1      chr10 102780159 102780160 10:104539917:T:G  1  \n",
       "2      chr10 102780159 102780160 10:104539917:T:G  2  \n",
       "3      chr10 102780159 102780160 10:104539917:T:G  3  \n",
       "4      chr10 102780159 102780160 10:104539917:T:G  4  \n",
       "5      chr10 102780159 102780160 10:104539917:T:G  5  \n",
       "6      chr10 102780159 102780160 10:104539917:T:G  6  \n",
       "7      chr10 102780159 102780160 10:104539917:T:G  7  \n",
       "8      chr10 102780159 102780160 10:104539917:T:G  8  \n",
       "9      chr10 102780159 102780160 10:104539917:T:G  9  \n",
       "10     chr10 102780159 102780160 10:104539917:T:G 10  \n",
       "11     chr10 102780159 102780160 10:104539917:T:G 11  \n",
       "12     chr10 102780159 102780160 10:104539917:T:G 12  \n",
       "13     chr10 102780159 102780160 10:104539917:T:G 13  \n",
       "14     chr10 102780159 102780160 10:104539917:T:G 14  \n",
       "15     chr10 102780159 102780160 10:104539917:T:G 15  \n",
       "16     chr10 102780159 102780160 10:104539917:T:G 16  \n",
       "17     chr10 102780159 102780160 10:104539917:T:G 17  \n",
       "18     chr10 102780159 102780160 10:104539917:T:G 18  \n",
       "19     chr10 102780159 102780160 10:104539917:T:G 19  \n",
       "20     chr10 102780159 102780160 10:104539917:T:G 20  \n",
       "21     chr10 102780159 102780160 10:104539917:T:G 21  \n",
       "22     chr10 102780703 102780704 10:104540461:C:T  1  \n",
       "23     chr10 102780703 102780704 10:104540461:C:T  2  \n",
       "24     chr10 102780703 102780704 10:104540461:C:T  3  \n",
       "25     chr10 102780703 102780704 10:104540461:C:T  4  \n",
       "26     chr10 102780703 102780704 10:104540461:C:T  5  \n",
       "27     chr10 102780703 102780704 10:104540461:C:T  6  \n",
       "28     chr10 102780703 102780704 10:104540461:C:T  7  \n",
       "29     chr10 102780703 102780704 10:104540461:C:T  8  \n",
       "30     chr10 102780703 102780704 10:104540461:C:T  9  \n",
       "⋮      ⋮     ⋮         ⋮         ⋮                ⋮   \n",
       "275046 chr9  93461753  93461754  9:96224036:T:C   8970\n",
       "275047 chr9  93461753  93461754  9:96224036:T:C   8971\n",
       "275048 chr9  93462018  93462019  9:96224301:C:A   8970\n",
       "275049 chr9  93462018  93462019  9:96224301:C:A   8971\n",
       "275050 chr9  93463195  93463196  9:96225478:A:T   8970\n",
       "275051 chr9  93463195  93463196  9:96225478:A:T   8971\n",
       "275052 chr9  93464228  93464229  9:96226511:A:T   8970\n",
       "275053 chr9  93464228  93464229  9:96226511:A:T   8971\n",
       "275054 chr9  93464817  93464818  9:96227100:G:A   8970\n",
       "275055 chr9  93464817  93464818  9:96227100:G:A   8971\n",
       "275056 chr9  93466539  93466540  9:96228822:C:T   8970\n",
       "275057 chr9  93466539  93466540  9:96228822:C:T   8971\n",
       "275058 chr9  93466614  93466615  9:96228897:T:A   8970\n",
       "275059 chr9  93466614  93466615  9:96228897:T:A   8971\n",
       "275060 chr9  93466669  93466670  9:96228952:G:A   8970\n",
       "275061 chr9  93466669  93466670  9:96228952:G:A   8971\n",
       "275062 chr9  93467279  93467280  9:96229562:T:C   8970\n",
       "275063 chr9  93467279  93467280  9:96229562:T:C   8971\n",
       "275064 chr9  93467911  93467912  9:96230194:G:A   8970\n",
       "275065 chr9  93467911  93467912  9:96230194:G:A   8971\n",
       "275066 chr9  93468888  93468889  9:96231171:T:C   8970\n",
       "275067 chr9  93468888  93468889  9:96231171:T:C   8971\n",
       "275068 chr9  93468958  93468959  9:96231241:C:T   8970\n",
       "275069 chr9  93468958  93468959  9:96231241:C:T   8971\n",
       "275070 chr9  93470119  93470120  9:96232402:A:G   8970\n",
       "275071 chr9  93470119  93470120  9:96232402:A:G   8971\n",
       "275072 chr9  93470459  93470460  9:96232742:T:C   8970\n",
       "275073 chr9  93470459  93470460  9:96232742:T:C   8971\n",
       "275074 chr9  93470591  93470592  9:96232874:G:T   8970\n",
       "275075 chr9  93470591  93470592  9:96232874:G:T   8971"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Get sets bed file for Leo's analysis\n",
    "library(data.table)\n",
    "sets <- fread(\"~/HRJ_monocytes/leo_triplets/input/alex_inputs/Helen_SNP_sets.txt\")\n",
    "snplocs <- read.csv(\"~/HRJ_monocytes/leo_triplets/input/alex_inputs/snp_chromlocations.csv\")\n",
    "snplocs <- as.data.table(snplocs)\n",
    "setkey(snplocs, annot)\n",
    "sets_locs <- sets[snplocs, on = c(hg19_ID = \"annot\"), nomatch = NULL]\n",
    "#sets_locs\n",
    "sets_locs[, position0 := position-1]\n",
    "sets_bed <- sets_locs[, .(chrom, position0, position, hg19_ID, set)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "### Example of GUESS for Leo: CBX3 locus. Some evidence in BaseQTL but not sig. SEE NEXT CHUNK FOR FULLER PICTURE\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/CBX3_plotGardenerGUESS.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"7:26095784:C:T\" # hg38SNP ID\n",
    "myGene = \"CBX3\"\n",
    "mychrom = \"chr7\"\n",
    "mystart = 26090000\n",
    "myend = 26218192\n",
    "myset = 8155\n",
    "###\n",
    "# hg19 proxy ID 7:26135404:C:T (Also the lead eQTL then)\n",
    "# hg38 proxy ID 7:26095784:C:T\n",
    "# ATAC peak chr7 26096710 - 26098550 (Peak_134539 )\n",
    "mypeakstart = 26096710\n",
    "mypeakend = 26098550\n",
    "\n",
    "### restrict loops to my eQTL - eGene combo: now need to\n",
    "loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "### restrict eQTLs to mine\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### Get the GUESS variants for that set.\n",
    "mysetbed <- sets_bed[set == myset]\n",
    "###\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "# pinpoint the ATAC peak as well as the chic interaction\n",
    "\n",
    "pageCreate(width = 16, height = 7, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 2.8, x = 1.5, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 5.2)\n",
    "\n",
    "\n",
    "\n",
    "## plot loops\n",
    "archplot <- plotPairsArches(\n",
    "    data = loops_eqtl, params = params_main, y = 3.5, height = 1.5, fill = \"#0072B2\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"eQTL-eGene\\nCHi-C interaction\", \n",
    "    fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params = params_text, y = 3.8, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"GUESS best \\nmodel SNP\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 1.6, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = mysetbed, params = params_main, linecolor=\"black\", fill = \"black\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"Tested SNPs\", \n",
    "    fontsize = 7, fontcolor = \"black\", rot = 0, params = params_text, y = 2.3, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "#plotRanges( \n",
    "#    data = baits, params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.2, alpha = 0.4\n",
    "#)\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mypeakstart,\n",
    "                        chromend = mypeakend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_main, height = 1, y = 0.5, units = \"cm\",\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC signal at\\npeak (-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.7, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,8), label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\", units = \"cm\"\n",
    ")\n",
    "\n",
    "## annotate with directions:\n",
    "plotText(\n",
    "    label = \"ALT allele associated with:\", fontsize = 8, fontcolor = \"black\",\n",
    "    rot = 0, params = params_main, y = 0, x = 12.5, height = 0.5\n",
    ") \n",
    "plotText(\n",
    "    label = \"Decreased accessibility\\n(beta = -0.292)\", fontsize = 7, fontcolor = \"dodgerblue4\",\n",
    "    rot = 0, params = params_main, y = 1, x = 12.5, height = 0.5\n",
    ")\n",
    "plotText(\n",
    "    label = \"Decreased CBX3 expression\\n(beta = -0.013)\", fontsize = 7, fontcolor = \"dodgerblue4\",\n",
    "    rot = 0, params = params_main, y = 2.5, x = 12.5, height = 0.5\n",
    ")\n",
    "plotText(\n",
    "    label = \"Decreased CHi-C contact\\n(beta = -0.14)\", fontsize = 7, fontcolor = \"dodgerblue4\",\n",
    "    rot = 0, params = params_main, y = 4, x = 12.5, height = 0.5\n",
    ")\n",
    "# have to add arrows manually\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr7</td><td>26095447</td><td>26095795</td><td>chr7</td><td>26200630</td><td>26211694</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 6\n",
       "\\begin{tabular}{llllll}\n",
       " V1 & V2 & V3 & V1 & V2 & V3\\\\\n",
       " <chr> & <int> & <int> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr7 & 26095447 & 26095795 & chr7 & 26200630 & 26211694\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 6\n",
       "\n",
       "| V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| chr7 | 26095447 | 26095795 | chr7 | 26200630 | 26211694 |\n",
       "\n"
      ],
      "text/plain": [
       "  V1   V2       V3       V1   V2       V3      \n",
       "1 chr7 26095447 26095795 chr7 26200630 26211694"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "arches[arches2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges4]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "signal[signal2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Example of GUESS for Leo: CBX3 locus. Some evidence in BaseQTL but not sig. SEE NEXT CHUNK FOR FULLER PICTURE\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/NFE2L3_CBX3_plotGardenerGUESS.pdf\")\n",
    "\n",
    "### Set region parameters\n",
    "mychrom = \"chr7\"\n",
    "mystart = 26090000\n",
    "myend = 26218192\n",
    "###\n",
    "\n",
    "########## Set parameters main and text\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "########################## set parameters for CBX3\n",
    "GUESSvarloc1 = 26095784\n",
    "myGene1 = \"CBX3\"\n",
    "myset1 = 8155\n",
    "mypeakstart1 = 26096710\n",
    "mypeakend1 = 26098550\n",
    "mybait1 = \"5775056\"\n",
    "###\n",
    "mysetbed1 <- sets_bed[set == 8155]\n",
    "\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep1 <- regions[V7 %like% mybait1 & V7 %like% myGene1] \n",
    "regions_keep_LHS1 <- regions_keep1[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS1 <- regions_keep1[, .(chrom2, V5, V6)]\n",
    "# merge regions if right nect to each other! \n",
    "regions_keep_RHS_Sort1 <- setorder(regions_keep_RHS1, cols = \"V5\")\n",
    "regions_keep_LHS_merge1 <- regions_keep_LHS1[1:2, ]\n",
    "RHS_merge1 <- bedtoolsr::bt.merge(regions_keep_RHS_Sort1, d = 1)\n",
    "LHS_merge1 <- bedtoolsr::bt.merge(regions_keep_LHS1, d = 1)\n",
    "both_regions1 <- cbind(LHS_merge1, RHS_merge1)\n",
    "both_regions1\n",
    "\n",
    "\n",
    "\n",
    "###\n",
    "########################## Set parameters for NFE2L3\n",
    "#chr7:26106730-26106731\n",
    "#chr7:26106442-26106443\n",
    "#chr7:26109942-26109943\n",
    "# chic ids:\n",
    "#5775084_ENSG00000050344_NFE2L3, 5775085, 5775087, 5775094, 5775096, 5775100, 5775101, 5775103\n",
    "\n",
    "#Peak_134544, chr7:26104130-26106010\n",
    "\n",
    "# variants and sets just put in manually, too many\n",
    "myGene2 = \"NFE2L3\"\n",
    "mypeakstart2 = 26104130\n",
    "mypeakend2 = 26106010\n",
    "mybait2 = \"5775056\"\n",
    "###\n",
    "\n",
    "### restrict called regions to that in BaseQTL\n",
    "regions_keep2 <- regions[V7 %like% 5775084 & V7 %like% myGene2 | V7 %like% 5775085 & V7 %like% myGene2 | \n",
    "                        V7 %like% 5775087 & V7 %like% myGene2 | V7 %like% 5775094 & V7 %like% myGene2 | \n",
    "                        V7 %like% 5775096 & V7 %like% myGene2 | V7 %like% 5775100 & V7 %like% myGene2 | \n",
    "                        V7 %like% 5775101 & V7 %like% myGene2 | V7 %like% 5775103 & V7 %like% myGene2] \n",
    "regions_keep_LHS2 <- regions_keep2[, .(chrom1, V2, V3)]\n",
    "regions_keep_RHS2 <- regions_keep2[, .(chrom2, V5, V6)]\n",
    "# merge regions if right nect to each other! \n",
    "regions_keep_RHS_Sort2 <- setorder(regions_keep_RHS2, cols = \"V5\")\n",
    "regions_keep_LHS_merge2 <- regions_keep_LHS2[1:2, ]\n",
    "RHS_merge2 <- bedtoolsr::bt.merge(regions_keep_RHS_Sort2, d = 1)\n",
    "LHS_merge2 <- bedtoolsr::bt.merge(regions_keep_LHS2, d = 1)\n",
    "both_regions2 <- cbind(LHS_merge2, RHS_merge2)\n",
    "#both_regions2\n",
    "\n",
    "########################## \n",
    "### Get the GUESS variants for that set.\n",
    "mysetbed2 <- sets_bed[set == 8157 | set == 8158 | set == 8159 | set == 8161 | set == 8162 | set == 8163 | set == 8164 | set == 8165]\n",
    "\n",
    "\n",
    "# Insert NFE2L3 above as well.\n",
    "\n",
    "pageCreate(width = 16, height = 7, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = c(myGene1, myGene2), \"color\" = c(\"chartreuse4\", \"#e69F00\")) # make these different colors, corresponding to SNPs\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 3.5, height = 1)\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 3.8, x = 1.5, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6)\n",
    "\n",
    "####################### LOOPS\n",
    "## plot loops\n",
    "archplot <- plotPairsArches(\n",
    "    data = both_regions1, params = params_main, y = 4.5, height = 1.2, fill = \"chartreuse4\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "\n",
    "## plot loops\n",
    "archplot <- plotPairsArches(\n",
    "    data = both_regions2, params = params_main, y = 4.5, height = 1, fill = \"#e69F00\", linecolor = \"fill\", flip = TRUE, alpha = 0.2\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"Tested contacts\", \n",
    "    fontsize = 7, fontcolor = \"black\", rot = 0, params = params_text, y = 4.8, height = 0.5,\n",
    ")\n",
    "#######################\n",
    "\n",
    "####################### EQTLS\n",
    "\n",
    "# Plot eQTLs\n",
    "GUESS1 <- data.table(chrom = mychrom, \n",
    "                    start = GUESSvarloc1 - 1, \n",
    "                    end= GUESSvarloc1)\n",
    "plotRanges( \n",
    "    data = GUESS1, params = params_main, linecolor=\"chartreuse4\", fill = \"chartreuse4\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"GUESS best model\\nSNP, CBX3\", \n",
    "    fontsize = 7, fontcolor = \"chartreuse4\", rot = 0, params = params_text, y = 1.6, height = 0.5,\n",
    ")\n",
    "GUESS2 <- data.table(chrom = c(mychrom, mychrom, mychrom), \n",
    "                    start = c(26106730, 26106442, 26109942), \n",
    "                    end= c(26106731, 26106443, 26109943))\n",
    "plotRanges( \n",
    "    data = GUESS2, params = params_main, linecolor=\"#e69F00\", fill = \"#e69F00\", collapse = T,\n",
    "    y = 2.5, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"GUESS best model \\nSNP, CBX3\", \n",
    "    fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 2.5, height = 0.5,\n",
    ")\n",
    "\n",
    "#######################\n",
    "\n",
    "####################### SETS\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = mysetbed1, params = params_main, linecolor=\"gray\", fill = \"gray\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"SNP set, CBX3\", \n",
    "    fontsize = 7, fontcolor = \"black\", rot = 0, params = params_text, y = 2.3, height = 0.5\n",
    ")\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = mysetbed2, params = params_main, linecolor=\"gray\", fill = \"gray\", collapse = T,\n",
    "    y = 3, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"SNP set, NFE2L3\", \n",
    "    fontsize = 7, fontcolor = \"black\", rot = 0, params = params_text, y = 3.2, height = 0.5\n",
    ")\n",
    "#######################\n",
    "\n",
    "####################### ATAC\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion1 <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mypeakstart1,\n",
    "                        chromend = mypeakend1)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion1,\n",
    "    params = params_main, height = 1, y = 0.5, units = \"cm\",\n",
    "    linecolor = \"chartreuse4\", baseline = F, baseline.color = \"white\" # cannot seem to remove baseline.\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC signal at\\npeaks (-log10P)\", fontsize = 7, fontcolor = \"black\",\n",
    "    rot = 0, params = params_text, y = 0.7, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,8), label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"chartreuse4\", units = \"cm\"\n",
    ")\n",
    "atac_myregion2 <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mypeakstart2,\n",
    "                        chromend = mypeakend2)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion2,\n",
    "    params = params_main, height = 1, y = 0.5, units = \"cm\",\n",
    "    linecolor = \"#e69F00\", baseline = F, baseline.color = \"white\"\n",
    ")\n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### PTGER4 locus\n",
    "pdf(file = \"~/HRJ_monocytes/paper/PTGER4_plotGardener_V03.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"5:40040641:T:G\"\n",
    "myGene = \"PTGER4\"\n",
    "mychrom = \"chr5\"\n",
    "mystart = 39851080\n",
    "myend = 40747278\n",
    "###\n",
    "\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "### restrict eQTLs to mine\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "pageCreate(width = 12, height = 7, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 2.8, x = 1.5, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.2)\n",
    "\n",
    "        \n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/PTGER4_DpnII_4966107.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(\n",
    "    data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 3.5,\n",
    "    linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 5.5, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"PTGER4 CHi-C \\nsignal, N reads\", \n",
    "    fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 3.7, height = 0.5\n",
    ")\n",
    "\n",
    "###\n",
    "\n",
    "## plot loops\n",
    "plotPairsArches(\n",
    "    data = loops_eqtl, params = params_main, y = 4.5, height = 1.5, fill = \"#0072B2\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"eGene-eQTL \\nsignificant \\n contacts\", \n",
    "    fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params = params_text, y = 4.8, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead \\neQTLs\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 1.6, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD variants\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.3, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "plotRanges( \n",
    "    data = baits, params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.2, alpha = 0.4\n",
    ")\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = paste0(\"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus_34reps/consensus_34reps_Pval_\", mychrom, \".bw\"),\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_main, height = 1, y = 0.5,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.7, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,62), label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### VIM region\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/VIM_plotGardener_V03.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"10:16963755:C:T\"\n",
    "myGene = \"VIM\"\n",
    "mychrom = \"chr10\"\n",
    "mystart = 16871203\n",
    "myend = 17270031\n",
    "###\n",
    "\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "pageCreate(width = 12, height = 7, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 2.8, x = 1.5, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.2)\n",
    "\n",
    "        \n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/VIM_DpnII_628844.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(\n",
    "    data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 3.5,\n",
    "    linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 5.5, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"VIM CHi-C \\nsignal, N reads\", \n",
    "    fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 3.7, height = 0.5\n",
    ")\n",
    "\n",
    "###\n",
    "\n",
    "## plot loops\n",
    "plotPairsArches(\n",
    "    data = loops_eqtl, params = params_main, y = 4.5, height = 1.5, fill = \"#0072B2\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"eGene-eQTL \\nsignificant \\n contacts\", \n",
    "    fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params = params_text, y = 4.8, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead \\neQTLs\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 1.6, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD variants\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.3, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "plotRanges( \n",
    "    data = baits, params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.2, alpha = 0.4\n",
    ")\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = paste0(\"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus_34reps/consensus_34reps_Pval_\", mychrom, \".bw\"),\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_main, height = 1, y = 0.5,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.7, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,72), label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "genes[genes1]\n",
      "\n",
      "text[text1]\n",
      "\n",
      "Warning message:\n",
      "“Start label is rounded.”\n",
      "Warning message:\n",
      "“End label is rounded.”\n",
      "genomeLabel[genomeLabel1]\n",
      "\n",
      "signal[signal1]\n",
      "\n",
      "yaxis[yaxis1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "arches[arches1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges1]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "ranges[ranges3]\n",
      "\n",
      "signal[signal2]\n",
      "\n",
      "text[text2]\n",
      "\n",
      "yaxis[yaxis2]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<strong>png:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{png:} 2"
      ],
      "text/markdown": [
       "**png:** 2"
      ],
      "text/plain": [
       "png \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### SGK1 locus\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/SGK1_plotGardener_V03.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"6:134721003:G:A\"\n",
    "myGene = \"SGK1\"\n",
    "mychrom = \"chr6\"\n",
    "mystart = 134011126\n",
    "myend = 134850000\n",
    "###\n",
    "\n",
    "### restrict loops to my eQTL - eGene combo\n",
    "loops_eqtl <- eQTL_eGene_score5[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "### restrict eQTLs to the ones we are intersted in?\n",
    "myeqtls <- eqtls[V4 %like% myeQTL]\n",
    "### restrict proxies to the ones we are interested in\n",
    "myproxies <- proxies_plot[hg38SNP_ID == myeQTL]\n",
    "\n",
    "\n",
    "pageCreate(width = 12, height = 7, default.units = \"cm\", showGuides = F)\n",
    "\n",
    "########## Set parameters\n",
    "params_main <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "\n",
    "params_text <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.5, width = 10, just = c(\"right\", \"top\"), default.units = \"cm\")\n",
    "##########\n",
    "\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_main, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "plotText(\n",
    "    label = \"Genes\", \n",
    "    fontsize = 7, fontcolor = \"grey48\", rot = 0, params = params_text,\n",
    "    y = 2.8, x = 1.5, height = 0.5\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_main, scale = \"Mb\", fontsize = 7, y = 6.2)\n",
    "\n",
    "        \n",
    "### NEW: Get plotBaits for the eGene\n",
    "plotbaits <- fread(\"~/HRJ_monocytes/CHiC/chicago/findings/bedgraphs/SGK1_DpnII_5621688.bg\")\n",
    "setnames(plotbaits, \"N\", \"score\")\n",
    "plotBaitsPlot <- plotSignal(\n",
    "    data = plotbaits, binSize = 50, binCap = F, \n",
    "    params = params_main, height = 1, y = 3.5,\n",
    "    linecolor = \"#e69F00\", fill = \"#e69F00\"\n",
    ")\n",
    "annoYaxis(plotBaitsPlot, at = NULL, label = T, main = TRUE, axisLine = T, params = params_main, y = 5.5, height = 0.5, fontsize = 7, fontcolor = \"#37a7db\")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"SGK1 CHi-C \\nsignal, N reads\", \n",
    "    fontsize = 7, fontcolor = \"#e69F00\", rot = 0, params = params_text, y = 3.7, height = 0.5\n",
    ")\n",
    "\n",
    "###\n",
    "\n",
    "## plot loops\n",
    "plotPairsArches(\n",
    "    data = loops_eqtl, params = params_main, y = 4.5, height = 1.5, fill = \"#0072B2\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"eGene-eQTL \\nsignificant \\n contacts\", \n",
    "    fontsize = 7, fontcolor = \"#0072B2\", rot = 0, params = params_text, y = 4.8, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = myeqtls, params = params_main, linecolor=\"tomato3\", fill = \"tomato3\", collapse = T,\n",
    "    y = 1.7, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"Lead \\neQTLs\", \n",
    "    fontsize = 7, fontcolor = \"tomato3\", rot = 0, params = params_text, y = 1.6, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot proxies\n",
    "plotRanges( \n",
    "    data = myproxies, params = params_main, linecolor=\"fill\", fill = \"gray40\", collapse = T,\n",
    "    y = 2.2, height = 0.3\n",
    ")\n",
    "# text for proxies\n",
    "plotText(\n",
    "    label = \"LD variants\", \n",
    "    fontsize = 7, fontcolor = \"gray40\", rot = 0, params = params_text, y = 2.3, height = 0.5\n",
    ")\n",
    "\n",
    "## Add baits as a highlight\n",
    "plotRanges( \n",
    "    data = baits, params = params_main, fill = \"gray\", collapse = T, y = 0, height = 6.2, alpha = 0.4\n",
    ")\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = paste0(\"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus_34reps/consensus_34reps_Pval_\", mychrom, \".bw\"),\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_main, height = 1, y = 0.5,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0, params = params_text, y = 0.7, height = 0.5\n",
    ") \n",
    "\n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,68), label = T, # top -log10P val\n",
    "    main = TRUE, axisLine = T, params = params_main, y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "### NAAA locus\n",
    "pdf(file = \"~/HRJ_monocytes/paper/NAAA_plotGardener.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL1 <- \"4:75897880:A:G\"\n",
    "myeQTL2 <- \"4:75896259:G:A\"\n",
    "myGene = \"NAAA\"\n",
    "mychrom = \"chr4\"\n",
    "mystart = 75850000\n",
    "myend = 76000000\n",
    "###\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL1 | hg38SNP_ID == myeQTL2]\n",
    "###\n",
    "### restrict eQTL locus to my (distal) one\n",
    "proxy_sets_thisone <- proxy_sets[V4 == myeQTL1 | V4 == myeQTL2]\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 6, default.units = \"cm\", showGuides = F)\n",
    "params_a <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.7, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_a, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Genes\", fontsize = 7, fontcolor = \"grey48\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.8, x = 0.4, height = 0.5,\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_a, \n",
    "                   scale = \"Mb\", fontsize = 7,\n",
    "                   y = 5.2)\n",
    "## plot loops\n",
    "plotPairsArches(\n",
    "    data = loops_eqtl,\n",
    "    params = params_a,\n",
    "    y = 3.5, height = 1.5,\n",
    "    fill = \"#37a7db\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"CHi-C\", fontsize = 7, fontcolor = \"#37a7db\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 4, x = 0.4, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = proxy_sets_thisone,\n",
    "    params = params_a, linecolor=\"tomato3\", fill = \"tomato3\", \n",
    "    collapse = T,\n",
    "    y = 2.1, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"eQTLs\", fontsize = 7, fontcolor = \"tomato3\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.2, x = 0.4, height = 0.5,\n",
    ")\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_a, height = 1, y = 1,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 1.3, x = 0.4, height = 0.5,\n",
    ") \n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,11),\n",
    "    label = c(0, 11), # top -log10P val\n",
    "    main = TRUE, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## THBS1 locus, different genotypes\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/thbs1_plotGardener_genoSpecific.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"15:39312634:C:T\"\n",
    "myGene = \"THBS1\"\n",
    "mychrom = \"chr15\"\n",
    "mystart = 39200000\n",
    "myend = 39615829\n",
    "###\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "### restrcit eQTLs to the one we are testing\n",
    "proxy_sets_thisone <- proxy_sets[V4 == myeQTL]\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 9, default.units = \"cm\", showGuides = F)\n",
    "params_a <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_a, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Genes\", fontsize = 7, fontcolor = \"grey48\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.8, x = 0.4, height = 0.5,\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_a, \n",
    "                   scale = \"Mb\", fontsize = 7,\n",
    "                   y = 7.7)\n",
    "## plot loops\n",
    "#plotPairsArches(\n",
    "#    data = loops_eqtl,\n",
    "#    params = params_a,\n",
    "#    y = 3.5, height = 1.5,\n",
    "#    fill = \"#37a7db\", linecolor = \"fill\", flip = TRUE\n",
    "#)\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "\n",
    "\n",
    "#################### ADD GENO SPECIFIC CHICAGO PLOTBAITS\n",
    "REF <- fread(file = \"~/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/15:39606026:C:T_homozygous_REF/15:39606026:C:T_homozygous_REFallReplicates_Chicago_154237.bg\")\n",
    "setnames(REF, \"V4\", \"score\")\n",
    "CHiC_signal_REF <- plotSignal(\n",
    "    data = REF, binSize = 50, binCap = F, range = c(0,70),\n",
    "    params = params_a, height = 1, y = 3.5,\n",
    "    linecolor = \"#67a9cf\", fill = \"#67a9cf\"\n",
    ")\n",
    "## Bring in chic bw data from a bait for the SNP\n",
    "ALT <- fread(file = \"~/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/15:39606026:C:T_homozygous_ALT/15:39606026:C:T_homozygous_ALTallReplicates_Chicago_154237.bg\")\n",
    "setnames(ALT, \"V4\", \"score\")\n",
    "CHiC_signal_ALT <- plotSignal(\n",
    "    data = ALT, binSize = 50, binCap = F, range = c(0,70),\n",
    "    params = params_a, height = 1, y = 4.7,\n",
    "    linecolor = \"#b2182b\", fill = \"#b2182b\"\n",
    ")\n",
    "\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"THBS1 CHi-C\", fontsize = 7, fontcolor = \"black\",\n",
    "    rot = 90,\n",
    "    params = params_a,\n",
    "    y = 6.5, x = 0.3, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom REF \\nN reads\", fontsize = 7, fontcolor = \"#67a9cf\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 4, x = 0.6, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom ALT \\nN reads\", fontsize = 7, fontcolor = \"#b2182b\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 5, x = 0.6, height = 0.5,\n",
    ")\n",
    "\n",
    "# range for loops\n",
    "annoYaxis(\n",
    "    CHiC_signal_REF,\n",
    "    at = NULL,\n",
    "    label = T, \n",
    "    main = T, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 3.5, height = 1, fontsize = 7, fontcolor = \"#67a9cf\"\n",
    ")\n",
    "annoYaxis(\n",
    "    CHiC_signal_ALT,\n",
    "    at = NULL,\n",
    "    label = T, \n",
    "    main = T, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 4.7, height = 1, fontsize = 7, fontcolor = \"#b2182b\"\n",
    ")\n",
    "#############################\n",
    "## plot loops\n",
    "plotPairsArches(\n",
    "    data = loops_eqtl,\n",
    "    params = params_a,\n",
    "    y = 6, height = 1.5,\n",
    "    fill = \"#37a7db\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "plotText(\n",
    "    label = \"Consensus \\ninteractions\", fontsize = 7, fontcolor = \"#37a7db\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 6.5, x = 0.6, height = 0.5,\n",
    ")\n",
    "#############################\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = proxy_sets_thisone,\n",
    "    params = params_a, linecolor=\"tomato3\", fill = \"tomato3\", \n",
    "    collapse = T,\n",
    "    y = 2.1, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"eQTLs\", fontsize = 7, fontcolor = \"tomato3\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.2, x = 0.4, height = 0.5,\n",
    ")\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_a, height = 1, y = 1,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 1.3, x = 0.4, height = 0.5,\n",
    ") \n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0, 12),\n",
    "    label = c(0, 12), # top -log10P val\n",
    "    main = TRUE, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "dev.off()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "### C7orf50 locus\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/C7orf50_plotGardener_genoSpecific.pdf\")\n",
    "\n",
    "### set parameters\n",
    "myeQTL <- \"7:1172575:C:A\"\n",
    "myGene = \"C7orf50\"\n",
    "mychrom = \"chr7\"\n",
    "mystart = 960500\n",
    "myend = 1397102\n",
    "###\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "### restrcit eQTLs to the one we are testing\n",
    "proxy_sets_thisone <- proxy_sets[V4 == myeQTL]\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 7, default.units = \"cm\", showGuides = F)\n",
    "params_a <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 2, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_a, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Genes\", fontsize = 7, fontcolor = \"grey48\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.8, x = 0.4, height = 0.5,\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_a, \n",
    "                   scale = \"Mb\", fontsize = 7,\n",
    "                   y = 6)\n",
    "## plot loops\n",
    "#plotPairsArches(\n",
    "#    data = loops_eqtl,\n",
    "#    params = params_a,\n",
    "#    y = 3.5, height = 1.5,\n",
    "#    fill = \"#37a7db\", linecolor = \"fill\", flip = TRUE\n",
    "#)\n",
    "\n",
    "\n",
    "\n",
    "#################### ADD GENO SPECIFIC CHICAGO PLOTBAITS\n",
    "# process these!!!\n",
    "REF <- fread(file = \"~/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/7:1210755:C:T_homozygous_REF/7:1210755:C:T_homozygous_REFallReplicates_Chicago_426778.bg\")\n",
    "setnames(REF, \"V4\", \"score\")\n",
    "CHiC_signal_REF <- plotSignal(\n",
    "    data = REF, binSize = 50, binCap = F, range = c(0, 84),\n",
    "    params = params_a, height = 1, y = 3.5,\n",
    "    linecolor = \"deepskyblue2\", fill = \"deepskyblue2\"\n",
    ")\n",
    "## Bring in chic bw data from a bait for the SNP\n",
    "ALT <- fread(file = \"~/HRJ_monocytes/CHiC/chicago/combinations_output/5K-bins/genotype_specific_runs/7:1210755:C:T_homozygous_ALT/7:1210755:C:T_homozygous_ALTallReplicates_Chicago_426778.bg\")\n",
    "setnames(ALT, \"V4\", \"score\")\n",
    "CHiC_signal_ALT <- plotSignal(\n",
    "    data = ALT,binSize = 50, binCap = F, range = c(0, 84),\n",
    "    params = params_a, height = 1, y = 4.7,\n",
    "    linecolor = \"firebrick2\", fill = \"firebrick2\"\n",
    ")\n",
    "####################\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"rs2960848 CHi-C\", fontsize = 7, fontcolor = \"black\",\n",
    "    rot = 90,\n",
    "    params = params_a,\n",
    "    y = 5.5, x = 0.3, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom REF \\nN reads\", fontsize = 7, fontcolor = \"deepskyblue2\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 4, x = 0.6, height = 0.5,\n",
    ")\n",
    "plotText(\n",
    "    label = \"Hom ALT \\nN reads\", fontsize = 7, fontcolor = \"firebrick2\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 5, x = 0.6, height = 0.5,\n",
    ")\n",
    "\n",
    "# range for loops\n",
    "annoYaxis(\n",
    "    CHiC_signal_REF,\n",
    "    at = NULL,\n",
    "    label = T, # limit we have set - not going to work, though!\n",
    "    main = T, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 3.5, height = 1, fontsize = 7, fontcolor = \"deepskyblue2\"\n",
    ")\n",
    "annoYaxis(\n",
    "    CHiC_signal_ALT,\n",
    "    at = NULL,\n",
    "    label = T, # limit we have set - not going to work, though!\n",
    "    main = T, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 4.7, height = 1, fontsize = 7, fontcolor = \"firebrick2\"\n",
    ")\n",
    "#############################\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = proxy_sets_thisone,\n",
    "    params = params_a, linecolor=\"tomato3\", fill = \"tomato3\", \n",
    "    collapse = T,\n",
    "    y = 2.1, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"eQTLs\", fontsize = 7, fontcolor = \"tomato3\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.2, x = 0.4, height = 0.5,\n",
    ")\n",
    "\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal <- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_a, height = 1, y = 1,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 1.3, x = 0.4, height = 0.5,\n",
    ") \n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0, 12),\n",
    "    label = c(0, 12), # top -log10P val\n",
    "    main = TRUE, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "### CEBPB locus\n",
    "\n",
    "pdf(file = \"~/HRJ_monocytes/paper/CEBPB_plotGardener.pdf\")\n",
    "\n",
    "\n",
    "### restrict loops to my eQTL\n",
    "loops_eqtl <- loops_proxies[hg38SNP_ID == myeQTL]\n",
    "###\n",
    "\n",
    "pageCreate(width = 12, height = 6, default.units = \"cm\", showGuides = F)\n",
    "params_a <- pgParams(chrom = mychrom, chromstart = mystart, chromend = myend, \n",
    "                      assembly = \"hg38\", , default.units = \"cm\",\n",
    "                      x = 1.7, width = 10, just = c(\"left\", \"top\"), default.units = \"cm\")\n",
    "## specify the gene target\n",
    "geneHighlights <- data.frame(\"geneName\" = myGene, \"color\" = \"chartreuse4\")\n",
    "\n",
    "## Plot gene track\n",
    "genes_a <- plotGenes(params = params_a, stroke = 1, fontsize = 8,\n",
    "                     geneHighlights = geneHighlights, geneBackground = \"grey48\",\n",
    "                        y = 2.5, height = 1)\n",
    "\n",
    "# text for genes\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"Genes\", fontsize = 7, fontcolor = \"grey48\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.8, x = 0.4, height = 0.5,\n",
    ") \n",
    "\n",
    "## Annotate genome label\n",
    "annoGenomeLabel(plot = genes_a, params = params_a, \n",
    "                   scale = \"Mb\", fontsize = 7,\n",
    "                   y = 5.2)\n",
    "## plot loops\n",
    "plotPairsArches(\n",
    "    data = loops_eqtl,\n",
    "    params = params_a,\n",
    "    y = 3.5, height = 1.5,\n",
    "    fill = \"#37a7db\", linecolor = \"fill\", flip = TRUE\n",
    ")\n",
    "# text for loops\n",
    "plotText(\n",
    "    label = \"CHi-C\", fontsize = 7, fontcolor = \"#37a7db\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 4, x = 0.4, height = 0.5,\n",
    ")\n",
    "\n",
    "# Plot eQTLs\n",
    "plotRanges( \n",
    "    data = proxy_sets,\n",
    "    params = params_a, linecolor=\"tomato3\", fill = \"tomato3\", \n",
    "    collapse = T,\n",
    "    y = 2.1, height = 0.3\n",
    ")\n",
    "# text for eQTLs\n",
    "plotText(\n",
    "    label = \"eQTLs\", fontsize = 7, fontcolor = \"tomato3\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 2.2, x = 0.4, height = 0.5,\n",
    ")\n",
    "\n",
    "## plot ATAC\n",
    "atac_myregion <- readBigwig(file = \"~/HRJ_monocytes/ATAC-seq/analyses/Genrich/consensus/consensus_Genrich.bw\",\n",
    "                        chrom = mychrom,\n",
    "                        chromstart = mystart,\n",
    "                        chromend = myend)\n",
    "ATAC_signal<- plotSignal(\n",
    "    data = atac_myregion,\n",
    "    params = params_a, height = 1, y = 1,\n",
    "    linecolor = \"seagreen\"\n",
    ")\n",
    "# text for ATAC\n",
    "plotText(\n",
    "    label = \"ATAC \\n(-log10P)\", fontsize = 7, fontcolor = \"seagreen\",\n",
    "    rot = 0,\n",
    "    params = params_a,\n",
    "    y = 1.3, x = 0.4, height = 0.5,\n",
    ") \n",
    "# range for ATAC\n",
    "annoYaxis(\n",
    "    ATAC_signal,\n",
    "    at = c(0,12),\n",
    "    label = c(0, 12), # top -log10P val\n",
    "    main = TRUE, \n",
    "    axisLine = T,\n",
    "    params = params_a, \n",
    "    y = 1, height = 1, fontsize = 7, fontcolor = \"maroon3\"\n",
    ")\n",
    "\n",
    "dev.off()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:plotgardener]",
   "language": "R",
   "name": "conda-env-plotgardener-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
